---
title: "Pig_Mouse_Comparison_Study"
author: "Nirosh D. Aluthge"
date: "3/27/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = '/Users/niroshaluthge/Desktop/NDA/manuscripts/GitHub_repositories/men_mice_and_pigs/')
knitr::opts_chunk$set(echo = TRUE)
```

<div style="margin-bottom:50px;">
</div>
#### **INTRODUCTION** 

This R Markdown file details all the data analysis steps used in the manuscript **"Differential longitudinal establishment of human fecal bacterial communities in germ-free porcine and murine models"** authored by Aluthge et al. and published in the journal 'Communications Biology' (Commun Biol 3, 760 (2020)).


#### **DATA ANALYSIS**

All the steps of this analysis (unless mentioned otherwise) was performed on an Apple MacBook pro with a 2.3 GHz Intel core i5 processor and an 8 GB memory 

**Note on sequencing: ** The data used in this analysis was generated through barcoded multiplexed amplicon sequencing of the V4 hypervariable region of the bacterial 16S rRNA gene using an Illumina MiSeq sequencing platform. Since there were > 490 samples, the initial sequencing was carried out as two sequencing runs (pig_mouse_comp_sequencing_run1 and pig_mouse_comp_sequencing_run2). Subsequently, based on sequencing depths, some samples which had yielded low numbers of sequence reads were resequenced in two separate 'resequencing' runs (pig_mouse_comp_resequencing_run1 and pig_mouse_comp_resequencing_run2) with the aim of increasing the sequencing depth of these samples. Since the error profiles determined in the DADA2 data analysis pipeline is run-dependent, the initial denoising steps were conducted separately for each sequencing run and at the end all the individual 'feature tables' were merged to make a final 'master' feature table containing all the sample x feature count data.

**Notes on R coding:** I learned a lot of R coding during the data analysis described in this RMarkdown file. Not being an experienced user of the R programming language, I have little doubt that some of the manipulations I have performed using several lines of code could be done more simply with fewer lines of code in the hands of an experienced R. It is also possible that such users will be able to identify R packages which would have been more suitable for a particular type of analysis. Notwithstanding such possible limitations, a lot of effort has been expended to ensure that the code described in this document is accurate. Many input files as well as important phyloseq objects have been uploaded to the 'men_mice_and_pigs' GitHub page for the convenience of those who don't wish to go through the entire analysis but are more interested in specific types of analyses.


```{r, echo=TRUE}
#This data analysis pipeline follows the analysis of microbiome data using ASVs as described by Callahan et al. in  https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html

#Loading or installing required packages
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if (any(!.inst)) {
  install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
if (any(!.inst)) {
  source("http://bioconductor.org/biocLite.R")
  biocLite(.bioc_packages[!.inst], ask = F)
}

sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
library("BiocStyle"); packageVersion("BiocStyle")
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("DECIPHER")
library("DECIPHER");packageVersion("DECIPHER")
```
Now that the required main packages have been loaded, we can proceed with the data analysis steps.

We will first pick ASVs using the DADA2 pipeline for the '.fastq' files from our very first sequencing run. These '.fastq' files are in the folder 'pig_mouse_comparisons_r2_first_run_fastq_files_no_pre_inoc'. There are **274** files corresponding to **137** samples.

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

#change to required work directory
setwd("/Users/niroshaluthge/Desktop/NDA/manuscripts/GitHub_repositories/men_mice_and_pigs")
getwd()

MiSeq_path <- "./pig_mouse_comparisons_r2_first_run_fastq_files_no_pre_inoc" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

#Separate the 'forward' and 'reverse' reads
fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

#Look at base quality plots
plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

#filter the forward and reverse reads
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(250,140), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) #the trunLen values were determined based off of the quality profile plots from above
write.table(out, "out_file_run_1.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 3061044 input reads and 2821030 output reads, so we have retained ~ 92.1% of the reads after this filtering and trim step

#dereplicating to eliminate redundancy (multiple occurances of the same identical sequence)
derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

#Learning error rates
errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

#inferring sequences through DADA2 method
dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] #checking how many sequence variants were inferred from first sample

#Construct sample by sequence feature table
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_run1 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_run1)))
sum(seqtabAll_run1) # 2673152 reads had remained after the mergers. Since we had 3061044 total input reads, this means that ~ 87.3% of reads were retained

#remove non-target-length sequences
seqtabAll2_run1 <- seqtabAll_run1[,nchar(colnames(seqtabAll_run1)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_run1)))
dim(seqtabAll2_run1)
sum(seqtabAll2_run1) # 2673118 sequences retained

#Store seqtabAll2 from run 1 as a serialized R object
saveRDS(seqtabAll2_run1, "seqtabAll2_run1.rds")
```

Now let's run the same pipeline for the '.fastq' files from our second sequencing run. These fastq files are in the folder "pig_mouse_comparisons_r2_second_run_fastq_files" and consist of **492** .fastq files from **246** samples

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

#change to required work directory
setwd("/Users/niroshaluthge/Desktop/NDA/manuscripts/pig_mouse_comparison_study")
getwd()

MiSeq_path <- "./pig_mouse_comparisons_r2_second_run_fastq_files" 
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") 
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(240,100), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_run_2.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 12907795 input reads and 10492999 output reads, so we have retained ~ 81.2 % of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_run2 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_run2)))
sum(seqtabAll_run2) # 9793401 reads had remained after the mergers. Since we had 12907795 total input reads, this means that ~ 76 % of reads were retained

seqtabAll2_run2 <- seqtabAll_run2[,nchar(colnames(seqtabAll_run2)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_run2)))
dim(seqtabAll2_run2)
sum(seqtabAll2_run2) # 9790523 sequences retained

#Store seqtabAll2 from run 2 as a serialized R object
saveRDS(seqtabAll2_run2, "seqtabAll2_run2.rds")
```

Now we can merge the two sequence tables together:

```{r, echo=TRUE}
seqtab_run1 <- readRDS("seqtabAll2_run1.rds")
seqtab_run2 <- readRDS("seqtabAll2_run2.rds")
seqtab_runs_1_and_2 <- mergeSequenceTables(seqtab_run1, seqtab_run2)
```

Let's make sure that the merging has been done properly :

```{r, echo=TRUE}
dim(seqtab_run1)
sum(seqtab_run1)
dim(seqtab_run2)
sum(seqtab_run2)
dim(seqtab_runs_1_and_2)
sum(seqtab_runs_1_and_2)
saveRDS(seqtab_runs_1_and_2, "seqtab_runs_1_and_2.rds")
```

We can see that the number of samples and the total number of sequences matchup. The number of ASVs don't add-up because there appear to be shared ASVs between the two tables.

After looking at the read depth for each sample, we realized that there were quite a few samples with low sequencing depths (including some of the donor samples and inocula). In addition, we didn't have samples 97-192 in our first two sequencing runs either. Therefore, all these samples were 'resequenced' using the same Illumina MiSeq instrument and targeting the same V4 hypervariable region in order to increase read depths of these samples. The '.fastq' files from these 'resequencing' runs are in folders 'pig_mouse_comparison_resequencing_run_1_fastq_files' and 'pig_mouse_comparison_resequencing_run_2_fastq_files', respectively.

Running the fastq files (**614** fastq files from **307** samples) from the first resequencing run using the dada2 pipeline:

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

MiSeq_path <- "./pig_mouse_comparison_resequencing_run_1_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(245,145), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_reseq_run_1.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 4344562 input reads and 3991675 output reads, so we have retained ~92% of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_reseq_run1 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_reseq_run1)))
sum(seqtabAll_reseq_run1) # 3680068 reads had remained after the mergers. Since we had 4309906 total input reads, this means that 84.7% of reads were retained

seqtabAll2_reseq_run1 <- seqtabAll_reseq_run1[,nchar(colnames(seqtabAll_reseq_run1)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_reseq_run1)))
dim(seqtabAll2_reseq_run1)
sum(seqtabAll2_reseq_run1) # 3679625 sequences retained

#Store seqtabAll2 from resequencing run 1 as a serialized R object
saveRDS(seqtabAll2_reseq_run1, "seqtabAll2_reseq_run1.rds")
seqtabAll2_reseq_run1 <- readRDS("seqtabAll2_reseq_run1.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_reseq_1 <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs,getN), sapply (mergers, getN), rowSums(seqtabAll2_reseq_run1))
colnames(track_reseq_1) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "seqtable")
rownames(track_reseq_1) <- sampleNames
head(track_reseq_1)
write.table(track_reseq_1, 'tracking_reads_through_pipeline_reseq_plate_1.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```


The developers of phyloseq recently developed the 'decontam R package' for detecting reads coming from external contaminants (reagent contaminants, lab-workers, etc.). We'll implement the 'prevalence' function of this decontam package to identify possible contaminant ASVs based on their prevalence in negative controls:

```{r, echo=TRUE}
# Detection of contaminants using 'decontam'
library("decontam") ; packageVersion("decontam") 
decontam_sample_data_reseq_1 <- read.table("pig_mouse_comp_resequencing_run_1_decont_mapp_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(decontam_sample_data_reseq_1) <- as.character(decontam_sample_data_reseq_1[,1])
decontam_mp_reseq_1 <- phyloseq(otu_table(seqtabAll2_reseq_run1, taxa_are_rows = FALSE), sample_data(decontam_sample_data_reseq_1))

## ANOTHER way to get mapping file read into R
# decontam_sample_data_reseq_1 <- read.table("pig_mouse_comp_resequencing_run_1_decont_mapp_file.txt", sep = "\t", header = TRUE)
# rownames(decontam_sample_data_reseq_1) = decontam_sample_data_reseq_1$SampleID

# identifying contaminants using the 'prevalence' method
sample_data(decontam_mp_reseq_1)$is.neg <- sample_data(decontam_mp_reseq_1)$Sample_or_Control == "negative_control"
contamdf.prev <- isContaminant(decontam_mp_reseq_1, method = "prevalence", neg = "is.neg", threshold = 0.1) # default threshold of 0.1 used
hist(contamdf.prev$p, 100, ylim = c(0,400), xlim = c(0,1))
table(contamdf.prev$contaminant) # TRUE ones are the contaminants. 14 such potential contaminants have been identified.

# let's filter out these potential contaminants from the sequence table
seqtab_reseq_run_1_no_contam <- prune_taxa(!contamdf.prev$contaminant, otu_table(decontam_mp_reseq_1))
str(seqtab_reseq_run_1_no_contam) # 3306 ASVs have been retained

# Let's check how many reads we lost as a result of removing the contaminants
sum(seqtab_reseq_run_1_no_contam)/sum(seqtabAll2_reseq_run1) # about ~0.3% of reads were removed as contaminants
dim(seqtab_reseq_run_1_no_contam)

# Let's save the 'decontaminated' sequence table as an R serial object
saveRDS(seqtab_reseq_run_1_no_contam, "seqtab_reseq_run_1_no_contam.rds")
```


Performing dada2 pipeline to pick sequence variants from resequencing run 2 which included **306** fastq files from **153** samples :

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

MiSeq_path <- "./pig_mouse_comparison_resequencing_run_2_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") 
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(240,120), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_reseq_run_2.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 6539111 input reads and 5811681 output reads, so we have retained ~ 88.9 % of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_reseq_run2 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_reseq_run2)))
sum(seqtabAll_reseq_run2) # 5344804 reads had remained after the mergers. Since we had 6539111 total input reads, this means that ~ 81.7% of reads were retained

seqtabAll2_reseq_run2 <- seqtabAll_reseq_run2[,nchar(colnames(seqtabAll_reseq_run2)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_reseq_run2)))
dim(seqtabAll2_reseq_run2)
sum(seqtabAll2_reseq_run2) # 5338176 sequences retained

#Store seqtabAll2 from resequencing run 2 as a serialized R object
saveRDS(seqtabAll2_reseq_run2, "seqtabAll2_reseq_run2.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_reseq_2 <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs,getN), sapply (mergers, getN), rowSums(seqtabAll2_reseq_run2))
colnames(track_reseq_2) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "seqtable")
rownames(track_reseq_2) <- sampleNames
head(track_reseq_2)
write.table(track_reseq_2, 'tracking_reads_through_pipeline_reseq_plate_2.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```

Removal of potential contaminants using 'decontam' R package for resequencing run 2 :

```{r, echo=TRUE}
seqtabAll2_reseq_run2 <- readRDS ("seqtabAll2_reseq_run2.rds") 
dim(seqtabAll2_reseq_run2)
decontam_sample_data_reseq_2 <- read.table("pig_mouse_comp_resequencing_run_2_decont_mapp_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(decontam_sample_data_reseq_2) <- as.character(decontam_sample_data_reseq_2[,1])
decontam_mp_reseq_2 <- phyloseq(otu_table(seqtabAll2_reseq_run2, taxa_are_rows = FALSE), sample_data(decontam_sample_data_reseq_2))
decontam_mp_reseq_2

# identifying contaminants using the 'prevalence' method
sample_data(decontam_mp_reseq_2)$is.neg <- sample_data(decontam_mp_reseq_2)$Sample_or_Control == "negative_control"
contamdf.prev <- isContaminant(decontam_mp_reseq_2, method = "prevalence", neg = "is.neg", threshold = 0.1)
hist(contamdf.prev$p, 100, ylim = c(0,400), xlim = c(0,1))
table(contamdf.prev$contaminant) # TRUE ones are the contaminants. 38 such potential contaminants have been identified.

# now let's filter out these potential contam ASVs from the sequence table
seqtab_reseq_run_2_no_contam <- prune_taxa(!contamdf.prev$contaminant, otu_table(decontam_mp_reseq_2))
dim(seqtab_reseq_run_2_no_contam)

# Let's check how many reads we lost as a result of removing the contaminants
sum(seqtab_reseq_run_2_no_contam)/sum(seqtabAll2_reseq_run2) # about ~0.6% of reads were removed as contaminants

# Let's save the 'decontaminated' sequence table as an R serial object
saveRDS(seqtab_reseq_run_2_no_contam, "seqtab_reseq_run_2_no_contam.rds")
```


Now that we have determined the ASVs through the DADA2 pipeline for all the samples from all of our sequencing runs, we can merge the individual sequence tables to make a single 'main' sequence table. 

```{r, echo=TRUE}
# reading in the .rds files of the sequence tables
seqtab_runs_1_and_2 <- readRDS("seqtab_runs_1_and_2.rds")
dim(seqtab_runs_1_and_2)
seqtab_reseq_run_1_no_contams <- readRDS("seqtab_reseq_run_1_no_contam.rds")
dim(seqtab_reseq_run_1_no_contams)
seqtab_reseq_run_2_no_contams <- readRDS("seqtab_reseq_run_2_no_contam.rds")
dim(seqtab_reseq_run_2_no_contams)

# Let's merge these sequence tables :
merged_seqtab_all_runs <- mergeSequenceTables(seqtab_runs_1_and_2,seqtab_reseq_run_1_no_contams,seqtab_reseq_run_2_no_contams, repeats = "sum")
dim(merged_seqtab_all_runs) # There are 491 samples with 15055 ASVs
write.table(merged_seqtab_all_runs, "merged_seqtab_all_runs.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
```

Let's double check to see if the numbers add-up :

```{r, echo=TRUE}
sum(seqtab_runs_1_and_2)
sum(seqtab_reseq_run_1_no_contams)
sum(seqtab_reseq_run_2_no_contams)
```

The merged table should have a total of **21,440,384** reads. Let's see if this is the case :

```{r,echo=TRUE}
sum(merged_seqtab_all_runs)
```

The numbers add-up! Can continue onto chimera identification and filtering using this merged sequence table :

```{r, echo=TRUE}
# chimera filtering
merged_seqtab_all_runs_nochim <- removeBimeraDenovo(merged_seqtab_all_runs, method = "consensus", multithread = TRUE, verbose = TRUE)
dim(merged_seqtab_all_runs_nochim) # 8773 chimeras were identified (which is a majority of ASVs but not abnormal apparently)
sum(merged_seqtab_all_runs_nochim)/sum(merged_seqtab_all_runs) # only about 5.5% of the total READS were removed as chimeras
```

Assigning taxonomy to the remaining ASVs using the SILVA database files :

```{r, echo=TRUE}
fastaRef <- "./silva_nr_v138_train_set.fa" 
taxTab <- assignTaxonomy(merged_seqtab_all_runs_nochim, refFasta = fastaRef, multithread = TRUE)
taxTab <- addSpecies(taxTab, "./silva_species_assignment_v138.fa")
unname(head(taxTab))
head(taxTab)
write.table(taxTab, "full_taxa_table.txt", sep = "\t", row.names = T, col.names = NA, quote = F)

# Let's output the taxonomy as a .txt file
taxa_cols <- colnames(taxTab)
taxa_table <- as.data.frame(taxTab)
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";")) # code obtained from https://forum.qiime2.org/t/importing-dada2-and-phyloseq-objects-to-qiime-2/4683
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
dim(taxa_table)

taxa_table$ASV_seq <- rownames(taxa_table)
ASV_id <- paste0('ASV_', seq(rownames(taxa_table))) # replacing the long sequence names with "ASV_#" id
ASV_id
taxa_table$ASV_id <- ASV_id
dim(taxa_table)
View(taxa_table)
taxa_table <- taxa_table[,c(3,2,1)] # arrange table columns in order
View(taxa_table) # check to see that everything looks good
saveRDS(taxTab, "taxTab.rds")
taxTab <- readRDS("taxTab.rds")

write.table(taxa_table, "ASV_full_taxonomy_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

Next a phylogenetic tree needs to be built using the ASV sequences. Need the ASVs to be in FASTA format to do the alignments. This can be done using the 'uniquesToFasta' function in phyloseq :

```{r, echo=TRUE}
uniquesToFasta(merged_seqtab_all_runs_nochim, fout = "main_rep_set.fna", ids = colnames(merged_seqtab_all_runs_nochim))
```

This FASTA file is used for alignment to the 'Silva_nr' database and subsequent construction of a phylogenetic tree using MOTHUR as follows :

```
# Open a terminal and type the following commands to download and unzip the Silva_nr database :
wget https://mothur.s3.us-east-2.amazonaws.com/wiki/silva.nr_v138.tgz
tar -zxvf silva.nr_v138.tgz 

# move the unzipped 'silva.nr_v132.align' database and the FASTA formatted file from running the 'make_repset_fasta_dada2.py' script to the MOTHUR folder and run the following MOTHUR commands :

pcr.seqs(fasta=silva.nr_v138.align, start=11894, end=25319, keepdots=F, processors=8)
system(mv silva.nr_v138.pcr.align silva.v4_138.fasta)
align.seqs(fasta=main_rep_set.fna, reference=silva.v4_138.fasta)

# The 'main_rep_set.align' file needs to be modified to make it compatible with MOTHUR for running the commands for distance calculations and making the phylogenetic tree using 'clearcut'. The '.' in the alignment file need to be replaced with '-' . Need to open a terminal and type these commands (outside of MOTHUR).

sed -i -e 's/\./-/g' main_rep_set.align # replaces '.' with '-'

# making phylogenetic tree using 'clearcut' in MOTHUR (takes a while so I ran these commands on a server. The script is in 'phylo_tree_MOTHUR.pbs'):
dist.seqs(fasta=main_rep_set.align, processors=2, cutoff=.10, output=phylip)
clearcut(phylip=main_rep_set.phylip.dist)
```

Let's read in the sample data file (mapping file) which contains all the metadata and the newly constructed phylogenetic tree:

```{r, echo=TRUE}
mapping_file <- read.table("pig_mouse_comp_r2_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "") # has been uploaded to Github 
row.names(mapping_file) <- as.character(mapping_file[,1])
View(mapping_file)
tree_file <- 'main_rep_set.phylip.tre' # has been uploaded to GitHub for convenience
phylo_tree <- read_tree(tree_file)
phylo_tree
```

Now we can combine the merged sequence table, mapping file, taxa table, and phylogenetic tree into a single pyloseq object :

```{r, echo=TRUE}
merged_pig_mouse_comp_ASV_as_seqs_ps <- phyloseq(otu_table(merged_seqtab_all_runs_nochim, taxa_are_rows = FALSE), sample_data(mapping_file), tax_table(taxTab), phy_tree(phylo_tree))
head(taxa_names(merged_pig_mouse_comp_ASV_as_seqs_ps)) # ASVs are in sequence format

# replacing the long sequence name with short ID 
taxa_names(merged_pig_mouse_comp_ASV_as_seqs_ps) <- paste0("ASV_", seq(ntaxa(merged_pig_mouse_comp_ASV_as_seqs_ps))) # replacing the long sequence name with short ID 

merged_pig_mouse_comp_ps <- merged_pig_mouse_comp_ASV_as_seqs_ps 
head(taxa_names(merged_pig_mouse_comp_ps)) # now ASV_IDs are in 'ASV_##' format 

merged_pig_mouse_comp_ps <- readRDS("merged_pig_mouse_comp_ps.rds") # .rds file has been provided for convenience for anyone who wants to start the analysis from this point onwards
```

Let's look at the prevalence of taxa in the samples 
 
```{r, echo=TRUE}
prevdf = apply(X = otu_table(merged_pig_mouse_comp_ps),
               MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf = data.frame(Prevalence=prevdf, TotalAbundance=taxa_sums(merged_pig_mouse_comp_ps))
View(prevdf) # minimum prevalence is '1'
```

Removing 'negative control' samples :

```{r, echo=TRUE}
merged_pig_mouse_comp_ps_no_neg <- subset_samples(merged_pig_mouse_comp_ps, Description != 'negative_control')
merged_pig_mouse_comp_ps_no_neg

prevdf_no_neg = apply(X = otu_table(merged_pig_mouse_comp_ps_no_neg),
               MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps_no_neg), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_no_neg = data.frame(Prevalence=prevdf_no_neg, TotalAbundance=taxa_sums(merged_pig_mouse_comp_ps_no_neg))
View(prevdf_no_neg)

# There appear to be quite a few ASVs (around 350) which show a prevalence of '0' after filtering out the negative controls
# this seems to suggest that these ASVs would have been present ONLY in negative controls, and are no longer present after they are filtered out

negative_control_taxa <- rownames(prevdf_no_neg)[(prevdf_no_neg$Prevalence < 1)]
str(negative_control_taxa) # 348 ASVs
negative_control_taxa_table <- prune_taxa(negative_control_taxa, tax_table(merged_pig_mouse_comp_ps_no_neg))
```

Filtering out ASVs that are classified in the kingdoms 'Archaea' or 'Eukaryota' as well as those ASVs that have a Mitochondrial classification :

```{r, echo=TRUE}
remov_kingdoms <- c('Archaea', 'Eukaryota')
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_pig_mouse_comp_ps_no_neg, !Kingdom %in% remov_kingdoms & (Family != 'Mitochondria' | is.na(Family))) 
merged_pig_mouse_comp_ps_master
head(taxa_names(merged_pig_mouse_comp_ps_master))
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_pig_mouse_comp_ps_master, Kingdom == 'Bacteria') # removed 4 ASVs classified as 'NA' at the kingdom level
merged_pig_mouse_comp_ps_master
```

In the past, we've filtered out any OTU which had a Cyanobacterial classification. However, recent research (Di Rienzi et al., 2013) has suggested that a non-photosynthetic group of 'Cyanobacteria' called 'Melainabacteria' might be human gut-associated. Therefore, we're going to filter out only those Cyanobacterial ASVs that are not classified as belonging to the class 'Melainabacteria':

```{r, echo=TRUE}
cyano_ps <- subset_taxa(merged_pig_mouse_comp_ps_master, Phylum == "Cyanobacteria")
cyano_ps 

remov_cyano_ps <- subset_taxa(cyano_ps, Class != "Melainabacteria" | is.na(Class)) # Keep only Melainabacteria (also remove any cyano ASVs with 'NA' as class)
cyano_to_remove <- taxa_names(remov_cyano_ps) # apparently, no melainabacteria are present in these samples

# filtering out the Cyanobacteria 
all_taxa <- taxa_names(merged_pig_mouse_comp_ps_master)
good_taxa <- all_taxa[!(all_taxa %in% cyano_to_remove)]
good_taxa

merged_pig_mouse_comp_ps_master_no_cyano <- prune_taxa(good_taxa,merged_pig_mouse_comp_ps_master)
merged_pig_mouse_comp_ps_master_no_cyano # 6178 ASVs remain
```

Computing prevalence of each ASV to filter out those that are present in only a single sample :

```{r, echo=TRUE}
# Calculating prevalence for each ASV
prevdf <- apply(X = otu_table(merged_pig_mouse_comp_ps_master_no_cyano), MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps_master_no_cyano), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
View(prevdf)
prevdf_with_taxa <- data.frame(Prevalence = prevdf, TotalAbundance = taxa_sums(merged_pig_mouse_comp_ps_master_no_cyano), tax_table(merged_pig_mouse_comp_ps_master_no_cyano))
View(prevdf_with_taxa)

# Retaining only those ASVs found in more than one sample
keepTaxa <- rownames(prevdf_with_taxa)[(prevdf_with_taxa$Prevalence > 1)]
merged_pig_mouse_comp_ps_master_all_filtered <- prune_taxa(keepTaxa, merged_pig_mouse_comp_ps_master_no_cyano)
merged_pig_mouse_comp_ps_master_all_filtered # We've filtered out a great majority of ASVs (5043)

sum(otu_table(merged_pig_mouse_comp_ps_master_no_cyano))
sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))
sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))/sum(otu_table(merged_pig_mouse_comp_ps_master_no_cyano)) # However, those 5043 ASVs only accounted for ~0.21% of reads (41,111). This means we have retained 99.79% of our reads even after filtering 5043 out of 6178 ASVs.
```

Looking at the distribution of reads per sample :

```{r, echo=TRUE}
reads_per_sample_df <- data.frame(Reads = sample_sums(merged_pig_mouse_comp_ps_master_all_filtered)) 
reads_per_sample_df$sampleID <- rownames(reads_per_sample_df)
View(reads_per_sample_df)
reads_per_sample_df <- reads_per_sample_df[,c(2,1)] # reorder columns
reads_per_sample_df_sorted <- reads_per_sample_df[order(reads_per_sample_df$Reads),] # order samples by read count (ascending)
View(reads_per_sample_df_sorted) # check that it looks right

# Making a histogram plot with sequence depth information
library('ggplot2'); packageVersion('ggplot2')
read_depth_histo <- ggplot(reads_per_sample_df_sorted, aes(x = Reads)) + geom_histogram(color = "black", fill = "indianred", binwidth = 2500) + ggtitle("Sequence depth distribution across samples") + xlab("Number of reads") + theme(axis.title.y = element_blank())
read_depth_histo

# Depth information
min_read_depth <- min(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered))
min_read_depth # 5,415 reads
max_read_depth <- max(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered))
max_read_depth # 142,321 reads
avg_read_depth <- sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))/nsamples(merged_pig_mouse_comp_ps_master_all_filtered)
avg_read_depth # ~43,510 reads/sample
```

Filtering out samples which have < 10,500 total reads :

```{r, echo=TRUE}
merged_pig_mouse_comp_complete_ps <- prune_samples(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered) > 10500, merged_pig_mouse_comp_ps_master_all_filtered)
merged_pig_mouse_comp_complete_ps # 459 samples remain

# Save as an R serial object for use in future analyses
saveRDS(merged_pig_mouse_comp_complete_ps, "merged_pig_mouse_comp_complete_ps.rds")

merged_pig_mouse_comp_complete_ps <- readRDS("merged_pig_mouse_comp_complete_ps.rds") # .rds file provided for convenience
```

We will also remove fecal samples collected at euthanization since these samples were collected differently from the other samples

```{r, echo = TRUE}
merged_pig_mouse_comp_complete_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Time_point != "at_euthanization" )
merged_pig_mouse_comp_complete_ps # 392 samples remain
sum(otu_table(merged_pig_mouse_comp_complete_ps)) # 17016489 total reads; mean=43,409 (SD=21,178)

write.table(t(otu_table(merged_pig_mouse_comp_complete_ps)), "complete_otu_table.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE) # having this main OTU table as a .txt file might be useful to verify downstream results
```

Looking at sample diversity in the human donor samples :

```{r, echo=TRUE} 
# Since alpha diversity measures are sensitive to the rare species, we're going to use the ASV table which still has ASVs which are found in only a single sample 
merged_pig_mouse_comp_human_donors_ps <- subset_samples(merged_pig_mouse_comp_ps_master_no_cyano, Species == 'human')
merged_pig_mouse_comp_human_donors_ps

# alpha diversity comparison of donor samples (Shannon index)

# Let's make these plots without rarefying since phyloseq authors recommend not rarefying as it 'throws out' data (https://github.com/joey711/phyloseq/issues/287 and https://github.com/joey711/phyloseq/issues/447)

# Estimating alpha diversity using Shannon index only for inoculae actually used
human_donor_inoculae_ps <- subset_samples(merged_pig_mouse_comp_human_donors_ps, Actual_inoculum == "yes")
human_donor_inoculae_ps <- subset_samples(human_donor_inoculae_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
human_donor_inoculae_ps

# just to check that we have the correct donor inoculae
write.table(sample_data(human_donor_inoculae_ps), "donor sample data.txt", sep = "\t", col.names = T, row.names = F, quote = F)

results_donors = estimate_richness(human_donor_inoculae_ps, measures = "Shannon")
results_donors_df <- as.data.frame(results_donors)
results_donors_df$SampleID <- rownames(results_donors_df)
results_donors_df <- results_donors_df[c(2,1)]
View(results_donors_df)
write.table(results_donors_df, 'donor_alpha_div_source_data.txt', sep = "\t", row.names = F, col.names = T, quote = F) # source data required for publication

# import mapping file which would be used for alpha diversity analyses
alpha_mapping_file <- read.table("alpha_div_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "") # has been uploaded to GitHub
row.names(alpha_mapping_file) <- as.character(alpha_mapping_file[,1])

donors_alpha_div_df <- merge(results_donors_df, alpha_mapping_file, by='SampleID')
View(donors_alpha_div_df)

#pdf("donor_inocula_alpha_diversity.pdf", title = "Figure1")
theme_set(theme_gray())
p1 <- ggplot(donors_alpha_div_df, aes(x=Donor, y=Shannon, fill=Donor)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits = c("INF2", "CHLD2", "ADLT2", "ELD2"), labels = c("Donor_1", "Donor_2", "Donor_3", "Donor_4")) + theme(legend.position = "none", axis.text = element_text(size = 10), axis.title = element_text(size = 12)) + labs(x = "Donor", y = "Shannon index")
p2
ggsave('donor_inocula_alpha_diversity.tiff', p2, device = "tiff", dpi = 500)
#dev.off
                            
# Looking at statistical significance between different donor inocula
d = sample_data(human_donor_inoculae_ps)

# calculate stats using Wilcoxon test
INF = results_donors[d[,'Donor'] == 'INF2',]
CHLD = results_donors[d[,'Donor'] == 'CHLD2',]
ADLT = results_donors[d[,'Donor'] == 'ADLT2',]
ELD = results_donors[d[,'Donor'] == 'ELD2',]
wilcox.test(INF, CHLD)
wilcox.test(INF, ADLT)
wilcox.test(INF, ELD)
wilcox.test(CHLD, ADLT)
wilcox.test(CHLD, ELD)
wilcox.test(ADLT, ELD)
```


## **Donor-based Analysis** ##

Let's look at each donor separately and compare the humanized pig and mouse microbiotas to the original human donor microbiotas. For the first trial, we will refer to Donor_1 as 'INF2', Donor_2 as 'CHLD2', Donor_3 as 'ADLT2', and Donor_4 as 'ELD2'.


**Donor_1 (INF2) Analysis**

For the donor_1 (infant) microbiota, there were two groups of pigs (3 pigs/group): one group (early inoculations) were inoculated while they were on a milk diet, while the other group (late inoculations) were inoculated after they had been weaned and were on a solid diet. For this study we'll only concentrate on the late-inoculation piglets. Let's go ahead and compare INF2 donor, HMA mice, and the late-inoculated piglet fecal bacterial communities.

Let's first have a look at how the humanized animal model fecal bacterial communities cluster relative to the human donor inocula communities. As mentioned earlier, we'll compare only the 'late inoculation' piglets to the mouse and human donor samples:

```{r, echo=TRUE}
merged_pig_mouse_comp_no_cecal_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, SampleType != 'cecal')
pig_mouse_comp_INF2_comparison_ps <- subset_samples(merged_pig_mouse_comp_no_cecal_ps, Donor=="INF2")
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
pig_mouse_comp_INF2_comparison_ps

early_inoculation_donor_samples <- c('299', '300')
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, !SampleID %in% early_inoculation_donor_samples) # keeping only the donor inocula used to inoculate the mice and 'late' piglets
pig_mouse_comp_INF2_comparison_ps

rel_abund_INF2_ps <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function (OTU) OTU/sum(OTU))
rel_abund_INF2_ps
```

Let's see which ASVs were present in both the inocula used to inoculate the mice as well as the inoculum used to inoculate the piglets and thus define a 'core' INF2 microbiota. We can subsequently compare how these core ASVs are distributed between human donor and the humanized animal models:

```{r, echo=TRUE}
# Let's define a new core only considering the taxa found in the three donor samples which went to the mice and the HMA late inoculation piglets
human_donors_INF2_inocula_relevant_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == "human") # removing 'early' piglet inoculae
human_donors_INF2_inocula_relevant_ps
prevdf_INF2_donor = apply(X = otu_table(human_donors_INF2_inocula_relevant_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_INF2_inocula_relevant_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_INF2_donor = data.frame(Prevalence=prevdf_INF2_donor, TotalAbundance=taxa_sums(human_donors_INF2_inocula_relevant_ps))
View(prevdf_INF2_donor)

core_threshold = nsamples(human_donors_INF2_inocula_relevant_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_INF2_taxa <- rownames(prevdf_INF2_donor)[prevdf_INF2_donor$Prevalence >= core_threshold]
core_INF2_taxa # We have 26 ASVs which were detected in all 3 inocula

# Calculating the number of reads accounted for by the core ASVs 
human_donors_INF2_inocula_relevant_core_ps <- prune_taxa(core_INF2_taxa, human_donors_INF2_inocula_relevant_ps)
human_donors_INF2_inocula_relevant_core_ps

sum(otu_table(human_donors_INF2_inocula_relevant_core_ps))/sum(otu_table(human_donors_INF2_inocula_relevant_ps)) # core accounts for 98.9% of total reads
```

Now let's see how these core donor_1(INF2) donor ASVs are established in the two HMA animal models by looking at beta diversity :

```{r, echo=TRUE}
rel_abund_INF2_ps
rel_abund_INF2_core_ps <- prune_taxa(core_INF2_taxa, rel_abund_INF2_ps)
rel_abund_INF2_core_ps
write.table(otu_table(rel_abund_INF2_core_ps), "INF2_core_rel_abund.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE) # source data for publication

# plotting PCoA using unweighted unifrac distances for core taxa - donor_1
unw_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, unw_ord_INF2_core, color = "Species", shape = "Species") + ggtitle("Donor_1 PCoA plot - Unweighted UniFrac") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #+ scale_color_manual(values = c("orange", "green4", "dodgerblue3"))#legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") 
unweighted_pcoa_INF2_core

# plotting PCoA using Bray-Curtis distances for core taxa - donor_1
bray_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, bray_ord_INF2_core, color = "Species", shape = "Species") + ggtitle("Donor_1 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species") # + scale_color_manual(values = c("orange", "green4", "dodgerblue3")) #labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
bray_pcoa_INF2_core

INF2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_INF2_core, bray_pcoa_INF2_core)
INF2_core_unifrac_bray_comb
ggsave('Donor_1_PCoA_unw.pdf', unweighted_pcoa_INF2_core, device = "pdf", dpi = 500)
```


Hierarchical clustering based on 'Bray-Curtis' distances

```{r, echo= TRUE}
## INF2 Donor

# Use the relative abundance table
rel_abund_INF2_core_ps
bray_dist_INF2 <- phyloseq::distance(rel_abund_INF2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_INF2 <- hclust(bray_dist_INF2, method = "ward.D2")
plot(bray_clust_INF2)

# need to color plot by 'Species' to see clustering patterns 
install.packages("dendextend"); packageVersion("dendextend") 
library(dendextend)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_INF2 <- read.table("mapping_files_for_dendrograms/INF2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "") # 'mapping_files_for_dendrograms' folder has been uploaded to GitHub
View(mapping_file_dendro_INF2)
mapping_file_dendro_INF2$Color_Sp <- as.character(mapping_file_dendro_INF2$Color_Sp)

# making dendrogram
bray_dend_INF2 <- as.dendrogram(bray_clust_INF2, hang=0.1)
SampleIDs_dend_INF2 <- labels(bray_dend_INF2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_INF2

sorted_mapping_INF2 <- mapping_file_dendro_INF2[match(SampleIDs_dend_INF2, mapping_file_dendro_INF2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_INF2)

# color dendrogram labels 
labels_colors(bray_dend_INF2) <- sorted_mapping_INF2$Color_Sp
labels_colors(bray_dend_INF2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_INF2) <- sorted_mapping_INF2$Time_point
bray_dend_INF2 <- set(bray_dend_INF2, "labels_cex", 0.5)
labels(bray_dend_INF2)
plot(bray_dend_INF2, ylab="Bray-Curtis distances")
```


Visualizing differences in abundances of ASVs using a heatmap :

```{r, echo=TRUE}

# DESeq2 conversion and call
# Install DESeq2 if not already installed
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")

# load DESeq2
library('DESeq2'); packageVersion('DESeq2')

## Using variance stabilized count table
pig_mouse_comp_INF2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_INF2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
INF2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_INF2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_INF2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_INF2_comparison_ps)))

INF2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
INF2_vst_core_ps <- prune_taxa(core_INF2_taxa, INF2_vst_ps)
INF2_vst_core_ps
```


Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
install.packages("gplots")
library("gplots"); packageVersion('gplots')

asv_tab <- otu_table(INF2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
write.table(asv_tab_df, "INF2_ASV_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
write.table(tax_table(INF2_vst_core_ps), "INF2_taxa_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)

## We want to combine the ASV_ID with the Genus or Family name (or the lowest taxonomic rank available) available for a given ASV.
## Therefore, we can open "INF2_taxa_table.txt" from above in MS Excel and merge the ASV_ID column with the appropriate taxonomic rank column (for example using =A2&" "&F2).
## We can then paste this 'merged' column as the ASV_ID column in "INF2_ASV_table.txt". We'll then rename this ASV table as "INF2_ASV_table_heatmap.txt" and read it into R. These files can be found in the folder 'mapping_files_for_heatmaps'
asv_tab_w_taxa_INF <- read.table("mapping_files_for_heatmaps/INF2_ASV_table_heatmap.txt", header = T, sep = "\t", row.names = 1, check.names = F) # "mapping_files_for_heatmaps" folder has been uploaded to GitHub
View(asv_tab_w_taxa_INF) 

input <- as.matrix(asv_tab_w_taxa_INF)
dim(input)

col_labels <- labels(bray_dend_INF2)
str(bray_dend_INF2)
View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_INF2))]
View(col_labels)

col_cols <- labels_colors(bray_dend_INF2)
View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_INF2))]
View(col_cols)

#par(mgp=c(0.5,0.4,0))

#pdf("INF_heatmap.pdf", title = "Figure1")
heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.40, cexCol=0.7, scale="none", Colv = bray_dend_INF2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.5, 9.5), key = T, dendrogram = "column", xlab = "inocula and fecal samples", ylab = "Donor_1 core ASVs") 

legend(0.85, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.64)

#dev.off()
```


**Donor_2 (CHLD2) Analysis**

Let's separate out the relevant 'Child' donor samples :

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_master_ps <- subset_samples(merged_pig_mouse_comp_no_cecal_ps, Donor == 'CHLD2')
pig_mouse_comp_CHLD2_master_ps 

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animal fecal bacterial communities
pig_mouse_comp_CHLD2_comparison_ps <- subset_samples(pig_mouse_comp_CHLD2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_CHLD2_comparison_ps

# Transform count table to relative abundance table
rel_abund_CHLD2_ps <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x){x / sum(x)}) 
rel_abund_CHLD2_ps
```


Let's see how the donor_2 samples look like

```{r, echo=TRUE}
human_donors_CHLD2_ps <- subset_samples(pig_mouse_comp_CHLD2_master_ps, Species == 'human')
human_donors_CHLD2_ps
human_donors_CHLD2_inocula_ps <- subset_samples(human_donors_CHLD2_ps, Actual_inoculum == 'yes')
human_donors_CHLD2_inocula_ps
```

Defining a 'core' donor_2 microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' CHLD2 donor microbiota
prevdf_CHLD2 = apply(X = otu_table(human_donors_CHLD2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_CHLD2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_CHLD2 = data.frame(Prevalence=prevdf_CHLD2, TotalAbundance=taxa_sums(human_donors_CHLD2_inocula_ps))
View(prevdf_CHLD2)

core_threshold = nsamples(human_donors_CHLD2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_CHLD2_taxa <- rownames(prevdf_CHLD2)[prevdf_CHLD2$Prevalence >= core_threshold]
human_donors_CHLD2_inocula_core_ps <- prune_taxa(core_CHLD2_taxa, human_donors_CHLD2_inocula_ps)
human_donors_CHLD2_inocula_core_ps # 76 core ASVs identified
sum(otu_table(human_donors_CHLD2_inocula_core_ps))/sum(otu_table(human_donors_CHLD2_inocula_ps)) # the 76 core ASVs account for 96.37% of total CHLD2 donor reads
```

Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula bacterial communities for donor_2 (CHLD2). 

```{r, echo=TRUE}
rel_abund_CHLD2_ps
rel_abund_CHLD2_core_ps <- prune_taxa(core_CHLD2_taxa, rel_abund_CHLD2_ps)
rel_abund_CHLD2_core_ps 
write.table(otu_table(rel_abund_CHLD2_core_ps), "CHLD2_rel_abund.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # source data for publication

# plotting PCoA using unweighted unifrac distances for core taxa - CHLD2
unw_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, unw_ord_CHLD2_core, color = "Species", shape = "Species") + ggtitle("Donor_2 PCoA plot - Unweighted UniFrac") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") 
unweighted_pcoa_CHLD2_core

# plotting PCoA using Bray-Curtis distances for core taxa - CHLD2
bray_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, bray_ord_CHLD2_core, color = "Species", shape = "Species") + ggtitle("Donor_2 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
bray_pcoa_CHLD2_core

CHLD_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_CHLD2_core, bray_pcoa_CHLD2_core)
CHLD_core_unifrac_bray_comb
```


Hierarchical clustering based on 'Bray-Curtis' distances

```{r, echo= TRUE}
## CHLD2 Donor

# Use the relative abundance table
rel_abund_CHLD2_core_ps
bray_dist_CHLD2 <- phyloseq::distance(rel_abund_CHLD2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_CHLD2 <- hclust(bray_dist_CHLD2, method = "ward.D2")
plot(bray_clust_CHLD2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_CHLD2 <- read.table("mapping_files_for_dendrograms/CHLD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_CHLD2)
mapping_file_dendro_CHLD2$Color_Sp <- as.character(mapping_file_dendro_CHLD2$Color_Sp)

# making dendrogram
bray_dend_CHLD2 <- as.dendrogram(bray_clust_CHLD2, hang=0.1)
SampleIDs_dend_CHLD2 <- labels(bray_dend_CHLD2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_CHLD2

sorted_mapping_CHLD2 <- mapping_file_dendro_CHLD2[match(SampleIDs_dend_CHLD2, mapping_file_dendro_CHLD2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_CHLD2)

# color dendrogram labels 
labels_colors(bray_dend_CHLD2) <- sorted_mapping_CHLD2$Color_Sp
labels_colors(bray_dend_CHLD2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_CHLD2) <- sorted_mapping_CHLD2$Time_point
bray_dend_CHLD2 <- set(bray_dend_CHLD2, "labels_cex", 0.5)
labels(bray_dend_CHLD2)
plot(bray_dend_CHLD2, ylab="Bray-Curtis distances")
```


Making heatmaps to visualize distribution of core ASVs :

```{r, echo=TRUE}
## Using variance stabilized count table
pig_mouse_comp_CHLD2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_CHLD2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
CHLD2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_CHLD2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_CHLD2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_CHLD2_comparison_ps)))

CHLD2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
CHLD2_vst_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_vst_ps)
CHLD2_vst_core_ps
#plot_heatmap(CHLD2_vst_core_ps, sample.label = "Species", sample.order = "Species")
#plot_heatmap(CHLD2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
CHLD2_vst_core_ps
asv_tab <- otu_table(CHLD2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
write.table(asv_tab_df, "CHLD2_ASV_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
write.table(tax_table(CHLD2_vst_core_ps), "CHLD2_taxa_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
asv_tab_w_taxa_CHLD <- read.table("mapping_files_for_heatmaps/CHLD2_ASV_table_heatmap.txt", header = T, sep = "\t", row.names = 1, check.names = F)
View(asv_tab_w_taxa_CHLD)

input <- as.matrix(asv_tab_w_taxa_CHLD)
dim(input)

col_labels <- labels(bray_dend_CHLD2)
View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_CHLD2))]
View(col_labels)

col_cols <- labels_colors(bray_dend_CHLD2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_CHLD2))]
#View(col_cols)

#pdf("CHLD_heatmap.pdf")
heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.3, cexCol=0.65, scale="none", Colv = bray_dend_CHLD2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.0,8.0), xlab = "inocula and fecal samples", ylab = "Donor_2 core ASVs", dendrogram = "column", key = T) 

legend(0.92, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.5)
#dev.off()
```
 
 
 **Donor_3 (ADLT2) Analysis**
 

Separating out the relevant Donor_3 samples :

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_master_ps <- subset_samples(merged_pig_mouse_comp_no_cecal_ps, Donor == 'ADLT2')
pig_mouse_comp_ADLT2_master_ps
sample_names(pig_mouse_comp_ADLT2_master_ps)

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animal fecal bacterial communities
pig_mouse_comp_ADLT2_comparison_ps <- subset_samples(pig_mouse_comp_ADLT2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_ADLT2_comparison_ps 

# Sample 105 was identified as an outlier so it shall be removed
pig_mouse_comp_ADLT2_comparison_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, SampleID != '105')
pig_mouse_comp_ADLT2_comparison_ps

# Transform count table to relative abundance table
rel_abund_ADLT2_ps <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x){x / sum(x)}) 
rel_abund_ADLT2_ps
```


Let's see how the donor_3 samples look like

```{r, echo=TRUE}
human_donors_ADLT2_ps <- subset_samples(pig_mouse_comp_ADLT2_master_ps, Species == 'human')
human_donors_ADLT2_ps
human_donors_ADLT2_inocula_ps <- subset_samples(human_donors_ADLT2_ps, Actual_inoculum == 'yes')
human_donors_ADLT2_inocula_ps
```

Defining a 'core' donor_3 microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' ADLT2 donor microbiota
prevdf_ADLT2 = apply(X = otu_table(human_donors_ADLT2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ADLT2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ADLT2 = data.frame(Prevalence=prevdf_ADLT2, TotalAbundance=taxa_sums(human_donors_ADLT2_inocula_ps))
View(prevdf_ADLT2)

core_threshold = nsamples(human_donors_ADLT2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_ADLT2_taxa <- rownames(prevdf_ADLT2)[prevdf_ADLT2$Prevalence >= core_threshold]
human_donors_ADLT2_inocula_core_ps <- prune_taxa(core_ADLT2_taxa, human_donors_ADLT2_inocula_ps)
human_donors_ADLT2_inocula_core_ps # 140 core ASVs identified
sum(otu_table(human_donors_ADLT2_inocula_core_ps))/sum(otu_table(human_donors_ADLT2_inocula_ps)) # the 140 core ASVs account for ~ 97.58 % of total ADLT2 donor reads
```


Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula bacterial communities for donor_3. 

```{r, echo=TRUE}
rel_abund_ADLT2_ps
rel_abund_ADLT2_core_ps <- prune_taxa(core_ADLT2_taxa, rel_abund_ADLT2_ps)
rel_abund_ADLT2_core_ps 
write.table(otu_table(rel_abund_ADLT2_core_ps), "ADLT2_rel_abund_table.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE) # source data for publication

# plotting PCoA using unweighted unifrac distances for core taxa - donor_3
unw_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, unw_ord_ADLT2_core, color = "Species", shape = "Species") + ggtitle("Donor_3 donor PCoA plot - Unweighted UniFrac") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
unweighted_pcoa_ADLT2_core

# plotting PCoA using Bray-Curtis distances for core taxa - donor_3
bray_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, bray_ord_ADLT2_core, color = "Species", shape = "Species" ) + ggtitle("Donor_3 donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
bray_pcoa_ADLT2_core

ADLT2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ADLT2_core, bray_pcoa_ADLT2_core)
ADLT2_core_unifrac_bray_comb
```


Hierarchical clustering based on Bray-Curtis distances

```{r, echo=TRUE}

# Use the relative abundance table
rel_abund_ADLT2_core_ps
bray_dist_ADLT2 <- phyloseq::distance(rel_abund_ADLT2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_ADLT2 <- hclust(bray_dist_ADLT2, method = "ward.D2")
plot(bray_clust_ADLT2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_ADLT2 <- read.table("mapping_files_for_dendrograms/ADLT2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_ADLT2)
mapping_file_dendro_ADLT2$Color_Sp <- as.character(mapping_file_dendro_ADLT2$Color_Sp)

# making dendrogram
bray_dend_ADLT2 <- as.dendrogram(bray_clust_ADLT2, hang=0.1)
SampleIDs_dend_ADLT2 <- labels(bray_dend_ADLT2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ADLT2

sorted_mapping_ADLT2 <- mapping_file_dendro_ADLT2[match(SampleIDs_dend_ADLT2, mapping_file_dendro_ADLT2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_ADLT2)

# color dendrogram labels 
labels_colors(bray_dend_ADLT2) <- sorted_mapping_ADLT2$Color_Sp
labels_colors(bray_dend_ADLT2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_ADLT2) <- sorted_mapping_ADLT2$Time_point
bray_dend_ADLT2 <- set(bray_dend_ADLT2, "labels_cex", 0.5)
plot(bray_dend_ADLT2, ylab="Bray-Curtis distances")
```


Making heatmaps to visualize distriution of core ASVs among the 3 host species :

```{r, echo=TRUE}

## Using variance stabilized count table
pig_mouse_comp_ADLT2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_ADLT2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
ADLT2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_ADLT2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_ADLT2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_ADLT2_comparison_ps)))

ADLT2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
ADLT2_vst_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_vst_ps)
ADLT2_vst_core_ps
plot_heatmap(ADLT2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(ADLT2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
asv_tab <- otu_table(ADLT2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
dim(asv_tab_df)
write.table(asv_tab_df, "ADLT2_ASV_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
write.table(tax_table(ADLT2_vst_core_ps), "ADLT2_taxa_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
asv_tab_w_taxa_ADLT <- read.table("mapping_files_for_heatmaps/ADLT2_ASV_table_heatmap.txt", header = T, sep = "\t", row.names = 1, check.names = F)
View(asv_tab_w_taxa_ADLT)

input <- as.matrix(asv_tab_w_taxa_ADLT)
dim(input)

col_labels <- labels(bray_dend_ADLT2)
#View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_ADLT2))]
#View(col_labels)

col_cols <- labels_colors(bray_dend_ADLT2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_ADLT2))]
#View(col_cols)

#pdf("ADLT_heatmap.pdf")
heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.30, cexCol=0.65, scale="none", Colv = bray_dend_ADLT2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.0,8.0), xlab = "inocula and fecal samples", ylab = "Donor_3 core ASVs", key = T, dendrogram = "column") # margins = c(7.0, 7.5)

legend(0.92, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.6)
#dev.off()
```


**Donor_4 (ELD2/SNR2) Analysis**

Separating out the relevant Donor_4 samples :

```{r, echo=TRUE}
pig_mouse_comp_ELD2_master_ps <- subset_samples(merged_pig_mouse_comp_no_cecal_ps, Donor == 'ELD2')
pig_mouse_comp_ELD2_master_ps

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animal fecal bacterial communities
pig_mouse_comp_ELD2_comparison_ps <- subset_samples(pig_mouse_comp_ELD2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_ELD2_comparison_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, SampleType != 'cecal')
pig_mouse_comp_ELD2_comparison_ps

# Transform count table to relative abundance table
rel_abund_ELD2_ps <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x){x / sum(x)})
rel_abund_ELD2_ps
```

Let's see how the donor Donor_4 samples look like

```{r, echo=TRUE}
human_donors_ELD2_ps <- subset_samples(pig_mouse_comp_ELD2_master_ps, Species == 'human')
human_donors_ELD2_ps
human_donors_ELD2_inocula_ps <- subset_samples(human_donors_ELD2_ps, Actual_inoculum == 'yes')
human_donors_ELD2_inocula_ps
```

Defining a 'core' donor_4 microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' ELD2 donor microbiota
prevdf_ELD2 = apply(X = otu_table(human_donors_ELD2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ELD2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ELD2 = data.frame(Prevalence=prevdf_ELD2, TotalAbundance=taxa_sums(human_donors_ELD2_inocula_ps))
View(prevdf_ELD2)

core_threshold = nsamples(human_donors_ELD2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_ELD2_taxa <- rownames(prevdf_ELD2)[prevdf_ELD2$Prevalence >= core_threshold]
human_donors_ELD2_inocula_core_ps <- prune_taxa(core_ELD2_taxa, human_donors_ELD2_inocula_ps)
human_donors_ELD2_inocula_core_ps # 134 core ASVs identified
sum(otu_table(human_donors_ELD2_inocula_core_ps))/sum(otu_table(human_donors_ELD2_inocula_ps)) # the 134 core ASVs account for ~ 96.04 % of total ADLT2 donor reads
```


Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula bacterial communities for donor_4. 

```{r, echo=TRUE}
rel_abund_ELD2_ps
rel_abund_ELD2_core_ps <- prune_taxa(core_ELD2_taxa, rel_abund_ELD2_ps)
rel_abund_ELD2_core_ps 
write.table(otu_table(rel_abund_ELD2_core_ps), "SNR2_rel_abund_table.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = F) # source data for publication

# plotting PCoA using unweighted unifrac distances for core taxa - ELD2
unw_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, unw_ord_ELD2_core, color = "Species", shape = "Species") + ggtitle("Donor_4 PCoA plot - Unweighted UniFrac") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
unweighted_pcoa_ELD2_core

# plotting PCoA using Bray-Curtis distances for core taxa - ELD2
bray_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, bray_ord_ELD2_core, color = "Species", shape = "Species") + ggtitle("Donor_4 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
bray_pcoa_ELD2_core

ELD2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ELD2_core, bray_pcoa_ELD2_core)
ELD2_core_unifrac_bray_comb
```


Hierarchical clustering based on Bray-Curtis distances

```{r, echo=TRUE}

# Use the relative abundance table
rel_abund_ELD2_core_ps
bray_dist_ELD2 <- phyloseq::distance(rel_abund_ELD2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_ELD2 <- hclust(bray_dist_ELD2, method = "ward.D2")
plot(bray_clust_ELD2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_ELD2 <- read.table("mapping_files_for_dendrograms/ELD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_ELD2)
mapping_file_dendro_ELD2$Color_Sp <- as.character(mapping_file_dendro_ELD2$Color_Sp)

# making dendrogram
bray_dend_ELD2 <- as.dendrogram(bray_clust_ELD2, hang=0.1)
SampleIDs_dend_ELD2 <- labels(bray_dend_ELD2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ELD2

sorted_mapping_ELD2 <- mapping_file_dendro_ELD2[match(SampleIDs_dend_ELD2, mapping_file_dendro_ELD2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_ELD2)

# color dendrogram labels 
labels_colors(bray_dend_ELD2) <- sorted_mapping_ELD2$Color_Sp
labels_colors(bray_dend_ELD2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_ELD2) <- sorted_mapping_ELD2$Time_point
bray_dend_ELD2 <- set(bray_dend_ELD2, "labels_cex", 0.5)
plot(bray_dend_ELD2, ylab="Bray-Curtis distances")
```


Visualizing differences in relative abundances of ASVs using a heatmap :

```{r, echo=TRUE}

## Using variance stabilized count table
pig_mouse_comp_ELD2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_ELD2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
ELD2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_ELD2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_ELD2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_ELD2_comparison_ps)))

ELD2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
ELD2_vst_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_vst_ps)
ELD2_vst_core_ps
plot_heatmap(ELD2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(ELD2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```

Let's make a heatmap using hierarchical clustering :

```{r, echo=TRUE}
asv_tab <- otu_table(ELD2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
write.table(asv_tab_df, "ELD2_ASV_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
write.table(tax_table(ELD2_vst_core_ps), "ELD2_taxa_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
asv_tab_w_taxa_ELD <- read.table("mapping_files_for_heatmaps/ELD2_ASV_table_heatmap.txt", header = T, sep = "\t", row.names = 1, check.names = F)

input <- as.matrix(asv_tab_w_taxa_ELD)
dim(input)

col_labels <- labels(bray_dend_ELD2)
#View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_ELD2))]
#View(col_labels)

col_cols <- labels_colors(bray_dend_ELD2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_ELD2))]
#View(col_cols)

#pdf("test_heatmap_plot.pdf")

#par(mgp=c(0.5,0.4,0))
#pdf("SNR_heatmap.pdf")

heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.2, cexCol=0.65, scale="none", Colv = bray_dend_ELD2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.0,8.0),  xlab = "inocula and fecal samples", ylab = "Donor_4 core ASVs", dendrogram = "column", key = T) 

legend(0.95, 1.13, xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.5)
#dev.off()
```



##**Determining the establishment of core donor ASVs in the  HMA animal models**##

Let's see how many of these core ASVs have established in the HMA animal models. To be a colonizer, an ASV needs to be found in at least a single fecal sample. In order to be considered 'persistently colonizing', an ASV needs to be detected in at least **4** sampling time points and in at least **2/3 of animals** in each of those sampling time points:


**Donor_1 (INF2)**

```{r, echo=TRUE}

# establishment of donor_1 core ASVs in HMA mice

core_INF2_taxa # 26 core ASVs

INF2_HMA_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
INF2_HMA_mice_ps
INF2_HMA_mice_core_ps <- prune_taxa(core_INF2_taxa, INF2_HMA_mice_ps)
INF2_HMA_mice_core_ps

# Let's turn the making of a 'Prevalence' data frame into a function since we have to determine prevalence quite a number of times
makePrevDF <- function(phyloseqobject, prevthresh) {
  prevdf = apply(X = otu_table(phyloseqobject),
               MARGIN = ifelse(taxa_are_rows(phyloseqobject), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
  prevdf = data.frame(Prevalence=prevdf, TotalAbundance=taxa_sums(phyloseqobject))
  prev_threshold = prevthresh
  colonized_taxa <- rownames(prevdf)[prevdf$Prevalence >= prev_threshold]
  return(list(prevdf, colonized_taxa))
}


# Let's see how many donor_1 core ASVs were found in at least one INF2 HMA mouse fecal sample 
prevdf_INF2_core_mice <- makePrevDF(INF2_HMA_mice_core_ps, 1)
prevdf_INF2_core_mice_table <- as.data.frame(prevdf_INF2_core_mice[1])
colonized_prevdf_INF2_core_mice <- prevdf_INF2_core_mice_table[prevdf_INF2_core_mice_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_core_mice) # 21 of the 26 core INF2 donor ASVs found in at least one INF2 mouse fecal sample. These are defined as 'COLONIZERS' in the manuscript
core_INF2_taxa_colonizing_mice <-prevdf_INF2_core_mice[2]
core_INF2_taxa_colonizing_mice <- unlist(core_INF2_taxa_colonizing_mice) 
core_INF2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF2_HMA_mice_core_ps_48HRS <- subset_samples(INF2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
INF2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_mice_core_ps_48HRS,5) # This time point as well as the 1 and 2week time points had 8 animals
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
View(prevdf_48HRS_table)
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 5,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
View(colonized_prevdf_48HRS)
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_mice_core_ps_1week <- subset_samples(INF2_HMA_mice_core_ps, Description == '1week_post_inoc')
INF2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(INF2_HMA_mice_core_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 5,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_mice_core_ps_2weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
INF2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_mice_core_ps_2weeks,5)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 5,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_mice_core_ps_3weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
INF2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_mice_core_ps_3weeks,4) # from this sampling time point onwards there were only 7 mice
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 4,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_mice_core_ps_4weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
INF2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_mice_core_ps_4weeks,4)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 4,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_mice_core_ps_5weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
INF2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_mice_core_ps_5weeks,4)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 4,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_mice_core_ps_40days <- subset_samples(INF2_HMA_mice_core_ps, Description == 'Final_collection')
INF2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_mice_core_ps_40days,4)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 4,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_INF2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
View(merged_prev_df_INF2_core_mice)
str(merged_prev_df_INF2_core_mice) # All columns considered factors
merged_prev_df_INF2_core_mice[2:8] <- lapply(merged_prev_df_INF2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_core_mice)
View(merged_prev_df_INF2_core_mice)
     
merged_prev_df_INF2_core_mice[is.na(merged_prev_df_INF2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_core_mice$total <- rowSums(merged_prev_df_INF2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_core_mice)
merged_prev_df_INF2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_core_mice)
write.table(merged_prev_df_INF2_core_mice, "merged_prev_df_INF2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in 2/3 of animals
perst_colonizing_core_INF2_mice_df <- subset(merged_prev_df_INF2_core_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_INF2_mice_df)
perst_colonizing_core_INF2_mice <- perst_colonizing_core_INF2_mice_df[,1]
perst_colonizing_core_INF2_mice # 16 ASVs persistently colonizing mice. These are the 'PERSISTENT COLONIZERS' referred to in the manuscript
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of donor_1 core ASVs in HMA piglets

core_INF2_taxa # 26 core ASVs

INF2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
INF2_HMA_piglets_ps <- subset_samples(INF2_HMA_piglets_ps, Timingofinocu != "early" & SampleType != "cecal")
INF2_HMA_piglets_ps
INF2_HMA_piglets_core_ps <- prune_taxa(core_INF2_taxa, INF2_HMA_piglets_ps)
INF2_HMA_piglets_core_ps

# Let's see how many donor_1 core ASVs were found in at least one INF2 HMA piglet fecal sample 
prevdf_INF2_core_piglets <- makePrevDF(INF2_HMA_piglets_core_ps, 1)
prevdf_INF2_core_piglets_table <- as.data.frame(prevdf_INF2_core_piglets[1])
colonized_prevdf_INF2_core_piglets <- prevdf_INF2_core_piglets_table[prevdf_INF2_core_piglets_table$Prevalence >= 1,]
View(colonized_prevdf_INF2_core_piglets) # 20 of the 26 core INF2 donor ASVs found in at least one INF2 piglet fecal sample. These are the 'COLONIZERS'
core_INF2_taxa_colonizing_piglets <-prevdf_INF2_core_piglets[2]
core_INF2_taxa_colonizing_piglets <- unlist(core_INF2_taxa_colonizing_piglets) 
core_INF2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF2_HMA_piglets_core_ps_48HRS <- subset_samples(INF2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
INF2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_piglets_core_ps_48HRS,3) # This time point had 4 animals
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 3,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_piglets_core_ps_1week <- subset_samples(INF2_HMA_piglets_core_ps, Description == '1week_post_inoc')
INF2_HMA_piglets_core_ps_1week 

prevdf_1week <- makePrevDF(INF2_HMA_piglets_core_ps_1week,2) # there were only 3 piglets from this time point onwards
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 2,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_piglets_core_ps_2weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
INF2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 2,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_piglets_core_ps_3weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
INF2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 2,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_piglets_core_ps_4weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
INF2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 2,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_piglets_core_ps_5weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
INF2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 2,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_piglets_core_ps_40days <- subset_samples(INF2_HMA_piglets_core_ps, Description == 'Final_collection')
INF2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 2,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_INF2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_INF2_core_piglets) # All columns considered factors
merged_prev_df_INF2_core_piglets[2:8] <- lapply(merged_prev_df_INF2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_core_piglets)
View(merged_prev_df_INF2_core_piglets)
     
merged_prev_df_INF2_core_piglets[is.na(merged_prev_df_INF2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_core_piglets$total <- rowSums(merged_prev_df_INF2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_core_piglets)
merged_prev_df_INF2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_core_piglets)
write.table(merged_prev_df_INF2_core_piglets, "merged_prev_df_INF2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_INF2_piglets_df <- subset(merged_prev_df_INF2_core_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_INF2_piglets_df)
perst_colonizing_core_INF2_piglets <- perst_colonizing_core_INF2_piglets_df[,1]
perst_colonizing_core_INF2_piglets # 15 ASVs persistently colonizing piglets. These are the 'PERSISTENT COLONIZERS'

## core INF2 donor ASVs that are persistently colonizing each HMA animal model based on colonization criteria (detected in > 50% of animals in >= 4 sampling time points) with taxonomy assignments

# for INF2 HMA piglets
perst_colonizing_core_INF2_piglets_ps <- prune_taxa(perst_colonizing_core_INF2_piglets,INF2_HMA_piglets_core_ps)
perst_colonizing_core_INF2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_INF2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_INF2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_INF2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for INF2 HMA mice
perst_colonizing_core_INF2_mice_ps <- prune_taxa(perst_colonizing_core_INF2_mice,INF2_HMA_mice_core_ps)
perst_colonizing_core_INF2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_INF2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_INF2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_INF2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_2 (CHLD2)**

```{r, echo=TRUE}

# establishment of donor_2 core ASVs in HMA mice

core_CHLD2_taxa # 76 core ASVs

pig_mouse_comp_CHLD2_comparison_ps
CHLD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
CHLD2_HMA_mice_ps
CHLD2_HMA_mice_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_HMA_mice_ps)
CHLD2_HMA_mice_core_ps

# Let's see how many core CHLD2 donor ASVs were found in at least one CHLD2 HMA mouse fecal sample 
prevdf_CHLD2_core_mice <- makePrevDF(CHLD2_HMA_mice_core_ps, 1)
prevdf_CHLD2_core_mice_table <- as.data.frame(prevdf_CHLD2_core_mice[1])
View(prevdf_CHLD2_core_mice_table)
colonized_prevdf_CHLD2_core_mice <- prevdf_CHLD2_core_mice_table[prevdf_CHLD2_core_mice_table$Prevalence >= 1,]
View(colonized_prevdf_CHLD2_core_mice) # 37 of the 76 core CHLD2 donor ASVs found in at least one CHLD2 mouse fecal sample. These are the 'COLONIZERS'
core_CHLD2_taxa_colonizing_mice <-prevdf_CHLD2_core_mice[2]
core_CHLD2_taxa_colonizing_mice <- unlist(core_CHLD2_taxa_colonizing_mice) 
core_CHLD2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD2_HMA_mice_core_ps_48HRS <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD2_HMA_mice_core_ps_48HRS,6) 
View(prevdf_48HRS)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
View(prevdf_48HRS_table)
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 6,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
View(colonized_prevdf_48HRS)
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_mice_core_ps_1week <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '1week_post_inoc')
CHLD2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_mice_core_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
View(prevdf_1week_table)
#colonized_prevdf_1week <- filter(prevdf_1week_table, Prevalence >= 6)
#View(colonized_prevdf_1week)
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
View(colonized_prevdf_1week)
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
View(colonized_prevdf_1week)
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_mice_core_ps_2weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
View(prevdf_2weeks_table)
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
View(colonized_prevdf_2weeks)
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_mice_core_ps_3weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 6,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_mice_core_ps_4weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_mice_core_ps_5weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 6,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_mice_core_ps_40days <- subset_samples(CHLD2_HMA_mice_core_ps, Description == 'Final_collection')
CHLD2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 6,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_CHLD2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_CHLD2_core_mice) # All columns considered factors
merged_prev_df_CHLD2_core_mice[2:8] <- lapply(merged_prev_df_CHLD2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_core_mice)
View(merged_prev_df_CHLD2_core_mice)
     
merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_core_mice$total <- rowSums(merged_prev_df_CHLD2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_core_mice)
merged_prev_df_CHLD2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
View(merged_prev_df_CHLD2_core_mice)
write.table(merged_prev_df_CHLD2_core_mice, "merged_prev_df_CHLD2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_core_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_CHLD2_mice_df)
perst_colonizing_core_CHLD2_mice <- perst_colonizing_core_CHLD2_mice_df[,1]
perst_colonizing_core_CHLD2_mice # 21 ASVs persistently colonizing mice. These are the 'PERSISTENT COLONIZERS'.
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of donor_2 core ASVs in HMA piglets

core_CHLD2_taxa # 76 core ASVs

CHLD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
CHLD2_HMA_piglets_ps
CHLD2_HMA_piglets_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_HMA_piglets_ps)
CHLD2_HMA_piglets_core_ps

# Let's see how many core CHLD2 donor ASVs were found in at least one CHLD2 HMA piglet fecal sample 
prevdf_CHLD2_core_piglets <- makePrevDF(CHLD2_HMA_piglets_core_ps, 1)
prevdf_CHLD2_core_piglets_table <- as.data.frame(prevdf_CHLD2_core_piglets[1])
colonized_prevdf_CHLD2_core_piglets <- prevdf_CHLD2_core_piglets_table[prevdf_CHLD2_core_piglets_table$Prevalence >= 1,]
View(colonized_prevdf_CHLD2_core_piglets) # 65 of the 76 core CHLD2 donor ASVs found in at least one CHLD2 piglet fecal sample. These are the 'COLONIZERS'.
core_CHLD2_taxa_colonizing_piglets <-prevdf_CHLD2_core_piglets[2]
core_CHLD2_taxa_colonizing_piglets <- unlist(core_CHLD2_taxa_colonizing_piglets) 
core_CHLD2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD2_HMA_piglets_core_ps_48HRS <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_piglets_core_ps_48HRS
prevdf_48HRS <- makePrevDF(CHLD2_HMA_piglets_core_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
View(prevdf_48HRS_table)
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 2,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_piglets_core_ps_1week <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '1week_post_inoc')
CHLD2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 2,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_piglets_core_ps_2weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 2,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_piglets_core_ps_3weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 2,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_piglets_core_ps_4weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 2,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_piglets_core_ps_5weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 2,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_piglets_core_ps_40days <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == 'Final_collection')
CHLD2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 2,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_CHLD2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_CHLD2_core_piglets) # All columns considered factors
merged_prev_df_CHLD2_core_piglets[2:8] <- lapply(merged_prev_df_CHLD2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_core_piglets)
View(merged_prev_df_CHLD2_core_piglets)
     
merged_prev_df_CHLD2_core_piglets[is.na(merged_prev_df_CHLD2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_core_piglets$total <- rowSums(merged_prev_df_CHLD2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_core_piglets)
merged_prev_df_CHLD2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD2_core_piglets)
write.table(merged_prev_df_CHLD2_core_piglets, "merged_prev_df_CHLD2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_core_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_CHLD2_piglets_df)
perst_colonizing_core_CHLD2_piglets <- perst_colonizing_core_CHLD2_piglets_df[,1]
perst_colonizing_core_CHLD2_piglets # 43 ASVs persistently colonizing piglets. These are the 'PERSISTENT COLONIZERS'

# core CHLD2 donor ASVs that are persistently colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for CHLD2 HMA piglets
perst_colonizing_core_CHLD2_piglets_ps <- prune_taxa(perst_colonizing_core_CHLD2_piglets,CHLD2_HMA_piglets_core_ps)
perst_colonizing_core_CHLD2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_CHLD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_CHLD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_CHLD2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for CHLD2 HMA mice
perst_colonizing_core_CHLD2_mice_ps <- prune_taxa(perst_colonizing_core_CHLD2_mice,CHLD2_HMA_mice_core_ps)
perst_colonizing_core_CHLD2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_CHLD2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_CHLD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_CHLD2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_3 (ADLT2)**

```{r, echo=TRUE}

# establishment of donor_3 core ASVs in HMA mice

core_ADLT2_taxa # 140 core ASVs

pig_mouse_comp_ADLT2_comparison_ps
ADLT2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
ADLT2_HMA_mice_ps
ADLT2_HMA_mice_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_HMA_mice_ps)
ADLT2_HMA_mice_core_ps

# Let's see how many core ADLT2 donor ASVs were found in at least one ADLT2 HMA mouse fecal sample 
prevdf_ADLT2_core_mice <- makePrevDF(ADLT2_HMA_mice_core_ps, 1)
prevdf_ADLT2_core_mice_table <- as.data.frame(prevdf_ADLT2_core_mice[1])
colonized_prevdf_ADLT2_core_mice <- prevdf_ADLT2_core_mice_table[prevdf_ADLT2_core_mice_table$Prevalence >= 1,]
View(colonized_prevdf_ADLT2_core_mice) # 66 of the 140 core ADLT2 donor ASVs found in at least one ADLT2 mouse fecal sample. These are the 'COLONIZERS'
core_ADLT2_taxa_colonizing_mice <-prevdf_ADLT2_core_mice[2]
core_ADLT2_taxa_colonizing_mice <- unlist(core_ADLT2_taxa_colonizing_mice) 
core_ADLT2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT2_HMA_mice_core_ps_48HRS <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT2_HMA_mice_core_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
View(prevdf_48HRS_table)
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 6,]
View(colonized_prevdf_48HRS)
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
View(colonized_prevdf_48HRS)
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_mice_core_ps_1week <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '1week_post_inoc')
ADLT2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_mice_core_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_mice_core_ps_2weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_mice_core_ps_3weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 6,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_mice_core_ps_4weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_mice_core_ps_5weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 6,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_mice_core_ps_40days <- subset_samples(ADLT2_HMA_mice_core_ps, Description == 'Final_collection')
ADLT2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 6,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ADLT2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ADLT2_core_mice) # All columns considered factors
merged_prev_df_ADLT2_core_mice[2:8] <- lapply(merged_prev_df_ADLT2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_core_mice)
View(merged_prev_df_ADLT2_core_mice)
     
merged_prev_df_ADLT2_core_mice[is.na(merged_prev_df_ADLT2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_core_mice$total <- rowSums(merged_prev_df_ADLT2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_core_mice)
merged_prev_df_ADLT2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_core_mice)
write.table(merged_prev_df_ADLT2_core_mice, "merged_prev_df_ADLT2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_core_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_ADLT2_mice_df)
perst_colonizing_core_ADLT2_mice <- perst_colonizing_core_ADLT2_mice_df[,1]
perst_colonizing_core_ADLT2_mice # 31 ASVs persistently colonizing mice. These are the 'PERSISTENT COLONIZERS'.
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of core ADLT2 ASVs in HMA piglets

core_ADLT2_taxa # 140 core ASVs

ADLT2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
ADLT2_HMA_piglets_ps
ADLT2_HMA_piglets_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_HMA_piglets_ps)
ADLT2_HMA_piglets_core_ps

# Let's see how many core ADLT2 donor ASVs were found in at least one ADLT2 HMA piglet fecal sample 
prevdf_ADLT2_core_piglets <- makePrevDF(ADLT2_HMA_piglets_core_ps, 1)
prevdf_ADLT2_core_piglets_table <- as.data.frame(prevdf_ADLT2_core_piglets[1])
colonized_prevdf_ADLT2_core_piglets <- prevdf_ADLT2_core_piglets_table[prevdf_ADLT2_core_piglets_table$Prevalence >= 1,]
View(colonized_prevdf_ADLT2_core_piglets) # 106 of the 140 core ADLT2 donor ASVs found in at least one ADLT2 piglet fecal sample. These are the 'COLONIZERS'
core_ADLT2_taxa_colonizing_piglets <-prevdf_ADLT2_core_piglets[2]
core_ADLT2_taxa_colonizing_piglets <- unlist(core_ADLT2_taxa_colonizing_piglets) 
core_ADLT2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT2_HMA_piglets_core_ps_48HRS <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT2_HMA_piglets_core_ps_48HRS,2) # there were only two fecal samples left for analysis for this time point since 'sample 105' was an outlier.
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 2,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_piglets_core_ps_1week <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '1week_post_inoc')
ADLT2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_piglets_core_ps_1week,2) # from week 1 onwards there were 3 samples/time point
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 2,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_piglets_core_ps_2weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 2,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_piglets_core_ps_3weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 2,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_piglets_core_ps_4weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 2,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_piglets_core_ps_5weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 2,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_piglets_core_ps_40days <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == 'Final_collection')
ADLT2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 2,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ADLT2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ADLT2_core_piglets) # All columns considered factors
merged_prev_df_ADLT2_core_piglets[2:8] <- lapply(merged_prev_df_ADLT2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_core_piglets)
View(merged_prev_df_ADLT2_core_piglets)
     
merged_prev_df_ADLT2_core_piglets[is.na(merged_prev_df_ADLT2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_core_piglets$total <- rowSums(merged_prev_df_ADLT2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_core_piglets)
merged_prev_df_ADLT2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_core_piglets)
write.table(merged_prev_df_ADLT2_core_piglets, "merged_prev_df_ADLT2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_core_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_ADLT2_piglets_df)
perst_colonizing_core_ADLT2_piglets <- perst_colonizing_core_ADLT2_piglets_df[,1]
perst_colonizing_core_ADLT2_piglets # 69 ASVs persistently colonizing piglets. These are the 'PERSISTENT COLONIZERS'.

# core ADLT2 donor ASVs that are persistently colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ADLT2 HMA piglets
perst_colonizing_core_ADLT2_piglets_ps <- prune_taxa(perst_colonizing_core_ADLT2_piglets,ADLT2_HMA_piglets_core_ps)
perst_colonizing_core_ADLT2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_ADLT2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_ADLT2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_ADLT2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ADLT2 HMA mice
perst_colonizing_core_ADLT2_mice_ps <- prune_taxa(perst_colonizing_core_ADLT2_mice,ADLT2_HMA_mice_core_ps)
perst_colonizing_core_ADLT2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_ADLT2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_ADLT2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_ADLT2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_4 (ELD2/SNR2)**

```{r, echo=TRUE}

# establishment of donor_4 ASVs in HMA mice

core_ELD2_taxa # 134 core ASVs

pig_mouse_comp_ELD2_comparison_ps
ELD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
ELD2_HMA_mice_ps
ELD2_HMA_mice_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_HMA_mice_ps)
ELD2_HMA_mice_core_ps

# Let's see how many core ELD2 donor ASVs were found in at least one ELD2 HMA mouse fecal sample 
prevdf_ELD2_core_mice <- makePrevDF(ELD2_HMA_mice_core_ps, 1)
prevdf_ELD2_core_mice_table <- as.data.frame(prevdf_ELD2_core_mice[1])
colonized_prevdf_ELD2_core_mice <- prevdf_ELD2_core_mice_table[prevdf_ELD2_core_mice_table$Prevalence >= 1,]
View(colonized_prevdf_ELD2_core_mice) # 55 of the 134 core ELD2 donor ASVs found in at least one ELD2 mouse fecal sample. These are the 'COLONIZERS'.
core_ELD2_taxa_colonizing_mice <-prevdf_ELD2_core_mice[2]
core_ELD2_taxa_colonizing_mice <- unlist(core_ELD2_taxa_colonizing_mice) 
core_ELD2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ELD2_HMA_mice_core_ps_48HRS <- subset_samples(ELD2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
ELD2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_mice_core_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
View(prevdf_48HRS_table)
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 6,]
View(colonized_prevdf_48HRS)
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_mice_core_ps_1week <- subset_samples(ELD2_HMA_mice_core_ps, Description == '1week_post_inoc')
ELD2_HMA_mice_core_ps_1week # only 9 samples at this time point

prevdf_1week <- makePrevDF(ELD2_HMA_mice_core_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 5,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_mice_core_ps_2weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
ELD2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_mice_core_ps_3weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
ELD2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 6,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_mice_core_ps_4weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
ELD2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_mice_core_ps_5weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
ELD2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 6,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_mice_core_ps_40days <- subset_samples(ELD2_HMA_mice_core_ps, Description == 'Final_collection')
ELD2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 6,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ELD2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ELD2_core_mice) # All columns considered factors
merged_prev_df_ELD2_core_mice[2:8] <- lapply(merged_prev_df_ELD2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_core_mice)
View(merged_prev_df_ELD2_core_mice)
     
merged_prev_df_ELD2_core_mice[is.na(merged_prev_df_ELD2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_core_mice$total <- rowSums(merged_prev_df_ELD2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_core_mice)
merged_prev_df_ELD2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
View(merged_prev_df_ELD2_core_mice)
write.table(merged_prev_df_ELD2_core_mice, "merged_prev_df_ELD2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals
perst_colonizing_core_ELD2_mice_df <- subset(merged_prev_df_ELD2_core_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_ELD2_mice_df)
perst_colonizing_core_ELD2_mice <- perst_colonizing_core_ELD2_mice_df[,1]
perst_colonizing_core_ELD2_mice # 28 ASVs persistently colonizing mice. These are the 'PERSISTENT COLONIZERS'.
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of donor_4 core ASVs in HMA piglets

core_ELD2_taxa # 134 core ASVs

ELD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
ELD2_HMA_piglets_ps
ELD2_HMA_piglets_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_HMA_piglets_ps)
ELD2_HMA_piglets_core_ps

# Let's see how many core ELD2 donor ASVs were found in at least one ELD2 HMA piglet fecal samples 
prevdf_ELD2_core_piglets <- makePrevDF(ELD2_HMA_piglets_core_ps, 1)
prevdf_ELD2_core_piglets_table <- as.data.frame(prevdf_ELD2_core_piglets[1])
colonized_prevdf_ELD2_core_piglets <- prevdf_ELD2_core_piglets_table[prevdf_ELD2_core_piglets_table$Prevalence >= 1,]
View(colonized_prevdf_ELD2_core_piglets) # 92 of the 134 core ADLT2 donor ASVs found in at least one ELD2 piglet fecal sample
core_ELD2_taxa_colonizing_piglets <-prevdf_ELD2_core_piglets[2]
core_ELD2_taxa_colonizing_piglets <- unlist(core_ELD2_taxa_colonizing_piglets) 
core_ELD2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ELD2_HMA_piglets_core_ps_48HRS <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
ELD2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_piglets_core_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 2,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_piglets_core_ps_1week <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '1week_post_inoc')
ELD2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(ELD2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 2,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_piglets_core_ps_2weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
ELD2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 2,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_piglets_core_ps_3weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
ELD2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 2,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_piglets_core_ps_4weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
ELD2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 2,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_piglets_core_ps_5weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
ELD2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 2,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_piglets_core_ps_40days <- subset_samples(ELD2_HMA_piglets_core_ps, Description == 'Final_collection')
ELD2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 2,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ELD2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ELD2_core_piglets) # All columns considered factors
merged_prev_df_ELD2_core_piglets[2:8] <- lapply(merged_prev_df_ELD2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_core_piglets)
View(merged_prev_df_ELD2_core_piglets)
     
merged_prev_df_ELD2_core_piglets[is.na(merged_prev_df_ELD2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_core_piglets$total <- rowSums(merged_prev_df_ELD2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_core_piglets)
merged_prev_df_ELD2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ELD2_core_piglets)
write.table(merged_prev_df_ELD2_core_piglets, "merged_prev_df_ELD2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in >= 4 time points in at least 2/3 of animals 
perst_colonizing_core_ELD2_piglets_df <- subset(merged_prev_df_ELD2_core_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_core_ELD2_piglets_df)
perst_colonizing_core_ELD2_piglets <- perst_colonizing_core_ELD2_piglets_df[,1]
perst_colonizing_core_ELD2_piglets # 38 ASVs persistently colonizing piglets. These are the 'PERSISTENT COLONIZERS'.

# core ELD2 donor ASVs that are persistently colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ELD2 HMA piglets
perst_colonizing_core_ELD2_piglets_ps <- prune_taxa(perst_colonizing_core_ELD2_piglets,ELD2_HMA_piglets_core_ps)
perst_colonizing_core_ELD2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_ELD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_ELD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_ELD2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ELD2 HMA mice
perst_colonizing_core_ELD2_mice_ps <- prune_taxa(perst_colonizing_core_ELD2_mice,ELD2_HMA_mice_core_ps)
perst_colonizing_core_ELD2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_ELD2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_ELD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_ELD2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


##**Weekly variation in alpha diversity for each donor group among the HMA animals**##

Since the previous analyses have shown that there is a time effect when it comes to the establishment of taxa, we would like to see how the alpha diversity (as determined by the Shannon Index) varies with sampling time point for each HMA animal model within each donor.

```{r, echo=TRUE}
# Need to import modified mapping file and then remake a new phyloseq object
alpha_mapping_file <- read.table("alpha_div_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(alpha_mapping_file) <- as.character(alpha_mapping_file[,1])
View(alpha_mapping_file) # this mapping file doesn't include the four samples which had been removed due to having <10,500 reads
                         # as well as it doesn't have sample '105' (outlier - 48 hrs) from donor_3

# Need to merge with a minimally-filtered ASV table
dim(merged_seqtab_all_runs_nochim) # make sure that this is minimally filtered - has 6282 ASVs
alpha_div_master_ps <- phyloseq(otu_table(merged_seqtab_all_runs_nochim, taxa_are_rows = FALSE), sample_data(alpha_mapping_file), tax_table(taxTab), phy_tree(phylo_tree))
alpha_div_master_ps # 391 samples with 6282 ASVs
#taxa_names(alpha_div_master_ps) <- paste0("ASV_", seq(ntaxa(alpha_div_master_ps)))
#taxa_names(phy_tree(phylo_tree))

remov_kingdoms <- c('Archaea', 'Eukaryota')
alpha_div_master_ps_1 <- subset_taxa(alpha_div_master_ps, !Kingdom %in% remov_kingdoms & (Family != 'Mitochondria' | is.na(Family))) 
alpha_div_master_ps_1
alpha_div_master_ps_2 <- subset_taxa(alpha_div_master_ps_1, Kingdom == 'Bacteria') # removed 4 ASVs classified as 'NA' at the kingdom level
alpha_div_master_ps_2

Cyano_ps <- subset_taxa(alpha_div_master_ps_2, Phylum == 'Cyanobacteria') # Isolating the Cyanobacterial ASVs
remov_cyano_ps <- subset_taxa(Cyano_ps, Class != "Melainabacteria" | is.na(Class)) # Keep only Melainabacteria (also remove any cyano ASVs with 'NA' as class)
Cyano_to_remove <- taxa_names(remov_cyano_ps) 

# Filtering out the non-Melainabacterial Cyanobacterial ASVs
alltaxa = taxa_names(alpha_div_master_ps_2)
goodtaxa = alltaxa[!(alltaxa %in% Cyano_to_remove)]
alpha_div_master_all_filt_ps <- prune_taxa(goodtaxa, alpha_div_master_ps_2)
alpha_div_master_all_filt_ps # 6178 ASVs retained
```


**Performing weekly alpha diversity comparisons for Donor_1(INF2)**

```{r, echo=TRUE}

### separating out relevant donor_1 samples
alpha_div_INF2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'INF2')
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Timingofinocu != 'early'| is.na(Timingofinocu))
alpha_div_INF2_ps

## only keeping inoculae
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, SampleType != 'cecal')
alpha_div_INF2_ps

results_INF2 = estimate_richness(alpha_div_INF2_ps, measures = "Shannon")
results_INF2_df <- as.data.frame(results_INF2)
results_INF2_df$SampleID <- rownames(results_INF2_df)
View(results_INF2_df) # 'SampleID' is second column
results_INF2_df <- results_INF2_df[c(2,1)] # re-order dataframe so that 'SampleID' is first column 

INF2_alpha_div_df <- merge(results_INF2_df, alpha_mapping_file, by='SampleID')
View(INF2_alpha_div_df)
write.table(INF2_alpha_div_df, "source_data_INF2_weekly_alpha_diversity.txt", sep = "\t", row.names = FALSE, quote = FALSE) # source data for publication

p1 <- ggplot(INF2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot() + labs(fill = "host species")
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "2_days_post_inoc", "7_days_post_inoc", "14_days_post_inoc","21_days_post_inoc", "28_days_post_inoc", "35_days_post_inoc", "40_days_post_inoc")) + theme(axis.text.x=element_text(angle=45,hjust=1), legend.title = element_text(size = 10)) + xlab("Sampling Time Point") + ylab("Shannon Index") + ylim(1.0,5.0)
p2

## Performing statistical comparisons using the 'Wilcoxon test (Mann-Whitney U test)' - Based on discussion on https://github.com/joey711/phyloseq/issues/578

## For Mice
alpha_div_INF2_ps_mice <- subset_samples(alpha_div_INF2_ps, Species != "piglet")
results_INF2_mice = estimate_richness(alpha_div_INF2_ps_mice, measures = "Shannon")

d = sample_data(alpha_div_INF2_ps_mice)

donors = results_INF2_mice[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_INF2_mice[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_INF2_mice[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_INF2_mice[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_INF2_mice[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_INF2_mice[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_INF2_mice[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_INF2_mice[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # not significant (ns)
wilcox.test(donors, week_1) # ns
wilcox.test(donors, week_2) # ns
wilcox.test(donors, week_3) # ns
wilcox.test(donors, week_4) # significant
wilcox.test(donors, week_5) # ns
wilcox.test(donors, final) # ns

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the mice
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # significant 
wilcox.test(two_days, week_5) # significant 
wilcox.test(two_days, final) # ns

# let's also compare alpha diversity between each time point for the mice
wilcox.test(two_days, week_1) # ns 
wilcox.test(week_1, week_2) # ns 
wilcox.test(week_2, week_3) # ns 
wilcox.test(week_3, week_4) # significant 
wilcox.test(week_4, week_5) # ns 
wilcox.test(week_5, final) # ns 

# For Piglets
alpha_div_INF2_ps_piglets <- subset_samples(alpha_div_INF2_ps, Species != "mouse")
results_INF2_piglets = estimate_richness(alpha_div_INF2_ps_piglets, measures = "Shannon")

d = sample_data(alpha_div_INF2_ps_piglets)

donors = results_INF2_piglets[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_INF2_piglets[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_INF2_piglets[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_INF2_piglets[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_INF2_piglets[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_INF2_piglets[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_INF2_piglets[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_INF2_piglets[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # significant 
wilcox.test(donors, week_1) # significant 
wilcox.test(donors, week_2) # significant 
wilcox.test(donors, week_3) # ns
wilcox.test(donors, week_4) # significant 
wilcox.test(donors, week_5) # ns
wilcox.test(donors, final)  # significant 

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the piglets
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # ns
wilcox.test(two_days, week_5) # ns
wilcox.test(two_days, final) # ns

# let's also compare alpha diversity between each time point for the piglets
wilcox.test(two_days, week_1) # ns 
wilcox.test(week_1, week_2) # ns 
wilcox.test(week_2, week_3) # ns 
wilcox.test(week_3, week_4) # ns 
wilcox.test(week_4, week_5) # ns 
wilcox.test(week_5, final) # ns 
```


**Performing weekly alpha diversity comparisons for Donor_2(CHLD2)**

```{r, echo=TRUE}
### separating out relevant donor_2 samples
alpha_div_CHLD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'CHLD2')
alpha_div_CHLD2_ps

## only keeping inoculae
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, SampleType != 'cecal')
alpha_div_CHLD2_ps

results_CHLD2 = estimate_richness(alpha_div_CHLD2_ps, measures = "Shannon")
results_CHLD2_df <- as.data.frame(results_CHLD2)
results_CHLD2_df$SampleID <- rownames(results_CHLD2_df)
results_CHLD2_df <- results_CHLD2_df[c(2,1)]
View(results_CHLD2_df)

CHLD2_alpha_div_df <- merge(results_CHLD2_df, alpha_mapping_file, by='SampleID')
View(CHLD2_alpha_div_df)
write.table(CHLD2_alpha_div_df, "source_data_CHLD2_weekly_alpha_diversity.txt", sep = "\t", row.names = FALSE, quote = FALSE) # source data for publication

p1 <- ggplot(CHLD2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot() + labs(fill = "host species")
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "2_days_post_inoc", "7_days_post_inoc", "14_days_post_inoc","21_days_post_inoc", "28_days_post_inoc", "35_days_post_inoc", "40_days_post_inoc")) + theme(axis.text.x=element_text(angle=45,hjust=1), legend.title = element_text(size = 10)) + xlab("Sampling Time Point") + ylab("Shannon Index") + ylim(1.0,5.0)
p2

## Performing statistical comparisons using the 'Wilcoxon test (Mann-Whitney U test)'

# For Mice
alpha_div_CHLD2_ps_mice <- subset_samples(alpha_div_CHLD2_ps, Species != "piglet")
results_CHLD2_mice = estimate_richness(alpha_div_CHLD2_ps_mice, measures = "Shannon")

d = sample_data(alpha_div_CHLD2_ps_mice)

donors = results_CHLD2_mice[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_CHLD2_mice[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_CHLD2_mice[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_CHLD2_mice[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_CHLD2_mice[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_CHLD2_mice[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_CHLD2_mice[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_CHLD2_mice[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # significant (p=0.002)
wilcox.test(donors, week_1) # significant (p=0.002)
wilcox.test(donors, week_2) # significant (p=0.002)
wilcox.test(donors, week_3) # significant (p=0.002)
wilcox.test(donors, week_4) # significant (p=0.002)
wilcox.test(donors, week_5) # significant (p=0.002)
wilcox.test(donors, final) # significant (p=0.002)

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the mice
wilcox.test(two_days, week_1) # significant (p=1.083e-05)
wilcox.test(two_days, week_2) # significant (p=2.165e-05)
wilcox.test(two_days, week_3) # significant (p=1.083e-05)
wilcox.test(two_days, week_4) # significant (p=4.33e-05)
wilcox.test(two_days, week_5) # significant (p=7.578e-05)
wilcox.test(two_days, final) # significant (p=0.0004871)


# For Piglets
alpha_div_CHLD2_ps_piglets <- subset_samples(alpha_div_CHLD2_ps, Species != "mouse")
results_CHLD2_piglets = estimate_richness(alpha_div_CHLD2_ps_piglets, measures = "Shannon")

d = sample_data(alpha_div_CHLD2_ps_piglets)

donors = results_CHLD2_piglets[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_CHLD2_piglets[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_CHLD2_piglets[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_CHLD2_piglets[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_CHLD2_piglets[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_CHLD2_piglets[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_CHLD2_piglets[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_CHLD2_piglets[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # ns (p=0.057 - 'tendency')
wilcox.test(donors, week_1) # ns
wilcox.test(donors, week_2) # ns
wilcox.test(donors, week_3) # ns
wilcox.test(donors, week_4) # ns
wilcox.test(donors, week_5) # ns
wilcox.test(donors, final) # ns

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the piglets
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # ns
wilcox.test(two_days, week_5) # ns
wilcox.test(two_days, final) # ns
```


**Performing weekly alpha diversity comparisons for Donor_3(ADLT2)**

```{r, echo=TRUE}
### separating out the relevant donor_3 samples
alpha_div_ADLT2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ADLT2')
alpha_div_ADLT2_ps

## only keeping inoculae
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, SampleType != 'cecal')
alpha_div_ADLT2_ps

results_ADLT2 = estimate_richness(alpha_div_ADLT2_ps, measures = "Shannon")
View(results_ADLT2)

results_ADLT2_df <- as.data.frame(results_ADLT2)
View(results_ADLT2_df)

results_ADLT2_df$SampleID <- rownames(results_ADLT2_df)
View(results_ADLT2_df)

results_ADLT2_df <- results_ADLT2_df[c(2,1)]
View(results_ADLT2_df)

ADLT2_alpha_div_df <- merge(results_ADLT2_df, alpha_mapping_file, by='SampleID')
View(ADLT2_alpha_div_df)
write.table(ADLT2_alpha_div_df, "source_data_ADLT2_weekly_alpha_diversity.txt", sep = "\t", row.names = FALSE, quote = FALSE) # source data for publication

p1 <- ggplot(ADLT2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot() + labs(fill = "host species")
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "2_days_post_inoc", "7_days_post_inoc", "14_days_post_inoc","21_days_post_inoc", "28_days_post_inoc", "35_days_post_inoc", "40_days_post_inoc")) + theme(axis.text.x=element_text(angle=45,hjust=1), legend.title = element_text(size = 10)) + xlab("Sampling Time Point") + ylab("Shannon Index") + ylim(1.0,5.0)
p2

## Performing statistical comparisons using the 'Wilcoxon test (Mann-Whitney U test)'

# For Mice
alpha_div_ADLT2_ps_mice <- subset_samples(alpha_div_ADLT2_ps, Species != "piglet")
results_ADLT2_mice = estimate_richness(alpha_div_ADLT2_ps_mice, measures = "Shannon")

d = sample_data(alpha_div_ADLT2_ps_mice)

donors = results_ADLT2_mice[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_ADLT2_mice[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_ADLT2_mice[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_ADLT2_mice[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_ADLT2_mice[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_ADLT2_mice[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_ADLT2_mice[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_ADLT2_mice[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # significant (p=0.002)
wilcox.test(donors, week_1) # significant (p=0.002)
wilcox.test(donors, week_2) # significant (p=0.002)
wilcox.test(donors, week_3) # significant (p=0.002)
wilcox.test(donors, week_4) # significant (p=0.002)
wilcox.test(donors, week_5) # significant (p=0.002)
wilcox.test(donors, final) # significant (p=0.002)

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the mice
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # ns
wilcox.test(two_days, week_5) # ns
wilcox.test(two_days, final) # ns


# For Piglets
alpha_div_ADLT2_ps_piglets <- subset_samples(alpha_div_ADLT2_ps, Species != "mouse")
results_ADLT2_piglets = estimate_richness(alpha_div_ADLT2_ps_piglets, measures = "Shannon")

d = sample_data(alpha_div_ADLT2_ps_piglets)

donors = results_ADLT2_piglets[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_ADLT2_piglets[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_ADLT2_piglets[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_ADLT2_piglets[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_ADLT2_piglets[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_ADLT2_piglets[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_ADLT2_piglets[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_ADLT2_piglets[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # ns
wilcox.test(donors, week_1) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_2) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_3) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_4) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_5) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, final) # ns (p=0.05714 - 'tendency')

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the piglets
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # ns
wilcox.test(two_days, week_5) # ns
wilcox.test(two_days, final) # ns
```


**Performing weekly alpha diversity comparisons for Donor_4(ELD2/SNR2)**

```{r, echo=TRUE}
### Separating out the relevant donor_4 samples
alpha_div_ELD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ELD2')
alpha_div_ELD2_ps

## only keeping inoculae
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, SampleType != 'cecal')
alpha_div_ELD2_ps

results_ELD2 = estimate_richness(alpha_div_ELD2_ps, measures = "Shannon")
results_ELD2_df <- as.data.frame(results_ELD2)
results_ELD2_df$SampleID <- rownames(results_ELD2_df)
results_ELD2_df <- results_ELD2_df[c(2,1)]
View(results_ELD2_df)

ELD2_alpha_div_df <- merge(results_ELD2_df, alpha_mapping_file, by='SampleID')
View(ELD2_alpha_div_df)
write.table(ELD2_alpha_div_df, "source_data_ELD2_weekly_alpha_diversity.txt", sep = "\t", row.names = FALSE, quote = FALSE) # source data for publication

p1 <- ggplot(ELD2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot() + labs(fill = "host species")
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "2_days_post_inoc", "7_days_post_inoc", "14_days_post_inoc","21_days_post_inoc", "28_days_post_inoc", "35_days_post_inoc", "40_days_post_inoc")) + theme(axis.text.x=element_text(angle=45,hjust=1), legend.title = element_text(size = 10)) + xlab("Sampling Time Point") + ylab("Shannon Index") + ylim(1.0,5.0)
p2 

## Performing statistical comparisons using the 'Wilcoxon test (Mann-Whitney U test)'

# For Mice
alpha_div_ELD2_ps_mice <- subset_samples(alpha_div_ELD2_ps, Species != "piglet")
results_ELD2_mice = estimate_richness(alpha_div_ELD2_ps_mice, measures = "Shannon")

d = sample_data(alpha_div_ELD2_ps_mice)

donors = results_ELD2_mice[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_ELD2_mice[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_ELD2_mice[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_ELD2_mice[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_ELD2_mice[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_ELD2_mice[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_ELD2_mice[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_ELD2_mice[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # significant(p=0.002)
wilcox.test(donors, week_1) # significant(p=0.003)
wilcox.test(donors, week_2) # significant(p=0.002)
wilcox.test(donors, week_3) # significant(p=0.002)
wilcox.test(donors, week_4) # significant(p=0.002)
wilcox.test(donors, week_5) # significant(p=0.002)
wilcox.test(donors, final) # significant(p=0.002)

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the mice
wilcox.test(two_days, week_1) # significant(p=0.003)
wilcox.test(two_days, week_2) # significant(p=1.083e-05)
wilcox.test(two_days, week_3) # significant(p=0.0002057)
wilcox.test(two_days, week_4) # significant(p=1.083e-05)
wilcox.test(two_days, week_5) # significant(p=2.165e-05)
wilcox.test(two_days, final) # significant(p=0.0002057)


# For Piglets
alpha_div_ELD2_ps_piglets <- subset_samples(alpha_div_ELD2_ps, Species != "mouse")
results_ELD2_piglets = estimate_richness(alpha_div_ELD2_ps_piglets, measures = "Shannon")

d = sample_data(alpha_div_ELD2_ps_piglets)

donors = results_ELD2_piglets[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results_ELD2_piglets[d[,'Alpha_Desc'] == '2_days_post_inoc',]
week_1 = results_ELD2_piglets[d[,'Alpha_Desc'] == '7_days_post_inoc',]
week_2 = results_ELD2_piglets[d[,'Alpha_Desc'] == '14_days_post_inoc',]
week_3 = results_ELD2_piglets[d[,'Alpha_Desc'] == '21_days_post_inoc',]
week_4 = results_ELD2_piglets[d[,'Alpha_Desc'] == '28_days_post_inoc',]
week_5 = results_ELD2_piglets[d[,'Alpha_Desc'] == '35_days_post_inoc',]
final =  results_ELD2_piglets[d[,'Alpha_Desc'] == '40_days_post_inoc',]

wilcox.test(donors, two_days) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_1) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_2) # ns
wilcox.test(donors, week_3) # ns
wilcox.test(donors, week_4) # ns (p=0.05714 - 'tendency')
wilcox.test(donors, week_5) # ns
wilcox.test(donors, final) # ns

# let's also compare alpha diversity between the 'two_days' time point and the other time points for the piglets
wilcox.test(two_days, week_1) # ns
wilcox.test(two_days, week_2) # ns
wilcox.test(two_days, week_3) # ns
wilcox.test(two_days, week_4) # ns
wilcox.test(two_days, week_5) # ns
wilcox.test(two_days, final) # ns
```


##**Variation of the relative abundances of persistent colonizers with time point relative to donor**##

We also thought that it would be useful to compare the relative abundances of the persistent colonizer ASVs of the HMA animal models with the relative abundances of these ASVs in the respective donors at each time point.

**Donor_1(INF2)**

```{r, echo=TRUE}

library('DESeq2'); packageVersion('DESeq2')

pig_mouse_comp_INF2_comparison_ps # phyloseq object with late inoc piglets, donors, and mice

### for mice
pig_mouse_comp_INF2_donor_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species != 'piglet')
pig_mouse_comp_INF2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_INF2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_INF2_mice_48hrs_ps # phyloseq object with the eight 48 hrs tp samples

pig_mouse_comp_INF2_donors <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_INF2_donors

pig_mouse_comp_INF2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_48hrs_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with both donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the perst. colonizer ASVs for mice 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice_perst_col <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
View(sigtab_subset_48hrs_mice_perst_col)
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_INF2_mice,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_mice_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_mice_df)[colnames(sigtab_0.05_48hrs_perst_col_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_mice_df)

# Performing mean,SD calculations for relative abundances (as described in https://github.com/joey711/phyloseq/issues/1085)
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_INF2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% dplyr::ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_INF2_mice,]
View(ps_mean_stats_perst_col_48hrs)
ps_mean_stats_perst_col_48hrs_rounded <- ps_mean_stats_perst_col_48hrs %>% dplyr::mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_48hrs_rounded)
ps_mean_stats_perst_col_48hrs_rounded_donors <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_perst_col_48hrs_rounded_donors, "DONORS_INF2_perst_col_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_perst_col_48hrs_rounded_mice <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_perst_col_48hrs_rounded_mice, "MICE_INF2_perst_col_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_INF2_mice_1week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_INF2_mice_1week_ps # phyloseq object with the eight 1week tp samples

pig_mouse_comp_INF2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_1week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_INF2_mice,]
sigtab_0.05_1week_perst_col_mice_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_mice_df)[colnames(sigtab_0.05_1week_perst_col_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_1week_rounded_mice <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_mice, "MICE_INF2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_INF2_mice_2week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_INF2_mice_2week_ps # phyloseq object with the eight 2weeks tp samples

pig_mouse_comp_INF2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_2week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_2weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_perst_col <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% perst_colonizing_core_INF2_mice,]
sigtab_0.05_2weeks_perst_col_mice_df <- subset(sigtab_0.05_2weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_perst_col_mice_df)[colnames(sigtab_0.05_2weeks_perst_col_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_perst_col_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_2weeks_rounded_mice <- ps_mean_stats_perst_col_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_2weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_2weeks_rounded_mice, "MICE_INF2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_INF2_mice_3week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_INF2_mice_3week_ps # phyloseq object with the seven 3weeks tp samples

pig_mouse_comp_INF2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_3week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_perst_col <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% perst_colonizing_core_INF2_mice,]
sigtab_0.05_3weeks_perst_col_mice_df <- subset(sigtab_0.05_3weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_perst_col_mice_df)[colnames(sigtab_0.05_3weeks_perst_col_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_perst_col_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_3weeks_rounded_mice <- ps_mean_stats_perst_col_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_3weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_3weeks_rounded_mice, "MICE_INF2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_INF2_mice_4week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_INF2_mice_4week_ps # phyloseq object with the seven 4weeks tp samples

pig_mouse_comp_INF2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_4week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_perst_col <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% perst_colonizing_core_INF2_mice,]
sigtab_0.05_4weeks_perst_col_mice_df <- subset(sigtab_0.05_4weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_perst_col_mice_df)[colnames(sigtab_0.05_4weeks_perst_col_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_perst_col_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_4weeks_rounded_mice <- ps_mean_stats_perst_col_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_4weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_4weeks_rounded_mice, "MICE_INF2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_INF2_mice_5week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_INF2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_INF2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_5week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_perst_col <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% perst_colonizing_core_INF2_mice,]
sigtab_0.05_5weeks_perst_col_mice_df <- subset(sigtab_0.05_5weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_perst_col_mice_df)[colnames(sigtab_0.05_5weeks_perst_col_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_perst_col_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_5weeks_rounded_mice <- ps_mean_stats_perst_col_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_5weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_5weeks_rounded_mice, "MICE_INF2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_INF2_mice_40_days_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_INF2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_INF2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_40_days_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_mice_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_perst_col_INF2_mice,]
sigtab_0.05_40days_perst_col_mice_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_mice_df)[colnames(sigtab_0.05_40days_perst_col_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_INF2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_INF2_mice,]
ps_mean_stats_perst_col_40days_rounded_mice <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_mice)
write.table(ps_mean_stats_perst_col_40days_rounded_mice, "MICE_INF2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of INF2 donor
merged_sigtabs_0.05_perst_col_mice_INF2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_mice_df,sigtab_0.05_1week_perst_col_mice_df,sigtab_0.05_2weeks_perst_col_mice_df,sigtab_0.05_3weeks_perst_col_mice_df,sigtab_0.05_4weeks_perst_col_mice_df,sigtab_0.05_5weeks_perst_col_mice_df,sigtab_0.05_40days_perst_col_mice_df))
View(merged_sigtabs_0.05_perst_col_mice_INF2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_mice_INF2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_mice_INF2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_mice_INF2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
INF2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_perst_col_mice_INF2_df, non_sig_time_points >= '4') 
View(INF2_perst_rel_adund_mice_df)
perst_rel_abund_perst_col_INF2_taxa_mice <- INF2_perst_rel_adund_mice_df[,1]
perst_rel_abund_perst_col_INF2_taxa_mice

# ASVs 3975, 5999, and 6000 didn't have a single time point where their abundances were significantly different from those in the donor
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_3975","ASV_5999","ASV_6000")
perst_rel_abund_perst_col_INF2_taxa_mice <- c(perst_rel_abund_perst_col_INF2_taxa_mice,not_sig_diff_ASVs)
str(perst_rel_abund_perst_col_INF2_taxa_mice) # 7 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_perst_col_mice_INF2_df, "overall_sigdiff_ASVs_perst_col_mice_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

perst_colonizing_core_INF2_piglets
pig_mouse_comp_INF2_donor_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species != 'mouse')
pig_mouse_comp_INF2_donor_piglets_ps # has donor samples and piglet samples

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_INF2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_INF2_piglets_48hrs_ps # phyloseq object with the four 48 hrs tp samples

pig_mouse_comp_INF2_donors <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_INF2_donors

pig_mouse_comp_INF2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_48hrs_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_piglets_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_piglets_df)[colnames(sigtab_0.05_48hrs_perst_col_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_piglets_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_INF2_piglets,]
ps_mean_stats_perst_col_48hrs_rounded_piglets <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_48hrs_rounded_piglets)
write.table(ps_mean_stats_perst_col_48hrs_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_INF2_piglets_1week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_INF2_piglets_1week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_INF2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_1week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_1week_perst_col_piglets_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_piglets_df)[colnames(sigtab_0.05_1week_perst_col_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_1week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_INF2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_1week_rounded_piglets <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_INF2_piglets_2week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_INF2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_INF2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_2week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_2week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_perst_col <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_2week_perst_col_piglets_df <- subset(sigtab_0.05_2week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_perst_col_piglets_df)[colnames(sigtab_0.05_2week_perst_col_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_perst_col_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_2week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% perst_colonizing_core_INF2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_2week_rounded_piglets <- ps_mean_stats_perst_col_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_2week_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_INF2_piglets_3week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_INF2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_INF2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_3week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_perst_col <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_3week_perst_col_piglets_df <- subset(sigtab_0.05_3week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_perst_col_piglets_df)[colnames(sigtab_0.05_3week_perst_col_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_perst_col_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_3week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% perst_colonizing_core_INF2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_3week_rounded_piglets <- ps_mean_stats_perst_col_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_3week_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_INF2_piglets_4week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_INF2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_INF2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_4week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_perst_col <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_4week_perst_col_piglets_df <- subset(sigtab_0.05_4week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_perst_col_piglets_df)[colnames(sigtab_0.05_4week_perst_col_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_perst_col_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_4week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% perst_colonizing_core_INF2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_4week_rounded_piglets <- ps_mean_stats_perst_col_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_4week_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_INF2_piglets_5week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_INF2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_INF2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_5week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_5week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_perst_col <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_5week_perst_col_piglets_df <- subset(sigtab_0.05_5week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_perst_col_piglets_df)[colnames(sigtab_0.05_5week_perst_col_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_perst_col_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% perst_colonizing_core_INF2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_5week_rounded_piglets <- ps_mean_stats_perst_col_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_5week_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_INF2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_INF2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_INF2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_40_days_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_INF2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_INF2_donors_vs_piglets_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_INF2_piglets,]
sigtab_0.05_40days_perst_col_piglets_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_piglets_df)[colnames(sigtab_0.05_40days_perst_col_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_piglets_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_INF2_piglets,]
ps_mean_stats_perst_col_40days_rounded_piglets <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_piglets)
write.table(ps_mean_stats_perst_col_40days_rounded_piglets, "PIGLETS_INF2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for piglets of INF2 donor
merged_sigtabs_0.05_perst_col_piglets_INF2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_piglets_df,sigtab_0.05_1week_perst_col_piglets_df,sigtab_0.05_2week_perst_col_piglets_df,sigtab_0.05_3week_perst_col_piglets_df,sigtab_0.05_4week_perst_col_piglets_df,sigtab_0.05_5week_perst_col_piglets_df,sigtab_0.05_40days_perst_col_piglets_df))
View(merged_sigtabs_0.05_perst_col_piglets_INF2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_piglets_INF2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_piglets_INF2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_piglets_INF2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
INF2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_perst_col_piglets_INF2_df, non_sig_time_points >= '4') 
View(INF2_perst_rel_adund_piglets_df)
perst_rel_abund_perst_col_INF2_taxa_piglets <- INF2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_perst_col_INF2_taxa_piglets

# ASVs 264, 268, 3368, and 5452 didn't have a single time point where their relative abundances were significantly different from those of the donor
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_264", "ASV_268", "ASV_3368", "ASV_5452")
perst_rel_abund_perst_col_INF2_taxa_piglets <- c(perst_rel_abund_perst_col_INF2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_perst_col_INF2_taxa_piglets) 
perst_rel_abund_perst_col_INF2_taxa_piglets # 8 ASVs with 'persistent' relative abundances 

write.table(merged_sigtabs_0.05_perst_col_piglets_INF2_df, "overall_sigdiff_ASVs_perst_col_piglets_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_perst_col_perst_ASVs_INF2 <- Reduce(intersect, list(perst_rel_abund_perst_col_INF2_taxa_mice, perst_rel_abund_perst_col_INF2_taxa_piglets))
common_perst_col_perst_ASVs_INF2 # no common ASVs


## Taxonomy for the persistent colonizers with donor-like abundances

# for the HMA mice
pig_mouse_comp_INF2_perst_col_donor_like_abund_mice_ps <- prune_taxa(perst_rel_abund_perst_col_INF2_taxa_mice, pig_mouse_comp_INF2_donor_mice_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_INF2_perst_col_donor_like_abund_mice_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_INF2_perst_col_donor_like_abund_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_INF2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_INF2_mice) # check to see that everything looks good

write.table(taxa_table_perst_coloniz_donor_like_abund_INF2_mice, "taxonomy_table_for_INF2_perst_colonizers_with_donor-like_abundances_HMA_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)


# for the HMA piglets
pig_mouse_comp_INF2_perst_col_donor_like_abund_piglets_ps <- prune_taxa(perst_rel_abund_perst_col_INF2_taxa_piglets, pig_mouse_comp_INF2_donor_piglets_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_INF2_perst_col_donor_like_abund_piglets_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_INF2_perst_col_donor_like_abund_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_INF2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_INF2_piglets) # check to see that everything looks good

write.table(taxa_table_perst_coloniz_donor_like_abund_INF2_piglets, "taxonomy_table_for_INF2_perst_colonizers_with_donor-like_abundances_HMA_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_2(CHLD2)**

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_comparison_ps

### for mice
pig_mouse_comp_CHLD2_donor_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species != 'piglet')
pig_mouse_comp_CHLD2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_CHLD2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_CHLD2_mice_48hrs_ps # phyloseq object with the ten 48 hrs tp samples

pig_mouse_comp_CHLD2_donors <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_CHLD2_donors

pig_mouse_comp_CHLD2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_48hrs_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_mice_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_mice_df)[colnames(sigtab_0.05_48hrs_perst_col_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_mice_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_CHLD2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_48hrs_rounded <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_perst_col_48hrs_rounded_donors <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_perst_col_48hrs_rounded_donors, "DONORS_CHLD2_perst_col_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_perst_col_48hrs_rounded_mice <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == '48_hrs')
View(ps_mean_stats_perst_col_48hrs_rounded_mice)
write.table(ps_mean_stats_perst_col_48hrs_rounded_mice, "MICE_CHLD2_perst_col_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_CHLD2_mice_1week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_CHLD2_mice_1week_ps # phyloseq object with ten 1week tp samples

pig_mouse_comp_CHLD2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_1week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_1week_perst_col_mice_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_mice_df)[colnames(sigtab_0.05_1week_perst_col_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_1week_rounded_mice <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_mice, "MICE_CHLD2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_CHLD2_mice_2week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_CHLD2_mice_2week_ps # phyloseq object with the ten 2weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_2week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_2weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_perst_col <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_2weeks_perst_col_mice_df <- subset(sigtab_0.05_2weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_perst_col_mice_df)[colnames(sigtab_0.05_2weeks_perst_col_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_perst_col_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_2weeks_rounded_mice <- ps_mean_stats_perst_col_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_2weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_2weeks_rounded_mice, "MICE_CHLD2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_CHLD2_mice_3week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_CHLD2_mice_3week_ps # phyloseq object with the ten 3weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_3week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_perst_col <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_3weeks_perst_col_mice_df <- subset(sigtab_0.05_3weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_perst_col_mice_df)[colnames(sigtab_0.05_3weeks_perst_col_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_perst_col_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_3weeks_rounded_mice <- ps_mean_stats_perst_col_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_3weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_3weeks_rounded_mice, "MICE_CHLD2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_CHLD2_mice_4week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_CHLD2_mice_4week_ps # phyloseq object with the ten 4weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_4week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_perst_col <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_4weeks_perst_col_mice_df <- subset(sigtab_0.05_4weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_perst_col_mice_df)[colnames(sigtab_0.05_4weeks_perst_col_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_perst_col_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_4weeks_rounded_mice <- ps_mean_stats_perst_col_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_4weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_4weeks_rounded_mice, "MICE_CHLD2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_CHLD2_mice_5week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_CHLD2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_5week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_perst_col <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_5weeks_perst_col_mice_df <- subset(sigtab_0.05_5weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_perst_col_mice_df)[colnames(sigtab_0.05_5weeks_perst_col_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_perst_col_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_5weeks_rounded_mice <- ps_mean_stats_perst_col_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_5weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_5weeks_rounded_mice, "MICE_CHLD2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_CHLD2_mice_40_days_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_CHLD2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_CHLD2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_40_days_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_mice_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_CHLD2_mice,]
sigtab_0.05_40days_perst_col_mice_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_mice_df)[colnames(sigtab_0.05_40days_perst_col_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_CHLD2_mice,]
ps_mean_stats_perst_col_40days_rounded_mice <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_mice)
write.table(ps_mean_stats_perst_col_40days_rounded_mice, "MICE_CHLD2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of CHLD2 donor
merged_sigtabs_0.05_perst_col_mice_CHLD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_mice_df,sigtab_0.05_1week_perst_col_mice_df,sigtab_0.05_2weeks_perst_col_mice_df,sigtab_0.05_3weeks_perst_col_mice_df,sigtab_0.05_4weeks_perst_col_mice_df,sigtab_0.05_5weeks_perst_col_mice_df,sigtab_0.05_40days_perst_col_mice_df))
View(merged_sigtabs_0.05_perst_col_mice_CHLD2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_mice_CHLD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_mice_CHLD2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_mice_CHLD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
CHLD2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_perst_col_mice_CHLD2_df, non_sig_time_points >= '4') 
View(CHLD2_perst_rel_adund_mice_df)
perst_rel_abund_perst_col_CHLD2_taxa_mice <- CHLD2_perst_rel_adund_mice_df[,1]
perst_rel_abund_perst_col_CHLD2_taxa_mice # only 2 (ASV_139 and ASV_5251) of the 21 persistent colonizers had donor-like abundances

write.table(merged_sigtabs_0.05_perst_col_mice_CHLD2_df, "overall_sigdiff_ASVs_perst_col_mice_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_CHLD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species != 'mouse')
pig_mouse_comp_CHLD2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_CHLD2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_CHLD2_piglets_48hrs_ps # phyloseq object with the three 48 hrs tp samples

pig_mouse_comp_CHLD2_donors <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_CHLD2_donors

pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_48hrs_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_piglets_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_piglets_df)[colnames(sigtab_0.05_48hrs_perst_col_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_piglets_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_CHLD2_piglets,]
ps_mean_stats_perst_col_48hrs_rounded_piglets <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_48hrs_rounded_piglets)
write.table(ps_mean_stats_perst_col_48hrs_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_CHLD2_piglets_1week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_CHLD2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_CHLD2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_1week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_1week_perst_col_piglets_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_piglets_df)[colnames(sigtab_0.05_1week_perst_col_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_1week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_CHLD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_1week_rounded_piglets <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_CHLD2_piglets_2week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_CHLD2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_CHLD2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_2week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESeq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_2week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_perst_col <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_2week_perst_col_piglets_df <- subset(sigtab_0.05_2week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_perst_col_piglets_df)[colnames(sigtab_0.05_2week_perst_col_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_perst_col_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_2week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
View(ps_mean_stats_2week)
ps_mean_stats_perst_col_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% perst_colonizing_core_CHLD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_2week_rounded_piglets <- ps_mean_stats_perst_col_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_2week_rounded_piglets)
write.table(ps_mean_stats_perst_col_2week_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_CHLD2_piglets_3week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_CHLD2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_CHLD2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_3week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_3week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_perst_col <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_3week_perst_col_piglets_df <- subset(sigtab_0.05_3week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_perst_col_piglets_df)[colnames(sigtab_0.05_3week_perst_col_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_perst_col_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_3week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% perst_colonizing_core_CHLD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_3week_rounded_piglets <- ps_mean_stats_perst_col_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_3week_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_CHLD2_piglets_4week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_CHLD2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_CHLD2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_4week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent_colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_perst_col <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_4week_perst_col_piglets_df <- subset(sigtab_0.05_4week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_perst_col_piglets_df)[colnames(sigtab_0.05_4week_perst_col_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_perst_col_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_4week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% perst_colonizing_core_CHLD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_4week_rounded_piglets <- ps_mean_stats_perst_col_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_4week_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_CHLD2_piglets_5week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_CHLD2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_CHLD2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_5week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_perst_col <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_5week_perst_col_piglets_df <- subset(sigtab_0.05_5week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_perst_col_piglets_df)[colnames(sigtab_0.05_5week_perst_col_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_perst_col_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% perst_colonizing_core_CHLD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_5week_rounded_piglets <- ps_mean_stats_perst_col_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_5week_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_CHLD2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_CHLD2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_CHLD2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_40_days_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_CHLD2_donors_vs_piglets_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_CHLD2_piglets,]
sigtab_0.05_40days_perst_col_piglets_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_piglets_df)[colnames(sigtab_0.05_40days_perst_col_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_piglets_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_CHLD2_piglets,]
ps_mean_stats_perst_col_40days_rounded_piglets <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_piglets)
write.table(ps_mean_stats_perst_col_40days_rounded_piglets, "PIGLETS_CHLD2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


# Merging core significance tables (sigtab_0.05) for piglets of CHLD2 donor
merged_sigtabs_0.05_perst_col_piglets_CHLD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_piglets_df,sigtab_0.05_1week_perst_col_piglets_df,sigtab_0.05_2week_perst_col_piglets_df,sigtab_0.05_3week_perst_col_piglets_df,sigtab_0.05_4week_perst_col_piglets_df,sigtab_0.05_5week_perst_col_piglets_df,sigtab_0.05_40days_perst_col_piglets_df))
View(merged_sigtabs_0.05_perst_col_piglets_CHLD2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_piglets_CHLD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_piglets_CHLD2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_piglets_CHLD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
CHLD2_perst_rel_abund_piglets_df <- subset(merged_sigtabs_0.05_perst_col_piglets_CHLD2_df, non_sig_time_points >= '4') 
View(CHLD2_perst_rel_abund_piglets_df)
perst_rel_abund_perst_col_CHLD2_taxa_piglets <- CHLD2_perst_rel_abund_piglets_df[,1]
perst_rel_abund_perst_col_CHLD2_taxa_piglets

# ASVs 55 & 95 were not siginificantly different in abundance from the donor in any of the time points
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <-c( "ASV_55", "ASV_95")
perst_rel_abund_perst_col_CHLD2_taxa_piglets <- c(perst_rel_abund_perst_col_CHLD2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_perst_col_CHLD2_taxa_piglets) # 15 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_perst_col_piglets_CHLD2_df, "overall_sigdiff_ASVs_perst_col_piglets_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_perst_col_perst_ASVs_CHLD2 <- Reduce(intersect, list(perst_rel_abund_perst_col_CHLD2_taxa_mice, perst_rel_abund_perst_col_CHLD2_taxa_piglets))
common_perst_col_perst_ASVs_CHLD2 # 2 ASVs in both (ASV_139 and ASV_5251)


## Taxonomy for the persistent colonizers with donor-like abundances

# for the HMA mice
pig_mouse_comp_CHLD2_perst_col_donor_like_abund_mice_ps <- prune_taxa(perst_rel_abund_perst_col_CHLD2_taxa_mice, pig_mouse_comp_CHLD2_donor_mice_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_CHLD2_perst_col_donor_like_abund_mice_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_CHLD2_perst_col_donor_like_abund_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_CHLD2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_CHLD2_mice) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_CHLD2_mice, "taxonomy_table_for_CHLD2_perst_colonizers_with_donor-like_abundances_HMA_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for the HMA piglets
pig_mouse_comp_CHLD2_perst_col_donor_like_abund_piglets_ps <- prune_taxa(perst_rel_abund_perst_col_CHLD2_taxa_piglets, pig_mouse_comp_CHLD2_donor_piglets_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_CHLD2_perst_col_donor_like_abund_piglets_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_CHLD2_perst_col_donor_like_abund_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_CHLD2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_CHLD2_piglets) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_CHLD2_piglets, "taxonomy_table_for_CHLD2_perst_colonizers_with_donor-like_abundances_HMA_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_3(ADLT2)**

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_comparison_ps

### for mice
pig_mouse_comp_ADLT2_donor_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species != 'piglet')
pig_mouse_comp_ADLT2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ADLT2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_ADLT2_mice_48hrs_ps # phyloseq object with the ten 48 hrs tp samples

pig_mouse_comp_ADLT2_donors <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ADLT2_donors

pig_mouse_comp_ADLT2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_48hrs_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_mice_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_mice_df)[colnames(sigtab_0.05_48hrs_perst_col_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_mice_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ADLT2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_48hrs_rounded <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_perst_col_48hrs_rounded_donors <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_perst_col_48hrs_rounded_donors, "DONORS_ADLT2_perst_col_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_perst_col_48hrs_rounded_mice <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_perst_col_48hrs_rounded_mice, "MICE_ADLT2_perst_col_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ADLT2_mice_1week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_ADLT2_mice_1week_ps # phyloseq object with the ten 1week tp samples

pig_mouse_comp_ADLT2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_1week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_1week_perst_col_mice_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_mice_df)[colnames(sigtab_0.05_1week_perst_col_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_1week_rounded_mice <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_mice, "MICE_ADLT2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_ADLT2_mice_2week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_ADLT2_mice_2week_ps # phyloseq object with the ten 2weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_2week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_2weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_perst_col <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_2weeks_perst_col_mice_df <- subset(sigtab_0.05_2weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_perst_col_mice_df)[colnames(sigtab_0.05_2weeks_perst_col_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_perst_col_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_2weeks_rounded_mice <- ps_mean_stats_perst_col_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_2weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_2weeks_rounded_mice, "MICE_ADLT2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_ADLT2_mice_3week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_ADLT2_mice_3week_ps # phyloseq object with the ten 3weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_3week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_perst_col <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_3weeks_perst_col_mice_df <- subset(sigtab_0.05_3weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_perst_col_mice_df)[colnames(sigtab_0.05_3weeks_perst_col_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_perst_col_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_3weeks_rounded_mice <- ps_mean_stats_perst_col_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_3weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_3weeks_rounded_mice, "MICE_ADLT2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_ADLT2_mice_4week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_ADLT2_mice_4week_ps # phyloseq object with the ten 4weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_4week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_perst_col <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_4weeks_perst_col_mice_df <- subset(sigtab_0.05_4weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_perst_col_mice_df)[colnames(sigtab_0.05_4weeks_perst_col_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_perst_col_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_4weeks_rounded_mice <- ps_mean_stats_perst_col_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_4weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_4weeks_rounded_mice, "MICE_ADLT2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_ADLT2_mice_5week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_ADLT2_mice_5week_ps # phyloseq object with the ten 5weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_5week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_perst_col <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_5weeks_perst_col_mice_df <- subset(sigtab_0.05_5weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_perst_col_mice_df)[colnames(sigtab_0.05_5weeks_perst_col_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_perst_col_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_5weeks_rounded_mice <- ps_mean_stats_perst_col_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_5weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_5weeks_rounded_mice, "MICE_ADLT2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ADLT2_mice_40_days_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_ADLT2_mice_40_days_ps # phyloseq object with the ten 40 days tp samples

pig_mouse_comp_ADLT2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_40_days_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_mice_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_ADLT2_mice,]
sigtab_0.05_40days_perst_col_mice_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_mice_df)[colnames(sigtab_0.05_40days_perst_col_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_ADLT2_mice,]
ps_mean_stats_perst_col_40days_rounded_mice <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_mice)
write.table(ps_mean_stats_perst_col_40days_rounded_mice, "MICE_ADLT2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of ADLT2 donor
merged_sigtabs_0.05_perst_col_mice_ADLT2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_mice_df,sigtab_0.05_1week_perst_col_mice_df,sigtab_0.05_2weeks_perst_col_mice_df,sigtab_0.05_3weeks_perst_col_mice_df,sigtab_0.05_4weeks_perst_col_mice_df,sigtab_0.05_5weeks_perst_col_mice_df,sigtab_0.05_40days_perst_col_mice_df))
View(merged_sigtabs_0.05_perst_col_mice_ADLT2_df) # All 31 persistent colonizer ASVs had at least 1 time point in which their abundance was significantly different from that of the donor.

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_mice_ADLT2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_mice_ADLT2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_mice_ADLT2_df)

# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ADLT2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_perst_col_mice_ADLT2_df, non_sig_time_points >= '4') 
View(ADLT2_perst_rel_adund_mice_df)
perst_rel_abund_perst_col_ADLT2_taxa_mice <- ADLT2_perst_rel_adund_mice_df[,1]
perst_rel_abund_perst_col_ADLT2_taxa_mice # 6 perst. colonizers with donor-like abundances

write.table(merged_sigtabs_0.05_perst_col_mice_ADLT2_df, "overall_sigdiff_ASVs_perst_col_mice_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_ADLT2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species != 'mouse')
pig_mouse_comp_ADLT2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ADLT2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_ADLT2_piglets_48hrs_ps # phyloseq object with the two 48 hrs tp samples (the 3rd sample had been removed as an outlier)

pig_mouse_comp_ADLT2_donors <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ADLT2_donors

pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_48hrs_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_piglets_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_piglets_df)[colnames(sigtab_0.05_48hrs_perst_col_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_piglets_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_ADLT2_piglets,]
ps_mean_stats_perst_col_48hrs_rounded_piglets <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_48hrs_rounded_piglets)
write.table(ps_mean_stats_perst_col_48hrs_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ADLT2_piglets_1week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_ADLT2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_ADLT2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_1week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_1week_perst_col_piglets_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_piglets_df)[colnames(sigtab_0.05_1week_perst_col_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_1week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_ADLT2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_1week_rounded_piglets <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_ADLT2_piglets_2week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_ADLT2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_ADLT2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_2week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_2week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_perst_col <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_2week_perst_col_piglets_df <- subset(sigtab_0.05_2week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_perst_col_piglets_df)[colnames(sigtab_0.05_2week_perst_col_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_perst_col_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_2week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% perst_colonizing_core_ADLT2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_2week_rounded_piglets <- ps_mean_stats_perst_col_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_2week_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_ADLT2_piglets_3week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_ADLT2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_ADLT2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_3week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_perst_col <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_3week_perst_col_piglets_df <- subset(sigtab_0.05_3week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_perst_col_piglets_df)[colnames(sigtab_0.05_3week_perst_col_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_perst_col_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_3week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% perst_colonizing_core_ADLT2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_3week_rounded_piglets <- ps_mean_stats_perst_col_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_3week_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_ADLT2_piglets_4week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_ADLT2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_ADLT2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_4week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_perst_col <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_4week_perst_col_piglets_df <- subset(sigtab_0.05_4week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_perst_col_piglets_df)[colnames(sigtab_0.05_4week_perst_col_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_perst_col_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_4week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% perst_colonizing_core_ADLT2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_4week_rounded_piglets <- ps_mean_stats_perst_col_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_4week_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_ADLT2_piglets_5week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_ADLT2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_ADLT2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_5week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_perst_col <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_5week_perst_col_piglets_df <- subset(sigtab_0.05_5week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_perst_col_piglets_df)[colnames(sigtab_0.05_5week_perst_col_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_perst_col_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% perst_colonizing_core_ADLT2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_5week_rounded_piglets <- ps_mean_stats_perst_col_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_5week_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ADLT2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_ADLT2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_ADLT2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_40_days_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ADLT2_donors_vs_piglets_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_ADLT2_piglets,]
sigtab_0.05_40days_perst_col_piglets_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_piglets_df)[colnames(sigtab_0.05_40days_perst_col_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_piglets_df)

# calculating mean and SD for 40days for piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_ADLT2_piglets,]
ps_mean_stats_perst_col_40days_rounded_piglets <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_piglets)
write.table(ps_mean_stats_perst_col_40days_rounded_piglets, "PIGLETS_ADLT2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging perst_col significance tables (sigtab_0.05) for piglets of ADLT2 donor
merged_sigtabs_0.05_perst_col_piglets_ADLT2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_piglets_df,sigtab_0.05_1week_perst_col_piglets_df,sigtab_0.05_2week_perst_col_piglets_df,sigtab_0.05_3week_perst_col_piglets_df,sigtab_0.05_4week_perst_col_piglets_df,sigtab_0.05_5week_perst_col_piglets_df,sigtab_0.05_40days_perst_col_piglets_df))
View(merged_sigtabs_0.05_perst_col_piglets_ADLT2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_piglets_ADLT2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_piglets_ADLT2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_piglets_ADLT2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ADLT2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_perst_col_piglets_ADLT2_df, non_sig_time_points >= '4') 
View(ADLT2_perst_rel_adund_piglets_df)
perst_rel_abund_perst_col_ADLT2_taxa_piglets <- ADLT2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_perst_col_ADLT2_taxa_piglets

# ASVs 851 & 3643 were not significantly diffrent in abundance from the donor in any of the time points
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <-c( "ASV_851", "ASV_3643")
perst_rel_abund_perst_col_ADLT2_taxa_piglets <- c(perst_rel_abund_perst_col_ADLT2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_perst_col_ADLT2_taxa_piglets) # 40 persistent colonizer ASVs with donor-like relative abundances

write.table(merged_sigtabs_0.05_perst_col_piglets_ADLT2_df, "overall_sigdiff_ASVs_perst_col_piglets_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_perst_col_perst_ASVs_ADLT2 <- Reduce(intersect, list(perst_rel_abund_perst_col_ADLT2_taxa_mice, perst_rel_abund_perst_col_ADLT2_taxa_piglets))
common_perst_col_perst_ASVs_ADLT2 # 2 ASVs common (ASV_1036 and ASV_5256)


## Taxonomy for the persistent colonizers with donor-like abundances

# for the HMA mice
pig_mouse_comp_ADLT2_perst_col_donor_like_abund_mice_ps <- prune_taxa(perst_rel_abund_perst_col_ADLT2_taxa_mice, pig_mouse_comp_ADLT2_donor_mice_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_ADLT2_perst_col_donor_like_abund_mice_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_ADLT2_perst_col_donor_like_abund_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_ADLT2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_ADLT2_mice) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_ADLT2_mice, "taxonomy_table_for_ADLT2_perst_colonizers_with_donor-like_abundances_HMA_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for the HMA piglets
pig_mouse_comp_ADLT2_perst_col_donor_like_abund_piglets_ps <- prune_taxa(perst_rel_abund_perst_col_ADLT2_taxa_piglets, pig_mouse_comp_ADLT2_donor_piglets_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_ADLT2_perst_col_donor_like_abund_piglets_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_ADLT2_perst_col_donor_like_abund_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_ADLT2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_ADLT2_piglets) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_ADLT2_piglets, "taxonomy_table_for_ADLT2_perst_colonizers_with_donor-like_abundances_HMA_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Donor_4(ELD2/SNR2)**

```{r, echo=TRUE}
pig_mouse_comp_ELD2_comparison_ps

### for mice
pig_mouse_comp_ELD2_donor_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species != 'piglet')
pig_mouse_comp_ELD2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ELD2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_ELD2_mice_48hrs_ps # phyloseq object with the ten 48 hrs tp samples

pig_mouse_comp_ELD2_donors <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ELD2_donors

pig_mouse_comp_ELD2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_48hrs_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_mice_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_mice_df)[colnames(sigtab_0.05_48hrs_perst_col_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_mice_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ELD2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_48hrs_rounded <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_perst_col_48hrs_rounded_donors <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_perst_col_48hrs_rounded_donors, "DONORS_ELD2_perst_col_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_perst_col_48hrs_rounded_mice <- subset(ps_mean_stats_perst_col_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_perst_col_48hrs_rounded_mice, "MICE_ELD2_perst_col_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ELD2_mice_1week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_ELD2_mice_1week_ps # phyloseq object with the nine 1week tp samples

pig_mouse_comp_ELD2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_1week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_1week_perst_col_mice_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_mice_df)[colnames(sigtab_0.05_1week_perst_col_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_1week_rounded_mice <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_mice, "MICE_ELD2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_ELD2_mice_2week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_ELD2_mice_2week_ps # phyloseq object with the ten 2weeks tp samples

pig_mouse_comp_ELD2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_2week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_2weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_perst_col <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_2weeks_perst_col_mice_df <- subset(sigtab_0.05_2weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_perst_col_mice_df)[colnames(sigtab_0.05_2weeks_perst_col_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_perst_col_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_2weeks_rounded_mice <- ps_mean_stats_perst_col_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_2weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_2weeks_rounded_mice, "MICE_ELD2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_ELD2_mice_3week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_ELD2_mice_3week_ps # phyloseq object with the ten 3weeks tp samples

pig_mouse_comp_ELD2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_3week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_3weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_perst_col <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_3weeks_perst_col_mice_df <- subset(sigtab_0.05_3weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_perst_col_mice_df)[colnames(sigtab_0.05_3weeks_perst_col_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_perst_col_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_3weeks_rounded_mice <- ps_mean_stats_perst_col_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_3weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_3weeks_rounded_mice, "MICE_ELD2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_ELD2_mice_4week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_ELD2_mice_4week_ps # phyloseq object with the ten 4weeks tp samples

pig_mouse_comp_ELD2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_4week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_4weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_perst_col <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_4weeks_perst_col_mice_df <- subset(sigtab_0.05_4weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_perst_col_mice_df)[colnames(sigtab_0.05_4weeks_perst_col_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_perst_col_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_4weeks_rounded_mice <- ps_mean_stats_perst_col_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_4weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_4weeks_rounded_mice, "MICE_ELD2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_ELD2_mice_5week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_ELD2_mice_5week_ps # phyloseq object with the ten 5weeks tp samples

pig_mouse_comp_ELD2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_5week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_perst_col <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_5weeks_perst_col_mice_df <- subset(sigtab_0.05_5weeks_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_perst_col_mice_df)[colnames(sigtab_0.05_5weeks_perst_col_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_perst_col_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_5weeks_rounded_mice <- ps_mean_stats_perst_col_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_5weeks_rounded_mice)
write.table(ps_mean_stats_perst_col_5weeks_rounded_mice, "MICE_ELD2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ELD2_mice_40_days_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_ELD2_mice_40_days_ps # phyloseq object with the ten 40 days tp samples

pig_mouse_comp_ELD2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_40_days_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_mice,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_mice_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_ELD2_mice,]
sigtab_0.05_40days_perst_col_mice_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_mice_df)[colnames(sigtab_0.05_40days_perst_col_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_ELD2_mice,]
ps_mean_stats_perst_col_40days_rounded_mice <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_mice)
write.table(ps_mean_stats_perst_col_40days_rounded_mice, "MICE_ELD2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging perst_col significance tables (sigtab_0.05) for mice of ELD2 donor
merged_sigtabs_0.05_perst_col_mice_ELD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_mice_df,sigtab_0.05_1week_perst_col_mice_df,sigtab_0.05_2weeks_perst_col_mice_df,sigtab_0.05_3weeks_perst_col_mice_df,sigtab_0.05_4weeks_perst_col_mice_df,sigtab_0.05_5weeks_perst_col_mice_df,sigtab_0.05_40days_perst_col_mice_df))
View(merged_sigtabs_0.05_perst_col_mice_ELD2_df) # All 28 persistent colonizers had at least 1 time point in which their relative abundances were significantly different from that of the donor.

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_mice_ELD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_mice_ELD2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_mice_ELD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ELD2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_perst_col_mice_ELD2_df, non_sig_time_points >= '4') 
View(ELD2_perst_rel_adund_mice_df)
perst_rel_abund_perst_col_ELD2_taxa_mice <- ELD2_perst_rel_adund_mice_df[,1]
perst_rel_abund_perst_col_ELD2_taxa_mice # 7 persistent colonizer ASVs with donor-like abundances

write.table(merged_sigtabs_0.05_perst_col_mice_ELD2_df, "overall_sigdiff_ASVs_perst_col_mice_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_ELD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species != 'mouse')
pig_mouse_comp_ELD2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ELD2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_ELD2_piglets_48hrs_ps # phyloseq object with the three 48 hrs tp samples 

pig_mouse_comp_ELD2_donors <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ELD2_donors

pig_mouse_comp_ELD2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_48hrs_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_48hrs_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_perst_col <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
View(sigtab_0.05_48hrs_perst_col)
sigtab_0.05_48hrs_perst_col_piglets_df <- subset(sigtab_0.05_48hrs_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_perst_col_piglets_df)[colnames(sigtab_0.05_48hrs_perst_col_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_perst_col_piglets_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% perst_colonizing_core_ELD2_piglets,]
ps_mean_stats_perst_col_48hrs_rounded_piglets <- ps_mean_stats_perst_col_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_48hrs_rounded_piglets)
write.table(ps_mean_stats_perst_col_48hrs_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ELD2_piglets_1week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_ELD2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_ELD2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_1week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_1week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_perst_col <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_1week_perst_col_piglets_df <- subset(sigtab_0.05_1week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_perst_col_piglets_df)[colnames(sigtab_0.05_1week_perst_col_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_perst_col_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_1week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% perst_colonizing_core_ELD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_1week_rounded_piglets <- ps_mean_stats_perst_col_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_1week_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_ELD2_piglets_2week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_ELD2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_ELD2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_2week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_2week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_perst_col <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_2week_perst_col_piglets_df <- subset(sigtab_0.05_2week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_perst_col_piglets_df)[colnames(sigtab_0.05_2week_perst_col_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_perst_col_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_2week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% perst_colonizing_core_ELD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_2week_rounded_piglets <- ps_mean_stats_perst_col_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_2week_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_ELD2_piglets_3week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_ELD2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_ELD2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_3week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_3week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_perst_col <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_3week_perst_col_piglets_df <- subset(sigtab_0.05_3week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_perst_col_piglets_df)[colnames(sigtab_0.05_3week_perst_col_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_perst_col_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_3week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% perst_colonizing_core_ELD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_3week_rounded_piglets <- ps_mean_stats_perst_col_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_3week_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_ELD2_piglets_4week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_ELD2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_ELD2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_4week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_4week_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_perst_col <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_4week_perst_col_piglets_df <- subset(sigtab_0.05_4week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_perst_col_piglets_df)[colnames(sigtab_0.05_4week_perst_col_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_perst_col_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_4week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% perst_colonizing_core_ELD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_4week_rounded_piglets <- ps_mean_stats_perst_col_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_4week_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_ELD2_piglets_5week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_ELD2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_ELD2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_5week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_5weeks_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_perst_col <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_5week_perst_col_piglets_df <- subset(sigtab_0.05_5week_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_perst_col_piglets_df)[colnames(sigtab_0.05_5week_perst_col_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_perst_col_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% perst_colonizing_core_ELD2_piglets,]
View(ps_mean_stats)
ps_mean_stats_perst_col_5week_rounded_piglets <- ps_mean_stats_perst_col_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_perst_col_5week_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ELD2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_ELD2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_ELD2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_40_days_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_perst_col <- sigtab[sigtab$ASV_ID %in% perst_colonizing_core_ELD2_piglets,] # Now let's separate out the persistent colonizer ASVs 
View(sigtab_perst_col)
write.table(sigtab_perst_col, 'DESeq2_results_ELD2_donors_vs_piglets_40_days_persistent_colonizer_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_perst_col, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_perst_col <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% perst_colonizing_core_ELD2_piglets,]
sigtab_0.05_40days_perst_col_piglets_df <- subset(sigtab_0.05_40days_perst_col, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_perst_col_piglets_df)[colnames(sigtab_0.05_40days_perst_col_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_perst_col_piglets_df)

# calculating mean and SD for 40days for piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_perst_col_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% perst_colonizing_core_ELD2_piglets,]
ps_mean_stats_perst_col_40days_rounded_piglets <- ps_mean_stats_perst_col_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_perst_col_40days_rounded_piglets)
write.table(ps_mean_stats_perst_col_40days_rounded_piglets, "PIGLETS_ELD2_perst_col_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


# Merging perst_col significance tables (sigtab_0.05) for piglets of ELD2 donor
merged_sigtabs_0.05_perst_col_piglets_ELD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_perst_col_piglets_df,sigtab_0.05_1week_perst_col_piglets_df,sigtab_0.05_2week_perst_col_piglets_df,sigtab_0.05_3week_perst_col_piglets_df,sigtab_0.05_4week_perst_col_piglets_df,sigtab_0.05_5week_perst_col_piglets_df,sigtab_0.05_40days_perst_col_piglets_df))
View(merged_sigtabs_0.05_perst_col_piglets_ELD2_df) # All persistent colonizer ASVs had at least 1 time point in which their abundances were significantly different from those in the donor.

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_perst_col_piglets_ELD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_perst_col_piglets_ELD2_df),1,sum)
View(merged_sigtabs_0.05_perst_col_piglets_ELD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ELD2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_perst_col_piglets_ELD2_df, non_sig_time_points >= '4') 
View(ELD2_perst_rel_adund_piglets_df)
perst_rel_abund_perst_col_ELD2_taxa_piglets <- ELD2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_perst_col_ELD2_taxa_piglets # 16 persistent colonizer ASVs with donor-like relative abundances

write.table(merged_sigtabs_0.05_perst_col_piglets_ELD2_df, "overall_sigdiff_ASVs_perst_col_piglets_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_perst_col_perst_ASVs_ELD2 <- Reduce(intersect, list(perst_rel_abund_perst_col_ELD2_taxa_mice, perst_rel_abund_perst_col_ELD2_taxa_piglets))
common_perst_col_perst_ASVs_ELD2 # 2 ASVs common (ASV_3368 and ASV_3942)

## Taxonomy for the persistent colonizers with donor-like abundances

# for the HMA mice
pig_mouse_comp_ELD2_perst_col_donor_like_abund_mice_ps <- prune_taxa(perst_rel_abund_perst_col_ELD2_taxa_mice, pig_mouse_comp_ELD2_donor_mice_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_ELD2_perst_col_donor_like_abund_mice_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_ELD2_perst_col_donor_like_abund_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_ELD2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_ELD2_mice) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_ELD2_mice, "taxonomy_table_for_ELD2_perst_colonizers_with_donor-like_abundances_HMA_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for the HMA piglets
pig_mouse_comp_ELD2_perst_col_donor_like_abund_piglets_ps <- prune_taxa(perst_rel_abund_perst_col_ELD2_taxa_piglets, pig_mouse_comp_ELD2_donor_piglets_ps)
taxa_cols <- colnames(tax_table(pig_mouse_comp_ELD2_perst_col_donor_like_abund_piglets_ps))
taxa_table <- as.data.frame(tax_table(pig_mouse_comp_ELD2_perst_col_donor_like_abund_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_perst_coloniz_donor_like_abund_ELD2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_perst_coloniz_donor_like_abund_ELD2_piglets) # check to see that everything looks good
write.table(taxa_table_perst_coloniz_donor_like_abund_ELD2_piglets, "taxonomy_table_for_ELD2_perst_colonizers_with_donor-like_abundances_HMA_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


##**Common core ASVs across donors**##

We wanted to look at which ASVs were core ASVs across the four different donors :

```{r, echo=TRUE}
str(core_INF2_taxa)
str(core_CHLD2_taxa)
str(core_ADLT2_taxa)
str(core_ELD2_taxa)
common_core_ASVs <- Reduce(intersect, list(core_INF2_taxa,core_CHLD2_taxa,core_ADLT2_taxa,core_ELD2_taxa))
common_core_ASVs # only ASV 353 is commonly found in all the cores!

# Let's try looking at a common core omitting the INF2 donor core
common_core_ASVs_no_INF2 <- Reduce(intersect, list(core_CHLD2_taxa,core_ADLT2_taxa,core_ELD2_taxa))
common_core_ASVs_no_INF2 # there are 27 shared core ASVs among CHLD2, ADLT2, and ELD2 cores

# taxonomy of these 27 'common core' ASVs
human_donor_inoculae_common_core_ps <- prune_taxa(common_core_ASVs_no_INF2, human_donor_inoculae_ps)
tax<-as(tax_table(human_donor_inoculae_common_core_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "taxonomy_of_27_common_core_ASVs.txt", quote=FALSE, col.names=FALSE, sep="\t")


## Let's see how much of each donor core (Donor_2, Donor_3, and Donor_4) is accounted for by these 27 core ASVs

# Donor_2(CHLD2)
human_donors_CHLD2_inocula_core_ps
overall_common_core_CHLD2_inocula_ps <- prune_taxa(common_core_ASVs_no_INF2, human_donors_CHLD2_inocula_core_ps)

common_overall_core_CHLD2_mice <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_CHLD2_taxa_colonizing_mice)) # 9 ASVs found in mice
common_overall_core_CHLD2_piglets <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_CHLD2_taxa_colonizing_piglets)) # 24 ASVs found in piglets

persistent_coloniz_core_27_CHLD2_mice <- Reduce(intersect, list(perst_colonizing_core_CHLD2_mice,common_core_ASVs_no_INF2)) # 5 ASVs persistently colonizing the mice
persistent_coloniz_core_27_CHLD2_piglets <- Reduce(intersect, list(perst_colonizing_core_CHLD2_piglets,common_core_ASVs_no_INF2)) # 16 ASVs persistently colonizing the piglets


# Donor_3(ADLT2)
human_donors_ADLT2_inocula_core_ps
overall_common_core_ADLT2_inocula_ps <- prune_taxa(common_core_ASVs_no_INF2, human_donors_ADLT2_inocula_core_ps)

common_overall_core_ADLT2_mice <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_ADLT2_taxa_colonizing_mice)) # 15 ASVs found in mice
common_overall_core_ADLT2_piglets <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_ADLT2_taxa_colonizing_piglets)) # 25 ASVs found in piglets

persistent_coloniz_core_27_ADLT2_mice <- Reduce(intersect, list(perst_colonizing_core_ADLT2_mice,common_core_ASVs_no_INF2)) # 6 ASVs persistently colonizing the mice
persistent_coloniz_core_27_ADLT2_piglets <- Reduce(intersect, list(perst_colonizing_core_ADLT2_piglets,common_core_ASVs_no_INF2)) # 20 ASVs persistently colonizing the piglets


# Donor_4(SNR2/ELD2)
human_donors_ELD2_inocula_core_ps
overall_common_core_ELD2_inocula_ps <- prune_taxa(common_core_ASVs_no_INF2, human_donors_ELD2_inocula_core_ps)

common_overall_core_ELD2_mice <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_ELD2_taxa_colonizing_mice)) # 10 ASVs found in mice
common_overall_core_ELD2_piglets <- Reduce(intersect, list(common_core_ASVs_no_INF2,core_ELD2_taxa_colonizing_piglets)) # 19 ASVs found in piglets

persistent_coloniz_core_27_ELD2_mice <- Reduce(intersect, list(perst_colonizing_core_ELD2_mice,common_core_ASVs_no_INF2)) # 5 ASVs persistently colonizing the mice
persistent_coloniz_core_27_ELD2_piglets <- Reduce(intersect, list(perst_colonizing_core_ELD2_piglets,common_core_ASVs_no_INF2)) # 6 ASVs persistently colonizing the piglets
```


##**core ASVs not colonizing animal models**##

```{r, echo=TRUE}

## Donor_1(INF2)
core_INF2_taxa
core_INF2_taxa_colonizing_mice
core_INF2_taxa_colonizing_piglets

# core INF2 ASVs not colonizing mice (using 'setdiff' function)
core_INF2_taxa_not_colonizing_mice <- setdiff(core_INF2_taxa, core_INF2_taxa_colonizing_mice)
core_INF2_taxa_not_colonizing_mice # 5 ASVs 

# core INF2 ASVs not colonizing piglets
core_INF2_taxa_not_colonizing_piglets <- setdiff(core_INF2_taxa, core_INF2_taxa_colonizing_piglets)
core_INF2_taxa_not_colonizing_piglets # 6 ASVs

# core INF2 ASVs not colonizing mice but colonizing pigs
core_INF2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_INF2_taxa_colonizing_piglets,core_INF2_taxa_not_colonizing_mice))
core_INF2_taxa_no_mice_yes_pigs # only 1 (ASV_51)

# core INF2 ASVs not colonizing piglets but colonizing mice
core_INF2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_INF2_taxa_colonizing_mice,core_INF2_taxa_not_colonizing_piglets))
core_INF2_taxa_no_pigs_yes_mice # 2 ASVs 

# core INF2 ASVs not colonizing either animal model
core_INF2_taxa_not_colonizing <- Reduce(intersect, list(core_INF2_taxa_not_colonizing_mice,core_INF2_taxa_not_colonizing_piglets))
core_INF2_taxa_not_colonizing # 4 ASVs

# Let's get the taxonomic classifications for these ASVs which didn't colonize either animal model
pig_mouse_comp_INF2_comparison_ps
INF2_not_colonizing_ps <- prune_taxa(core_INF2_taxa_not_colonizing,pig_mouse_comp_INF2_comparison_ps)
INF2_not_colonizing_ps

tax<-as(tax_table(INF2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax_INF2.txt", quote=FALSE, col.names=FALSE, sep="\t")


## Donor_2(CHLD2) 
core_CHLD2_taxa
core_CHLD2_taxa_colonizing_mice
core_CHLD2_taxa_colonizing_piglets

# core CHLD2 ASVs not colonizing mice (using 'setdiff' function)
core_CHLD2_taxa_not_colonizing_mice <- setdiff(core_CHLD2_taxa, core_CHLD2_taxa_colonizing_mice)
core_CHLD2_taxa_not_colonizing_mice # 39 ASVs 

# core CHLD2 ASVs not colonizing piglets
core_CHLD2_taxa_not_colonizing_piglets <- setdiff(core_CHLD2_taxa, core_CHLD2_taxa_colonizing_piglets)
core_CHLD2_taxa_not_colonizing_piglets # 11 ASVs

# core CHLD2 ASVs not colonizing mice but colonizing pigs
core_CHLD2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_CHLD2_taxa_colonizing_piglets,core_CHLD2_taxa_not_colonizing_mice))
core_CHLD2_taxa_no_mice_yes_pigs # 28 ASVs

# core CHLD2 ASVs not colonizing piglets but colonizing mice
core_CHLD2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_CHLD2_taxa_colonizing_mice,core_CHLD2_taxa_not_colonizing_piglets))
core_CHLD2_taxa_no_pigs_yes_mice # 0 ASVs

# core CHLD2 ASVs not colonizing either animal model
core_CHLD2_taxa_not_colonizing <- Reduce(intersect, list(core_CHLD2_taxa_not_colonizing_mice,core_CHLD2_taxa_not_colonizing_piglets))  
core_CHLD2_taxa_not_colonizing # 11 ASVs

# Let's get the taxonomic classifications for these ASVs which failed to colonize both animal models
pig_mouse_comp_CHLD2_comparison_ps
CHLD2_not_colonizing_ps <- prune_taxa(core_CHLD2_taxa_not_colonizing,pig_mouse_comp_CHLD2_comparison_ps)
CHLD2_not_colonizing_ps

tax<-as(tax_table(CHLD2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax_CHLD2.txt", quote=FALSE, col.names=FALSE, sep="\t")


## Donor_3(ADLT2) 
core_ADLT2_taxa
core_ADLT2_taxa_colonizing_mice
core_ADLT2_taxa_colonizing_piglets

# core ADLT2 ASVs not colonizing mice (using 'setdiff' function)
core_ADLT2_taxa_not_colonizing_mice <- setdiff(core_ADLT2_taxa, core_ADLT2_taxa_colonizing_mice)
core_ADLT2_taxa_not_colonizing_mice # 74 ASVs 

# core ADLT2 ASVs not colonizing piglets
core_ADLT2_taxa_not_colonizing_piglets <- setdiff(core_ADLT2_taxa, core_ADLT2_taxa_colonizing_piglets)
core_ADLT2_taxa_not_colonizing_piglets # 34 ASVs

# core ADLT2 ASVs not colonizing mice but colonizing pigs
core_ADLT2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_ADLT2_taxa_colonizing_piglets,core_ADLT2_taxa_not_colonizing_mice))
core_ADLT2_taxa_no_mice_yes_pigs # 43 ASVs

# core CHLD2 ASVs not colonizing piglets but colonizing mice
core_ADLT2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_ADLT2_taxa_colonizing_mice,core_ADLT2_taxa_not_colonizing_piglets))
core_ADLT2_taxa_no_pigs_yes_mice # 3 ASVs
                   
# core ADLT2 ASVs not colonizing either animal model
core_ADLT2_taxa_not_colonizing <- Reduce(intersect, list(core_ADLT2_taxa_not_colonizing_mice,core_ADLT2_taxa_not_colonizing_piglets))
core_ADLT2_taxa_not_colonizing # 31 core ASVs

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_ADLT2_comparison_ps
ADLT2_not_colonizing_ps <- prune_taxa(core_ADLT2_taxa_not_colonizing,pig_mouse_comp_ADLT2_comparison_ps)
ADLT2_not_colonizing_ps

tax<-as(tax_table(ADLT2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax_ADLT2.txt", quote=FALSE, col.names=FALSE, sep="\t")


## Donor_4(ELD2)
core_ELD2_taxa
core_ELD2_taxa_colonizing_mice
core_ELD2_taxa_colonizing_piglets

# core ELD2 ASVs not colonizing mice (using 'setdiff' function)
core_ELD2_taxa_not_colonizing_mice <- setdiff(core_ELD2_taxa, core_ELD2_taxa_colonizing_mice)
core_ELD2_taxa_not_colonizing_mice # 79 ASVs 

# core ELD2 ASVs not colonizing piglets
core_ELD2_taxa_not_colonizing_piglets <- setdiff(core_ELD2_taxa, core_ELD2_taxa_colonizing_piglets)
core_ELD2_taxa_not_colonizing_piglets # 42 ASVs

# core ELD2 ASVs not colonizing mice but colonizing pigs
core_ELD2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_ELD2_taxa_colonizing_piglets,core_ELD2_taxa_not_colonizing_mice))
core_ELD2_taxa_no_mice_yes_pigs # 41 ASVs

# core ELD2 ASVs not colonizing piglets but colonizing mice
core_ELD2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_ELD2_taxa_colonizing_mice,core_ELD2_taxa_not_colonizing_piglets))
core_ELD2_taxa_no_pigs_yes_mice # 4 ASVs

# core ELD2 ASVs not colonizing either animal model
core_ELD2_taxa_not_colonizing <- Reduce(intersect, list(core_ELD2_taxa_not_colonizing_mice,core_ELD2_taxa_not_colonizing_piglets))
core_ELD2_taxa_not_colonizing # 38 ASVs not colonizing either animal model

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_ELD2_comparison_ps
ELD2_not_colonizing_ps <- prune_taxa(core_ELD2_taxa_not_colonizing,pig_mouse_comp_ELD2_comparison_ps)
ELD2_not_colonizing_ps

tax<-as(tax_table(ELD2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax_ELD2.txt", quote=FALSE, col.names=FALSE, sep="\t")
```



##**Calculating mean and standard deviation for core ASVs in each donor among the three different host species and assigning taxonomy**## 


**Donor_1(INF2)**

```{r, echo=TRUE}

### for INF2 donor
human_donors_INF2_inocula_relevant_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_INF2_inocula_table_PHYLUM <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Phylum"], exclude = NULL)
human_donors_INF2_PHYLUM_table <- as.data.frame(human_donors_INF2_inocula_table_PHYLUM)
colnames(human_donors_INF2_PHYLUM_table)[colnames(human_donors_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_INF2_PHYLUM_table)[colnames(human_donors_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_PHYLUM_table)

# at FAMILY level
human_donors_INF2_inocula_table_FAMILY <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Family"], exclude = NULL)
View(human_donors_INF2_inocula_table_FAMILY)
human_donors_INF2_FAMILY_table <- as.data.frame(human_donors_INF2_inocula_table_FAMILY)
colnames(human_donors_INF2_FAMILY_table)[colnames(human_donors_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_INF2_FAMILY_table)[colnames(human_donors_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_FAMILY_table)

# at GENUS level
human_donors_INF2_inocula_table_GENUS <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_INF2_inocula_table_GENUS)
human_donors_INF2_GENUS_table <- as.data.frame(human_donors_INF2_inocula_table_GENUS)
colnames(human_donors_INF2_GENUS_table)[colnames(human_donors_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_INF2_GENUS_table)[colnames(human_donors_INF2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_GENUS_table)

pig_mouse_comp_INF2_comparison_ps # cecal samples and 'early' inoculation piglet samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_INF2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_INF2_comparison_ps_core.prop <- prune_taxa(core_INF2_taxa, pig_mouse_comp_INF2_comparison_ps.prop)
pig_mouse_comp_INF2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_INF2_core_ASVs <- psmelt(pig_mouse_comp_INF2_comparison_ps_core.prop) %>% as_tibble
tb_INF2_core_ASVs_0 <- tb_INF2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_0_rounded <- tb_INF2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_0_rounded)

df_donor_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_INF2)[colnames(df_mouse_INF2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_INF2)[colnames(df_piglet_INF2) == "OTU"] <- "ASV_ID"
View(df_donor_INF2)
View(df_mouse_INF2)
View(df_piglet_INF2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_mouse_INF2,df_piglet_INF2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = NA, row.names = TRUE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_INF2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Phylum")

tb_INF2_core_ASVs_PHYLUM_df <- psmelt(tb_INF2_core_ASVs_PHYLUM) %>% as_tibble
tb_INF2_core_ASVs_PHYLUM_df_0 <- tb_INF2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_PHYLUM_df_0_rounded <- tb_INF2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_PHYLUM)[colnames(df_mouse_INF2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_PHYLUM)[colnames(df_mouse_INF2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_PHYLUM)[colnames(df_piglet_INF2_PHYLUM) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_INF2_PHYLUM)[colnames(df_piglet_INF2_PHYLUM) == "SD"] <- "SD_piglets"

View(df_donor_INF2_PHYLUM)
df_donor_INF2_PHYLUM$sample_Species <- NULL

INF2_donor_PHYLUM_df <- merge(human_donors_INF2_PHYLUM_table,df_donor_INF2_PHYLUM, by="Phylum")
View(INF2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_INF2_mice_ps <- prune_taxa(core_INF2_taxa_colonizing_mice, INF2_HMA_mice_ps)
colonizing_core_taxa_INF2_mice_ps

# at PHYLUM level
INF2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_INF2_PHYLUM_table <- as.data.frame(INF2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_PHYLUM_table)

df_mouse_INF2_PHYLUM$sample_Species <- NULL
INF2_mouse_PHYLUM_df <- merge(core_taxa_mice_INF2_PHYLUM_table,df_mouse_INF2_PHYLUM, by="Phylum")
View(INF2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_INF2_piglets_ps <- prune_taxa(core_INF2_taxa_colonizing_piglets, INF2_HMA_piglets_ps)
colonizing_core_taxa_INF2_piglets_ps

# at PHYLUM level
INF2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_INF2_PHYLUM_table <- as.data.frame(INF2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_PHYLUM_table)

df_piglet_INF2_PHYLUM$sample_Species <- NULL
INF2_piglet_PHYLUM_df <- merge(core_taxa_piglets_INF2_PHYLUM_table,df_piglet_INF2_PHYLUM, by="Phylum")
View(INF2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_INF2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_PHYLUM_df,INF2_mouse_PHYLUM_df,INF2_piglet_PHYLUM_df))
View(merged_INF2_PHYLUM)

write.table(merged_INF2_PHYLUM, "merged_core_PHYLUM_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_INF2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Family")

tb_INF2_core_ASVs_FAMILY_df <- psmelt(tb_INF2_core_ASVs_FAMILY) %>% as_tibble
tb_INF2_core_ASVs_FAMILY_df_0 <- tb_INF2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_FAMILY_df_0_rounded <- tb_INF2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_FAMILY_df_0_rounded)

df_donor_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_FAMILY)[colnames(df_mouse_INF2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_FAMILY)[colnames(df_mouse_INF2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_FAMILY)[colnames(df_piglet_INF2_FAMILY) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_INF2_FAMILY)[colnames(df_piglet_INF2_FAMILY) == "SD"] <- "SD_piglets"


View(df_donor_INF2_FAMILY)
df_donor_INF2_FAMILY$sample_Species <- NULL

INF2_donor_FAMILY_df <- merge(human_donors_INF2_FAMILY_table,df_donor_INF2_FAMILY, by="Family")
View(INF2_donor_FAMILY_df)


## for mice

# at FAMILY level
INF2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_INF2_FAMILY_table <- as.data.frame(INF2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_FAMILY_table)

df_mouse_INF2_FAMILY$sample_Species <- NULL
INF2_mouse_FAMILY_df <- merge(core_taxa_mice_INF2_FAMILY_table,df_mouse_INF2_FAMILY, by="Family")
View(INF2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
INF2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_INF2_FAMILY_table <- as.data.frame(INF2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_FAMILY_table)

df_piglet_INF2_FAMILY$sample_Species <- NULL
INF2_piglet_FAMILY_df <- merge(core_taxa_piglets_INF2_FAMILY_table,df_piglet_INF2_FAMILY, by="Family")
View(INF2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_INF2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_FAMILY_df,INF2_mouse_FAMILY_df,INF2_piglet_FAMILY_df))
View(merged_INF2_FAMILY)

write.table(merged_INF2_FAMILY, "merged_core_FAMILY_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_INF2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Genus", NArm = FALSE) # NArm = FALSE so that unclassified genus level ASVs are not removed
tb_INF2_core_ASVs_GENUS_df <- psmelt(tb_INF2_core_ASVs_GENUS) %>% as_tibble
View(tb_INF2_core_ASVs_GENUS_df)
tb_INF2_core_ASVs_GENUS_df_0 <- tb_INF2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_GENUS_df_0_rounded <- tb_INF2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_GENUS_df_0_rounded)

df_donor_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_GENUS)[colnames(df_mouse_INF2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_GENUS)[colnames(df_mouse_INF2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_GENUS)[colnames(df_piglet_INF2_GENUS) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_INF2_GENUS)[colnames(df_piglet_INF2_GENUS) == "SD"] <- "SD_piglets"

View(df_donor_INF2_GENUS)
df_donor_INF2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
INF2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_INF2_GENUS_table,df_donor_INF2_GENUS))
View(INF2_donor_GENUS_df)


## for mice

# at GENUS level
INF2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_INF2_GENUS_table <- as.data.frame(INF2_core_taxa_mouse_table_GENUS)
View(INF2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_GENUS_table)

df_mouse_INF2_GENUS$sample_Species <- NULL
View(df_mouse_INF2_GENUS)
INF2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_INF2_GENUS_table,df_mouse_INF2_GENUS))
View(INF2_mouse_GENUS_df)

## for piglets

# at GENUS level
INF2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_INF2_GENUS_table <- as.data.frame(INF2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_GENUS_table)

df_piglet_INF2_GENUS$sample_Species <- NULL
INF2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_INF2_GENUS_table,df_piglet_INF2_GENUS))
View(INF2_piglet_GENUS_df)


## merged the data frames from all three species
merged_INF2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_GENUS_df,INF2_mouse_GENUS_df,INF2_piglet_GENUS_df))
View(merged_INF2_GENUS)

write.table(merged_INF2_GENUS, "merged_core_GENUS_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_2(CHLD2)**

```{r, echo=TRUE}

### for CHLD2 donor
human_donors_CHLD2_inocula_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_CHLD2_inocula_table_PHYLUM <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_CHLD2_PHYLUM_table <- as.data.frame(human_donors_CHLD2_inocula_table_PHYLUM)
colnames(human_donors_CHLD2_PHYLUM_table)[colnames(human_donors_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_CHLD2_PHYLUM_table)[colnames(human_donors_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_PHYLUM_table)

# at FAMILY level
human_donors_CHLD2_inocula_table_FAMILY <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_CHLD2_inocula_table_FAMILY)
human_donors_CHLD2_FAMILY_table <- as.data.frame(human_donors_CHLD2_inocula_table_FAMILY)
colnames(human_donors_CHLD2_FAMILY_table)[colnames(human_donors_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_CHLD2_FAMILY_table)[colnames(human_donors_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_FAMILY_table)

# at GENUS level
human_donors_CHLD2_inocula_table_GENUS <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_CHLD2_inocula_table_GENUS)
human_donors_CHLD2_GENUS_table <- as.data.frame(human_donors_CHLD2_inocula_table_GENUS)
colnames(human_donors_CHLD2_GENUS_table)[colnames(human_donors_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_CHLD2_GENUS_table)[colnames(human_donors_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_GENUS_table)

pig_mouse_comp_CHLD2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_CHLD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_CHLD2_comparison_ps_core.prop <- prune_taxa(core_CHLD2_taxa, pig_mouse_comp_CHLD2_comparison_ps.prop)
pig_mouse_comp_CHLD2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_CHLD2_core_ASVs <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_core.prop) %>% as_tibble
tb_CHLD2_core_ASVs_0 <- tb_CHLD2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_0_rounded <- tb_CHLD2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_CHLD2)[colnames(df_mouse_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_CHLD2)[colnames(df_piglet_CHLD2) == "OTU"] <- "ASV_ID"
View(df_donor_CHLD2)
View(df_mouse_CHLD2)
View(df_piglet_CHLD2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_mouse_CHLD2,df_piglet_CHLD2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_CHLD2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Phylum")

tb_CHLD2_core_ASVs_PHYLUM_df <- psmelt(tb_CHLD2_core_ASVs_PHYLUM) %>% as_tibble
tb_CHLD2_core_ASVs_PHYLUM_df_0 <- tb_CHLD2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded <- tb_CHLD2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_PHYLUM)[colnames(df_mouse_CHLD2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_PHYLUM)[colnames(df_mouse_CHLD2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_PHYLUM)[colnames(df_piglet_CHLD2_PHYLUM) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_CHLD2_PHYLUM)[colnames(df_piglet_CHLD2_PHYLUM) == "SD"] <- "SD_piglets"

df_donor_CHLD2_PHYLUM$sample_Species <- NULL

CHLD2_donor_PHYLUM_df <- merge(human_donors_CHLD2_PHYLUM_table,df_donor_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_CHLD2_mice_ps <- prune_taxa(core_CHLD2_taxa_colonizing_mice, CHLD2_HMA_mice_ps)
colonizing_core_taxa_CHLD2_mice_ps # 37 ASVs

# at PHYLUM level
CHLD2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_PHYLUM_table)

df_mouse_CHLD2_PHYLUM$sample_Species <- NULL
CHLD2_mouse_PHYLUM_df <- merge(core_taxa_mice_CHLD2_PHYLUM_table,df_mouse_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_CHLD2_piglets_ps <- prune_taxa(core_CHLD2_taxa_colonizing_piglets, CHLD2_HMA_piglets_ps)
colonizing_core_taxa_CHLD2_piglets_ps # 65 ASVs

# at PHYLUM level
CHLD2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_PHYLUM_table)

df_piglet_CHLD2_PHYLUM$sample_Species <- NULL
CHLD2_piglet_PHYLUM_df <- merge(core_taxa_piglets_CHLD2_PHYLUM_table,df_piglet_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_CHLD2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_PHYLUM_df,CHLD2_mouse_PHYLUM_df,CHLD2_piglet_PHYLUM_df))
View(merged_CHLD2_PHYLUM)

write.table(merged_CHLD2_PHYLUM, "merged_core_PHYLUM_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_CHLD2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_CHLD2_core_ASVs_FAMILY_df <- psmelt(tb_CHLD2_core_ASVs_FAMILY) %>% as_tibble
tb_CHLD2_core_ASVs_FAMILY_df_0 <- tb_CHLD2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_FAMILY_df_0_rounded <- tb_CHLD2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded)

df_donor_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_FAMILY)[colnames(df_mouse_CHLD2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_FAMILY)[colnames(df_mouse_CHLD2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_FAMILY)[colnames(df_piglet_CHLD2_FAMILY) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_CHLD2_FAMILY)[colnames(df_piglet_CHLD2_FAMILY) == "SD"] <- "SD_piglets"


View(df_donor_CHLD2_FAMILY)
df_donor_CHLD2_FAMILY$sample_Species <- NULL

CHLD2_donor_FAMILY_df <- merge(human_donors_CHLD2_FAMILY_table,df_donor_CHLD2_FAMILY, by="Family")
View(CHLD2_donor_FAMILY_df)


## for mice

# at FAMILY level
CHLD2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_CHLD2_FAMILY_table <- as.data.frame(CHLD2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_FAMILY_table)

df_mouse_CHLD2_FAMILY$sample_Species <- NULL
CHLD2_mouse_FAMILY_df <- merge(core_taxa_mice_CHLD2_FAMILY_table,df_mouse_CHLD2_FAMILY, by="Family")
View(CHLD2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
CHLD2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_CHLD2_FAMILY_table <- as.data.frame(CHLD2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_FAMILY_table)

df_piglet_CHLD2_FAMILY$sample_Species <- NULL
CHLD2_piglet_FAMILY_df <- merge(core_taxa_piglets_CHLD2_FAMILY_table,df_piglet_CHLD2_FAMILY, by="Family")
View(CHLD2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_CHLD2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_FAMILY_df,CHLD2_mouse_FAMILY_df,CHLD2_piglet_FAMILY_df))
View(merged_CHLD2_FAMILY)

write.table(merged_CHLD2_FAMILY, "merged_core_FAMILY_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_CHLD2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_CHLD2_core_ASVs_GENUS_df <- psmelt(tb_CHLD2_core_ASVs_GENUS) %>% as_tibble
tb_CHLD2_core_ASVs_GENUS_df_0 <- tb_CHLD2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_GENUS_df_0_rounded <- tb_CHLD2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_GENUS_df_0_rounded)

df_donor_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_GENUS)[colnames(df_mouse_CHLD2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_GENUS)[colnames(df_mouse_CHLD2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_GENUS)[colnames(df_piglet_CHLD2_GENUS) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_CHLD2_GENUS)[colnames(df_piglet_CHLD2_GENUS) == "SD"] <- "SD_piglets"

View(df_donor_CHLD2_GENUS)
df_donor_CHLD2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
CHLD2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_CHLD2_GENUS_table,df_donor_CHLD2_GENUS))
View(CHLD2_donor_GENUS_df)


## for mice

# at GENUS level
CHLD2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_CHLD2_GENUS_table <- as.data.frame(CHLD2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_GENUS_table)

df_mouse_CHLD2_GENUS$sample_Species <- NULL
View(df_mouse_CHLD2_GENUS)
CHLD2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_CHLD2_GENUS_table,df_mouse_CHLD2_GENUS))
View(CHLD2_mouse_GENUS_df)

## for piglets

# at GENUS level
CHLD2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_CHLD2_GENUS_table <- as.data.frame(CHLD2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_GENUS_table)

df_piglet_CHLD2_GENUS$sample_Species <- NULL
CHLD2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_CHLD2_GENUS_table,df_piglet_CHLD2_GENUS))
View(CHLD2_piglet_GENUS_df)


## merged the data frames from all three species
merged_CHLD2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_GENUS_df,CHLD2_mouse_GENUS_df,CHLD2_piglet_GENUS_df))
View(merged_CHLD2_GENUS)

write.table(merged_CHLD2_GENUS, "merged_core_GENUS_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_3(ADLT2)**

```{r, echo=TRUE}

### for ADLT2 donor
human_donors_ADLT2_inocula_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_ADLT2_inocula_table_PHYLUM <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_ADLT2_PHYLUM_table <- as.data.frame(human_donors_ADLT2_inocula_table_PHYLUM)
colnames(human_donors_ADLT2_PHYLUM_table)[colnames(human_donors_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_ADLT2_PHYLUM_table)[colnames(human_donors_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_PHYLUM_table)

# at FAMILY level
human_donors_ADLT2_inocula_table_FAMILY <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_ADLT2_inocula_table_FAMILY)
human_donors_ADLT2_FAMILY_table <- as.data.frame(human_donors_ADLT2_inocula_table_FAMILY)
colnames(human_donors_ADLT2_FAMILY_table)[colnames(human_donors_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_ADLT2_FAMILY_table)[colnames(human_donors_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_FAMILY_table)

# at GENUS level
human_donors_ADLT2_inocula_table_GENUS <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_ADLT2_inocula_table_GENUS)
human_donors_ADLT2_GENUS_table <- as.data.frame(human_donors_ADLT2_inocula_table_GENUS)
colnames(human_donors_ADLT2_GENUS_table)[colnames(human_donors_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_ADLT2_GENUS_table)[colnames(human_donors_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_GENUS_table)
pig_mouse_comp_ADLT2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_ADLT2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_ADLT2_comparison_ps_core.prop <- prune_taxa(core_ADLT2_taxa, pig_mouse_comp_ADLT2_comparison_ps.prop)
pig_mouse_comp_ADLT2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_ADLT2_core_ASVs <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_core.prop) %>% as_tibble
tb_ADLT2_core_ASVs_0 <- tb_ADLT2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_0_rounded <- tb_ADLT2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ADLT2)[colnames(df_mouse_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ADLT2)[colnames(df_piglet_ADLT2) == "OTU"] <- "ASV_ID"
View(df_donor_ADLT2)
View(df_mouse_ADLT2)
View(df_piglet_ADLT2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_mouse_ADLT2,df_piglet_ADLT2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_ADLT2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Phylum")

tb_ADLT2_core_ASVs_PHYLUM_df <- psmelt(tb_ADLT2_core_ASVs_PHYLUM) %>% as_tibble
tb_ADLT2_core_ASVs_PHYLUM_df_0 <- tb_ADLT2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded <- tb_ADLT2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_PHYLUM)[colnames(df_mouse_ADLT2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_PHYLUM)[colnames(df_mouse_ADLT2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_PHYLUM)[colnames(df_piglet_ADLT2_PHYLUM) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ADLT2_PHYLUM)[colnames(df_piglet_ADLT2_PHYLUM) == "SD"] <- "SD_piglets"

df_donor_ADLT2_PHYLUM$sample_Species <- NULL

ADLT2_donor_PHYLUM_df <- merge(human_donors_ADLT2_PHYLUM_table,df_donor_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_ADLT2_mice_ps <- prune_taxa(core_ADLT2_taxa_colonizing_mice, ADLT2_HMA_mice_ps)
colonizing_core_taxa_ADLT2_mice_ps # 66 ASVs

# at PHYLUM level
ADLT2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_PHYLUM_table)

df_mouse_ADLT2_PHYLUM$sample_Species <- NULL
ADLT2_mouse_PHYLUM_df <- merge(core_taxa_mice_ADLT2_PHYLUM_table,df_mouse_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_ADLT2_piglets_ps <- prune_taxa(core_ADLT2_taxa_colonizing_piglets, ADLT2_HMA_piglets_ps)
colonizing_core_taxa_ADLT2_piglets_ps # 106 ASVs

# at PHYLUM level
ADLT2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_PHYLUM_table)

df_piglet_ADLT2_PHYLUM$sample_Species <- NULL
ADLT2_piglet_PHYLUM_df <- merge(core_taxa_piglets_ADLT2_PHYLUM_table,df_piglet_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_ADLT2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_PHYLUM_df,ADLT2_mouse_PHYLUM_df,ADLT2_piglet_PHYLUM_df))
View(merged_ADLT2_PHYLUM)

write.table(merged_ADLT2_PHYLUM, "merged_core_PHYLUM_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_ADLT2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_ADLT2_core_ASVs_FAMILY_df <- psmelt(tb_ADLT2_core_ASVs_FAMILY) %>% as_tibble
tb_ADLT2_core_ASVs_FAMILY_df_0 <- tb_ADLT2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_FAMILY_df_0_rounded <- tb_ADLT2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded)

df_donor_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_FAMILY)[colnames(df_mouse_ADLT2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_FAMILY)[colnames(df_mouse_ADLT2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_FAMILY)[colnames(df_piglet_ADLT2_FAMILY) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ADLT2_FAMILY)[colnames(df_piglet_ADLT2_FAMILY) == "SD"] <- "SD_piglets"

View(df_donor_ADLT2_FAMILY)
df_donor_ADLT2_FAMILY$sample_Species <- NULL

ADLT2_donor_FAMILY_df <- merge(human_donors_ADLT2_FAMILY_table,df_donor_ADLT2_FAMILY, by="Family")
View(ADLT2_donor_FAMILY_df)


## for mice

# at FAMILY level
ADLT2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_ADLT2_FAMILY_table <- as.data.frame(ADLT2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_FAMILY_table)

df_mouse_ADLT2_FAMILY$sample_Species <- NULL
ADLT2_mouse_FAMILY_df <- merge(core_taxa_mice_ADLT2_FAMILY_table,df_mouse_ADLT2_FAMILY, by="Family")
View(ADLT2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
ADLT2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_ADLT2_FAMILY_table <- as.data.frame(ADLT2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_FAMILY_table)

df_piglet_ADLT2_FAMILY$sample_Species <- NULL
ADLT2_piglet_FAMILY_df <- merge(core_taxa_piglets_ADLT2_FAMILY_table,df_piglet_ADLT2_FAMILY, by="Family")
View(ADLT2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_ADLT2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_FAMILY_df,ADLT2_mouse_FAMILY_df,ADLT2_piglet_FAMILY_df))
View(merged_ADLT2_FAMILY)

write.table(merged_ADLT2_FAMILY, "merged_core_FAMILY_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_ADLT2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_ADLT2_core_ASVs_GENUS_df <- psmelt(tb_ADLT2_core_ASVs_GENUS) %>% as_tibble
tb_ADLT2_core_ASVs_GENUS_df_0 <- tb_ADLT2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_GENUS_df_0_rounded <- tb_ADLT2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_GENUS_df_0_rounded)

df_donor_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_GENUS)[colnames(df_mouse_ADLT2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_GENUS)[colnames(df_mouse_ADLT2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_GENUS)[colnames(df_piglet_ADLT2_GENUS) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ADLT2_GENUS)[colnames(df_piglet_ADLT2_GENUS) == "SD"] <- "SD_piglets"

View(df_donor_ADLT2_GENUS)
df_donor_ADLT2_GENUS$sample_Species <- NULL

# merged_prev_df_ADLT2_core_mice[is.na(merged_prev_df_ADLT2_core_mice)] <- 0
ADLT2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ADLT2_GENUS_table,df_donor_ADLT2_GENUS))
View(ADLT2_donor_GENUS_df)


## for mice

# at GENUS level
ADLT2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_ADLT2_GENUS_table <- as.data.frame(ADLT2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_GENUS_table)

df_mouse_ADLT2_GENUS$sample_Species <- NULL
View(df_mouse_ADLT2_GENUS)
ADLT2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ADLT2_GENUS_table,df_mouse_ADLT2_GENUS))
View(ADLT2_mouse_GENUS_df)

## for piglets

# at GENUS level
ADLT2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_ADLT2_GENUS_table <- as.data.frame(ADLT2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_GENUS_table)

df_piglet_ADLT2_GENUS$sample_Species <- NULL
ADLT2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ADLT2_GENUS_table,df_piglet_ADLT2_GENUS))
View(ADLT2_piglet_GENUS_df)


## merged the data frames from all three species
merged_ADLT2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_GENUS_df,ADLT2_mouse_GENUS_df,ADLT2_piglet_GENUS_df))
View(merged_ADLT2_GENUS)

write.table(merged_ADLT2_GENUS, "merged_core_GENUS_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_4(SNR2/ELD2)**

```{r, echo=TRUE}

### for ELD2 donor
human_donors_ELD2_inocula_core_ps 

# core taxa found in all donor samples
# at PHYLUM level
human_donors_ELD2_inocula_table_PHYLUM <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_ELD2_PHYLUM_table <- as.data.frame(human_donors_ELD2_inocula_table_PHYLUM)
colnames(human_donors_ELD2_PHYLUM_table)[colnames(human_donors_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_ELD2_PHYLUM_table)[colnames(human_donors_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_PHYLUM_table)

# at FAMILY level
human_donors_ELD2_inocula_table_FAMILY <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_ELD2_inocula_table_FAMILY)
human_donors_ELD2_FAMILY_table <- as.data.frame(human_donors_ELD2_inocula_table_FAMILY)
colnames(human_donors_ELD2_FAMILY_table)[colnames(human_donors_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_ELD2_FAMILY_table)[colnames(human_donors_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_FAMILY_table)

# at GENUS level
human_donors_ELD2_inocula_table_GENUS <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_ELD2_inocula_table_GENUS)
human_donors_ELD2_GENUS_table <- as.data.frame(human_donors_ELD2_inocula_table_GENUS)
colnames(human_donors_ELD2_GENUS_table)[colnames(human_donors_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_ELD2_GENUS_table)[colnames(human_donors_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_GENUS_table)

pig_mouse_comp_ELD2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_ELD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x) x/sum(x))

# retain just the core ELD2 ASVs
pig_mouse_comp_ELD2_comparison_ps_core.prop <- prune_taxa(core_ELD2_taxa, pig_mouse_comp_ELD2_comparison_ps.prop)
pig_mouse_comp_ELD2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_ELD2_core_ASVs <- psmelt(pig_mouse_comp_ELD2_comparison_ps_core.prop) %>% as_tibble
tb_ELD2_core_ASVs_0 <- tb_ELD2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_0_rounded <- tb_ELD2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ELD2)[colnames(df_mouse_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ELD2)[colnames(df_piglet_ELD2) == "OTU"] <- "ASV_ID"
View(df_donor_ELD2)
View(df_mouse_ELD2)
View(df_piglet_ELD2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_mouse_ELD2,df_piglet_ELD2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_ELD2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Phylum")

tb_ELD2_core_ASVs_PHYLUM_df <- psmelt(tb_ELD2_core_ASVs_PHYLUM) %>% as_tibble
tb_ELD2_core_ASVs_PHYLUM_df_0 <- tb_ELD2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_PHYLUM_df_0_rounded <- tb_ELD2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_PHYLUM)[colnames(df_mouse_ELD2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_PHYLUM)[colnames(df_mouse_ELD2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_PHYLUM)[colnames(df_piglet_ELD2_PHYLUM) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ELD2_PHYLUM)[colnames(df_piglet_ELD2_PHYLUM) == "SD"] <- "SD_piglets"

df_donor_ELD2_PHYLUM$sample_Species <- NULL

ELD2_donor_PHYLUM_df <- merge(human_donors_ELD2_PHYLUM_table,df_donor_ELD2_PHYLUM, by="Phylum")
View(ELD2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_ELD2_mice_ps <- prune_taxa(core_ELD2_taxa_colonizing_mice, ELD2_HMA_mice_ps)
colonizing_core_taxa_ELD2_mice_ps # 55 ASVs

# at PHYLUM level
ELD2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_ELD2_PHYLUM_table <- as.data.frame(ELD2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_PHYLUM_table)

df_mouse_ELD2_PHYLUM$sample_Species <- NULL
ELD2_mouse_PHYLUM_df <- merge(core_taxa_mice_ELD2_PHYLUM_table,df_mouse_ELD2_PHYLUM, by="Phylum")
View(ELD2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_ELD2_piglets_ps <- prune_taxa(core_ELD2_taxa_colonizing_piglets, ELD2_HMA_piglets_ps)
colonizing_core_taxa_ELD2_piglets_ps # 92 ASVs

# at PHYLUM level
ELD2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_ELD2_PHYLUM_table <- as.data.frame(ELD2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_PHYLUM_table)

df_piglet_ELD2_PHYLUM$sample_Species <- NULL
ELD2_piglet_PHYLUM_df <- merge(core_taxa_piglets_ELD2_PHYLUM_table,df_piglet_ELD2_PHYLUM, by="Phylum")
View(ELD2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_ELD2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_PHYLUM_df,ELD2_mouse_PHYLUM_df,ELD2_piglet_PHYLUM_df))
View(merged_ELD2_PHYLUM)

write.table(merged_ELD2_PHYLUM, "merged_core_PHYLUM_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_ELD2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_ELD2_core_ASVs_FAMILY_df <- psmelt(tb_ELD2_core_ASVs_FAMILY) %>% as_tibble
tb_ELD2_core_ASVs_FAMILY_df_0 <- tb_ELD2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_FAMILY_df_0_rounded <- tb_ELD2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_FAMILY_df_0_rounded)

df_donor_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_FAMILY)[colnames(df_mouse_ELD2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_FAMILY)[colnames(df_mouse_ELD2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_FAMILY)[colnames(df_piglet_ELD2_FAMILY) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ELD2_FAMILY)[colnames(df_piglet_ELD2_FAMILY) == "SD"] <- "SD_piglets"

View(df_donor_ELD2_FAMILY)
df_donor_ELD2_FAMILY$sample_Species <- NULL

ELD2_donor_FAMILY_df <- merge(human_donors_ELD2_FAMILY_table,df_donor_ELD2_FAMILY, by="Family")
View(ELD2_donor_FAMILY_df)


## for mice

# at FAMILY level
ELD2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_ELD2_FAMILY_table <- as.data.frame(ELD2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_FAMILY_table)

df_mouse_ELD2_FAMILY$sample_Species <- NULL
ELD2_mouse_FAMILY_df <- merge(core_taxa_mice_ELD2_FAMILY_table,df_mouse_ELD2_FAMILY, by="Family")
View(ELD2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
ELD2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_ELD2_FAMILY_table <- as.data.frame(ELD2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_FAMILY_table)

df_piglet_ELD2_FAMILY$sample_Species <- NULL
ELD2_piglet_FAMILY_df <- merge(core_taxa_piglets_ELD2_FAMILY_table,df_piglet_ELD2_FAMILY, by="Family")
View(ELD2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_ELD2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_FAMILY_df,ELD2_mouse_FAMILY_df,ELD2_piglet_FAMILY_df))
View(merged_ELD2_FAMILY)

write.table(merged_ELD2_FAMILY, "merged_core_FAMILY_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_ELD2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_ELD2_core_ASVs_GENUS_df <- psmelt(tb_ELD2_core_ASVs_GENUS) %>% as_tibble
tb_ELD2_core_ASVs_GENUS_df_0 <- tb_ELD2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_GENUS_df_0_rounded <- tb_ELD2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_GENUS_df_0_rounded)

df_donor_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_GENUS)[colnames(df_mouse_ELD2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_GENUS)[colnames(df_mouse_ELD2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_GENUS)[colnames(df_piglet_ELD2_GENUS) == "Mean"] <- "Mean_piglets" 
colnames(df_piglet_ELD2_GENUS)[colnames(df_piglet_ELD2_GENUS) == "SD"] <- "SD_piglets"

View(df_donor_ELD2_GENUS)
df_donor_ELD2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
ELD2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ELD2_GENUS_table,df_donor_ELD2_GENUS))
View(ELD2_donor_GENUS_df)


## for mice

# at GENUS level
ELD2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_ELD2_GENUS_table <- as.data.frame(ELD2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_GENUS_table)

df_mouse_ELD2_GENUS$sample_Species <- NULL
View(df_mouse_ELD2_GENUS)
ELD2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ELD2_GENUS_table,df_mouse_ELD2_GENUS))
View(ELD2_mouse_GENUS_df)

## for piglets

# at GENUS level
ELD2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_ELD2_GENUS_table <- as.data.frame(ELD2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_GENUS_table)

df_piglet_ELD2_GENUS$sample_Species <- NULL
ELD2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ELD2_GENUS_table,df_piglet_ELD2_GENUS))
View(ELD2_piglet_GENUS_df)


## merged the data frames from all three species
merged_ELD2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_GENUS_df,ELD2_mouse_GENUS_df,ELD2_piglet_GENUS_df))
View(merged_ELD2_GENUS)

write.table(merged_ELD2_GENUS, "merged_core_GENUS_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


##**Calculating mean and standard deviation for persistently colonizing core ASVs in each donor among the three different host species and assigning taxonomy**## 

**INF2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_INF2_comparison_ps # cecal samples and 'early' inoculation piglet samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_INF2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_INF2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_INF2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_INF2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_INF2_mice
pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_INF2_mice, pig_mouse_comp_INF2_comparison_ps_mice.prop)

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_INF2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_INF2_perst_core_ASVs_mice_0 <- tb_INF2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_perst_core_ASVs_mice_0_rounded <- tb_INF2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_perst_core_ASVs_mice_0_rounded)

df_donor_INF2 <- subset(tb_INF2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_INF2 <- subset(tb_INF2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_INF2)[colnames(df_mouse_INF2) == "OTU"] <- "ASV_ID"

View(df_donor_INF2)
View(df_mouse_INF2)

merged_INF2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_mouse_INF2, by="ASV_ID"))
View(merged_INF2_perst_mice)
merged_INF2_perst_mice[[5]] <- NULL
merged_INF2_perst_mice[[9]] <- NULL
merged_INF2_perst_mice[[10]] <- NULL
merged_INF2_perst_mice[[11]] <- NULL
View(merged_INF2_perst_mice)
write.table(merged_INF2_perst_mice, "merged_INF2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglets

# retain just the piglet and donor samples
pig_mouse_comp_INF2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_INF2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_INF2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_INF2_piglets
pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_INF2_piglets, pig_mouse_comp_INF2_comparison_ps_piglets.prop)

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_INF2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_INF2_perst_core_ASVs_piglets_0 <- tb_INF2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_perst_core_ASVs_piglets_0_rounded <- tb_INF2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_perst_core_ASVs_piglets_0_rounded)

df_donor_INF2 <- subset(tb_INF2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_INF2 <- subset(tb_INF2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_INF2)[colnames(df_piglet_INF2) == "OTU"] <- "ASV_ID"

View(df_donor_INF2)
View(df_piglet_INF2)

merged_INF2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_piglet_INF2, by="ASV_ID"))
View(merged_INF2_perst_piglets)
merged_INF2_perst_piglets[[5]] <- NULL
merged_INF2_perst_piglets[[9]] <- NULL
merged_INF2_perst_piglets[[10]] <- NULL
merged_INF2_perst_piglets[[11]] <- NULL
View(merged_INF2_perst_piglets)
write.table(merged_INF2_perst_piglets, "merged_INF2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**Donor_2(CHLD2)**

```{r, echo=TRUE}

pig_mouse_comp_CHLD2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_CHLD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_CHLD2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_CHLD2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_CHLD2_mice
pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_CHLD2_mice,pig_mouse_comp_CHLD2_comparison_ps_mice.prop)
pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_CHLD2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_CHLD2_perst_core_ASVs_mice_0 <- tb_CHLD2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_perst_core_ASVs_mice_0_rounded <- tb_CHLD2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_perst_core_ASVs_mice_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_CHLD2)[colnames(df_mouse_CHLD2) == "OTU"] <- "ASV_ID"

View(df_donor_CHLD2)
View(df_mouse_CHLD2)

merged_CHLD2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_mouse_CHLD2, by="ASV_ID"))
View(merged_CHLD2_perst_mice)
merged_CHLD2_perst_mice[[5]] <- NULL
merged_CHLD2_perst_mice[[9]] <- NULL
merged_CHLD2_perst_mice[[10]] <- NULL
merged_CHLD2_perst_mice[[11]] <- NULL
View(merged_CHLD2_perst_mice)
write.table(merged_CHLD2_perst_mice, "merged_CHLD2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglets

# retain just the mouse and donor samples
pig_mouse_comp_CHLD2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_CHLD2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_CHLD2_piglets
pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_CHLD2_piglets, pig_mouse_comp_CHLD2_comparison_ps_piglets.prop)
pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_CHLD2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_CHLD2_perst_core_ASVs_piglets_0 <- tb_CHLD2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_perst_core_ASVs_piglets_0_rounded <- tb_CHLD2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_perst_core_ASVs_piglets_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_CHLD2)[colnames(df_piglet_CHLD2) == "OTU"] <- "ASV_ID"

View(df_donor_CHLD2)
View(df_piglet_CHLD2)

merged_CHLD2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_piglet_CHLD2, by="ASV_ID"))
View(merged_CHLD2_perst_piglets)
merged_CHLD2_perst_piglets[[5]] <- NULL
merged_CHLD2_perst_piglets[[9]] <- NULL
merged_CHLD2_perst_piglets[[10]] <- NULL
merged_CHLD2_perst_piglets[[11]] <- NULL
View(merged_CHLD2_perst_piglets)
write.table(merged_CHLD2_perst_piglets, "merged_CHLD2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**Donor_3(ADLT2)**

```{r, echo=TRUE}

pig_mouse_comp_ADLT2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_ADLT2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_ADLT2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_ADLT2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_ADLT2_mice
pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_ADLT2_mice,pig_mouse_comp_ADLT2_comparison_ps_mice.prop)
pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ADLT2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_ADLT2_perst_core_ASVs_mice_0 <- tb_ADLT2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_perst_core_ASVs_mice_0_rounded <- tb_ADLT2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_perst_core_ASVs_mice_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ADLT2)[colnames(df_mouse_ADLT2) == "OTU"] <- "ASV_ID"

View(df_donor_ADLT2)
View(df_mouse_ADLT2)

merged_ADLT2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_mouse_ADLT2, by="ASV_ID"))
View(merged_ADLT2_perst_mice)
merged_ADLT2_perst_mice[[5]] <- NULL
merged_ADLT2_perst_mice[[9]] <- NULL
merged_ADLT2_perst_mice[[10]] <- NULL
merged_ADLT2_perst_mice[[11]] <- NULL
View(merged_ADLT2_perst_mice)
write.table(merged_ADLT2_perst_mice, "merged_ADLT2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglets

# retain just the mouse and donor samples
pig_mouse_comp_ADLT2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_ADLT2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_ADLT2_piglets
pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_ADLT2_piglets, pig_mouse_comp_ADLT2_comparison_ps_piglets.prop)
pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ADLT2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_ADLT2_perst_core_ASVs_piglets_0 <- tb_ADLT2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_perst_core_ASVs_piglets_0_rounded <- tb_ADLT2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_perst_core_ASVs_piglets_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ADLT2)[colnames(df_piglet_ADLT2) == "OTU"] <- "ASV_ID"

View(df_donor_ADLT2)
View(df_piglet_ADLT2)

merged_ADLT2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_piglet_ADLT2, by="ASV_ID"))
View(merged_ADLT2_perst_piglets)
merged_ADLT2_perst_piglets[[5]] <- NULL
merged_ADLT2_perst_piglets[[9]] <- NULL
merged_ADLT2_perst_piglets[[10]] <- NULL
merged_ADLT2_perst_piglets[[11]] <- NULL
View(merged_ADLT2_perst_piglets)
write.table(merged_ADLT2_perst_piglets, "merged_ADLT2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**Donor_4(SNR2/ELD2)**

```{r, echo=TRUE}

pig_mouse_comp_ELD2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_ELD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_ELD2_comparison_ps.prop

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_ELD2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_ELD2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_ELD2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_ELD2_mice
pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_ELD2_mice,pig_mouse_comp_ELD2_comparison_ps_mice.prop)
pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ELD2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_ELD2_perst_core_ASVs_mice_0 <- tb_ELD2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_perst_core_ASVs_mice_0_rounded <- tb_ELD2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_perst_core_ASVs_mice_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_ELD2 <- subset(tb_ELD2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ELD2)[colnames(df_mouse_ELD2) == "OTU"] <- "ASV_ID"

View(df_donor_ELD2)
View(df_mouse_ELD2)

merged_ELD2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_mouse_ELD2, by="ASV_ID"))
View(merged_ELD2_perst_mice)
merged_ELD2_perst_mice[[5]] <- NULL
merged_ELD2_perst_mice[[9]] <- NULL
merged_ELD2_perst_mice[[10]] <- NULL
merged_ELD2_perst_mice[[11]] <- NULL
View(merged_ELD2_perst_mice)
write.table(merged_ELD2_perst_mice, "merged_ELD2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglets

# retain just the mouse and donor samples
pig_mouse_comp_ELD2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_ELD2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_ELD2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_ELD2_piglets
pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_ELD2_piglets, pig_mouse_comp_ELD2_comparison_ps_piglets.prop)
pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ELD2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_ELD2_perst_core_ASVs_piglets_0 <- tb_ELD2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_perst_core_ASVs_piglets_0_rounded <- tb_ELD2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_perst_core_ASVs_piglets_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_ELD2 <- subset(tb_ELD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ELD2)[colnames(df_piglet_ELD2) == "OTU"] <- "ASV_ID"

View(df_donor_ELD2)
View(df_piglet_ELD2)

merged_ELD2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_piglet_ELD2, by="ASV_ID"))
View(merged_ELD2_perst_piglets)
merged_ELD2_perst_piglets[[5]] <- NULL
merged_ELD2_perst_piglets[[9]] <- NULL
merged_ELD2_perst_piglets[[10]] <- NULL
merged_ELD2_perst_piglets[[11]] <- NULL
View(merged_ELD2_perst_piglets)
write.table(merged_ELD2_perst_piglets, "merged_ELD2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```



##**Making Circos Plots in the form of Chord Diagrams**##


## Generating data for Circos plots ##


Getting relative abundance values for making Circos plots :


**Donor_1 (INF2)**

```{r, echo=TRUE}
pig_mouse_comp_INF2_comparison_ps

## Calculating abundance values for donor inocula
INF2_donor_inocula_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "human")
INF2_donor_inocula_ps

# Calculate relative abundances
INF2_donor_inocula_ps.prop <- transform_sample_counts(INF2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_INF2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
INF2_donor_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "mouse")
INF2_donor_mice_ps

# Calculate relative abundances
INF2_donor_mice_ps.prop <- transform_sample_counts(INF2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_INF2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
INF2_donor_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "piglet")
INF2_donor_piglets_ps

# Calculate relative abundances
INF2_donor_piglets_ps.prop <- transform_sample_counts(INF2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_INF2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_2 (CHLD2)** 

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_comparison_ps

## Calculating abundance values for donor inocula
CHLD2_donor_inocula_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "human")
CHLD2_donor_inocula_ps

# Calculate relative abundances
CHLD2_donor_inocula_ps.prop <- transform_sample_counts(CHLD2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
CHLD2_donor_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "mouse")
CHLD2_donor_mice_ps

# Calculate relative abundances
CHLD2_donor_mice_ps.prop <- transform_sample_counts(CHLD2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
CHLD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "piglet")
CHLD2_donor_piglets_ps

# Calculate relative abundances
CHLD2_donor_piglets_ps.prop <- transform_sample_counts(CHLD2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_3 (ADLT2)**

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_comparison_ps

## Calculating abundance values for donor inocula
ADLT2_donor_inocula_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "human")
ADLT2_donor_inocula_ps

# Calculate relative abundances
ADLT2_donor_inocula_ps.prop <- transform_sample_counts(ADLT2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
ADLT2_donor_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "mouse")
ADLT2_donor_mice_ps

# Calculate relative abundances
ADLT2_donor_mice_ps.prop <- transform_sample_counts(ADLT2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
ADLT2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "piglet")
ADLT2_donor_piglets_ps

# Calculate relative abundances
ADLT2_donor_piglets_ps.prop <- transform_sample_counts(ADLT2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_4 (ELD2/SNR2)**

```{r, echo=TRUE}
pig_mouse_comp_ELD2_comparison_ps

## Calculating abundance values for donor inocula
ELD2_donor_inocula_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "human")
ELD2_donor_inocula_ps

# Calculate relative abundances
ELD2_donor_inocula_ps.prop <- transform_sample_counts(ELD2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_ELD2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
ELD2_donor_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "mouse")
ELD2_donor_mice_ps

# Calculate relative abundances
ELD2_donor_mice_ps.prop <- transform_sample_counts(ELD2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_ELD2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
ELD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "piglet")
ELD2_donor_piglets_ps

# Calculate relative abundances
ELD2_donor_piglets_ps.prop <- transform_sample_counts(ELD2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_ELD2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


Drawing circos plots (Chord diagrams) with the relative abundance data generated above (Based on information in https://jokergoo.github.io/circlize_book/book/):

**Donor_1 (INF2)**

```{r, echo=TRUE}

## Folder with all the input files necessary to generate the Chord diagrams ('Circos_plots' folder) has been uploaded to GitHub.

# Install 'circlize' package
install.packages("circlize"); packageVersion('circlize')

# Load circlize package and initiate
library(circlize); packageVersion('circlize')
circos.par("track.height" = 0.1)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/INF2_core/INF2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)
View(input_df)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/INF2_core/INF2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame
View(cols_df)

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/INF2_core/INF2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)
View(grid_cols_df)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)
is.vector(grid_cols) # can check to see if it is indeed a vector

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/INF2_core/INF2_df1.txt", sep = "\t", header = T)
INF2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/INF2_core/INF2_df2.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed
INF2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/INF2_core/INF2_df3.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed
INF2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(INF2_df1$Samp)
M_ASVs <- as.vector(INF2_df2$Samp)
P_ASVs <- as.vector(INF2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.5, font = 2)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "Donor_1", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
chord_diag_INF2 <- highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "HMA mice ", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
chord_diag_INF2 <- highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "HMA piglets", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**Donor_2 (CHLD2)**

```{r, echo=TRUE}
# Load circlize package and initiate
circos.par("track.height" = 0.1)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/CHLD2_core/CHLD2_df1.txt", sep = "\t", header = T)
CHLD2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/CHLD2_core/CHLD2_df2.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
CHLD2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/CHLD2_core/CHLD2_df3.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
CHLD2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(CHLD2_df1$Samp)
M_ASVs <- as.vector(CHLD2_df2$Samp)
View(M_ASVs)
P_ASVs <- as.vector(CHLD2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.4, font = 2)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "Donor_2", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "HMA mice ", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "HMA piglets", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**Donor_3 (ADLT2)**

```{r, echo=TRUE}

circos.par("track.height" = 0.1, gap.degree = 0.8)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)
View(grid_cols)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]
View(link_cols)

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/ADLT2_core/ADLT2_df1.txt", sep = "\t", header = T)
ADLT2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/ADLT2_core/ADLT2_df2.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
ADLT2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/ADLT2_core/ADLT2_df3.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
ADLT2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(ADLT2_df1$Samp)
M_ASVs <- as.vector(ADLT2_df2$Samp)
P_ASVs <- as.vector(ADLT2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.4, font = 2)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "Donor_3", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "HMA mice", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "HMA piglets", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**Donor_4 (ELD2/SNR2)**

```{r, echo=TRUE}

circos.par("track.height" = 0.1, gap.degree = 0.8)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/ELD2_core/ELD2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/ELD2_core/ELD2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/ELD2_core/ELD2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)
View(grid_cols)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]
View(link_cols)

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/ELD2_core/ELD2_df1.txt", sep = "\t", header = T)
ELD2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/ELD2_core/ELD2_df2.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
ELD2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/ELD2_core/ELD2_df3.txt", sep = "\t", header = T) # any ASV which has a zero relative abundance needs to be removed. In addition, need to remove anything with a rel. abundance < 10-4 (<0.0001)
ELD2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(ELD2_df1$Samp)
M_ASVs <- as.vector(ELD2_df2$Samp)
P_ASVs <- as.vector(ELD2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.4, font = 2)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "Donor_4", cex = 1.0, font = 2,text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "HMA mice", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "HMA piglets", cex = 1.0, font = 2, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


##**Comparing unweighted UniFrac and Bray-Curtis distances**##


**Comparing unweighted Unifrac Distances**

```{r, echo=TRUE}

### Based on code provided in https://github.com/joey711/phyloseq/issues/714
install.packages("dplyr") # if not alraedy installed
library("dplyr")
library("reshape2")
library("dada2"); packageVersion("dada2")


### for INF2 donor

# calculate distances
INF2_dist_comp_unf <- phyloseq::distance(rel_abund_INF2_core_ps, method = 'unifrac', weighted = F)
INF2_dist_comp.m <- melt(as.matrix(INF2_dist_comp_unf))

# remove self-comparisons
INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
INF2_donor_samples <- c("R2D269","R2D273", "323")
INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(!as.character(Var1) %in% INF2_donor_samples) %>% mutate_if(is.factor,as.character)

INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(as.character(Var2) %in% INF2_donor_samples) %>% mutate_if(is.factor,as.character)
INF2_dist_comp.m.df <- as.data.frame(INF2_dist_comp.m) 

# get sample data 
#sample_data_CHLD2 <- sample_data(rel_abund_CHLD2_core_ps) %>% select('SampleID','Species','Donor') %>% mutate_if(is.factor,as.character) ## For some reason (possibly due to a conflict with other packages) this is not working. So I'm doing the following :
sample_data_INF2 <- as.data.frame(sample_data(rel_abund_INF2_core_ps))
View(sample_data_INF2)
sample_data_INF2 <- subset(sample_data_INF2, select = c('SampleID','Species','Donor'))
View(sample_data_INF2)
sample_data_INF2[] <- lapply(sample_data_INF2, as.character)

# combine unifrac distance data with sample data
names(sample_data_INF2)[names(sample_data_INF2) == "SampleID"]  <- "Var1"
INF2_dist_comp.sd = dplyr::left_join(INF2_dist_comp.m, sample_data_INF2, by = "Var1")
View(INF2_dist_comp.sd)


### for CHLD2 donor

# calculate distances
CHLD2_dist_comp_unf <- phyloseq::distance(rel_abund_CHLD2_core_ps, method = 'unifrac', weighted = F)
CHLD2_dist_comp.m <- melt(as.matrix(CHLD2_dist_comp_unf))

# remove self-comparisons
CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
CHLD2_donor_samples <- c("R2D270","R2D274", "360", "371")
CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(!as.character(Var1) %in% CHLD2_donor_samples) %>% mutate_if(is.factor,as.character)

CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(as.character(Var2) %in% CHLD2_donor_samples) %>% mutate_if(is.factor,as.character)
CHLD2_dist_comp.m.df <- as.data.frame(CHLD2_dist_comp.m) 

# get sample data 
sample_data_CHLD2 <- as.data.frame(sample_data(rel_abund_CHLD2_core_ps))
View(sample_data_CHLD2)
sample_data_CHLD2 <- subset(sample_data_CHLD2, select = c('SampleID','Species','Donor'))
View(sample_data_CHLD2)
sample_data_CHLD2[] <- lapply(sample_data_CHLD2, as.character)

# combine unifrac distance data with sample data
names(sample_data_CHLD2)[names(sample_data_CHLD2) == "SampleID"]  <- "Var1"
CHLD2_dist_comp.sd = left_join(CHLD2_dist_comp.m, sample_data_CHLD2, by = "Var1")
View(CHLD2_dist_comp.sd)


### for ADLT2 donor

# calculate distances
ADLT2_dist_comp_unf <- phyloseq::distance(rel_abund_ADLT2_core_ps, method = 'unifrac', weighted = F)
ADLT2_dist_comp.m <- melt(as.matrix(ADLT2_dist_comp_unf))

# remove self-comparisons
ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
ADLT2_donor_samples <- c("R2D271","R2D275", "348", "372")
ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(!as.character(Var1) %in% ADLT2_donor_samples) %>% mutate_if(is.factor,as.character)

ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(as.character(Var2) %in% ADLT2_donor_samples) %>% mutate_if(is.factor,as.character)
ADLT2_dist_comp.m.df <- as.data.frame(ADLT2_dist_comp.m) 

# get sample data 
sample_data_ADLT2 <- as.data.frame(sample_data(rel_abund_ADLT2_core_ps))
View(sample_data_ADLT2)
sample_data_ADLT2 <- subset(sample_data_ADLT2, select = c('SampleID','Species','Donor'))
View(sample_data_ADLT2)
sample_data_ADLT2[] <- lapply(sample_data_ADLT2, as.character)

# combine unifrac distance data with sample data
names(sample_data_ADLT2)[names(sample_data_ADLT2) == "SampleID"]  <- "Var1"
ADLT2_dist_comp.sd = left_join(ADLT2_dist_comp.m, sample_data_ADLT2, by = "Var1")
View(ADLT2_dist_comp.sd)


### for SNR2 donor

# calculate distances
ELD2_dist_comp_unf <- phyloseq::distance(rel_abund_ELD2_core_ps, method = 'unifrac', weighted = F)
ELD2_dist_comp.m <- melt(as.matrix(ELD2_dist_comp_unf))

# remove self-comparisons
ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
DONORS_snr <- subset_samples(rel_abund_ELD2_core_ps, Species == 'human')
DONORS_snr
sample_names(DONORS_snr)
ELD2_donor_samples <- c("R2D272","R2D276", "324", "359")
ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(!as.character(Var1) %in% ELD2_donor_samples) %>% mutate_if(is.factor,as.character)

ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(as.character(Var2) %in% ELD2_donor_samples) %>% mutate_if(is.factor,as.character)
ELD2_dist_comp.m.df <- as.data.frame(ELD2_dist_comp.m) 

# get sample data 
sample_data_ELD2 <- as.data.frame(sample_data(rel_abund_ELD2_core_ps))
View(sample_data_ELD2)
sample_data_ELD2 <- subset(sample_data_ELD2, select = c('SampleID','Species','Donor'))
View(sample_data_ELD2)
sample_data_ELD2[] <- lapply(sample_data_ELD2, as.character)

# combine unifrac distance data with sample data
names(sample_data_ELD2)[names(sample_data_ELD2) == "SampleID"]  <- "Var1"
ELD2_dist_comp.sd = left_join(ELD2_dist_comp.m, sample_data_ELD2, by = "Var1")
View(ELD2_dist_comp.sd)


## combine overall results
comb_unifrac_df <- rbind(INF2_dist_comp.sd,CHLD2_dist_comp.sd,ADLT2_dist_comp.sd,ELD2_dist_comp.sd)
View(comb_unifrac_df)
write.table(comb_unifrac_df, "unifrac_dist_comp_source_data.txt", sep = "\t", row.names = F, quote = F, col.names = T) # source data for publication

plot_unifrac <- ggplot(comb_unifrac_df, aes(x = Donor, y = value, fill = Species)) + geom_boxplot() + scale_x_discrete(limits=c("INF2", "CHLD2", "ADLT2", "ELD2"), labels = c("Donor_1", "Donor_2", "Donor_3", "Donor_4")) + theme(axis.text.x=element_text(angle=45,hjust=1, size = 10), legend.title = element_text(size = 10, face = "bold"), axis.title = element_text(size = 10,face = "bold")) + xlab("Donor") + scale_fill_manual(values = c("forestgreen", "dodgerblue3"), name = "host species", labels = c("Donor inocula vs. HMA mouse fecal samples","Donor inocula vs. HMA piglet fecal samples")) + scale_y_continuous(name = "Unweighted UniFrac distance from donor",  limits = c(0,1), labels = scales::number_format(accuracy=0.1))

plot_unifrac


## Performing comparisons with Wilocoxon test

# INF2 donor
INF_mice = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'INF2' & comb_unifrac_df[,'Species'] == 'mouse',]
INF_mice <- INF_mice$value

INF_piglets = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'INF2' & comb_unifrac_df[,'Species'] == 'piglet',]
INF_piglets <- INF_piglets$value

wilcox.test(INF_mice,INF_piglets)

# CHLD2 donor
CHLD_mice = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'CHLD2' & comb_unifrac_df[,'Species'] == 'mouse',]
CHLD_mice <- CHLD_mice$value

CHLD_piglets = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'CHLD2' & comb_unifrac_df[,'Species'] == 'piglet',]
CHLD_piglets <- CHLD_piglets$value

wilcox.test(CHLD_mice,CHLD_piglets)

# ADLT2 donor
ADLT_mice = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'ADLT2' & comb_unifrac_df[,'Species'] == 'mouse',]
ADLT_mice <- ADLT_mice$value

ADLT_piglets = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'ADLT2' & comb_unifrac_df[,'Species'] == 'piglet',]
ADLT_piglets <- ADLT_piglets$value

wilcox.test(ADLT_mice,ADLT_piglets)

# SNR2 donor
ELD_mice = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'ELD2' & comb_unifrac_df[,'Species'] == 'mouse',]
ELD_mice <- ELD_mice$value

ELD_piglets = comb_unifrac_df[comb_unifrac_df[,'Donor'] == 'ELD2' & comb_unifrac_df[,'Species'] == 'piglet',]
ELD_piglets <- ELD_piglets$value

wilcox.test(ELD_mice,ELD_piglets)
```


**Comparing Bray-Curtis distances**

```{r, echo=TRUE}
### for INF2 donor

# calculate distances
INF2_dist_comp_BC <- phyloseq::distance(rel_abund_INF2_core_ps, method = 'bray')
INF2_dist_comp.m <- melt(as.matrix(INF2_dist_comp_BC))

# remove self-comparisons
INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
INF2_donor_samples <- c("R2D269","R2D273", "323")
INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(!as.character(Var1) %in% INF2_donor_samples) %>% mutate_if(is.factor,as.character)

INF2_dist_comp.m <- INF2_dist_comp.m %>% filter(as.character(Var2) %in% INF2_donor_samples) %>% mutate_if(is.factor,as.character)
INF2_dist_comp.m.df <- as.data.frame(INF2_dist_comp.m) 

# combine unifrac distance data with sample data
names(sample_data_INF2)[names(sample_data_INF2) == "SampleID"]  <- "Var1"
INF2_dist_comp.sd = left_join(INF2_dist_comp.m, sample_data_INF2, by = "Var1")


### for CHLD2 donor

# calculate distances
CHLD2_dist_comp_BC <- phyloseq::distance(rel_abund_CHLD2_core_ps, method = 'bray')
CHLD2_dist_comp.m <- melt(as.matrix(CHLD2_dist_comp_BC))

# remove self-comparisons
CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
CHLD2_donor_samples <- c("R2D270","R2D274", "360", "371")
CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(!as.character(Var1) %in% CHLD2_donor_samples) %>% mutate_if(is.factor,as.character)

CHLD2_dist_comp.m <- CHLD2_dist_comp.m %>% filter(as.character(Var2) %in% CHLD2_donor_samples) %>% mutate_if(is.factor,as.character)
CHLD2_dist_comp.m.df <- as.data.frame(CHLD2_dist_comp.m) 

# combine unifrac distance data with sample data
names(sample_data_CHLD2)[names(sample_data_CHLD2) == "SampleID"]  <- "Var1"
CHLD2_dist_comp.sd = left_join(CHLD2_dist_comp.m, sample_data_CHLD2, by = "Var1")


### for ADLT2 donor

# calculate distances
ADLT2_dist_comp_BC <- phyloseq::distance(rel_abund_ADLT2_core_ps, method = 'bray')
ADLT2_dist_comp.m <- melt(as.matrix(ADLT2_dist_comp_BC))

# remove self-comparisons
ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
ADLT2_donor_samples <- c("R2D271","R2D275", "348", "372")
ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(!as.character(Var1) %in% ADLT2_donor_samples) %>% mutate_if(is.factor,as.character)

ADLT2_dist_comp.m <- ADLT2_dist_comp.m %>% filter(as.character(Var2) %in% ADLT2_donor_samples) %>% mutate_if(is.factor,as.character)
ADLT2_dist_comp.m.df <- as.data.frame(ADLT2_dist_comp.m) 

# combine unifrac distance data with sample data
names(sample_data_ADLT2)[names(sample_data_ADLT2) == "SampleID"]  <- "Var1"
ADLT2_dist_comp.sd = left_join(ADLT2_dist_comp.m, sample_data_ADLT2, by = "Var1")


### for SNR2 donor

# calculate distances
ELD2_dist_comp_BC <- phyloseq::distance(rel_abund_ELD2_core_ps, method = 'bray')
ELD2_dist_comp.m <- melt(as.matrix(ELD2_dist_comp_BC))

# remove self-comparisons
ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(as.character(Var1) != as.character(Var2)) %>% mutate_if(is.factor,as.character)  

# remove donor-donor comparisons
DONORS_snr <- subset_samples(rel_abund_ELD2_core_ps, Species == 'human')
DONORS_snr
sample_names(DONORS_snr)
ELD2_donor_samples <- c("R2D272","R2D276", "324", "359")
ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(!as.character(Var1) %in% ELD2_donor_samples) %>% mutate_if(is.factor,as.character)

ELD2_dist_comp.m <- ELD2_dist_comp.m %>% filter(as.character(Var2) %in% ELD2_donor_samples) %>% mutate_if(is.factor,as.character)
ELD2_dist_comp.m.df <- as.data.frame(ELD2_dist_comp.m) 

# combine unifrac distance data with sample data
names(sample_data_ELD2)[names(sample_data_ELD2) == "SampleID"]  <- "Var1"
ELD2_dist_comp.sd = left_join(ELD2_dist_comp.m, sample_data_ELD2, by = "Var1")
View(ELD2_dist_comp.sd)

## combine overall results
comb_BC_df <- rbind(INF2_dist_comp.sd,CHLD2_dist_comp.sd,ADLT2_dist_comp.sd,ELD2_dist_comp.sd)
write.table(comb_BC_df, "bc_dist_comp_source_data.txt", sep = "\t", row.names = F, quote = F, col.names = T) # source data for publication

plot_BC <- ggplot(comb_BC_df, aes(x = Donor, y = value, fill = Species)) + geom_boxplot() + scale_x_discrete(limits=c("INF2", "CHLD2", "ADLT2", "ELD2"), labels = c("Donor_1", "Donor_2", "Donor_3", "Donor_4")) + theme(axis.text.x=element_text(angle=45,hjust=1, size = 10), legend.title = element_text(size = 10, face = "bold"), axis.title = element_text(size = 10, face = "bold")) + xlab("Donor") + scale_fill_manual(values = c("forestgreen", "dodgerblue3"), name = "host species", labels = c("Donor inocula vs. HMA mouse fecal samples","Donor inocula vs. HMA piglet fecal samples")) + scale_y_continuous(name = "Bray-Curtis distance from donor", limits = c(0,1), labels = scales::number_format(accuracy = 0.1))

plot_BC


## Performing comparisons with Wilocoxon test

# INF2 donor
INF_mice = comb_BC_df[comb_BC_df[,'Donor'] == 'INF2' & comb_BC_df[,'Species'] == 'mouse',]
INF_mice <- INF_mice$value

INF_piglets = comb_BC_df[comb_BC_df[,'Donor'] == 'INF2' & comb_BC_df[,'Species'] == 'piglet',]
INF_piglets <- INF_piglets$value

wilcox.test(INF_mice,INF_piglets)

# CHLD2 donor
CHLD_mice = comb_BC_df[comb_BC_df[,'Donor'] == 'CHLD2' & comb_BC_df[,'Species'] == 'mouse',]
CHLD_mice <- CHLD_mice$value

CHLD_piglets = comb_BC_df[comb_BC_df[,'Donor'] == 'CHLD2' & comb_BC_df[,'Species'] == 'piglet',]
CHLD_piglets <- CHLD_piglets$value

wilcox.test(CHLD_mice,CHLD_piglets)

# ADLT2 donor
ADLT_mice = comb_BC_df[comb_BC_df[,'Donor'] == 'ADLT2' & comb_BC_df[,'Species'] == 'mouse',]
ADLT_mice <- ADLT_mice$value

ADLT_piglets = comb_BC_df[comb_BC_df[,'Donor'] == 'ADLT2' & comb_BC_df[,'Species'] == 'piglet',]
ADLT_piglets <- ADLT_piglets$value

wilcox.test(ADLT_mice,ADLT_piglets)

# SNR2 donor
ELD_mice = comb_BC_df[comb_BC_df[,'Donor'] == 'ELD2' & comb_BC_df[,'Species'] == 'mouse',]
ELD_mice <- ELD_mice$value

ELD_piglets = comb_BC_df[comb_BC_df[,'Donor'] == 'ELD2' & comb_BC_df[,'Species'] == 'piglet',]
ELD_piglets <- ELD_piglets$value

wilcox.test(ELD_mice,ELD_piglets)
```



##**PICRUSt Analysis**##

```{r, echo=TRUE}

mapping_file <- read.table("pig_mouse_comp_r2_mapping_file.txt", sep = "\t", header = TRUE) # read-in mapping file
rownames(mapping_file) <- mapping_file$SampleID

taxa_table <- readRDS("taxTab.rds")

merged_phyloseq_ps <- merge_phyloseq(otu_table(merged_seqtab_all_runs_nochim, taxa_are_rows = FALSE), sample_data(mapping_file), tax_table(taxa_table)) # 'decontam' as well as chimera removal have already been done in 'merged_seqtab_all_runs_nochim'
merged_phyloseq_ps

head(taxa_names(merged_phyloseq_ps)) # check to make sure that taxa names are the actual sequences (not 'ASV_#' format)

# remove negative controls
merged_phyloseq_ps_no_neg <- subset_samples(merged_phyloseq_ps, Description != 'negative_control')
merged_phyloseq_ps_no_neg

# filtering out Archaea, Eukaryota, and Mitochondria sequences
remov_kingdoms <- c('Archaea', 'Eukaryota')
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_phyloseq_ps_no_neg, !Kingdom %in% remov_kingdoms & (Family != 'Mitochondria' | is.na(Family))) 
merged_pig_mouse_comp_ps_master
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_pig_mouse_comp_ps_master, Kingdom == 'Bacteria') # removed 4 ASVs classified as 'NA' at the kingdom level
merged_pig_mouse_comp_ps_master

# Now let's remove Cyanobacterial sequences except for those that belong to the Class Melainabacteria (which are thought to be human gut cyanobacteria)
cyano_ps <- subset_taxa(merged_pig_mouse_comp_ps_master, Phylum == "Cyanobacteria")
cyano_ps 

remov_cyano_ps <- subset_taxa(cyano_ps, Class != "Melainabacteria" | is.na(Class)) # Keep only Melainabacteria (also remove any cyano ASVs with 'NA' as class)
cyano_to_remove <- taxa_names(remov_cyano_ps) 

# filtering out the non-Melainabacterial Cyanobacteria 
all_taxa <- taxa_names(merged_pig_mouse_comp_ps_master)
good_taxa <- all_taxa[!(all_taxa %in% cyano_to_remove)]
good_taxa

merged_pig_mouse_comp_ps_master_no_cyano <- prune_taxa(good_taxa,merged_pig_mouse_comp_ps_master)
merged_pig_mouse_comp_ps_master_no_cyano

# Let's also perform a prevalence calculation and filter out any ASV which is present in only a single sample
prevdf <- apply(X = otu_table(merged_pig_mouse_comp_ps_master_no_cyano), MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps_master_no_cyano), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
View(prevdf)
prevdf_with_taxa <- data.frame(Prevalence = prevdf, TotalAbundance = taxa_sums(merged_pig_mouse_comp_ps_master_no_cyano), tax_table(merged_pig_mouse_comp_ps_master_no_cyano))
View(prevdf_with_taxa)

# Retaining only those ASVs found in at least one sample
keepTaxa <- rownames(prevdf_with_taxa)[(prevdf_with_taxa$Prevalence >1)]
pig_mouse_comp_PICRUSt_ps <- prune_taxa(keepTaxa, merged_pig_mouse_comp_ps_master_no_cyano)
pig_mouse_comp_PICRUSt_ps

# Let's look at read depth
reads_per_sample_df <- data.frame(Reads = sample_sums(pig_mouse_comp_PICRUSt_ps)) 
reads_per_sample_df$sampleID <- rownames(reads_per_sample_df)
View(reads_per_sample_df)
reads_per_sample_df <- reads_per_sample_df[,c(2,1)] # reorder columns
reads_per_sample_df_sorted <- reads_per_sample_df[order(reads_per_sample_df$Reads),] # order samples by read count (ascending)
View(reads_per_sample_df_sorted)

# filtering out samples with <10,500 reads
pig_mouse_comp_PICRUSt_comp_ps <- prune_samples(sample_sums(pig_mouse_comp_PICRUSt_ps) > 10500, pig_mouse_comp_PICRUSt_ps)
pig_mouse_comp_PICRUSt_comp_ps 

# remove fecal samples collected at euthanization
pig_mouse_comp_PICRUSt_comp_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_ps, Time_point != "at_euthanization" )
pig_mouse_comp_PICRUSt_comp_ps # main phyloseq object to be used as the starting point for the following analyses based on individual donors
```


**Donor_1 (INF Donor)**

```{r, echo=TRUE}
pig_mouse_comp_PICRUSt_comp_no_cecal_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_ps, SampleType != 'cecal')
pig_mouse_comp_PICRUSt_INF_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_no_cecal_ps, Donor=="INF2")
pig_mouse_comp_PICRUSt_INF_ps <- subset_samples(pig_mouse_comp_PICRUSt_INF_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_PICRUSt_INF_ps <- subset_samples(pig_mouse_comp_PICRUSt_INF_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
pig_mouse_comp_PICRUSt_INF_ps 

early_inoculation_donor_samples <- c('299', '300')
pig_mouse_comp_PICRUSt_INF_ps <- subset_samples(pig_mouse_comp_PICRUSt_INF_ps, !SampleID %in% early_inoculation_donor_samples) # keeping only the donor inocula used to inoculate the mice and 'late' piglets
pig_mouse_comp_PICRUSt_INF_ps

# Let's keep only those ASVs found in > 1 sample
prevdf_INF <- apply(X = otu_table(pig_mouse_comp_PICRUSt_INF_ps), MARGIN = ifelse(taxa_are_rows(pig_mouse_comp_PICRUSt_INF_ps), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
prevdf_INF <- data.frame(Prevalence = prevdf_INF, TotalAbundance = taxa_sums(pig_mouse_comp_PICRUSt_INF_ps))
View(prevdf_INF)

# Retaining only those ASVs found in at least one sample
keepTaxa <- rownames(prevdf_INF)[(prevdf_INF$Prevalence >1)]
pig_mouse_comp_PICRUSt_INF_ps <- prune_taxa(keepTaxa, pig_mouse_comp_PICRUSt_INF_ps)
pig_mouse_comp_PICRUSt_INF_ps

## getting the necessary input files for PICRUSt
seqtab_PICRUSt_INF <- as(otu_table(pig_mouse_comp_PICRUSt_INF_ps), "matrix") 

# making FASTA file
uniquesToFasta(seqtab_PICRUSt_INF, fout = "PICRUSt_INF_ASV.fna", ids = colnames(seqtab_PICRUSt_INF))

# write sequence table which would be used as one of the input files for PICRUSt
write.table(t(seqtab_PICRUSt_INF), "PICRUSt_INF_seqtab.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
```

A new terminal is opened and the following steps are performed in order to carry out the PICRUSt2 analysis (based on the PICRUSt2 pipeline described in the tutorial in https://github.com/picrust/picrust2/wiki/PICRUSt2-Tutorial-(v2.2.0-beta)) :

```
conda create -n picrust2 -c bioconda -c conda-forge picrust2=2.2.0_b # install PICRUSt2 as a conda environment

conda activate picrust2 # activate PICRUSt2

## the input files for running PICRUSt2 are "PICRUSt_INF_ASV.fna" (ASVs in FASTA format) and "PICRUSt_INF_seqtab.txt" (seq. table). I made a folder called 'PICRUSt2' and within that folder made a subfolder called 'INF' which had these two input files. The outputs from running the PICRUSt2 commands will be written to this folder.

(picrust2) niroshs-mbp:INF niroshaluthge$ place_seqs.py -s PICRUSt_INF_ASV.fna -o out_INF.tre -p 1 --intermediate intermediate/place_seqs
(picrust2) niroshs-mbp:INF niroshaluthge$ hsp.py -i 16S -t out_INF.tre -o INF_marker_predicted_and_nsti.tsv.gz -p 1 -n 
(picrust2) niroshs-mbp:INF niroshaluthge$ hsp.py -i KO -t out_INF.tre -o INF_KO_predicted.tsv.gz # uses KEGG orthology database

# for the next command, need to convert the '.txt' format "PICRUSt_INF_seqtab.txt" file into a biom format table
(picrust2) niroshs-mbp:INF niroshaluthge$ echo -n "#OTU Table" | cat - PICRUSt_INF_seqtab.txt > PICRUSt_INF_seqtab-biom-table.txt 
(picrust2) niroshs-mbp:INF niroshaluthge$ biom convert -i PICRUSt_INF_seqtab-biom-table.txt -o PICRUSt_INF_seqtab.biom --table-type="OTU table" --to-hdf5

(picrust2) niroshs-mbp:INF niroshaluthge$ metagenome_pipeline.py -i PICRUSt_INF_seqtab.biom -m INF_marker_predicted_and_nsti.tsv.gz -f INF_KO_predicted.tsv.gz -o INF_KO_metagenome_out --strat_out
0 of 171 ASVs were above the max NSTI cut-off of 2.0 and were removed.

# Within the resulting 'INF_KO_metagenome_out/' folder there is a file called 'KO_pred_metagenome_unstrat.tsv.gz' which is very similar to an OTU/ASV table but has KO features instead of ASVs as rows. The columns are the sample IDs. This is the file we need to read into phyloseq. We'll first unzip the file and then convert it into a JSON-format biom table and read it into phyloseq.

(picrust2) niroshs-mbp:INF_KO_metagenome_out niroshaluthge$ gunzip KO_pred_metagenome_unstrat.tsv.gz
(picrust2) niroshs-mbp:INF_KO_metagenome_out niroshaluthge$ biom convert -i KO_pred_metagenome_unstrat.tsv -o INF_KO_pred_metagenome.biom --table-type="OTU table" --to-json
```

Let's read in the PICRUSt output, which is a function-based seqtab in BIOM format, to phyloseq

```{r, echo=TRUE}
picrust_biomfile <- "PICRUSt/INF/INF_KO_pred_metagenome.biom"
picrust_biom = biomformat::read_biom(biom_file = picrust_biomfile) # might need to install 'biomformat' R package
picrust_biom_table <- import_biom(picrust_biom) 
picrust_INF_seqtab <- t(as(otu_table(picrust_biom_table), "matrix")) # Need to transform it back so that the 'taxa' are columns and the 'samples' are rows since that's how phyloseq wants it
dim(picrust_INF_seqtab)
sum(picrust_INF_seqtab) # check that everything is as expected

picrust_INF_ps <- merge_phyloseq(otu_table(picrust_INF_seqtab, taxa_are_rows = FALSE), sample_data(sample_data(pig_mouse_comp_PICRUSt_INF_ps))) # make a new phyloseq object

picrust_INF_ps

picrust_INF_ps.prop <- transform_sample_counts(picrust_INF_ps, function (OTU) OTU/sum(OTU)) # transform sample counts into relative abundances
picrust_INF_ps.prop 

## Let's look at the 'core' KO functions coded for by the INF donor samples. These are defined as those functions detected in all donor inocula

picrust_INF_donor_ps <- subset_samples(picrust_INF_ps, Species == "human")
prevdf_INF_donor = apply(X = otu_table(picrust_INF_donor_ps),
               MARGIN = ifelse(taxa_are_rows(picrust_INF_donor_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_INF_donor = data.frame(Prevalence=prevdf_INF_donor, TotalAbundance=taxa_sums(picrust_INF_donor_ps))
View(prevdf_INF_donor)

core_threshold = nsamples(picrust_INF_donor_ps) # set threshold for the number of samples in which the KO function needs to be present in order for it to be considered as part of the core
core_INF_KOs <- rownames(prevdf_INF_donor)[prevdf_INF_donor$Prevalence >= core_threshold]
core_INF_KOs

# phyloseq object with just the core donor KOs 
picrust_INF_core_KO_ps <- prune_taxa(core_INF_KOs,picrust_INF_ps)
picrust_INF_core_KO_ps # 4163 core KOs identified

sum(otu_table(picrust_INF_core_KO_ps))/sum(otu_table(picrust_INF_ps)) # 99.19% of reads are retained

picrust_INF_core_KO_ps.prop <- prune_taxa(core_INF_KOs,picrust_INF_ps.prop)
picrust_INF_core_KO_ps.prop

# plotting PCoA using Bray-Curtis distances 
bray_ord_picrust_INF <- ordinate(picrust_INF_core_KO_ps.prop, method = "PCoA", distance = "bray")
bray_picrust_INF_core_KO <- plot_ordination(picrust_INF_core_KO_ps.prop, bray_ord_picrust_INF, color = "Species", shape = "Species") + ggtitle("PICRUSt Donor_1 PCoA plot") + theme(plot.title = element_text(hjust = 0.5, size = 10), legend.title = element_text(size = 10), legend.text = element_text(size = 10), axis.title = element_text(size = 10)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species")
bray_picrust_INF_core_KO
```


**Donor_2 (CHLD Donor)**

```{r, echo=TRUE}
pig_mouse_comp_PICRUSt_CHLD_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_no_cecal_ps, Donor=="CHLD2")
pig_mouse_comp_PICRUSt_CHLD_ps <- subset_samples(pig_mouse_comp_PICRUSt_CHLD_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_PICRUSt_CHLD_ps

# Let's keep only those ASVs found in > 1 sample
prevdf_CHLD <- apply(X = otu_table(pig_mouse_comp_PICRUSt_CHLD_ps), MARGIN = ifelse(taxa_are_rows(pig_mouse_comp_PICRUSt_CHLD_ps), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
prevdf_CHLD <- data.frame(Prevalence = prevdf_CHLD, TotalAbundance = taxa_sums(pig_mouse_comp_PICRUSt_CHLD_ps))
View(prevdf_CHLD)

# Retaining only those ASVs found in at least one sample
keepTaxa <- rownames(prevdf_CHLD)[(prevdf_CHLD$Prevalence >1)]
pig_mouse_comp_PICRUSt_CHLD_ps <- prune_taxa(keepTaxa, pig_mouse_comp_PICRUSt_CHLD_ps)
pig_mouse_comp_PICRUSt_CHLD_ps

## getting the necessary input files for PICRUSt
seqtab_PICRUSt_CHLD <- as(otu_table(pig_mouse_comp_PICRUSt_CHLD_ps), "matrix") 

# making FASTA file
uniquesToFasta(seqtab_PICRUSt_CHLD, fout = "PICRUSt_CHLD_ASV.fna", ids = colnames(seqtab_PICRUSt_CHLD))

# write the seq. table as a .txt file which will be used as input for PICRUST2
write.table(t(seqtab_PICRUSt_CHLD), "PICRUSt_CHLD_seqtab.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)

## The PICRUSt2 steps are run similar to how they were done for the INF donor previously

# Let's read in the PICRUSt output, which is a function-based seqtab (which i converted to .biom format) to phyloseq
picrust_biomfile <- "PICRUSt/CHLD/CHLD_KO_pred_metagenome.biom" 
picrust_biom = biomformat::read_biom(biom_file = picrust_biomfile)
picrust_biom_table <- import_biom(picrust_biom) 
picrust_CHLD_seqtab <- t(as(otu_table(picrust_biom_table), "matrix")) # Need to transform it back so that the 'taxa' are columns and the 'samples' are rows since that's how phyloseq wants it
dim(picrust_CHLD_seqtab)
sum(picrust_CHLD_seqtab)

picrust_CHLD_ps <- merge_phyloseq(otu_table(picrust_CHLD_seqtab, taxa_are_rows = FALSE), sample_data(sample_data(pig_mouse_comp_PICRUSt_CHLD_ps)))

picrust_CHLD_ps

picrust_CHLD_ps.prop <- transform_sample_counts(picrust_CHLD_ps, function (OTU) OTU/sum(OTU))
picrust_CHLD_ps.prop

## Let's look at the 'core' KO functions coded for by the CHLD donor samples. These are defined as those functions detected in all donor inocula

picrust_CHLD_donor_ps <- subset_samples(picrust_CHLD_ps, Species == "human")
prevdf_CHLD_donor = apply(X = otu_table(picrust_CHLD_donor_ps),
               MARGIN = ifelse(taxa_are_rows(picrust_CHLD_donor_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_CHLD_donor = data.frame(Prevalence=prevdf_CHLD_donor, TotalAbundance=taxa_sums(picrust_CHLD_donor_ps))
View(prevdf_CHLD_donor)

core_threshold = nsamples(picrust_CHLD_donor_ps) # set threshold for the number of samples in which the function needs to be present in order for it to be considered as part of the core
core_CHLD_KOs <- rownames(prevdf_CHLD_donor)[prevdf_CHLD_donor$Prevalence >= core_threshold]
core_CHLD_KOs

# phyloseq object with just the core donor KOs 
picrust_CHLD_core_KO_ps <- prune_taxa(core_CHLD_KOs,picrust_CHLD_ps)
picrust_CHLD_core_KO_ps # 4426 core KOs identified

sum(otu_table(picrust_CHLD_core_KO_ps))/sum(otu_table(picrust_CHLD_ps)) # 99.80% of reads are retained

picrust_CHLD_core_KO_ps.prop <- prune_taxa(core_CHLD_KOs,picrust_CHLD_ps.prop)
picrust_CHLD_core_KO_ps.prop

# plotting PCoA using Bray-Curtis distances 
bray_ord_picrust_CHLD <- ordinate(picrust_CHLD_core_KO_ps.prop, method = "PCoA", distance = "bray")
bray_picrust_CHLD_core_KO <- plot_ordination(picrust_CHLD_core_KO_ps.prop, bray_ord_picrust_CHLD, color = "Species", shape = "Species") + ggtitle("PICRUSt Donor_2 PCoA plot") + theme(plot.title = element_text(hjust = 0.5, size = 10), legend.title = element_text(size = 10)) + labs(colour = "", shape = "host species")
bray_picrust_CHLD_core_KO
```


**Donor_3 (ADLT Donor)**

```{r, echo=TRUE}
pig_mouse_comp_PICRUSt_ADLT_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_no_cecal_ps, Donor=="ADLT2")
pig_mouse_comp_PICRUSt_ADLT_ps <- subset_samples(pig_mouse_comp_PICRUSt_ADLT_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))

# removing sample '105' since it's an outlier
pig_mouse_comp_PICRUSt_ADLT_ps <- subset_samples(pig_mouse_comp_PICRUSt_ADLT_ps, SampleID != '105')
pig_mouse_comp_PICRUSt_ADLT_ps

# Let's keep only those ASVs found in > 1 sample
prevdf_ADLT <- apply(X = otu_table(pig_mouse_comp_PICRUSt_ADLT_ps), MARGIN = ifelse(taxa_are_rows(pig_mouse_comp_PICRUSt_ADLT_ps), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
prevdf_ADLT <- data.frame(Prevalence = prevdf_ADLT, TotalAbundance = taxa_sums(pig_mouse_comp_PICRUSt_ADLT_ps))
View(prevdf_ADLT)

# Retaining only those ASVs found in at least one sample
keepTaxa <- rownames(prevdf_ADLT)[(prevdf_ADLT$Prevalence >1)]
pig_mouse_comp_PICRUSt_ADLT_ps <- prune_taxa(keepTaxa, pig_mouse_comp_PICRUSt_ADLT_ps)
pig_mouse_comp_PICRUSt_ADLT_ps

## getting the necessary input files for PICRUSt
seqtab_PICRUSt_ADLT <- as(otu_table(pig_mouse_comp_PICRUSt_ADLT_ps), "matrix") 

# making FASTA file
uniquesToFasta(seqtab_PICRUSt_ADLT, fout = "PICRUSt_ADLT_ASV.fna", ids = colnames(seqtab_PICRUSt_ADLT))

write.table(t(seqtab_PICRUSt_ADLT), "PICRUSt_ADLT_seqtab.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)

## The PICRUSt2 steps are run similar to how they were done for the INF donor previously

# Let's read in the PICRUSt output, which is a function-based seqtab (which i converted to .biom format) to phyloseq
picrust_biomfile <- "PICRUSt/ADLT/ADLT_KO_metagenome_out/ADLT_KO_pred_metagenome.biom" 
picrust_biom = biomformat::read_biom(biom_file = picrust_biomfile)
picrust_biom_table <- import_biom(picrust_biom) 
picrust_ADLT_seqtab <- t(as(otu_table(picrust_biom_table), "matrix")) # Need to transform it back so that the 'taxa' are columns and the 'samples' are rows since that's how phyloseq wants it
dim(picrust_ADLT_seqtab)
sum(picrust_ADLT_seqtab)

picrust_ADLT_ps <- merge_phyloseq(otu_table(picrust_ADLT_seqtab, taxa_are_rows = FALSE), sample_data(sample_data(pig_mouse_comp_PICRUSt_ADLT_ps)))

picrust_ADLT_ps

picrust_ADLT_ps.prop <- transform_sample_counts(picrust_ADLT_ps, function (OTU) OTU/sum(OTU))
picrust_ADLT_ps.prop

## Let's look at the 'core' KO functions coded for by the ADLT donor samples. These are defined as those functions detected in all donor inocula

picrust_ADLT_donor_ps <- subset_samples(picrust_ADLT_ps, Species == "human")
prevdf_ADLT_donor = apply(X = otu_table(picrust_ADLT_donor_ps),
               MARGIN = ifelse(taxa_are_rows(picrust_ADLT_donor_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ADLT_donor = data.frame(Prevalence=prevdf_ADLT_donor, TotalAbundance=taxa_sums(picrust_ADLT_donor_ps))
View(prevdf_ADLT_donor)

core_threshold = nsamples(picrust_ADLT_donor_ps) # set threshold for the number of samples in which the KO function needs to be present in order for it to be considered as part of the core
core_ADLT_KOs <- rownames(prevdf_ADLT_donor)[prevdf_ADLT_donor$Prevalence >= core_threshold]
core_ADLT_KOs

# phyloseq object with just the core donor KOs 
picrust_ADLT_core_KO_ps <- prune_taxa(core_ADLT_KOs,picrust_ADLT_ps)
picrust_ADLT_core_KO_ps # 4470 core KOs identified

sum(otu_table(picrust_ADLT_core_KO_ps))/sum(otu_table(picrust_ADLT_ps)) # 99.65% of reads are retained

picrust_ADLT_core_KO_ps.prop <- prune_taxa(core_ADLT_KOs,picrust_ADLT_ps.prop)
picrust_ADLT_core_KO_ps.prop

# plotting PCoA using Bray-Curtis distances 
bray_ord_picrust_ADLT <- ordinate(picrust_ADLT_core_KO_ps.prop, method = "PCoA", distance = "bray")
bray_picrust_ADLT_core_KO <- plot_ordination(picrust_ADLT_core_KO_ps.prop, bray_ord_picrust_ADLT, color = "Species", shape = "Species") + ggtitle("PICRUSt Donor_3 PCoA plot") + theme(plot.title = element_text(hjust = 0.5, size = 10), legend.title = element_text(size = 10)) + labs(colour = "", shape = "host species")
bray_picrust_ADLT_core_KO
```


**Donor_4 (ELD Donor)** 

```{r, echo=TRUE}
pig_mouse_comp_PICRUSt_ELD_ps <- subset_samples(pig_mouse_comp_PICRUSt_comp_no_cecal_ps, Donor=="ELD2")
pig_mouse_comp_PICRUSt_ELD_ps <- subset_samples(pig_mouse_comp_PICRUSt_ELD_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_PICRUSt_ELD_ps

# Let's keep only those ASVs found in > 1 sample
prevdf_ELD <- apply(X = otu_table(pig_mouse_comp_PICRUSt_ELD_ps), MARGIN = ifelse(taxa_are_rows(pig_mouse_comp_PICRUSt_ELD_ps), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
prevdf_ELD <- data.frame(Prevalence = prevdf_ELD, TotalAbundance = taxa_sums(pig_mouse_comp_PICRUSt_ELD_ps))
View(prevdf_ELD)

# Retaining only those ASVs found in at least one sample
keepTaxa <- rownames(prevdf_ELD)[(prevdf_ELD$Prevalence >1)]
pig_mouse_comp_PICRUSt_ELD_ps <- prune_taxa(keepTaxa, pig_mouse_comp_PICRUSt_ELD_ps)
pig_mouse_comp_PICRUSt_ELD_ps

## getting the necessary input files for PICRUSt
seqtab_PICRUSt_ELD <- as(otu_table(pig_mouse_comp_PICRUSt_ELD_ps), "matrix") 

# making FASTA file
uniquesToFasta(seqtab_PICRUSt_ELD, fout = "PICRUSt_ELD_ASV.fna", ids = colnames(seqtab_PICRUSt_ELD))

write.table(t(seqtab_PICRUSt_ELD), "PICRUSt_ELD_seqtab.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)

## The PICRUSt2 steps are run similar to how they were done for the INF donor previously

# Let's read in the PICRUSt output, which is a function-based seqtab (which i converted to .biom format) to phyloseq
picrust_biomfile <- "PICRUSt/ELD/ELD_KO_pred_metagenome.biom" 
picrust_biom = biomformat::read_biom(biom_file = picrust_biomfile)
picrust_biom_table <- import_biom(picrust_biom) 
picrust_ELD_seqtab <- t(as(otu_table(picrust_biom_table), "matrix")) # Need to transform it back so that the 'taxa' are columns and the 'samples' are rows since that's how phyloseq wants it
dim(picrust_ELD_seqtab)
sum(picrust_ELD_seqtab)

picrust_ELD_ps <- merge_phyloseq(otu_table(picrust_ELD_seqtab, taxa_are_rows = FALSE), sample_data(sample_data(pig_mouse_comp_PICRUSt_ELD_ps)))

picrust_ELD_ps

picrust_ELD_ps.prop <- transform_sample_counts(picrust_ELD_ps, function (OTU) OTU/sum(OTU))
picrust_ELD_ps.prop

## Let's look at the 'core' KO functions coded for by the ELD donor samples. These are defined as those functions detected in all donor inocula

picrust_ELD_donor_ps <- subset_samples(picrust_ELD_ps, Species == "human")
prevdf_ELD_donor = apply(X = otu_table(picrust_ELD_donor_ps),
               MARGIN = ifelse(taxa_are_rows(picrust_ELD_donor_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ELD_donor = data.frame(Prevalence=prevdf_ELD_donor, TotalAbundance=taxa_sums(picrust_ELD_donor_ps))
View(prevdf_ELD_donor)

core_threshold = nsamples(picrust_ELD_donor_ps) # set threshold for the number of samples in which the KO function needs to be present in order for it to be considered as part of the core
core_ELD_KOs <- rownames(prevdf_ELD_donor)[prevdf_ELD_donor$Prevalence >= core_threshold]
core_ELD_KOs

# phyloseq object with just the core donor KOs 
picrust_ELD_core_KO_ps <- prune_taxa(core_ELD_KOs,picrust_ELD_ps)
picrust_ELD_core_KO_ps # 4329 core KOs identified

sum(otu_table(picrust_ELD_core_KO_ps))/sum(otu_table(picrust_ELD_ps)) # 99.59% of reads are retained

picrust_ELD_core_KO_ps.prop <- prune_taxa(core_ELD_KOs,picrust_ELD_ps.prop)
picrust_ELD_core_KO_ps.prop

# plotting PCoA using Bray-Curtis distances 
bray_ord_picrust_ELD <- ordinate(picrust_ELD_core_KO_ps.prop, method = "PCoA", distance = "bray")
bray_picrust_ELD_core_KO <- plot_ordination(picrust_ELD_core_KO_ps.prop, bray_ord_picrust_ELD, color = "Species", shape = "Species") + ggtitle("PICRUSt Donor_4 PCoA plot") + theme(plot.title = element_text(hjust = 0.5, size = 10), legend.title = element_text(size = 10)) + labs(colour = "", shape = "host species")
bray_picrust_ELD_core_KO
```



#################################################################################
DATA ANALYSIS FOR SECOND STUDY
#################################################################################

The work below shows the data analysis steps for the **second study** which was performed with a different set of donors. This includes the follow-up comparison between HMA piglets and HMA mice for the establishment of a second infant donor microbiota (Donor_8) as well as the evaluation of the establishment of core donor taxa from Donor_5, Donor_6, and Donor_7 in a different group of HMA mice (C3H/HeN).

The first sequencing run had **432** .fastq files corresponding to **216** samples. These fastq files were in the folder 'round_2_JUNE_fastq_files' :

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

#change to required work directory
setwd("/Users/niroshaluthge/Desktop/NDA/manuscripts/pig_mouse_comparison_study/supp_materials_data_analysis")
getwd()

MiSeq_path <- "./supp_materials_data_analysis/round_2_JUNE_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(225,70), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_round_2_JUNE_FWD_only.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 16676415 input reads and 14652855 output reads, so we have retained ~ 87.9% of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

# make sequence table from just the Forward reads (Since we'll be using just the forward reads from the 'AUGUST' run; see further below in the next section):
seqtabAll_round_2_JUNE_FWD <- makeSequenceTable(dadaFs)
table(nchar(getSequences(seqtabAll_round_2_JUNE_FWD)))
sum(seqtabAll_round_2_JUNE_FWD) # There were 14487858
dim(seqtabAll_round_2_JUNE_FWD)
saveRDS(seqtabAll_round_2_JUNE_FWD, "seqtabAll_round_2_JUNE_FWD.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_round_2_JUNE <- cbind(out, sapply(dadaFs, getN), rowSums(seqtabAll_round_2_JUNE_FWD))
colnames(track_round_2_JUNE) <- c("input", "filtered", "denoisedF", "seqtable")
rownames(track_round_2_JUNE) <- sampleNames
head(track_round_2_JUNE)
write.table(track_round_2_JUNE, 'tracking_reads_through_pipeline_round_2_JUNE_FWD.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```


The second sequencing run had **194** .fastq files corresponding to **97** samples. These fastq files were in the folder 'round_2_AUGUST_fastq_files' :

```{r, echo=TRUE}
MiSeq_path <- "./supp_materials_data_analysis/round_2_AUGUST_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

# the reverse read quality is really bad so we'll use only the forward reads

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, truncLen = 225, maxN=0, maxEE=2, truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_round_2_AUGUST_FWD_only.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 5463278 input reads and 4483935 output reads, so we have retained ~ 82% of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
names(derepFs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
plotErrors(errF)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaFs[1] 

seqtabAll_round_2_AUGUST_FWD <- makeSequenceTable(dadaFs) # 4416740 sequences retained
table(nchar(getSequences(seqtabAll_round_2_AUGUST_FWD)))
sum(seqtabAll_round_2_AUGUST_FWD)
dim(seqtabAll_round_2_AUGUST_FWD)

#Store seqtabAll2 from round1 August sequencing run as a .rds file
saveRDS(seqtabAll_round_2_AUGUST_FWD, "seqtabAll_round_2_AUGUST_FWD.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_round_2_AUGUST <- cbind(out, sapply(dadaFs, getN), rowSums(seqtabAll_round_2_AUGUST_FWD))
colnames(track_round_2_AUGUST) <- c("input", "filtered", "denoisedF", "seqtable")
rownames(track_round_2_AUGUST) <- sampleNames
head(track_round_2_AUGUST)
write.table(track_round_2_AUGUST, 'tracking_reads_through_pipeline_round_2_AUGUST_FWD.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```


Let's merge the sequence tables made from just the forward reads ('FWD') for the 'June' and 'August' sequencing runs:

```{r,echo=TRUE}
seqtab_JUNE_FWD <- readRDS("seqtabAll_round_2_JUNE_FWD.rds")
dim(seqtab_JUNE_FWD)
seqtab_AUGUST_FWD <- readRDS("seqtabAll_round_2_AUGUST_FWD.rds")
dim(seqtab_AUGUST_FWD)

# merging the two sequence tables
merged_seqtabs_FWD <- mergeSequenceTables(seqtab_JUNE_FWD,seqtab_AUGUST_FWD)
dim(merged_seqtabs_FWD) 
```

Removing chimeras :

```{r,echo=TRUE}
merged_seqtabs_FWD_NoC <- removeBimeraDenovo(merged_seqtabs_FWD, method = "consensus", multithread = TRUE, verbose = TRUE)
dim(merged_seqtabs_FWD_NoC) # 8115 ASVs removed as bimeras
sum(merged_seqtabs_FWD_NoC)/sum(merged_seqtabs_FWD) # chimeras accounted for ~ 15% of total reads
```

Assigning taxonomy to the remaining ASVs using the SILVA database files :

```{r, echo=TRUE}
fastaRef <- "./silva_nr_v138_train_set.fa" 
taxTab_r2 <- assignTaxonomy(merged_seqtabs_FWD_NoC, refFasta = fastaRef, multithread = TRUE)
taxTab_r2 <- addSpecies(taxTab_r2, "./silva_species_assignment_v138.fa")
unname(head(taxTab_r2))
head(taxTab_r2)
write.table(taxTab_r2, "round_2_full_taxa_table.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
```

The next step is to make the phylogenetic tree. This requires a FASTA file from the ASV sequences which will be generated using the following code from https://forum.qiime2.org/t/importing-dada2-and-phyloseq-objects-to-qiime-2/4683 :

```{r, echo=TRUE}
uniquesToFasta(merged_seqtabs_FWD_NoC, fout = 'study_2_ASVs.fna', ids = colnames(merged_seqtabs_FWD_NoC))
```

Using the 'study_2_ASVs.fna' FASTA file as the input file, a phylogenetic tree was made using the SILVA database and MOTHUR as described previously in this markdown file. Reading in the phylogenetic tree made in MOTHUR to R and phyloseq environment :

```{r, echo = TRUE}
tree_file_r2 <- 'study_2_ASVs.phylip.tre'
phylo_tree_r2 <- read_tree(tree_file_r2)
```

Let's read in the mapping file:

```{r, echo=TRUE}
mapping_file <- read.table('mapping_file_round_2.txt', sep = "\t", header = TRUE, comment.char = "") # has been uploaded to GitHub
row.names(mapping_file) <- as.character(mapping_file[,1])
View(mapping_file)
```

Now we can merge the sequence table with the mapping file, phylogenetic tree, and the taxonomy table to obtain the main phyloseq object:

```{r, echo=TRUE}
study_2_ps1 <- phyloseq(otu_table(merged_seqtabs_FWD_NoC, taxa_are_rows = FALSE), sample_data(mapping_file), phy_tree(phylo_tree_r2), tax_table(taxTab_r2))
study_2_ps1 # there are 3632 ASVs across 313 samples
#write.table(sample_data(supp_mater_ps1), "mapping_file_supp_materials-2.txt", sep = "\t", row.names = F, col.names = T, quote = F)
```

Removing potential contaminant ASVs using the 'decontam' program of phyloseq:
 
```{r, echo=TRUE}
library(decontam); packageVersion('decontam')

sample_data(study_2_ps1)$is.neg <- sample_data(study_2_ps1)$Sample_or_Control == "negative_control"

# identifying potential contaminants using the 'prevalence' method
contamdf.prev_batch <- isContaminant(study_2_ps1, method = "prevalence", neg = "is.neg", batch = sample_data(study_2_ps1)$Batch, batch.combine = "minimum")
table(contamdf.prev_batch$contaminant) # 11 contaminant ASVs identified 

# removing contaminants
study_2_ps1_no_contams <- prune_taxa(!contamdf.prev_batch$contaminant, study_2_ps1)
study_2_ps1_no_contams # 3621 ASVS are retained

sum(otu_table(study_2_ps1_no_contams))/sum(otu_table(study_2_ps1)) # >99.99% of reads remain after removing potential contaminants
```

Removing negative control samples :

```{r, echo=TRUE}
study_2_no_negs_ps1 <- subset_samples(study_2_ps1_no_contams, Sample_or_Control != 'negative_control')
study_2_no_negs_ps1
```

Filtering out ASVs that are classified in the kingdoms 'Archaea' or 'Eukaryota' as well as those ASVs that have a Mitochondrial classification :

```{r, echo=TRUE}
remov_kingdoms <- c('Archaea', 'Eukaryota')
study_2_no_negs_ps1_taxa_filt <- subset_taxa(study_2_no_negs_ps1, !Kingdom %in% remov_kingdoms) 
#(Family != 'Mitochondria' | is.na(Family))) 

# remove Mitochondrial sequences
study_2_no_negs_ps1_taxa_filt <- subset_taxa(study_2_no_negs_ps1_taxa_filt, Family != 'Mitochondria' | is.na(Family))
study_2_no_negs_ps1_taxa_filt

# remove any ASV which has a non-Bacterial classification
study_2_no_negs_ps1_taxa_filt <- subset_taxa(study_2_no_negs_ps1_taxa_filt, Kingdom == 'Bacteria') 
study_2_no_negs_ps1_taxa_filt

# removing non-Melainabacterial Cyanobacterial ASVs
Cyano_ps <- subset_taxa(study_2_no_negs_ps1_taxa_filt, Phylum == 'Cyanobacteria') # Isolating the Cyanobacterial ASVs
remov_Cyano_ps <- subset_taxa(Cyano_ps, Class != 'Melainabacteria' | is.na(Class)) # identifying Cyanobacterial ASVs that are not 'Melainabacteria'
Cyano_to_remov <- taxa_names(remov_Cyano_ps)

# Filtering out the non-Melainabacterial Cyanobacterial ASVs
alltaxa = taxa_names(study_2_no_negs_ps1_taxa_filt)
goodtaxa = alltaxa[!(alltaxa %in% Cyano_to_remov)]
study_2_no_negs_ps1_taxa_filt_no_Cyano <- prune_taxa(goodtaxa, study_2_no_negs_ps1_taxa_filt)
study_2_no_negs_ps1_taxa_filt_no_Cyano

write.table(tax_table(study_2_no_negs_ps1_taxa_filt_no_Cyano), 'round_2_taxa_table_after_taxa_filtering.txt', sep = "\t")
```

Computing prevalence of each ASV to filter out those that are present in only a single sample :

```{r, echo=TRUE}
# Calculating prevalence for each ASV
prevdf <- apply(X = otu_table(study_2_no_negs_ps1_taxa_filt_no_Cyano), MARGIN = ifelse(taxa_are_rows(study_2_no_negs_ps1_taxa_filt_no_Cyano), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
View(prevdf)
prevdf_with_taxa <- data.frame(Prevalence = prevdf, TotalAbundance = taxa_sums(study_2_no_negs_ps1_taxa_filt_no_Cyano), tax_table(study_2_no_negs_ps1_taxa_filt_no_Cyano))
View(prevdf_with_taxa)

# Retaining only those ASVs found in more than one sample
keepTaxa <- rownames(prevdf_with_taxa)[(prevdf_with_taxa$Prevalence > 1)]
round_2_ps_all_filtd <- prune_taxa(keepTaxa, study_2_no_negs_ps1_taxa_filt_no_Cyano)
round_2_ps_all_filtd # We've filtered out 2754 ASVs

sum(otu_table(round_2_ps_all_filtd))/sum(otu_table(study_2_no_negs_ps1_taxa_filt_no_Cyano)) # However, those 2754 ASVs only accounted for ~0.2% of reads
```


Looking at the distribution of reads per sample :

```{r, echo=TRUE}
reads_per_sample_df <- data.frame(Reads = sample_sums(round_2_ps_all_filtd)) 
reads_per_sample_df$sampleID <- rownames(reads_per_sample_df)
View(reads_per_sample_df)
reads_per_sample_df <- reads_per_sample_df[,c(2,1)] # reorder columns
reads_per_sample_df_sorted <- reads_per_sample_df[order(reads_per_sample_df$Reads),] # order samples by read count (ascending)
View(reads_per_sample_df_sorted) # check that it looks right
write.table(reads_per_sample_df_sorted, 'Reads_per_sample_after_all_the_filtering_steps_round_2.txt', sep = "\t", col.names = T, row.names = F, quote = F)

# Depth information
min_read_depth <- min(sample_sums(round_2_ps_all_filtd))
min_read_depth # 3,174
max_read_depth <- max(sample_sums(round_2_ps_all_filtd))
max_read_depth # 127,180 reads
avg_read_depth <- sum(otu_table(round_2_ps_all_filtd))/nsamples(round_2_ps_all_filtd)
avg_read_depth # 52,336 reads/sample
```

Filtering out samples which have < 10,000 total reads :

```{r, echo=TRUE}
round_2_complete_ps <- prune_samples(sample_sums(round_2_ps_all_filtd) > 10000, round_2_ps_all_filtd)
round_2_complete_ps  # 302 samples remain

# Save as an R serial object for use in future analyses
saveRDS(round_2_complete_ps, "round_2_complete_ps.rds")
round_2_complete_ps <- readRDS("round_2_complete_ps.rds") # 'round_2_complete_ps.rds' has been uploaded to GitHub for convenience
```


##**Alpha diversity comparisons for donors**##

```{r, echo=TRUE}
round_2_human_donors_ps <- subset_samples(study_2_no_negs_ps1_taxa_filt_no_Cyano, Species == 'human') # Using a minimally filtered ASV table
round_2_human_donors_ps

## keep only inocula used for colonizing 
round_2_human_donors_ps <- subset_samples(round_2_human_donors_ps, Description != "unused inoculum for mice")
#write.table(sample_data(round_2_human_donors_ps), "round_2_donors_mapping_file.txt", sep = "\t", row.names = T, col.names = NA, quote = F)

#unused_inocula <- c("109", "121", "13", "133", "193", "25", "37")
#round_2_human_donor_inocula_ps <- subset_samples(round_2_human_donors_ps, !SampleID %in% unused_inocula)
#sample_names(round_2_human_donor_inocula_ps)

# alpha diversity comparison of donor samples (Shannon index)
# Since it is recommended to calculate alpha diversity before filtering out any taxa, we'll use the sequence table with just the negative control samples removed as the starting point for this analysis

results_donors_r2 = estimate_richness(round_2_human_donors_ps, measures = "Shannon")
results_donors_r2_df <- as.data.frame(results_donors_r2)
results_donors_r2_df$SampleID <- rownames(results_donors_r2_df)
results_donors_r2_df <- results_donors_r2_df[c(2,1)]
View(results_donors_r2_df)

donors_r2_alpha_div_df <- merge(results_donors_r2_df, mapping_file, by='SampleID')
View(donors_r2_alpha_div_df) # check that it looks correct

pdf("donor_inocula_alpha_diversity_r2.pdf", title = "Additional_file#")
theme_set(theme_gray())
p1 <- ggplot(donors_r2_alpha_div_df, aes(x=Donor, y=Shannon, fill=Donor)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits = c("child", "adult", "elderly", "infant"), labels = c("Donor_5", "Donor_6", "Donor_7", "Donor_8")) + theme(legend.position = "none", axis.text = element_text(size = 10), axis.title = element_text(size = 12)) + labs(x = "Donor", y = "Shannon index")
p2
dev.off()
                            
# Looking at statistical significance between different donor inocula
d = sample_data(round_2_human_donors_ps)

# calculate stats using Wilcoxon test
INF_r2 = results_donors[d[,'Donor'] == 'infant',]
CHLD_r2 = results_donors[d[,'Donor'] == 'child',]
ADLT_r2 = results_donors[d[,'Donor'] == 'adult',]
ELD_r2 = results_donors[d[,'Donor'] == 'elderly',]
wilcox.test(INF_r2, CHLD_r2)
wilcox.test(INF_r2, ADLT_r2)
wilcox.test(INF_r2, ELD_r2)
wilcox.test(CHLD_r2, ADLT_r2)
wilcox.test(CHLD_r2, ELD_r2)
wilcox.test(ADLT_r2, ELD_r2)
```


##**Performing comparisons between HMA mice and HMA piglets for second INF donor (Donor_8)**##

```{r, echo=TRUE}
round_2_INF_ps <- subset_samples(round_2_complete_ps, Donor == 'infant')
round_2_INF_ps

# removing sample '193' which is an unused inoculum
round_2_INF_ps <- subset_samples(round_2_INF_ps, SampleID != '193')
round_2_INF_ps

# transforming samples counts prior to carrying out performing ordination
round_2_INF_ps.prop <- transform_sample_counts(round_2_INF_ps, function(x){x/sum(x)})
round_2_INF_ps.prop

# removing piglet samples on milk diet
round_2_INF_no_milk_ps.prop <- subset_samples(round_2_INF_ps.prop, Diet != 'milk' | is.na(Diet))
round_2_INF_no_milk_ps.prop

# plotting using Bray-Curtis distances
bray_ord_INF <- ordinate(round_2_INF_no_milk_ps.prop, method="PCoA", distance = "bray")
bray_pcoa_INF_round_2 <- plot_ordination(round_2_INF_no_milk_ps.prop, bray_ord_INF, color="Species", shape = "Species") + ggtitle("Donor_8 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species")
bray_pcoa_INF_round_2
```


Let's look at donor taxa that are found in all four donor inocula (core ASVs for INF donor)

```{r, echo=TRUE}
round_2_INF_DONORS_ps <- subset_samples(round_2_INF_ps, Species == 'human')
round_2_INF_DONORS_ps
sample_names(round_2_INF_DONORS_ps)

# Calculating prevalence to determine which ASVs are found in all 4 donor inocula
prevdf_INF_round_2_donors = apply(X = otu_table(round_2_INF_DONORS_ps),
               MARGIN = ifelse(taxa_are_rows(round_2_INF_DONORS_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_INF_round_2_donors = data.frame(Prevalence=prevdf_INF_round_2_donors, TotalAbundance=taxa_sums(round_2_INF_DONORS_ps))
View(prevdf_INF_round_2_donors)

core_threshold = nsamples(round_2_INF_DONORS_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_INF_round_2_donors_taxa <- rownames(prevdf_INF_round_2_donors)[prevdf_INF_round_2_donors$Prevalence == core_threshold]
core_INF_round_2_donors_taxa # 20 core ASVs identified that were detected in all 4 donor inocula

round_2_INF_DONORS_core_ps <- prune_taxa(core_INF_round_2_donors_taxa, round_2_INF_DONORS_ps)
round_2_INF_DONORS_core_ps
sum(otu_table(round_2_INF_DONORS_core_ps))/sum(otu_table(round_2_INF_DONORS_ps)) # the 20 core ASVs account for ~95.4% of donor reads

# Let's see what percentage of reads of the 2 INF inocula used to inoculate the mice was represented by these 20 core ASVs
INF_r2_mouse_inocula <- c('49', '145')
INF_r2_mouse_inocula_ps <- prune_samples(INF_r2_mouse_inocula, round_2_INF_DONORS_ps)
INF_r2_mouse_inocula_ps

INF_r2_mouse_inocula_core_ps <- prune_taxa(core_INF_round_2_donors_taxa, INF_r2_mouse_inocula_ps)
INF_r2_mouse_inocula_core_ps # only the 20 core ASVs are retained

sum(otu_table(INF_r2_mouse_inocula_core_ps))/sum(otu_table(INF_r2_mouse_inocula_ps)) # 93.3% of reads are accounted for by the core ASVs

# Let's see what percentage of reads of the 2 INF inocula used to inoculate the piglets was represented by these 20 core ASVs
INF_r2_piglet_inocula <- c('1', '97')
INF_r2_piglet_inocula_ps <- prune_samples(INF_r2_piglet_inocula, round_2_INF_DONORS_ps)
INF_r2_piglet_inocula_ps

INF_r2_piglet_inocula_core_ps <- prune_taxa(core_INF_round_2_donors_taxa, INF_r2_piglet_inocula_ps)
INF_r2_piglet_inocula_core_ps # only the 20 core ASVs are retained

sum(otu_table(INF_r2_piglet_inocula_core_ps))/sum(otu_table(INF_r2_piglet_inocula_ps)) # 97.4% of reads are accounted for by the core ASVs
```

Let's see how the establishment of core INF ASVs compare among the two animal models :

```{r, echo=TRUE}
round_2_INF_no_milk_ps.prop

# Retaining only the core INF ASVs
round_2_INF_no_milk_core_ps.prop <- prune_taxa(core_INF_round_2_donors_taxa, round_2_INF_no_milk_ps.prop)
round_2_INF_no_milk_core_ps.prop

# PCoA plotting using unweighted unifrac distances
unw_ord_INF_core <- ordinate(round_2_INF_no_milk_core_ps.prop, method = "PCoA", distance = "unifrac", weighted = F)
unwunifrac_INF_round_2_core <- plot_ordination(round_2_INF_no_milk_core_ps.prop, unw_ord_INF_core, color = "Species", shape = "Species") + ggtitle("Infant donor core PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5), legend.title = element_text(size = 10)) + labs(colour = " ", shape = "host species")
unwunifrac_INF_round_2_core 

bray_pcoa_INF_round_2 <- plot_ordination(round_2_INF_no_milk_ps.prop, bray_ord_INF, color="Species", shape = "Species") + ggtitle("Donor_8 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species")

bray_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, bray_ord_INF2_core, color = "Species", shape = "Species") + ggtitle("Donor_1 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species") # + scale_color_manual(values = c("orange", "green4", "dodgerblue3")) #labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.position = "none", axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species") #theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + labs(colour = "", shape = "host species")
bray_pcoa_INF2_core

# plotting using Bray-Curtis distances
bray_ord_INF_core <- ordinate(round_2_INF_no_milk_core_ps.prop, method="PCoA", distance = "bray")
bray_pcoa_INF_round_2_core <- plot_ordination(round_2_INF_no_milk_core_ps.prop, bray_ord_INF_core, color="Species", shape = "Species") + ggtitle("Donor_8 PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 8), legend.title = element_text(size = 8), legend.text = element_text(size = 8), axis.title = element_text(size = 8)) + geom_point(aes(shape = Species, color = Species)) + labs(color = "host species", shape = "host species")
bray_pcoa_INF_round_2_core
```


**Core taxa establishment for Donor_8 (INF)**

Establishment in GF mice :

```{r, echo=TRUE}
INF_round_2_core_ps <- prune_taxa(core_INF_round_2_donors_taxa, round_2_INF_ps)
INF_round_2_core_ps

round_2_INF_ps.prop <- transform_sample_counts(round_2_INF_ps, function(x) x/sum(x)) # converting to relative abundances
round_2_INF_core_ps.prop <- prune_taxa(core_INF_round_2_donors_taxa, round_2_INF_ps.prop) # retaining only the core ASVs
 
# Checking how many of the 20 Donor_8 core ASVs were detected in at least one of the mouse samples
# subsetting phyloseq object to only include mice
INF_round_2_core_mice_ps <- subset_samples(INF_round_2_core_ps, Species == "mouse")
INF_round_2_core_mice_ps

prevdf_INF_round_2_donor_taxa_detect_mice <- makePrevDF(INF_round_2_core_mice_ps, 1)
prevdf_INF_round_2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_INF_round_2_donor_taxa_detect_mice[1])
View(prevdf_INF_round_2_donor_taxa_detect_mice_table)
colonized_prevdf_INF_round_2_donor_mice <- prevdf_INF_round_2_donor_taxa_detect_mice_table[prevdf_INF_round_2_donor_taxa_detect_mice_table$Prevalence >= 1,]
View(colonized_prevdf_INF_round_2_donor_mice)
all_INF_round_2_taxa_colonizing_mice <-prevdf_INF_round_2_donor_taxa_detect_mice[2]
all_INF_round_2_taxa_colonizing_mice <- unlist(all_INF_round_2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_INF_round_2_taxa_colonizing_mice # 16 of the 20 core Donor_8 ASVs have colonized the mice 

Donor_8_core_taxa_colonizing_mice_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_mice,INF_round_2_core_mice_ps)
Donor_8_core_taxa_colonizing_mice_ps

tax_mice_donor_8 <- as(tax_table(Donor_8_core_taxa_colonizing_mice_ps), "matrix")
tax_cols <- colnames(tax_mice_donor_8)
tax_mice_donor_8 <- as.data.frame(tax_mice_donor_8)
View(tax_mice_donor_8)
tax_mice_donor_8$taxonomy <- do.call(paste, c(tax_mice_donor_8[tax_cols], sep = ";"))
for (co in tax_cols) tax_mice_donor_8[co] <- NULL
write.table(tax_mice_donor_8, "taxa_colonizing_mice_donor_8.txt", quote = FALSE, col.names = FALSE, sep = "\t")

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF_round_2_core_mice_ps
INF_round_2_HMA_mice_ps_48HRS <- subset_samples(INF_round_2_core_mice_ps, Description == '2_days_post_inoculation')
INF_round_2_HMA_mice_ps_48HRS # 8 samples at this time point

prevdf_48HRS <- makePrevDF(INF_round_2_HMA_mice_ps_48HRS,5) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 5,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF_round_2_HMA_mice_ps_1week <- subset_samples(INF_round_2_core_mice_ps, Description == '7_days_post_inoculation')
INF_round_2_HMA_mice_ps_1week # there are 10 mouse fecal samples

prevdf_1week <- makePrevDF(INF_round_2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF_round_2_HMA_mice_ps_2weeks <- subset_samples(INF_round_2_core_mice_ps, Description == '14_days_post_inoculation')
INF_round_2_HMA_mice_ps_2weeks # there are 9 samples at this time point
 
prevdf_2weeks <- makePrevDF(INF_round_2_HMA_mice_ps_2weeks,5)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 5,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF_round_2_HMA_mice_ps_3weeks <- subset_samples(INF_round_2_core_mice_ps, Description == '21_days_post_inoculation')
INF_round_2_HMA_mice_ps_3weeks # 10 samples at this time point

prevdf_3weeks <- makePrevDF(INF_round_2_HMA_mice_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 6,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF_round_2_HMA_mice_ps_4weeks <- subset_samples(INF_round_2_core_mice_ps, Description == '28_days_post_inoculation')
INF_round_2_HMA_mice_ps_4weeks # 10 samples at this time point

prevdf_4weeks <- makePrevDF(INF_round_2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF_round_2_HMA_mice_ps_5weeks <- subset_samples(INF_round_2_core_mice_ps, Description == '35_days_post_inoculation')
INF_round_2_HMA_mice_ps_5weeks # 10 mouse samples at this time point

prevdf_5weeks <- makePrevDF(INF_round_2_HMA_mice_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 6,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF_round_2_HMA_mice_ps_40days <- subset_samples(INF_round_2_core_mice_ps, Description == 'final fecal collection')
INF_round_2_HMA_mice_ps_40days # 10 mouse samples at this time point

prevdf_40days <- makePrevDF(INF_round_2_HMA_mice_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 6,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

merged_prev_df_INF_round_2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_INF_round_2_mice) # All columns considered factors
merged_prev_df_INF_round_2_mice[2:8] <- lapply(merged_prev_df_INF_round_2_mice[2:8], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_INF_round_2_mice)
View(merged_prev_df_INF_round_2_mice)

merged_prev_df_INF_round_2_mice[is.na(merged_prev_df_INF_round_2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF_round_2_mice$total <- rowSums(merged_prev_df_INF_round_2_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF_round_2_mice)
merged_prev_df_INF_round_2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_INF_round_2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF_round_2_mice)
write.table(merged_prev_df_INF_round_2_mice, "merged_prev_df_INF_round_2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_INF_round_2_mice_df <- subset(merged_prev_df_INF_round_2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_taxa_INF_round_2_mice_df)
perst_colonizing_taxa_INF_round_2_mice <- perst_colonizing_taxa_INF_round_2_mice_df[,1]
perst_colonizing_taxa_INF_round_2_mice # 13 ASVs persistently colonizing mice 

Donor_8_core_taxa_perst_colonizing_mice_ps <- prune_taxa(perst_colonizing_taxa_INF_round_2_mice, INF_round_2_core_mice_ps)
Donor_8_core_taxa_perst_colonizing_mice_ps

perst_tax_mice_donor_8 <- as(tax_table(Donor_8_core_taxa_perst_colonizing_mice_ps), "matrix")
tax_cols <- colnames(perst_tax_mice_donor_8)
perst_tax_mice_donor_8 <- as.data.frame(perst_tax_mice_donor_8)
View(perst_tax_mice_donor_8)
perst_tax_mice_donor_8$taxonomy <- do.call(paste, c(perst_tax_mice_donor_8[tax_cols], sep = ";"))
for (co in tax_cols) perst_tax_mice_donor_8[co] <- NULL
write.table(perst_tax_mice_donor_8, "taxa_perst_colonizing_mice_donor_8.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

Establishment in GF piglets :

```{r, echo=TRUE}
# Checking how many of the 20 Donor_8 core ASVs were detected in at least one of the piglet samples
# subsetting phyloseq object to only include mice
INF_round_2_core_piglets_ps <- subset_samples(INF_round_2_core_ps, Species == "piglet")
INF_round_2_core_piglets_ps

prevdf_INF_round_2_donor_taxa_detect_piglets <- makePrevDF(INF_round_2_core_piglets_ps, 1)
prevdf_INF_round_2_donor_taxa_detect_piglets_table <- as.data.frame(prevdf_INF_round_2_donor_taxa_detect_piglets[1])
View(prevdf_INF_round_2_donor_taxa_detect_piglets_table)
colonized_prevdf_INF_round_2_donor_piglets <- prevdf_INF_round_2_donor_taxa_detect_piglets_table[prevdf_INF_round_2_donor_taxa_detect_piglets_table$Prevalence >= 1,]
View(colonized_prevdf_INF_round_2_donor_piglets)
all_INF_round_2_taxa_colonizing_piglets <-prevdf_INF_round_2_donor_taxa_detect_piglets[2]
all_INF_round_2_taxa_colonizing_piglets <- unlist(all_INF_round_2_taxa_colonizing_piglets) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_INF_round_2_taxa_colonizing_piglets # 18 of the 20 core Donor_8 ASVs have colonized the piglets 

Donor_8_core_taxa_colonizing_piglets_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_piglets,INF_round_2_core_piglets_ps)
Donor_8_core_taxa_colonizing_piglets_ps

tax_piglets_donor_8 <- as(tax_table(Donor_8_core_taxa_colonizing_piglets_ps), "matrix")
tax_cols <- colnames(tax_piglets_donor_8)
tax_piglets_donor_8 <- as.data.frame(tax_piglets_donor_8)
View(tax_piglets_donor_8)
tax_piglets_donor_8$taxonomy <- do.call(paste, c(tax_piglets_donor_8[tax_cols], sep = ";"))
for (co in tax_cols) tax_piglets_donor_8[co] <- NULL
write.table(tax_piglets_donor_8, "taxa_colonizing_piglets_donor_8.txt", quote = FALSE, col.names = FALSE, sep = "\t")

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF_round_2_core_piglets_ps
INF_round_2_HMA_piglets_ps_48HRS <- subset_samples(INF_round_2_core_piglets_ps, Description == '2_days_post_inoculation')
INF_round_2_HMA_piglets_ps_48HRS # 3 samples at this time point

prevdf_48HRS <- makePrevDF(INF_round_2_HMA_piglets_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 2,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF_round_2_HMA_piglets_ps_1week <- subset_samples(INF_round_2_core_piglets_ps, Description == '7_days_post_inoculation')
INF_round_2_HMA_piglets_ps_1week # there are 4 piglet fecal samples

prevdf_1week <- makePrevDF(INF_round_2_HMA_piglets_ps_1week,3)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 3,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF_round_2_HMA_piglets_ps_2weeks <- subset_samples(INF_round_2_core_piglets_ps, Description == '14_days_post_inoculation')
INF_round_2_HMA_piglets_ps_2weeks # there are 2 samples at this time point
 
prevdf_2weeks <- makePrevDF(INF_round_2_HMA_piglets_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 2,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF_round_2_HMA_piglets_ps_3weeks <- subset_samples(INF_round_2_core_piglets_ps, Description == '21_days_post_inoculation')
INF_round_2_HMA_piglets_ps_3weeks # there are 2 samples at this time point

prevdf_3weeks <- makePrevDF(INF_round_2_HMA_piglets_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 2,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF_round_2_HMA_piglets_ps_4weeks <- subset_samples(INF_round_2_core_piglets_ps, Description == '28_days_post_inoculation')
INF_round_2_HMA_piglets_ps_4weeks # 3 samples at this time point

prevdf_4weeks <- makePrevDF(INF_round_2_HMA_piglets_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 2,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF_round_2_HMA_piglets_ps_5weeks <- subset_samples(INF_round_2_core_piglets_ps, Description == '35_days_post_inoculation')
INF_round_2_HMA_piglets_ps_5weeks # 3 piglet samples at this time point

prevdf_5weeks <- makePrevDF(INF_round_2_HMA_piglets_ps_5weeks,3)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 3,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF_round_2_HMA_piglets_ps_40days <- subset_samples(INF_round_2_core_piglets_ps, Description == 'last sample inside bubble')
INF_round_2_HMA_piglets_ps_40days # 4 piglet samples at this time point

prevdf_40days <- makePrevDF(INF_round_2_HMA_piglets_ps_40days,3)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 3,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

merged_prev_df_INF_round_2_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_INF_round_2_piglets) # All columns considered factors
merged_prev_df_INF_round_2_piglets[2:8] <- lapply(merged_prev_df_INF_round_2_piglets[2:8], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_INF_round_2_piglets)
View(merged_prev_df_INF_round_2_piglets)

merged_prev_df_INF_round_2_piglets[is.na(merged_prev_df_INF_round_2_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF_round_2_piglets$total <- rowSums(merged_prev_df_INF_round_2_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF_round_2_piglets)
merged_prev_df_INF_round_2_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_INF_round_2_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF_round_2_piglets)
write.table(merged_prev_df_INF_round_2_piglets, "merged_prev_df_INF_round_2_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_INF_round_2_piglets_df <- subset(merged_prev_df_INF_round_2_piglets, total >= 4) # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_taxa_INF_round_2_piglets_df)
perst_colonizing_taxa_INF_round_2_piglets <- perst_colonizing_taxa_INF_round_2_piglets_df[,1]
perst_colonizing_taxa_INF_round_2_piglets # 11 ASVs persistently colonizing piglets 

Donor_8_core_taxa_perst_colonizing_piglets_ps <- prune_taxa(perst_colonizing_taxa_INF_round_2_piglets, INF_round_2_core_piglets_ps)
Donor_8_core_taxa_perst_colonizing_piglets_ps

perst_tax_piglets_donor_8 <- as(tax_table(Donor_8_core_taxa_perst_colonizing_piglets_ps), "matrix")
tax_cols <- colnames(perst_tax_piglets_donor_8)
perst_tax_piglets_donor_8 <- as.data.frame(perst_tax_piglets_donor_8)
View(perst_tax_piglets_donor_8)
perst_tax_piglets_donor_8$taxonomy <- do.call(paste, c(perst_tax_piglets_donor_8[tax_cols], sep = ";"))
for (co in tax_cols) perst_tax_piglets_donor_8[co] <- NULL
write.table(perst_tax_piglets_donor_8, "taxa_perst_colonizing_piglets_donor_8.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```


Mean abundance calculations and taxonomic breakdown of Donor_8(INF) core ASVs among the two HMA animal models :

```{r, echo=TRUE}
round_2_INF_DONORS_core_ps

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_r2_INF_inocula_table_PHYLUM <- table(tax_table(round_2_INF_DONORS_core_ps)[, "Phylum"], exclude = NULL)
human_donors_r2_INF_PHYLUM_table <- as.data.frame(human_donors_r2_INF_inocula_table_PHYLUM)
colnames(human_donors_r2_INF_PHYLUM_table)[colnames(human_donors_r2_INF_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_r2_INF_PHYLUM_table)[colnames(human_donors_r2_INF_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_r2_INF_PHYLUM_table)

# at FAMILY level
human_donors_r2_INF_inocula_table_FAMILY <- table(tax_table(round_2_INF_DONORS_core_ps)[, "Family"], exclude = NULL)
View(human_donors_r2_INF_inocula_table_FAMILY)
human_donors_r2_INF_FAMILY_table <- as.data.frame(human_donors_r2_INF_inocula_table_FAMILY)
colnames(human_donors_r2_INF_FAMILY_table)[colnames(human_donors_r2_INF_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_r2_INF_FAMILY_table)[colnames(human_donors_r2_INF_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_r2_INF_FAMILY_table)

# at GENUS level
human_donors_r2_INF_inocula_table_GENUS <- table(tax_table(round_2_INF_DONORS_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_r2_INF_inocula_table_GENUS)
human_donors_r2_INF_GENUS_table <- as.data.frame(human_donors_r2_INF_inocula_table_GENUS)
colnames(human_donors_r2_INF_GENUS_table)[colnames(human_donors_r2_INF_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_r2_INF_GENUS_table)[colnames(human_donors_r2_INF_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_r2_INF_GENUS_table)


### at PHYLUM level
tb_r2_INF_core_ASVs_PHYLUM <- tax_glom(round_2_INF_core_ps.prop, "Phylum") 
tb_r2_INF_core_ASVs_PHYLUM_df <- psmelt(tb_r2_INF_core_ASVs_PHYLUM) %>% as_tibble
tb_r2_INF_core_ASVs_PHYLUM_df_0 <- tb_r2_INF_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_r2_INF_core_ASVs_PHYLUM_df_0_rounded <- tb_r2_INF_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_r2_INF_core_ASVs_PHYLUM_df_0_rounded)

df_donor_r2_INF_PHYLUM <- subset(tb_r2_INF_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_r2_INF_PHYLUM <- subset(tb_r2_INF_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_r2_INF_PHYLUM <- subset(tb_r2_INF_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_r2_INF_PHYLUM)[colnames(df_mouse_r2_INF_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_r2_INF_PHYLUM)[colnames(df_mouse_r2_INF_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_r2_INF_PHYLUM)[colnames(df_piglet_r2_INF_PHYLUM) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_r2_INF_PHYLUM)[colnames(df_piglet_r2_INF_PHYLUM) == "SD"] <- "SD_piglet"

View(df_donor_r2_INF_PHYLUM)
df_donor_r2_INF_PHYLUM$sample_Species <- NULL

r2_INF_donor_PHYLUM_df <- merge(human_donors_r2_INF_PHYLUM_table,df_donor_r2_INF_PHYLUM, by="Phylum")
View(r2_INF_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_mice_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_mice, INF_round_2_core_mice_ps)
colonizing_core_taxa_r2_INF_mice_ps

# at PHYLUM level
r2_INF_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_r2_INF_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_r2_INF_PHYLUM_table <- as.data.frame(r2_INF_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_r2_INF_PHYLUM_table)[colnames(core_taxa_mice_r2_INF_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_r2_INF_PHYLUM_table)[colnames(core_taxa_mice_r2_INF_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_r2_INF_PHYLUM_table)

df_mouse_r2_INF_PHYLUM$sample_Species <- NULL
r2_INF_mouse_PHYLUM_df <- merge(core_taxa_mice_r2_INF_PHYLUM_table,df_mouse_r2_INF_PHYLUM, by="Phylum")
View(r2_INF_mouse_PHYLUM_df)


## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_piglets_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_piglets, INF_round_2_core_piglets_ps)
colonizing_core_taxa_r2_INF_piglets_ps

# at PHYLUM level
r2_INF_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_r2_INF_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_r2_INF_PHYLUM_table <- as.data.frame(r2_INF_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_r2_INF_PHYLUM_table)[colnames(core_taxa_piglets_r2_INF_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_r2_INF_PHYLUM_table)[colnames(core_taxa_piglets_r2_INF_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_r2_INF_PHYLUM_table)

df_piglet_r2_INF_PHYLUM$sample_Species <- NULL
r2_INF_piglet_PHYLUM_df <- merge(core_taxa_piglets_r2_INF_PHYLUM_table,df_piglet_r2_INF_PHYLUM, by="Phylum")
View(r2_INF_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_r2_INF_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(r2_INF_donor_PHYLUM_df,r2_INF_mouse_PHYLUM_df,r2_INF_piglet_PHYLUM_df))
View(merged_r2_INF_PHYLUM)

write.table(merged_r2_INF_PHYLUM, "merged_core_PHYLUM_r2_INF.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_r2_INF_core_ASVs_FAMILY <- tax_glom(round_2_INF_core_ps.prop, "Family") 

tb_r2_INF_core_ASVs_FAMILY_df <- psmelt(tb_r2_INF_core_ASVs_FAMILY) %>% as_tibble
tb_r2_INF_core_ASVs_FAMILY_df_0 <- tb_r2_INF_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_r2_INF_core_ASVs_FAMILY_df_0_rounded <- tb_r2_INF_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_r2_INF_core_ASVs_FAMILY_df_0_rounded)

df_donor_r2_INF_FAMILY <- subset(tb_r2_INF_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_r2_INF_FAMILY <- subset(tb_r2_INF_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_r2_INF_FAMILY <- subset(tb_r2_INF_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_r2_INF_FAMILY)[colnames(df_mouse_r2_INF_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_r2_INF_FAMILY)[colnames(df_mouse_r2_INF_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_r2_INF_FAMILY)[colnames(df_piglet_r2_INF_FAMILY) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_r2_INF_FAMILY)[colnames(df_piglet_r2_INF_FAMILY) == "SD"] <- "SD_piglet"

View(df_donor_r2_INF_FAMILY)
df_donor_r2_INF_FAMILY$sample_Species <- NULL

r2_INF_donor_FAMILY_df <- merge(human_donors_r2_INF_FAMILY_table,df_donor_r2_INF_FAMILY, by="Family")
View(r2_INF_donor_FAMILY_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_mice_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_mice, INF_round_2_core_mice_ps)
colonizing_core_taxa_r2_INF_mice_ps

# at FAMILY level
r2_INF_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_r2_INF_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_r2_INF_FAMILY_table <- as.data.frame(r2_INF_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_r2_INF_FAMILY_table)[colnames(core_taxa_mice_r2_INF_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_r2_INF_FAMILY_table)[colnames(core_taxa_mice_r2_INF_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_r2_INF_FAMILY_table)

df_mouse_r2_INF_FAMILY$sample_Species <- NULL
r2_INF_mouse_FAMILY_df <- merge(core_taxa_mice_r2_INF_FAMILY_table,df_mouse_r2_INF_FAMILY, by="Family")
View(r2_INF_mouse_FAMILY_df)


## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_piglets_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_piglets, INF_round_2_core_piglets_ps)
colonizing_core_taxa_r2_INF_piglets_ps

# at FAMILY level
r2_INF_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_r2_INF_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_r2_INF_FAMILY_table <- as.data.frame(r2_INF_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_r2_INF_FAMILY_table)[colnames(core_taxa_piglets_r2_INF_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_r2_INF_FAMILY_table)[colnames(core_taxa_piglets_r2_INF_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_r2_INF_FAMILY_table)

df_piglet_r2_INF_FAMILY$sample_Species <- NULL
r2_INF_piglet_FAMILY_df <- merge(core_taxa_piglets_r2_INF_FAMILY_table,df_piglet_r2_INF_FAMILY, by="Family")
View(r2_INF_piglet_FAMILY_df)

# merged the data frames from all three species
merged_r2_INF_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(r2_INF_donor_FAMILY_df,r2_INF_mouse_FAMILY_df,r2_INF_piglet_FAMILY_df))
View(merged_r2_INF_FAMILY)

write.table(merged_r2_INF_FAMILY, "merged_core_FAMILY_r2_INF.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level 
tb_r2_INF_core_ASVs_GENUS <- tax_glom(round_2_INF_core_ps.prop, "Genus") 

tb_r2_INF_core_ASVs_GENUS_df <- psmelt(tb_r2_INF_core_ASVs_GENUS) %>% as_tibble
tb_r2_INF_core_ASVs_GENUS_df_0 <- tb_r2_INF_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
View(tb_r2_INF_core_ASVs_GENUS_df_0)
tb_r2_INF_core_ASVs_GENUS_df_0_rounded <- tb_r2_INF_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_r2_INF_core_ASVs_GENUS_df_0_rounded)

df_donor_r2_INF_GENUS <- subset(tb_r2_INF_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_r2_INF_GENUS <- subset(tb_r2_INF_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_r2_INF_GENUS <- subset(tb_r2_INF_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_r2_INF_GENUS)[colnames(df_mouse_r2_INF_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_r2_INF_GENUS)[colnames(df_mouse_r2_INF_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_r2_INF_GENUS)[colnames(df_piglet_r2_INF_GENUS) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_r2_INF_GENUS)[colnames(df_piglet_r2_INF_GENUS) == "SD"] <- "SD_piglet"

View(df_donor_r2_INF_GENUS)
df_donor_r2_INF_GENUS$sample_Species <- NULL

r2_INF_donor_GENUS_df <- merge(human_donors_r2_INF_GENUS_table,df_donor_r2_INF_GENUS, by="Genus")
View(r2_INF_donor_GENUS_df)


## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_mice_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_mice, INF_round_2_core_mice_ps)
colonizing_core_taxa_r2_INF_mice_ps

# at GENUS level
r2_INF_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_r2_INF_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_r2_INF_GENUS_table <- as.data.frame(r2_INF_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_r2_INF_GENUS_table)[colnames(core_taxa_mice_r2_INF_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_r2_INF_GENUS_table)[colnames(core_taxa_mice_r2_INF_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_r2_INF_GENUS_table)

df_mouse_r2_INF_GENUS$sample_Species <- NULL
r2_INF_mouse_GENUS_df <- merge(core_taxa_mice_r2_INF_GENUS_table,df_mouse_r2_INF_GENUS, by="Genus")
View(r2_INF_mouse_GENUS_df)


## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_r2_INF_piglets_ps <- prune_taxa(all_INF_round_2_taxa_colonizing_piglets, INF_round_2_core_piglets_ps)
colonizing_core_taxa_r2_INF_piglets_ps

# at GENUS level
r2_INF_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_r2_INF_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_r2_INF_GENUS_table <- as.data.frame(r2_INF_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_r2_INF_GENUS_table)[colnames(core_taxa_piglets_r2_INF_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_r2_INF_GENUS_table)[colnames(core_taxa_piglets_r2_INF_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_r2_INF_GENUS_table)

df_piglet_r2_INF_GENUS$sample_Species <- NULL
r2_INF_piglet_GENUS_df <- merge(core_taxa_piglets_r2_INF_GENUS_table,df_piglet_r2_INF_GENUS, by="Genus")
View(r2_INF_piglet_GENUS_df)

# merged the data frames from all three species
merged_r2_INF_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(r2_INF_donor_GENUS_df,r2_INF_mouse_GENUS_df,r2_INF_piglet_GENUS_df))
View(merged_r2_INF_GENUS)

write.table(merged_r2_INF_GENUS, "merged_core_GENUS_r2_INF.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```



##**Looking at taxa establishment for Donor_5(CHLD), Donor_6(ADLT), and Donor_7(SNR) donors in the HMA mice**##


**Donor_5 (CHLD)**

```{r, echo=TRUE}
round_2_CHLD_ps <- subset_samples(round_2_complete_ps, Donor == 'child')
round_2_CHLD_ps

# for mice
samples_used_to_inoculate_mice <- c('61', '157') 
human_donors_CHLD_round_2_inocula_for_mice_ps <- subset_samples(round_2_CHLD_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_CHLD_round_2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_CHLD_round_2_mouse_donors <- sum(otu_table(human_donors_CHLD_round_2_inocula_for_mice_ps))/2
avg_num_of_reads_CHLD_round_2_mouse_donors # 16,744 reads/sample

# Determining taxa prevalence
prevdf_CHLD_round_2_mice_inocula = apply(X = otu_table(human_donors_CHLD_round_2_inocula_for_mice_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_CHLD_round_2_inocula_for_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_CHLD_round_2_mice_inocula = data.frame(Prevalence=prevdf_CHLD_round_2_mice_inocula, TotalAbundance=taxa_sums(human_donors_CHLD_round_2_inocula_for_mice_ps))
View(prevdf_CHLD_round_2_mice_inocula)

# Determining the number of core ASVs (ASVs found in both donor inocula)
core_CHLD_round_2_donor_taxa_mice <- rownames(prevdf_CHLD_round_2_mice_inocula)[prevdf_CHLD_round_2_mice_inocula$Prevalence == 2]
core_CHLD_round_2_donor_taxa_mice # 107 core ASVs found in both inocula


#overall_common_core_CHLD2_inocula_ps
#merged_pig_mouse_comp_ASV_as_seqs_ps <- readRDS("merged_pig_mouse_comp_ASV_as_seqs_ps.rds") 
#CHLD2_merged_pig_mouse_comp_ASV_as_seqs_ps <- subset_samples(merged_pig_mouse_comp_ASV_as_seqs_ps, Donor == "CHLD2")
#CHLD2_donors <- sample_names(overall_common_core_CHLD2_inocula_ps)
#CHLD2_merged_pig_mouse_comp_ASV_as_seqs_donors_ps <- subset_samples(CHLD2_merged_pig_mouse_comp_ASV_as_seqs_ps, SampleID %in% CHLD2_donors)
#CHLD2_merged_pig_mouse_comp_ASV_as_seqs_ps_cyano <- subset_taxa(CHLD2_merged_pig_mouse_comp_ASV_as_seqs_donors_ps, Phylum == "Cyanobacteria" ) 


#CHLD_round_2_donors_core_ps <- prune_taxa(core_CHLD_round_2_donor_taxa_mice,human_donors_CHLD_round_2_inocula_for_mice_ps)

#seqtab_r1 <- otu_table(CHLD2_merged_pig_mouse_comp_ASV_as_seqs_ps, taxa_are_rows = F)
#dim(seqtab_r1)

#seqtab_r2 <- otu_table(CHLD2_merged_pig_mouse_comp_ASV_as_seqs_ps_cyano, taxa_are_rows = F)
#dim(seqtab_r2)

#merged_r1_r2_seqtabs <- mergeSequenceTables(seqtab_r1, seqtab_r2, repeats = "sum")
#dim(merged_r1_r2_seqtabs)

# Let's see how the donor taxa have established in the mice
CHLD_round_2_HMA_mice_ps <- subset_samples(round_2_CHLD_ps, Species == 'mouse') # keeping just the mouse samples
CHLD_round_2_HMA_mice_ps
avg_num_of_reads_CHLD_round_2_mice <- sum(otu_table(CHLD_round_2_HMA_mice_ps))/nsamples(CHLD_round_2_HMA_mice_ps)
avg_num_of_reads_CHLD_round_2_mice # 54,621 reads/sample

CHLD_round_2_HMA_mice_donor_taxa_ps <- prune_taxa(core_CHLD_round_2_donor_taxa_mice, CHLD_round_2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
CHLD_round_2_HMA_mice_donor_taxa_ps

# Calculating the percentage of the total HMA mouse reads that is represented by these 107 donor ASVs
sum(otu_table(CHLD_round_2_HMA_mice_donor_taxa_ps))/sum(otu_table(CHLD_round_2_HMA_mice_ps)) # ~69% of reads represented by core Donor_5 ASVs

# Checking how many of the 107 core Donor_5 ASVs were detected in at least one of the mouse fecal samples
prevdf_CHLD_round_2_donor_taxa_detect_mice <- makePrevDF(CHLD_round_2_HMA_mice_donor_taxa_ps, 1)
prevdf_CHLD_round_2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_CHLD_round_2_donor_taxa_detect_mice[1])
View(prevdf_CHLD_round_2_donor_taxa_detect_mice_table)
colonized_prevdf_CHLD_round_2_donor_mice <- prevdf_CHLD_round_2_donor_taxa_detect_mice_table[prevdf_CHLD_round_2_donor_taxa_detect_mice_table$Prevalence >= 1,]
View(colonized_prevdf_CHLD_round_2_donor_mice)
core_CHLD_round_2_taxa_colonizing_mice <-prevdf_CHLD_round_2_donor_taxa_detect_mice[2]
core_CHLD_round_2_taxa_colonizing_mice <- unlist(core_CHLD_round_2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
core_CHLD_round_2_taxa_colonizing_mice # 68 of the 107 core Donor_5 ASVs have colonized the mice 

Donor_5_core_taxa_colonizing_mice_ps <- prune_taxa(core_CHLD_round_2_taxa_colonizing_mice,CHLD_round_2_HMA_mice_donor_taxa_ps)
Donor_5_core_taxa_colonizing_mice_ps

tax_mice_donor_5 <- as(tax_table(Donor_5_core_taxa_colonizing_mice_ps), "matrix")
tax_cols <- colnames(tax_mice_donor_5)
tax_mice_donor_5 <- as.data.frame(tax_mice_donor_5)
View(tax_mice_donor_5)
tax_mice_donor_5$taxonomy <- do.call(paste, c(tax_mice_donor_5[tax_cols], sep = ";"))
for (co in tax_cols) tax_mice_donor_5[co] <- NULL
write.table(tax_mice_donor_5, "taxa_colonizing_mice_donor_5.txt", quote = FALSE, col.names = FALSE, sep = "\t")

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD_round_2_HMA_mice_donor_taxa_ps
CHLD_round_2_HMA_mice_ps_48HRS <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '2_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD_round_2_HMA_mice_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 6,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD_round_2_HMA_mice_ps_1week <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '7_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_1week

prevdf_1week <- makePrevDF(CHLD_round_2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD_round_2_HMA_mice_ps_2weeks <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '14_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD_round_2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD_round_2_HMA_mice_ps_3weeks <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '21_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_3weeks # only 9 samples at this time point

prevdf_3weeks <- makePrevDF(CHLD_round_2_HMA_mice_ps_3weeks,5)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 5,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD_round_2_HMA_mice_ps_4weeks <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '28_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD_round_2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD_round_2_HMA_mice_ps_5weeks <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == '35_days_post_inoculation')
CHLD_round_2_HMA_mice_ps_5weeks # only 9 samples at this time point

prevdf_5weeks <- makePrevDF(CHLD_round_2_HMA_mice_ps_5weeks,5)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 5,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD_round_2_HMA_mice_ps_40days <- subset_samples(CHLD_round_2_HMA_mice_donor_taxa_ps, Description == 'final fecal collection')
CHLD_round_2_HMA_mice_ps_40days # only 8 samples at this time point

prevdf_40days <- makePrevDF(CHLD_round_2_HMA_mice_ps_40days,5)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 5,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

merged_prev_df_CHLD_round_2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_CHLD_round_2_mice) # All columns considered factors
merged_prev_df_CHLD_round_2_mice[2:8] <- lapply(merged_prev_df_CHLD_round_2_mice[2:8], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD_round_2_mice)
View(merged_prev_df_CHLD_round_2_mice)

merged_prev_df_CHLD_round_2_mice[is.na(merged_prev_df_CHLD_round_2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD_round_2_mice$total <- rowSums(merged_prev_df_CHLD_round_2_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD_round_2_mice)
merged_prev_df_CHLD_round_2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD_round_2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD_round_2_mice)
write.table(merged_prev_df_CHLD_round_2_mice, "merged_prev_df_CHLD_round_2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_CHLD_round_2_mice_df <- subset(merged_prev_df_CHLD_round_2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_taxa_CHLD_round_2_mice_df)
perst_colonizing_taxa_CHLD_round_2_mice <- perst_colonizing_taxa_CHLD_round_2_mice_df[,1]
perst_colonizing_taxa_CHLD_round_2_mice # 29 core ASVs persistently colonizing mice 

Donor_5_core_taxa_perst_colonizing_mice_ps <- prune_taxa(perst_colonizing_taxa_CHLD_round_2_mice,CHLD_round_2_HMA_mice_donor_taxa_ps)
Donor_5_core_taxa_perst_colonizing_mice_ps

perst_tax_mice_donor_5 <- as(tax_table(Donor_5_core_taxa_perst_colonizing_mice_ps), "matrix")
tax_cols <- colnames(perst_tax_mice_donor_5)
perst_tax_mice_donor_5 <- as.data.frame(perst_tax_mice_donor_5)
View(perst_tax_mice_donor_5)
perst_tax_mice_donor_5$taxonomy <- do.call(paste, c(perst_tax_mice_donor_5[tax_cols], sep = ";"))
for (co in tax_cols) perst_tax_mice_donor_5[co] <- NULL
write.table(perst_tax_mice_donor_5, "taxa_perst_colonizing_mice_donor_5.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```


**Donor_6(ADLT)**

```{r, echo=TRUE}
round_2_ADLT_ps <- subset_samples(round_2_complete_ps, Donor == 'adult')
round_2_ADLT_ps

# for mice
samples_used_to_inoculate_mice <- c('73', '169') 
human_donors_ADLT_round_2_inocula_for_mice_ps <- subset_samples(round_2_ADLT_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_ADLT_round_2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_ADLT_round_2_mouse_donors <- sum(otu_table(human_donors_ADLT_round_2_inocula_for_mice_ps))/2
avg_num_of_reads_ADLT_round_2_mouse_donors # 39,962 reads/sample

# Determining taxa prevalence
prevdf_ADLT_round_2_mice_inocula = apply(X = otu_table(human_donors_ADLT_round_2_inocula_for_mice_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ADLT_round_2_inocula_for_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ADLT_round_2_mice_inocula = data.frame(Prevalence=prevdf_ADLT_round_2_mice_inocula, TotalAbundance=taxa_sums(human_donors_ADLT_round_2_inocula_for_mice_ps))
View(prevdf_ADLT_round_2_mice_inocula)

core_ADLT_round_2_donor_taxa_mice <- rownames(prevdf_ADLT_round_2_mice_inocula)[prevdf_ADLT_round_2_mice_inocula$Prevalence == 2]
str(core_ADLT_round_2_donor_taxa_mice) # 117 core ASVs found in both donor inocula 

# Let's see how the donor taxa have established in the mice
ADLT_round_2_HMA_mice_ps <- subset_samples(round_2_ADLT_ps, Species == 'mouse') # keeping just the mouse samples
ADLT_round_2_HMA_mice_ps
avg_num_of_reads_ADLT_round_2_mice <- sum(otu_table(ADLT_round_2_HMA_mice_ps))/nsamples(ADLT_round_2_HMA_mice_ps)
avg_num_of_reads_ADLT_round_2_mice # 58,610 reads/sample

ADLT_round_2_HMA_mice_donor_taxa_ps <- prune_taxa(core_ADLT_round_2_donor_taxa_mice, ADLT_round_2_HMA_mice_ps) # retaining only the core ASVs that were found in the donor inoculae
ADLT_round_2_HMA_mice_donor_taxa_ps

# Calculating the percentage of the total HMA mouse reads that is represented by these 117 core donor ASVs
sum(otu_table(ADLT_round_2_HMA_mice_donor_taxa_ps))/sum(otu_table(ADLT_round_2_HMA_mice_ps)) # ~68.9% of reads originating from core donor ASVs 

# Checking how many of the 117 ADLT supp donor ASVs were detected in at least one of the mouse samples
prevdf_ADLT_round_2_donor_taxa_detect_mice <- makePrevDF(ADLT_round_2_HMA_mice_donor_taxa_ps, 1)
prevdf_ADLT_round_2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_ADLT_round_2_donor_taxa_detect_mice[1])
View(prevdf_ADLT_round_2_donor_taxa_detect_mice_table)
colonized_prevdf_ADLT_round_2_donor_mice <- prevdf_ADLT_round_2_donor_taxa_detect_mice_table[prevdf_ADLT_round_2_donor_taxa_detect_mice_table$Prevalence >= 1,]
View(colonized_prevdf_ADLT_round_2_donor_mice)
core_ADLT_round_2_taxa_colonizing_mice <-prevdf_ADLT_round_2_donor_taxa_detect_mice[2]
core_ADLT_round_2_taxa_colonizing_mice <- unlist(core_ADLT_round_2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
core_ADLT_round_2_taxa_colonizing_mice # 85 of the 117 round 2 ADLT donor core ASVs have colonized the mice 

Donor_6_core_taxa_colonizing_mice_ps <- prune_taxa(core_ADLT_round_2_taxa_colonizing_mice,ADLT_round_2_HMA_mice_donor_taxa_ps)
Donor_6_core_taxa_colonizing_mice_ps

tax_mice_donor_6 <- as(tax_table(Donor_6_core_taxa_colonizing_mice_ps), "matrix")
tax_cols <- colnames(tax_mice_donor_6)
tax_mice_donor_6 <- as.data.frame(tax_mice_donor_6)
View(tax_mice_donor_6)
tax_mice_donor_6$taxonomy <- do.call(paste, c(tax_mice_donor_6[tax_cols], sep = ";"))
for (co in tax_cols) tax_mice_donor_6[co] <- NULL
write.table(tax_mice_donor_6, "taxa_colonizing_mice_donor_6.txt", quote = FALSE, col.names = FALSE, sep = "\t")

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT_round_2_HMA_mice_donor_taxa_ps
ADLT_round_2_HMA_mice_ps_48HRS <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '2_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT_round_2_HMA_mice_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 6,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT_round_2_HMA_mice_ps_1week <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '7_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_1week

prevdf_1week <- makePrevDF(ADLT_round_2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT_round_2_HMA_mice_ps_2weeks <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '14_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT_round_2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT_round_2_HMA_mice_ps_3weeks <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '21_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_3weeks # only 9 samples at this time point

prevdf_3weeks <- makePrevDF(ADLT_round_2_HMA_mice_ps_3weeks,5)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 5,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT_round_2_HMA_mice_ps_4weeks <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '28_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_4weeks # only 8 samples at this time point

prevdf_4weeks <- makePrevDF(ADLT_round_2_HMA_mice_ps_4weeks,5)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 5,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT_round_2_HMA_mice_ps_5weeks <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == '35_days_post_inoculation')
ADLT_round_2_HMA_mice_ps_5weeks # 9 samples at tis time point

prevdf_5weeks <- makePrevDF(ADLT_round_2_HMA_mice_ps_5weeks,5)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 5,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT_round_2_HMA_mice_ps_40days <- subset_samples(ADLT_round_2_HMA_mice_donor_taxa_ps, Description == 'final fecal collection')
ADLT_round_2_HMA_mice_ps_40days # 9 samples at tis time point

prevdf_40days <- makePrevDF(ADLT_round_2_HMA_mice_ps_40days,5)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 5,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

merged_prev_df_ADLT_round_2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ADLT_round_2_mice) # All columns considered factors
merged_prev_df_ADLT_round_2_mice[2:8] <- lapply(merged_prev_df_ADLT_round_2_mice[2:8], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT_round_2_mice)
View(merged_prev_df_ADLT_round_2_mice)

merged_prev_df_ADLT_round_2_mice[is.na(merged_prev_df_ADLT_round_2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT_round_2_mice$total <- rowSums(merged_prev_df_ADLT_round_2_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT_round_2_mice)
merged_prev_df_ADLT_round_2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT_round_2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT_round_2_mice)
write.table(merged_prev_df_ADLT_round_2_mice, "merged_prev_df_ADLT_round_2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_ADLT_round_2_mice_df <- subset(merged_prev_df_ADLT_round_2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_taxa_ADLT_round_2_mice_df)
perst_colonizing_taxa_ADLT_round_2_mice <- perst_colonizing_taxa_ADLT_round_2_mice_df[,1]
perst_colonizing_taxa_ADLT_round_2_mice # 43 core Donor_6 ASVs persistently colonizing mice 

Donor_6_core_taxa_perst_colonizing_mice_ps <- prune_taxa(perst_colonizing_taxa_ADLT_round_2_mice,ADLT_round_2_HMA_mice_donor_taxa_ps)
Donor_6_core_taxa_perst_colonizing_mice_ps

perst_tax_mice_donor_6 <- as(tax_table(Donor_6_core_taxa_perst_colonizing_mice_ps), "matrix")
tax_cols <- colnames(perst_tax_mice_donor_6)
perst_tax_mice_donor_6 <- as.data.frame(perst_tax_mice_donor_6)
View(perst_tax_mice_donor_6)
perst_tax_mice_donor_6$taxonomy <- do.call(paste, c(perst_tax_mice_donor_6[tax_cols], sep = ";"))
for (co in tax_cols) perst_tax_mice_donor_6[co] <- NULL
write.table(perst_tax_mice_donor_6, "taxa_perst_colonizing_mice_donor_6.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```


**Donor_7 (SNR)**

```{r, echo=TRUE}
round_2_SNR_ps <- subset_samples(round_2_complete_ps, Donor == 'elderly')
round_2_SNR_ps

# for mice
samples_used_to_inoculate_mice <- c('85', '181') 
human_donors_SNR_round_2_inocula_for_mice_ps <- subset_samples(round_2_SNR_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_SNR_round_2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_SNR_round_2_mouse_donors <- sum(otu_table(human_donors_SNR_round_2_inocula_for_mice_ps))/2
avg_num_of_reads_SNR_round_2_mouse_donors # 42,946 reads/sample

# Determining taxa prevalence
prevdf_SNR_round_2_mice_inocula = apply(X = otu_table(human_donors_SNR_round_2_inocula_for_mice_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_SNR_round_2_inocula_for_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_SNR_round_2_mice_inocula = data.frame(Prevalence=prevdf_SNR_round_2_mice_inocula, TotalAbundance=taxa_sums(human_donors_SNR_round_2_inocula_for_mice_ps))
View(prevdf_SNR_round_2_mice_inocula)

core_SNR_round_2_donor_taxa_mice <- rownames(prevdf_SNR_round_2_mice_inocula)[prevdf_SNR_round_2_mice_inocula$Prevalence == 2]
core_SNR_round_2_donor_taxa_mice # 151 core ASVs found in the donor inocula for Donor_7

# Let's see how the donor taxa have established in the mice
SNR_round_2_HMA_mice_ps <- subset_samples(round_2_SNR_ps, Species == 'mouse') # keeping just the mouse samples
SNR_round_2_HMA_mice_ps
avg_num_of_reads_SNR_round_2_mice <- sum(otu_table(SNR_round_2_HMA_mice_ps))/nsamples(SNR_round_2_HMA_mice_ps)
avg_num_of_reads_SNR_round_2_mice # 53,741 reads/sample

SNR_round_2_HMA_mice_donor_taxa_ps <- prune_taxa(core_SNR_round_2_donor_taxa_mice, SNR_round_2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
SNR_round_2_HMA_mice_donor_taxa_ps

# Calculating the percentage of the total HMA mouse reads that is represented by these 151 core donor ASVs
sum(otu_table(SNR_round_2_HMA_mice_donor_taxa_ps))/sum(otu_table(SNR_round_2_HMA_mice_ps)) # ~64.4% of reads originating from core donor ASVs 

# Checking how many of the 151 SNR supp donor ASVs were detected in at least one of the mouse samples
prevdf_SNR_round_2_donor_taxa_detect_mice <- makePrevDF(SNR_round_2_HMA_mice_donor_taxa_ps, 1)
prevdf_SNR_round_2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_SNR_round_2_donor_taxa_detect_mice[1])
View(prevdf_SNR_round_2_donor_taxa_detect_mice_table)
colonized_prevdf_SNR_round_2_donor_mice <- prevdf_SNR_round_2_donor_taxa_detect_mice_table[prevdf_SNR_round_2_donor_taxa_detect_mice_table$Prevalence >= 1,]
View(colonized_prevdf_SNR_round_2_donor_mice)
core_SNR_round_2_taxa_colonizing_mice <-prevdf_SNR_round_2_donor_taxa_detect_mice[2]
core_SNR_round_2_taxa_colonizing_mice <- unlist(core_SNR_round_2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
core_SNR_round_2_taxa_colonizing_mice # 68 out of 151 core Donor_7 ASVs detected in at least 1 mouse fecal sample

Donor_7_core_taxa_colonizing_mice_ps <- prune_taxa(core_SNR_round_2_taxa_colonizing_mice,SNR_round_2_HMA_mice_donor_taxa_ps)
Donor_7_core_taxa_colonizing_mice_ps

tax_mice_donor_7 <- as(tax_table(Donor_7_core_taxa_colonizing_mice_ps), "matrix")
tax_cols <- colnames(tax_mice_donor_7)
tax_mice_donor_7 <- as.data.frame(tax_mice_donor_7)
View(tax_mice_donor_7)
tax_mice_donor_7$taxonomy <- do.call(paste, c(tax_mice_donor_7[tax_cols], sep = ";"))
for (co in tax_cols) tax_mice_donor_7[co] <- NULL
write.table(tax_mice_donor_7, "taxa_colonizing_mice_donor_7.txt", quote = FALSE, col.names = FALSE, sep = "\t")

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
SNR_round_2_HMA_mice_donor_taxa_ps
SNR_round_2_HMA_mice_ps_48HRS <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '2_days_post_inoculation')
SNR_round_2_HMA_mice_ps_48HRS # 7 samples at this time point

prevdf_48HRS <- makePrevDF(SNR_round_2_HMA_mice_ps_48HRS,4) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= 4,]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
SNR_round_2_HMA_mice_ps_1week <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '7_days_post_inoculation')
SNR_round_2_HMA_mice_ps_1week # 10 samples at this time point

prevdf_1week <- makePrevDF(SNR_round_2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= 6,]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
SNR_round_2_HMA_mice_ps_2weeks <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '14_days_post_inoculation')
SNR_round_2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(SNR_round_2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= 6,]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
SNR_round_2_HMA_mice_ps_3weeks <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '21_days_post_inoculation')
SNR_round_2_HMA_mice_ps_3weeks

prevdf_3weeks <- makePrevDF(SNR_round_2_HMA_mice_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= 6,]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
SNR_round_2_HMA_mice_ps_4weeks <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '28_days_post_inoculation')
SNR_round_2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(SNR_round_2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= 6,]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
SNR_round_2_HMA_mice_ps_5weeks <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == '35_days_post_inoculation')
SNR_round_2_HMA_mice_ps_5weeks # 9 samples at this time point

prevdf_5weeks <- makePrevDF(SNR_round_2_HMA_mice_ps_5weeks,5)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= 5,]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
SNR_round_2_HMA_mice_ps_40days <- subset_samples(SNR_round_2_HMA_mice_donor_taxa_ps, Description == 'final fecal collection')
SNR_round_2_HMA_mice_ps_40days

prevdf_40days <- makePrevDF(SNR_round_2_HMA_mice_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= 6,]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

merged_prev_df_SNR_round_2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_SNR_round_2_mice) # All columns considered factors
merged_prev_df_SNR_round_2_mice[2:8] <- lapply(merged_prev_df_SNR_round_2_mice[2:8], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_SNR_round_2_mice)
View(merged_prev_df_SNR_round_2_mice)

merged_prev_df_SNR_round_2_mice[is.na(merged_prev_df_SNR_round_2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_SNR_round_2_mice$total <- rowSums(merged_prev_df_SNR_round_2_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_SNR_round_2_mice)
merged_prev_df_SNR_round_2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_SNR_round_2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_SNR_round_2_mice)
write.table(merged_prev_df_SNR_round_2_mice, "merged_prev_df_SNR_round_2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_SNR_round_2_mice_df <- subset(merged_prev_df_SNR_round_2_mice, total >= 4) # select the ASVs which were detected in at least 4 time points
View(perst_colonizing_taxa_SNR_round_2_mice_df)
perst_colonizing_taxa_SNR_round_2_mice <- perst_colonizing_taxa_SNR_round_2_mice_df[,1]
str(perst_colonizing_taxa_SNR_round_2_mice) # 36 core Donor_7 ASVs persistently colonizing mice 

Donor_7_core_taxa_perst_colonizing_mice_ps <- prune_taxa(perst_colonizing_taxa_SNR_round_2_mice,SNR_round_2_HMA_mice_donor_taxa_ps)
Donor_7_core_taxa_perst_colonizing_mice_ps

perst_tax_mice_donor_7 <- as(tax_table(Donor_7_core_taxa_perst_colonizing_mice_ps), "matrix")
tax_cols <- colnames(perst_tax_mice_donor_7)
perst_tax_mice_donor_7 <- as.data.frame(perst_tax_mice_donor_7)
View(perst_tax_mice_donor_7)
perst_tax_mice_donor_7$taxonomy <- do.call(paste, c(perst_tax_mice_donor_7[tax_cols], sep = ";"))
for (co in tax_cols) perst_tax_mice_donor_7[co] <- NULL
write.table(perst_tax_mice_donor_7, "taxa_perst_colonizing_mice_donor_7.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```



## **Looking at the taxonomic classifications to which the colonizing taxa belong to** ##

**Donor_5(CHLD)**

```{r,echo=TRUE}

### for CHLD donor
human_donor_inocula_with_rel_taxa_ps <- prune_taxa(core_CHLD_round_2_donor_taxa_mice, human_donors_CHLD_round_2_inocula_for_mice_ps) # 107 core Donor_5 ASVs

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_CHLD_inocula_table_PHYLUM <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Phylum"], exclude = NULL)
human_donors_CHLD_PHYLUM_table <- as.data.frame(human_donors_CHLD_inocula_table_PHYLUM)
colnames(human_donors_CHLD_PHYLUM_table)[colnames(human_donors_CHLD_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_CHLD_PHYLUM_table)[colnames(human_donors_CHLD_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD_PHYLUM_table)

# at FAMILY level
human_donors_CHLD_inocula_table_FAMILY <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Family"], exclude = NULL)
View(human_donors_CHLD_inocula_table_FAMILY)
human_donors_CHLD_FAMILY_table <- as.data.frame(human_donors_CHLD_inocula_table_FAMILY)
colnames(human_donors_CHLD_FAMILY_table)[colnames(human_donors_CHLD_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_CHLD_FAMILY_table)[colnames(human_donors_CHLD_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD_FAMILY_table)

# at GENUS level
human_donors_CHLD_inocula_table_GENUS <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(human_donors_CHLD_inocula_table_GENUS)
human_donors_CHLD_GENUS_table <- as.data.frame(human_donors_CHLD_inocula_table_GENUS)
colnames(human_donors_CHLD_GENUS_table)[colnames(human_donors_CHLD_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_CHLD_GENUS_table)[colnames(human_donors_CHLD_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for CHLD donor 

## at PHYLUM level for donors
human_donors_CHLD_round_2_inocula_for_mice_ps.prop <- transform_sample_counts(human_donors_CHLD_round_2_inocula_for_mice_ps, function(x) x/sum(x)) 

CHLD_round_2_donor_core_ps.prop <- prune_taxa(core_CHLD_round_2_donor_taxa_mice,human_donors_CHLD_round_2_inocula_for_mice_ps.prop) # filtering the core ASVs from the overall ASV table

tb_CHLD_core_ASVs_PHYLUM <- tax_glom(CHLD_round_2_donor_core_ps.prop, "Phylum", NArm = FALSE)

tb_CHLD_core_ASVs_PHYLUM_df <- psmelt(tb_CHLD_core_ASVs_PHYLUM) %>% as_tibble
tb_CHLD_core_ASVs_PHYLUM_df_0 <- tb_CHLD_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_PHYLUM_df_0_rounded <- tb_CHLD_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_PHYLUM_df_0_rounded)

df_donor_CHLD_PHYLUM <- subset(tb_CHLD_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
View(df_donor_CHLD_PHYLUM)
df_donor_CHLD_PHYLUM$sample_Species <- NULL
View(df_donor_CHLD_PHYLUM)


## at FAMILY level for donors
tb_CHLD_core_ASVs_FAMILY <- tax_glom(CHLD_round_2_donor_core_ps.prop, "Family", NArm = FALSE)

tb_CHLD_core_ASVs_FAMILY_df <- psmelt(tb_CHLD_core_ASVs_FAMILY) %>% as_tibble
tb_CHLD_core_ASVs_FAMILY_df_0 <- tb_CHLD_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_FAMILY_df_0_rounded <- tb_CHLD_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_FAMILY_df_0_rounded)

df_donor_CHLD_FAMILY <- subset(tb_CHLD_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")

View(df_donor_CHLD_FAMILY)
df_donor_CHLD_FAMILY$sample_Species <- NULL
View(df_donor_CHLD_FAMILY)


## at GENUS level for donors
tb_CHLD_core_ASVs_GENUS <- tax_glom(CHLD_round_2_donor_core_ps.prop, "Genus", NArm = FALSE)

tb_CHLD_core_ASVs_GENUS_df <- psmelt(tb_CHLD_core_ASVs_GENUS) %>% as_tibble
tb_CHLD_core_ASVs_GENUS_df_0 <- tb_CHLD_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_GENUS_df_0_rounded <- tb_CHLD_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_GENUS_df_0_rounded)

df_donor_CHLD_GENUS <- subset(tb_CHLD_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")

View(df_donor_CHLD_GENUS)
df_donor_CHLD_GENUS$sample_Species <- NULL
View(df_donor_CHLD_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
CHLD_donor_PHYLUM_df <- merge(human_donors_CHLD_PHYLUM_table,df_donor_CHLD_PHYLUM, by="Phylum")
View(CHLD_donor_PHYLUM_df)

CHLD_donor_FAMILY_df <- merge(human_donors_CHLD_FAMILY_table,df_donor_CHLD_FAMILY, by="Family")
View(CHLD_donor_FAMILY_df)

CHLD_donor_GENUS_df <- merge(human_donors_CHLD_GENUS_table,df_donor_CHLD_GENUS, by="Genus")
View(CHLD_donor_GENUS_df)


## for mice
core_CHLD_round_2_taxa_colonizing_mice # 68 of the 107 core Donor_5 ASVs have colonized the mice

CHLD_round_2_HMA_mice_donor_rel_taxa_ps <- prune_taxa(core_CHLD_round_2_taxa_colonizing_mice, CHLD_round_2_HMA_mice_donor_taxa_ps)

# at PHYLUM level
CHLD_rel_taxa_mouse_table_PHYLUM <- table(tax_table(CHLD_round_2_HMA_mice_donor_rel_taxa_ps)[, "Phylum"], exclude = NULL)
View(CHLD_rel_taxa_mouse_table_PHYLUM)
rel_taxa_mice_CHLD_PHYLUM_table <- as.data.frame(CHLD_rel_taxa_mouse_table_PHYLUM)
colnames(rel_taxa_mice_CHLD_PHYLUM_table)[colnames(rel_taxa_mice_CHLD_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(rel_taxa_mice_CHLD_PHYLUM_table)[colnames(rel_taxa_mice_CHLD_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_CHLD_PHYLUM_table)

# at FAMILY level
CHLD_rel_taxa_mouse_table_FAMILY <- table(tax_table(CHLD_round_2_HMA_mice_donor_rel_taxa_ps)[, "Family"], exclude = NULL)
View(CHLD_rel_taxa_mouse_table_FAMILY)
rel_taxa_mice_CHLD_FAMILY_table <- as.data.frame(CHLD_rel_taxa_mouse_table_FAMILY)
colnames(rel_taxa_mice_CHLD_FAMILY_table)[colnames(rel_taxa_mice_CHLD_FAMILY_table) == "Var1"] <- "Family"
colnames(rel_taxa_mice_CHLD_FAMILY_table)[colnames(rel_taxa_mice_CHLD_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_CHLD_FAMILY_table)

# at GENUS level
CHLD_rel_taxa_mouse_table_GENUS <- table(tax_table(CHLD_round_2_HMA_mice_donor_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(CHLD_rel_taxa_mouse_table_GENUS)
rel_taxa_mice_CHLD_GENUS_table <- as.data.frame(CHLD_rel_taxa_mouse_table_GENUS)
colnames(rel_taxa_mice_CHLD_GENUS_table)[colnames(rel_taxa_mice_CHLD_GENUS_table) == "Var1"] <- "Genus"
colnames(rel_taxa_mice_CHLD_GENUS_table)[colnames(rel_taxa_mice_CHLD_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_CHLD_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for HMA mice

# at PHYLUM level for mice
CHLD_round_2_HMA_mice_ps.prop <- transform_sample_counts(CHLD_round_2_HMA_mice_ps, function(x) x/sum(x)) # convert to relative abundance

# retain only the ASVs colonizing mice
CHLD_round_2_mice_ps.prop <- prune_taxa(core_CHLD_round_2_taxa_colonizing_mice, CHLD_round_2_HMA_mice_ps.prop)

tb_CHLD_core_ASVs_PHYLUM <- tax_glom(CHLD_round_2_mice_ps.prop, "Phylum", NArm = FALSE)

tb_CHLD_core_ASVs_PHYLUM_df <- psmelt(tb_CHLD_core_ASVs_PHYLUM) %>% as_tibble
tb_CHLD_core_ASVs_PHYLUM_df_0 <- tb_CHLD_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_PHYLUM_df_0_rounded <- tb_CHLD_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_PHYLUM_df_0_rounded)

df_mice_CHLD_PHYLUM <- subset(tb_CHLD_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
View(df_mice_CHLD_PHYLUM)
df_mice_CHLD_PHYLUM$sample_Species <- NULL
View(df_mice_CHLD_PHYLUM)


## at FAMILY level for mice
tb_CHLD_core_ASVs_FAMILY <- tax_glom(CHLD_round_2_mice_ps.prop, "Family", NArm = FALSE)

tb_CHLD_core_ASVs_FAMILY_df <- psmelt(tb_CHLD_core_ASVs_FAMILY) %>% as_tibble
tb_CHLD_core_ASVs_FAMILY_df_0 <- tb_CHLD_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_FAMILY_df_0_rounded <- tb_CHLD_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_FAMILY_df_0_rounded)

df_mice_CHLD_FAMILY <- subset(tb_CHLD_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")

View(df_mice_CHLD_FAMILY)
df_mice_CHLD_FAMILY$sample_Species <- NULL
View(df_mice_CHLD_FAMILY)


## at GENUS level for mices
tb_CHLD_core_ASVs_GENUS <- tax_glom(CHLD_round_2_mice_ps.prop, "Genus", NArm = FALSE)

tb_CHLD_core_ASVs_GENUS_df <- psmelt(tb_CHLD_core_ASVs_GENUS) %>% as_tibble
tb_CHLD_core_ASVs_GENUS_df_0 <- tb_CHLD_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD_core_ASVs_GENUS_df_0_rounded <- tb_CHLD_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD_core_ASVs_GENUS_df_0_rounded)

df_mice_CHLD_GENUS <- subset(tb_CHLD_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")

View(df_mice_CHLD_GENUS)
df_mice_CHLD_GENUS$sample_Species <- NULL
View(df_mice_CHLD_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
CHLD_mice_PHYLUM_df <- merge(rel_taxa_mice_CHLD_PHYLUM_table,df_mice_CHLD_PHYLUM, by="Phylum")
colnames(CHLD_mice_PHYLUM_df)[colnames(CHLD_mice_PHYLUM_df) == "Mean"] <- "Mean-mouse"
colnames(CHLD_mice_PHYLUM_df)[colnames(CHLD_mice_PHYLUM_df) == "SD"] <- "SD-mouse"
View(CHLD_mice_PHYLUM_df)

CHLD_mice_FAMILY_df <- merge(rel_taxa_mice_CHLD_FAMILY_table,df_mice_CHLD_FAMILY, by="Family")
colnames(CHLD_mice_FAMILY_df)[colnames(CHLD_mice_FAMILY_df) == "Mean"] <- "Mean-mouse"
colnames(CHLD_mice_FAMILY_df)[colnames(CHLD_mice_FAMILY_df) == "SD"] <- "SD-mouse"
View(CHLD_mice_FAMILY_df)

CHLD_mice_GENUS_df <- merge(rel_taxa_mice_CHLD_GENUS_table,df_mice_CHLD_GENUS, by="Genus")
colnames(CHLD_mice_GENUS_df)[colnames(CHLD_mice_GENUS_df) == "Mean"] <- "Mean-mouse"
colnames(CHLD_mice_GENUS_df)[colnames(CHLD_mice_GENUS_df) == "SD"] <- "SD-mouse"
View(CHLD_mice_GENUS_df)


# merged the data frames from human donors and HMA mice
merged_CHLD_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD_donor_PHYLUM_df,CHLD_mice_PHYLUM_df))
View(merged_CHLD_PHYLUM)

merged_CHLD_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD_donor_FAMILY_df,CHLD_mice_FAMILY_df))
View(merged_CHLD_FAMILY)

merged_CHLD_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD_donor_GENUS_df,CHLD_mice_GENUS_df))
View(merged_CHLD_GENUS)

write.table(merged_CHLD_PHYLUM, "merged_CHLD_round_2_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_CHLD_FAMILY, "merged_CHLD_round_2_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_CHLD_GENUS, "merged_CHLD_round_2_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_6(ADLT)**

```{r,echo=TRUE}

### for ADLT donor
human_donor_inocula_with_rel_taxa_ps <- prune_taxa(core_ADLT_round_2_donor_taxa_mice, human_donors_ADLT_round_2_inocula_for_mice_ps)

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_ADLT_inocula_table_PHYLUM <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Phylum"], exclude = NULL)
human_donors_ADLT_PHYLUM_table <- as.data.frame(human_donors_ADLT_inocula_table_PHYLUM)
colnames(human_donors_ADLT_PHYLUM_table)[colnames(human_donors_ADLT_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_ADLT_PHYLUM_table)[colnames(human_donors_ADLT_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT_PHYLUM_table)

# at FAMILY level
human_donors_ADLT_inocula_table_FAMILY <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Family"], exclude = NULL)
View(human_donors_ADLT_inocula_table_FAMILY)
human_donors_ADLT_FAMILY_table <- as.data.frame(human_donors_ADLT_inocula_table_FAMILY)
colnames(human_donors_ADLT_FAMILY_table)[colnames(human_donors_ADLT_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_ADLT_FAMILY_table)[colnames(human_donors_ADLT_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT_FAMILY_table)

# at GENUS level
human_donors_ADLT_inocula_table_GENUS <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(human_donors_ADLT_inocula_table_GENUS)
human_donors_ADLT_GENUS_table <- as.data.frame(human_donors_ADLT_inocula_table_GENUS)
colnames(human_donors_ADLT_GENUS_table)[colnames(human_donors_ADLT_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_ADLT_GENUS_table)[colnames(human_donors_ADLT_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for ADLT donor 

human_donors_ADLT_round_2_inocula_for_mice_ps.prop <- transform_sample_counts(human_donors_ADLT_round_2_inocula_for_mice_ps, function(x) x/sum(x))

ADLT_round_2_donor_core_ps.prop <- prune_taxa(core_ADLT_round_2_donor_taxa_mice, human_donors_ADLT_round_2_inocula_for_mice_ps.prop)
ADLT_round_2_donor_core_ps.prop 

## at PHYLUM level for donors
tb_ADLT_core_ASVs_PHYLUM <- tax_glom(ADLT_round_2_donor_core_ps.prop , "Phylum", NArm = FALSE)

tb_ADLT_core_ASVs_PHYLUM_df <- psmelt(tb_ADLT_core_ASVs_PHYLUM) %>% as_tibble
tb_ADLT_core_ASVs_PHYLUM_df_0 <- tb_ADLT_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_PHYLUM_df_0_rounded <- tb_ADLT_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_PHYLUM_df_0_rounded)

df_donor_ADLT_PHYLUM <- subset(tb_ADLT_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
View(df_donor_ADLT_PHYLUM)
df_donor_ADLT_PHYLUM$sample_Species <- NULL
View(df_donor_ADLT_PHYLUM)


## at FAMILY level for donors
tb_ADLT_core_ASVs_FAMILY <- tax_glom(ADLT_round_2_donor_core_ps.prop, "Family", NArm = FALSE)

tb_ADLT_core_ASVs_FAMILY_df <- psmelt(tb_ADLT_core_ASVs_FAMILY) %>% as_tibble
tb_ADLT_core_ASVs_FAMILY_df_0 <- tb_ADLT_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_FAMILY_df_0_rounded <- tb_ADLT_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_FAMILY_df_0_rounded)

df_donor_ADLT_FAMILY <- subset(tb_ADLT_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")

View(df_donor_ADLT_FAMILY)
df_donor_ADLT_FAMILY$sample_Species <- NULL
View(df_donor_ADLT_FAMILY)


## at GENUS level for donors
tb_ADLT_core_ASVs_GENUS <- tax_glom(ADLT_round_2_donor_core_ps.prop, "Genus", NArm = FALSE)

tb_ADLT_core_ASVs_GENUS_df <- psmelt(tb_ADLT_core_ASVs_GENUS) %>% as_tibble
tb_ADLT_core_ASVs_GENUS_df_0 <- tb_ADLT_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_GENUS_df_0_rounded <- tb_ADLT_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_GENUS_df_0_rounded)

df_donor_ADLT_GENUS <- subset(tb_ADLT_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")

View(df_donor_ADLT_GENUS)
df_donor_ADLT_GENUS$sample_Species <- NULL
View(df_donor_ADLT_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
ADLT_donor_PHYLUM_df <- merge(human_donors_ADLT_PHYLUM_table,df_donor_ADLT_PHYLUM, by="Phylum")
View(ADLT_donor_PHYLUM_df)

ADLT_donor_FAMILY_df <- merge(human_donors_ADLT_FAMILY_table,df_donor_ADLT_FAMILY, by="Family")
View(ADLT_donor_FAMILY_df)

ADLT_donor_GENUS_df <- merge(human_donors_ADLT_GENUS_table,df_donor_ADLT_GENUS, by="Genus")
View(ADLT_donor_GENUS_df)


## for mice
core_ADLT_round_2_taxa_colonizing_mice # 85 of the 117 Donor_6 core ASVs have colonized the mice

ADLT_round_2_HMA_mice_donor_rel_taxa_ps <- prune_taxa(core_ADLT_round_2_taxa_colonizing_mice, ADLT_round_2_HMA_mice_donor_taxa_ps)

# at PHYLUM level
ADLT_rel_taxa_mouse_table_PHYLUM <- table(tax_table(ADLT_round_2_HMA_mice_donor_rel_taxa_ps)[, "Phylum"], exclude = NULL)
View(ADLT_rel_taxa_mouse_table_PHYLUM)
rel_taxa_mice_ADLT_PHYLUM_table <- as.data.frame(ADLT_rel_taxa_mouse_table_PHYLUM)
colnames(rel_taxa_mice_ADLT_PHYLUM_table)[colnames(rel_taxa_mice_ADLT_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(rel_taxa_mice_ADLT_PHYLUM_table)[colnames(rel_taxa_mice_ADLT_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_ADLT_PHYLUM_table)

# at FAMILY level
ADLT_rel_taxa_mouse_table_FAMILY <- table(tax_table(ADLT_round_2_HMA_mice_donor_rel_taxa_ps)[, "Family"], exclude = NULL)
View(ADLT_rel_taxa_mouse_table_FAMILY)
rel_taxa_mice_ADLT_FAMILY_table <- as.data.frame(ADLT_rel_taxa_mouse_table_FAMILY)
colnames(rel_taxa_mice_ADLT_FAMILY_table)[colnames(rel_taxa_mice_ADLT_FAMILY_table) == "Var1"] <- "Family"
colnames(rel_taxa_mice_ADLT_FAMILY_table)[colnames(rel_taxa_mice_ADLT_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_ADLT_FAMILY_table)

# at GENUS level
ADLT_rel_taxa_mouse_table_GENUS <- table(tax_table(ADLT_round_2_HMA_mice_donor_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(ADLT_rel_taxa_mouse_table_GENUS)
rel_taxa_mice_ADLT_GENUS_table <- as.data.frame(ADLT_rel_taxa_mouse_table_GENUS)
colnames(rel_taxa_mice_ADLT_GENUS_table)[colnames(rel_taxa_mice_ADLT_GENUS_table) == "Var1"] <- "Genus"
colnames(rel_taxa_mice_ADLT_GENUS_table)[colnames(rel_taxa_mice_ADLT_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_ADLT_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for HMA mice

ADLT_round_2_HMA_mice_ps.prop <- transform_sample_counts(ADLT_round_2_HMA_mice_ps, function(x) x/sum(x)) # convert to relative abundance

# retain only the ASVs colonizing mice
ADLT_round_2_mice_ps.prop <- prune_taxa(core_ADLT_round_2_taxa_colonizing_mice, ADLT_round_2_HMA_mice_ps.prop)

# at PHYLUM level for mice
tb_ADLT_core_ASVs_PHYLUM <- tax_glom(ADLT_round_2_mice_ps.prop, "Phylum", NArm = FALSE)

tb_ADLT_core_ASVs_PHYLUM_df <- psmelt(tb_ADLT_core_ASVs_PHYLUM) %>% as_tibble
tb_ADLT_core_ASVs_PHYLUM_df_0 <- tb_ADLT_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_PHYLUM_df_0_rounded <- tb_ADLT_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_PHYLUM_df_0_rounded)

df_mice_ADLT_PHYLUM <- subset(tb_ADLT_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
View(df_mice_ADLT_PHYLUM)
df_mice_ADLT_PHYLUM$sample_Species <- NULL
View(df_mice_ADLT_PHYLUM)


## at FAMILY level for mice
tb_ADLT_core_ASVs_FAMILY <- tax_glom(ADLT_round_2_mice_ps.prop, "Family", NArm = FALSE)

tb_ADLT_core_ASVs_FAMILY_df <- psmelt(tb_ADLT_core_ASVs_FAMILY) %>% as_tibble
tb_ADLT_core_ASVs_FAMILY_df_0 <- tb_ADLT_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_FAMILY_df_0_rounded <- tb_ADLT_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_FAMILY_df_0_rounded)

df_mice_ADLT_FAMILY <- subset(tb_ADLT_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")

View(df_mice_ADLT_FAMILY)
df_mice_ADLT_FAMILY$sample_Species <- NULL
View(df_mice_ADLT_FAMILY)


## at GENUS level for mice
tb_ADLT_core_ASVs_GENUS <- tax_glom(ADLT_round_2_mice_ps.prop, "Genus", NArm = FALSE)

tb_ADLT_core_ASVs_GENUS_df <- psmelt(tb_ADLT_core_ASVs_GENUS) %>% as_tibble
tb_ADLT_core_ASVs_GENUS_df_0 <- tb_ADLT_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT_core_ASVs_GENUS_df_0_rounded <- tb_ADLT_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT_core_ASVs_GENUS_df_0_rounded)

df_mice_ADLT_GENUS <- subset(tb_ADLT_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")

View(df_mice_ADLT_GENUS)
df_mice_ADLT_GENUS$sample_Species <- NULL
View(df_mice_ADLT_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
ADLT_mice_PHYLUM_df <- merge(rel_taxa_mice_ADLT_PHYLUM_table,df_mice_ADLT_PHYLUM, by="Phylum")
colnames(ADLT_mice_PHYLUM_df)[colnames(ADLT_mice_PHYLUM_df) == "Mean"] <- "Mean-mouse"
colnames(ADLT_mice_PHYLUM_df)[colnames(ADLT_mice_PHYLUM_df) == "SD"] <- "SD-mouse"
View(ADLT_mice_PHYLUM_df)

ADLT_mice_FAMILY_df <- merge(rel_taxa_mice_ADLT_FAMILY_table,df_mice_ADLT_FAMILY, by="Family")
colnames(ADLT_mice_FAMILY_df)[colnames(ADLT_mice_FAMILY_df) == "Mean"] <- "Mean-mouse"
colnames(ADLT_mice_FAMILY_df)[colnames(ADLT_mice_FAMILY_df) == "SD"] <- "SD-mouse"
View(ADLT_mice_FAMILY_df)

ADLT_mice_GENUS_df <- merge(rel_taxa_mice_ADLT_GENUS_table,df_mice_ADLT_GENUS, by="Genus")
colnames(ADLT_mice_GENUS_df)[colnames(ADLT_mice_GENUS_df) == "Mean"] <- "Mean-mouse"
colnames(ADLT_mice_GENUS_df)[colnames(ADLT_mice_GENUS_df) == "SD"] <- "SD-mouse"
View(ADLT_mice_GENUS_df)


# merged the data frames from human donors and HMA mice
merged_ADLT_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT_donor_PHYLUM_df,ADLT_mice_PHYLUM_df))
View(merged_ADLT_PHYLUM)

merged_ADLT_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT_donor_FAMILY_df,ADLT_mice_FAMILY_df))
View(merged_ADLT_FAMILY)

merged_ADLT_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT_donor_GENUS_df,ADLT_mice_GENUS_df))
View(merged_ADLT_GENUS)

write.table(merged_ADLT_PHYLUM, "merged_ADLT_round_2_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_ADLT_FAMILY, "merged_ADLT_round_2_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_ADLT_GENUS, "merged_ADLT_round_2_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**Donor_7(SNR)**

```{r,echo=TRUE}

### for SNR donor
human_donor_inocula_with_rel_taxa_ps <- prune_taxa(core_SNR_round_2_donor_taxa_mice, human_donors_SNR_round_2_inocula_for_mice_ps)

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_SNR_inocula_table_PHYLUM <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Phylum"], exclude = NULL)
human_donors_SNR_PHYLUM_table <- as.data.frame(human_donors_SNR_inocula_table_PHYLUM)
colnames(human_donors_SNR_PHYLUM_table)[colnames(human_donors_SNR_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_SNR_PHYLUM_table)[colnames(human_donors_SNR_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_SNR_PHYLUM_table)

# at FAMILY level
human_donors_SNR_inocula_table_FAMILY <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Family"], exclude = NULL)
View(human_donors_SNR_inocula_table_FAMILY)
human_donors_SNR_FAMILY_table <- as.data.frame(human_donors_SNR_inocula_table_FAMILY)
colnames(human_donors_SNR_FAMILY_table)[colnames(human_donors_SNR_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_SNR_FAMILY_table)[colnames(human_donors_SNR_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_SNR_FAMILY_table)

# at GENUS level
human_donors_SNR_inocula_table_GENUS <- table(tax_table(human_donor_inocula_with_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(human_donors_SNR_inocula_table_GENUS)
human_donors_SNR_GENUS_table <- as.data.frame(human_donors_SNR_inocula_table_GENUS)
colnames(human_donors_SNR_GENUS_table)[colnames(human_donors_SNR_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_SNR_GENUS_table)[colnames(human_donors_SNR_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_SNR_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for SNR donor 

human_donors_SNR_round_2_inocula_for_mice_ps.prop <- transform_sample_counts(human_donors_SNR_round_2_inocula_for_mice_ps, function(x) x/sum(x))

SNR_round_2_donor_core_ps.prop <- prune_taxa(core_SNR_round_2_donor_taxa_mice, human_donors_SNR_round_2_inocula_for_mice_ps.prop)
SNR_round_2_donor_core_ps.prop # 151 core ASVs

## at PHYLUM level for donors
tb_SNR_core_ASVs_PHYLUM <- tax_glom(SNR_round_2_donor_core_ps.prop, "Phylum", NArm = FALSE)

tb_SNR_core_ASVs_PHYLUM_df <- psmelt(tb_SNR_core_ASVs_PHYLUM) %>% as_tibble
tb_SNR_core_ASVs_PHYLUM_df_0 <- tb_SNR_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_PHYLUM_df_0_rounded <- tb_SNR_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_PHYLUM_df_0_rounded)

df_donor_SNR_PHYLUM <- subset(tb_SNR_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
View(df_donor_SNR_PHYLUM)
df_donor_SNR_PHYLUM$sample_Species <- NULL
View(df_donor_SNR_PHYLUM)


## at FAMILY level for donors
tb_SNR_core_ASVs_FAMILY <- tax_glom(SNR_round_2_donor_core_ps.prop, "Family", NArm = FALSE)

tb_SNR_core_ASVs_FAMILY_df <- psmelt(tb_SNR_core_ASVs_FAMILY) %>% as_tibble
tb_SNR_core_ASVs_FAMILY_df_0 <- tb_SNR_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_FAMILY_df_0_rounded <- tb_SNR_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_FAMILY_df_0_rounded)

df_donor_SNR_FAMILY <- subset(tb_SNR_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")

View(df_donor_SNR_FAMILY)
df_donor_SNR_FAMILY$sample_Species <- NULL
View(df_donor_SNR_FAMILY)


## at GENUS level for donors
tb_SNR_core_ASVs_GENUS <- tax_glom(SNR_round_2_donor_core_ps.prop, "Genus", NArm = FALSE)

tb_SNR_core_ASVs_GENUS_df <- psmelt(tb_SNR_core_ASVs_GENUS) %>% as_tibble
tb_SNR_core_ASVs_GENUS_df_0 <- tb_SNR_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_GENUS_df_0_rounded <- tb_SNR_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_GENUS_df_0_rounded)

df_donor_SNR_GENUS <- subset(tb_SNR_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")

View(df_donor_SNR_GENUS)
df_donor_SNR_GENUS$sample_Species <- NULL
View(df_donor_SNR_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
SNR_donor_PHYLUM_df <- merge(human_donors_SNR_PHYLUM_table,df_donor_SNR_PHYLUM, by="Phylum")
View(SNR_donor_PHYLUM_df)

SNR_donor_FAMILY_df <- merge(human_donors_SNR_FAMILY_table,df_donor_SNR_FAMILY, by="Family")
View(SNR_donor_FAMILY_df)

SNR_donor_GENUS_df <- merge(human_donors_SNR_GENUS_table,df_donor_SNR_GENUS, by="Genus")
View(SNR_donor_GENUS_df)


## for mice
core_SNR_round_2_taxa_colonizing_mice # 68 of the 151 Donor_7 core ASVs have colonized the mice

SNR_round_2_HMA_mice_donor_rel_taxa_ps <- prune_taxa(core_SNR_round_2_taxa_colonizing_mice, SNR_round_2_HMA_mice_donor_taxa_ps)

# at PHYLUM level
SNR_rel_taxa_mouse_table_PHYLUM <- table(tax_table(SNR_round_2_HMA_mice_donor_rel_taxa_ps)[, "Phylum"], exclude = NULL)
View(SNR_rel_taxa_mouse_table_PHYLUM)
rel_taxa_mice_SNR_PHYLUM_table <- as.data.frame(SNR_rel_taxa_mouse_table_PHYLUM)
colnames(rel_taxa_mice_SNR_PHYLUM_table)[colnames(rel_taxa_mice_SNR_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(rel_taxa_mice_SNR_PHYLUM_table)[colnames(rel_taxa_mice_SNR_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_SNR_PHYLUM_table)

# at FAMILY level
SNR_rel_taxa_mouse_table_FAMILY <- table(tax_table(SNR_round_2_HMA_mice_donor_rel_taxa_ps)[, "Family"], exclude = NULL)
View(SNR_rel_taxa_mouse_table_FAMILY)
rel_taxa_mice_SNR_FAMILY_table <- as.data.frame(SNR_rel_taxa_mouse_table_FAMILY)
colnames(rel_taxa_mice_SNR_FAMILY_table)[colnames(rel_taxa_mice_SNR_FAMILY_table) == "Var1"] <- "Family"
colnames(rel_taxa_mice_SNR_FAMILY_table)[colnames(rel_taxa_mice_SNR_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_SNR_FAMILY_table)

# at GENUS level
SNR_rel_taxa_mouse_table_GENUS <- table(tax_table(SNR_round_2_HMA_mice_donor_rel_taxa_ps)[, "Genus"], exclude = NULL)
View(SNR_rel_taxa_mouse_table_GENUS)
rel_taxa_mice_SNR_GENUS_table <- as.data.frame(SNR_rel_taxa_mouse_table_GENUS)
colnames(rel_taxa_mice_SNR_GENUS_table)[colnames(rel_taxa_mice_SNR_GENUS_table) == "Var1"] <- "Genus"
colnames(rel_taxa_mice_SNR_GENUS_table)[colnames(rel_taxa_mice_SNR_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(rel_taxa_mice_SNR_GENUS_table)


## Calculating mean and standard deviation for each phylum, family, and genus level for HMA mice

SNR_round_2_HMA_mice_ps.prop <- transform_sample_counts(SNR_round_2_HMA_mice_ps, function(x) x/sum(x)) # convert to relative abundance

# retain only the ASVs colonizing mice
SNR_round_2_mice_ps.prop <- prune_taxa(core_SNR_round_2_taxa_colonizing_mice, SNR_round_2_HMA_mice_ps.prop)

# at PHYLUM level for mice
tb_SNR_core_ASVs_PHYLUM <- tax_glom(SNR_round_2_mice_ps.prop, "Phylum", NArm = FALSE)

tb_SNR_core_ASVs_PHYLUM_df <- psmelt(tb_SNR_core_ASVs_PHYLUM) %>% as_tibble
tb_SNR_core_ASVs_PHYLUM_df_0 <- tb_SNR_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_PHYLUM_df_0_rounded <- tb_SNR_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_PHYLUM_df_0_rounded)

df_mice_SNR_PHYLUM <- subset(tb_SNR_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
View(df_mice_SNR_PHYLUM)
df_mice_SNR_PHYLUM$sample_Species <- NULL
View(df_mice_SNR_PHYLUM)

## at FAMILY level for mice
tb_SNR_core_ASVs_FAMILY <- tax_glom(SNR_round_2_mice_ps.prop, "Family", NArm = FALSE)

tb_SNR_core_ASVs_FAMILY_df <- psmelt(tb_SNR_core_ASVs_FAMILY) %>% as_tibble
tb_SNR_core_ASVs_FAMILY_df_0 <- tb_SNR_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_FAMILY_df_0_rounded <- tb_SNR_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_FAMILY_df_0_rounded)

df_mice_SNR_FAMILY <- subset(tb_SNR_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")

View(df_mice_SNR_FAMILY)
df_mice_SNR_FAMILY$sample_Species <- NULL
View(df_mice_SNR_FAMILY)

## at GENUS level for mices
tb_SNR_core_ASVs_GENUS <- tax_glom(SNR_round_2_mice_ps.prop, "Genus", NArm = FALSE)

tb_SNR_core_ASVs_GENUS_df <- psmelt(tb_SNR_core_ASVs_GENUS) %>% as_tibble
tb_SNR_core_ASVs_GENUS_df_0 <- tb_SNR_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_SNR_core_ASVs_GENUS_df_0_rounded <- tb_SNR_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_SNR_core_ASVs_GENUS_df_0_rounded)

df_mice_SNR_GENUS <- subset(tb_SNR_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")

View(df_mice_SNR_GENUS)
df_mice_SNR_GENUS$sample_Species <- NULL
View(df_mice_SNR_GENUS)


# Merging the count and abundance dataframes at the phylum, family, and genus levels
SNR_mice_PHYLUM_df <- merge(rel_taxa_mice_SNR_PHYLUM_table,df_mice_SNR_PHYLUM, by="Phylum")
colnames(SNR_mice_PHYLUM_df)[colnames(SNR_mice_PHYLUM_df) == "Mean"] <- "Mean-mouse"
colnames(SNR_mice_PHYLUM_df)[colnames(SNR_mice_PHYLUM_df) == "SD"] <- "SD-mouse"
View(SNR_mice_PHYLUM_df)

SNR_mice_FAMILY_df <- merge(rel_taxa_mice_SNR_FAMILY_table,df_mice_SNR_FAMILY, by="Family")
colnames(SNR_mice_FAMILY_df)[colnames(SNR_mice_FAMILY_df) == "Mean"] <- "Mean-mouse"
colnames(SNR_mice_FAMILY_df)[colnames(SNR_mice_FAMILY_df) == "SD"] <- "SD-mouse"
View(SNR_mice_FAMILY_df)

SNR_mice_GENUS_df <- merge(rel_taxa_mice_SNR_GENUS_table,df_mice_SNR_GENUS, by="Genus")
colnames(SNR_mice_GENUS_df)[colnames(SNR_mice_GENUS_df) == "Mean"] <- "Mean-mouse"
colnames(SNR_mice_GENUS_df)[colnames(SNR_mice_GENUS_df) == "SD"] <- "SD-mouse"
View(SNR_mice_GENUS_df)


# merged the data frames from human donors and HMA mice
merged_SNR_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(SNR_donor_PHYLUM_df,SNR_mice_PHYLUM_df))
View(merged_SNR_PHYLUM)

merged_SNR_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(SNR_donor_FAMILY_df,SNR_mice_FAMILY_df))
View(merged_SNR_FAMILY)

merged_SNR_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(SNR_donor_GENUS_df,SNR_mice_GENUS_df))
View(merged_SNR_GENUS)

write.table(merged_SNR_PHYLUM, "merged_SNR_round_2_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_SNR_FAMILY, "merged_SNR_round_2_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
write.table(merged_SNR_GENUS, "merged_SNR_round_2_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```



##**Making a phylogenetic tree for the core Firmicutes globally across donors**##

```{r, echo=TRUE}
## identifying the 'global' Firmicutes ASVs
core_INF2_taxa
core_CHLD2_taxa
core_ADLT2_taxa
core_ELD2_taxa

union_core_donor_ASVs <- Reduce(union, list(core_INF2_taxa,core_CHLD2_taxa,core_ADLT2_taxa,core_ELD2_taxa))
str(union_core_donor_ASVs) # 251 core ASVs represented across all donors ('union')

union_core_donor_ASVs_ps <- prune_taxa(union_core_donor_ASVs,merged_pig_mouse_comp_complete_ps)
union_core_donor_ASVs_Firmicutes_ps <- subset_taxa(union_core_donor_ASVs_ps, Phylum == "Firmicutes")
union_core_donor_ASVs_Firmicutes_ps
all_core_Firmicutes <- taxa_names(union_core_donor_ASVs_Firmicutes_ps) # 178 Firmicutes ASVs
write.csv(all_core_Firmicutes, "all_core_Firmicutes.txt", col.names = NA, quote = F, row.names = F)

phylo_tree_core_firmicutes <- phy_tree(union_core_donor_ASVs_Firmicutes_ps)
ape::write.tree(phylo_tree_core_firmicutes, "Firmicutes_tree.tre")

tax <- as(tax_table(union_core_donor_ASVs_Firmicutes_ps), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep=";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "all_core_firmicutes_taxonomy.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## checking how many of the core 'union' Firmicutes ASVs had colonized each HMA animal model

# for the HMA mice
core_INF2_taxa_colonizing_mice
core_CHLD2_taxa_colonizing_mice
core_ADLT2_taxa_colonizing_mice
core_ELD2_taxa_colonizing_mice

all_core_donor_ASVs_colonizing_mice <- Reduce(union, list(core_INF2_taxa_colonizing_mice,core_CHLD2_taxa_colonizing_mice,core_ADLT2_taxa_colonizing_mice,core_ELD2_taxa_colonizing_mice))
str(all_core_donor_ASVs_colonizing_mice)

all_core_firmicute_ASVs_colonizing_mice <- Reduce(intersect, list(all_core_Firmicutes,all_core_donor_ASVs_colonizing_mice))
all_core_firmicute_ASVs_colonizing_mice # 69 of the core 'union' firmicute ASVs have colonized the mice
write.csv(all_core_firmicute_ASVs_colonizing_mice, "all_core_firmicute_ASVs_colonizing_mice.txt", col.names = NA, quote = F, row.names = F)


# for the HMA piglets
core_INF2_taxa_colonizing_piglets
core_CHLD2_taxa_colonizing_piglets
core_ADLT2_taxa_colonizing_piglets
core_ELD2_taxa_colonizing_piglets

all_core_donor_ASVs_colonizing_piglets <- Reduce(union, list(core_INF2_taxa_colonizing_piglets,core_CHLD2_taxa_colonizing_piglets,core_ADLT2_taxa_colonizing_piglets,core_ELD2_taxa_colonizing_piglets))
str(all_core_donor_ASVs_colonizing_piglets)

all_core_firmicute_ASVs_colonizing_piglets <- Reduce(intersect, list(all_core_Firmicutes,all_core_donor_ASVs_colonizing_piglets))
all_core_firmicute_ASVs_colonizing_piglets # 131 of the ASVs have colonized the piglets
write.csv(all_core_firmicute_ASVs_colonizing_piglets, "all_core_firmicute_ASVs_colonizing_piglets.txt", col.names = NA, quote = F, row.names = F)

list_1 <- list(all_core_Firmicutes,all_core_firmicute_ASVs_colonizing_mice,all_core_firmicute_ASVs_colonizing_piglets)
x = unique(unlist(list_1))

df_test <- data.frame(lapply(list_1, function(y) ifelse(is.element(x,y),x,NA)))
View(df_test)

write.table(df_test, "firmicutes_distribution.txt", sep = "\t")


## checking how many of the core 'union' Firmicutes ASVs had persistently colonized each HMA animal model

# for the HMA mice
perst_colonizing_core_INF2_mice
perst_colonizing_core_CHLD2_mice
perst_colonizing_core_ADLT2_mice
perst_colonizing_core_ELD2_mice

all_core_donor_ASVs_perst_colonizing_mice <- Reduce(union, list(perst_colonizing_core_INF2_mice,perst_colonizing_core_CHLD2_mice,perst_colonizing_core_ADLT2_mice,perst_colonizing_core_ELD2_mice))

all_core_firmicute_ASVs_perst_colonizing_mice <- Reduce(intersect, list(all_core_Firmicutes,all_core_donor_ASVs_perst_colonizing_mice))
str(all_core_firmicute_ASVs_perst_colonizing_mice) # 24 of the core 'union' Firmicutes ASVs have persistently colonized the mice


# for the HMA piglets
perst_colonizing_core_INF2_piglets
perst_colonizing_core_CHLD2_piglets
perst_colonizing_core_ADLT2_piglets
perst_colonizing_core_ELD2_piglets

all_core_donor_ASVs_perst_colonizing_piglets <- Reduce(union, list(perst_colonizing_core_INF2_piglets,perst_colonizing_core_CHLD2_piglets,perst_colonizing_core_ADLT2_piglets,perst_colonizing_core_ELD2_piglets))

all_core_firmicute_ASVs_perst_colonizing_piglets <- Reduce(intersect, list(all_core_Firmicutes,all_core_donor_ASVs_perst_colonizing_piglets))
str(all_core_firmicute_ASVs_perst_colonizing_piglets) # 74 of the core 'union' Firmicute ASVs have persistently colonized the piglets

list_2 <- list(all_core_Firmicutes,all_core_firmicute_ASVs_perst_colonizing_mice,all_core_firmicute_ASVs_perst_colonizing_piglets)
x = unique(unlist(list_2))

df_test_2 <- data.frame(lapply(list_2, function(y) ifelse(is.element(x,y),x,NA)))
View(df_test_2)

write.table(df_test_2, "Firmicutes_distribution_perst_coloniz.txt", sep = "\t")
```


