---
title: "Pig_Mouse_Comparison_Study"
author: "Nirosh D. Aluthge"
date: "3/27/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = '/Users/niroshaluthge/Desktop/NDA/manuscripts/pig_mouse_comparison_study/')
knitr::opts_chunk$set(echo = TRUE)
```

<div style="margin-bottom:50px;">
</div>
#### **INTRODUCTION** 

This R Markdown file details all the data analysis steps used in the manuscript **"Longitudinal establishment of human fecal bacterial communities in human microbiota-associated (HMA) porcine and murine models"** authored by Aluthge et al.


#### **DATA ANALYSIS**

All the steps of this analysis (unless mentioned otherwise) was performed on an Apple MacBook pro with a 2.3 GHz Intel Core i5 processor and an 8 GB memory 

**Note on sequencing: ** The data used in this analysis was generated through barcoded multiplexed amplicon sequencing of the V4 hypervariable region of the bacterial 16S rRNA gene using an Illumina MiSeq sequencing platform. Since there were > 490 samples, the initial sequencing was carried out as two sequencing runs (pig_mouse_comp_sequencing_run1 and pig_mouse_comp_sequencing_run2). Subsequently, based on sequencing depths, some samples which had yielded low numbers of sequence reads were resequenced in two separate 'resequencing' runs (pig_mouse_comp_resequencing_run1 and pig_mouse_comp_resequencing_run2) with the aim of increasing the sequencing depth of these samples. Since the error profiles determined in the DADA2 data analysis pipeline is run-dependent, the initial data analysis steps were conducted separately for each sequencing run and at the end all the individual 'feature tables' were merged to make a final 'master' feature table containing all the sample x feature count data.

```{r, echo=TRUE}
#This data analysis pipeline follows the data analysis of microbiome data using ASVs as described by Callahan et al. in  https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html

#Loading or installing required packages
library("knitr")
library("BiocStyle")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if (any(!.inst)) {
  install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
if (any(!.inst)) {
  source("http://bioconductor.org/biocLite.R")
  biocLite(.bioc_packages[!.inst], ask = F)
}

sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
library("BiocStyle"); packageVersion("BiocStyle")
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("DECIPHER")
library("DECIPHER");packageVersion("DECIPHER")
```
Now that the required main packages have been loaded, we can proceed with the data analysis steps.

We will first pick ASVs using the DADA2 pipeline for the '.fastq' files from our very first sequencing run. These '.fastq' files are in the folder 'pig_mouse_comparisons_r2_first_run_fastq_files_no_pre_inoc'. There are **274** files corresponding to **137** samples.

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

#change to required work directory
setwd("/Users/niroshaluthge/Desktop/NDA/manuscripts/pig_mouse_comparison_study")
getwd()

MiSeq_path <- "./pig_mouse_comparisons_r2_first_run_fastq_files_no_pre_inoc" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

#Separate the 'forward' and 'reverse' reads
fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

#Look at base quality plots
plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

#filter the forward and reverse reads
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(250,140), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) #the trunLen values were determined based off of the quality profile plots from above
write.table(out, "out_file_run_1.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 3061044 input reads and 2821030 output reads, so we have retained ~ 92.1% of the reads after this filtering and trim step

#dereplicating to eliminate redundancy (multiple occurances of the same identical sequence)
derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

#Learning error rates
errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

#inferring sequences through DADA2 method
dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] #checking how many sequence variants were inferred from first sample

#Construct sample by sequence feature table
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_run1 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_run1)))
sum(seqtabAll_run1) # 2673152 reads had remained after the mergers. Since we had 3061044 total input reads, this means that ~ 87.3% of reads were retained

#remove non-target-length sequences
seqtabAll2_run1 <- seqtabAll_run1[,nchar(colnames(seqtabAll_run1)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_run1)))
dim(seqtabAll2_run1)
sum(seqtabAll2_run1) # 2673118 sequences retained

#Store seqtabAll2 from run 1 as a serialized R object
saveRDS(seqtabAll2_run1, "seqtabAll2_run1.rds")
```

Now let's run the same pipeline for the '.fastq' files from our second sequencing run. These fastq files are in the folder "pig_mouse_comparisons_r2_second_run_fastq_files" and consist of **492** .fastq files from **246** samples

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

#change to required work directory
setwd("/Users/niroshaluthge/Desktop/NDA/manuscripts/pig_mouse_comparison_study")
getwd()

MiSeq_path <- "./pig_mouse_comparisons_r2_second_run_fastq_files" 
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") 
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(240,100), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_run_2.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 12907795 input reads and 10492999 output reads, so we have retained ~ 81.2 % of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_run2 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_run2)))
sum(seqtabAll_run2) # 9793401 reads had remained after the mergers. Since we had 12907795 total input reads, this means that ~ 76 % of reads were retained

seqtabAll2_run2 <- seqtabAll_run2[,nchar(colnames(seqtabAll_run2)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_run2)))
dim(seqtabAll2_run2)
sum(seqtabAll2_run2) # 9790523 sequences retained

#Store seqtabAll2 from run 2 as a serialized R object
saveRDS(seqtabAll2_run2, "seqtabAll2_run2.rds")
```
Now we can merge the two sequence tables together:

```{r, echo=TRUE}
seqtab_run1 <- readRDS("seqtabAll2_run1.rds")
seqtab_run2 <- readRDS("seqtabAll2_run2.rds")
seqtab_runs_1_and_2 <- mergeSequenceTables(seqtab_run1, seqtab_run2)
```
Let's make sure that the merging has been done properly :

```{r, echo=TRUE}
dim(seqtab_run1)
sum(seqtab_run1)
dim(seqtab_run2)
sum(seqtab_run2)
dim(seqtab_runs_1_and_2)
sum(seqtab_runs_1_and_2)
saveRDS(seqtab_runs_1_and_2, "seqtab_runs_1_and_2.rds")
```
We can see that the number of samples and the total number of sequences matchup. The number of ASVs don't add-up because there appear to be shared ASVs between the two tables.

After looking at the read depth for each sample, we realized that there were quite a few samples with low sequencing depths (including some of the donor samples and innocula). In addition we didn't have samples 97-192 in our first two sequencing runs either. Therefore, all these samples were 'resequenced' using the same Illumina MiSeq instrument and targeting the same V4 hypervariable region in order to increase read depths of these samples. The '.fastq' files from these 'resequencing' runs are in folders 'pig_mouse_comparison_resequencing_run_1_fastq_files' and 'pig_mouse_comparison_resequencing_run_2_fastq_files', respectively.

Running the fastq files (**614** fastq files from **307** samples) from the first resequencing run using the dada2 pipeline:

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

MiSeq_path <- "./pig_mouse_comparison_resequencing_run_1_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") # folder for quality-filtered reads
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(245,145), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_reseq_run_1.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 4344562 input reads and 3991675 output reads, so we have retained ~92% of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_reseq_run1 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_reseq_run1)))
sum(seqtabAll_reseq_run1) # 3680068 reads had remained after the mergers. Since we had 4309906 total input reads, this means that 84.7% of reads were retained

seqtabAll2_reseq_run1 <- seqtabAll_reseq_run1[,nchar(colnames(seqtabAll_reseq_run1)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_reseq_run1)))
dim(seqtabAll2_reseq_run1)
sum(seqtabAll2_reseq_run1) # 3679625 sequences retained

#Store seqtabAll2 from resequencing run 1 as a serialized R object
saveRDS(seqtabAll2_reseq_run1, "seqtabAll2_reseq_run1.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_reseq_1 <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs,getN), sapply (mergers, getN), rowSums(seqtabAll2_reseq_run1))
colnames(track_reseq_1) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "seqtable")
rownames(track_reseq_1) <- sampleNames
head(track_reseq_1)
write.table(track_reseq_1, 'tracking_reads_through_pipeline_reseq_plate_1.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```

The developers of phyloseq recently developed the 'decontam R package' for detecting reads coming from external contaminants (reagent contaminants, lab-workers, etc.). We'll implement the 'prevalence' function of this decontam package to identify possible contaminant ASVs based on their prevalence in negative controls:

```{r, echo=TRUE}
# Detection of contaminants using 'decontam'
library("decontam") ; packageVersion("decontam") 
decontam_sample_data_reseq_1 <- read.table("pig_mouse_comp_resequencing_run_1_decont_mapp_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(decontam_sample_data_reseq_1) <- as.character(decontam_sample_data_reseq_1[,1])
decontam_mp_reseq_1 <- phyloseq(otu_table(seqtabAll2_reseq_run1, taxa_are_rows = FALSE), sample_data(decontam_sample_data_reseq_1))

## ANOTHER way to get mapping file read into R
# decontam_sample_data_reseq_1 <- read.table("pig_mouse_comp_resequencing_run_1_decont_mapp_file.txt", sep = "\t", header = TRUE)
# rownames(decontam_sample_data_reseq_1) = decontam_sample_data_reseq_1$SampleID

# identifying contaminants using the 'prevalence' method
sample_data(decontam_mp_reseq_1)$is.neg <- sample_data(decontam_mp_reseq_1)$Sample_or_Control == "negative_control"
contamdf.prev <- isContaminant(decontam_mp_reseq_1, method = "prevalence", neg = "is.neg", threshold = 0.1) # default threshold of 0.1 used
hist(contamdf.prev$p, 100, ylim = c(0,400), xlim = c(0,1))
table(contamdf.prev$contaminant) # TRUE ones are the contaminants
```

The package has identified **14** ASVs that are potential contaminants. Identification and removal of these contaminants:

```{r, echo=TRUE}
contaminants_reseq_1 <- which(contamdf.prev$contaminant) # Won't return the sequences but the numbers corresponding to the sequences like '4508'
contaminants_reseq_1 <- paste0("ASV_", contaminants_reseq_1)  # renaming contaminants so that they have the format 'ASV_4508'
contaminants_reseq_1

# Now we have a list of the ASVs which are contaminants
# However, our 'decontam_mp_reseq_1' doesn't have the ASVs labeled like 'ASV_4508'. They still have the actual sequences
# Therefore, let's relabel the exact sequences that we have in our feature table as 'ASV_1', 'ASV_2', 'ASV_3', etc.
taxa <- as(otu_table(decontam_mp_reseq_1), "matrix")
taxa_sequences <- colnames(taxa)
head(taxa_sequences) # shows the exact sequences
taxa_ASV <- paste0("ASV_", seq(taxa_sequences)) # replaces exact sequence with 'ASV_i' label 
head(taxa_ASV)

#Let's make a data frame which relates the 'ASV_' identifier to the corresponding exact sequence 
asv_df <- data.frame(taxa_ASV, taxa_sequences) 
nrow(asv_df)
ncol(asv_df)
class(asv_df)
str(asv_df) # We see that the columns are considered factors. Let's change them to characters since they are not really factors.
asv_df$taxa_ASV <- as.character(as.factor(asv_df$taxa_ASV))
asv_df$taxa_sequences <- as.character(as.factor(asv_df$taxa_sequences))
head(asv_df$taxa_ASV)
str(asv_df$taxa_sequences)
str(asv_df) # check to see that the variables are characters

# Now we can use the 'asv_df' to extract the exact sequences which corresponds to the ASVs identified as contaminants in 'contaminants'
# Let's make a function called 'getContamSeqs' so that we can use it for similar situations in the future
# We'll also make a data frame with the 'ASV_id' and corresponding ASV sequence for each contaminant

getContamSeqs <- function(asv_df, contam_ids) {
  contam_asv_id <- character(0) # Makes an empty 'character vector' to store the ASV_id
  contam_asv_seq <- character(0) # Makes an empty 'character vector' to store the exact sequences
  for (i in contam_ids) { # Looping through 'contaminants' using 'ASV_i'
    for (row in 1:nrow(asv_df)) {# Loop through 'asv_df' by row ('ASV_i')
      ASV_num <- asv_df[row, "taxa_ASV"] # Each row is 'separated' by asv_id and asv_sequence
      ASV_seq <- asv_df[row, "taxa_sequences"]
     if (ASV_num == i) { # If 'ASV_' in 'contaminants' matches 'ASV_i' in 'asv_df'
       contam_asv_id <- c(contam_asv_id, ASV_num)
       contam_asv_seq <- c(contam_asv_seq, ASV_seq) # 'Push' asv_sequence to character vector
      }
    }
  } 
  contam_asv_df <- data.frame(contam_asv_id, contam_asv_seq)
  return(list(contam_asv_seq, contam_asv_df))
}

contam_info <- getContamSeqs(asv_df, contaminants_reseq_1)
contam_sequences <- contam_info[[1]] # gets contaminant ASV sequences vector from 'List'
head(contam_sequences) # Now the contaminant sequences are in 'contam_sequences'
str(contam_sequences)
contam_asv_df_reseq_1 <- contam_info[[2]] # gets the data frame with just the contaminant sequences
str(contam_asv_df_reseq_1)
dim(contam_asv_df_reseq_1)
names(contam_asv_df_reseq_1) <- c("ASV_id", "ASV_sequence") # rename data frame columns

# Let's filter the contaminating sequences out of the sequence table
alltaxa = taxa_names(decontam_mp_reseq_1)
goodtaxa = alltaxa[!(alltaxa %in% contam_sequences)]
seqtab_reseq_run_1_no_contam <- prune_taxa(goodtaxa, otu_table(decontam_mp_reseq_1))
dim(seqtab_reseq_run_1_no_contam)

# Let's check how many reads we lost as a result of removing the contaminants
sum(otu_table(seqtab_reseq_run_1_no_contam))/sum(seqtabAll2_reseq_run1) # about ~0.3% of reads were removed as contaminants
seqtab_reseq_run_1_no_contams <- as(otu_table(seqtab_reseq_run_1_no_contam), "matrix") # separating out 'otu table' from phyloseq object
dim(seqtab_reseq_run_1_no_contams)
class(seqtab_reseq_run_1_no_contams)

# Let's save the 'decontaminated' sequence table as an R serial object
saveRDS(seqtab_reseq_run_1_no_contams, "seqtab_reseq_run_1_no_contams.rds")
```

Since identifying what these contaminants are taxonomically might be useful and interesting, let's assign them taxonomy:

```{r, echo=TRUE}
# retaining just the contaminants
alltaxa = taxa_names(decontam_mp_reseq_1)
goodtaxa = alltaxa[(alltaxa %in% contam_sequences)]
seqtab_reseq_run_1_contaminants <- prune_taxa(goodtaxa, otu_table(decontam_mp_reseq_1))
dim(seqtab_reseq_run_1_contaminants)

# assigning taxonomy to the contaminants
seqtab_reseq_run_1_contams <- as(otu_table(seqtab_reseq_run_1_contaminants), "matrix")
fastaRef <- "./silva_nr_v132_train_set.fa" 
taxTab_reseq_1_contams <- assignTaxonomy(seqtab_reseq_run_1_contams, refFasta = fastaRef, multithread = TRUE)
taxTab_reseq_1_contams <- addSpecies(taxTab_reseq_1_contams, "./silva_species_assignment_v132.fa")
dim(taxTab_reseq_1_contams)
View(taxTab_reseq_1_contams)
unname(head(taxTab_reseq_1_contams))

# output the taxonomy table as a .txt file
write.table(taxTab_reseq_1_contams, "contaminant_taxa_table_reseq_1.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
```

Performing dada2 pipeline to pick sequence variants from resequencing run 2 which included **306** fastq files from **153** samples :

```{r, echo=TRUE}
set.seed(100) #for reproducibility?

MiSeq_path <- "./pig_mouse_comparison_resequencing_run_2_fastq_files" #specify path to where the raw '.fastq' files are
list.files(MiSeq_path)

fnFs <- sort(list.files(MiSeq_path, pattern = "_R1_001.fastq"))
fnRs <- sort(list.files(MiSeq_path, pattern = "_R2_001.fastq"))

sampleNames <- sapply(strsplit(fnFs, "_"), `[`, 1)

fnFs <- file.path(MiSeq_path, fnFs)
fnRs <- file.path(MiSeq_path, fnRs)
fnFs[1:3]
fnRs[1:3]

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnRs[1:2])

filt_path <- file.path(MiSeq_path, "filtered") 
if(!file_test("-d", filt_path))  dir.create(filt_path)
filtFs <- file.path(filt_path, paste0(sampleNames, "_F_filt.fastq"))
filtRs <- file.path(filt_path, paste0(sampleNames, "_R_filt.fastq"))

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(240,120), maxN=0, maxEE=c(2,2), truncQ=2, rm.phix = TRUE, compress=TRUE, multithread=TRUE)
head(out) 
write.table(out, "out_file_reseq_run_2.txt", sep = "\t", col.names = NA, row.names = T, quote = F) 

# There were 6539111 input reads and 5811681 output reads, so we have retained ~ 88.9 % of the reads after this filtering and trim step

derepFs <- derepFastq(filtFs, verbose = TRUE)
derepRs <- derepFastq(filtRs, verbose = TRUE)
names(derepFs) <- sampleNames
names(derepRs) <- sampleNames

errF <- learnErrors(filtFs, multithread = TRUE)
errR <- learnErrors(filtRs, multithread = TRUE)
plotErrors(errF)
plotErrors(errR)

dadaFs <- dada(derepFs, err = errF, multithread = TRUE)
dadaRs <- dada(derepRs, err = errR, multithread = TRUE)
dadaFs[1] 

mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)
seqtabAll_reseq_run2 <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtabAll_reseq_run2)))
sum(seqtabAll_reseq_run2) # 5344804 reads had remained after the mergers. Since we had 6539111 total input reads, this means that ~ 81.7% of reads were retained

seqtabAll2_reseq_run2 <- seqtabAll_reseq_run2[,nchar(colnames(seqtabAll_reseq_run2)) %in% seq(250,255)]
table(nchar(getSequences(seqtabAll2_reseq_run2)))
dim(seqtabAll2_reseq_run2)
sum(seqtabAll2_reseq_run2) # 5338176 sequences retained

#Store seqtabAll2 from resequencing run 2 as a serialized R object
saveRDS(seqtabAll2_reseq_run2, "seqtabAll2_reseq_run2.rds")

# track reads through the process
getN <- function(x) sum(getUniques(x))
track_reseq_2 <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs,getN), sapply (mergers, getN), rowSums(seqtabAll2_reseq_run2))
colnames(track_reseq_2) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "seqtable")
rownames(track_reseq_2) <- sampleNames
head(track_reseq_2)
write.table(track_reseq_2, 'tracking_reads_through_pipeline_reseq_plate_2.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```

Removal of potential contaminants using 'decontam' R package for resequencing run 2 :

```{r, echo=TRUE}
seqtabAll2_reseq_run2 <- readRDS ("seqtabAll2_reseq_run2.rds") 
dim(seqtabAll2_reseq_run2)
decontam_sample_data_reseq_2 <- read.table("pig_mouse_comp_resequencing_run_2_decont_mapp_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(decontam_sample_data_reseq_2) <- as.character(decontam_sample_data_reseq_2[,1])
decontam_mp_reseq_2 <- phyloseq(otu_table(seqtabAll2_reseq_run2, taxa_are_rows = FALSE), sample_data(decontam_sample_data_reseq_2))
decontam_mp_reseq_2

# identifying contaminants using the 'prevalence' method
sample_data(decontam_mp_reseq_2)$is.neg <- sample_data(decontam_mp_reseq_2)$Sample_or_Control == "negative_control"
contamdf.prev <- isContaminant(decontam_mp_reseq_2, method = "prevalence", neg = "is.neg", threshold = 0.1)
hist(contamdf.prev$p, 100, ylim = c(0,400), xlim = c(0,1))
table(contamdf.prev$contaminant) # TRUE ones are the contaminants
```

The program identified **38** potential contaminant ASVs. These shall be identified and removed in the following steps :

```{r, echo=TRUE}
contaminants_reseq_2 <- which(contamdf.prev$contaminant) # Won't return the sequences but the numbers corresponding to the sequences like '4508'
contaminants_reseq_2 <- paste0("ASV_", contaminants_reseq_2)  # renaming contaminants so that they have the format 'ASV_4508'
contaminants_reseq_2

# Now we have a list of the ASVs which are contaminants
# However, our 'decontam_mp_reseq_2' doesn't have the ASVs labeled like 'ASV_4508'. They still have the actual sequences
# Therefore, let's relabel the exact sequences that we have in our feature table as 'ASV_1', 'ASV_2', 'ASV_3', etc.
taxa <- as(otu_table(decontam_mp_reseq_2), "matrix")
taxa_sequences <- colnames(taxa)
head(taxa_sequences) # shows the exact sequences
taxa_ASV <- paste0("ASV_", seq(taxa_sequences)) # replaces exact sequence with 'ASV_i' label 
head(taxa_ASV)

#Let's make a data frame which relates the 'ASV_' identifier to the corresponding exact sequence 
asv_df <- data.frame(taxa_ASV, taxa_sequences) 
nrow(asv_df)
ncol(asv_df)
class(asv_df)
str(asv_df) # We see that the columns are considered factors. Let's change them to characters since they are not really factors.
asv_df$taxa_ASV <- as.character(as.factor(asv_df$taxa_ASV))
asv_df$taxa_sequences <- as.character(as.factor(asv_df$taxa_sequences))
head(asv_df$taxa_ASV)
str(asv_df$taxa_sequences)
str(asv_df) # check to see that the variables are factors

# Now we can use the 'asv_df' to extract the exact sequences which corresponds to the ASVs identified as contaminants in 'contaminants'

# We'll also make a data frame with the 'ASV_id' and corresponding ASV sequence for each contaminant

# Retrieving contaminant info using 'getContamSeqs'
contam_info <- getContamSeqs(asv_df, contaminants_reseq_2)
contam_sequences <- contam_info[[1]] # gets contaminant ASV sequences vector from 'List'
head(contam_sequences) # Now the contaminant sequences are in 'contam_sequences'
str(contam_sequences)
contam_asv_df_reseq_2 <- contam_info[[2]] # gets the data frame with just the contaminant sequences
str(contam_asv_df_reseq_2)
dim(contam_asv_df_reseq_2)
names(contam_asv_df_reseq_2) <- c("ASV_id", "ASV_sequence") # rename data frame columns

# Let's filter the contaminating sequences out of the feature table
decontam_mp_reseq_2
alltaxa = taxa_names(decontam_mp_reseq_2)
goodtaxa = alltaxa[!(alltaxa %in% contam_sequences)]
seqtab_reseq_run_2_no_contam <- prune_taxa(goodtaxa, otu_table(decontam_mp_reseq_2))
dim(seqtab_reseq_run_2_no_contam)

# Let's check how many reads we lost as a result of removing the contaminants
sum(otu_table(seqtab_reseq_run_2_no_contam))/sum(seqtabAll2_reseq_run2) # about ~0.6% of reads were removed as contaminants
seqtab_reseq_run_2_no_contams <- as(otu_table(seqtab_reseq_run_2_no_contam), "matrix") # separating out 'otu table' from phyloseq object
dim(seqtab_reseq_run_2_no_contams)
class(seqtab_reseq_run_2_no_contams)

# Let's save the 'decontaminated' sequence table as an R serial object
saveRDS(seqtab_reseq_run_2_no_contams, "seqtab_reseq_run_2_no_contams.rds")

# Assigning taxonomy to the contaminants
alltaxa = taxa_names(decontam_mp_reseq_2)
goodtaxa = alltaxa[(alltaxa %in% contam_sequences)]
seqtab_reseq_run_2_contaminants <- prune_taxa(goodtaxa, otu_table(decontam_mp_reseq_2))
dim(seqtab_reseq_run_2_contaminants)

seqtab_reseq_run_2_contams <- as(otu_table(seqtab_reseq_run_2_contaminants), "matrix")
fastaRef <- "./silva_nr_v132_train_set.fa" 
taxTab_reseq_2_contams <- assignTaxonomy(seqtab_reseq_run_2_contams, refFasta = fastaRef, multithread = TRUE)
taxTab_reseq_2_contams <- addSpecies(taxTab_reseq_2_contams, "./silva_species_assignment_v132.fa")
dim(taxTab_reseq_2_contams)
View(taxTab_reseq_2_contams)
unname(head(taxTab_reseq_2_contams))

# output the taxonomy table as a .txt file
write.table(taxTab_reseq_2_contams, "contaminant_taxa_table_reseq_2.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
```

Now that we have determined the ASVs through the DADA2 pipeline for all the samples from all of our sequencing runs, we can merge the individual sequence tables to make a single 'main' sequence table. However, we are unable to perform this merging with the 'mergeSequenceTables' function, because this function won't work when the same sampleID is found in two or more sequence tables (which is our case since we have samples which have been resequenced). Therefore, this merging function will be done using 'QIIME2', since this pipeline includes commands which can merge sequence tables with overlapping sampleIDs (this is also possible to do in qiime1 using a different set of commands):

```{r, echo=TRUE}
# reading in the .rds files of the sequence tables
seqtab_runs_1_and_2 <- readRDS("seqtab_runs_1_and_2.rds")
dim(seqtab_runs_1_and_2)
seqtab_reseq_run_1_no_contams <- readRDS("seqtab_reseq_run_1_no_contams.rds")
dim(seqtab_reseq_run_1_no_contams)
seqtab_reseq_run_2_no_contams <- readRDS("seqtab_reseq_run_2_no_contams.rds")
dim(seqtab_reseq_run_2_no_contams)

# exporting out the sequence tables as .txt files
write.table(t(seqtab_runs_1_and_2), file = "seqtab_runs_1_and_2.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
write.table(t(seqtab_reseq_run_1_no_contams), file = "seqtab_reseq_run_1.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
write.table(t(seqtab_reseq_run_2_no_contams), file = "seqtab_reseq_run_2.txt", sep = "\t", row.names = TRUE, col.names = NA, quote = FALSE)
```

The remaining steps are carried out outside of R in qiime2 (Based on code available in https://forum.qiime2.org/t/importing-dada2-and-phyloseq-objects-to-qiime-2/4683)

```
# Open a terminal and change directory to the folder where the exported .txt files from the above steps are

# convert the .txt files to hdf5-format .biom tables:

echo -n "#OTU table" | cat - seqtab_runs_1_and_2.txt > seqtab-runs-1-and-2-biom-table.txt 

# Run in qiime2 environment
biom convert -i seqtab-runs-1-and-2-biom-table.txt -o seqtab-runs-1-and-2-biom-table.biom --table-type="OTU table" --to-hdf5

# can check to see if otu table is in the hdf5 format :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ file seqtab-runs-1-and-2-biom-table.biom
seqtab-runs-1-and-2-biom-table.biom: Hierarchical Data Format (version 5) data

echo -n "#OTU table" | cat - seqtab_reseq_run_1.txt > seqtab-reseq-run-1-biom-table.txt 
biom convert -i seqtab-reseq-run-1-biom-table.txt -o seqtab-reseq-run-1-biom-table.biom --table-type="OTU table" --to-hdf5

echo -n "#OTU table" | cat - seqtab_reseq_run_2.txt > seqtab-reseq-run-2-biom-table.txt 
biom convert -i seqtab-reseq-run-2-biom-table.txt -o seqtab-reseq-run-2-biom-table.biom --table-type="OTU table" --to-hdf5

# invoke qiime2 (e.g., source activate qiime2-2019.1) and import the .biom files into the qiime2 environment

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import \
> --input-path seqtab-runs-1-and-2-biom-table.biom \
> --type 'FeatureTable[Frequency]' \
> --input-format BIOMV210Format \
> --output-path seqtab-runs-1-and-2-biom-table.qza

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import \
> --input-path seqtab-reseq-run-1-biom-table.biom \
> --type 'FeatureTable[Frequency]' \
> --input-format BIOMV210Format \
> --output-path seqtab-reseq-run-1-biom-table.qza

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import \
> --input-path seqtab-reseq-run-2-biom-table.biom \
> --type 'FeatureTable[Frequency]' \
> --input-format BIOMV210Format \
> --output-path seqtab-reseq-run-2-biom-table.qza

# merge the three .qza files using 'qiime feature-table merge' :

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime feature-table merge \
> --i-tables seqtab-runs-1-and-2-biom-table.qza \
> --i-tables seqtab-reseq-run-1-biom-table.qza \
> --i-tables seqtab-reseq-run-2-biom-table.qza \
> --p-overlap-method sum \
> --o-merged-table merged-seqtabs-biom-table.qza

# exporting merged .qza table as a .biom file

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools export \
> --input-path merged-seqtabs-biom-table.qza \
> --output-path merged-qiime2-feature-table

# the merged sequence table file in .biom (hdf5 format) is in the 'merged-qiime2-feature-table' folder
# need to convert the 'hdf5' format .biom file to the 'json' format in order to use wih phyloseq in R :

biom convert -i merged-qiime2-feature-table/feature-table.biom -o merged-qiime2-feature-table/feature-table.txt --to-tsv
biom convert -i  merged-qiime2-feature-table/feature-table.txt -o  merged-qiime2-feature-table/merged-qiime2-feature-table-json-format.biom --table-type="OTU table" --to-json

# check format using file command 

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ file merged-qiime2-feature-table/merged-qiime2-feature-table-json-format.biom 
merged-qiime2-feature-table/merged-qiime2-feature-table-json-format.biom: ASCII text, with very long lines, with no line terminators
```

The 'merged-qiime2-feature-table-json-format.biom' can now be imported to R and used in phyloseq :

```{r, echo=TRUE}
merged_biomfile <- "merged-qiime2-feature-table/merged-qiime2-feature-table-json-format.biom" 
merged_biom = biomformat::read_biom(biom_file = merged_biomfile)
merged_qiime2_table <- import_biom(merged_biom) 
merged_seqtab_all_runs <- t(as(otu_table(merged_qiime2_table), "matrix")) # Need to transform it back so that the 'taxa' are columns and the 'samples' are rows since that's how phyloseq wants it
dim(merged_seqtab_all_runs)
```

Let's double check to see if the numbers add-up :

```{r, echo=TRUE}
sum(seqtab_runs_1_and_2)
sum(seqtab_reseq_run_1_no_contams)
sum(seqtab_reseq_run_2_no_contams)
```

The merged table should have a total of **21,440,384** reads. Let's see if this is the case :

```{r,echo=TRUE}
sum(merged_seqtab_all_runs)
```

The numbers add-up! Can continue onto chimera identification and filtering using this merged sequence table :

```{r, echo=TRUE}
# chimera filtering
typeof(merged_seqtab_all_runs) 
# comes out as 'Double' (not sure why this happens but may be something to do with converting and merging)
# need to convert to an 'Integer' matrix before chimera picking using 'removeBimeraDeNovo'
storage.mode(merged_seqtab_all_runs) <- "integer"
typeof(merged_seqtab_all_runs)

merged_seqtab_all_runs_nochim <- removeBimeraDenovo(merged_seqtab_all_runs, method = "consensus", multithread = TRUE, verbose = TRUE)
dim(merged_seqtab_all_runs_nochim) # 8773 chimeras were identified (which is a majority of ASVs but not abnormal apparently)
sum(merged_seqtab_all_runs_nochim)/sum(merged_seqtab_all_runs) # only about 5.5% of the total READS were removed as chimeras
```

Assigning taxonomy to the remaining ASVs using the SILVA database files :

```{r, echo=TRUE}
fastaRef <- "./silva_nr_v132_train_set.fa" 
taxTab <- assignTaxonomy(merged_seqtab_all_runs_nochim, refFasta = fastaRef, multithread = TRUE)
taxTab <- addSpecies(taxTab, "./silva_species_assignment_v132.fa")
unname(head(taxTab))
head(taxTab)
write.table(taxTab, "full_taxa_table.txt", sep = "\t", row.names = T, col.names = NA, quote = F)

# Let's output the taxonomy as a .txt file
taxa_cols <- colnames(taxTab)
taxa_table <- as.data.frame(taxTab)
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
dim(taxa_table)

taxa_table$ASV_seq <- rownames(taxa_table)
ASV_id <- paste0('ASV_', seq(rownames(taxa_table))) # replacing the long sequence names with "ASV_#" id
ASV_id
taxa_table$ASV_id <- ASV_id
dim(taxa_table)
View(taxa_table)
taxa_table <- taxa_table[,c(3,2,1)] # arrange table columns in order
View(taxa_table) # check to see that everything looks good
saveRDS(taxTab, "taxTab.rds")

write.table(taxa_table, "ASV_full_taxonomy_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

Next a phylogenetic tree needs to be built using the ASV sequences. Need the ASVs to be in FASTA format to do the alignments. This was done using the "ASV_full_taxonomy_table.txt" file and the python script 'make_repset_fasta_dada2.py'. 

##################################################################################################################################################
The FASTA format file may also be generated within R from the following code as shown in https://astrobiomike.github.io/amplicon/dada2_workflow_ex

asv_seqs <- colnames(merged_seqtab_all_runs_nochim)
asv_headers <- vector(dim(merged_seqtab_all_runs_nochim)[2], mode="character")

for (i in 1:dim(merged_seqtab_all_runs_nochim)[2]) {
  asv_headers[i] <- paste(">ASV", i, sep-"_")
}

asv_fasta <- c(rb"ind(asv_headers, asv_seqs))
write(asv_fasta, "ASV.fa")
##################################################################################################################################################

This FASTA file is used for alignment to the 'Silva_nr' database and subsequent construction of a phylogenetic tree using MOTHUR as follows :

```
# Open a terminal and type the following commands to download and unzip the Silva_nr database :
wget http://www.mothur.org/w/images/3/32/Silva.nr_v132.tgz
tar -zxvf Silva.nr_v132.tgz 

# move the unzipped 'silva.nr_v132.align' database and the FASTA formatted file from running the 'make_repset_fasta_dada2.py' script to the MOTHUR folder and run the following MOTHUR commands :

pcr.seqs(fasta=silva.nr_v132.align, start=11894, end=25319, keepdots=F, processors=8)
system(mv silva.nr_v132.pcr.align silva.v4.fasta)
align.seqs(fasta=ASV_rep_set.fasta, reference=silva.v4.fasta)

# The 'ASV_rep_set.align' file needs to be modified to make it compatible with MOTHUR for running the commands for distance calculations and making the phylogenetic tree using 'clearcut'. The FASTA headers need to have at least 10 characters and the '.' in the alignment file need to be replaced with '-' . Need to open a terminal and type these commands (outside of MOTHUR).

sed -i -e 's/>/>AAAAAAAAAA/g' ASV_rep_set.align # makes the fasta headers at least 10 characters
sed -i -e 's/\./-/g' ASV_rep_set.align # replaces '.' with '-'

# making phylogenetic tree using 'clearcut' in MOTHUR (takes a while so I ran these commands on a server):
dist.seqs(fasta=ASV_rep_set.align, processors=2, cutoff=.10, output=phylip)
clearcut(phylip=ASV_rep_set.phylip.dist)

# The 10 A's now need to be removed from the headers of the resulting phylogenetic tree :
sed -i -e 's/AAAAAAAAAA//g' ASV_rep_set.phylip.tre
```

Let's read in the sample data file (mapping file) which contains all the metadata and the newly constructed phylogenetic tree:

```{r, echo=TRUE}
mapping_file <- read.table("pig_mouse_comp_r2_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(mapping_file) <- as.character(mapping_file[,1])
View(mapping_file)
tree_file <- 'ASV_rep_set.phylip.tre'
phylo_tree <- read_tree(tree_file)
phylo_tree
```

Now we can combine the merged sequence table, mapping file, taxa table, and phylogenetic tree into a single pyloseq object :

```{r, echo=TRUE}
merged_pig_mouse_comp_ps <- phyloseq(otu_table(merged_seqtab_all_runs_nochim, taxa_are_rows = FALSE), sample_data(mapping_file), tax_table(taxTab))
merged_pig_mouse_comp_ps
taxa_names(merged_pig_mouse_comp_ps) <- paste0("ASV_", seq(ntaxa(merged_pig_mouse_comp_ps))) # replacing the long sequence name with short ID 
merged_pig_mouse_comp_ps <-merge_phyloseq(merged_pig_mouse_comp_ps, phylo_tree)
merged_pig_mouse_comp_ps

saveRDS(merged_pig_mouse_comp_ps, "merged_pig_mouse_comp_ps.rds")
#merged_pig_mouse_comp_ps <- readRDS("merged_pig_mouse_comp_ps.rds")
#merged_pig_mouse_comp_ps
```

Let's look at the prevalence of taxa in the samples 
 
```{r, echo=TRUE}
prevdf = apply(X = otu_table(merged_pig_mouse_comp_ps),
               MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf = data.frame(Prevalence=prevdf, TotalAbundance=taxa_sums(merged_pig_mouse_comp_ps))
View(prevdf) # minimum prevalence is '1'
```

Removing 'negative control' samples :

```{r, echo=TRUE}
merged_pig_mouse_comp_ps_no_neg <- subset_samples(merged_pig_mouse_comp_ps, Description != 'negative_control')
merged_pig_mouse_comp_ps_no_neg

prevdf_no_neg = apply(X = otu_table(merged_pig_mouse_comp_ps_no_neg),
               MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps_no_neg), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_no_neg = data.frame(Prevalence=prevdf_no_neg, TotalAbundance=taxa_sums(merged_pig_mouse_comp_ps_no_neg))
View(prevdf_no_neg)

# There appear to be quite a few ASVs (around 350) which show a prevalence of '0' after filtering out the negative controls
# this seems to suggest that these ASVs would have been present ONLY in negative controls, and are no longer present after they are filtered out
# Let's see what these ASVs are and assign them taxonomy in case it is of interest later on

negative_control_taxa <- rownames(prevdf_no_neg)[(prevdf_no_neg$Prevalence < 1)]
negative_control_taxa # 348 ASVs
negative_control_taxa_table <- prune_taxa(negative_control_taxa, tax_table(merged_pig_mouse_comp_ps_no_neg))
negative_control_taxa_df <- as.data.frame(negative_control_taxa_table)
View(negative_control_taxa_df)
write.table(negative_control_taxa_df, 'Taxa_found_only_in_NEG_samples.txt', sep = '\t', col.names = NA, row.names = TRUE, quote = FALSE)
```

Filtering out ASVs that are classified in the kingdoms 'Archaea' or 'Eukaryota' as well as those ASVs that have a Mitochondrial classification :

```{r, echo=TRUE}
remov_kingdoms <- c('Archaea', 'Eukaryota')
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_pig_mouse_comp_ps_no_neg, !Kingdom %in% remov_kingdoms & (Family != 'Mitochondria' | is.na(Family))) 
merged_pig_mouse_comp_ps_master
merged_pig_mouse_comp_ps_master <- subset_taxa(merged_pig_mouse_comp_ps_master, Kingdom == 'Bacteria') # removed 4 ASVs classified as 'NA' at the kingdom level
merged_pig_mouse_comp_ps_master
```

In the past, we've filtered out any OTU which had a Cyanobacterial classification. However, recent research (Di Rienzi et al., 2013) has suggested that a non-photosynthetic group of 'Cyanobacteria' called 'Melainabacteria' might be human gut-associated. Therefore, we're going to filter out only those Cyanobacterial ASVs that are not classified as belonging to the class 'Melainabacteria':

```{r, echo=TRUE}
Cyano_ps <- subset_taxa(merged_pig_mouse_comp_ps_master, Phylum == 'Cyanobacteria') # Isolating the Cyanobacterial ASVs
remov_Cyano_ps <- subset_taxa(Cyano_ps, Class != 'Melainabacteria') # identifying Cyanobacterial ASVs that are not 'Melainabacteria'
Cyano_to_remov <- taxa_names(remov_Cyano_ps)
Cyano_to_remov <- append(Cyano_to_remov, 'ASV_2842') # ASV_2842 has 'NA' as Class so we'll remove that one as well

# Filtering out the non-Melainabacterial Cyanobacterial ASVs
alltaxa = taxa_names(merged_pig_mouse_comp_ps_master)
goodtaxa = alltaxa[!(alltaxa %in% Cyano_to_remov)]
merged_pig_mouse_comp_ps_master_no_Cyano <- prune_taxa(goodtaxa, merged_pig_mouse_comp_ps_master)
merged_pig_mouse_comp_ps_master_no_Cyano
sum(otu_table(merged_pig_mouse_comp_ps_master_no_Cyano))

# Writing taxonomy table after filtering unwanted taxa
write.table(tax_table(merged_pig_mouse_comp_ps_master_no_Cyano), "no_cyano_taxa_table.txt", sep = "\t", col.names = NA, row.names = T, quote = F)
```

Computing prevalence of each ASV to filter out those that are present in only a single sample :

```{r, echo=TRUE}
# Calculating prevalence for each ASV
prevdf <- apply(X = otu_table(merged_pig_mouse_comp_ps_master_no_Cyano), MARGIN = ifelse(taxa_are_rows(merged_pig_mouse_comp_ps_master_no_Cyano), yes = 1, no = 2), FUN = function(x){sum(x > 0)})
View(prevdf)
prevdf_with_taxa <- data.frame(Prevalence = prevdf, TotalAbundance = taxa_sums(merged_pig_mouse_comp_ps_master_no_Cyano), tax_table(merged_pig_mouse_comp_ps_master_no_Cyano))
View(prevdf_with_taxa)

# Retaining only those ASVs found in more than one sample
keepTaxa <- rownames(prevdf_with_taxa)[(prevdf_with_taxa$Prevalence > 1)]
merged_pig_mouse_comp_ps_master_all_filtered <- prune_taxa(keepTaxa, merged_pig_mouse_comp_ps_master_no_Cyano)
merged_pig_mouse_comp_ps_master_all_filtered # We've filtered out a whopping 5064 ASVs

sum(otu_table(merged_pig_mouse_comp_ps_master_no_Cyano))
sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))
sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))/sum(otu_table(merged_pig_mouse_comp_ps_master_no_Cyano)) # However, those 5066 ASVs only accounted for ~0.3% of reads (41,238)
```

Looking at the distribution of reads per sample :

```{r, echo=TRUE}
reads_per_sample_df <- data.frame(Reads = sample_sums(merged_pig_mouse_comp_ps_master_all_filtered)) 
reads_per_sample_df$sampleID <- rownames(reads_per_sample_df)
View(reads_per_sample_df)
reads_per_sample_df <- reads_per_sample_df[,c(2,1)] # reorder columns
reads_per_sample_df_sorted <- reads_per_sample_df[order(reads_per_sample_df$Reads),] # order samples by read count (ascending)
View(reads_per_sample_df_sorted) # check that it looks right
write.table(reads_per_sample_df_sorted, 'Reads_per_sample_after_all_the_filtering_steps.txt', sep = "\t", col.names = T, row.names = F, quote = F)

# Making a histogram plot with sequence depth information
library('ggplot2'); packageVersion('ggplot2')
read_depth_histo <- ggplot(reads_per_sample_df_sorted, aes(x = Reads)) + geom_histogram(color = "black", fill = "indianred", binwidth = 2500) + ggtitle("Sequence depth distribution across samples") + xlab("Number of reads") + theme(axis.title.y = element_blank())
read_depth_histo

# Depth information
min_read_depth <- min(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered))
min_read_depth # 5,415 reads
max_read_depth <- max(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered))
max_read_depth # 142,321 reads
avg_read_depth <- sum(otu_table(merged_pig_mouse_comp_ps_master_all_filtered))/nsamples(merged_pig_mouse_comp_ps_master_all_filtered)
avg_read_depth # 43,528 reads/sample
```

Filtering out samples which have < 10,500 total reads :

```{r, echo=TRUE}
merged_pig_mouse_comp_complete_ps <- prune_samples(sample_sums(merged_pig_mouse_comp_ps_master_all_filtered) > 10500, merged_pig_mouse_comp_ps_master_all_filtered)
merged_pig_mouse_comp_complete_ps # 459 samples remain

# Save as an R serial object for use in future analyses
saveRDS(merged_pig_mouse_comp_complete_ps, "merged_pig_mouse_comp_complete_ps.rds")
merged_pig_mouse_comp_complete_ps <- readRDS("merged_pig_mouse_comp_complete_ps.rds")
merged_pig_mouse_comp_complete_ps
sum(otu_table(merged_pig_mouse_comp_complete_ps))
write.table (sample_data(merged_pig_mouse_comp_complete_ps), "after_filtering_mapping_file.txt", sep = "\t", row.names = F, col.names = T, quote = F)
```

We will also remove fecal samples collected at euthanization since these samples were collected differently from the others samples

```{r, echo = TRUE}
merged_pig_mouse_comp_complete_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Time_point != "at_euthanization" )
merged_pig_mouse_comp_complete_ps # 392 samples remain
```

Looking at alpha rarefactions and sample diversity in the human donor samples :

```{r, echo=TRUE} 
merged_pig_mouse_comp_human_donors_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Species == 'human')
merged_pig_mouse_comp_human_donors_ps

# making rarefaction curve for donore samples
library("vegan") ; packageVersion("vegan")
rarecurve(otu_table(merged_pig_mouse_comp_human_donors_ps), step = 50, cex = 0.5)

# alpha diversity comparison of donor samples (Shannon index)
# Since it is recommended to calculate alpha diversity before filtering out any taxa, we'll use the sequence table with just the negative control samples removed as the starting point for this analysis

# Let's also make these plots without rarefying since phyloseq authors recommend not rarefying because the alpha-diversity indices are sensitive to that
human_donor_alph_div <- plot_richness(merged_pig_mouse_comp_human_donors_ps, x = "Donor", measures = "Shannon")
human_donor_alph_div


# Estimating alpha diversity using Shannon index only for inoculae actually used
human_donor_inoculae_ps <- subset_samples(merged_pig_mouse_comp_human_donors_ps, Actual_inoculum == "yes")
human_donor_inoculae_ps <- subset_samples(human_donor_inoculae_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
human_donor_inoculae_ps

write.table(sample_data(human_donor_inoculae_ps), "donor sample data.txt", sep = "\t", col.names = T, row.names = F, quote = F)

results_donors = estimate_richness(human_donor_inoculae_ps, measures = "Shannon")
results_donors_df <- as.data.frame(results_donors)
results_donors_df$SampleID <- rownames(results_donors_df)
results_donors_df <- results_donors_df[c(2,1)]
View(results_donors_df)

# import mapping file which would be used for alpha diversity analyses
alpha_mapping_file <- read.table("alpha_div_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(alpha_mapping_file) <- as.character(alpha_mapping_file[,1])

donors_alpha_div_df <- merge(results_donors_df, alpha_mapping_file, by='SampleID')
View(donors_alpha_div_df)

p1 <- ggplot(donors_alpha_div_df, aes(x=Donor, y=Shannon, fill=Donor)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits = c("INF2", "CHLD2", "ADLT2", "ELD2"), labels = c("Infant", "Child", "Adult", "Elderly")) + theme(legend.position = "none") + labs(x = "Donor", y = "Shannon index")
p2
                            
# Looking at statistical significance between different donor inocula
d = sample_data(human_donor_inoculae_ps)

# calculate stats using Wilcoxon test
INF = results_donors[d[,'Donor'] == 'INF2',]
CHLD = results_donors[d[,'Donor'] == 'CHLD2',]
ADLT = results_donors[d[,'Donor'] == 'ADLT2',]
ELD = results_donors[d[,'Donor'] == 'ELD2',]
wilcox.test(INF, CHLD)
wilcox.test(INF, ADLT)
wilcox.test(INF, ELD)
wilcox.test(CHLD, ADLT)
wilcox.test(CHLD, ELD)
wilcox.test(ADLT, ELD)

## Plotting rarefaction curves and alpha diversity considering only those samples used as inoculae

# separating out the human donor inoculae
human_donor_inoculae_ps <- subset_samples(merged_pig_mouse_comp_human_donors_ps, Actual_inoculum == 'yes')
human_donor_inoculae_ps

# plotting rarefaction curve
rarecurve(otu_table(human_donor_inoculae_ps), step = 50, cex = 0.5)

# plotting alpha diversity
human_donor_inoculae_alph_div <- plot_richness(human_donor_inoculae_ps, x = "Donor", measures = "Shannon")
human_donor_inoculae_alph_div
```

Beta diversity analysis for human donors : ## Delete this part if not necessary for manuscript

```{r, echo=TRUE}

# Let's make these plots for those samples which were actually used as inoculae

# Keeping only the samples used as inoculae
human_donor_inoculae_ps <- subset_samples(merged_pig_mouse_comp_human_donors_ps, Actual_inoculum == 'yes')
human_donor_inoculae_ps

# Converting count table to relative abundances in order to account for differences in read depth
human_donor_inoculae_ps.prop <- transform_sample_counts(human_donor_inoculae_ps, function(x){x / sum(x)}) 

# plotting human donor inoculae samples using unweighted unifrac distances
unw_ord <- ordinate(human_donor_inoculae_ps.prop, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_inoculae <- plot_ordination(human_donor_inoculae_ps.prop, unw_ord, color = "Donor") 
unweighted_pcoa_inoculae

# plotting human donor inoculae samples using weighted unifrac distances
w_ord <- ordinate(human_donor_inoculae_ps.prop, method = "PCoA", distance = "wunifrac")
weighted_pcoa_inoculae <- plot_ordination(human_donor_inoculae_ps.prop, w_ord, color = "Donor") 
weighted_pcoa_inoculae
```

## **Donor-based Analysis** ##

Let's look at each donor separately and compare the humanized pig and mouse microbiotas to the original human donor microbiotas.

**INFANT Donor Analysis**

```{r, echo=TRUE}
## MAKING RAREFACTION PLOTS
merged_pig_mouse_comp_ps_no_neg_no_cecal <- subset_samples(merged_pig_mouse_comp_ps_no_neg, SampleType != 'cecal')
merged_pig_mouse_comp_INF_ps <- subset_samples(merged_pig_mouse_comp_ps_no_neg_no_cecal, Donor=="INF2")
merged_pig_mouse_comp_INF_ps <- subset_samples(merged_pig_mouse_comp_INF_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
merged_pig_mouse_comp_INF_ps <- subset_samples(merged_pig_mouse_comp_INF_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
merged_pig_mouse_comp_INF_ps <- subset_samples(merged_pig_mouse_comp_INF_ps, Time_point != 'at_euthanization')
merged_pig_mouse_comp_INF_ps <- subset_samples(merged_pig_mouse_comp_INF_ps, !SampleID %in% early_inoculation_donor_samples)
merged_pig_mouse_comp_INF_ps

## export as .txt files
write.table(t(otu_table(merged_pig_mouse_comp_INF_ps)), "INF2_otu_table.txt", sep = "\t", row.names = TRUE, col.names=NA, quote = FALSE)
write.table(sample_data(merged_pig_mouse_comp_INF_ps), "INF2_sample_data.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE) 
```

```
## Using QIIME2 to make rarefaction curves

# Export feature table and converting it into a biom file :

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ echo -n "#OTU Table" | cat - INF2_otu_table.txt > INF2_otu_biom_table.txt
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ biom convert -i INF2_otu_biom_table.txt -o INF2_otu_biom_table.biom --table-type="OTU table" --to-hdf5

# importing biom table to qiime2 :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import --input-path INF2_otu_biom_table.biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path INF2_otu_biom_table.biom.qza

# Use feature table summary to find out what the medium read depth is (for use in --p-max-depth of following command) :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime feature-table summarize --i-table INF2_otu_biom_table.biom.qza --o-visualization INF_table.qzv --m-sample-metadata-file INF2_sample_data.txt

# Performing alpha rarefactions :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime diversity alpha-rarefaction \
> --i-table INF2_otu_biom_table.biom.qza \
> --i-phylogeny ASV_rep_set.phylip.qza \
> --p-max-depth 34000 \
> --m-metadata-file INF2_sample_data.txt \
> --o-visualization INF2-alpha-rarefaction.qzv
```

For the infant microbiota, there were two groups of pigs (3 pigs/group): one group (early inoculations) were inoculated while they were on a milk diet, while the other group (late inoculations) were inoculated after they had been weaned and were on a solid diet. For this study we'll only concentrate on the late-inoculation piglets. Let's go ahead and compare INF2 donor, HMA mice, and the late-inoculated piglet fecal bacterial communities :

```{r, echo=TRUE}
pig_mouse_comp_INF2_master_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Donor == 'INF2')
pig_mouse_comp_INF2_master_ps

# Separating out the donor inocula which were used to inoculate the mice and piglets 
human_donors_INF2_ps <- subset_samples(pig_mouse_comp_INF2_master_ps, Species == 'human')
human_donors_INF2_ps
write.table(sample_data(human_donors_INF2_ps), "IND2_donors.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
human_donors_INF2_inocula_ps <- subset_samples(human_donors_INF2_ps, Actual_inoculum == 'yes')
human_donors_INF2_inocula_ps # contains only the samples used as inocula

# Making a box plot at the 'Family' level
human_donors_INF2_inocula_ps_trans <- transform_sample_counts(human_donors_INF2_inocula_ps, function(x) x/sum(x))
theme_set(theme_bw())
p1 <- plot_bar(human_donors_INF2_inocula_ps_trans, fill = "Family")
p2 <- p1 + theme (legend.text = element_text(size = 6), legend.title = element_text(size = 8))
p3 <- p2 + theme(legend.key.height = unit(0.5, "cm"), legend.key.width = unit(0.2, "cm"), legend.key.size = unit(0.5, "cm"))
p3
```

Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the human donor inocula communities. We'll compare only the 'late inoculation' piglets to the mice and human donor samples:

```{r, echo=TRUE}
pig_mouse_comp_INF2_comparison_ps <- subset_samples (pig_mouse_comp_INF2_master_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_INF2_comparison_ps
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Timingofinocu != 'early' | is.na(Timingofinocu)) # only 'late inoculation' piglets are considered
pig_mouse_comp_INF2_comparison_ps
early_inoculation_donor_samples <- c('299', '300')
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, !SampleID %in% early_inoculation_donor_samples) # keeping only the donor inocula used to inoculate the mice and 'late' piglets
pig_mouse_comp_INF2_comparison_ps

# Let's also remove cecal samples 
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, SampleType != 'cecal')
pig_mouse_comp_INF2_comparison_ps

## Now let's visualize beta diversity in the form of PCoA plots

# Transform counts to relative abundances
rel_abund_INF2_ps <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x){x / sum(x)})

# PCoA plotting using unweighted unifrac distances
unw_ord <- ordinate(rel_abund_INF2_ps, method = "PCoA", distance = "unifrac", weighted = F)
unwunifrac_INF2 <- plot_ordination(rel_abund_INF2_ps, unw_ord, color = "Species") + ggtitle("Infant donor PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_INF2

# PCoA plotting using weighted unifrac distances
w_ord <- ordinate(rel_abund_INF2_ps, method = "PCoA", distance = "wunifrac")
wunifrac_INF2 <- plot_ordination(rel_abund_INF2_ps, w_ord, color = "Species") + ggtitle("Infant donor PCoA plot - Weighted unifrac") + theme(plot.title = element_text(hjust = 0.5))
wunifrac_INF2

# Combining the two unifrac plots into a single graphic
INF2_unifrac_comb_plot <- grid.arrange(ncol = 2, unwunifrac_INF2, wunifrac_INF2)
INF2_unifrac_comb_plot

# plotting using binary jaccard distances
bin_jac_ord <- ordinate(rel_abund_INF2_ps, method = "PCoA", distance = "jaccard", binary = T)
binary_jaccard_pcoa_INF2 <- plot_ordination(rel_abund_INF2_ps, bin_jac_ord, color = "Species") + ggtitle("Infant donor PCoA plot - binary Jaccard") + theme(plot.title = element_text(hjust = 0.5))
binary_jaccard_pcoa_INF2

# plotting using Bray-Curtis distances
bray_ord_INF2 <- ordinate(rel_abund_INF2_ps, method="PCoA", distance = "bray")
bray_pcoa_INF2 <- plot_ordination(rel_abund_INF2_ps, bray_ord_INF2, color="Species") + ggtitle("Infant donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5))
bray_pcoa_INF2

## Let's also color the weighted and unweighted unifrac plots by sampling time point
unwunifrac_INF2_by_tp <- plot_ordination(rel_abund_INF2_ps, unw_ord, color = "Time_point", shape = "Species") + ggtitle("Infant donor PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_INF2_by_tp

wunifrac_INF2_by_tp <- plot_ordination(rel_abund_INF2_ps, w_ord, color = "Time_point", shape = "Species") + ggtitle("Infant donor PCoA plot - Weighted unifrac") + theme(plot.title = element_text(hjust = 0.5))
wunifrac_INF2_by_tp

### performing PERMANOVA analyses
rel_abund_INF2_ps
metadata <- as(sample_data(rel_abund_INF2_ps), "data.frame")
View(sample_data(rel_abund_INF2_ps))
adonis(phyloseq::distance(rel_abund_INF2_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_INF2_ps), "data.frame"), permutations = 999)

## defining a function for performing pair-wise permanovas
pairwise_permanova <- function(physeq,treatment1,treatment2,method) {
    metadata <- as(sample_data(physeq), "data.frame")
    cbn <- combn(x = levels(metadata[[treatment1]]), m = 2)
    ncol(cbn)
    print(1:ncol(cbn))
    results <- list()
    for (i in 1:ncol(cbn)) {
       print(i)
       print(cbn[,i])
       cbn1 <- cbn[,i]
       ids1 = as.character(get_variable(physeq, treatment1)) == cbn1[1] #& as.character(get_variable(physeq, "Species")) == cbn1[2]  
       ids2 = as.character(get_variable(physeq, treatment1)) == cbn1[2] 
       ps1 <- prune_samples(ids1, physeq)
       ps2 <- prune_samples(ids2, physeq)
       ps <- merge_phyloseq(ps1,ps2)
       #print(nsamples(ps))
       metadata_new <- as(sample_data(ps), "data.frame")
       result <- adonis(phyloseq::distance(ps, method = method, weighted = F) ~ metadata_new[[treatment1]] + metadata_new[[treatment2]]) #Species, data = metadata_new
       results[[i]] <- result
    }
    return(results)
 }
 
## performing pair-wise permanovas for INF2
 results_INF2_unweighted <- pairwise_permanova(rel_abund_INF2_ps,"Species","Time_point","unifrac") 
 print(results_INF2_unweighted)
```

Let's also plot comparing both the early and late inoculation piglets with mice (## remove if unnecessary) :

```{r, echo=TRUE}
write.table (sample_data(pig_mouse_comp_INF2_comparison_ps), 'INF2_samples_mapping_file.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

pig_mouse_comp_INF2_comparison_ps_2 <- subset_samples (pig_mouse_comp_INF2_master_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
pig_mouse_comp_INF2_comparison_ps_2 <- subset_samples(pig_mouse_comp_INF2_comparison_ps_2, SampleType != 'cecal')
pig_mouse_comp_INF2_comparison_ps_2

write.table (sample_data(pig_mouse_comp_INF2_comparison_ps_2), 'INF2_samples_mapping_file.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
# Reading in a modified mapping file which includes a 'Classification' column :
INF2_mapping_file <- read.table("INF2_samples_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(INF2_mapping_file) <- as.character(INF2_mapping_file[,1])
View(INF2_mapping_file)

# making a new 'phyloseq object' incorporating INF2 mapping file :
INF2_ps <- phyloseq(otu_table(pig_mouse_comp_INF2_comparison_ps_2, taxa_are_rows = FALSE), sample_data(INF2_mapping_file), phy_tree(phylo_tree))
INF2_ps

# Rarefying to an even depth
INF2_ps.rarefied <- rarefy_even_depth(INF2_ps, rngseed = 1, sample.size = min(sample_sums(INF2_ps)), replace = F) # rarefied at 15658 reads/sample

# plotting PCoA using unweighted unifrac distances 
unweighted_unifrac_dm_INF2 <- phyloseq::distance(INF2_ps.rarefied, method = "unifrac", weighted = F)
unw_ord <- ordinate(INF2_ps.rarefied, method = "PCoA", distance = unweighted_unifrac_dm_INF2)
unweighted_pcoa_INF2_early_and_late <- plot_ordination(INF2_ps.rarefied, unw_ord, color = "Species", shape = "Timingofinocu") 
unweighted_pcoa_INF2_early_and_late 

# plotting PCoA using weighted unifrac distances 
weighted_unifrac_dm_INF2 <- phyloseq::distance(INF2_ps.rarefied, method = "unifrac", weighted = T)
w_ord <- ordinate(INF2_ps.rarefied, method = "PCoA", distance = weighted_unifrac_dm_INF2)
weighted_pcoa_INF2_early_and_late <- plot_ordination(INF2_ps.rarefied, w_ord, color = "Species", shape = "Timingofinocu") 
weighted_pcoa_INF2_early_and_late
```

Let's see which ASVs were present in both the inocula used to inoculate the mice as well as the inoculum used to inoculate the 'late' piglets and thus define a 'core' INF2 microbiota. We can subsequently compare how these core ASVs are distributed between human donor and the humanized animal models :

```{r, echo=TRUE}
# Let's define a new core only considering the taxa found in the three donor samples which went to the mice and the HMA late inoculation piglets
human_donors_INF2_inocula_relevant_ps <- subset_samples(human_donors_INF2_inocula_ps, SampleID != c('299', '300')) # removing 'early' piglet inoculae
human_donors_INF2_inocula_relevant_ps
prevdf_INF2_donor = apply(X = otu_table(human_donors_INF2_inocula_relevant_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_INF2_inocula_relevant_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_INF2_donor = data.frame(Prevalence=prevdf_INF2_donor, TotalAbundance=taxa_sums(human_donors_INF2_inocula_relevant_ps))
View(prevdf_INF2_donor)

core_threshold = nsamples(human_donors_INF2_inocula_relevant_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_INF2_taxa <- rownames(prevdf_INF2_donor)[prevdf_INF2_donor$Prevalence >= core_threshold]
core_INF2_taxa # We have 26 ASVs which were detected in all 3 inocula

# Let's make a box plot at the 'Genus' level only considering these 26 ASVs among the 3 donor samples
human_donors_INF2_inocula_relevant_ps_trans <- transform_sample_counts(human_donors_INF2_inocula_relevant_ps, function(x) x/sum(x))
human_donors_INF2_inocula_relevant_core_ps_trans <- prune_taxa(core_INF2_taxa, human_donors_INF2_inocula_relevant_ps_trans)
human_donors_INF2_inocula_relevant_core_ps_trans

theme_set(theme_bw())
p1 <- plot_bar(human_donors_INF2_inocula_relevant_core_ps_trans, fill = "Genus")
p2 <- p1 + theme(legend.text = element_text(size = 6), legend.title = element_text(size = 8))
p3 <- p2 + theme(legend.key.height = unit(0.5, "cm"), legend.key.width = unit(0.2, "cm"), legend.key.size = unit(0.5, "cm"))
p3

# Calculating the number of reads accounted for by the core ASVs 
human_donors_INF2_inocula_relevant_core_ps <- prune_taxa(core_INF2_taxa, human_donors_INF2_inocula_relevant_ps)
human_donors_INF2_inocula_relevant_core_ps

sum(otu_table(human_donors_INF2_inocula_relevant_core_ps))/sum(otu_table(human_donors_INF2_inocula_relevant_ps))

# Calculating how much of each donor inocula group was represented by these core ASVs
# for mouse inoculae
INF2_mouse_inocula_ps <- subset_samples(human_donors_INF2_inocula_relevant_ps, SampleID != '323') # removes piglet inoculum
INF2_mouse_inocula_ps
INF2_mouse_inocula_core_ps <- prune_taxa(core_INF2_taxa,INF2_mouse_inocula_ps)
INF2_mouse_inocula_core_ps
sum(otu_table(INF2_mouse_inocula_core_ps))/sum(otu_table(INF2_mouse_inocula_ps)) # the 26 core ASVs represents 99% of these donor sequences

# for piglet inoculum
INF2_late_piglet_inoculum_ps <- subset_samples(human_donors_INF2_inocula_relevant_ps, SampleID == '323') # removes piglet inoculum
INF2_late_piglet_inoculum_ps
INF2_late_piglet_inoculum_core_ps <- prune_taxa(core_INF2_taxa,INF2_late_piglet_inoculum_ps)
INF2_late_piglet_inoculum_core_ps
sum(otu_table(INF2_late_piglet_inoculum_core_ps))/sum(otu_table(INF2_late_piglet_inoculum_ps)) # the 26 core ASVs represents 98.4% of these donor sequences
```

Now let's see how these core INF2 donor ASVs are established in the two HMA animal models by looking at beta diversity :

```{r, echo=TRUE}
rel_abund_INF2_ps
rel_abund_INF2_core_ps <- prune_taxa(core_INF2_taxa, rel_abund_INF2_ps)
rel_abund_INF2_core_ps

# plotting PCoA using unweighted unifrac distances for core taxa - INF2
unw_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, unw_ord_INF2_core, color = "Species", shape = "Species") + ggtitle("INF donor PCoA plot - Unweighted Unifrac") + theme(plot.title = element_text(hjust = 0.5, size = 10))
unweighted_pcoa_INF2_core

# plotting PCoA using weighted unifrac distances for core taxa - INF2
w_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, w_ord_INF2_core, color = "Species") 
weighted_pcoa_INF2_core

INF2_core_unifrac_comb <- grid.arrange(ncol = 2, unweighted_pcoa_INF2_core, weighted_pcoa_INF2_core)
INF2_core_unifrac_comb

# plotting PCoA using Jaccard distances for core taxa - INF2
bin_jac_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "jaccard", binary = T)
jaccard_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, bin_jac_ord_INF2_core, color = "Species") + ggtitle("INF donor PCoA plot - Binary Jaccard") + theme(plot.title = element_text(hjust = 0.5, size = 10))
jaccard_pcoa_INF2_core

# plotting PCoA using Bray-Curtis distances for core taxa - INF2
bray_ord_INF2_core <- ordinate(rel_abund_INF2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_INF2_core <- plot_ordination(rel_abund_INF2_core_ps, bray_ord_INF2_core, color = "Species", shape = "Species") + ggtitle("INF donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 10))
bray_pcoa_INF2_core

INF2_core_unifrac_comb <- grid.arrange(ncol = 2, unweighted_pcoa_INF2_core, weighted_pcoa_INF2_core)
INF2_core_unifrac_comb

INF2_core_jaccard_bray_comb <- grid.arrange(ncol = 2, jaccard_pcoa_INF2_core, bray_pcoa_INF2_core)
INF2_core_jaccard_bray_comb

INF2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_INF2_core, bray_pcoa_INF2_core)
INF2_core_unifrac_bray_comb

## Let's also color the unweighted unifrac and bray-curtis plots by sampling time point
unwunifrac_INF2_core_by_tp <- plot_ordination(rel_abund_INF2_core_ps, unw_ord_INF2_core, color = "Time_point", shape = "Species") + ggtitle("Infant donor PCoA plot for core ASVs - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_INF2_core_by_tp

bray_INF2_by_tp <- plot_ordination(rel_abund_INF2_core_ps, bray_ord_INF2_core, color = "Time_point", shape = "Species") + ggtitle("Infant donor PCoA plot core ASVs - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 10))
bray_INF2_by_tp

INF2_core_PCoA_by_tp_comb <- grid.arrange(ncol = 2, unwunifrac_INF2_core_by_tp, bray_INF2_by_tp)
INF2_core_PCoA_by_tp_comb
```

PERMANOVA analysis to check if clustering patterns are statistically siginificantly different

```{r, echo=TRUE}
### performing PERMANOVA analyses on unweighted unifrac
rel_abund_INF2_core_ps
metadata <- as(sample_data(rel_abund_INF2_core_ps), "data.frame")
View(sample_data(rel_abund_INF2_core_ps))
adonis(phyloseq::distance(rel_abund_INF2_core_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_INF2_core_ps), "data.frame"), permutations = 999)

## Carrying out pair-wise comparisons
results_INF2_unweighted <- pairwise_permanova(rel_abund_INF2_core_ps,"Species","Time_point","unifrac") 
print(results_INF2_unweighted)
```

Hierarchical clustering based on 'Bray-Curtis' distances

```{r, echo= TRUE}

## INF2 Donor

# Use the relative abundance table
rel_abund_INF2_core_ps
bray_dist_INF2 <- phyloseq::distance(rel_abund_INF2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_INF2 <- hclust(bray_dist_INF2, method = "ward.D2")
plot(bray_clust_INF2)

# need to color plot by 'Species' to see clustering patterns 
install.packages("dendextend"); packageVersion("dendextend") 
library(dendextend)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_INF2 <- read.table("mapping_files_for_dendrograms/INF2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_INF2)
mapping_file_dendro_INF2$Color_Sp <- as.character(mapping_file_dendro_INF2$Color_Sp)

# making dendrogram
bray_dend_INF2 <- as.dendrogram(bray_clust_INF2, hang=0.1)
SampleIDs_dend_INF2 <- labels(bray_dend_INF2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_INF2

sorted_mapping_INF2 <- mapping_file_dendro_INF2[match(SampleIDs_dend_INF2, mapping_file_dendro_INF2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_INF2)

# color dendrogram labels 
labels_colors(bray_dend_INF2) <- sorted_mapping_INF2$Color_Sp
labels_colors(bray_dend_INF2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_INF2) <- sorted_mapping_INF2$Time_point
bray_dend_INF2 <- set(bray_dend_INF2, "labels_cex", 0.5)
labels(bray_dend_INF2)
plot(bray_dend_INF2, ylab="Bray-Curtis distances")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
install.packages("gplots")
library("gplots"); packageVersion('gplots')
#install.packages("heatmap.plus")
#library("heatmap.plus"); packageVersion('heatmap.plus')
#library("RColorBrewer"); packageVersion("RColorBrewer")

asv_tab <- otu_table(INF2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
dim(asv_tab_df)

input <- as.matrix(asv_tab_df)
dim(input)


col_labels <- labels(bray_dend_INF2)
View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_INF2))]
View(col_labels)

col_cols <- labels_colors(bray_dend_INF2)
View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_INF2))]
View(col_cols)

par(mgp=c(0.5,0.4,0))

heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.7, cexCol=0.7, scale="none", Colv = bray_dend_INF2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.75,5.5), xlab = "inocula and fecal samples", ylab = "core INF donor ASVs", key = T, dendrogram = "column") 

legend(0.92, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.64)
```

Now let's use ANCOM and compare the relative abundances of the core INF2 ASVs between the donor inocula used to inoculate the mice and the piglets. We'll first run the ANCOM commands to get the functions setup according to the scripts provided by Dr. Siddhartha Mandal (Lead author of ANCOM publication)

```{r, echo=TRUE}
install.packages("exactRankTests")
library(exactRankTests)
library(nlme)
library(ggplot2)
#########################################################################################
##functions required for executing ANCOM as coded by Dr. Siddhartha Mandal
ancom.W = function(otu_data,var_data,
                   adjusted,repeated,
                   main.var,adj.formula,
                   repeat.var,long,rand.formula,
                   multcorr,sig){
  
  n_otu=dim(otu_data)[2]-1
  
  otu_ids=colnames(otu_data)[-1]
  
  if(repeated==F){
    data_comp=data.frame(merge(otu_data,var_data,by="Sample.ID",all.y=T),row.names=NULL)
    #data_comp=data.frame(merge(otu_data,var_data[,c("Sample.ID",main.var)],by="Sample.ID",all.y=T),row.names=NULL)
  }else if(repeated==T){
    data_comp=data.frame(merge(otu_data,var_data,by="Sample.ID"),row.names=NULL)
    # data_comp=data.frame(merge(otu_data,var_data[,c("Sample.ID",main.var,repeat.var)],by="Sample.ID"),row.names=NULL)
  }
  
  base.formula = paste0("lr ~ ",main.var)
  if(repeated==T){
    repeat.formula = paste0(base.formula," | ", repeat.var)
  }
  if(adjusted==T){
    adjusted.formula = paste0(base.formula," + ", adj.formula)
  }
  
  if( adjusted == F & repeated == F ){
    fformula  <- formula(base.formula)
  } else if( adjusted == F & repeated == T & long == T ){
    fformula  <- formula(base.formula)   
  }else if( adjusted == F & repeated == T & long == F ){
    fformula  <- formula(repeat.formula)   
  }else if( adjusted == T & repeated == F  ){
    fformula  <- formula(adjusted.formula)   
  }else if( adjusted == T & repeated == T  ){
    fformula  <- formula(adjusted.formula)   
  }else{
    stop("Problem with data. Dataset should contain OTU abundances, groups, 
         and optionally an ID for repeated measures.")
  }
  
  
  
  if( repeated==FALSE & adjusted == FALSE){
    if( length(unique(data_comp[,which(colnames(data_comp)==main.var)]))==2 ){
      tfun <- exactRankTests::wilcox.exact
    } else{
      tfun <- stats::kruskal.test
    }
  }else if( repeated==FALSE & adjusted == TRUE){
    tfun <- stats::aov
  }else if( repeated== TRUE & adjusted == FALSE & long == FALSE){
    tfun <- stats::friedman.test
  }else if( repeated== TRUE & adjusted == FALSE & long == TRUE){
    tfun <- nlme::lme
  }else if( repeated== TRUE & adjusted == TRUE){
    tfun <- nlme::lme
  }
  
  logratio.mat <- matrix(NA, nrow=n_otu, ncol=n_otu)
  for(ii in 1:(n_otu-1)){
    for(jj in (ii+1):n_otu){
      data.pair <- data_comp[,which(colnames(data_comp)%in%otu_ids[c(ii,jj)])]
      lr <- log((1+as.numeric(data.pair[,1]))/(1+as.numeric(data.pair[,2])))
      
      lr_dat <- data.frame( lr=lr, data_comp,row.names=NULL )
      
      if(adjusted==FALSE&repeated==FALSE){  ## Wilcox, Kruskal Wallis
        logratio.mat[ii,jj] <- tfun( formula=fformula, data = lr_dat)$p.value
      }else if(adjusted==FALSE&repeated==TRUE&long==FALSE){ ## Friedman's 
        logratio.mat[ii,jj] <- tfun( formula=fformula, data = lr_dat)$p.value
      }else if(adjusted==TRUE&repeated==FALSE){ ## ANOVA
        model=tfun(formula=fformula, data = lr_dat,na.action=na.omit)   
        picker=which(gsub(" ","",row.names(summary(model)[[1]]))==main.var)  
        logratio.mat[ii,jj] <- summary(model)[[1]][["Pr(>F)"]][picker]
      }else if(repeated==TRUE&long==TRUE){ ## GEE
        model=tfun(fixed=fformula,data = lr_dat,
                   random = formula(rand.formula),
                   correlation=corAR1(),
                   na.action=na.omit)   
        picker=which(gsub(" ","",row.names(anova(model)))==main.var)
        logratio.mat[ii,jj] <- anova(model)[["p-value"]][picker]
      }
      
    }
  } 
  
  ind <- lower.tri(logratio.mat)
  logratio.mat[ind] <- t(logratio.mat)[ind]
  
  
  logratio.mat[which(is.finite(logratio.mat)==FALSE)] <- 1
  
  mc.pval <- t(apply(logratio.mat,1,function(x){
    s <- p.adjust(x, method = "BH")
    return(s)
  }))
  
  a <- logratio.mat[upper.tri(logratio.mat,diag=FALSE)==TRUE]
  
  b <- matrix(0,ncol=n_otu,nrow=n_otu)
  b[upper.tri(b)==T] <- p.adjust(a, method = "BH")
  diag(b)  <- NA
  ind.1    <- lower.tri(b)
  b[ind.1] <- t(b)[ind.1]
  
  #########################################
  ### Code to extract surrogate p-value
  surr.pval <- apply(mc.pval,1,function(x){
    s0=quantile(x[which(as.numeric(as.character(x))<sig)],0.95)
    # s0=max(x[which(as.numeric(as.character(x))<alpha)])
    return(s0)
  })
  #########################################
  ### Conservative
  if(multcorr==1){
    W <- apply(b,1,function(x){
      subp <- length(which(x<sig))
    })
    ### Moderate
  } else if(multcorr==2){
    W <- apply(mc.pval,1,function(x){
      subp <- length(which(x<sig))
    })
    ### No correction
  } else if(multcorr==3){
    W <- apply(logratio.mat,1,function(x){
      subp <- length(which(x<sig))
    })
  }
  
  return(W)
  }




ANCOM.main = function(OTUdat,Vardat,
                      adjusted,repeated,
                      main.var,adj.formula,
                      repeat.var,longitudinal,
                      random.formula,
                      multcorr,sig,
                      prev.cut){
  
  p.zeroes=apply(OTUdat[,-1],2,function(x){
    s=length(which(x==0))/length(x)
  })
  
  zeroes.dist=data.frame(colnames(OTUdat)[-1],p.zeroes,row.names=NULL)
  colnames(zeroes.dist)=c("Taxon","Proportion_zero")
  
  zero.plot = ggplot(zeroes.dist, aes(x=Proportion_zero)) + 
    geom_histogram(binwidth=0.1,colour="black",fill="white") + 
    xlab("Proportion of zeroes") + ylab("Number of taxa") +
    theme_bw()
  
  #print(zero.plot)
  
  OTUdat.thinned=OTUdat
  OTUdat.thinned=OTUdat.thinned[,c(1,1+which(p.zeroes<prev.cut))]
  
  otu.names=colnames(OTUdat.thinned)[-1]
  
  W.detected   <- ancom.W(OTUdat.thinned,Vardat,
                          adjusted,repeated,
                          main.var,adj.formula,
                          repeat.var,longitudinal,random.formula,
                          multcorr,sig)
  
  W_stat       <- W.detected
  
  
  ### Bubble plot
  
  W_frame = data.frame(otu.names,W_stat,row.names=NULL)
  W_frame = W_frame[order(-W_frame$W_stat),]
  
  W_frame$detected_0.9=rep(FALSE,dim(W_frame)[1])
  W_frame$detected_0.8=rep(FALSE,dim(W_frame)[1])
  W_frame$detected_0.7=rep(FALSE,dim(W_frame)[1])
  W_frame$detected_0.6=rep(FALSE,dim(W_frame)[1])
  
  W_frame$detected_0.9[which(W_frame$W_stat>0.9*(dim(OTUdat.thinned[,-1])[2]-1))]=TRUE
  W_frame$detected_0.8[which(W_frame$W_stat>0.8*(dim(OTUdat.thinned[,-1])[2]-1))]=TRUE
  W_frame$detected_0.7[which(W_frame$W_stat>0.7*(dim(OTUdat.thinned[,-1])[2]-1))]=TRUE
  W_frame$detected_0.6[which(W_frame$W_stat>0.6*(dim(OTUdat.thinned[,-1])[2]-1))]=TRUE
  
  final_results=list(W_frame,zero.plot)
  names(final_results)=c("W.taxa","PLot.zeroes")
  return(final_results)
}
#########################################################################################
```

ANCOM on the donor samples comparing the abundances of the 26 core INF2 ASVs among the 3 donor inoculae :

```{r, echo=TRUE}
human_donors_INF2_inocula_relevant_ps
human_donors_INF2_inocula_relevant_core_ps <- prune_taxa(core_INF2_taxa, human_donors_INF2_inocula_relevant_ps)
human_donors_INF2_inocula_relevant_core_ps
write.table (otu_table(human_donors_INF2_inocula_relevant_core_ps), 'ANCOM_INF2_inocula_relevant_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
# type echo -n "Sample.ID" | cat - ANCOM_INF2_inocula.txt > ANCOM_INF2_inocula_INPUT.txt in a terminal to add required header to 'ANCOM_CHLD2_inocula_INPUT.txt'
write.table(sample_data(human_donors_INF2_inocula_relevant_core_ps), 'ANCOM_INF2_inocula_relevant_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)
# running ANCOM 
otu_table <- read.table("ANCOM_INF2_inocula_relevant_INPUT.txt", sep = "\t", header = TRUE)
View(otu_table)

map_file <- read.table("ANCOM_INF2_inocula_relevant_comp_mapping_file.txt", sep = "\t", header = TRUE)
View(map_file)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Which_animal%in%c("pig", "mouse"), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Which_animal",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)

comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_INF2_human_inocula_relevant_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # None of the ASVs are significantly different
```

Performing the comparison of relative abundances for the donor samples using DESeq2 :

```{r,echo=TRUE}
# DESeq2 conversion and call
# Install DESeq2 if not already installed
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")

# load DESeq2
library('DESeq2'); packageVersion('DESeq2')

# import mapping file with relevant donor info
map_file <- read.table("ANCOM_INF2_inocula_relevant_comp_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(map_file) <- as.character(map_file[,1])

# merge into a new phyloseq object
DESeq2_INF2_donors_ps <- phyloseq(otu_table(human_donors_INF2_inocula_relevant_core_ps), taxa_are_rows = FALSE, sample_data(map_file), tax_table(human_donors_INF2_inocula_relevant_core_ps))
DESeq2_INF2_donors_ps
diagadds = phyloseq_to_deseq2(DESeq2_INF2_donors_ps, ~Which_animal)
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
diagdds = DESeq(diagdds, fitType="local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(DESeq2_INF2_donors_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_INF2_donors.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 1 ASV (ASV_65) was significantly different
```

Now let's do the ANCOM comparisons between the donor samples and the respective HMA animal models :

```{r, echo=TRUE}
#INF2 donor vs HMA piglets
pig_mouse_comp_INF2_comparison_core_ps
ANCOM_donor_vs_HMA_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_core_ps, Species != 'mouse')
ANCOM_donor_vs_HMA_piglets_ps

## Do ANCOM only on the core ASVs that had actually established

write.table(otu_table(ANCOM_donor_vs_HMA_piglets_ps), 'ANCOM_INF2_donor_relevant_vs_HMA_piglets_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_piglets_ps)
write.table(map_file, 'ANCOM_INF2_inocula_relevant_HMA_piglets_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_INF2_donor_relevant_vs_HMA_piglets_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_INF2_inocula_relevant_HMA_piglets_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','piglet'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.99) # only 21 ASVs considered at prev.cut = 0.99
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_INF2_human_relevant_inocula_vs_HMA_piglets_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 2 ASVs significantly different

# INF2 donor vs HMA mice
pig_mouse_comp_INF2_comparison_core_ps
ANCOM_donor_vs_HMA_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_core_ps, Species != 'piglet')
ANCOM_donor_vs_HMA_mice_ps
write.table (otu_table(ANCOM_donor_vs_HMA_mice_ps), 'ANCOM_INF2_donor_relevant_vs_HMA_mice_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_mice_ps)
write.table(map_file, 'ANCOM_INF2_relevant_inocula_HMA_mice_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_INF2_donor_relevant_vs_HMA_mice_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_INF2_relevant_inocula_HMA_mice_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','mouse'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.99) # had to increase to 0.99 so that all 26 ASVs are included in the analysis (if set to the default 0.9, only 21 ASVs are considered)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_INF2_relevant_human_inocula_vs_HMA_mice_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # no ASVs were significantly differentially abundant
```

Let's also do these abundance comparisons between the human donor taxa and the taxa established in the HMAs using the DESeq2 package through phyloseq :

```{r, echo=TRUE}
# INF2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps

# DESeq2 conversion and call
library('DESeq2'); packageVersion('DESeq2')

diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_piglets_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_piglets_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_INF2_donor_vs_late_piglets.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 7 ASVs were significantly different


# INF2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps

diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_mice_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_mice_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_INF2_donor_vs_late_mice.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 6 ASVs were significantly different
```

Visualizing differences in abundances of ASVs using a heatmap :

```{r, echo=TRUE}
## using raw counts
plot_heatmap(pig_mouse_comp_INF2_comparison_core_ps, sample.label = 'Species', sample.order = 'Species')

## using relative abundances
pig_mouse_comp_INF2_comparison_ps_trans <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_INF2_comparison_core_ps_trans <- prune_taxa(core_INF2_taxa,pig_mouse_comp_INF2_comparison_ps_trans)
pig_mouse_comp_INF2_comparison_core_ps_trans
export_INF2_asv_table <- otu_table(pig_mouse_comp_INF2_comparison_core_ps_trans)
write.table(t(export_INF2_asv_table), "INF2_core_asv_table-2.txt", sep = "\t", row.names = T, col.names = NA)
write.table(otu_table(pig_mouse_comp_INF2_comparison_core_ps_trans), "export_INF2_asv_table_rabund.txt", sep = "\t", row.names = T, col.names = NA)
plot_heatmap(pig_mouse_comp_INF2_comparison_core_ps_trans, sample.label = 'Species', sample.order = 'Species')

## Using variance stabilized count table
pig_mouse_comp_INF2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_INF2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
INF2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_INF2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_INF2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_INF2_comparison_ps)))

INF2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
INF2_vst_core_ps <- prune_taxa(core_INF2_taxa, INF2_vst_ps)
INF2_vst_core_ps
export_INF2_asv_table <- otu_table(INF2_vst_core_ps)
export_INF2_asv_table_df <- as.data.frame(export_INF2_asv_table)
write.table(export_INF2_asv_table, "INF2_core_asv_table.txt", sep = "\t", row.names = T, col.names = NA)
write.table(tax_table(INF2_vst_core_ps), "INF2_core_taxa_table.txt", sep = "\t", row.names = T, col.names = NA)
plot_heatmap(INF2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(INF2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```

Let's compare the late-inoculation piglet and early-inoculation piglet fecal samples to the human donor samples

```{r, echo=TRUE}
# removing the donor samples not used as inocula as well as the donor inocula used to inoculate mice
pig_mouse_comp_INF2_comparison_ps <- subset_samples (pig_mouse_comp_INF2_master_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
mouse_inoculae <- c('R2D269', 'R2D273')
pig_mouse_comp_INF2_comparison_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, !SampleID %in% mouse_inoculae)
pig_mouse_comp_INF2_comparison_ps
pig_mouse_comp_INF2_early_late_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species != 'mouse')
pig_mouse_comp_INF2_early_late_piglets_ps <- subset_samples(pig_mouse_comp_INF2_early_late_piglets_ps, SampleType != 'cecal')
pig_mouse_comp_INF2_early_late_piglets_ps

pig_mouse_comp_INF2_early_late_piglets_ps.rarefied <- rarefy_even_depth(pig_mouse_comp_INF2_early_late_piglets_ps, rngseed = 1, sample.size = min(sample_sums(pig_mouse_comp_INF2_early_late_piglets_ps)), replace = F) # rarefied at 20637 reads/sample

# plotting PCoA using unweighted unifrac distances - INF2
unweighted_unifrac_dm_INF2_early_vs_late <- phyloseq::distance(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, method = "unifrac", weighted = F)
unw_ord <- ordinate(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, method = "PCoA", distance = unweighted_unifrac_dm_INF2_early_vs_late)
unweighted_pcoa_INF2_early_vs_late <- plot_ordination(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, unw_ord, color = "Timingofinocu") 
unweighted_pcoa_INF2_early_vs_late

# plotting PCoA using weighted unifrac distances - INF2
weighted_unifrac_dm_INF2_early_vs_late <- phyloseq::distance(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, method = "unifrac", weighted = T)
w_ord <- ordinate(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, method = "PCoA", distance = weighted_unifrac_dm_INF2_early_vs_late)
weighted_pcoa_INF2_early_vs_late <- plot_ordination(pig_mouse_comp_INF2_early_late_piglets_ps.rarefied, w_ord, color = "Timingofinocu") 
p1 <- weighted_pcoa_INF2_early_vs_late + ggtitle("PCoA plot based on weighted unifrac distance matrices")
p2 <- p1 + labs(color = 'Timing of inoculum')
p3 <- p2 + theme(plot.title = element_text(face = "bold", size = (12), hjust = 0.5)) 
p4 <- p3 + theme(legend.title = element_text(face = "bold", size = (10)))
p4
```

We don't see any clear separation between early and late inoculation piglet samples in both the weighted and unweighted unifrac distance matrix-based PCoA plots


**CHILD Donor Analysis**

```{r, echo = TRUE}
## MAKING RAREFCATION PLOTS
merged_pig_mouse_comp_ps_no_neg_no_cecal <- subset_samples(merged_pig_mouse_comp_ps_no_neg, SampleType != 'cecal')
merged_pig_mouse_comp_CHLD_ps <- subset_samples(merged_pig_mouse_comp_ps_no_neg_no_cecal, Donor=="CHLD2")
merged_pig_mouse_comp_CHLD_ps <- subset_samples(merged_pig_mouse_comp_CHLD_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
merged_pig_mouse_comp_CHLD_ps <- subset_samples(merged_pig_mouse_comp_CHLD_ps, Time_point != "at_euthanization")
merged_pig_mouse_comp_CHLD_ps

## export as .txt files
write.table(t(otu_table(merged_pig_mouse_comp_CHLD_ps)), "CHLD2_otu_table.txt", sep = "\t", row.names = TRUE, col.names=NA, quote = FALSE)
write.table(sample_data(merged_pig_mouse_comp_CHLD_ps), "CHLD2_sample_data.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

```
## Using QIIME2 to make rarefaction curves

# Export feature table and converting it into a biom file :

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ echo -n "#OTU Table" | cat - CHLD2_otu_table.txt > CHLD2_otu_biom_table.txt
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ biom convert -i CHLD2_otu_biom_table.txt -o CHLD2_otu_biom_table.biom --table-type="OTU table" --to-hdf5

# importing biom table to qiime2 :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import --input-path CHLD2_otu_biom_table.biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path CHLD2_otu_biom_table.biom.qza

# Use feature table summary to find out what the medium read depth is (for use in --p-max-depth of following command) :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime feature-table summarize --i-table CHLD2_otu_biom_table.biom.qza --o-visualization CHLD_table.qzv --m-sample-metadata-file CHLD2_sample_data.txt

# Performing alpha rarefactions :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime diversity alpha-rarefaction \
> --i-table CHLD2_otu_biom_table.biom.qza \
> --i-phylogeny ASV_rep_set.phylip.qza \
> --p-max-depth 37000 \
> --m-metadata-file CHLD2_sample_data.txt \
> --o-visualization CHLD2-alpha-rarefaction.qzv
```

Let's separate out the relevant 'Child' donor samples :

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_master_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Donor == 'CHLD2')
pig_mouse_comp_CHLD2_master_ps 

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animal fecal bacterial communities
pig_mouse_comp_CHLD2_comparison_ps <- subset_samples(pig_mouse_comp_CHLD2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_CHLD2_comparison_ps
pig_mouse_comp_CHLD2_comparison_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, SampleType != 'cecal')
pig_mouse_comp_CHLD2_comparison_ps

# Transform count table to relative abundance table
rel_abund_CHLD2_ps <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x){x / sum(x)}) 
rel_abund_CHLD2_ps

# plotting PCoA using unweighted unifrac distances - CHLD2
unw_ord_CHLD2 <- ordinate(rel_abund_CHLD2_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_CHLD2 <- plot_ordination(rel_abund_CHLD2_ps, unw_ord_CHLD2, color = "Species") 
unweighted_pcoa_CHLD2

# plotting PCoA using weighted unifrac distances - CHLD2
w_ord_CHLD2 <- ordinate(rel_abund_CHLD2_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_CHLD2 <- plot_ordination(rel_abund_CHLD2_ps, w_ord_CHLD2, color = "Species") 
weighted_pcoa_CHLD2

# plotting using binary jaccard distances
bin_jac_ord_CHLD2 <- ordinate(rel_abund_CHLD2_ps, method = "PCoA", distance = "jaccard", binary = TRUE)
binary_jaccard_pcoa_CHLD2 <- plot_ordination(rel_abund_CHLD2_ps, bin_jac_ord_CHLD2, color = "Species") 
binary_jaccard_pcoa_CHLD2

# plotting using Bray-Curtis distances
bray_ord_CHLD2 <- ordinate(rel_abund_CHLD2_ps, method="PCoA", distance = "bray")
bray_pcoa_CHLD2 <- plot_ordination(rel_abund_CHLD2_ps, bray_ord_CHLD2, color="Species")
bray_pcoa_CHLD2

## Let's also color the weighted and unweighted unifrac plots by sampling time point
unwunifrac_CHLD2_by_tp <- plot_ordination(rel_abund_CHLD2_ps, unw_ord_CHLD2, color = "Time_point", shape = "Species") + ggtitle("Child donor PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_CHLD2_by_tp

wunifrac_CHLD2_by_tp <- plot_ordination(rel_abund_CHLD2_ps, w_ord_CHLD2, color = "Time_point", shape = "Species") + ggtitle("Child donor PCoA plot - Weighted unifrac") + theme(plot.title = element_text(hjust = 0.5))
wunifrac_CHLD2_by_tp

### performing PERMANOVA analyses
adonis(phyloseq::distance(rel_abund_CHLD2_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_CHLD2_ps), "data.frame"), permutations = 999)

results_CHLD2_unweighted <- pairwise_permanova(rel_abund_CHLD2_ps,"Species","Time_point","unifrac") 
print(results_CHLD2_unweighted)
```

Let's see how the donor samples look like

```{r, echo=TRUE}
human_donors_CHLD2_ps <- subset_samples(pig_mouse_comp_CHLD2_master_ps, Species == 'human')
human_donors_CHLD2_ps
write.table(sample_data(human_donors_CHLD2_ps), "CHLD2_donors.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
human_donors_CHLD2_inocula_ps <- subset_samples(human_donors_CHLD2_ps, Actual_inoculum == 'yes')
human_donors_CHLD2_inocula_ps
```

Defining a 'core' CHLD2 donor microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' CHLD2 donor microbiota
prevdf_CHLD2 = apply(X = otu_table(human_donors_CHLD2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_CHLD2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_CHLD2 = data.frame(Prevalence=prevdf_CHLD2, TotalAbundance=taxa_sums(human_donors_CHLD2_inocula_ps))
View(prevdf_CHLD2)

core_threshold = nsamples(human_donors_CHLD2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_CHLD2_taxa <- rownames(prevdf_CHLD2)[prevdf_CHLD2$Prevalence >= core_threshold]
human_donors_CHLD2_inocula_core_ps <- prune_taxa(core_CHLD2_taxa, human_donors_CHLD2_inocula_ps)
human_donors_CHLD2_inocula_core_ps # 76 core ASVs identified
sum(otu_table(human_donors_CHLD2_inocula_core_ps))/sum(otu_table(human_donors_CHLD2_inocula_ps)) # the 76 core ASVs account for 96% of total CHLD2 donor reads

# making box plots for the donor samples
human_donors_CHLD2_inocula_ps_trans <- transform_sample_counts(human_donors_CHLD2_inocula_ps, function(x) x/sum(x))
human_donors_CHLD2_inocula_core_ps_trans <- prune_taxa(core_CHLD2_taxa, human_donors_CHLD2_inocula_ps_trans)
theme_set(theme_bw())
p1 <- plot_bar(human_donors_CHLD2_inocula_core_ps_trans, fill = "Genus")
p2 <- p1 + theme (legend.text = element_text(size = 6), legend.title = element_text(size = 8))
p3 <- p2 + theme(legend.key.height = unit(0.5, "cm"), legend.key.width = unit(0.2, "cm"))
p3
```

Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula communities for CHLD2 donor. 

```{r, echo=TRUE}
rel_abund_CHLD2_ps
rel_abund_CHLD2_core_ps <- prune_taxa(core_CHLD2_taxa, rel_abund_CHLD2_ps)
rel_abund_CHLD2_core_ps 

# plotting PCoA using unweighted unifrac distances for core taxa - CHLD2
unw_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, unw_ord_CHLD2_core, color = "Species", shape = "Species") + ggtitle("CHLD donor PCoA plot - Unweighted Unifrac") + theme(plot.title = element_text(hjust = 0.5, size = 10)) 
unweighted_pcoa_CHLD2_core

# plotting PCoA using weighted unifrac distances for core taxa - CHLD2
w_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, w_ord_CHLD2_core, color = "Species") 
weighted_pcoa_CHLD2_core

# plotting PCoA using Jaccard distances for core taxa - CHLD2
bin_jac_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "jaccard", binary = T)
jaccard_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, bin_jac_ord_CHLD2_core, color = "Species") 
jaccard_pcoa_CHLD2_core

# plotting PCoA using Bray-Curtis distances for core taxa - CHLD2
bray_ord_CHLD2_core <- ordinate(rel_abund_CHLD2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_CHLD2_core <- plot_ordination(rel_abund_CHLD2_core_ps, bray_ord_CHLD2_core, color = "Species", shape = "Species") + ggtitle("CHLD donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 10))
bray_pcoa_CHLD2_core

CHLD2_core_unifrac_comb <- grid.arrange(ncol = 2, unweighted_pcoa_CHLD2_core, weighted_pcoa_CHLD2_core)
CHLD2_core_unifrac_comb

CHLD2_core_jaccard_bray_comb <- grid.arrange(ncol = 2, jaccard_pcoa_CHLD2_core, bray_pcoa_CHLD2_core)
CHLD2_core_jaccard_bray_comb

CHLD_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_CHLD2_core, bray_pcoa_CHLD2_core)
CHLD_core_unifrac_bray_comb

## Let's also color the unweighted unifrac and bray-curtis plots by sampling time point
unwunifrac_CHLD2_core_by_tp <- plot_ordination(rel_abund_CHLD2_core_ps, unw_ord_CHLD2_core, color = "Time_point", shape = "Species") + ggtitle("Child donor PCoA plot for core ASVs - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_CHLD2_core_by_tp

bray_CHLD2_by_tp <- plot_ordination(rel_abund_CHLD2_core_ps, bray_ord_CHLD2_core, color = "Time_point", shape = "Species") + ggtitle("Child donor PCoA plot core ASVs - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5))
bray_CHLD2_by_tp

CHLD2_core_PCoA_by_tp_comb <- grid.arrange(ncol = 2, unwunifrac_CHLD2_core_by_tp, bray_CHLD2_by_tp)
CHLD2_core_PCoA_by_tp_comb
```

PERMANOVA analysis to check if clustering patterns are statistically siginificantly different

```{r, echo=TRUE}
### performing PERMANOVA analyses on unweighted unifrac
rel_abund_CHLD2_core_ps
metadata <- as(sample_data(rel_abund_CHLD2_core_ps), "data.frame")
View(sample_data(rel_abund_CHLD2_core_ps))
adonis(phyloseq::distance(rel_abund_CHLD2_core_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_CHLD2_core_ps), "data.frame"), permutations = 999)

## Carrying out pair-wise comparisons
results_CHLD2_unweighted <- pairwise_permanova(rel_abund_CHLD2_core_ps,"Species","Time_point","unifrac") 
print(results_CHLD2_unweighted)
```


Hierarchical clustering based on 'Bray-Curtis' distances

```{r, echo= TRUE}

## CHLD2 Donor

# Use the relative abundance table
rel_abund_CHLD2_core_ps
bray_dist_CHLD2 <- phyloseq::distance(rel_abund_CHLD2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_CHLD2 <- hclust(bray_dist_CHLD2, method = "ward.D2")
plot(bray_clust_CHLD2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_CHLD2 <- read.table("mapping_files_for_dendrograms/CHLD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_CHLD2)
mapping_file_dendro_CHLD2$Color_Sp <- as.character(mapping_file_dendro_CHLD2$Color_Sp)

# making dendrogram
bray_dend_CHLD2 <- as.dendrogram(bray_clust_CHLD2, hang=0.1)
SampleIDs_dend_CHLD2 <- labels(bray_dend_CHLD2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_CHLD2

sorted_mapping_CHLD2 <- mapping_file_dendro_CHLD2[match(SampleIDs_dend_CHLD2, mapping_file_dendro_CHLD2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_CHLD2)

# color dendrogram labels 
labels_colors(bray_dend_CHLD2) <- sorted_mapping_CHLD2$Color_Sp
labels_colors(bray_dend_CHLD2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_CHLD2) <- sorted_mapping_CHLD2$Time_point
bray_dend_CHLD2 <- set(bray_dend_CHLD2, "labels_cex", 0.5)
plot(bray_dend_CHLD2, ylab="Bray-Curtis distances")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
asv_tab <- otu_table(CHLD2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
dim(asv_tab_df)

input <- as.matrix(asv_tab_df)
dim(input)


col_labels <- labels(bray_dend_CHLD2)
View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_CHLD2))]
View(col_labels)

col_cols <- labels_colors(bray_dend_CHLD2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_CHLD2))]
#View(col_cols)

par(mgp=c(0.5,0.4,0))

heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.4, cexCol=0.65, scale="none", Colv = bray_dend_CHLD2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.5,4.5), xlab = "inocula and fecal samples", ylab = "core CHLD donor ASVs", dendrogram = "column", key = T) 

legend(0.92, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.6)
```

ANCOM on the donor samples :

```{r, echo=TRUE}
write.table (otu_table(human_donors_CHLD2_inocula_core_ps), 'ANCOM_CHLD2_inocula_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
# type echo -n "Sample.ID" | cat - ANCOM_INF2_inocula.txt > ANCOM_INF2_inocula_INPUT.txt in a terminal to add required header to 'ANCOM_CHLD2_inocula_INPUT.txt'
write.table(sample_data(human_donors_CHLD2_inocula_core_ps), 'ANCOM_CHLD2_inocula_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)
# running ANCOM 
otu_table <- read.table("ANCOM_CHLD2_inocula_INPUT.txt", sep = "\t", header = TRUE)
#View(otu_table)

map_file <- read.table("ANCOM_CHLD2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE)
#View(map_file)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Which_animal%in%c("pig", "mouse"), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Which_animal",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)

comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_CHLD2_human_inocula_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # None of the ASVs are significantly different
```


Performing the comparison of relative abundances for the donor samples using DESeq2 :

```{r,echo=TRUE}
# DESeq2 conversion and call
# Install DESeq2 if not already installed
# load DESeq2
library('DESeq2'); packageVersion('DESeq2')

# important mapping file with relevant donor info
map_file <- read.table("ANCOM_CHLD2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(map_file) <- as.character(map_file[,1])

# merge into a new phyloseq object
DESeq2_CHLD2_donors_ps <- phyloseq(otu_table(human_donors_CHLD2_inocula_core_ps), taxa_are_rows = FALSE, sample_data(map_file), tax_table(human_donors_CHLD2_inocula_core_ps))
DESeq2_CHLD2_donors_ps

diagadds = phyloseq_to_deseq2(DESeq2_CHLD2_donors_ps, ~Which_animal)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj > alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(DESeq2_CHLD2_donors_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
View(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_CHLD2_donors.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 

# none of the core ASVs are sig. different in their relative abundances between the inoculae
```


Now let's do the ANCOM comparisons between the donor samples and the respective HMA animal models :

```{r, echo=TRUE}
#CHLD2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_core_ps, Species != 'mouse')
ANCOM_donor_vs_HMA_piglets_ps
write.table (otu_table(ANCOM_donor_vs_HMA_piglets_ps), 'ANCOM_CHLD2_donor_vs_HMA_piglets_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_piglets_ps)
write.table(map_file, 'ANCOM_CHLD2_inocula_HMA_piglets_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_CHLD2_donor_vs_HMA_piglets_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_CHLD2_inocula_HMA_piglets_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','piglet'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_CHLD2_human_inocula_vs_HMA_piglets_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 3 ASVs significantly different

#CHLD2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_core_ps, Species != 'piglet')
ANCOM_donor_vs_HMA_mice_ps
write.table (otu_table(ANCOM_donor_vs_HMA_mice_ps), 'ANCOM_CHLD2_donor_vs_HMA_mice_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_mice_ps)
write.table(map_file, 'ANCOM_CHLD2_inocula_HMA_mice_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_CHLD2_donor_vs_HMA_mice_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_CHLD2_inocula_HMA_mice_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','mouse'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.99) # had to increase to 0.99 so that all 76 ASVs are included in the analysis (if set to the default 0.9, only 36 ASVs are considered meaning that the other ASVs (40) are present in <= 10% of samples)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_CHLD2_human_inocula_vs_HMA_mice_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 27 ASVs significantly different 
```

Using DESeq2 to compare abundances between donor and HMAs

```{r, echo=TRUE}
# CHLD2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_piglets_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_piglets_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_CHLD2_donor_vs_piglets.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 40 ASVs were significantly differentially abundant


# CHLD2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_mice_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_mice_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_CHLD2_donor_vs_mice.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 14 ASVs were significantly differentially abundant
```

According to the ANCOM analysis, more than half the core donor ASVs are not prevalent in at least 90% of the mouse fecal samples. Let's make a heatmap and see if this is the case.

```{r, echo=TRUE}
# using raw counts
pig_mouse_comp_CHLD2_comparison_core_ps
plot_heatmap(pig_mouse_comp_CHLD2_comparison_core_ps, sample.label = 'Species', sample.order = 'Species')

# using relative abundances
pig_mouse_comp_CHLD2_comparison_ps_trans <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_CHLD2_comparison_core_ps_trans <- prune_taxa(core_CHLD2_taxa,pig_mouse_comp_CHLD2_comparison_ps_trans)
pig_mouse_comp_CHLD2_comparison_core_ps_trans
plot_heatmap(pig_mouse_comp_CHLD2_comparison_core_ps_trans, sample.label = 'Species', sample.order = 'Species')

## Using variance stabilized count table
pig_mouse_comp_CHLD2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_CHLD2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
CHLD2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_CHLD2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_CHLD2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_CHLD2_comparison_ps)))

CHLD2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
CHLD2_vst_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_vst_ps)
CHLD2_vst_core_ps
plot_heatmap(CHLD2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(CHLD2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```

 
 **ADULT Donor Analysis**
 
```{r, echo = TRUE}
## MAKING RAREFCATION PLOTS
merged_pig_mouse_comp_ps_no_neg_no_cecal <- subset_samples(merged_pig_mouse_comp_ps_no_neg, SampleType != 'cecal')
merged_pig_mouse_comp_ADLT_ps <- subset_samples(merged_pig_mouse_comp_ps_no_neg_no_cecal, Donor=="ADLT2")
merged_pig_mouse_comp_ADLT_ps <- subset_samples(merged_pig_mouse_comp_ADLT_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
merged_pig_mouse_comp_ADLT_ps <- subset_samples(merged_pig_mouse_comp_ADLT_ps, Time_point != "at_euthanization")
merged_pig_mouse_comp_ADLT_ps

## export as .txt files
write.table(t(otu_table(merged_pig_mouse_comp_ADLT_ps)), "ADLT2_otu_table.txt", sep = "\t", row.names = TRUE, col.names=NA, quote = FALSE)
write.table(sample_data(merged_pig_mouse_comp_ADLT_ps), "ADLT2_sample_data.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

```
## Using QIIME2 to make rarefaction curves

# Export feature table and converting it into a biom file :

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ echo -n "#OTU Table" | cat - ADLT2_otu_table.txt > ADLT2_otu_biom_table.txt
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ biom convert -i ADLT2_otu_biom_table.txt -o ADLT2_otu_biom_table.biom --table-type="OTU table" --to-hdf5

# importing biom table to qiime2 :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import --input-path ADLT2_otu_biom_table.biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path ADLT2_otu_biom_table.biom.qza

# Use feature table summary to find out what the medium read depth is (for use in --p-max-depth of following command) :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime feature-table summarize --i-table ADLT2_otu_biom_table.biom.qza --o-visualization ADLT_table.qzv --m-sample-metadata-file ADLT2_sample_data.txt

# Performing alpha rarefactions :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime diversity alpha-rarefaction \
> --i-table ADLT2_otu_biom_table.biom.qza \
> --i-phylogeny ASV_rep_set.phylip.qza \
> --p-max-depth 38000 \
> --m-metadata-file ADLT2_sample_data.txt \
> --o-visualization ADLT2-alpha-rarefaction.qzv
``` 

Separating out the relevant ADULT2 samples :

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_master_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Donor == 'ADLT2')
pig_mouse_comp_ADLT2_master_ps

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animal fecal bacterial communities
pig_mouse_comp_ADLT2_comparison_ps <- subset_samples(pig_mouse_comp_ADLT2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_ADLT2_comparison_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, SampleType != 'cecal') # removing cecal samples
pig_mouse_comp_ADLT2_comparison_ps 

# Sample 105 was identified as an outlier so it shall be removed
pig_mouse_comp_ADLT2_comparison_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, SampleID != '105')
pig_mouse_comp_ADLT2_comparison_ps

# Transform count table to relative abundance table
rel_abund_ADLT2_ps <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x){x / sum(x)}) 
rel_abund_ADLT2_ps

# plotting PCoA using unweighted unifrac distances - ADLT2
unw_ord_ADLT2 <- ordinate(rel_abund_ADLT2_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ADLT2 <- plot_ordination(rel_abund_ADLT2_ps, unw_ord_ADLT2, color = "Species") 
unweighted_pcoa_ADLT2

# plotting PCoA using weighted unifrac distances - ADLT2
w_ord_ADLT2 <- ordinate(rel_abund_ADLT2_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_ADLT2 <- plot_ordination(rel_abund_ADLT2_ps, w_ord_ADLT2, color = "Species") 
weighted_pcoa_ADLT2

# plotting using binary jaccard distances
bin_jac_ord_ADLT2 <- ordinate(rel_abund_ADLT2_ps, method = "PCoA", distance = "jaccard", binary = TRUE)
binary_jaccard_pcoa_ADLT2 <- plot_ordination(rel_abund_ADLT2_ps, bin_jac_ord_ADLT2, color = "Species") 
binary_jaccard_pcoa_ADLT2

# plotting using Bray-Curtis distances
bray_ord_ADLT2 <- ordinate(rel_abund_ADLT2_ps, method="PCoA", distance = "bray")
bray_pcoa_ADLT2 <- plot_ordination(rel_abund_ADLT2_ps, bray_ord_ADLT2, color="Species")
bray_pcoa_ADLT2

## Let's also color the weighted and unweighted unifrac plots by sampling time point
unwunifrac_ADLT2_by_tp <- plot_ordination(rel_abund_ADLT2_ps, unw_ord_ADLT2, color = "Time_point", shape = "Species") + ggtitle("Adult donor PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_ADLT2_by_tp

wunifrac_ADLT2_by_tp <- plot_ordination(rel_abund_ADLT2_ps, w_ord_ADLT2, color = "Time_point", shape = "Species") + ggtitle("Adult donor PCoA plot - Weighted unifrac") + theme(plot.title = element_text(hjust = 0.5))
wunifrac_ADLT2_by_tp

### performing PERMANOVA analyses
adonis(phyloseq::distance(rel_abund_ADLT2_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_ADLT2_ps), "data.frame"), permutations = 999)

results_ADLT2_unweighted <- pairwise_permanova(rel_abund_ADLT2_ps,"Species","Time_point","unifrac") 
print(results_ADLT2_unweighted)
```

Let's see how the ADULT2 donor samples look like

```{r, echo=TRUE}
human_donors_ADLT2_ps <- subset_samples(pig_mouse_comp_ADLT2_master_ps, Species == 'human')
human_donors_ADLT2_ps
write.table(sample_data(human_donors_ADLT2_ps), "ADLT2_donors.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
human_donors_ADLT2_inocula_ps <- subset_samples(human_donors_ADLT2_ps, Actual_inoculum == 'yes')
human_donors_ADLT2_inocula_ps
```

Defining a 'core' ADLT2 donor microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' ADLT2 donor microbiota
prevdf_ADLT2 = apply(X = otu_table(human_donors_ADLT2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ADLT2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ADLT2 = data.frame(Prevalence=prevdf_ADLT2, TotalAbundance=taxa_sums(human_donors_ADLT2_inocula_ps))
View(prevdf_ADLT2)

core_threshold = nsamples(human_donors_ADLT2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_ADLT2_taxa <- rownames(prevdf_ADLT2)[prevdf_ADLT2$Prevalence >= core_threshold]
human_donors_ADLT2_inocula_core_ps <- prune_taxa(core_ADLT2_taxa, human_donors_ADLT2_inocula_ps)
human_donors_ADLT2_inocula_core_ps # 140 core ASVs identified
sum(otu_table(human_donors_ADLT2_inocula_core_ps))/sum(otu_table(human_donors_ADLT2_inocula_ps)) # the 140 core ASVs account for ~ 97.5 % of total ADLT2 donor reads

# box plots for human donor samples
human_donors_ADLT2_inocula_ps_trans <- transform_sample_counts(human_donors_ADLT2_inocula_ps, function(x) x/sum(x))
human_donors_ADLT2_inocula_core_ps_trans <- prune_taxa(core_ADLT2_taxa, human_donors_ADLT2_inocula_ps_trans)
theme_set(theme_bw())
p1 <- plot_bar(human_donors_ADLT2_inocula_core_ps_trans, fill = "Genus")
p2 <- p1 + theme (legend.text = element_text(size = 6), legend.title = element_text(size = 8))
p3 <- p2 + theme(legend.key.height = unit(0.5, "cm"), legend.key.width = unit(0.2, "cm"))
p4 <- p3 + theme(legend.position = 'none')
p4
```

Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula communities for ADLT2 donor. 

```{r, echo=TRUE}
rel_abund_ADLT2_ps
rel_abund_ADLT2_core_ps <- prune_taxa(core_ADLT2_taxa, rel_abund_ADLT2_ps)
rel_abund_ADLT2_core_ps 

# plotting PCoA using unweighted unifrac distances for core taxa - ADLT2
unw_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, unw_ord_ADLT2_core, color = "Species", shape = "Species") + ggtitle("ADLT donor PCoA plot - Unweighted Unifrac") + theme(plot.title = element_text(hjust = 0.5, size = 10))
unweighted_pcoa_ADLT2_core

# plotting PCoA using weighted unifrac distances for core taxa - ADLT2
w_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, w_ord_ADLT2_core, color = "Species") 
weighted_pcoa_ADLT2_core

# plotting PCoA using Jaccard distances for core taxa - ADLT2
bin_jac_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "jaccard", binary = T)
jaccard_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, bin_jac_ord_ADLT2_core, color = "Species") 
jaccard_pcoa_ADLT2_core

# plotting PCoA using Bray-Curtis distances for core taxa - ADLT2
bray_ord_ADLT2_core <- ordinate(rel_abund_ADLT2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_ADLT2_core <- plot_ordination(rel_abund_ADLT2_core_ps, bray_ord_ADLT2_core, color = "Species", shape = "Species" ) + ggtitle("ADLT donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 10))
bray_pcoa_ADLT2_core

ADLT2_core_unifrac_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ADLT2_core, weighted_pcoa_ADLT2_core)
ADLT2_core_unifrac_comb

ADLT2_core_jaccard_bray_comb <- grid.arrange(ncol = 2, jaccard_pcoa_ADLT2_core, bray_pcoa_ADLT2_core)
ADLT2_core_jaccard_bray_comb

ADLT2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ADLT2_core, bray_pcoa_ADLT2_core)
ADLT2_core_unifrac_bray_comb

## Let's also color the unweighted unifrac and bray-curtis plots by sampling time point
unwunifrac_ADLT2_core_by_tp <- plot_ordination(rel_abund_ADLT2_core_ps, unw_ord_ADLT2_core, color = "Time_point", shape = "Species") + ggtitle("Adult donor PCoA plot for core ASVs - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_ADLT2_core_by_tp

bray_ADLT2_by_tp <- plot_ordination(rel_abund_ADLT2_core_ps, bray_ord_ADLT2_core, color = "Time_point", shape = "Species") + ggtitle("Adult donor PCoA plot core ASVs - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5))
bray_ADLT2_by_tp

ADLT2_core_PCoA_by_tp_comb <- grid.arrange(ncol = 2, unwunifrac_ADLT2_core_by_tp, bray_ADLT2_by_tp)
ADLT2_core_PCoA_by_tp_comb
```


PERMANOVA analysis to check if clustering patterns are statistically siginificantly different

```{r, echo=TRUE}
### performing PERMANOVA analyses on unweighted unifrac
rel_abund_ADLT2_core_ps
metadata <- as(sample_data(rel_abund_ADLT2_core_ps), "data.frame")
View(sample_data(rel_abund_ADLT2_core_ps))
adonis(phyloseq::distance(rel_abund_ADLT2_core_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_ADLT2_core_ps), "data.frame"), permutations = 999)

## Carrying out pair-wise comparisons
results_ADLT2_unweighted <- pairwise_permanova(rel_abund_ADLT2_core_ps,"Species","Time_point","unifrac") 
print(results_ADLT2_unweighted)
```


Hierarchical clustering based on Bray-Curtis distances

```{r, echo=TRUE}

# Use the relative abundance table
rel_abund_ADLT2_core_ps
bray_dist_ADLT2 <- phyloseq::distance(rel_abund_ADLT2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_ADLT2 <- hclust(bray_dist_ADLT2, method = "ward.D2")
plot(bray_clust_ADLT2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_ADLT2 <- read.table("mapping_files_for_dendrograms/ADLT2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_ADLT2)
mapping_file_dendro_ADLT2$Color_Sp <- as.character(mapping_file_dendro_ADLT2$Color_Sp)

# making dendrogram
bray_dend_ADLT2 <- as.dendrogram(bray_clust_ADLT2, hang=0.1)
SampleIDs_dend_ADLT2 <- labels(bray_dend_ADLT2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ADLT2

sorted_mapping_ADLT2 <- mapping_file_dendro_ADLT2[match(SampleIDs_dend_ADLT2, mapping_file_dendro_ADLT2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_ADLT2)

# color dendrogram labels 
labels_colors(bray_dend_ADLT2) <- sorted_mapping_ADLT2$Color_Sp
labels_colors(bray_dend_ADLT2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_ADLT2) <- sorted_mapping_ADLT2$Time_point
bray_dend_ADLT2 <- set(bray_dend_ADLT2, "labels_cex", 0.5)
plot(bray_dend_ADLT2, ylab="Bray-Curtis distances")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
asv_tab <- otu_table(ADLT2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
dim(asv_tab_df)

input <- as.matrix(asv_tab_df)
dim(input)


col_labels <- labels(bray_dend_ADLT2)
#View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_ADLT2))]
#View(col_labels)

col_cols <- labels_colors(bray_dend_ADLT2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_ADLT2))]
#View(col_cols)
par(mgp=c(0.5,0.4,0))

heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.45, cexCol=0.65, scale="none", Colv = bray_dend_ADLT2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.5,4.5), xlab = "inocula and fecal samples", ylab = "ADLT donor core ASVs", key = T, dendrogram = "column") 

legend(0.92, 1.1,xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.6)
```

ANCOM on the ADLT2 donor samples :

```{r, echo=TRUE}
write.table (otu_table(human_donors_ADLT2_inocula_core_ps), 'ANCOM_ADLT2_inocula_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
# type echo -n "Sample.ID" | cat - ANCOM_INF2_inocula.txt > ANCOM_INF2_inocula_INPUT.txt in a terminal to add required header to 'ANCOM_CHLD2_inocula_INPUT.txt'
write.table(sample_data(human_donors_ADLT2_inocula_core_ps), 'ANCOM_ADLT2_inocula_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

# running ANCOM 
otu_table <- read.table("ANCOM_ADLT2_inocula_INPUT.txt", sep = "\t", header = TRUE)
#View(otu_table)

map_file <- read.table("ANCOM_ADLT2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE)
#View(map_file)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Which_animal%in%c("pig", "mouse"), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Which_animal",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)

comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ADLT2_human_inocula_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # None of the ASVs are significantly different
```

Performing the comparison of relative abundances for the donor samples using DESeq2 :

```{r,echo=TRUE}
# DESeq2 conversion and call
# Install DESeq2 if not already installed
# load DESeq2
library('DESeq2'); packageVersion('DESeq2')

# important mapping file with relevant donor info
map_file <- read.table("ANCOM_ADLT2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(map_file) <- as.character(map_file[,1])

# merge into a new phyloseq object
DESeq2_ADLT2_donors_ps <- phyloseq(otu_table(human_donors_ADLT2_inocula_core_ps), taxa_are_rows = FALSE, sample_data(map_file), tax_table(human_donors_ADLT2_inocula_core_ps))
DESeq2_ADLT2_donors_ps

diagadds = phyloseq_to_deseq2(DESeq2_ADLT2_donors_ps, ~Which_animal)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(DESeq2_ADLT2_donors_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
View(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ADLT2_donors.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 

# ASVs 187, 363, and 478 of the core ASVs were sig. different in their relative abundances between the inoculae
```


Now let's do the ANCOM comparisons for the establishment of core donor ASVs between the donor samples and the respective HMA animal models :

```{r, echo=TRUE}
#ADLT2 donor vs HMA piglets
pig_mouse_comp_ADLT2_comparison_core_ps
ANCOM_donor_vs_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_core_ps, Species != 'mouse')
ANCOM_donor_vs_HMA_piglets_ps
write.table (otu_table(ANCOM_donor_vs_HMA_piglets_ps), 'ANCOM_ADLT2_donor_vs_HMA_piglets_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_piglets_ps)
write.table(map_file, 'ANCOM_ADLT2_inocula_HMA_piglets_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_ADLT2_donor_vs_HMA_piglets_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_ADLT2_inocula_HMA_piglets_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','piglet'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ADLT2_human_inocula_vs_HMA_piglets_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 13 ASVs significantly different

#CHLD2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_core_ps, Species != 'piglet')
ANCOM_donor_vs_HMA_mice_ps
write.table (otu_table(ANCOM_donor_vs_HMA_mice_ps), 'ANCOM_ADLT2_donor_vs_HMA_mice_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_mice_ps)
write.table(map_file, 'ANCOM_ADLT2_inocula_HMA_mice_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_ADLT2_donor_vs_HMA_mice_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_ADLT2_inocula_HMA_mice_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','mouse'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.99) # had to increase to 0.99 so that all 140 ASVs are included in the analysis (if set to the default 0.9, only 47 ASVs are considered meaning that the other ASVs (93) are present in <= 10% of samples)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ADLT2_human_inocula_vs_HMA_mice_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 41 ASVs significantly different 
```

Using DESeq2 to compare abundances between donor and HMAs

```{r, echo=TRUE}
# ADLT2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_piglets_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_piglets_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ADLT2_donor_vs_piglets.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 64 ASVs were significantly differentially abundant


# ADLT2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_mice_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_mice_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ADLT2_donor_vs_mice.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 11 ASVs were significantly differentially abundant
```


According to the ANCOM analysis, only about 1/3 of the donor core ASVs are establishing in the mice. Let's make a heatmap and see if this is the case.

```{r, echo=TRUE}
# using raw counts
pig_mouse_comp_ADLT2_comparison_core_ps
plot_heatmap(pig_mouse_comp_ADLT2_comparison_core_ps, sample.label = 'Species', sample.order = 'Species')

# using relative abundances
pig_mouse_comp_ADLT2_comparison_ps_trans <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_ADLT2_comparison_ps_trans
pig_mouse_comp_ADLT2_comparison_core_ps_trans <- prune_taxa(core_ADLT2_taxa, pig_mouse_comp_ADLT2_comparison_ps_trans)
pig_mouse_comp_ADLT2_comparison_core_ps_trans

plot_heatmap(pig_mouse_comp_ADLT2_comparison_core_ps_trans, sample.label = 'Species', sample.order = 'Species')

## Using variance stabilized count table
pig_mouse_comp_ADLT2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_ADLT2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
ADLT2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_ADLT2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_ADLT2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_ADLT2_comparison_ps)))

ADLT2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
ADLT2_vst_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_vst_ps)
ADLT2_vst_core_ps
plot_heatmap(ADLT2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(ADLT2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```


**Elderly Donor Analysis**

```{r, echo = TRUE}
## MAKING RAREFCATION PLOTS
merged_pig_mouse_comp_ps_no_neg_no_cecal <- subset_samples(merged_pig_mouse_comp_ps_no_neg, SampleType != 'cecal')
merged_pig_mouse_comp_ELD_ps <- subset_samples(merged_pig_mouse_comp_ps_no_neg_no_cecal, Donor=="ELD2")
merged_pig_mouse_comp_ELD_ps <- subset_samples(merged_pig_mouse_comp_ELD_ps, Actual_inoculum != 'no'|is.na(Actual_inoculum))
merged_pig_mouse_comp_ELD_ps <- subset_samples(merged_pig_mouse_comp_ELD_ps, Time_point != "at_euthanization")
merged_pig_mouse_comp_ELD_ps

## export as .txt files
write.table(t(otu_table(merged_pig_mouse_comp_ELD_ps)), "ELD2_otu_table.txt", sep = "\t", row.names = TRUE, col.names=NA, quote = FALSE)
write.table(sample_data(merged_pig_mouse_comp_ELD_ps), "ELD2_sample_data.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
```

```
## Using QIIME2 to make rarefaction curves

# Export feature table and converting it into a biom file :

(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ echo -n "#OTU Table" | cat - ELD2_otu_table.txt > ELD2_otu_biom_table.txt
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ biom convert -i ELD2_otu_biom_table.txt -o ELD2_otu_biom_table.biom --table-type="OTU table" --to-hdf5

# importing biom table to qiime2 :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime tools import --input-path ELD2_otu_biom_table.biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path ELD2_otu_biom_table.biom.qza

# Use feature table summary to find out what the medium read depth is (for use in --p-max-depth of following command) :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime feature-table summarize --i-table ELD2_otu_biom_table.biom.qza --o-visualization ELD_table.qzv --m-sample-metadata-file ELD2_sample_data.txt

# Performing alpha rarefactions :
(qiime2-2019.1) Niroshs-MacBook-Pro:pig_mouse_comparison_study niroshaluthge$ qiime diversity alpha-rarefaction \
> --i-table ELD2_otu_biom_table.biom.qza \
> --i-phylogeny ASV_rep_set.phylip.qza \
> --p-max-depth 37000 \
> --m-metadata-file ELD2_sample_data.txt \
> --o-visualization ELD2-alpha-rarefaction.qzv
``` 

Separating out the relevant ADULT2 samples :

```{r, echo=TRUE}
pig_mouse_comp_ELD2_master_ps <- subset_samples(merged_pig_mouse_comp_complete_ps, Donor == 'ELD2')
pig_mouse_comp_ELD2_master_ps

# Let's see how the overall beta-diversity compare between the human donor inocula and the HMA animalfecal bacterial communities
pig_mouse_comp_ELD2_comparison_ps <- subset_samples(pig_mouse_comp_ELD2_master_ps, Actual_inoculum!= 'no' | is.na(Actual_inoculum)) 
pig_mouse_comp_ELD2_comparison_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, SampleType != 'cecal')
pig_mouse_comp_ELD2_comparison_ps

# Transform count table to relative abundance table
rel_abund_ELD2_ps <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x){x / sum(x)})
rel_abund_ELD2_ps

# plotting PCoA using unweighted unifrac distances - ELD2
unw_ord_ELD2 <- ordinate(rel_abund_ELD2_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ELD2 <- plot_ordination(rel_abund_ELD2_ps, unw_ord_ELD2, color = "Species") 
unweighted_pcoa_ELD2

# plotting PCoA using weighted unifrac distances - ELD2
w_ord_ELD2 <- ordinate(rel_abund_ELD2_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_ELD2 <- plot_ordination(rel_abund_ELD2_ps, w_ord_ELD2, color = "Species") 
weighted_pcoa_ELD2

# plotting using binary jaccard distances
bin_jac_ord_ELD2 <- ordinate(rel_abund_ELD2_ps, method = "PCoA", distance = "jaccard", binary = TRUE)
binary_jaccard_pcoa_ELD2 <- plot_ordination(rel_abund_ELD2_ps, bin_jac_ord_ELD2, color = "Species") 
binary_jaccard_pcoa_ELD2

# plotting using Bray-Curtis distances
bray_ord_ELD2 <- ordinate(rel_abund_ELD2_ps, method="PCoA", distance = "bray")
bray_pcoa_ELD2 <- plot_ordination(rel_abund_ELD2_ps, bray_ord_ELD2, color="Species")
bray_pcoa_ELD2

## Let's also color the weighted and unweighted unifrac plots by sampling time point
unwunifrac_ELD2_by_tp <- plot_ordination(rel_abund_ELD2_ps, unw_ord_ELD2, color = "Time_point", shape = "Species") + ggtitle("Senior donor PCoA plot - unweighted unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_ELD2_by_tp

wunifrac_ELD2_by_tp <- plot_ordination(rel_abund_ELD2_ps, w_ord_ELD2, color = "Time_point", shape = "Species") + ggtitle("Senior donor PCoA plot - Weighted unifrac") + theme(plot.title = element_text(hjust = 0.5))
wunifrac_ELD2_by_tp

### performing PERMANOVA analyses
rel_abund_ELD2_ps
adonis(phyloseq::distance(rel_abund_ELD2_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_ELD2_ps), "data.frame"), permutations = 999)

results_ELD2_unweighted <- pairwise_permanova(rel_abund_ELD2_ps,"Species","Time_point","unifrac") 
print(results_ELD2_unweighted)
```

Let's see how the donor ELD2 samples look like

```{r, echo=TRUE}
human_donors_ELD2_ps <- subset_samples(pig_mouse_comp_ELD2_master_ps, Species == 'human')
human_donors_ELD2_ps
write.table(sample_data(human_donors_ELD2_ps), "ELD2_donors.txt", sep = "\t", row.names = FALSE, col.names = TRUE, quote = FALSE)
human_donors_ELD2_inocula_ps <- subset_samples(human_donors_ELD2_ps, Actual_inoculum == 'yes')
human_donors_ELD2_inocula_ps
```

Defining a 'core' ELD2 donor microbiota :

```{r, echo=TRUE}
# Let's look at ASVs that are found in all the samples used as inocula and thus define a 'core' ELD2 donor microbiota
prevdf_ELD2 = apply(X = otu_table(human_donors_ELD2_inocula_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ELD2_inocula_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_ELD2 = data.frame(Prevalence=prevdf_ELD2, TotalAbundance=taxa_sums(human_donors_ELD2_inocula_ps))
View(prevdf_ELD2)

core_threshold = nsamples(human_donors_ELD2_inocula_ps) # set threshold for the number of samples in which the ASV needs to be present in order for it to be considered as part of the core
core_ELD2_taxa <- rownames(prevdf_ELD2)[prevdf_ELD2$Prevalence >= core_threshold]
human_donors_ELD2_inocula_core_ps <- prune_taxa(core_ELD2_taxa, human_donors_ELD2_inocula_ps)
human_donors_ELD2_inocula_core_ps # 134 core ASVs identified
sum(otu_table(human_donors_ELD2_inocula_core_ps))/sum(otu_table(human_donors_ELD2_inocula_ps)) # the 134 core ASVs account for ~ 96.04 % of total ADLT2 donor reads

# making box plots for donor samples
human_donors_ELD2_inocula_ps_trans <- transform_sample_counts(human_donors_ELD2_inocula_ps, function(x) x/sum(x))
human_donors_ELD2_inocula_core_ps_trans <- prune_taxa(core_ELD2_taxa, human_donors_ELD2_inocula_ps_trans)
theme_set(theme_bw())
p1 <- plot_bar(human_donors_ELD2_inocula_core_ps_trans, fill = "Genus")
p2 <- p1 + theme (legend.text = element_text(size = 6), legend.title = element_text(size = 8))
p3 <- p2 + theme(legend.key.height = unit(0.5, "cm"), legend.key.width = unit(0.2, "cm"))
p4 <- p3 + theme(legend.position = 'none')
p4
```

Let's have a look at how the humanized animal model fecal bacterial communities cluster relative to the core human donor inocula communities for ELD2 donor. 
```{r, echo=TRUE}
rel_abund_ELD2_ps
rel_abund_ELD2_core_ps <- prune_taxa(core_ELD2_taxa, rel_abund_ELD2_ps)
rel_abund_ELD2_core_ps 

# plotting PCoA using unweighted unifrac distances for core taxa - ELD2
unw_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "unifrac", weighted = F)
unweighted_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, unw_ord_ELD2_core, color = "Species", shape = "Species") + ggtitle("SNR donor PCoA plot - Unweighted Unifrac") + theme(plot.title = element_text(hjust = 0.5, size = 10))
unweighted_pcoa_ELD2_core

# plotting PCoA using weighted unifrac distances for core taxa - ELD2
w_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "wunifrac")
weighted_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, w_ord_ELD2_core, color = "Species") 
weighted_pcoa_ELD2_core

# plotting PCoA using Jaccard distances for core taxa - ELD2
bin_jac_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "jaccard", binary = T)
jaccard_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, bin_jac_ord_ELD2_core, color = "Species") 
jaccard_pcoa_ELD2_core

# plotting PCoA using Bray-Curtis distances for core taxa - ELD2
bray_ord_ELD2_core <- ordinate(rel_abund_ELD2_core_ps, method = "PCoA", distance = "bray")
bray_pcoa_ELD2_core <- plot_ordination(rel_abund_ELD2_core_ps, bray_ord_ELD2_core, color = "Species", shape = "Species") + ggtitle("SNR donor PCoA plot - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5, size = 10))
bray_pcoa_ELD2_core

ELD2_core_unifrac_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ELD2_core, weighted_pcoa_ELD2_core)
ELD2_core_unifrac_comb

ELD2_core_jaccard_bray_comb <- grid.arrange(ncol = 2, jaccard_pcoa_ELD2_core, bray_pcoa_ELD2_core)
ELD2_core_jaccard_bray_comb

ELD2_core_unifrac_bray_comb <- grid.arrange(ncol = 2, unweighted_pcoa_ELD2_core, bray_pcoa_ELD2_core)
ELD2_core_unifrac_bray_comb

## Let's also color the unweighted unifrac and bray-curtis plots by sampling time point
unwunifrac_ELD2_core_by_tp <- plot_ordination(rel_abund_ELD2_core_ps, unw_ord_ELD2_core, color = "Time_point", shape = "Species") + ggtitle("Senior donor PCoA plot core ASVs - Unweighted Unifrac") + theme(plot.title = element_text(hjust = 0.5)) 
unwunifrac_ELD2_core_by_tp

bray_ELD2_by_tp <- plot_ordination(rel_abund_ELD2_core_ps, bray_ord_ELD2_core, color = "Time_point", shape = "Species") + ggtitle("Senior donor PCoA plot core ASVs - Bray-Curtis") + theme(plot.title = element_text(hjust = 0.5))
bray_ELD2_by_tp

ELD2_core_PCoA_by_tp_comb <- grid.arrange(ncol = 2, unwunifrac_ELD2_core_by_tp, bray_ELD2_by_tp)
ELD2_core_PCoA_by_tp_comb
```

PERMANOVA analysis to check if clustering patterns are statistically siginificantly different

```{r, echo=TRUE}
### performing PERMANOVA analyses on unweighted unifrac
rel_abund_ELD2_core_ps
metadata <- as(sample_data(rel_abund_ELD2_core_ps), "data.frame")
View(sample_data(rel_abund_ELD2_core_ps))
adonis(phyloseq::distance(rel_abund_ELD2_core_ps, method = "unifrac", weighted = F) ~ Species + Time_point, data=as(sample_data(rel_abund_ELD2_core_ps), "data.frame"), permutations = 999)

## Carrying out pair-wise comparisons
results_ELD2_unweighted <- pairwise_permanova(rel_abund_ELD2_core_ps,"Species","Time_point","unifrac") 
print(results_ELD2_unweighted)
```


Hierarchical clustering based on Bray-Curtis distances

```{r, echo=TRUE}

# Use the relative abundance table
rel_abund_ELD2_core_ps
bray_dist_ELD2 <- phyloseq::distance(rel_abund_ELD2_core_ps, method = "bray")

# performing hierarchical clustering and plotting
bray_clust_ELD2 <- hclust(bray_dist_ELD2, method = "ward.D2")
plot(bray_clust_ELD2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_ELD2 <- read.table("mapping_files_for_dendrograms/ELD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_ELD2)
mapping_file_dendro_ELD2$Color_Sp <- as.character(mapping_file_dendro_ELD2$Color_Sp)

# making dendrogram
bray_dend_ELD2 <- as.dendrogram(bray_clust_ELD2, hang=0.1)
SampleIDs_dend_ELD2 <- labels(bray_dend_ELD2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ELD2

sorted_mapping_ELD2 <- mapping_file_dendro_ELD2[match(SampleIDs_dend_ELD2, mapping_file_dendro_ELD2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_ELD2)

# color dendrogram labels 
labels_colors(bray_dend_ELD2) <- sorted_mapping_ELD2$Color_Sp
labels_colors(bray_dend_ELD2) # check if sample IDs and colors correspond

# change labels to reflect time point, change label size and plot dendrogram
labels(bray_dend_ELD2) <- sorted_mapping_ELD2$Time_point
bray_dend_ELD2 <- set(bray_dend_ELD2, "labels_cex", 0.5)
plot(bray_dend_ELD2, ylab="Bray-Curtis distances")
```

Let's make a Heatmap based on hierarchical clustering :

```{r, echo=TRUE}
asv_tab <- otu_table(ELD2_vst_core_ps)
asv_tab_df <- as.data.frame(asv_tab)
dim(asv_tab_df)

input <- as.matrix(asv_tab_df)
dim(input)


col_labels <- labels(bray_dend_ELD2)
#View(col_labels)
col_labels <- col_labels[order(order.dendrogram(bray_dend_ELD2))]
#View(col_labels)

col_cols <- labels_colors(bray_dend_ELD2)
#View(col_cols)
col_cols <- col_cols[order(order.dendrogram(bray_dend_ELD2))]
#View(col_cols)

#pdf("test_heatmap_plot.pdf")

par(mgp=c(0.5,0.4,0))

heatmap.2(input, col = colorpanel(100,low = "#000033", high = "#66B2FF"), cexRow=0.5, cexCol=0.65, scale="none", Colv = bray_dend_ELD2, trace = "none", density = "none", breaks = seq(from=min(range(input)), to=max(range(input)), length.out = 101), symm = F, symkey = F, symbreaks = T, labCol = col_labels, colCol = col_cols, ColSideColors = col_cols, margins = c(7.5,4), xlab = "inocula and fecal samples", ylab = "core SNR donor ASVs", dendrogram = "column", key = T) 

legend(0.95, 1.13, xpd = TRUE, legend=c("donors","HMA mice", "HMA piglets"),fill=c('red','dark green', 'blue'),cex=0.5)

#dev.off()
```

ANCOM on ELD2 donor inocula to see if there are any differentially abundant ASVs between the donor inocula

```{r, echo=TRUE}
write.table (otu_table(human_donors_ELD2_inocula_core_ps), 'ANCOM_ELD2_inocula_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
# type echo -n "Sample.ID" | cat - ANCOM_INF2_inocula.txt > ANCOM_INF2_inocula_INPUT.txt in a terminal to add required header to 'ANCOM_CHLD2_inocula_INPUT.txt'
write.table(sample_data(human_donors_ELD2_inocula_core_ps), 'ANCOM_ELD2_inocula_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

# running ANCOM 
otu_table <- read.table("ANCOM_ELD2_inocula_INPUT.txt", sep = "\t", header = TRUE)
#View(otu_table)

map_file <- read.table("ANCOM_ELD2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE)
#View(map_file)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Which_animal%in%c("pig", "mouse"), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Which_animal",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)

comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ELD2_human_inocula_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F) # None of the ASVs are significantly different
```

Comparing donor inoculae relative abundances using DESeq2 :

```{r,echo=TRUE}
# DESeq2 conversion and call
# Install DESeq2 if not already installed
# load DESeq2
library('DESeq2'); packageVersion('DESeq2')

# important mapping file with relevant donor info
map_file <- read.table("ANCOM_ELD2_inocula_comp_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(map_file) <- as.character(map_file[,1])

# merge into a new phyloseq object
DESeq2_ELD2_donors_ps <- phyloseq(otu_table(human_donors_ELD2_inocula_core_ps), taxa_are_rows = FALSE, sample_data(map_file), tax_table(human_donors_ELD2_inocula_core_ps))
DESeq2_ELD2_donors_ps

diagadds = phyloseq_to_deseq2(DESeq2_ELD2_donors_ps, ~Which_animal)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(DESeq2_ELD2_donors_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
View(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ELD2_donors.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 

# None of the core ASVs were sig. different in their relative abundances between the inoculae
```

Now let's do the ANCOM comparisons for the establishment of core donor ASVs between the donor samples and the respective HMA animal models :

```{r, echo=TRUE}
# ADLT2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_core_ps, Species != 'mouse')
ANCOM_donor_vs_HMA_piglets_ps
write.table (otu_table(ANCOM_donor_vs_HMA_piglets_ps), 'ANCOM_ELD2_donor_vs_HMA_piglets_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_piglets_ps)
write.table(map_file, 'ANCOM_ELD2_inocula_HMA_piglets_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_ELD2_donor_vs_HMA_piglets_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_ELD2_inocula_HMA_piglets_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','piglet'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.90)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ELD2_human_inocula_vs_HMA_piglets_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 11 ASVs significantly different

# ADLT2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_core_ps, Species != 'piglet')
ANCOM_donor_vs_HMA_mice_ps
write.table (otu_table(ANCOM_donor_vs_HMA_mice_ps), 'ANCOM_ELD2_donor_vs_HMA_mice_INPUT.txt', sep = "\t", row.names = T, col.names = NA, quote = F)
map_file <- sample_data(ANCOM_donor_vs_HMA_mice_ps)
write.table(map_file, 'ANCOM_ELD2_inocula_HMA_mice_comp_mapping_file.txt', sep = "\t", row.names = F, col.names = T, quote = F)

otu_table <- read.table('ANCOM_ELD2_donor_vs_HMA_mice_INPUT.txt', sep = '\t', header = TRUE)
map_file <- read.table('ANCOM_ELD2_inocula_HMA_mice_comp_mapping_file.txt', sep = '\t', header = TRUE)

comparison_test=ANCOM.main(OTUdat=otu_table,Vardat=map_file[map_file$Species%in%c('human','mouse'), ],  adjusted=FALSE,
                                                                                       
                           repeated=F,
                           main.var="Species",
                           adj.formula=NULL,
                           repeat.var=NULL,
                           longitudinal=FALSE,
                           random.formula=NULL,
                           multcorr=2,
                           sig=0.05,
                           prev.cut=0.99) # had to increase to 0.99 so that all 134 ASVs are included in the analysis (if set to the default 0.9, only 49 ASVs are considered meaning that the other ASVs (85) are present in <= 10% of samples)
comparison_test$W.taxa

write.table (comparison_test$W.taxa,"ANCOM_output_ELD2_human_inocula_vs_HMA_mice_comparison.txt", sep = "\t", row.names = T, col.names = NA, quote = F)
# 33 ASVs significantly different 
```

Using DESeq2 to compare abundances between donor and HMAs

```{r, echo=TRUE}
# ADLT2 donor vs HMA piglets
ANCOM_donor_vs_HMA_piglets_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_piglets_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_piglets_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ELD2_donor_vs_piglets.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 97 ASVs were significantly differentially abundant


# ADLT2 donor vs HMA mice
ANCOM_donor_vs_HMA_mice_ps

# DESeq2 conversion and call
diagadds = phyloseq_to_deseq2(ANCOM_donor_vs_HMA_mice_ps, ~Species)
diagadds = DESeq(diagadds, test = 'Wald', fitType = "parametric") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
alpha = 0.05
sigtab <- res_DESeq2[which(res_DESeq2$padj < alpha), ]
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(ANCOM_donor_vs_HMA_mice_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab)
sigtab <- sigtab[, c(14,1:13)]
View(sigtab)
write.table(sigtab, 'DESeq2_results_ELD2_donor_vs_mice.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) # 9 ASVs were significantly differentially abundant
```

Visualizing differences in relative abundances of ASVs using a heatmap :

```{r, echo=TRUE}
# using raw counts
pig_mouse_comp_ELD2_comparison_core_ps
plot_heatmap(pig_mouse_comp_ELD2_comparison_core_ps, sample.label = 'Species', sample.order = 'Species')

# using relative abundances
pig_mouse_comp_ELD2_comparison_ps
pig_mouse_comp_ELD2_comparison_ps_trans <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_ELD2_comparison_ps_trans
pig_mouse_comp_ELD2_comparison_core_ps_trans <- prune_taxa(core_ELD2_taxa, pig_mouse_comp_ELD2_comparison_ps_trans)
pig_mouse_comp_ELD2_comparison_core_ps_trans

plot_heatmap(pig_mouse_comp_ELD2_comparison_core_ps_trans, sample.label = 'Species', sample.order = 'Species')

## Using variance stabilized count table
pig_mouse_comp_ELD2_comparison_ps
diagadds <- phyloseq_to_deseq2(pig_mouse_comp_ELD2_comparison_ps, ~Species)
diagadds = estimateSizeFactors(diagadds)
diagadds = estimateDispersions(diagadds)
diagvst = getVarianceStabilizedData(diagadds)
dim(diagvst)

# make new phyloseq object with variance stabilized count table as the ASV table
ELD2_vst_ps <- phyloseq(otu_table(diagvst, taxa_are_rows = TRUE), sample_data(sample_data(pig_mouse_comp_ELD2_comparison_ps)), tax_table(tax_table(pig_mouse_comp_ELD2_comparison_ps)), phy_tree(phy_tree(pig_mouse_comp_ELD2_comparison_ps)))

ELD2_vst_ps # Check to make sure that phyloseq object has been made properly

# retain only the core ASVs
ELD2_vst_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_vst_ps)
ELD2_vst_core_ps
plot_heatmap(ELD2_vst_core_ps, sample.label = "Species", sample.order = "Species")
plot_heatmap(ELD2_vst_core_ps, taxa.order = "Phylum", taxa.label = "Genus",  sample.label = "Species", sample.order = "Species")
```


##**Determining which taxa established and which taxa were not able to establish in the HMA animal models**##

**Infant donor**

Separating out the taxa that were found in at least one of the two samples used to inoculate the mice :

```{r, echo=TRUE}
samples_used_to_inoculate_mice <- c('R2D269', 'R2D273') 
human_donors_INF2_inocula_for_mice_ps <- subset_samples(human_donors_INF2_inocula_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_INF2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_INF2_mouse_donors <- sum(otu_table(human_donors_INF2_inocula_for_mice_ps))/2
avg_num_of_reads_INF2_mouse_donors # 32,561 reads/sample

# Determining taxa prevalence
prevdf_INF2_mice_inocula = apply(X = otu_table(human_donors_INF2_inocula_for_mice_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_INF2_inocula_for_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
prevdf_INF2_mice_inocula = data.frame(Prevalence=prevdf_INF2_mice_inocula, TotalAbundance=taxa_sums(human_donors_INF2_inocula_for_mice_ps))
View(prevdf_INF2_mice_inocula)

all_INF2_donor_taxa_mice <- rownames(prevdf_INF2_mice_inocula)[prevdf_INF2_mice_inocula$Prevalence >= 1]
all_INF2_donor_taxa_mice # 40 taxa were detected in at least one of the two donor inocula for mice. 
                         # This represents the 'pool' of donor ASVs that could have colonized the mice

# Let's make a taxonomy table for these ASVs
all_INF2_donor_taxa_mice_ps <- prune_taxa(all_INF2_donor_taxa_mice,human_donors_INF2_inocula_for_mice_ps)
all_INF2_donor_taxa_mice_ps
taxa_cols <- colnames(tax_table(all_INF2_donor_taxa_mice_ps))
taxa_table <- as.data.frame(tax_table(all_INF2_donor_taxa_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_INF2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_INF2_mice) # check to see that everything looks good
write.table(taxa_table_all_INF2_mice, "taxonomy_table_all_donor_INF2_ASVs_inoculae_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```
 
Let's see how many of these ASVs have established in the mice. In order to be considered 'persistently colonizing', an ASV needs to be detected in at least **4** samplings in at least **2/3 of animals** :

```{r, echo=TRUE}
INF2_HMA_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
INF2_HMA_mice_ps
avg_num_of_reads_INF2_mice <- sum(otu_table(INF2_HMA_mice_ps))/59
avg_num_of_reads_INF2_mice # 45,238 reads/sample

# Let's turn the making of a 'Prevalence' data frame into a function
makePrevDF <- function(phyloseqobject, prevthresh) {
  prevdf = apply(X = otu_table(phyloseqobject),
               MARGIN = ifelse(taxa_are_rows(phyloseqobject), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
  prevdf = data.frame(Prevalence=prevdf, TotalAbundance=taxa_sums(phyloseqobject))
  prev_threshold = prevthresh
  colonized_taxa <- rownames(prevdf)[prevdf$Prevalence >= prev_threshold]
  return(list(prevdf, colonized_taxa))
}

# Let's see how many total ASVs were found in at least one INF2 HMA mouse fecal sample 
prevdf_INF2_mice <- makePrevDF(INF2_HMA_mice_ps, 1)
prevdf_INF2_mice_table <- as.data.frame(prevdf_INF2_mice[1])
colonized_prevdf_INF2_mice <- prevdf_INF2_mice_table[prevdf_INF2_mice_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_mice) # 167 total ASVs found across INF2 mice

INF2_HMA_mice_donor_taxa_ps <- prune_taxa(all_INF2_donor_taxa_mice, INF2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
INF2_HMA_mice_donor_taxa_ps

# Calculating the percentage of the total HMA mouse reads is represented by these 40 donor ASVs
sum(otu_table(INF2_HMA_mice_donor_taxa_ps))/sum(otu_table(INF2_HMA_mice_ps)) # ~86% of reads originating from donor ASVs

# Checking how many of the 40 INF2 donor ASVs were detected in at least one of the mouse samples
prevdf_INF2_donor_taxa_detect_mice <- makePrevDF(INF2_HMA_mice_donor_taxa_ps, 1)
prevdf_INF2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_INF2_donor_taxa_detect_mice[1])
colonized_prevdf_INF2_donor_mice <- prevdf_INF2_donor_taxa_detect_mice_table[prevdf_INF2_donor_taxa_detect_mice_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_donor_mice)
all_INF2_taxa_colonizing_mice <-prevdf_INF2_donor_taxa_detect_mice[2]
all_INF2_taxa_colonizing_mice <- unlist(all_INF2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_INF2_taxa_colonizing_mice # 27 out of the possible 40 INF2 donor ASVs have colonized at least one mouse fecal sample

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF2_HMA_mice_ps_48HRS <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '48hrs_post_inoc')
INF2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_mice_ps_48HRS,5) # This time point as well as the 1 and 2week time points had 8 animals
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '5',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_mice_ps_1week <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '1week_post_inoc')
INF2_HMA_mice_ps_1week

prevdf_1week <- makePrevDF(INF2_HMA_mice_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '5',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_mice_ps_2weeks <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '2weeks_post_inoc')
INF2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_mice_ps_2weeks,5)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '5',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_mice_ps_3weeks <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '3weeks_post_inoc')
INF2_HMA_mice_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_mice_ps_3weeks,4)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '4',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_mice_ps_4weeks <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '4weeks_post_inoc')
INF2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_mice_ps_4weeks,4)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '4',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_mice_ps_5weeks <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '5weeks_post_inoc')
INF2_HMA_mice_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_mice_ps_5weeks,4)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '4',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_mice_ps_40days <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == 'Final_collection')
INF2_HMA_mice_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_mice_ps_40days,4)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '4',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
INF2_HMA_mice_ps_euthan <- subset_samples(INF2_HMA_mice_donor_taxa_ps, Description == '24h_before_euthanization')
INF2_HMA_mice_ps_euthan

prevdf_euthan <- makePrevDF(INF2_HMA_mice_ps_euthan,4)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '4',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_INF2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv, colonized_prevdf_euthan_relv))
str(merged_prev_df_INF2_mice) # All columns considered factors
merged_prev_df_INF2_mice[2:9] <- lapply(merged_prev_df_INF2_mice[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_mice)
View(merged_prev_df_INF2_mice)
     
merged_prev_df_INF2_mice[is.na(merged_prev_df_INF2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_mice$total <- rowSums(merged_prev_df_INF2_mice[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_mice)
merged_prev_df_INF2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_mice)
write.table(merged_prev_df_INF2_mice, "merged_prev_df_INF2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_INF2_mice <- merged_prev_df_INF2_mice[,1]
temporally_colonizing_taxa_INF2_mice

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_INF2_mice_df)
actually_colonizing_taxa_INF2_mice <- actually_colonizing_taxa_INF2_mice_df[,1]
actually_colonizing_taxa_INF2_mice # 22 ASVs persistently colonizing mice

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_mice, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_taxa_INF2_mice_df)
perst_colonizing_taxa_INF2_mice <- perst_colonizing_taxa_INF2_mice_df[,1]
perst_colonizing_taxa_INF2_mice # 19 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_mice, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_taxa_INF2_mice_df)
truly_colonizing_taxa_INF2_mice <- truly_colonizing_taxa_INF2_mice_df[,1]
truly_colonizing_taxa_INF2_mice # 18 ASVs 'truly' colonizing mice
```

Let's do a similar analysis for the HMA piglets inoculated with INF2 donor (`late' inoculation piglets)

```{r, echo=TRUE}

# Determining taxa found in the human donor sample used to inoculate the piglets
human_donors_INF2_sample_to_inoculate_pigs_ps <- subset_samples(human_donors_INF2_inocula_relevant_ps, SampleID == '323')
human_donors_INF2_sample_to_inoculate_pigs_ps
sum(otu_table(human_donors_INF2_sample_to_inoculate_pigs_ps)) # 24,455 reads

prevdf_INF2_donor_pigs <- makePrevDF(human_donors_INF2_sample_to_inoculate_pigs_ps, 1)
prevdf_INF2_donor_pigs_table <- as.data.frame(prevdf_INF2_donor_pigs[1])
colonized_prevdf_INF2_donor_pigs <- prevdf_INF2_donor_pigs_table[prevdf_INF2_donor_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_donor_pigs)
all_taxa_in_donor_INF2_pigs <- prevdf_INF2_donor_pigs[2]
all_taxa_in_donor_INF2_pigs <- unlist(all_taxa_in_donor_INF2_pigs) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_INF2_pigs # the INF2 donor inoculum used to inoculate piglets had 31 ASVs

# Let's make a taxonomy table for these ASVs
all_INF2_donor_taxa_piglets_ps <- prune_taxa(all_taxa_in_donor_INF2_pigs, human_donors_INF2_sample_to_inoculate_pigs_ps)
all_INF2_donor_taxa_piglets_ps
taxa_cols <- colnames(tax_table(all_INF2_donor_taxa_piglets_ps))
taxa_table <- as.data.frame(tax_table(all_INF2_donor_taxa_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_INF2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_INF2_piglets) # check to see that everything looks good
write.table(taxa_table_all_INF2_piglets, "taxonomy_table_all_donor_INF2_ASVs_inoculum_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


# Let's only keep those ASVs which came from the donor for the HMA piglet fecal samples
pig_mouse_comp_INF2_comparison_ps
INF2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
INF2_HMA_piglets_ps <- subset_samples(INF2_HMA_piglets_ps, Timingofinocu != "early" & SampleType != "cecal") # removing early inoculation and cecal samples
INF2_HMA_piglets_ps 

avg_num_of_reads_INF2_piglets <- sum(otu_table(INF2_HMA_piglets_ps))/25 
avg_num_of_reads_INF2_piglets # 37,341 reads/sample

# Let's see how many total ASVs were found in at least one INF2 HMA piglet fecal sample 
prevdf_INF2_pigs <- makePrevDF(INF2_HMA_piglets_ps, 1)
prevdf_INF2_pigs_table <- as.data.frame(prevdf_INF2_pigs[1])
colonized_prevdf_INF2_pigs <- prevdf_INF2_pigs_table[prevdf_INF2_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_pigs) # 439 total ASVs found across INF2 piglets

INF2_HMA_piglets_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_INF2_pigs, INF2_HMA_piglets_ps) # retaining only the ASVs that were found in the donor inoculae
INF2_HMA_piglets_donor_taxa_ps 

# Calculating the percentage of the total HMA piglet reads is represented by these 31 donor ASVs
sum(otu_table(INF2_HMA_piglets_donor_taxa_ps))/sum(otu_table(INF2_HMA_piglets_ps)) # 37.4% of reads originating from donor ASVs
```

Let's see how many of the possible 31 INF2 donor ASVs are detected in at least one pig fecal sample :

```{r, echo=TRUE}
INF2_HMA_piglets_donor_taxa_ps
prevdf_INF2_donor_taxa_detect_pigs <- makePrevDF(INF2_HMA_piglets_donor_taxa_ps, 1)
prevdf_INF2_donor_taxa_detect_pigs_table <- as.data.frame(prevdf_INF2_donor_taxa_detect_pigs[1])
View(prevdf_INF2_donor_taxa_detect_pigs_table)
colonized_prevdf_INF2_donor_pigs <- prevdf_INF2_donor_taxa_detect_pigs_table[prevdf_INF2_donor_taxa_detect_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_donor_pigs)
all_INF2_taxa_colonizing_pigs <-prevdf_INF2_donor_taxa_detect_pigs[2]
all_INF2_taxa_colonizing_pigs <- unlist(all_INF2_taxa_colonizing_pigs) # need to 'unlist' in order to use in the 'prune_taxa' function below 
all_INF2_taxa_colonizing_pigs # 23 of the possible 31 INF2 donor ASVs were detected in at least one piglet fecal sample
```

Determining the colonization of INF2 donor ASVs in piglets by sampling time point :

```{r, echo=TRUE}

# Prevalence of ASVs at 48 HRS
INF2_HMA_piglets_donor_taxa_ps
INF2_HMA_piglets_ps_48HRS <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '48hrs_post_inoc')
INF2_HMA_piglets_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_piglets_ps_48HRS,3) # there were 4 samples for this TP while for the rest of the TPs it is 3 samples
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '3',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_piglets_ps_1week <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '1week_post_inoc')
INF2_HMA_piglets_ps_1week

prevdf_1week <- makePrevDF(INF2_HMA_piglets_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_piglets_ps_2weeks <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '2weeks_post_inoc')
INF2_HMA_piglets_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_piglets_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_piglets_ps_3weeks <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '3weeks_post_inoc')
INF2_HMA_piglets_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_piglets_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_piglets_ps_4weeks <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '4weeks_post_inoc')
INF2_HMA_piglets_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_piglets_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_piglets_ps_5weeks <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == '5weeks_post_inoc')
INF2_HMA_piglets_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_piglets_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_piglets_ps_40days <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == 'Final_collection')
INF2_HMA_piglets_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_piglets_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
INF2_HMA_piglets_ps_euthan <- subset_samples(INF2_HMA_piglets_donor_taxa_ps, Description == 'at_euthanization')
INF2_HMA_piglets_ps_euthan

prevdf_euthan <- makePrevDF(INF2_HMA_piglets_ps_euthan,2)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '2',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)


# Merging all dataframes together
merged_prev_df_INF2_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv,colonized_prevdf_1week_relv,colonized_prevdf_2weeks_relv,colonized_prevdf_3weeks_relv,colonized_prevdf_4weeks_relv,colonized_prevdf_5weeks_relv,colonized_prevdf_40days_relv,colonized_prevdf_euthan_relv))
str(merged_prev_df_INF2_piglets) # All columns considered factors
merged_prev_df_INF2_piglets[2:9] <- lapply(merged_prev_df_INF2_piglets[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_piglets)
View(merged_prev_df_INF2_piglets)
     
merged_prev_df_INF2_piglets[is.na(merged_prev_df_INF2_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_piglets$total <- rowSums(merged_prev_df_INF2_piglets[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_piglets)
merged_prev_df_INF2_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_piglets)
write.table(merged_prev_df_INF2_piglets, "merged_prev_df_INF2_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_INF2_piglets <- merged_prev_df_INF2_piglets[,1]
temporally_colonizing_taxa_INF2_piglets

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_INF2_piglets_df)
actually_colonizing_taxa_INF2_piglets <- actually_colonizing_taxa_INF2_piglets_df[,1]
actually_colonizing_taxa_INF2_piglets # 20 ASVs persistently colonizing pigs

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_piglets, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_taxa_INF2_piglets_df)
perst_colonizing_taxa_INF2_piglets <- perst_colonizing_taxa_INF2_piglets_df[,1]
perst_colonizing_taxa_INF2_piglets # 17 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_piglets, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_INF2_piglets_df)
truly_colonizing_taxa_INF2_piglets <- truly_colonizing_taxa_INF2_piglets_df[,1]
truly_colonizing_taxa_INF2_piglets # 17 ASVs 'truly' colonizing pigs

# ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for INF2 HMA piglets
actually_colonizing_taxa_INF2_piglets_ps <- prune_taxa(actually_colonizing_taxa_INF2_piglets,INF2_HMA_piglets_ps)
actually_colonizing_taxa_INF2_piglets_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_INF2_piglets_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_INF2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_coloniz_piglets, "taxonomy_table_for_INF2_ASVs_actually_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for INF2 HMA mice
actually_colonizing_taxa_INF2_mice_ps <- prune_taxa(actually_colonizing_taxa_INF2_mice,INF2_HMA_mice_ps)
actually_colonizing_taxa_INF2_mice_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_INF2_mice_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_INF2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_coloniz_mice, "taxonomy_table_for_INF2_ASVs_actually_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```

**Child donor**

Separating out the taxa that were found in at least one of the two samples used to inoculate the mice :

```{r, echo=TRUE}
samples_used_to_inoculate_mice <- c('R2D270', 'R2D274') 
human_donors_CHLD2_inocula_for_mice_ps <- subset_samples(human_donors_CHLD2_inocula_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_CHLD2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_CHLD2_mouse_donors <- sum(otu_table(human_donors_CHLD2_inocula_for_mice_ps))/2
avg_num_of_reads_CHLD2_mouse_donors # 32,941 reads/sample

# Determining taxa prevalence
prevdf_CHLD2_donor_taxa_mice <- makePrevDF(human_donors_CHLD2_inocula_for_mice_ps, 1)
prevdf_CHLD2_donor_taxa_mice_table <- as.data.frame(prevdf_CHLD2_donor_taxa_mice[1])
View(prevdf_CHLD2_donor_taxa_mice_table)
all_taxa_in_donor_CHLD2_mice <- prevdf_CHLD2_donor_taxa_mice[2]
all_taxa_in_donor_CHLD2_mice <- unlist(all_taxa_in_donor_CHLD2_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_CHLD2_mice # the CHLD2 donor inoculae used to inoculate mice had 129 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_CHLD2_donor_taxa_mice_ps <- prune_taxa(all_taxa_in_donor_CHLD2_mice, human_donors_CHLD2_inocula_for_mice_ps)
all_CHLD2_donor_taxa_mice_ps
taxa_cols <- colnames(tax_table(all_CHLD2_donor_taxa_mice_ps))
taxa_table <- as.data.frame(tax_table(all_CHLD2_donor_taxa_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_CHLD2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_CHLD2_mice) # check to see that everything looks good
write.table(taxa_table_all_CHLD2_mice, "taxonomy_table_all_donor_CHLD2_ASVs_inoculae_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

Let's see how many of these ASVs have established in the mice. In order to be considered 'established', an ASV needs to be detected in at least **3** samplings in at least **2/3 of animals** :

```{r, echo=TRUE}
CHLD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_master_ps, Species == 'mouse') # keeping just the mouse samples
CHLD2_HMA_mice_ps
avg_num_of_reads_CHLD2_mice <- sum(otu_table(CHLD2_HMA_mice_ps))/80
avg_num_of_reads_CHLD2_mice # 47814 reads/sample

# Let's see how many total ASVs were found in at least one CHLD2 HMA mouse fecal sample 
prevdf_CHLD2_mice <- makePrevDF(CHLD2_HMA_mice_ps, 1)
prevdf_CHLD2_mice_table <- as.data.frame(prevdf_CHLD2_mice[1])
colonized_prevdf_CHLD2_mice <- prevdf_CHLD2_mice_table[prevdf_CHLD2_mice_table$Prevalence >= '1',]
View(colonized_prevdf_CHLD2_mice) # 290 total ASVs found across CHLD2 mice

CHLD2_HMA_mice_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_CHLD2_mice, CHLD2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
CHLD2_HMA_mice_donor_taxa_ps 

# Calculating the percentage of the total HMA mouse reads that is represented by these 129 donor ASVs
sum(otu_table(CHLD2_HMA_mice_donor_taxa_ps))/sum(otu_table(CHLD2_HMA_mice_ps)) # 79% of reads originating from donor ASVs

# Checking how many of the 129 CHLD2 donor ASVs were detected in at least one of the mouse samples
prevdf_CHLD2_donor_taxa_detect_mice <- makePrevDF(CHLD2_HMA_mice_donor_taxa_ps, 1)
prevdf_CHLD2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_CHLD2_donor_taxa_detect_mice[1])
View(prevdf_CHLD2_donor_taxa_detect_mice_table)
all_CHLD2_taxa_colonizing_mice <-prevdf_CHLD2_donor_taxa_detect_mice[2]
all_CHLD2_taxa_colonizing_mice <- unlist(all_CHLD2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_CHLD2_taxa_colonizing_mice # 56 out of the possible 129 CHLD2 donor ASVs have colonized at least one mouse fecal sample
                               # It's actually these 56 ASVs that are making up the 79% of the total reads contributed by the donor


# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD2_HMA_mice_ps_48HRS <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD2_HMA_mice_ps_48HRS,6)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_mice_ps_1week <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '1week_post_inoc')
CHLD2_HMA_mice_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '6',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_mice_ps_2weeks <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_mice_ps_3weeks <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_mice_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_mice_ps_3weeks,3)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '3',]
View(colonized_prevdf_3weeks)
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_mice_ps_4weeks <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_mice_ps_5weeks <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_mice_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_mice_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
View(colonized_prevdf_5weeks)
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_mice_ps_40days <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == 'Final_collection')
CHLD2_HMA_mice_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_mice_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
CHLD2_HMA_mice_ps_euthan <- subset_samples(CHLD2_HMA_mice_donor_taxa_ps, Description == '24h_before_euthanization')
CHLD2_HMA_mice_ps_euthan

prevdf_euthan <- makePrevDF(CHLD2_HMA_mice_ps_euthan,6)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '6',]
View(colonized_prevdf_euthan)
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_CHLD2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv, colonized_prevdf_euthan_relv))
str(merged_prev_df_CHLD2_mice) # All columns considered factors
merged_prev_df_CHLD2_mice[2:9] <- lapply(merged_prev_df_CHLD2_mice[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_mice)
View(merged_prev_df_CHLD2_mice)
     
merged_prev_df_CHLD2_mice[is.na(merged_prev_df_CHLD2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_mice$total <- rowSums(merged_prev_df_CHLD2_mice[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_mice)
merged_prev_df_CHLD2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD2_mice)
write.table(merged_prev_df_CHLD2_mice, "merged_prev_df_CHLD2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_CHLD2_mice <- merged_prev_df_CHLD2_mice[,1]
temporally_colonizing_taxa_CHLD2_mice # 36 ASVs temporarily colonize mice

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_CHLD2_mice_df)
actually_colonizing_taxa_CHLD2_mice <- actually_colonizing_taxa_CHLD2_mice_df[,1]
actually_colonizing_taxa_CHLD2_mice # 14 ASVs persistently colonizing mice

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points (50% of time points)
persistently_colonizing_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_CHLD2_mice_df)
persistently_colonizing_taxa_CHLD2_mice <- persistently_colonizing_taxa_CHLD2_mice_df[,1]
persistently_colonizing_taxa_CHLD2_mice # 6 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_mice, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_CHLD2_mice_df)
truly_colonizing_taxa_CHLD2_mice <- truly_colonizing_taxa_CHLD2_mice_df[,1]
truly_colonizing_taxa_CHLD2_mice # 3 ASVs 'truly' colonizing mice
```

Performing same analysis for CHLD2 HMA piglets

Separating out the taxa that were found in at least one of the two samples used to inoculate the piglets :

```{r, echo=TRUE}
samples_used_to_inoculate_piglets <- c('360', '371') 
human_donors_CHLD2_inocula_for_piglets_ps <- subset_samples(human_donors_CHLD2_inocula_ps, SampleID %in% samples_used_to_inoculate_piglets)
human_donors_CHLD2_inocula_for_piglets_ps # only retains the inocula used for piglets
avg_number_of_reads <- sum(otu_table(human_donors_CHLD2_inocula_for_piglets_ps))/2
avg_number_of_reads # 47,582 reads/sample

# Determining taxa prevalence
prevdf_CHLD2_donor_taxa_piglets <- makePrevDF(human_donors_CHLD2_inocula_for_piglets_ps, 1)
prevdf_CHLD2_donor_taxa_piglets_table <- as.data.frame(prevdf_CHLD2_donor_taxa_piglets[1])
View(prevdf_CHLD2_donor_taxa_piglets_table)
all_taxa_in_donor_CHLD2_piglets <- prevdf_CHLD2_donor_taxa_piglets[2]
all_taxa_in_donor_CHLD2_piglets <- unlist(all_taxa_in_donor_CHLD2_piglets) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_CHLD2_piglets # the CHLD2 donor inoculae used to inoculate piglets had 157 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_CHLD2_donor_taxa_piglets_ps <- prune_taxa(all_taxa_in_donor_CHLD2_piglets, human_donors_CHLD2_inocula_for_piglets_ps)
all_CHLD2_donor_taxa_piglets_ps
taxa_cols <- colnames(tax_table(all_CHLD2_donor_taxa_piglets_ps))
taxa_table <- as.data.frame(tax_table(all_CHLD2_donor_taxa_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_CHLD2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_CHLD2_piglets) # check to see that everything looks good
write.table(taxa_table_all_CHLD2_piglets, "taxonomy_table_all_donor_CHLD2_ASVs_inoculae_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# Let's only keep those ASVs which came from the donor for the HMA piglet fecal samples
pig_mouse_comp_CHLD2_comparison_ps
CHLD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
CHLD2_HMA_piglets_ps
avg_num_of_reads_CHLD2_piglets <- sum(otu_table(CHLD2_HMA_piglets_ps))/24 # 44,321 reads/sample
avg_num_of_reads_CHLD2_piglets

# Let's see how many total ASVs were found in at least one CHLD2 HMA piglet fecal sample
prevdf_CHLD2_pigs <- makePrevDF(CHLD2_HMA_piglets_ps, 1)
prevdf_CHLD2_pigs_table <- as.data.frame(prevdf_CHLD2_pigs[1])
colonized_prevdf_CHLD2_pigs <- prevdf_CHLD2_pigs_table[prevdf_CHLD2_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_CHLD2_pigs) # 226 total ASVs found across CHLD2 piglets

CHLD2_HMA_piglets_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_CHLD2_piglets, CHLD2_HMA_piglets_ps) # retaining only the ASVs that were found in the donor inoculae
CHLD2_HMA_piglets_donor_taxa_ps 

# Calculating the percentage of the total HMA mouse reads that is represented by these 157 donor ASVs
sum(otu_table(CHLD2_HMA_piglets_donor_taxa_ps))/sum(otu_table(CHLD2_HMA_piglets_ps)) # 81% of reads originating from donor ASVs
```

Let's see how many of the possible 157 CHLD2 donor ASVs are detected in at least one pig fecal sample :

```{r, echo=TRUE}
prevdf_CHLD2_donor_taxa_detect_pigs <- makePrevDF(CHLD2_HMA_piglets_donor_taxa_ps, 1)
prevdf_CHLD2_donor_taxa_detect_pigs_table <- as.data.frame(prevdf_CHLD2_donor_taxa_detect_pigs[1])
View(prevdf_CHLD2_donor_taxa_detect_pigs_table)
all_CHLD2_taxa_colonizing_pigs <-prevdf_CHLD2_donor_taxa_detect_pigs[2]
all_CHLD2_taxa_colonizing_pigs <- unlist(all_CHLD2_taxa_colonizing_pigs) # need to 'unlist' in order to use in the 'prune_taxa' function below 
all_CHLD2_taxa_colonizing_pigs # 113 of the possible 157 CHLD2 donor ASVs were detected in at least one piglet fecal sample
```

Determining the colonization of CHLD2 donor ASVs in piglets by sampling time point :

```{r, echo=TRUE}

# Prevalence of ASVs at 48 HRS
CHLD2_HMA_piglets_ps_48HRS <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_piglets_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD2_HMA_piglets_ps_48HRS,2)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_piglets_ps_1week <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '1week_post_inoc')
CHLD2_HMA_piglets_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_piglets_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_piglets_ps_2weeks <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_piglets_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_piglets_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_piglets_ps_3weeks <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_piglets_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_piglets_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_piglets_ps_4weeks <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_piglets_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_piglets_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_piglets_ps_5weeks <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_piglets_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_piglets_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_piglets_ps_40days <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == 'Final_collection')
CHLD2_HMA_piglets_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_piglets_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
CHLD2_HMA_piglets_ps_euthan <- subset_samples(CHLD2_HMA_piglets_donor_taxa_ps, Description == 'at_euthanization')
CHLD2_HMA_piglets_ps_euthan

prevdf_euthan <- makePrevDF(CHLD2_HMA_piglets_ps_euthan,2)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '2',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_CHLD2_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv,colonized_prevdf_1week_relv,colonized_prevdf_2weeks_relv,colonized_prevdf_3weeks_relv,colonized_prevdf_4weeks_relv,colonized_prevdf_5weeks_relv,colonized_prevdf_40days_relv,colonized_prevdf_euthan_relv))
str(merged_prev_df_CHLD2_piglets) # All columns considered factors
merged_prev_df_CHLD2_piglets[2:9] <- lapply(merged_prev_df_CHLD2_piglets[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_piglets)
View(merged_prev_df_CHLD2_piglets)
     
merged_prev_df_CHLD2_piglets[is.na(merged_prev_df_CHLD2_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_piglets$total <- rowSums(merged_prev_df_CHLD2_piglets[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_piglets)
merged_prev_df_CHLD2_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD2_piglets)
write.table(merged_prev_df_CHLD2_piglets, "merged_prev_df_CHLD2_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_CHLD2_piglets <- merged_prev_df_CHLD2_piglets[,1]
temporally_colonizing_taxa_CHLD2_piglets # 105 ASVs temporarily colonizing the piglets

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_CHLD2_piglets_df)
actually_colonizing_taxa_CHLD2_piglets <- actually_colonizing_taxa_CHLD2_piglets_df[,1]
actually_colonizing_taxa_CHLD2_piglets # 80 ASVs persistently colonizing pigs

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points (50% of time points)
persistently_colonizing_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_CHLD2_piglets_df)
persistently_colonizing_taxa_CHLD2_piglets <- persistently_colonizing_taxa_CHLD2_piglets_df[,1]
persistently_colonizing_taxa_CHLD2_piglets # 75 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_piglets, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_CHLD2_piglets_df)
truly_colonizing_taxa_CHLD2_piglets <- truly_colonizing_taxa_CHLD2_piglets_df[,1]
truly_colonizing_taxa_CHLD2_piglets # 71 ASVs truly colonizing pigs


# ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for CHLD2 HMA piglets
actually_colonizing_taxa_CHLD2_piglets_ps <- prune_taxa(actually_colonizing_taxa_CHLD2_piglets,CHLD2_HMA_piglets_ps)
actually_colonizing_taxa_CHLD2_piglets_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_CHLD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_CHLD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_coloniz_piglets, "taxonomy_table_for_CHLD2_ASVs_actually_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for CHLD2 HMA mice
actually_colonizing_taxa_CHLD2_mice_ps <- prune_taxa(actually_colonizing_taxa_CHLD2_mice,CHLD2_HMA_mice_ps)
actually_colonizing_taxa_CHLD2_mice_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_CHLD2_mice_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_CHLD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_coloniz_mice, "taxonomy_table_for_CHLD2_ASVs_actually_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


**Adult donor**

Separating out the taxa that were found in at least one of the two samples used to inoculate the mice :

```{r, echo=TRUE}
samples_used_to_inoculate_mice <- c('R2D271', 'R2D275') 
human_donors_ADLT2_inocula_for_mice_ps <- subset_samples(human_donors_ADLT2_inocula_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_ADLT2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_ADLT2_mouse_donors <- sum(otu_table(human_donors_ADLT2_inocula_for_mice_ps))/2
avg_num_of_reads_ADLT2_mouse_donors # 43,779 reads/sample

# Determining taxa prevalence
prevdf_ADLT2_donor_taxa_mice <- makePrevDF(human_donors_ADLT2_inocula_for_mice_ps, 1)
prevdf_ADLT2_donor_taxa_mice_table <- as.data.frame(prevdf_ADLT2_donor_taxa_mice[1])
View(prevdf_ADLT2_donor_taxa_mice_table)
all_taxa_in_donor_ADLT2_mice <- prevdf_ADLT2_donor_taxa_mice[2]
all_taxa_in_donor_ADLT2_mice <- unlist(all_taxa_in_donor_ADLT2_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_ADLT2_mice # the ADLT2 donor inoculae used to inoculate mice had 214 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_ADLT2_donor_taxa_mice_ps <- prune_taxa(all_taxa_in_donor_ADLT2_mice, human_donors_ADLT2_inocula_for_mice_ps)
all_ADLT2_donor_taxa_mice_ps
taxa_cols <- colnames(tax_table(all_ADLT2_donor_taxa_mice_ps))
taxa_table <- as.data.frame(tax_table(all_ADLT2_donor_taxa_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_ADLT2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_ADLT2_mice) # check to see that everything looks good
write.table(taxa_table_all_ADLT2_mice, "taxonomy_table_all_donor_ADLT2_ASVs_inoculae_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

Let's see how many of these ASVs have established in the mice. In order to be considered 'established', an ASV needs to be detected in at least **3** samplings in at least **2/3 of animals** :

```{r, echo=TRUE}
ADLT2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_master_ps, Species == 'mouse') # keeping just the mouse samples
ADLT2_HMA_mice_ps
avg_num_of_reads_ADLT2_mice <- sum(otu_table(ADLT2_HMA_mice_ps))/80
avg_num_of_reads_ADLT2_mice # 45,216 reads/sample

# Let's see how many total ASVs were found in at least one ADLT2 HMA mouse fecal sample
prevdf_ADLT2_mice <- makePrevDF(ADLT2_HMA_mice_ps, 1)
prevdf_ADLT2_mice_table <- as.data.frame(prevdf_ADLT2_mice[1])
colonized_prevdf_ADLT2_mice <- prevdf_ADLT2_mice_table[prevdf_ADLT2_mice_table$Prevalence >= '1',]
View(colonized_prevdf_ADLT2_mice) # 267 total ASVs found across ADLT2 mice

ADLT2_HMA_mice_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_ADLT2_mice, ADLT2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
ADLT2_HMA_mice_donor_taxa_ps 

sum(otu_table(ADLT2_HMA_mice_donor_taxa_ps))/sum(otu_table(ADLT2_HMA_mice_ps)) # 71.16% of reads originating from donor ASVs

# Checking how many of the 214 ADLT2 donor ASVs were detected in at least one of the mouse samples
prevdf_ADLT2_donor_taxa_detect_mice <- makePrevDF(ADLT2_HMA_mice_donor_taxa_ps, 1)
prevdf_ADLT2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_ADLT2_donor_taxa_detect_mice[1])
View(prevdf_ADLT2_donor_taxa_detect_mice_table)
all_ADLT2_taxa_colonizing_mice <-prevdf_ADLT2_donor_taxa_detect_mice[2]
all_ADLT2_taxa_colonizing_mice <- unlist(all_ADLT2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_ADLT2_taxa_colonizing_mice # 92 out of the possible 214 ADLT2 donor ASVs have colonized at least one mouse fecal sample


# split by Description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT2_HMA_mice_ps_48HRS <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT2_HMA_mice_ps_48HRS,6)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_mice_ps_1week <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '1week_post_inoc')
ADLT2_HMA_mice_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_mice_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '6',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_mice_ps_2weeks <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_mice_ps_3weeks <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_mice_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_mice_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '6',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_mice_ps_4weeks <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_mice_ps_5weeks <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_mice_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_mice_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
View(colonized_prevdf_5weeks)
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_mice_ps_40days <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == 'Final_collection')
ADLT2_HMA_mice_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_mice_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
ADLT2_HMA_mice_ps_euthan <- subset_samples(ADLT2_HMA_mice_donor_taxa_ps, Description == '24h_before_euthanization')
ADLT2_HMA_mice_ps_euthan

prevdf_euthan <- makePrevDF(ADLT2_HMA_mice_ps_euthan,6)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '6',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_ADLT2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv, colonized_prevdf_euthan_relv))
str(merged_prev_df_ADLT2_mice) # All columns considered factors
merged_prev_df_ADLT2_mice[2:9] <- lapply(merged_prev_df_ADLT2_mice[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_mice)
View(merged_prev_df_ADLT2_mice)
     
merged_prev_df_ADLT2_mice[is.na(merged_prev_df_ADLT2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_mice$total <- rowSums(merged_prev_df_ADLT2_mice[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_mice)
merged_prev_df_ADLT2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_mice)
write.table(merged_prev_df_ADLT2_mice, "merged_prev_df_ADLT2_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_ADLT2_mice <- merged_prev_df_ADLT2_mice[,1]
temporally_colonizing_taxa_ADLT2_mice # 38 ASVs temporarily colonizing mice

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_ADLT2_mice_df)
actually_colonizing_taxa_ADLT2_mice <- actually_colonizing_taxa_ADLT2_mice_df[,1]
actually_colonizing_taxa_ADLT2_mice # 18 ASVs persistently colonizing mice

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points
persistently_colonizing_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_ADLT2_mice_df)
persistently_colonizing_taxa_ADLT2_mice <- persistently_colonizing_taxa_ADLT2_mice_df[,1]
persistently_colonizing_taxa_ADLT2_mice # 13 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_mice, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_ADLT2_mice_df)
truly_colonizing_taxa_ADLT2_mice <- truly_colonizing_taxa_ADLT2_mice_df[,1]
truly_colonizing_taxa_ADLT2_mice # 7 ASVs persistently colonizing mice
```

Performing same analysis for ADLT2 HMA piglets

Separating out the taxa that were found in at least one of the two samples used to inoculate the piglets :

```{r, echo=TRUE}
samples_used_to_inoculate_piglets <- c('348', '372') 
human_donors_ADLT2_inocula_for_piglets_ps <- subset_samples(human_donors_ADLT2_inocula_ps, SampleID %in% samples_used_to_inoculate_piglets)
human_donors_ADLT2_inocula_for_piglets_ps # only retains the inocula used for piglets
avg_num_of_reads_ADLT2_piglet_donors <- sum(otu_table(human_donors_ADLT2_inocula_for_piglets_ps))/2
avg_num_of_reads_ADLT2_piglet_donors # 36,901 reads/sample

# Determining taxa prevalence
prevdf_ADLT2_donor_taxa_piglets <- makePrevDF(human_donors_ADLT2_inocula_for_piglets_ps, 1)
prevdf_ADLT2_donor_taxa_piglets_table <- as.data.frame(prevdf_ADLT2_donor_taxa_piglets[1])
View(prevdf_ADLT2_donor_taxa_piglets_table)
all_taxa_in_donor_ADLT2_piglets <- prevdf_ADLT2_donor_taxa_piglets[2]
all_taxa_in_donor_ADLT2_piglets <- unlist(all_taxa_in_donor_ADLT2_piglets) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_ADLT2_piglets # the ADLT2 donor inoculae used to inoculate piglets had 219 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_ADLT2_donor_taxa_piglets_ps <- prune_taxa(all_taxa_in_donor_ADLT2_piglets, human_donors_ADLT2_inocula_for_piglets_ps)
all_ADLT2_donor_taxa_piglets_ps
taxa_cols <- colnames(tax_table(all_ADLT2_donor_taxa_piglets_ps))
taxa_table <- as.data.frame(tax_table(all_ADLT2_donor_taxa_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_ADLT2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_ADLT2_piglets) # check to see that everything looks good
write.table(taxa_table_all_ADLT2_piglets, "taxonomy_table_all_donor_ADLT2_ASVs_inoculae_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# Let's only keep those ASVs which came from the donor for the HMA piglet fecal samples
ADLT2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'piglet') # keeping just the mouse samples
ADLT2_HMA_piglets_ps
avg_num_of_reads_ADLT2_piglets <- sum(otu_table(ADLT2_HMA_piglets_ps))/23
avg_num_of_reads_ADLT2_piglets  # 40,288 reads/sample

# Let's see how many total ASVs were found in at least one ADLT2 HMA piglet fecal sample
prevdf_ADLT2_pigs <- makePrevDF(ADLT2_HMA_piglets_ps, 1)
prevdf_ADLT2_pigs_table <- as.data.frame(prevdf_ADLT2_pigs[1])
colonized_prevdf_ADLT2_pigs <- prevdf_ADLT2_pigs_table[prevdf_ADLT2_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_ADLT2_pigs) # 395 total ASVs found across ADLT2 piglets

ADLT2_HMA_piglets_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_ADLT2_piglets, ADLT2_HMA_piglets_ps) # retaining only the ASVs that were found in the donor inoculae
ADLT2_HMA_piglets_donor_taxa_ps 

# Calculating the percentage of the total HMA piglet reads that is represented by these 219 donor ASVs
sum(otu_table(ADLT2_HMA_piglets_donor_taxa_ps))/sum(otu_table(ADLT2_HMA_piglets_ps)) # 73.5% of reads originating from donor ASVs
```

Let's see how many of the possible 219 ADLT2 donor ASVs are detected in at least one pig fecal sample :

```{r, echo=TRUE}
prevdf_ADLT2_donor_taxa_detect_pigs <- makePrevDF(ADLT2_HMA_piglets_donor_taxa_ps, 1)
prevdf_ADLT2_donor_taxa_detect_pigs_table <- as.data.frame(prevdf_ADLT2_donor_taxa_detect_pigs[1])
View(prevdf_ADLT2_donor_taxa_detect_pigs_table)
all_ADLT2_taxa_colonizing_pigs <-prevdf_ADLT2_donor_taxa_detect_pigs[2]
all_ADLT2_taxa_colonizing_pigs <- unlist(all_ADLT2_taxa_colonizing_pigs) # need to 'unlist' in order to use in the 'prune_taxa' function below 
all_ADLT2_taxa_colonizing_pigs # 151 of the possible 219 ADLT2 donor ASVs were detected in at least one piglet fecal sample
```

Determining the colonization of ADLT2 donor ASVs in piglets by sampling time point :

```{r, echo=TRUE}

# Prevalence of ASVs at 48 HRS
ADLT2_HMA_piglets_ps_48HRS <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_piglets_ps_48HRS # only 2 samples because one was taken out as an outlier

prevdf_48HRS <- makePrevDF(ADLT2_HMA_piglets_ps_48HRS,2)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_piglets_ps_1week <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '1week_post_inoc')
ADLT2_HMA_piglets_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_piglets_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_piglets_ps_2weeks <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_piglets_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_piglets_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_piglets_ps_3weeks <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_piglets_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_piglets_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_piglets_ps_4weeks <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_piglets_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_piglets_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_piglets_ps_5weeks <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_piglets_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_piglets_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_piglets_ps_40days <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == 'Final_collection')
ADLT2_HMA_piglets_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_piglets_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
ADLT2_HMA_piglets_ps_euthan <- subset_samples(ADLT2_HMA_piglets_donor_taxa_ps, Description == 'at_euthanization')
ADLT2_HMA_piglets_ps_euthan 

prevdf_euthan <- makePrevDF(ADLT2_HMA_piglets_ps_euthan,2)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '2',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_ADLT2_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv,colonized_prevdf_1week_relv,colonized_prevdf_2weeks_relv,colonized_prevdf_3weeks_relv,colonized_prevdf_4weeks_relv,colonized_prevdf_5weeks_relv,colonized_prevdf_40days_relv,colonized_prevdf_euthan_relv))
str(merged_prev_df_ADLT2_piglets) # All columns considered factors
merged_prev_df_ADLT2_piglets[2:9] <- lapply(merged_prev_df_ADLT2_piglets[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_piglets)
View(merged_prev_df_ADLT2_piglets)
     
merged_prev_df_ADLT2_piglets[is.na(merged_prev_df_ADLT2_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_piglets$total <- rowSums(merged_prev_df_ADLT2_piglets[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_piglets)
merged_prev_df_ADLT2_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_piglets)
write.table(merged_prev_df_ADLT2_piglets, "merged_prev_df_ADLT2_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_ADLT2_piglets <- merged_prev_df_ADLT2_piglets[,1]
temporally_colonizing_taxa_ADLT2_piglets # 126 ASVs temporarily colonizing the piglets

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_ADLT2_piglets_df)
actually_colonizing_taxa_ADLT2_piglets <- actually_colonizing_taxa_ADLT2_piglets_df[,1]
actually_colonizing_taxa_ADLT2_piglets # 97 ASVs persistently colonizing pigs

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points
persistently_colonizing_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_ADLT2_piglets_df)
persistently_colonizing_taxa_ADLT2_piglets <- persistently_colonizing_taxa_ADLT2_piglets_df[,1]
persistently_colonizing_taxa_ADLT2_piglets # 88 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_piglets, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_ADLT2_piglets_df)
truly_colonizing_taxa_ADLT2_piglets <- truly_colonizing_taxa_ADLT2_piglets_df[,1]
truly_colonizing_taxa_ADLT2_piglets # 78 ASVs truly colonizing pigs

# ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ADLT2 HMA piglets
actually_colonizing_taxa_ADLT2_piglets_ps <- prune_taxa(actually_colonizing_taxa_ADLT2_piglets,ADLT2_HMA_piglets_ps)
actually_colonizing_taxa_ADLT2_piglets_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_ADLT2_piglets_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_ADLT2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_coloniz_piglets, "taxonomy_table_for_ADLT2_ASVs_actually_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ADLT2 HMA mice
actually_colonizing_taxa_ADLT2_mice
actually_colonizing_taxa_ADLT2_mice_ps <- prune_taxa(actually_colonizing_taxa_ADLT2_mice, ADLT2_HMA_mice_ps)
actually_colonizing_taxa_ADLT2_mice_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_ADLT2_mice_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_ADLT2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_coloniz_mice, "taxonomy_table_for_ADLT2_ASVs_actually_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```

**Elderly Donor**

Separating out the taxa that were found in at least one of the two samples used to inoculate the mice :

```{r, echo=TRUE}
samples_used_to_inoculate_mice <- c('R2D272', 'R2D276') 
human_donors_ELD2_inocula_for_mice_ps <- subset_samples(human_donors_ELD2_inocula_ps, SampleID %in% samples_used_to_inoculate_mice)
human_donors_ELD2_inocula_for_mice_ps # only retains the inocula used for mice
avg_num_of_reads_ELD2_mouse_donors <- sum(otu_table(human_donors_ELD2_inocula_for_mice_ps))/2
avg_num_of_reads_ELD2_mouse_donors # 34,379 reads/sample

# Determining taxa prevalence
prevdf_ELD2_donor_taxa_mice <- makePrevDF(human_donors_ELD2_inocula_for_mice_ps, 1)
prevdf_ELD2_donor_taxa_mice_table <- as.data.frame(prevdf_ELD2_donor_taxa_mice[1])
View(prevdf_ELD2_donor_taxa_mice_table)
all_taxa_in_donor_ELD2_mice <- prevdf_ELD2_donor_taxa_mice[2]
all_taxa_in_donor_ELD2_mice <- unlist(all_taxa_in_donor_ELD2_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_ELD2_mice # the ELD2 donor inoculae used to inoculate mice had 202 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_ELD2_donor_taxa_mice_ps <- prune_taxa(all_taxa_in_donor_ELD2_mice, human_donors_ELD2_inocula_for_mice_ps)
all_ELD2_donor_taxa_mice_ps
taxa_cols <- colnames(tax_table(all_ELD2_donor_taxa_mice_ps))
taxa_table <- as.data.frame(tax_table(all_ELD2_donor_taxa_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_ELD2_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_ELD2_mice) # check to see that everything looks good
write.table(taxa_table_all_ELD2_mice, "taxonomy_table_all_donor_ELD2_ASVs_inoculae_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

Let's see how many of these ASVs have established in the mice. In order to be considered 'established', an ASV needs to be detected in at least **3** samplings in at least **2/3 of animals** :

```{r, echo=TRUE}
ELD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ELD2_master_ps, Species == 'mouse') # keeping just the mouse samples
ELD2_HMA_mice_ps
avg_num_of_reads_ELD2_mice <- sum(otu_table(ELD2_HMA_mice_ps))/79
avg_num_of_reads_ELD2_mice # 45,833 reads/sample

# Let's see how many total ASVs were found in at least one ELD2 HMA mouse fecal sample
prevdf_ELD2_mice <- makePrevDF(ELD2_HMA_mice_ps, 1)
prevdf_ELD2_mice_table <- as.data.frame(prevdf_ELD2_mice[1])
colonized_prevdf_ELD2_mice <- prevdf_ELD2_mice_table[prevdf_ELD2_mice_table$Prevalence >= '1',]
View(colonized_prevdf_ELD2_mice) # 237 total ASVs found across ADLT2 mice

ELD2_HMA_mice_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_ELD2_mice, ELD2_HMA_mice_ps) # retaining only the ASVs that were found in the donor inoculae
ELD2_HMA_mice_donor_taxa_ps 

# Calculating the percentage of the total HMA mouse reads that is represented by these 202 donor ASVs
sum(otu_table(ELD2_HMA_mice_donor_taxa_ps))/sum(otu_table(ELD2_HMA_mice_ps)) # 59.3% of reads originating from donor ASVs

# Checking how many of the 202 ELD2 donor ASVs were detected in at least one of the mouse samples
prevdf_ELD2_donor_taxa_detect_mice <- makePrevDF(ELD2_HMA_mice_donor_taxa_ps, 1)
prevdf_ELD2_donor_taxa_detect_mice_table <- as.data.frame(prevdf_ELD2_donor_taxa_detect_mice[1])
View(prevdf_ELD2_donor_taxa_detect_mice_table)
all_ELD2_taxa_colonizing_mice <-prevdf_ELD2_donor_taxa_detect_mice[2]
all_ELD2_taxa_colonizing_mice <- unlist(all_ELD2_taxa_colonizing_mice) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_ELD2_taxa_colonizing_mice # 83 out of the possible 202 ELD2 donor ASVs have colonized at least one mouse fecal sample


# split by Description to get weekly .....
# Prevalence of ASVs at 48 HRS
ELD2_HMA_mice_ps_48HRS <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '48hrs_post_inoc')
ELD2_HMA_mice_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_mice_ps_48HRS,6)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_mice_ps_1week <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '1week_post_inoc')
ELD2_HMA_mice_ps_1week # only 9 samples (1 sample wuth <10,500 reads was removed earlier)

prevdf_1week <- makePrevDF(ELD2_HMA_mice_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '5',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_mice_ps_2weeks <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '2weeks_post_inoc')
ELD2_HMA_mice_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_mice_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_mice_ps_3weeks <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '3weeks_post_inoc')
ELD2_HMA_mice_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_mice_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '6',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_mice_ps_4weeks <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '4weeks_post_inoc')
ELD2_HMA_mice_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_mice_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_mice_ps_5weeks <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '5weeks_post_inoc')
ELD2_HMA_mice_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_mice_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_mice_ps_40days <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == 'Final_collection')
ELD2_HMA_mice_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_mice_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
ELD2_HMA_mice_ps_euthan <- subset_samples(ELD2_HMA_mice_donor_taxa_ps, Description == '24h_before_euthanization')
ELD2_HMA_mice_ps_euthan

prevdf_euthan <- makePrevDF(ELD2_HMA_mice_ps_euthan,6)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '6',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_ELD2_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv, colonized_prevdf_euthan_relv))
str(merged_prev_df_ELD2_mice) # All columns considered factors
merged_prev_df_ELD2_mice[2:9] <- lapply(merged_prev_df_ELD2_mice[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_mice)
View(merged_prev_df_ELD2_mice)
     
merged_prev_df_ELD2_mice[is.na(merged_prev_df_ELD2_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_mice$total <- rowSums(merged_prev_df_ELD2_mice[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_mice)
merged_prev_df_ELD2_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ELD2_mice)
write.table(merged_prev_df_ELD2_mice, "merged_prev_df_ELD_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_ELD2_mice <- merged_prev_df_ELD2_mice[,1]
temporally_colonizing_taxa_ELD2_mice # 58 ASVs temporarily colonizing mice

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_ELD2_mice_df)
actually_colonizing_taxa_ELD2_mice <- actually_colonizing_taxa_ELD2_mice_df[,1]
actually_colonizing_taxa_ELD2_mice # 25 ASVs persistently colonizing mice

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points
persistently_colonizing_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_mice, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_ELD2_mice_df)
persistently_colonizing_taxa_ELD2_mice <- persistently_colonizing_taxa_ELD2_mice_df[,1]
persistently_colonizing_taxa_ELD2_mice # 16 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_mice, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_ELD2_mice_df)
truly_colonizing_taxa_ELD2_mice <- truly_colonizing_taxa_ELD2_mice_df[,1]
truly_colonizing_taxa_ELD2_mice # 8 ASVs truly colonizing ELD2 mice
``` 


Performing same analysis for ELD2 HMA piglets

Separating out the taxa that were found in at least one of the two samples used to inoculate the piglets :

```{r, echo=TRUE}
samples_used_to_inoculate_piglets <- c('324', '359') 
human_donors_ELD2_inocula_for_piglets_ps <- subset_samples(human_donors_ELD2_inocula_ps, SampleID %in% samples_used_to_inoculate_piglets)
human_donors_ELD2_inocula_for_piglets_ps # only retains the inocula used for piglets
avg_num_of_reads_ELD2_piglet_donors <- sum(otu_table(human_donors_ELD2_inocula_for_piglets_ps))/2
avg_num_of_reads_ELD2_piglet_donors # 64,164 reads/sample

# Determining taxa prevalence
prevdf_ELD2_donor_taxa_piglets <- makePrevDF(human_donors_ELD2_inocula_for_piglets_ps, 1)
prevdf_ELD2_donor_taxa_piglets_table <- as.data.frame(prevdf_ELD2_donor_taxa_piglets[1])
View(prevdf_ELD2_donor_taxa_piglets_table)
all_taxa_in_donor_ELD2_piglets <- prevdf_ELD2_donor_taxa_piglets[2]
all_taxa_in_donor_ELD2_piglets <- unlist(all_taxa_in_donor_ELD2_piglets) # need to 'unlist' in order to use in the 'prune_taxa' function below
all_taxa_in_donor_ELD2_piglets # the ELD2 donor inoculae used to inoculate piglets had 281 ASVs in at least 1 of the samples used to inoculate

# Let's make a taxonomy table for these ASVs
all_ELD2_donor_taxa_piglets_ps <- prune_taxa(all_taxa_in_donor_ELD2_piglets, human_donors_ELD2_inocula_for_piglets_ps)
all_ELD2_donor_taxa_piglets_ps
taxa_cols <- colnames(tax_table(all_ELD2_donor_taxa_piglets_ps))
taxa_table <- as.data.frame(tax_table(all_ELD2_donor_taxa_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_all_ELD2_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_all_ELD2_piglets) # check to see that everything looks good
write.table(taxa_table_all_ELD2_piglets, "taxonomy_table_all_donor_ELD2_ASVs_inoculae_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# Let's only keep those ASVs which came from the donor for the HMA piglet fecal samples
ELD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'piglet') # keeping just the mouse samples
ELD2_HMA_piglets_ps 
avg_num_of_reads_ELD2_piglets <- sum(otu_table(ELD2_HMA_piglets_ps))/24
avg_num_of_reads_ELD2_piglets # 34,309 reads/sample

# Let's see how many total ASVs were found in at least one ELD2 HMA mouse fecal sample
prevdf_ELD2_pigs <- makePrevDF(ELD2_HMA_piglets_ps, 1)
prevdf_ELD2_pigs_table <- as.data.frame(prevdf_ELD2_pigs[1])
colonized_prevdf_ELD2_pigs <- prevdf_ELD2_pigs_table[prevdf_ELD2_pigs_table$Prevalence >= '1',]
View(colonized_prevdf_ELD2_pigs) # 428 ASVs found in at least across ELD2 HMA piglet fecal samples

ELD2_HMA_piglets_donor_taxa_ps <- prune_taxa(all_taxa_in_donor_ELD2_piglets, ELD2_HMA_piglets_ps) # retaining only the ASVs that were found in the donor inoculae
ELD2_HMA_piglets_donor_taxa_ps 

# Calculating the percentage of the total HMA piglet reads that is represented by these 202 donor ASVs
sum(otu_table(ELD2_HMA_piglets_donor_taxa_ps))/sum(otu_table(ELD2_HMA_piglets_ps)) # 62.3% of reads originating from donor ASVs
```

Let's see how many of the possible 281 ELD2 donor ASVs are detected in at least one pig fecal sample :

```{r, echo=TRUE}
prevdf_ELD2_donor_taxa_detect_pigs <- makePrevDF(ELD2_HMA_piglets_donor_taxa_ps, 1)
prevdf_ELD2_donor_taxa_detect_pigs_table <- as.data.frame(prevdf_ELD2_donor_taxa_detect_pigs[1])
View(prevdf_ELD2_donor_taxa_detect_pigs_table)
all_ELD2_taxa_colonizing_pigs <-prevdf_ELD2_donor_taxa_detect_pigs[2]
all_ELD2_taxa_colonizing_pigs <- unlist(all_ELD2_taxa_colonizing_pigs) # need to 'unlist' in order to use in the 'prune_taxa' function below 
all_ELD2_taxa_colonizing_pigs # 170 of the possible 281 ELD2 donor ASVs were detected in at least one piglet fecal sample
```

Determining the colonization of ELD2 donor ASVs in piglets by sampling time point :

```{r, echo=TRUE}

# Prevalence of ASVs at 48 HRS
ELD2_HMA_piglets_ps_48HRS <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '48hrs_post_inoc')
ELD2_HMA_piglets_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_piglets_ps_48HRS,2)
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_piglets_ps_1week <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '1week_post_inoc')
ELD2_HMA_piglets_ps_1week

prevdf_1week <- makePrevDF(ELD2_HMA_piglets_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_piglets_ps_2weeks <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '2weeks_post_inoc')
ELD2_HMA_piglets_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_piglets_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_piglets_ps_3weeks <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '3weeks_post_inoc')
ELD2_HMA_piglets_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_piglets_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_piglets_ps_4weeks <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '4weeks_post_inoc')
ELD2_HMA_piglets_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_piglets_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_piglets_ps_5weeks <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == '5weeks_post_inoc')
ELD2_HMA_piglets_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_piglets_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_piglets_ps_40days <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == 'Final_collection')
ELD2_HMA_piglets_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_piglets_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Prevalence of ASVs at euthanization
ELD2_HMA_piglets_ps_euthan <- subset_samples(ELD2_HMA_piglets_donor_taxa_ps, Description == 'at_euthanization')
ELD2_HMA_piglets_ps_euthan

prevdf_euthan <- makePrevDF(ELD2_HMA_piglets_ps_euthan,2)
prevdf_euthan_table <- as.data.frame(prevdf_euthan[1])
colonized_prevdf_euthan <- prevdf_euthan_table[prevdf_euthan_table$Prevalence >= '2',]
colonized_prevdf_euthan$ASV_ID <- rownames(colonized_prevdf_euthan)
colonized_prevdf_euthan <- colonized_prevdf_euthan[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_euthan)))
colonized_prevdf_euthan_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_euthan), at_euthanization =  tp_list)
dim(colonized_prevdf_euthan_relv)
View(colonized_prevdf_euthan_relv)

# Merging all dataframes together
merged_prev_df_ELD2_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv,colonized_prevdf_1week_relv,colonized_prevdf_2weeks_relv,colonized_prevdf_3weeks_relv,colonized_prevdf_4weeks_relv,colonized_prevdf_5weeks_relv,colonized_prevdf_40days_relv,colonized_prevdf_euthan_relv))
str(merged_prev_df_ELD2_piglets) # All columns considered factors
merged_prev_df_ELD2_piglets[2:9] <- lapply(merged_prev_df_ELD2_piglets[2:9], as.numeric) # need to convert columns 2-9 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_piglets)
View(merged_prev_df_ELD2_piglets)
write.table(merged_prev_df_ELD2_piglets, "merged_prev_df_ELD_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
     
merged_prev_df_ELD2_piglets[is.na(merged_prev_df_ELD2_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_piglets$total <- rowSums(merged_prev_df_ELD2_piglets[2:9]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_piglets)
merged_prev_df_ELD2_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ELD2_piglets)
write.table(merged_prev_df_ELD2_piglets, "merged_prev_df_ELD2_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_taxa_ELD2_piglets <- merged_prev_df_ELD2_piglets[,1]
temporally_colonizing_taxa_ELD2_piglets # 136 ASVs temporarily colonizing the piglets

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_taxa_ELD2_piglets_df)
actually_colonizing_taxa_ELD2_piglets <- actually_colonizing_taxa_ELD2_piglets_df[,1]
actually_colonizing_taxa_ELD2_piglets # 101 ASVs actually colonizing pigs

# persistently colonizing taxa defined as taxa found in a majority of animals in at least 4 time points
persistently_colonizing_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_piglets, total >= '4') # select the ASVs which were detected in at least 4 time points
View(persistently_colonizing_taxa_ELD2_piglets_df)
persistently_colonizing_taxa_ELD2_piglets <- persistently_colonizing_taxa_ELD2_piglets_df[,1]
persistently_colonizing_taxa_ELD2_piglets # 81 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_piglets, total >= '5') # select the ASVs which were detected in at least 3 time points
View(truly_colonizing_taxa_ELD2_piglets_df)
truly_colonizing_taxa_ELD2_piglets <- truly_colonizing_taxa_ELD2_piglets_df[,1]
truly_colonizing_taxa_ELD2_piglets # 50 ASVs truly colonizing pigs


# ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ELD2 HMA piglets
actually_colonizing_taxa_ELD2_piglets_ps <- prune_taxa(actually_colonizing_taxa_ELD2_piglets,ELD2_HMA_piglets_ps)
actually_colonizing_taxa_ELD2_piglets_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_ELD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_ELD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_coloniz_piglets, "taxonomy_table_for_ELD2_ASVs_actually_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ELD2 HMA mice
actually_colonizing_taxa_ELD2_mice
actually_colonizing_taxa_ELD2_mice_ps <- prune_taxa(actually_colonizing_taxa_ELD2_mice, ELD2_HMA_mice_ps)
actually_colonizing_taxa_ELD2_mice_ps
taxa_cols <- colnames(tax_table(actually_colonizing_taxa_ELD2_mice_ps))
taxa_table <- as.data.frame(tax_table(actually_colonizing_taxa_ELD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_coloniz_mice, "taxonomy_table_for_ELD2_ASVs_actually_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


##**Determining the establishment of core donor ASVs in the  HMA animal models**##

Let's see how the core ASVs (ASVs found in all the donor samples) have established between the two animal models :

**Infant donor**

```{r, echo=TRUE}

# establishment of core INF2 ASVs in HMA mice

core_INF2_taxa # 26 core ASVs

INF2_HMA_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
INF2_HMA_mice_ps
INF2_HMA_mice_core_ps <- prune_taxa(core_INF2_taxa, INF2_HMA_mice_ps)
INF2_HMA_mice_core_ps

# Let's see how many core INF2 donor ASVs were found in at least one INF2 HMA mouse fecal sample 
prevdf_INF2_core_mice <- makePrevDF(INF2_HMA_mice_core_ps, 1)
prevdf_INF2_core_mice_table <- as.data.frame(prevdf_INF2_core_mice[1])
colonized_prevdf_INF2_core_mice <- prevdf_INF2_core_mice_table[prevdf_INF2_core_mice_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_core_mice) # 21 of the 26 core INF2 donor ASVs found in at least one INF2 mouse fecal sample
core_INF2_taxa_colonizing_mice <-prevdf_INF2_core_mice[2]
core_INF2_taxa_colonizing_mice <- unlist(core_INF2_taxa_colonizing_mice) 
core_INF2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF2_HMA_mice_core_ps_48HRS <- subset_samples(INF2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
INF2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_mice_core_ps_48HRS,5) # This time point as well as the 1 and 2week time points had 8 animals
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '5',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_mice_core_ps_1week <- subset_samples(INF2_HMA_mice_core_ps, Description == '1week_post_inoc')
INF2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(INF2_HMA_mice_core_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '5',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_mice_core_ps_2weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
INF2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_mice_core_ps_2weeks,5)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '5',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_mice_core_ps_3weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
INF2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_mice_core_ps_3weeks,4)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '4',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_mice_core_ps_4weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
INF2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_mice_core_ps_4weeks,4)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '4',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_mice_core_ps_5weeks <- subset_samples(INF2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
INF2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_mice_core_ps_5weeks,4)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '4',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_mice_core_ps_40days <- subset_samples(INF2_HMA_mice_core_ps, Description == 'Final_collection')
INF2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_mice_core_ps_40days,4)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '4',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_INF2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_INF2_core_mice) # All columns considered factors
merged_prev_df_INF2_core_mice[2:8] <- lapply(merged_prev_df_INF2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_core_mice)
View(merged_prev_df_INF2_core_mice)
     
merged_prev_df_INF2_core_mice[is.na(merged_prev_df_INF2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_core_mice$total <- rowSums(merged_prev_df_INF2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_core_mice)
merged_prev_df_INF2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_core_mice)
write.table(merged_prev_df_INF2_core_mice, "merged_prev_df_INF2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_INF2_mice <- merged_prev_df_INF2_core_mice[,1]
temporally_colonizing_core_taxa_INF2_mice # 19 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_core_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_INF2_mice_df)
actually_colonizing_core_taxa_INF2_mice <- actually_colonizing_core_taxa_INF2_mice_df[,1]
actually_colonizing_core_taxa_INF2_mice # 18 ASVs persistently colonizing mice

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_core_mice, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_INF2_mice_df)
perst_colonizing_core_taxa_INF2_mice <- perst_colonizing_core_taxa_INF2_mice_df[,1]
perst_colonizing_core_taxa_INF2_mice # 16 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_INF2_mice_df <- subset(merged_prev_df_INF2_core_mice, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_INF2_mice_df)
truly_colonizing_core_taxa_INF2_mice <- truly_colonizing_core_taxa_INF2_mice_df[,1]
truly_colonizing_core_taxa_INF2_mice # 14 ASVs 'truly' colonizing mice
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of core INF2 ASVs in HMA piglets

core_INF2_taxa # 26 core ASVs

INF2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'piglet') # keeping just the mouse samples
INF2_HMA_piglets_ps <- subset_samples(INF2_HMA_piglets_ps, Timingofinocu != "early" & SampleType != "cecal")
INF2_HMA_piglets_ps
INF2_HMA_piglets_core_ps <- prune_taxa(core_INF2_taxa, INF2_HMA_piglets_ps)
INF2_HMA_piglets_core_ps

# Let's see how many core INF2 donor ASVs were found in at least one INF2 HMA mouse fecal sample 
prevdf_INF2_core_piglets <- makePrevDF(INF2_HMA_piglets_core_ps, 1)
prevdf_INF2_core_piglets_table <- as.data.frame(prevdf_INF2_core_piglets[1])
colonized_prevdf_INF2_core_piglets <- prevdf_INF2_core_piglets_table[prevdf_INF2_core_piglets_table$Prevalence >= '1',]
View(colonized_prevdf_INF2_core_piglets) # 20 of the 26 core INF2 donor ASVs found in at least one INF2 mouse fecal sample
core_INF2_taxa_colonizing_piglets <-prevdf_INF2_core_piglets[2]
core_INF2_taxa_colonizing_piglets <- unlist(core_INF2_taxa_colonizing_piglets) 
core_INF2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
INF2_HMA_piglets_core_ps_48HRS <- subset_samples(INF2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
INF2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(INF2_HMA_piglets_core_ps_48HRS,3) # This time point had 4 animals
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '3',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
INF2_HMA_piglets_core_ps_1week <- subset_samples(INF2_HMA_piglets_core_ps, Description == '1week_post_inoc')
INF2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(INF2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
INF2_HMA_piglets_core_ps_2weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
INF2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(INF2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
INF2_HMA_piglets_core_ps_3weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
INF2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(INF2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
INF2_HMA_piglets_core_ps_4weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
INF2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(INF2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
INF2_HMA_piglets_core_ps_5weeks <- subset_samples(INF2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
INF2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(INF2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
INF2_HMA_piglets_core_ps_40days <- subset_samples(INF2_HMA_piglets_core_ps, Description == 'Final_collection')
INF2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(INF2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)

# Merging all dataframes together
merged_prev_df_INF2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_INF2_core_piglets) # All columns considered factors
merged_prev_df_INF2_core_piglets[2:8] <- lapply(merged_prev_df_INF2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_INF2_core_piglets)
View(merged_prev_df_INF2_core_piglets)
     
merged_prev_df_INF2_core_piglets[is.na(merged_prev_df_INF2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_INF2_core_piglets$total <- rowSums(merged_prev_df_INF2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_INF2_core_piglets)
merged_prev_df_INF2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_INF2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_INF2_core_piglets)
write.table(merged_prev_df_INF2_core_piglets, "merged_prev_df_INF2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_INF2_piglets <- merged_prev_df_INF2_core_piglets[,1]
temporally_colonizing_core_taxa_INF2_piglets # 19 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_core_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_INF2_piglets_df)
actually_colonizing_core_taxa_INF2_piglets <- actually_colonizing_core_taxa_INF2_piglets_df[,1]
actually_colonizing_core_taxa_INF2_piglets # 18 ASVs colonizing piglets

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_core_piglets, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_INF2_piglets_df)
perst_colonizing_core_taxa_INF2_piglets <- perst_colonizing_core_taxa_INF2_piglets_df[,1]
perst_colonizing_core_taxa_INF2_piglets # 15 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_INF2_piglets_df <- subset(merged_prev_df_INF2_core_piglets, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_INF2_piglets_df)
truly_colonizing_core_taxa_INF2_piglets <- truly_colonizing_core_taxa_INF2_piglets_df[,1]
truly_colonizing_core_taxa_INF2_piglets # 15 ASVs 'truly' colonizing piglets

# Core INF2 donor ASVs that are 'persistently' colonizing each HMA animal model based on colonization criteria (detected in > 50% of animals in >= 4 sampling time points) with taxonomy assignments

# for INF2 HMA piglets
perst_colonizing_core_taxa_INF2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_piglets,INF2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_INF2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_INF2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_INF2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_INF2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for INF2 HMA mice
perst_colonizing_core_taxa_INF2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_mice,INF2_HMA_mice_core_ps)
perst_colonizing_core_taxa_INF2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_INF2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_INF2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_INF2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```

**Child donor**

```{r, echo=TRUE}

# establishment of core CHLD2 ASVs in HMA mice

core_CHLD2_taxa # 76 core ASVs

pig_mouse_comp_CHLD2_comparison_ps
CHLD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
CHLD2_HMA_mice_ps
CHLD2_HMA_mice_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_HMA_mice_ps)
CHLD2_HMA_mice_core_ps

# Let's see how many core CHLD2 donor ASVs were found in at least one CHLD2 HMA mouse fecal sample 
prevdf_CHLD2_core_mice <- makePrevDF(CHLD2_HMA_mice_core_ps, 1)
prevdf_CHLD2_core_mice_table <- as.data.frame(prevdf_CHLD2_core_mice[1])
colonized_prevdf_CHLD2_core_mice <- prevdf_CHLD2_core_mice_table[prevdf_CHLD2_core_mice_table$Prevalence >= '1',]
View(colonized_prevdf_CHLD2_core_mice) # 37 of the 76 core CHLD2 donor ASVs found in at least one CHLD2 mouse fecal sample
core_CHLD2_taxa_colonizing_mice <-prevdf_CHLD2_core_mice[2]
core_CHLD2_taxa_colonizing_mice <- unlist(core_CHLD2_taxa_colonizing_mice) 
core_CHLD2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD2_HMA_mice_core_ps_48HRS <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD2_HMA_mice_core_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_mice_core_ps_1week <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '1week_post_inoc')
CHLD2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_mice_core_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '6',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_mice_core_ps_2weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_mice_core_ps_3weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '6',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_mice_core_ps_4weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_mice_core_ps_5weeks <- subset_samples(CHLD2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_mice_core_ps_40days <- subset_samples(CHLD2_HMA_mice_core_ps, Description == 'Final_collection')
CHLD2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_CHLD2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_CHLD2_core_mice) # All columns considered factors
merged_prev_df_CHLD2_core_mice[2:8] <- lapply(merged_prev_df_CHLD2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_core_mice)
View(merged_prev_df_CHLD2_core_mice)
     
merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_core_mice$total <- rowSums(merged_prev_df_CHLD2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_core_mice)
merged_prev_df_CHLD2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD2_core_mice)
write.table(merged_prev_df_CHLD2_core_mice, "merged_prev_df_CHLD2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_CHLD2_mice <- merged_prev_df_CHLD2_core_mice[,1]
temporally_colonizing_core_taxa_CHLD2_mice # 20 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_core_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_CHLD2_mice_df)
actually_colonizing_core_taxa_CHLD2_mice <- actually_colonizing_core_taxa_CHLD2_mice_df[,1]
actually_colonizing_core_taxa_CHLD2_mice # 5 ASVs colonizing mice

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_core_mice, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_CHLD2_mice_df)
perst_colonizing_core_taxa_CHLD2_mice <- perst_colonizing_core_taxa_CHLD2_mice_df[,1]
perst_colonizing_core_taxa_CHLD2_mice # 3 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_CHLD2_mice_df <- subset(merged_prev_df_CHLD2_core_mice, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_CHLD2_mice_df)
truly_colonizing_core_taxa_CHLD2_mice <- truly_colonizing_core_taxa_CHLD2_mice_df[,1]
truly_colonizing_core_taxa_CHLD2_mice # 0 ASVs 'truly' colonizing mice
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of core CHLD2 ASVs in HMA piglets

core_CHLD2_taxa # 76 core ASVs

CHLD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'piglet') # keeping just the mouse samples
CHLD2_HMA_piglets_ps
CHLD2_HMA_piglets_core_ps <- prune_taxa(core_CHLD2_taxa, CHLD2_HMA_piglets_ps)
CHLD2_HMA_piglets_core_ps

# Let's see how many core CHLD2 donor ASVs were found in at least one CHLD2 HMA mouse fecal sample 
prevdf_CHLD2_core_piglets <- makePrevDF(CHLD2_HMA_piglets_core_ps, 1)
prevdf_CHLD2_core_piglets_table <- as.data.frame(prevdf_CHLD2_core_piglets[1])
colonized_prevdf_CHLD2_core_piglets <- prevdf_CHLD2_core_piglets_table[prevdf_CHLD2_core_piglets_table$Prevalence >= '1',]
View(colonized_prevdf_CHLD2_core_piglets) # 65 of the 76 core CHLD2 donor ASVs found in at least one CHLD2 mouse fecal sample
core_CHLD2_taxa_colonizing_piglets <-prevdf_CHLD2_core_piglets[2]
core_CHLD2_taxa_colonizing_piglets <- unlist(core_CHLD2_taxa_colonizing_piglets) 
core_CHLD2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
CHLD2_HMA_piglets_core_ps_48HRS <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
CHLD2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(CHLD2_HMA_piglets_core_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
CHLD2_HMA_piglets_core_ps_1week <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '1week_post_inoc')
CHLD2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(CHLD2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
CHLD2_HMA_piglets_core_ps_2weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
CHLD2_HMA_piglets_core_ps_3weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
CHLD2_HMA_piglets_core_ps_4weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
CHLD2_HMA_piglets_core_ps_5weeks <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
CHLD2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(CHLD2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
CHLD2_HMA_piglets_core_ps_40days <- subset_samples(CHLD2_HMA_piglets_core_ps, Description == 'Final_collection')
CHLD2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(CHLD2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_CHLD2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_CHLD2_core_piglets) # All columns considered factors
merged_prev_df_CHLD2_core_piglets[2:8] <- lapply(merged_prev_df_CHLD2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_CHLD2_core_piglets)
View(merged_prev_df_CHLD2_core_piglets)
     
merged_prev_df_CHLD2_core_piglets[is.na(merged_prev_df_CHLD2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_CHLD2_core_piglets$total <- rowSums(merged_prev_df_CHLD2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_CHLD2_core_piglets)
merged_prev_df_CHLD2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_CHLD2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_CHLD2_core_piglets)
write.table(merged_prev_df_CHLD2_core_piglets, "merged_prev_df_CHLD2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_CHLD2_piglets <- merged_prev_df_CHLD2_core_piglets[,1]
temporally_colonizing_core_taxa_CHLD2_piglets # 64 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_core_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_CHLD2_piglets_df)
actually_colonizing_core_taxa_CHLD2_piglets <- actually_colonizing_core_taxa_CHLD2_piglets_df[,1]
actually_colonizing_core_taxa_CHLD2_piglets # 49 ASVs actually colonizing piglets

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_core_piglets, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_CHLD2_piglets_df)
perst_colonizing_core_taxa_CHLD2_piglets <- perst_colonizing_core_taxa_CHLD2_piglets_df[,1]
perst_colonizing_core_taxa_CHLD2_piglets # 43 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_CHLD2_piglets_df <- subset(merged_prev_df_CHLD2_core_piglets, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_CHLD2_piglets_df)
truly_colonizing_core_taxa_CHLD2_piglets <- truly_colonizing_core_taxa_CHLD2_piglets_df[,1]
truly_colonizing_core_taxa_CHLD2_piglets # 38 ASVs 'truly' colonizing piglets

# Core CHLD2 donor ASVs that are 'persistently' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for CHLD2 HMA piglets
perst_colonizing_core_taxa_CHLD2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_piglets,CHLD2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_CHLD2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_CHLD2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for CHLD2 HMA mice
perst_colonizing_core_taxa_CHLD2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_mice,CHLD2_HMA_mice_core_ps)
perst_colonizing_core_taxa_CHLD2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_CHLD2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```

**Adult donor**

```{r, echo=TRUE}

# establishment of core ADLT2 ASVs in HMA mice

core_ADLT2_taxa # 140 core ASVs

pig_mouse_comp_ADLT2_comparison_ps
ADLT2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
ADLT2_HMA_mice_ps
ADLT2_HMA_mice_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_HMA_mice_ps)
ADLT2_HMA_mice_core_ps

# Let's see how many core ADLT2 donor ASVs were found in at least one ADLT2 HMA mouse fecal sample 
prevdf_ADLT2_core_mice <- makePrevDF(ADLT2_HMA_mice_core_ps, 1)
prevdf_ADLT2_core_mice_table <- as.data.frame(prevdf_ADLT2_core_mice[1])
colonized_prevdf_ADLT2_core_mice <- prevdf_ADLT2_core_mice_table[prevdf_ADLT2_core_mice_table$Prevalence >= '1',]
View(colonized_prevdf_ADLT2_core_mice) # 66 of the 140 core ADLT2 donor ASVs found in at least one ADLT2 mouse fecal sample
core_ADLT2_taxa_colonizing_mice <-prevdf_ADLT2_core_mice[2]
core_ADLT2_taxa_colonizing_mice <- unlist(core_ADLT2_taxa_colonizing_mice) 
core_ADLT2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT2_HMA_mice_core_ps_48HRS <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT2_HMA_mice_core_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_mice_core_ps_1week <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '1week_post_inoc')
ADLT2_HMA_mice_core_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_mice_core_ps_1week,6)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '6',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_mice_core_ps_2weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_mice_core_ps_3weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '6',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_mice_core_ps_4weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_mice_core_ps_5weeks <- subset_samples(ADLT2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_mice_core_ps_40days <- subset_samples(ADLT2_HMA_mice_core_ps, Description == 'Final_collection')
ADLT2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ADLT2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ADLT2_core_mice) # All columns considered factors
merged_prev_df_ADLT2_core_mice[2:8] <- lapply(merged_prev_df_ADLT2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_core_mice)
View(merged_prev_df_ADLT2_core_mice)
     
merged_prev_df_ADLT2_core_mice[is.na(merged_prev_df_ADLT2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_core_mice$total <- rowSums(merged_prev_df_ADLT2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_core_mice)
merged_prev_df_ADLT2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_core_mice)
write.table(merged_prev_df_ADLT2_core_mice, "merged_prev_df_ADLT2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_ADLT2_mice <- merged_prev_df_ADLT2_core_mice[,1]
temporally_colonizing_core_taxa_ADLT2_mice # 29 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_core_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_ADLT2_mice_df)
actually_colonizing_core_taxa_ADLT2_mice <- actually_colonizing_core_taxa_ADLT2_mice_df[,1]
actually_colonizing_core_taxa_ADLT2_mice # 11 ASVs persistently colonizing mice

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_core_mice, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_ADLT2_mice_df)
perst_colonizing_core_taxa_ADLT2_mice <- perst_colonizing_core_taxa_ADLT2_mice_df[,1]
perst_colonizing_core_taxa_ADLT2_mice # 7 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_ADLT2_mice_df <- subset(merged_prev_df_ADLT2_core_mice, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_ADLT2_mice_df)
truly_colonizing_core_taxa_ADLT2_mice <- truly_colonizing_core_taxa_ADLT2_mice_df[,1]
truly_colonizing_core_taxa_ADLT2_mice # 4 ASVs 'truly' colonizing mice
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of core ADLT2 ASVs in HMA piglets

core_ADLT2_taxa # 140 core ASVs

ADLT2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'piglet') # keeping just the mouse samples
ADLT2_HMA_piglets_ps
ADLT2_HMA_piglets_core_ps <- prune_taxa(core_ADLT2_taxa, ADLT2_HMA_piglets_ps)
ADLT2_HMA_piglets_core_ps

# Let's see how many core ADLT2 donor ASVs were found in at least one ADLT2 HMA mouse fecal sample 
prevdf_ADLT2_core_piglets <- makePrevDF(ADLT2_HMA_piglets_core_ps, 1)
prevdf_ADLT2_core_piglets_table <- as.data.frame(prevdf_ADLT2_core_piglets[1])
colonized_prevdf_ADLT2_core_piglets <- prevdf_ADLT2_core_piglets_table[prevdf_ADLT2_core_piglets_table$Prevalence >= '1',]
View(colonized_prevdf_ADLT2_core_piglets) # 106 of the 140 core ADLT2 donor ASVs found in at least one ADLT2 mouse fecal sample
core_ADLT2_taxa_colonizing_piglets <-prevdf_ADLT2_core_piglets[2]
core_ADLT2_taxa_colonizing_piglets <- unlist(core_ADLT2_taxa_colonizing_piglets) 
core_ADLT2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ADLT2_HMA_piglets_core_ps_48HRS <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
ADLT2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ADLT2_HMA_piglets_core_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ADLT2_HMA_piglets_core_ps_1week <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '1week_post_inoc')
ADLT2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(ADLT2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ADLT2_HMA_piglets_core_ps_2weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ADLT2_HMA_piglets_core_ps_3weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ADLT2_HMA_piglets_core_ps_4weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ADLT2_HMA_piglets_core_ps_5weeks <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
ADLT2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ADLT2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ADLT2_HMA_piglets_core_ps_40days <- subset_samples(ADLT2_HMA_piglets_core_ps, Description == 'Final_collection')
ADLT2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(ADLT2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ADLT2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ADLT2_core_piglets) # All columns considered factors
merged_prev_df_ADLT2_core_piglets[2:8] <- lapply(merged_prev_df_ADLT2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ADLT2_core_piglets)
View(merged_prev_df_ADLT2_core_piglets)
     
merged_prev_df_ADLT2_core_piglets[is.na(merged_prev_df_ADLT2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ADLT2_core_piglets$total <- rowSums(merged_prev_df_ADLT2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ADLT2_core_piglets)
merged_prev_df_ADLT2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ADLT2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ADLT2_core_piglets)
write.table(merged_prev_df_ADLT2_core_piglets, "merged_prev_df_ADLT2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_ADLT2_piglets <- merged_prev_df_ADLT2_core_piglets[,1]
temporally_colonizing_core_taxa_ADLT2_piglets # 93 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_core_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_ADLT2_piglets_df)
actually_colonizing_core_taxa_ADLT2_piglets <- actually_colonizing_core_taxa_ADLT2_piglets_df[,1]
actually_colonizing_core_taxa_ADLT2_piglets # 74 ASVs persistently colonizing piglets

# persistent colonizers defined as taxa detected in >= 4 time points
perst_colonizing_core_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_core_piglets, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_ADLT2_piglets_df)
perst_colonizing_core_taxa_ADLT2_piglets <- perst_colonizing_core_taxa_ADLT2_piglets_df[,1]
perst_colonizing_core_taxa_ADLT2_piglets # 69 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in >= 5 time points
truly_colonizing_core_taxa_ADLT2_piglets_df <- subset(merged_prev_df_ADLT2_core_piglets, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_ADLT2_piglets_df)
truly_colonizing_core_taxa_ADLT2_piglets <- truly_colonizing_core_taxa_ADLT2_piglets_df[,1]
truly_colonizing_core_taxa_ADLT2_piglets # 61 ASVs 'truly' colonizing piglets

# Core ADLT2 donor ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ADLT2 HMA piglets
perst_colonizing_core_taxa_ADLT2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_piglets,ADLT2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_ADLT2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_ADLT2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ADLT2 HMA mice
perst_colonizing_core_taxa_ADLT2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_mice,ADLT2_HMA_mice_core_ps)
perst_colonizing_core_taxa_ADLT2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_ADLT2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```

**Elderly donor**

```{r, echo=TRUE}

# establishment of core ELD2 ASVs in HMA mice

core_ELD2_taxa # 134 core ASVs

pig_mouse_comp_ELD2_comparison_ps
ELD2_HMA_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'mouse') # keeping just the mouse samples
ELD2_HMA_mice_ps
ELD2_HMA_mice_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_HMA_mice_ps)
ELD2_HMA_mice_core_ps

# Let's see how many core ELD2 donor ASVs were found in at least one ELD2 HMA mouse fecal sample 
prevdf_ELD2_core_mice <- makePrevDF(ELD2_HMA_mice_core_ps, 1)
prevdf_ELD2_core_mice_table <- as.data.frame(prevdf_ELD2_core_mice[1])
colonized_prevdf_ELD2_core_mice <- prevdf_ELD2_core_mice_table[prevdf_ELD2_core_mice_table$Prevalence >= '1',]
View(colonized_prevdf_ELD2_core_mice) # 55 of the 140 core ELD2 donor ASVs found in at least one ELD2 mouse fecal sample
core_ELD2_taxa_colonizing_mice <-prevdf_ELD2_core_mice[2]
core_ELD2_taxa_colonizing_mice <- unlist(core_ELD2_taxa_colonizing_mice) 
core_ELD2_taxa_colonizing_mice

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ELD2_HMA_mice_core_ps_48HRS <- subset_samples(ELD2_HMA_mice_core_ps, Description == '48hrs_post_inoc')
ELD2_HMA_mice_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_mice_core_ps_48HRS,6) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '6',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_mice_core_ps_1week <- subset_samples(ELD2_HMA_mice_core_ps, Description == '1week_post_inoc')
ELD2_HMA_mice_core_ps_1week # only 9 samples at this time point

prevdf_1week <- makePrevDF(ADLT2_HMA_mice_core_ps_1week,5)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '5',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_mice_core_ps_2weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '2weeks_post_inoc')
ELD2_HMA_mice_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_mice_core_ps_2weeks,6)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '6',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_mice_core_ps_3weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '3weeks_post_inoc')
ELD2_HMA_mice_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_mice_core_ps_3weeks,6)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '6',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_mice_core_ps_4weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '4weeks_post_inoc')
ELD2_HMA_mice_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_mice_core_ps_4weeks,6)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '6',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_mice_core_ps_5weeks <- subset_samples(ELD2_HMA_mice_core_ps, Description == '5weeks_post_inoc')
ELD2_HMA_mice_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_mice_core_ps_5weeks,6)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '6',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_mice_core_ps_40days <- subset_samples(ELD2_HMA_mice_core_ps, Description == 'Final_collection')
ELD2_HMA_mice_core_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_mice_core_ps_40days,6)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '6',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ELD2_core_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ELD2_core_mice) # All columns considered factors
merged_prev_df_ELD2_core_mice[2:8] <- lapply(merged_prev_df_ELD2_core_mice[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_core_mice)
View(merged_prev_df_ELD2_core_mice)
     
merged_prev_df_ELD2_core_mice[is.na(merged_prev_df_ELD2_core_mice)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_core_mice$total <- rowSums(merged_prev_df_ELD2_core_mice[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_core_mice)
merged_prev_df_ELD2_core_mice$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_core_mice$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ELD2_core_mice)
write.table(merged_prev_df_ELD2_core_mice, "merged_prev_df_ELD2_core_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_ELD2_mice <- merged_prev_df_ELD2_core_mice[,1]
temporally_colonizing_core_taxa_ELD2_mice # 37 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_core_mice, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_ELD2_mice_df)
actually_colonizing_core_taxa_ELD2_mice <- actually_colonizing_core_taxa_ELD2_mice_df[,1]
actually_colonizing_core_taxa_ELD2_mice # 11 ASVs Actually colonizing mice

# persistent colonizers defined as taxa detected in all >= 4 time points
perst_colonizing_core_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_core_mice, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_ELD2_mice_df)
perst_colonizing_core_taxa_ELD2_mice <- perst_colonizing_core_taxa_ELD2_mice_df[,1]
perst_colonizing_core_taxa_ELD2_mice # 2 ASVs persistently colonizing mice

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_ELD2_mice_df <- subset(merged_prev_df_ELD2_core_mice, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_ELD2_mice_df)
truly_colonizing_core_taxa_ELD2_mice <- truly_colonizing_core_taxa_ELD2_mice_df[,1]
truly_colonizing_core_taxa_ELD2_mice # 1 ASV 'truly' colonizing mice
```

Let's perform a similar analysis for the piglets :

```{r, echo=TRUE}
# establishment of core ELD2 ASVs in HMA piglets

core_ELD2_taxa # 134 core ASVs

ELD2_HMA_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'piglet') # keeping just the piglet samples
ELD2_HMA_piglets_ps
ELD2_HMA_piglets_core_ps <- prune_taxa(core_ELD2_taxa, ELD2_HMA_piglets_ps)
ELD2_HMA_piglets_core_ps

# Let's see how many core ELD2 donor ASVs were found in at least one ELD2 HMA mouse fecal sample 
prevdf_ELD2_core_piglets <- makePrevDF(ELD2_HMA_piglets_core_ps, 1)
prevdf_ELD2_core_piglets_table <- as.data.frame(prevdf_ELD2_core_piglets[1])
colonized_prevdf_ELD2_core_piglets <- prevdf_ELD2_core_piglets_table[prevdf_ELD2_core_piglets_table$Prevalence >= '1',]
View(colonized_prevdf_ELD2_core_piglets) # 92 of the 134 core ADLT2 donor ASVs found in at least one ELD2 mouse fecal sample
core_ELD2_taxa_colonizing_piglets <-prevdf_ELD2_core_piglets[2]
core_ELD2_taxa_colonizing_piglets <- unlist(core_ELD2_taxa_colonizing_piglets) 
core_ELD2_taxa_colonizing_piglets

# split by description to get weekly .....
# Prevalence of ASVs at 48 HRS
ELD2_HMA_piglets_core_ps_48HRS <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '48hrs_post_inoc')
ELD2_HMA_piglets_core_ps_48HRS

prevdf_48HRS <- makePrevDF(ELD2_HMA_piglets_core_ps_48HRS,2) 
prevdf_48HRS_table <- as.data.frame(prevdf_48HRS[1])
colonized_prevdf_48HRS <- prevdf_48HRS_table[prevdf_48HRS_table$Prevalence >= '2',]
colonized_prevdf_48HRS$ASV_ID <- rownames(colonized_prevdf_48HRS)
colonized_prevdf_48HRS <- colonized_prevdf_48HRS[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_48HRS)))
str(tp_list)
colonized_prevdf_48HRS_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_48HRS), Two_days =  tp_list)
dim(colonized_prevdf_48HRS_relv)
View(colonized_prevdf_48HRS_relv)

# Prevalence of ASVs at 1 week
ELD2_HMA_piglets_core_ps_1week <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '1week_post_inoc')
ELD2_HMA_piglets_core_ps_1week

prevdf_1week <- makePrevDF(ELD2_HMA_piglets_core_ps_1week,2)
prevdf_1week_table <- as.data.frame(prevdf_1week[1])
colonized_prevdf_1week <- prevdf_1week_table[prevdf_1week_table$Prevalence >= '2',]
colonized_prevdf_1week$ASV_ID <- rownames(colonized_prevdf_1week)
colonized_prevdf_1week <- colonized_prevdf_1week[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_1week))) # make a list which has '1' which can then be used as a column to indicate 'presence'
colonized_prevdf_1week_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_1week), Week_1 =  tp_list)
dim(colonized_prevdf_1week_relv)
View(colonized_prevdf_1week_relv)

# Prevalence of ASVs at 2 weeks
ELD2_HMA_piglets_core_ps_2weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '2weeks_post_inoc')
ELD2_HMA_piglets_core_ps_2weeks

prevdf_2weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_2weeks,2)
prevdf_2weeks_table <- as.data.frame(prevdf_2weeks[1])
colonized_prevdf_2weeks <- prevdf_2weeks_table[prevdf_2weeks_table$Prevalence >= '2',]
colonized_prevdf_2weeks$ASV_ID <- rownames(colonized_prevdf_2weeks)
colonized_prevdf_2weeks <- colonized_prevdf_2weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_2weeks)))
colonized_prevdf_2weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_2weeks), Week_2 =  tp_list)
dim(colonized_prevdf_2weeks_relv)
View(colonized_prevdf_2weeks_relv)

# Prevalence of ASVs at 3 weeks
ELD2_HMA_piglets_core_ps_3weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '3weeks_post_inoc')
ELD2_HMA_piglets_core_ps_3weeks

prevdf_3weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_3weeks,2)
prevdf_3weeks_table <- as.data.frame(prevdf_3weeks[1])
colonized_prevdf_3weeks <- prevdf_3weeks_table[prevdf_3weeks_table$Prevalence >= '2',]
colonized_prevdf_3weeks$ASV_ID <- rownames(colonized_prevdf_3weeks)
colonized_prevdf_3weeks <- colonized_prevdf_3weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_3weeks)))
colonized_prevdf_3weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_3weeks), Week_3 =  tp_list)
dim(colonized_prevdf_3weeks_relv)
View(colonized_prevdf_3weeks_relv)

# Prevalence of ASVs at 4 weeks
ELD2_HMA_piglets_core_ps_4weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '4weeks_post_inoc')
ELD2_HMA_piglets_core_ps_4weeks

prevdf_4weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_4weeks,2)
prevdf_4weeks_table <- as.data.frame(prevdf_4weeks[1])
colonized_prevdf_4weeks <- prevdf_4weeks_table[prevdf_4weeks_table$Prevalence >= '2',]
colonized_prevdf_4weeks$ASV_ID <- rownames(colonized_prevdf_4weeks)
colonized_prevdf_4weeks <- colonized_prevdf_4weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_4weeks)))
colonized_prevdf_4weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_4weeks), Week_4 =  tp_list)
dim(colonized_prevdf_4weeks_relv)
View(colonized_prevdf_4weeks_relv)

# Prevalence of ASVs at 5 weeks
ELD2_HMA_piglets_core_ps_5weeks <- subset_samples(ELD2_HMA_piglets_core_ps, Description == '5weeks_post_inoc')
ELD2_HMA_piglets_core_ps_5weeks

prevdf_5weeks <- makePrevDF(ELD2_HMA_piglets_core_ps_5weeks,2)
prevdf_5weeks_table <- as.data.frame(prevdf_5weeks[1])
colonized_prevdf_5weeks <- prevdf_5weeks_table[prevdf_5weeks_table$Prevalence >= '2',]
colonized_prevdf_5weeks$ASV_ID <- rownames(colonized_prevdf_5weeks)
colonized_prevdf_5weeks <- colonized_prevdf_5weeks[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_5weeks)))
colonized_prevdf_5weeks_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_5weeks), Week_5 =  tp_list)
dim(colonized_prevdf_5weeks_relv)
View(colonized_prevdf_5weeks_relv)

# Prevalence of ASVs at 40 days
ELD2_HMA_piglets_core_ps_40days <- subset_samples(ELD2_HMA_piglets_core_ps, Description == 'Final_collection')
ELD2_HMA_piglets_core_ps_40days

prevdf_40days <- makePrevDF(ELD2_HMA_piglets_core_ps_40days,2)
prevdf_40days_table <- as.data.frame(prevdf_40days[1])
colonized_prevdf_40days <- prevdf_40days_table[prevdf_40days_table$Prevalence >= '2',]
colonized_prevdf_40days$ASV_ID <- rownames(colonized_prevdf_40days)
colonized_prevdf_40days <- colonized_prevdf_40days[,c(3,1,2)]
tp_list <- rep('1', (nrow(colonized_prevdf_40days)))
colonized_prevdf_40days_relv <- data.frame(ASV_ID = rownames(colonized_prevdf_40days), final_collection =  tp_list)
dim(colonized_prevdf_40days_relv)
View(colonized_prevdf_40days_relv)


# Merging all dataframes together
merged_prev_df_ELD2_core_piglets <- Reduce(function(x, y) merge(x, y, all=TRUE), list(colonized_prevdf_48HRS_relv, colonized_prevdf_1week_relv, colonized_prevdf_2weeks_relv, colonized_prevdf_3weeks_relv, colonized_prevdf_4weeks_relv, colonized_prevdf_5weeks_relv, colonized_prevdf_40days_relv))
str(merged_prev_df_ELD2_core_piglets) # All columns considered factors
merged_prev_df_ELD2_core_piglets[2:8] <- lapply(merged_prev_df_ELD2_core_piglets[2:8], as.numeric) # need to convert columns 2-8 to numeric in order to do 'rowSums'
str(merged_prev_df_ELD2_core_piglets)
View(merged_prev_df_ELD2_core_piglets)
     
merged_prev_df_ELD2_core_piglets[is.na(merged_prev_df_ELD2_core_piglets)] <- 0 # Have to replace 'NA' (missing data) with '0' in order for 'rowSums' to work properly
merged_prev_df_ELD2_core_piglets$total <- rowSums(merged_prev_df_ELD2_core_piglets[2:8]) # add the values in the rows and use them to populate a new column called 'total'
View(merged_prev_df_ELD2_core_piglets)
merged_prev_df_ELD2_core_piglets$ASV_ID <- as.character(as.factor(merged_prev_df_ELD2_core_piglets$ASV_ID)) # convert 'ASV_ID' column to a character vector
str(merged_prev_df_ELD2_core_piglets)
write.table(merged_prev_df_ELD2_core_piglets, "merged_prev_df_ELD2_core_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# temporally colonizing taxa defined as taxa found in 2/3 of animals in at least 1 time point (these are all the ASVs included under the 'ASV_ID' column)
temporally_colonizing_core_taxa_ELD2_piglets <- merged_prev_df_ELD2_core_piglets[,1]
temporally_colonizing_core_taxa_ELD2_piglets # 80 core ASVs 

# actually colonizing taxa defined as taxa found in 2/3 of animals in at least 3 time points
actually_colonizing_core_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_core_piglets, total >= '3') # select the ASVs which were detected in at least 3 time points
View(actually_colonizing_core_taxa_ELD2_piglets_df)
actually_colonizing_core_taxa_ELD2_piglets <- actually_colonizing_core_taxa_ELD2_piglets_df[,1]
actually_colonizing_core_taxa_ELD2_piglets # 58 ASVs actually colonizing piglets

# persistent colonizers defined as taxa detected in >= 4 time points
perst_colonizing_core_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_core_piglets, total >= '4') # select the ASVs which were detected in at least 3 time points
View(perst_colonizing_core_taxa_ELD2_piglets_df)
perst_colonizing_core_taxa_ELD2_piglets <- perst_colonizing_core_taxa_ELD2_piglets_df[,1]
perst_colonizing_core_taxa_ELD2_piglets # 38 ASVs persistently colonizing piglets

# true colonizers defined as taxa detected in all >= 5 time points
truly_colonizing_core_taxa_ELD2_piglets_df <- subset(merged_prev_df_ELD2_core_piglets, total >= '5') # select the ASVs which were detected in at least 5 time points
View(truly_colonizing_core_taxa_ELD2_piglets_df)
truly_colonizing_core_taxa_ELD2_piglets <- truly_colonizing_core_taxa_ELD2_piglets_df[,1]
truly_colonizing_core_taxa_ELD2_piglets # 26 ASVs 'truly' colonizing piglets

# Core ELD2 donor ASVs that are 'actually' colonizing each HMA animal model based on colonization criteria with taxonomy assignments

# for ELD2 HMA piglets
perst_colonizing_core_taxa_ELD2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_piglets,ELD2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_ELD2_piglets_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_piglets <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_piglets) # check to see that everything looks good

write.table(taxa_table_core_coloniz_piglets, "taxonomy_table_for_core_ELD2_ASVs_persistently_colonizing_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE)

# for ADLT2 HMA mice
perst_colonizing_core_taxa_ELD2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_mice,ELD2_HMA_mice_core_ps)
perst_colonizing_core_taxa_ELD2_mice_ps
taxa_cols <- colnames(tax_table(perst_colonizing_core_taxa_ELD2_mice_ps))
taxa_table <- as.data.frame(tax_table(perst_colonizing_core_taxa_ELD2_mice_ps))
dim(taxa_table)
taxa_table$taxonomy <- do.call(paste, c(taxa_table[taxa_cols], sep = ";"))
for (co in taxa_cols) taxa_table[co]<- NULL
View(taxa_table)
taxa_table$ASV_ID <- rownames(taxa_table)
taxa_table_core_coloniz_mice <- taxa_table[,c(2,1)] # arrange table columns in order
View(taxa_table_core_coloniz_mice) # check to see that everything looks good

write.table(taxa_table_core_coloniz_mice, "taxonomy_table_for_core_ELD2_ASVs_persistently_colonizing_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE)
```


## **Looking at the taxonomic classifications to which the core colonizing taxa belong to** ##

**Infant donor**

```{r,echo=TRUE}

### for INF2 donor
human_donors_INF2_inocula_relevant_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_INF2_inocula_table_PHYLUM <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Phylum"], exclude = NULL)
human_donors_INF2_PHYLUM_table <- as.data.frame(human_donors_INF2_inocula_table_PHYLUM)
colnames(human_donors_INF2_PHYLUM_table)[colnames(human_donors_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_INF2_PHYLUM_table)[colnames(human_donors_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_PHYLUM_table)

# at FAMILY level
human_donors_INF2_inocula_table_FAMILY <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Family"], exclude = NULL)
View(human_donors_INF2_inocula_table_FAMILY)
human_donors_INF2_FAMILY_table <- as.data.frame(human_donors_INF2_inocula_table_FAMILY)
colnames(human_donors_INF2_FAMILY_table)[colnames(human_donors_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_INF2_FAMILY_table)[colnames(human_donors_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_FAMILY_table)

# at GENUS level
human_donors_INF2_inocula_table_GENUS <- table(tax_table(human_donors_INF2_inocula_relevant_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_INF2_inocula_table_GENUS)
human_donors_INF2_GENUS_table <- as.data.frame(human_donors_INF2_inocula_table_GENUS)
colnames(human_donors_INF2_GENUS_table)[colnames(human_donors_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_INF2_GENUS_table)[colnames(human_donors_INF2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_INF2_GENUS_table)

# Calculating the relative abundances of the taxa

# Compute prevalence of each feature, store as data.frame
prevdf_donor_INF2 = apply(X = otu_table(human_donors_INF2_inocula_relevant_core_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_INF2_inocula_relevant_core_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_donor_INF2 = data.frame(Prevalence = prevdf_donor_INF2,
                    TotalAbundance = taxa_sums(human_donors_INF2_inocula_relevant_core_ps),
                    tax_table(human_donors_INF2_inocula_relevant_core_ps))
View(prevdf_donor_INF2)

INF2_donor_total_reads <- sum(otu_table(human_donors_INF2_inocula_relevant_ps)) # all reads from INF2 donor samples (including reads from non-core ASVs)
INF2_donor_total_reads

# PHYLUM level
donor_df_phylum <- plyr::ddply(prevdf_donor_INF2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_donor_total_reads))})
View(donor_df_phylum)
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "1"] <- "read count in donors"
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "2"] <- "abundance in donors(%)"
View(donor_df_phylum)

merged_prev_by_PHYLUM_INF2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_INF2_PHYLUM_table,donor_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_INF2_table) 

# FAMILY level
donor_df_family <- plyr::ddply(prevdf_donor_INF2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_donor_total_reads))})
View(donor_df_family)
colnames(donor_df_family)[colnames(donor_df_family) == "1"] <- "read count in donors"
colnames(donor_df_family)[colnames(donor_df_family) == "2"] <- "abundance in donors(%)"
View(donor_df_family)

merged_prev_by_FAMILY_INF2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_INF2_FAMILY_table,donor_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_INF2_table) 


# GENUS level
donor_df_genus <- plyr::ddply(prevdf_donor_INF2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_donor_total_reads))})
View(donor_df_genus)
colnames(donor_df_genus)[colnames(donor_df_genus) == "1"] <- "read count in donors"
colnames(donor_df_genus)[colnames(donor_df_genus) == "2"] <- "abundance in donors(%)"
View(donor_df_genus)

merged_prev_by_GENUS_INF2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_INF2_GENUS_table,donor_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_INF2_table) 


## for mice
# core_taxa persistently colonizing mice (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_INF2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_mice, INF2_HMA_mice_core_ps)
perst_colonizing_core_taxa_INF2_mice_ps

# at PHYLUM level
INF2_perst_core_taxa_mouse_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_INF2_mice_ps)[, "Phylum"], exclude = NULL)
View(INF2_perst_core_taxa_mouse_table_PHYLUM)
core_taxa_mice_INF2_PHYLUM_table <- as.data.frame(INF2_perst_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_PHYLUM_table)

# at FAMILY level
INF2_perst_core_taxa_mouse_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_INF2_mice_ps)[, "Family"], exclude = NULL)
View(INF2_perst_core_taxa_mouse_table_FAMILY)
core_taxa_mice_INF2_FAMILY_table <- as.data.frame(INF2_perst_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_FAMILY_table)

# at GENUS level
INF2_perst_core_taxa_mouse_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_INF2_mice_ps)[, "Genus"], exclude = NULL)
View(INF2_perst_core_taxa_mouse_table_GENUS)
core_taxa_mice_INF2_GENUS_table <- as.data.frame(INF2_perst_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_GENUS_table)

# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_INF2_mice_ps
# Compute prevalence of each feature, store as data.frame
prevdf_mouse_INF2 = apply(X = otu_table(perst_colonizing_core_taxa_INF2_mice_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_INF2_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_mouse_INF2 = data.frame(Prevalence = prevdf_mouse_INF2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_INF2_mice_ps),
                    tax_table(perst_colonizing_core_taxa_INF2_mice_ps))
View(prevdf_mouse_INF2)

INF2_mice_total_reads <- sum(otu_table(INF2_HMA_mice_ps))
INF2_mice_total_reads

# PHYLUM level
mouse_df_phylum <- plyr::ddply(prevdf_mouse_INF2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_mice_total_reads))})
View(mouse_df_phylum)
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "1"] <- "read count in mice"
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "2"] <- "abundance in mice(%)"
View(mouse_df_phylum)

merged_prev_by_PHYLUM_INF2_table_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_INF2_PHYLUM_table,mouse_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_INF2_table_mice) 

# FAMILY level
mouse_df_family <- plyr::ddply(prevdf_mouse_INF2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_mice_total_reads))})
View(mouse_df_family)
colnames(mouse_df_family)[colnames(mouse_df_family) == "1"] <- "read count in mice"
colnames(mouse_df_family)[colnames(mouse_df_family) == "2"] <- "abundance in mice(%)"
View(mouse_df_family)

merged_prev_by_FAMILY_INF2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_INF2_FAMILY_table,mouse_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_INF2_table_mouse) 

# GENUS level
mouse_df_genus <- plyr::ddply(prevdf_mouse_INF2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_mice_total_reads))})
View(mouse_df_genus)
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "1"] <- "read count in mice"
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "2"] <- "abundance in mice(%)"
View(mouse_df_genus)

merged_prev_by_GENUS_INF2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_INF2_GENUS_table,mouse_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_INF2_table_mouse) 


## for piglets
# taxa persistently colonizing piglets (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_INF2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_piglets, INF2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_INF2_piglets_ps

# at PHYLUM level
INF2_perst_core_taxa_piglets_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_INF2_piglets_ps)[, "Phylum"], exclude = NULL)
View(INF2_perst_core_taxa_piglets_table_PHYLUM)
core_taxa_piglets_INF2_PHYLUM_table <- as.data.frame(INF2_perst_core_taxa_piglets_table_PHYLUM)
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_PHYLUM_table)

# at FAMILY level
INF2_perst_core_taxa_piglets_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_INF2_piglets_ps)[, "Family"], exclude = NULL)
View(INF2_perst_core_taxa_piglets_table_FAMILY)
core_taxa_piglets_INF2_FAMILY_table <- as.data.frame(INF2_perst_core_taxa_piglets_table_FAMILY)
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_FAMILY_table)

# at GENUS level
INF2_perst_core_taxa_piglets_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_INF2_piglets_ps)[, "Genus"], exclude = NULL)
View(INF2_perst_core_taxa_piglets_table_GENUS)
core_taxa_piglets_INF2_GENUS_table <- as.data.frame(INF2_perst_core_taxa_piglets_table_GENUS)
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_GENUS_table)


# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_INF2_piglets_ps
# Compute prevalence of each feature, store as data.frame
prevdf_piglets_INF2 = apply(X = otu_table(perst_colonizing_core_taxa_INF2_piglets_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_INF2_piglets_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_piglets_INF2 = data.frame(Prevalence = prevdf_piglets_INF2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_INF2_piglets_ps),
                    tax_table(perst_colonizing_core_taxa_INF2_piglets_ps))
View(prevdf_piglets_INF2)

INF2_piglets_total_reads <- sum(otu_table(INF2_HMA_piglets_ps))
INF2_piglets_total_reads

# PHYLUM level
piglets_df_phylum <- plyr::ddply(prevdf_piglets_INF2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_piglets_total_reads))})
View(piglets_df_phylum)
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "1"] <- "read count in piglets"
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "2"] <- "abundance in piglets(%)"
View(piglets_df_phylum)

merged_prev_by_PHYLUM_INF2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_INF2_PHYLUM_table,piglets_df_phylum)) # merging prevalence and abundance tables
View(merged_prev_by_PHYLUM_INF2_piglets_table)

# FAMILY level
piglets_df_family <- plyr::ddply(prevdf_piglets_INF2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_piglets_total_reads))})
View(piglets_df_family)
colnames(piglets_df_family)[colnames(piglets_df_family) == "1"] <- "read count in piglets"
colnames(piglets_df_family)[colnames(piglets_df_family) == "2"] <- "abundance in piglets(%)"
View(piglets_df_family)

merged_prev_by_FAMILY_INF2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_INF2_FAMILY_table,piglets_df_family)) # merging prevalence and abundance tables
View(merged_prev_by_FAMILY_INF2_piglets_table)

# GENUS level
piglets_df_genus <- plyr::ddply(prevdf_piglets_INF2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/INF2_piglets_total_reads))})
View(piglets_df_genus)
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "1"] <- "read count in piglets"
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "2"] <- "abundance in piglets(%)"
View(piglets_df_genus)

merged_prev_by_GENUS_INF2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_INF2_GENUS_table,piglets_df_genus)) # merging prevalence and abundance tables
View(merged_prev_by_GENUS_INF2_piglets_table)


## MERGED taxa level tables for piglets

install.packages('dplyr')
library("dplyr"); packageVersion("dplyr") # needed for 'rounding' decimal values

# PHYLUM level
merged_phyla_INF2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_PHYLUM_INF2_table,merged_prev_by_PHYLUM_INF2_table_mice,merged_prev_by_PHYLUM_INF2_piglets_table))
merged_phyla_INF2 <- merged_phyla_INF2 %>% mutate_if(is.numeric, round, 2) # rounding-off to 2 decimal places using 'dplyr'
View(merged_phyla_INF2)
write.table(merged_phyla_INF2, "INF2_prev_abundance_comparison_PHYLUM.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Family level
merged_families_INF2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_FAMILY_INF2_table,merged_prev_by_FAMILY_INF2_table_mouse,merged_prev_by_FAMILY_INF2_piglets_table))
merged_families_INF2 <- merged_families_INF2 %>% mutate_if(is.numeric, round, 2)
View(merged_families_INF2)
write.table(merged_families_INF2, "INF2_prev_abundance_comparison_FAMILY.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Genus level
merged_genera_INF2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_GENUS_INF2_table,merged_prev_by_GENUS_INF2_table_mouse,merged_prev_by_GENUS_INF2_piglets_table))
merged_genera_INF2 <- merged_genera_INF2 %>% mutate_if(is.numeric, round, 2)
View(merged_genera_INF2)
write.table(merged_genera_INF2, "INF2_prev_abundance_comparison_GENUS.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**Child donor**

```{r, echo=TRUE}
### for CHLD2 donor
human_donors_CHLD2_inocula_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_CHLD2_inocula_table_PHYLUM <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_CHLD2_PHYLUM_table <- as.data.frame(human_donors_CHLD2_inocula_table_PHYLUM)
colnames(human_donors_CHLD2_PHYLUM_table)[colnames(human_donors_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_CHLD2_PHYLUM_table)[colnames(human_donors_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_PHYLUM_table)

# at FAMILY level
human_donors_CHLD2_inocula_table_FAMILY <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_CHLD2_inocula_table_FAMILY)
human_donors_CHLD2_FAMILY_table <- as.data.frame(human_donors_CHLD2_inocula_table_FAMILY)
colnames(human_donors_CHLD2_FAMILY_table)[colnames(human_donors_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_CHLD2_FAMILY_table)[colnames(human_donors_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_FAMILY_table)

# at GENUS level
human_donors_CHLD2_inocula_table_GENUS <- table(tax_table(human_donors_CHLD2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_CHLD2_inocula_table_GENUS)
human_donors_CHLD2_GENUS_table <- as.data.frame(human_donors_CHLD2_inocula_table_GENUS)
colnames(human_donors_CHLD2_GENUS_table)[colnames(human_donors_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_CHLD2_GENUS_table)[colnames(human_donors_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_CHLD2_GENUS_table)

# Calculating the relative abundances of the taxa

# Compute prevalence of each feature, store as data.frame
prevdf_donor_CHLD2 = apply(X = otu_table(human_donors_CHLD2_inocula_core_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_CHLD2_inocula_core_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_donor_CHLD2 = data.frame(Prevalence = prevdf_donor_CHLD2,
                    TotalAbundance = taxa_sums(human_donors_CHLD2_inocula_core_ps),
                    tax_table(human_donors_CHLD2_inocula_core_ps))
View(prevdf_donor_CHLD2)

CHLD2_donor_total_reads <- sum(otu_table(human_donors_CHLD2_inocula_ps)) # all reads from CHLD2 donor samples (including raeds from non-core ASVs)
CHLD2_donor_total_reads

# PHYLUM level
donor_df_phylum <- plyr::ddply(prevdf_donor_CHLD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_donor_total_reads))})
View(donor_df_phylum)
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "1"] <- "read count in donors"
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "2"] <- "abundance in donors(%)"
View(donor_df_phylum)

merged_prev_by_PHYLUM_CHLD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_CHLD2_PHYLUM_table,donor_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_CHLD2_table) 

# FAMILY level
donor_df_family <- plyr::ddply(prevdf_donor_CHLD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_donor_total_reads))})
View(donor_df_family)
colnames(donor_df_family)[colnames(donor_df_family) == "1"] <- "read count in donors"
colnames(donor_df_family)[colnames(donor_df_family) == "2"] <- "abundance in donors(%)"
View(donor_df_family)

merged_prev_by_FAMILY_CHLD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_CHLD2_FAMILY_table,donor_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_CHLD2_table) 


# GENUS level
donor_df_genus <- plyr::ddply(prevdf_donor_CHLD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_donor_total_reads))})
View(donor_df_genus)
colnames(donor_df_genus)[colnames(donor_df_genus) == "1"] <- "read count in donors"
colnames(donor_df_genus)[colnames(donor_df_genus) == "2"] <- "abundance in donors(%)"
View(donor_df_genus)

merged_prev_by_GENUS_CHLD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_CHLD2_GENUS_table,donor_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_CHLD2_table) 


## for mice
# core_taxa persistently colonizing mice (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_CHLD2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_mice, CHLD2_HMA_mice_core_ps)
perst_colonizing_core_taxa_CHLD2_mice_ps

# at PHYLUM level
CHLD2_perst_core_taxa_mouse_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps)[, "Phylum"], exclude = NULL)
View(CHLD2_perst_core_taxa_mouse_table_PHYLUM)
core_taxa_mice_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_perst_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_PHYLUM_table)

# at FAMILY level
CHLD2_perst_core_taxa_mouse_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps)[, "Family"], exclude = NULL)
View(CHLD2_perst_core_taxa_mouse_table_FAMILY)
core_taxa_mice_CHLD2_FAMILY_table <- as.data.frame(CHLD2_perst_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_FAMILY_table)

# at GENUS level
CHLD2_perst_core_taxa_mouse_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps)[, "Genus"], exclude = NULL)
View(CHLD2_perst_core_taxa_mouse_table_GENUS)
core_taxa_mice_CHLD2_GENUS_table <- as.data.frame(CHLD2_perst_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_GENUS_table)

# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_CHLD2_mice_ps
# Compute prevalence of each feature, store as data.frame
prevdf_mouse_CHLD2 = apply(X = otu_table(perst_colonizing_core_taxa_CHLD2_mice_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_CHLD2_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_mouse_CHLD2 = data.frame(Prevalence = prevdf_mouse_CHLD2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_CHLD2_mice_ps),
                    tax_table(perst_colonizing_core_taxa_CHLD2_mice_ps))
View(prevdf_mouse_CHLD2)

CHLD2_mice_total_reads <- sum(otu_table(CHLD2_HMA_mice_ps))
CHLD2_mice_total_reads

# PHYLUM level
mouse_df_phylum <- plyr::ddply(prevdf_mouse_CHLD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_mice_total_reads))})
View(mouse_df_phylum)
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "1"] <- "read count in mice"
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "2"] <- "abundance in mice(%)"
View(mouse_df_phylum)

merged_prev_by_PHYLUM_CHLD2_table_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_CHLD2_PHYLUM_table,mouse_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_CHLD2_table_mice) 

# FAMILY level
mouse_df_family <- plyr::ddply(prevdf_mouse_CHLD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_mice_total_reads))})
View(mouse_df_family)
colnames(mouse_df_family)[colnames(mouse_df_family) == "1"] <- "read count in mice"
colnames(mouse_df_family)[colnames(mouse_df_family) == "2"] <- "abundance in mice(%)"
View(mouse_df_family)

merged_prev_by_FAMILY_CHLD2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_CHLD2_FAMILY_table,mouse_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_CHLD2_table_mouse) 

# GENUS level
mouse_df_genus <- plyr::ddply(prevdf_mouse_CHLD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_mice_total_reads))})
View(mouse_df_genus)
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "1"] <- "read count in mice"
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "2"] <- "abundance in mice(%)"
View(mouse_df_genus)

merged_prev_by_GENUS_CHLD2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_CHLD2_GENUS_table,mouse_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_CHLD2_table_mouse) 


## for piglets
# taxa persistently colonizing piglets (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_CHLD2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_piglets, CHLD2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_CHLD2_piglets_ps

# at PHYLUM level
CHLD2_perst_core_taxa_piglets_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps)[, "Phylum"], exclude = NULL)
View(CHLD2_perst_core_taxa_piglets_table_PHYLUM)
core_taxa_piglets_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_perst_core_taxa_piglets_table_PHYLUM)
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_PHYLUM_table)

# at FAMILY level
CHLD2_perst_core_taxa_piglets_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps)[, "Family"], exclude = NULL)
View(CHLD2_perst_core_taxa_piglets_table_FAMILY)
core_taxa_piglets_CHLD2_FAMILY_table <- as.data.frame(CHLD2_perst_core_taxa_piglets_table_FAMILY)
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_FAMILY_table)

# at GENUS level
CHLD2_perst_core_taxa_piglets_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps)[, "Genus"], exclude = NULL)
View(CHLD2_perst_core_taxa_piglets_table_GENUS)
core_taxa_piglets_CHLD2_GENUS_table <- as.data.frame(CHLD2_perst_core_taxa_piglets_table_GENUS)
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_GENUS_table)


# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_CHLD2_piglets_ps
# Compute prevalence of each feature, store as data.frame
prevdf_piglets_CHLD2 = apply(X = otu_table(perst_colonizing_core_taxa_CHLD2_piglets_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_CHLD2_piglets_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_piglets_CHLD2 = data.frame(Prevalence = prevdf_piglets_CHLD2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_CHLD2_piglets_ps),
                    tax_table(perst_colonizing_core_taxa_CHLD2_piglets_ps))
View(prevdf_piglets_CHLD2)

CHLD2_piglets_total_reads <- sum(otu_table(CHLD2_HMA_piglets_ps))
CHLD2_piglets_total_reads

# PHYLUM level
piglets_df_phylum <- plyr::ddply(prevdf_piglets_CHLD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_piglets_total_reads))})
View(piglets_df_phylum)
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "1"] <- "read count in piglets"
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "2"] <- "abundance in piglets(%)"
View(piglets_df_phylum)

merged_prev_by_PHYLUM_CHLD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_CHLD2_PHYLUM_table,piglets_df_phylum)) # merging prevalence and abundance tables
View(merged_prev_by_PHYLUM_CHLD2_piglets_table)

# FAMILY level
piglets_df_family <- plyr::ddply(prevdf_piglets_CHLD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_piglets_total_reads))})
View(piglets_df_family)
colnames(piglets_df_family)[colnames(piglets_df_family) == "1"] <- "read count in piglets"
colnames(piglets_df_family)[colnames(piglets_df_family) == "2"] <- "abundance in piglets(%)"
View(piglets_df_family)

merged_prev_by_FAMILY_CHLD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_CHLD2_FAMILY_table,piglets_df_family)) # merging prevalence and abundance tables
View(merged_prev_by_FAMILY_CHLD2_piglets_table)

# GENUS level
piglets_df_genus <- plyr::ddply(prevdf_piglets_CHLD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/CHLD2_piglets_total_reads))})
View(piglets_df_genus)
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "1"] <- "read count in piglets"
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "2"] <- "abundance in piglets(%)"
View(piglets_df_genus)

merged_prev_by_GENUS_CHLD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_CHLD2_GENUS_table,piglets_df_genus)) # merging prevalence and abundance tables
View(merged_prev_by_GENUS_CHLD2_piglets_table)


## MERGED taxa level tables for CHLD2

library("dplyr"); packageVersion("dplyr") # needed for 'rounding' decimal values

# PHYLUM level
merged_phyla_CHLD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_PHYLUM_CHLD2_table,merged_prev_by_PHYLUM_CHLD2_table_mice,merged_prev_by_PHYLUM_CHLD2_piglets_table))
merged_phyla_CHLD2 <- merged_phyla_CHLD2 %>% mutate_if(is.numeric, round, 2) # rounding-off to 2 decimal places using 'dplyr'
View(merged_phyla_CHLD2)
write.table(merged_phyla_CHLD2, "CHLD2_prev_abundance_comparison_PHYLUM.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Family level
merged_families_CHLD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_FAMILY_CHLD2_table,merged_prev_by_FAMILY_CHLD2_table_mouse,merged_prev_by_FAMILY_CHLD2_piglets_table))
merged_families_CHLD2 <- merged_families_CHLD2 %>% mutate_if(is.numeric, round, 2)
View(merged_families_CHLD2)
write.table(merged_families_CHLD2, "CHLD2_prev_abundance_comparison_FAMILY.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Genus level
merged_genera_CHLD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_GENUS_CHLD2_table,merged_prev_by_GENUS_CHLD2_table_mouse,merged_prev_by_GENUS_CHLD2_piglets_table))
merged_genera_CHLD2 <- merged_genera_CHLD2 %>% mutate_if(is.numeric, round, 2)
View(merged_genera_CHLD2)
write.table(merged_genera_CHLD2, "CHLD2_prev_abundance_comparison_GENUS.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

```

**Adult donor**

```{r, echo=TRUE}
### for ADLT2 donor
human_donors_ADLT2_inocula_core_ps 

# core taxa found in all relevant donor samples
# at PHYLUM level
human_donors_ADLT2_inocula_table_PHYLUM <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_ADLT2_PHYLUM_table <- as.data.frame(human_donors_ADLT2_inocula_table_PHYLUM)
colnames(human_donors_ADLT2_PHYLUM_table)[colnames(human_donors_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_ADLT2_PHYLUM_table)[colnames(human_donors_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_PHYLUM_table)

# at FAMILY level
human_donors_ADLT2_inocula_table_FAMILY <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_ADLT2_inocula_table_FAMILY)
human_donors_ADLT2_FAMILY_table <- as.data.frame(human_donors_ADLT2_inocula_table_FAMILY)
colnames(human_donors_ADLT2_FAMILY_table)[colnames(human_donors_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_ADLT2_FAMILY_table)[colnames(human_donors_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_FAMILY_table)

# at GENUS level
human_donors_ADLT2_inocula_table_GENUS <- table(tax_table(human_donors_ADLT2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_ADLT2_inocula_table_GENUS)
human_donors_ADLT2_GENUS_table <- as.data.frame(human_donors_ADLT2_inocula_table_GENUS)
colnames(human_donors_ADLT2_GENUS_table)[colnames(human_donors_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_ADLT2_GENUS_table)[colnames(human_donors_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ADLT2_GENUS_table)

# Calculating the relative abundances of the taxa

# Compute prevalence of each feature, store as data.frame
prevdf_donor_ADLT2 = apply(X = otu_table(human_donors_ADLT2_inocula_core_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ADLT2_inocula_core_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_donor_ADLT2 = data.frame(Prevalence = prevdf_donor_ADLT2,
                    TotalAbundance = taxa_sums(human_donors_ADLT2_inocula_core_ps),
                    tax_table(human_donors_ADLT2_inocula_core_ps))
View(prevdf_donor_ADLT2)

ADLT2_donor_total_reads <- sum(otu_table(human_donors_ADLT2_inocula_ps)) # all reads from CHLD2 donor samples (including raeds from non-core ASVs)
ADLT2_donor_total_reads

# PHYLUM level
donor_df_phylum <- plyr::ddply(prevdf_donor_ADLT2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_donor_total_reads))})
View(donor_df_phylum)
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "1"] <- "read count in donors"
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "2"] <- "abundance in donors(%)"
View(donor_df_phylum)

merged_prev_by_PHYLUM_ADLT2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ADLT2_PHYLUM_table,donor_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_ADLT2_table) 

# FAMILY level
donor_df_family <- plyr::ddply(prevdf_donor_ADLT2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_donor_total_reads))})
View(donor_df_family)
colnames(donor_df_family)[colnames(donor_df_family) == "1"] <- "read count in donors"
colnames(donor_df_family)[colnames(donor_df_family) == "2"] <- "abundance in donors(%)"
View(donor_df_family)

merged_prev_by_FAMILY_ADLT2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ADLT2_FAMILY_table,donor_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_ADLT2_table) 


# GENUS level
donor_df_genus <- plyr::ddply(prevdf_donor_ADLT2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_donor_total_reads))})
View(donor_df_genus)
colnames(donor_df_genus)[colnames(donor_df_genus) == "1"] <- "read count in donors"
colnames(donor_df_genus)[colnames(donor_df_genus) == "2"] <- "abundance in donors(%)"
View(donor_df_genus)

merged_prev_by_GENUS_ADLT2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ADLT2_GENUS_table,donor_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_ADLT2_table) 


## for mice
# core_taxa persistently colonizing mice (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_ADLT2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_mice, ADLT2_HMA_mice_core_ps)
perst_colonizing_core_taxa_ADLT2_mice_ps

# at PHYLUM level
ADLT2_perst_core_taxa_mouse_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps)[, "Phylum"], exclude = NULL)
View(ADLT2_perst_core_taxa_mouse_table_PHYLUM)
core_taxa_mice_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_perst_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_PHYLUM_table)

# at FAMILY level
ADLT2_perst_core_taxa_mouse_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps)[, "Family"], exclude = NULL)
View(ADLT2_perst_core_taxa_mouse_table_FAMILY)
core_taxa_mice_ADLT2_FAMILY_table <- as.data.frame(ADLT2_perst_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_FAMILY_table)

# at GENUS level
ADLT2_perst_core_taxa_mouse_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps)[, "Genus"], exclude = NULL)
View(ADLT2_perst_core_taxa_mouse_table_GENUS)
core_taxa_mice_ADLT2_GENUS_table <- as.data.frame(ADLT2_perst_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_GENUS_table)

# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_ADLT2_mice_ps
# Compute prevalence of each feature, store as data.frame
prevdf_mouse_ADLT2 = apply(X = otu_table(perst_colonizing_core_taxa_ADLT2_mice_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_ADLT2_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_mouse_ADLT2 = data.frame(Prevalence = prevdf_mouse_ADLT2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_ADLT2_mice_ps),
                    tax_table(perst_colonizing_core_taxa_ADLT2_mice_ps))
View(prevdf_mouse_ADLT2)

ADLT2_mice_total_reads <- sum(otu_table(ADLT2_HMA_mice_ps))
ADLT2_mice_total_reads

# PHYLUM level
mouse_df_phylum <- plyr::ddply(prevdf_mouse_ADLT2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_mice_total_reads))})
View(mouse_df_phylum)
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "1"] <- "read count in mice"
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "2"] <- "abundance in mice(%)"
View(mouse_df_phylum)

merged_prev_by_PHYLUM_ADLT2_table_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ADLT2_PHYLUM_table,mouse_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_ADLT2_table_mice) 

# FAMILY level
mouse_df_family <- plyr::ddply(prevdf_mouse_ADLT2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_mice_total_reads))})
View(mouse_df_family)
colnames(mouse_df_family)[colnames(mouse_df_family) == "1"] <- "read count in mice"
colnames(mouse_df_family)[colnames(mouse_df_family) == "2"] <- "abundance in mice(%)"
View(mouse_df_family)

merged_prev_by_FAMILY_ADLT2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ADLT2_FAMILY_table,mouse_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_ADLT2_table_mouse) 

# GENUS level
mouse_df_genus <- plyr::ddply(prevdf_mouse_ADLT2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_mice_total_reads))})
View(mouse_df_genus)
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "1"] <- "read count in mice"
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "2"] <- "abundance in mice(%)"
View(mouse_df_genus)

merged_prev_by_GENUS_ADLT2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ADLT2_GENUS_table,mouse_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_ADLT2_table_mouse) 


## for piglets
# taxa persistently colonizing piglets (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_ADLT2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_piglets, ADLT2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_ADLT2_piglets_ps

# at PHYLUM level
ADLT2_perst_core_taxa_piglets_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps)[, "Phylum"], exclude = NULL)
View(ADLT2_perst_core_taxa_piglets_table_PHYLUM)
core_taxa_piglets_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_perst_core_taxa_piglets_table_PHYLUM)
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_PHYLUM_table)

# at FAMILY level
ADLT2_perst_core_taxa_piglets_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps)[, "Family"], exclude = NULL)
View(ADLT2_perst_core_taxa_piglets_table_FAMILY)
core_taxa_piglets_ADLT2_FAMILY_table <- as.data.frame(ADLT2_perst_core_taxa_piglets_table_FAMILY)
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_FAMILY_table)

# at GENUS level
ADLT2_perst_core_taxa_piglets_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps)[, "Genus"], exclude = NULL)
View(ADLT2_perst_core_taxa_piglets_table_GENUS)
core_taxa_piglets_ADLT2_GENUS_table <- as.data.frame(ADLT2_perst_core_taxa_piglets_table_GENUS)
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_GENUS_table)


# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_ADLT2_piglets_ps
# Compute prevalence of each feature, store as data.frame
prevdf_piglets_ADLT2 = apply(X = otu_table(perst_colonizing_core_taxa_ADLT2_piglets_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_ADLT2_piglets_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_piglets_ADLT2 = data.frame(Prevalence = prevdf_piglets_ADLT2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_ADLT2_piglets_ps),
                    tax_table(perst_colonizing_core_taxa_ADLT2_piglets_ps))
View(prevdf_piglets_ADLT2)

ADLT2_piglets_total_reads <- sum(otu_table(ADLT2_HMA_piglets_ps))
ADLT2_piglets_total_reads

# PHYLUM level
piglets_df_phylum <- plyr::ddply(prevdf_piglets_ADLT2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_piglets_total_reads))})
View(piglets_df_phylum)
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "1"] <- "read count in piglets"
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "2"] <- "abundance in piglets(%)"
View(piglets_df_phylum)

merged_prev_by_PHYLUM_ADLT2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ADLT2_PHYLUM_table,piglets_df_phylum)) # merging prevalence and abundance tables
View(merged_prev_by_PHYLUM_ADLT2_piglets_table)

# FAMILY level
piglets_df_family <- plyr::ddply(prevdf_piglets_ADLT2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_piglets_total_reads))})
View(piglets_df_family)
colnames(piglets_df_family)[colnames(piglets_df_family) == "1"] <- "read count in piglets"
colnames(piglets_df_family)[colnames(piglets_df_family) == "2"] <- "abundance in piglets(%)"
View(piglets_df_family)

merged_prev_by_FAMILY_ADLT2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ADLT2_FAMILY_table,piglets_df_family)) # merging prevalence and abundance tables
View(merged_prev_by_FAMILY_ADLT2_piglets_table)

# GENUS level
piglets_df_genus <- plyr::ddply(prevdf_piglets_ADLT2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ADLT2_piglets_total_reads))})
View(piglets_df_genus)
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "1"] <- "read count in piglets"
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "2"] <- "abundance in piglets(%)"
View(piglets_df_genus)

merged_prev_by_GENUS_ADLT2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ADLT2_GENUS_table,piglets_df_genus)) # merging prevalence and abundance tables
View(merged_prev_by_GENUS_ADLT2_piglets_table)


## MERGED taxa level tables for ADLT2

library("dplyr"); packageVersion("dplyr") # needed for 'rounding' decimal values

# PHYLUM level
merged_phyla_ADLT2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_PHYLUM_ADLT2_table,merged_prev_by_PHYLUM_ADLT2_table_mice,merged_prev_by_PHYLUM_ADLT2_piglets_table))
merged_phyla_ADLT2 <- merged_phyla_ADLT2 %>% mutate_if(is.numeric, round, 2) # rounding-off to 2 decimal places using 'dplyr'
View(merged_phyla_ADLT2)
write.table(merged_phyla_ADLT2, "ADLT2_prev_abundance_comparison_PHYLUM.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Family level
merged_families_ADLT2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_FAMILY_ADLT2_table,merged_prev_by_FAMILY_ADLT2_table_mouse,merged_prev_by_FAMILY_ADLT2_piglets_table))
merged_families_ADLT2 <- merged_families_ADLT2 %>% mutate_if(is.numeric, round, 2)
View(merged_families_ADLT2)
write.table(merged_families_ADLT2, "ADLT2_prev_abundance_comparison_FAMILY.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Genus level
merged_genera_ADLT2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_GENUS_ADLT2_table,merged_prev_by_GENUS_ADLT2_table_mouse,merged_prev_by_GENUS_ADLT2_piglets_table))
merged_genera_ADLT2 <- merged_genera_ADLT2 %>% mutate_if(is.numeric, round, 2)
View(merged_genera_ADLT2)
write.table(merged_genera_ADLT2, "ADLT2_prev_abundance_comparison_GENUS.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**Elderly donor**

```{r, echo=TRUE}

### for ELD2 donor
human_donors_ELD2_inocula_core_ps 

# core taxa found in all donor samples
# at PHYLUM level
human_donors_ELD2_inocula_table_PHYLUM <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Phylum"], exclude = NULL)
human_donors_ELD2_PHYLUM_table <- as.data.frame(human_donors_ELD2_inocula_table_PHYLUM)
colnames(human_donors_ELD2_PHYLUM_table)[colnames(human_donors_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(human_donors_ELD2_PHYLUM_table)[colnames(human_donors_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_PHYLUM_table)

# at FAMILY level
human_donors_ELD2_inocula_table_FAMILY <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Family"], exclude = NULL)
View(human_donors_ELD2_inocula_table_FAMILY)
human_donors_ELD2_FAMILY_table <- as.data.frame(human_donors_ELD2_inocula_table_FAMILY)
colnames(human_donors_ELD2_FAMILY_table)[colnames(human_donors_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(human_donors_ELD2_FAMILY_table)[colnames(human_donors_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_FAMILY_table)

# at GENUS level
human_donors_ELD2_inocula_table_GENUS <- table(tax_table(human_donors_ELD2_inocula_core_ps)[, "Genus"], exclude = NULL)
View(human_donors_ELD2_inocula_table_GENUS)
human_donors_ELD2_GENUS_table <- as.data.frame(human_donors_ELD2_inocula_table_GENUS)
colnames(human_donors_ELD2_GENUS_table)[colnames(human_donors_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(human_donors_ELD2_GENUS_table)[colnames(human_donors_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in donor"
View(human_donors_ELD2_GENUS_table)


# Calculating the relative abundances of the taxa

# Compute prevalence of each feature, store as data.frame
prevdf_donor_ELD2 = apply(X = otu_table(human_donors_ELD2_inocula_core_ps),
               MARGIN = ifelse(taxa_are_rows(human_donors_ELD2_inocula_core_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_donor_ELD2 = data.frame(Prevalence = prevdf_donor_ELD2,
                    TotalAbundance = taxa_sums(human_donors_ELD2_inocula_core_ps),
                    tax_table(human_donors_ELD2_inocula_core_ps))
View(prevdf_donor_ELD2)

ELD2_donor_total_reads <- sum(otu_table(human_donors_ELD2_inocula_ps)) # all reads from CHLD2 donor samples (including reads from non-core ASVs)
ELD2_donor_total_reads

# PHYLUM level
donor_df_phylum <- plyr::ddply(prevdf_donor_ELD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_donor_total_reads))})
View(donor_df_phylum)
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "1"] <- "read count in donors"
colnames(donor_df_phylum)[colnames(donor_df_phylum) == "2"] <- "abundance in donors(%)"
View(donor_df_phylum)

merged_prev_by_PHYLUM_ELD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ELD2_PHYLUM_table,donor_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_ELD2_table) 

# FAMILY level
donor_df_family <- plyr::ddply(prevdf_donor_ELD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_donor_total_reads))})
View(donor_df_family)
colnames(donor_df_family)[colnames(donor_df_family) == "1"] <- "read count in donors"
colnames(donor_df_family)[colnames(donor_df_family) == "2"] <- "abundance in donors(%)"
View(donor_df_family)

merged_prev_by_FAMILY_ELD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ELD2_FAMILY_table,donor_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_ELD2_table) 


# GENUS level
donor_df_genus <- plyr::ddply(prevdf_donor_ELD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_donor_total_reads))})
View(donor_df_genus)
colnames(donor_df_genus)[colnames(donor_df_genus) == "1"] <- "read count in donors"
colnames(donor_df_genus)[colnames(donor_df_genus) == "2"] <- "abundance in donors(%)"
View(donor_df_genus)

merged_prev_by_GENUS_ELD2_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ELD2_GENUS_table,donor_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_ELD2_table)

## for mice
# core_taxa persistently colonizing mice (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_ELD2_mice_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_mice, ELD2_HMA_mice_core_ps)
perst_colonizing_core_taxa_ELD2_mice_ps

# at PHYLUM level
ELD2_perst_core_taxa_mouse_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_ELD2_mice_ps)[, "Phylum"], exclude = NULL)
View(ELD2_perst_core_taxa_mouse_table_PHYLUM)
core_taxa_mice_ELD2_PHYLUM_table <- as.data.frame(ELD2_perst_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_PHYLUM_table)

# at FAMILY level
ELD2_perst_core_taxa_mouse_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_ELD2_mice_ps)[, "Family"], exclude = NULL)
View(ELD2_perst_core_taxa_mouse_table_FAMILY)
core_taxa_mice_ELD2_FAMILY_table <- as.data.frame(ELD2_perst_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_FAMILY_table)

# at GENUS level
ELD2_perst_core_taxa_mouse_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_ELD2_mice_ps)[, "Genus"], exclude = NULL)
View(ELD2_perst_core_taxa_mouse_table_GENUS)
core_taxa_mice_ELD2_GENUS_table <- as.data.frame(ELD2_perst_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_GENUS_table)

# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_ELD2_mice_ps
# Compute prevalence of each feature, store as data.frame
prevdf_mouse_ELD2 = apply(X = otu_table(perst_colonizing_core_taxa_ELD2_mice_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_ELD2_mice_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_mouse_ELD2 = data.frame(Prevalence = prevdf_mouse_ELD2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_ELD2_mice_ps),
                    tax_table(perst_colonizing_core_taxa_ELD2_mice_ps))
View(prevdf_mouse_ELD2)

ELD2_mice_total_reads <- sum(otu_table(ELD2_HMA_mice_ps))
ELD2_mice_total_reads

# PHYLUM level
mouse_df_phylum <- plyr::ddply(prevdf_mouse_ELD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_mice_total_reads))})
View(mouse_df_phylum)
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "1"] <- "read count in mice"
colnames(mouse_df_phylum)[colnames(mouse_df_phylum) == "2"] <- "abundance in mice(%)"
View(mouse_df_phylum)

merged_prev_by_PHYLUM_ELD2_table_mice <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ELD2_PHYLUM_table,mouse_df_phylum)) # merging prevalence data and abundance data tables
View(merged_prev_by_PHYLUM_ELD2_table_mice) 

# FAMILY level
mouse_df_family <- plyr::ddply(prevdf_mouse_ELD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_mice_total_reads))})
View(mouse_df_family)
colnames(mouse_df_family)[colnames(mouse_df_family) == "1"] <- "read count in mice"
colnames(mouse_df_family)[colnames(mouse_df_family) == "2"] <- "abundance in mice(%)"
View(mouse_df_family)

merged_prev_by_FAMILY_ELD2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ELD2_FAMILY_table,mouse_df_family)) # merging prevalence data and abundance data tables
View(merged_prev_by_FAMILY_ELD2_table_mouse) 

# GENUS level
mouse_df_genus <- plyr::ddply(prevdf_mouse_ELD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_mice_total_reads))})
View(mouse_df_genus)
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "1"] <- "read count in mice"
colnames(mouse_df_genus)[colnames(mouse_df_genus) == "2"] <- "abundance in mice(%)"
View(mouse_df_genus)

merged_prev_by_GENUS_ELD2_table_mouse <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ELD2_GENUS_table,mouse_df_genus)) # merging prevalence data and abundance data tables
View(merged_prev_by_GENUS_ELD2_table_mouse) 


## for piglets
# taxa persistently colonizing piglets (>=4 time points in more than 50% of the animals)
perst_colonizing_core_taxa_ELD2_piglets_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_piglets, ELD2_HMA_piglets_core_ps)
perst_colonizing_core_taxa_ELD2_piglets_ps

# at PHYLUM level
ELD2_perst_core_taxa_piglets_table_PHYLUM <- table(tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps)[, "Phylum"], exclude = NULL)
View(ELD2_perst_core_taxa_piglets_table_PHYLUM)
core_taxa_piglets_ELD2_PHYLUM_table <- as.data.frame(ELD2_perst_core_taxa_piglets_table_PHYLUM)
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_PHYLUM_table)

# at FAMILY level
ELD2_perst_core_taxa_piglets_table_FAMILY <- table(tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps)[, "Family"], exclude = NULL)
View(ELD2_perst_core_taxa_piglets_table_FAMILY)
core_taxa_piglets_ELD2_FAMILY_table <- as.data.frame(ELD2_perst_core_taxa_piglets_table_FAMILY)
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_FAMILY_table)

# at GENUS level
ELD2_perst_core_taxa_piglets_table_GENUS <- table(tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps)[, "Genus"], exclude = NULL)
View(ELD2_perst_core_taxa_piglets_table_GENUS)
core_taxa_piglets_ELD2_GENUS_table <- as.data.frame(ELD2_perst_core_taxa_piglets_table_GENUS)
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_GENUS_table)


# Calculating the relative abundances of the taxa

perst_colonizing_core_taxa_ELD2_piglets_ps
# Compute prevalence of each feature, store as data.frame
prevdf_piglets_ELD2 = apply(X = otu_table(perst_colonizing_core_taxa_ELD2_piglets_ps),
               MARGIN = ifelse(taxa_are_rows(perst_colonizing_core_taxa_ELD2_piglets_ps), yes = 1, no = 2),
               FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf_piglets_ELD2 = data.frame(Prevalence = prevdf_piglets_ELD2,
                    TotalAbundance = taxa_sums(perst_colonizing_core_taxa_ELD2_piglets_ps),
                    tax_table(perst_colonizing_core_taxa_ELD2_piglets_ps))
View(prevdf_piglets_ELD2)

ELD2_piglets_total_reads <- sum(otu_table(ELD2_HMA_piglets_ps))
ELD2_piglets_total_reads

# PHYLUM level
piglets_df_phylum <- plyr::ddply(prevdf_piglets_ELD2, "Phylum", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_piglets_total_reads))})
View(piglets_df_phylum)
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "1"] <- "read count in piglets"
colnames(piglets_df_phylum)[colnames(piglets_df_phylum) == "2"] <- "abundance in piglets(%)"
View(piglets_df_phylum)

merged_prev_by_PHYLUM_ELD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ELD2_PHYLUM_table,piglets_df_phylum)) # merging prevalence and abundance tables
View(merged_prev_by_PHYLUM_ELD2_piglets_table)

# FAMILY level
piglets_df_family <- plyr::ddply(prevdf_piglets_ELD2, "Family", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_piglets_total_reads))})
View(piglets_df_family)
colnames(piglets_df_family)[colnames(piglets_df_family) == "1"] <- "read count in piglets"
colnames(piglets_df_family)[colnames(piglets_df_family) == "2"] <- "abundance in piglets(%)"
View(piglets_df_family)

merged_prev_by_FAMILY_ELD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ELD2_FAMILY_table,piglets_df_family)) # merging prevalence and abundance tables
View(merged_prev_by_FAMILY_ELD2_piglets_table)

# GENUS level
piglets_df_genus <- plyr::ddply(prevdf_piglets_ELD2, "Genus", function(df1){cbind(sum(df1$TotalAbundance), sum(df1$TotalAbundance*100/ELD2_piglets_total_reads))})
View(piglets_df_genus)
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "1"] <- "read count in piglets"
colnames(piglets_df_genus)[colnames(piglets_df_genus) == "2"] <- "abundance in piglets(%)"
View(piglets_df_genus)

merged_prev_by_GENUS_ELD2_piglets_table <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ELD2_GENUS_table,piglets_df_genus)) # merging prevalence and abundance tables
View(merged_prev_by_GENUS_ELD2_piglets_table)


## MERGED taxa level tables for ELD2

library("dplyr"); packageVersion("dplyr") # needed for 'rounding' decimal values

# PHYLUM level
merged_phyla_ELD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_PHYLUM_ELD2_table,merged_prev_by_PHYLUM_ELD2_table_mice,merged_prev_by_PHYLUM_ELD2_piglets_table))
merged_phyla_ELD2 <- merged_phyla_ELD2 %>% mutate_if(is.numeric, round, 2) # rounding-off to 2 decimal places using 'dplyr'
View(merged_phyla_ELD2)
write.table(merged_phyla_ELD2, "ELD2_prev_abundance_comparison_PHYLUM.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Family level
merged_families_ELD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_FAMILY_ELD2_table,merged_prev_by_FAMILY_ELD2_table_mouse,merged_prev_by_FAMILY_ELD2_piglets_table))
merged_families_ELD2 <- merged_families_ELD2 %>% mutate_if(is.numeric, round, 2)
View(merged_families_ELD2)
write.table(merged_families_ELD2, "ELD2_prev_abundance_comparison_FAMILY.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Genus level
merged_genera_ELD2 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(merged_prev_by_GENUS_ELD2_table,merged_prev_by_GENUS_ELD2_table_mouse,merged_prev_by_GENUS_ELD2_piglets_table))
merged_genera_ELD2 <- merged_genera_ELD2 %>% mutate_if(is.numeric, round, 2)
View(merged_genera_ELD2)
write.table(merged_genera_ELD2, "ELD2_prev_abundance_comparison_GENUS.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


##**Taxonomic breakdown of core persistent colonizers at the Phylum, Family, and Genus levels**##

**INF donor**

```{r, echo=TRUE}

## for mice
perst_colonizing_core_taxa_INF2_mice
INF2_HMA_mice_ps
INF2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_mice,INF2_HMA_mice_ps)
INF2_perst_colonizers_ps

# at PHYLUM level
INF2_perst_core_taxa_PHYLUM <- table(tax_table(INF2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
INF2_perst_core_taxa_PHYLUM_table_mice <- as.data.frame(INF2_perst_core_taxa_PHYLUM)
colnames(INF2_perst_core_taxa_PHYLUM_table_mice)[colnames(INF2_perst_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(INF2_perst_core_taxa_PHYLUM_table_mice)[colnames(INF2_perst_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_PHYLUM_table_mice)
write.table(INF2_perst_core_taxa_PHYLUM_table_mice, "INF2_perst_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
INF2_perst_core_taxa_FAMILY <- table(tax_table(INF2_perst_colonizers_ps)[, "Family"], exclude = NULL)
INF2_perst_core_taxa_FAMILY_table_mice <- as.data.frame(INF2_perst_core_taxa_FAMILY)
colnames(INF2_perst_core_taxa_FAMILY_table_mice)[colnames(INF2_perst_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(INF2_perst_core_taxa_FAMILY_table_mice)[colnames(INF2_perst_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_FAMILY_table_mice)
write.table(INF2_perst_core_taxa_FAMILY_table_mice, "INF2_perst_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
INF2_perst_core_taxa_GENUS <- table(tax_table(INF2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
INF2_perst_core_taxa_GENUS_table_mice <- as.data.frame(INF2_perst_core_taxa_GENUS)
colnames(INF2_perst_core_taxa_GENUS_table_mice)[colnames(INF2_perst_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(INF2_perst_core_taxa_GENUS_table_mice)[colnames(INF2_perst_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_GENUS_table_mice)
write.table(INF2_perst_core_taxa_GENUS_table_mice, "INF2_perst_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_colonizing_core_taxa_INF2_piglets
INF2_HMA_piglets_ps
INF2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_INF2_piglets,INF2_HMA_piglets_ps)
INF2_perst_colonizers_ps

# at PHYLUM level
INF2_perst_core_taxa_PHYLUM <- table(tax_table(INF2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
INF2_perst_core_taxa_PHYLUM_table_piglets <- as.data.frame(INF2_perst_core_taxa_PHYLUM)
colnames(INF2_perst_core_taxa_PHYLUM_table_piglets)[colnames(INF2_perst_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(INF2_perst_core_taxa_PHYLUM_table_piglets)[colnames(INF2_perst_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_PHYLUM_table_piglets)
write.table(INF2_perst_core_taxa_PHYLUM_table_piglets, "INF2_perst_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
INF2_perst_core_taxa_FAMILY <- table(tax_table(INF2_perst_colonizers_ps)[, "Family"], exclude = NULL)
INF2_perst_core_taxa_FAMILY_table_piglets <- as.data.frame(INF2_perst_core_taxa_FAMILY)
colnames(INF2_perst_core_taxa_FAMILY_table_piglets)[colnames(INF2_perst_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(INF2_perst_core_taxa_FAMILY_table_piglets)[colnames(INF2_perst_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_FAMILY_table_piglets)
write.table(INF2_perst_core_taxa_FAMILY_table_piglets, "INF2_perst_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
INF2_perst_core_taxa_GENUS <- table(tax_table(INF2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
INF2_perst_core_taxa_GENUS_table_piglets <- as.data.frame(INF2_perst_core_taxa_GENUS)
colnames(INF2_perst_core_taxa_GENUS_table_piglets)[colnames(INF2_perst_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(INF2_perst_core_taxa_GENUS_table_piglets)[colnames(INF2_perst_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_core_taxa_GENUS_table_piglets)
write.table(INF2_perst_core_taxa_GENUS_table_piglets, "INF2_perst_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**CHLD donor**

```{r, echo=TRUE}

## for mice
perst_colonizing_core_taxa_CHLD2_mice
CHLD2_HMA_mice_ps
CHLD2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_mice,CHLD2_HMA_mice_ps)
CHLD2_perst_colonizers_ps

# at PHYLUM level
CHLD2_perst_core_taxa_PHYLUM <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
CHLD2_perst_core_taxa_PHYLUM_table_mice <- as.data.frame(CHLD2_perst_core_taxa_PHYLUM)
colnames(CHLD2_perst_core_taxa_PHYLUM_table_mice)[colnames(CHLD2_perst_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(CHLD2_perst_core_taxa_PHYLUM_table_mice)[colnames(CHLD2_perst_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_PHYLUM_table_mice)
write.table(CHLD2_perst_core_taxa_PHYLUM_table_mice, "CHLD2_perst_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
CHLD2_perst_core_taxa_FAMILY <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Family"], exclude = NULL)
CHLD2_perst_core_taxa_FAMILY_table_mice <- as.data.frame(CHLD2_perst_core_taxa_FAMILY)
colnames(CHLD2_perst_core_taxa_FAMILY_table_mice)[colnames(CHLD2_perst_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(CHLD2_perst_core_taxa_FAMILY_table_mice)[colnames(CHLD2_perst_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_FAMILY_table_mice)
write.table(CHLD2_perst_core_taxa_FAMILY_table_mice, "CHLD2_perst_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
CHLD2_perst_core_taxa_GENUS <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
CHLD2_perst_core_taxa_GENUS_table_mice <- as.data.frame(CHLD2_perst_core_taxa_GENUS)
colnames(CHLD2_perst_core_taxa_GENUS_table_mice)[colnames(CHLD2_perst_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(CHLD2_perst_core_taxa_GENUS_table_mice)[colnames(CHLD2_perst_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_GENUS_table_mice)
write.table(CHLD2_perst_core_taxa_GENUS_table_mice, "CHLD2_perst_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_colonizing_core_taxa_CHLD2_piglets
CHLD2_HMA_piglets_ps
CHLD2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_CHLD2_piglets,CHLD2_HMA_piglets_ps)
CHLD2_perst_colonizers_ps

# at PHYLUM level
CHLD2_perst_core_taxa_PHYLUM <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
CHLD2_perst_core_taxa_PHYLUM_table_piglets <- as.data.frame(CHLD2_perst_core_taxa_PHYLUM)
colnames(CHLD2_perst_core_taxa_PHYLUM_table_piglets)[colnames(CHLD2_perst_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(CHLD2_perst_core_taxa_PHYLUM_table_piglets)[colnames(CHLD2_perst_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_PHYLUM_table_piglets)
write.table(CHLD2_perst_core_taxa_PHYLUM_table_piglets, "CHLD2_perst_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
CHLD2_perst_core_taxa_FAMILY <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Family"], exclude = NULL)
CHLD2_perst_core_taxa_FAMILY_table_piglets <- as.data.frame(CHLD2_perst_core_taxa_FAMILY)
colnames(CHLD2_perst_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(CHLD2_perst_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_FAMILY_table_piglets)
write.table(CHLD2_perst_core_taxa_FAMILY_table_piglets, "CHLD2_perst_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
CHLD2_perst_core_taxa_GENUS <- table(tax_table(CHLD2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
CHLD2_perst_core_taxa_GENUS_table_piglets <- as.data.frame(CHLD2_perst_core_taxa_GENUS)
colnames(CHLD2_perst_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(CHLD2_perst_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_core_taxa_GENUS_table_piglets)
write.table(CHLD2_perst_core_taxa_GENUS_table_piglets, "CHLD2_perst_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**ADLT donor**

```{r, echo=TRUE}

## for mice
perst_colonizing_core_taxa_ADLT2_mice
ADLT2_HMA_mice_ps
ADLT2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_mice,ADLT2_HMA_mice_ps)
ADLT2_perst_colonizers_ps

# at PHYLUM level
ADLT2_perst_core_taxa_PHYLUM <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
ADLT2_perst_core_taxa_PHYLUM_table_mice <- as.data.frame(ADLT2_perst_core_taxa_PHYLUM)
colnames(ADLT2_perst_core_taxa_PHYLUM_table_mice)[colnames(ADLT2_perst_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(ADLT2_perst_core_taxa_PHYLUM_table_mice)[colnames(ADLT2_perst_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_PHYLUM_table_mice)
write.table(ADLT2_perst_core_taxa_PHYLUM_table_mice, "ADLT2_perst_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ADLT2_perst_core_taxa_FAMILY <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Family"], exclude = NULL)
ADLT2_perst_core_taxa_FAMILY_table_mice <- as.data.frame(ADLT2_perst_core_taxa_FAMILY)
colnames(ADLT2_perst_core_taxa_FAMILY_table_mice)[colnames(ADLT2_perst_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(ADLT2_perst_core_taxa_FAMILY_table_mice)[colnames(ADLT2_perst_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_FAMILY_table_mice)
write.table(ADLT2_perst_core_taxa_FAMILY_table_mice, "ADLT2_perst_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ADLT2_perst_core_taxa_GENUS <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
ADLT2_perst_core_taxa_GENUS_table_mice <- as.data.frame(ADLT2_perst_core_taxa_GENUS)
colnames(ADLT2_perst_core_taxa_GENUS_table_mice)[colnames(ADLT2_perst_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(ADLT2_perst_core_taxa_GENUS_table_mice)[colnames(ADLT2_perst_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_GENUS_table_mice)
write.table(ADLT2_perst_core_taxa_GENUS_table_mice, "ADLT2_perst_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_colonizing_core_taxa_ADLT2_piglets
ADLT2_HMA_piglets_ps
ADLT2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_ADLT2_piglets,ADLT2_HMA_piglets_ps)
ADLT2_perst_colonizers_ps

# at PHYLUM level
ADLT2_perst_core_taxa_PHYLUM <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
ADLT2_perst_core_taxa_PHYLUM_table_piglets <- as.data.frame(ADLT2_perst_core_taxa_PHYLUM)
colnames(ADLT2_perst_core_taxa_PHYLUM_table_piglets)[colnames(ADLT2_perst_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(ADLT2_perst_core_taxa_PHYLUM_table_piglets)[colnames(ADLT2_perst_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_PHYLUM_table_piglets)
write.table(ADLT2_perst_core_taxa_PHYLUM_table_piglets, "ADLT2_perst_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ADLT2_perst_core_taxa_FAMILY <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Family"], exclude = NULL)
ADLT2_perst_core_taxa_FAMILY_table_piglets <- as.data.frame(ADLT2_perst_core_taxa_FAMILY)
colnames(ADLT2_perst_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ADLT2_perst_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_FAMILY_table_piglets)
write.table(ADLT2_perst_core_taxa_FAMILY_table_piglets, "ADLT2_perst_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ADLT2_perst_core_taxa_GENUS <- table(tax_table(ADLT2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
ADLT2_perst_core_taxa_GENUS_table_piglets <- as.data.frame(ADLT2_perst_core_taxa_GENUS)
colnames(ADLT2_perst_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ADLT2_perst_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_core_taxa_GENUS_table_piglets)
write.table(ADLT2_perst_core_taxa_GENUS_table_piglets, "ADLT2_perst_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**ELD donor**

```{r, echo=TRUE}

## for mice
perst_colonizing_core_taxa_ELD2_mice
ELD2_HMA_mice_ps
ELD2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_mice,ELD2_HMA_mice_ps)
ELD2_perst_colonizers_ps

# at PHYLUM level
ELD2_perst_core_taxa_PHYLUM <- table(tax_table(ELD2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
ELD2_perst_core_taxa_PHYLUM_table_mice <- as.data.frame(ELD2_perst_core_taxa_PHYLUM)
colnames(ELD2_perst_core_taxa_PHYLUM_table_mice)[colnames(ELD2_perst_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(ELD2_perst_core_taxa_PHYLUM_table_mice)[colnames(ELD2_perst_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_PHYLUM_table_mice)
write.table(ELD2_perst_core_taxa_PHYLUM_table_mice, "ELD2_perst_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ELD2_perst_core_taxa_FAMILY <- table(tax_table(ELD2_perst_colonizers_ps)[, "Family"], exclude = NULL)
ELD2_perst_core_taxa_FAMILY_table_mice <- as.data.frame(ELD2_perst_core_taxa_FAMILY)
colnames(ELD2_perst_core_taxa_FAMILY_table_mice)[colnames(ELD2_perst_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(ELD2_perst_core_taxa_FAMILY_table_mice)[colnames(ELD2_perst_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_FAMILY_table_mice)
write.table(ELD2_perst_core_taxa_FAMILY_table_mice, "ELD2_perst_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ELD2_perst_core_taxa_GENUS <- table(tax_table(ELD2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
ELD2_perst_core_taxa_GENUS_table_mice <- as.data.frame(ELD2_perst_core_taxa_GENUS)
colnames(ELD2_perst_core_taxa_GENUS_table_mice)[colnames(ELD2_perst_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(ELD2_perst_core_taxa_GENUS_table_mice)[colnames(ELD2_perst_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_GENUS_table_mice)
write.table(ELD2_perst_core_taxa_GENUS_table_mice, "ELD2_perst_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_colonizing_core_taxa_ELD2_piglets
ELD2_HMA_piglets_ps
ELD2_perst_colonizers_ps <- prune_taxa(perst_colonizing_core_taxa_ELD2_piglets,ELD2_HMA_piglets_ps)
ELD2_perst_colonizers_ps

# at PHYLUM level
ELD2_perst_core_taxa_PHYLUM <- table(tax_table(ELD2_perst_colonizers_ps)[, "Phylum"], exclude = NULL)
ELD2_perst_core_taxa_PHYLUM_table_piglets <- as.data.frame(ELD2_perst_core_taxa_PHYLUM)
colnames(ELD2_perst_core_taxa_PHYLUM_table_piglets)[colnames(ELD2_perst_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(ELD2_perst_core_taxa_PHYLUM_table_piglets)[colnames(ELD2_perst_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_PHYLUM_table_piglets)
write.table(ELD2_perst_core_taxa_PHYLUM_table_piglets, "ELD2_perst_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ELD2_perst_core_taxa_FAMILY <- table(tax_table(ELD2_perst_colonizers_ps)[, "Family"], exclude = NULL)
ELD2_perst_core_taxa_FAMILY_table_piglets <- as.data.frame(ELD2_perst_core_taxa_FAMILY)
colnames(ELD2_perst_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ELD2_perst_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_FAMILY_table_piglets)
write.table(ELD2_perst_core_taxa_FAMILY_table_piglets, "ELD2_perst_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ELD2_perst_core_taxa_GENUS <- table(tax_table(ELD2_perst_colonizers_ps)[, "Genus"], exclude = NULL)
ELD2_perst_core_taxa_GENUS_table_piglets <- as.data.frame(ELD2_perst_core_taxa_GENUS)
colnames(ELD2_perst_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ELD2_perst_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_core_taxa_GENUS_table_piglets)
write.table(ELD2_perst_core_taxa_GENUS_table_piglets, "ELD2_perst_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```




## **'Global' alpha diversity variation among donors and HMA animals** ##


```{r,echo=TRUE}

## for INF2 early
alpha_div_INF2_ps <- subset_samples(merged_pig_mouse_comp_ps_master_no_Cyano, Donor == 'INF2')
alpha_div_INF2_ps

# only keeping inoculae
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, SampleType != 'cecal')
alpha_div_INF2_ps

# removing late inoculation piglets
alpha_div_INF2_ps_early <- subset_samples(alpha_div_INF2_ps, Timingofinocu !='late' | is.na(Timingofinocu))
alpha_div_INF2_ps_early

# for early inoculation piglets
plot_richness(alpha_div_INF2_ps_early, x = 'Species', measures = 'Shannon')

results = estimate_richness(alpha_div_INF2_ps_early, measures = 'Shannon')
View(results)
d = sample_data(alpha_div_INF2_ps_early)

# calculate t-test
mice = results[d[,'Species'] == 'mouse',]
View(mice)
piglets = results[d[,'Species'] == 'piglet',]
donors = results[d[,'Species'] == 'human',]
wilcox.test(mice, piglets)
t.test(mice, piglets)

## for INF2 late

# removing early inoculation piglets
alpha_div_INF2_ps_late <- subset_samples(alpha_div_INF2_ps, Timingofinocu !='early' | is.na(Timingofinocu))
alpha_div_INF2_ps_late

# for early inoculation piglets
plot_richness(alpha_div_INF2_ps_late, x = 'Species', measures = 'Shannon')

results = estimate_richness(alpha_div_INF2_ps_late, measures = 'Shannon')
View(results)
d = sample_data(alpha_div_INF2_ps_late)

# calculate t-test
mice = results[d[,'Species'] == 'mouse',]
View(mice)
piglets = results[d[,'Species'] == 'piglet',]
donors = results[d[,'Species'] == 'human',]
wilcox.test(mice, piglets)
t.test(mice, piglets)


## for CHLD2 
alpha_div_CHLD2_ps <- subset_samples(merged_pig_mouse_comp_ps_master_no_Cyano, Donor == 'CHLD2')
alpha_div_CHLD2_ps

# only keeping inoculae
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, SampleType != 'cecal')
alpha_div_CHLD2_ps

plot_richness(alpha_div_CHLD2_ps, x = 'Species', measures = 'Shannon')

results = estimate_richness(alpha_div_CHLD2_ps, measures = 'Shannon')
View(results)
d = sample_data(alpha_div_CHLD2_ps)

# calculate t-test
mice = results[d[,'Species'] == 'mouse',]
View(mice)
piglets = results[d[,'Species'] == 'piglet',]
donors = results[d[,'Species'] == 'human',]
wilcox.test(mice, piglets)
t.test(mice, piglets)


## for ADLT2 
alpha_div_ADLT2_ps <- subset_samples(merged_pig_mouse_comp_ps_master_no_Cyano, Donor == 'ADLT2')
alpha_div_ADLT2_ps

# only keeping inoculae
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, SampleType != 'cecal')
alpha_div_ADLT2_ps

plot_richness(alpha_div_ADLT2_ps, x = 'Species', measures = 'Shannon')

results = estimate_richness(alpha_div_ADLT2_ps, measures = 'Shannon')
View(results)
d = sample_data(alpha_div_ADLT2_ps)

# calculate t-test
mice = results[d[,'Species'] == 'mouse',]
View(mice)
piglets = results[d[,'Species'] == 'piglet',]
donors = results[d[,'Species'] == 'human',]
wilcox.test(mice, piglets)
t.test(mice, piglets)


## for ELD2 
alpha_div_ELD2_ps <- subset_samples(merged_pig_mouse_comp_ps_master_no_Cyano, Donor == 'ELD2')
alpha_div_ELD2_ps

# only keeping inoculae
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, SampleType != 'cecal')
alpha_div_ELD2_ps

plot_richness(alpha_div_ELD2_ps, x = 'Species', measures = 'Shannon')

results = estimate_richness(alpha_div_ELD2_ps, measures = 'Shannon')
View(results)
d = sample_data(alpha_div_ELD2_ps)

# calculate t-test
mice = results[d[,'Species'] == 'mouse',]
View(mice)
piglets = results[d[,'Species'] == 'piglet',]
donors = results[d[,'Species'] == 'human',]
wilcox.test(mice, piglets)
t.test(mice, piglets)
```


##**Weekly variation in alpha diversity for each donor group among the HMA animals**##

Since the previous analyses have shown that there is a time effect when it comes to the establishment of taxa, we would like to see how the alpha diversity (as determined by the Shannon Index)  varies with sampling time point for each HMA animal model within each donor

```{r, echo=TRUE}

# EXPORT mapping file and modify it so that it can be better used for alpha diversity analysis
write.table(sample_data(merged_pig_mouse_comp_ps_master_no_Cyano), 'alpha_div_mapping_file.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# need to import modified mapping file and then remake a new phyloseq object

alpha_mapping_file <- read.table("alpha_div_mapping_file.txt", sep = "\t", header = TRUE, comment.char = "")
row.names(alpha_mapping_file) <- as.character(alpha_mapping_file[,1])

# Need to merge with a minimally-filtered ASV table
alpha_div_master_ps <- phyloseq(otu_table(merged_seqtab_all_runs_nochim, taxa_are_rows = FALSE), sample_data(alpha_mapping_file), tax_table(taxTab))
alpha_div_master_ps
taxa_names(alpha_div_master_ps) <- paste0("ASV_", seq(ntaxa(alpha_div_master_ps)))

alpha_div_master_ps <- merge_phyloseq(alpha_div_master_ps, phy_tree(phylo_tree))
alpha_div_master_ps

remov_kingdoms <- c('Archaea', 'Eukaryota')
alpha_div_master_ps_1 <- subset_taxa(alpha_div_master_ps, !Kingdom %in% remov_kingdoms & (Family != 'Mitochondria' | is.na(Family))) 
alpha_div_master_ps_1
alpha_div_master_ps_2 <- subset_taxa(alpha_div_master_ps_1, Kingdom == 'Bacteria') # removed 4 ASVs classified as 'NA' at the kingdom level
alpha_div_master_ps_2

Cyano_ps <- subset_taxa(alpha_div_master_ps_2, Phylum == 'Cyanobacteria') # Isolating the Cyanobacterial ASVs
remov_Cyano_ps <- subset_taxa(Cyano_ps, Class != 'Melainabacteria') # identifying Cyanobacterial ASVs that are not 'Melainabacteria'
Cyano_to_remov <- taxa_names(remov_Cyano_ps)
Cyano_to_remov <- append(Cyano_to_remov, 'ASV_2842') # ASV_2842 has 'NA' as Class so we'll remove that one as well

# Filtering out the non-Melainabacterial Cyanobacterial ASVs
alltaxa = taxa_names(alpha_div_master_ps_2)
goodtaxa = alltaxa[!(alltaxa %in% Cyano_to_remov)]
alpha_div_master_all_filt_ps <- prune_taxa(goodtaxa, alpha_div_master_ps_2)
alpha_div_master_all_filt_ps
```

**INF2 donor**

```{r, echo=TRUE}
## for INF2 HMA mice
alpha_div_INF2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'INF2')
alpha_div_INF2_ps

# only keeping inoculae
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, SampleType != 'cecal')
alpha_div_INF2_ps

# removing piglet samples
alpha_div_INF2_mice_ps <- subset_samples(alpha_div_INF2_ps, Species != 'piglet')
alpha_div_INF2_mice_ps

# removing euthanization samples
alpha_div_INF2_mice_ps <- subset_samples(alpha_div_INF2_mice_ps, Description != '24h_before_euthanization')
alpha_div_INF2_mice_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_INF2_mice_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

# Looks like there are two clear outliers for the 48hrs time point. These samples were identified as samples '29' and '37'

# Let's remove these and redo the plot
alpha_div_INF2_mice_ps_no_outliers <- subset_samples(alpha_div_INF2_mice_ps, SampleID != c('29', '37'))
alpha_div_INF2_mice_ps_no_outliers

p3 <- plot_richness(alpha_div_INF2_mice_ps_no_outliers, x = 'Alpha_Desc', measures = 'Shannon')
p4 <- p3 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot()
p4

results = estimate_richness(alpha_div_INF2_mice_ps_no_outliers, measures = 'Shannon')
d = sample_data(alpha_div_INF2_mice_ps_no_outliers)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]

View(two_days)
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

library('ggpubr')

p5 <- p4 + stat_compare_means(comparisons = my_comparisons, label = "p.signif" ) 
p5


## For early inoculation piglets

# removing mouse samples
alpha_div_INF2_piglets_ps <- subset_samples(alpha_div_INF2_ps, Species != 'mouse')
alpha_div_INF2_piglets_ps

# removing late inoculation piglets
alpha_div_INF2_early_piglets_ps <- subset_samples(alpha_div_INF2_piglets_ps, Timingofinocu != 'late' | is.na(Timingofinocu))
alpha_div_INF2_early_piglets_ps

# remove samples at euthanization
alpha_div_INF2_early_piglets_ps <- subset_samples(alpha_div_INF2_early_piglets_ps, Description != 'at_euthanization')
alpha_div_INF2_early_piglets_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_INF2_early_piglets_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_INF2_early_piglets_ps, measures = 'Shannon')
d = sample_data(alpha_div_INF2_early_piglets_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3                                                                           


## For late inoculation piglets

# removing early inoculation piglets
alpha_div_INF2_late_piglets_ps <- subset_samples(alpha_div_INF2_piglets_ps, Timingofinocu != 'early' | is.na(Timingofinocu))
alpha_div_INF2_late_piglets_ps

# remove samples at euthanization
alpha_div_INF2_late_piglets_ps <- subset_samples(alpha_div_INF2_late_piglets_ps, Description != 'at_euthanization')
alpha_div_INF2_late_piglets_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_INF2_late_piglets_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_INF2_late_piglets_ps, measures = 'Shannon')
d = sample_data(alpha_div_INF2_late_piglets_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3
```


**CHLD2 donor**

```{r,echo=TRUE}

alpha_div_CHLD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'CHLD2')
alpha_div_CHLD2_ps

# only keeping inoculae
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, SampleType != 'cecal')
alpha_div_CHLD2_ps

# removing piglet samples
alpha_div_CHLD2_mice_ps <- subset_samples(alpha_div_CHLD2_ps, Species != 'piglet')
alpha_div_CHLD2_mice_ps

# removing euthanization samples
alpha_div_CHLD2_mice_ps <- subset_samples(alpha_div_CHLD2_mice_ps, Description != '24h_before_euthanization')
alpha_div_CHLD2_mice_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_CHLD2_mice_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_CHLD2_mice_ps, measures = 'Shannon')
d = sample_data(alpha_div_CHLD2_mice_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3


## for CHLD2 piglets
alpha_div_CHLD2_ps

# removing mouse samples
alpha_div_CHLD2_piglets_ps <- subset_samples(alpha_div_CHLD2_ps, Species != 'mouse')
alpha_div_CHLD2_piglets_ps

# removing euthanization samples
alpha_div_CHLD2_piglets_ps <- subset_samples(alpha_div_CHLD2_piglets_ps, Description != 'at_euthanization')
alpha_div_CHLD2_piglets_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_CHLD2_piglets_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_CHLD2_piglets_ps, measures = 'Shannon')
d = sample_data(alpha_div_CHLD2_piglets_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3
```


**ADLT2 donor**

```{r, echo=TRUE}
alpha_div_ADLT2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ADLT2')
alpha_div_ADLT2_ps

# only keeping inoculae
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, SampleType != 'cecal')
alpha_div_ADLT2_ps

# removing piglet samples
alpha_div_ADLT2_mice_ps <- subset_samples(alpha_div_ADLT2_ps, Species != 'piglet')
alpha_div_ADLT2_mice_ps

# removing euthanization samples
alpha_div_ADLT2_mice_ps <- subset_samples(alpha_div_ADLT2_mice_ps, Description != '24h_before_euthanization')
alpha_div_ADLT2_mice_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_ADLT2_mice_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_ADLT2_mice_ps, measures = 'Shannon')
d = sample_data(alpha_div_ADLT2_mice_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons, label = "p.adj") 
p3



## for ADLT2 piglets
alpha_div_ADLT2_ps

# removing mouse samples
alpha_div_ADLT2_piglets_ps <- subset_samples(alpha_div_ADLT2_ps, Species != 'mouse')
alpha_div_ADLT2_piglets_ps

# removing euthanization samples
alpha_div_ADLT2_piglets_ps <- subset_samples(alpha_div_ADLT2_piglets_ps, Description != 'at_euthanization')
alpha_div_ADLT2_piglets_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_ADLT2_piglets_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_ADLT2_piglets_ps, measures = 'Shannon')
d = sample_data(alpha_div_ADLT2_piglets_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons, label = "p.adj") 
p3
```


**ELD2 donor**

```{r, echo=TRUE}
alpha_div_ELD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ELD2')
alpha_div_ELD2_ps

# only keeping inoculae
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, SampleType != 'cecal')
alpha_div_ELD2_ps

# removing piglet samples
alpha_div_ELD2_mice_ps <- subset_samples(alpha_div_ELD2_ps, Species != 'piglet')
alpha_div_ELD2_mice_ps

# removing euthanization samples
alpha_div_ELD2_mice_ps <- subset_samples(alpha_div_ELD2_mice_ps, Description != '24h_before_euthanization')
alpha_div_ELD2_mice_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_ELD2_mice_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_ELD2_mice_ps, measures = 'Shannon')
d = sample_data(alpha_div_ELD2_mice_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3


## for ELD2 piglets
alpha_div_ELD2_ps

# removing mouse samples
alpha_div_ELD2_piglets_ps <- subset_samples(alpha_div_ELD2_ps, Species != 'mouse')
alpha_div_ELD2_piglets_ps

# removing euthanization samples
alpha_div_ELD2_piglets_ps <- subset_samples(alpha_div_ELD2_piglets_ps, Description != 'at_euthanization')
alpha_div_ELD2_piglets_ps

## Performing alpha diversity analyses

# plotting richness
p1 <- plot_richness(alpha_div_ELD2_piglets_ps, x = 'Alpha_Desc', measures = 'Shannon')
p1 
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + geom_boxplot(outlier.shape = NA)
p2

results = estimate_richness(alpha_div_ELD2_piglets_ps, measures = 'Shannon')
d = sample_data(alpha_div_ELD2_piglets_ps)

# perform wilcoxon test
donors = results[d[,'Alpha_Desc'] == 'donor_inocula',]
two_days = results[d[,'Alpha_Desc'] == '48hrs_post_inoc',]
week_1 = results[d[,'Alpha_Desc'] == '1week_post_inoc',]
week_2 = results[d[,'Alpha_Desc'] == '2weeks_post_inoc',]
week_3 = results[d[,'Alpha_Desc'] == '3weeks_post_inoc',]
week_4 = results[d[,'Alpha_Desc'] == '4weeks_post_inoc',]
week_5 = results[d[,'Alpha_Desc'] == '5weeks_post_inoc',]
final =  results[d[,'Alpha_Desc'] == 'Final_collection',]

wilcox.test(donors, two_days)
wilcox.test(donors, week_1)
wilcox.test(donors, week_2)
wilcox.test(donors, week_3)
wilcox.test(donors, week_4)
wilcox.test(donors, week_5)
wilcox.test(donors, final)

my_comparisons <- list(c('donor_inocula', '48hrs_post_inoc'), c('donor_inocula', '1week_post_inoc'), c('donor_inocula', '2weeks_post_inoc'), c('donor_inocula', '3weeks_post_inoc'), c('donor_inocula', '4weeks_post_inoc'), c('donor_inocula', '5weeks_post_inoc'), c('donor_inocula', 'Final_collection'))

p3 <- p2 + stat_compare_means(comparisons = my_comparisons) 
p3
```


## Trying alpha diversity comparisons another way

```{r, echo=TRUE}

## for INF2 donor
alpha_div_INF2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'INF2')
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Timingofinocu != 'early'| is.na(Timingofinocu))
alpha_div_INF2_ps

# only keeping inoculae
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_INF2_ps <- subset_samples(alpha_div_INF2_ps, SampleType != 'cecal')
alpha_div_INF2_ps

results_INF2 = estimate_richness(alpha_div_INF2_ps, measures = "Shannon")
results_INF2_df <- as.data.frame(results_INF2)
results_INF2_df$SampleID <- rownames(results_INF2_df)
results_INF2_df <- results_INF2_df[c(2,1)]
View(results_INF2_df)

INF2_alpha_div_df <- merge(results_INF2_df, alpha_mapping_file, by='SampleID')
View(INF2_alpha_div_df)

p1 <- ggplot(INF2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + theme(axis.text.x=element_text(angle=45,hjust=1))
p2


## for ADULT2 donor
alpha_div_ADLT2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ADLT2')
alpha_div_ADLT2_ps

# only keeping inoculae
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ADLT2_ps <- subset_samples(alpha_div_ADLT2_ps, SampleType != 'cecal')
alpha_div_ADLT2_ps

results_ADLT2 = estimate_richness(alpha_div_ADLT2_ps, measures = "Shannon")
View(results_ADLT2)

results_ADLT2_df <- as.data.frame(results_ADLT2)
View(results_ADLT2_df)

results_ADLT2_df$SampleID <- rownames(results_ADLT2_df)
View(results_ADLT2_df)

results_ADLT2_df <- results_ADLT2_df[c(2,1)]
View(results_ADLT2_df)

ADLT2_alpha_div_df <- merge(results_ADLT2_df, alpha_mapping_file, by='SampleID')
View(ADLT2_alpha_div_df)

p1 <- ggplot(ADLT2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + theme(axis.text.x=element_text(angle=45,hjust=1))
p2


## for CHLD2 donor
alpha_div_CHLD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'CHLD2')
alpha_div_CHLD2_ps

# only keeping inoculae
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_CHLD2_ps <- subset_samples(alpha_div_CHLD2_ps, SampleType != 'cecal')
alpha_div_CHLD2_ps

results_CHLD2 = estimate_richness(alpha_div_CHLD2_ps, measures = "Shannon")
results_CHLD2_df <- as.data.frame(results_CHLD2)
results_CHLD2_df$SampleID <- rownames(results_CHLD2_df)
results_CHLD2_df <- results_CHLD2_df[c(2,1)]
View(results_CHLD2_df)

CHLD2_alpha_div_df <- merge(results_CHLD2_df, alpha_mapping_file, by='SampleID')
View(CHLD2_alpha_div_df)

p1 <- ggplot(CHLD2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + theme(axis.text.x=element_text(angle=45,hjust=1))
p2


## for ELD2 donor
alpha_div_ELD2_ps <- subset_samples(alpha_div_master_all_filt_ps, Donor == 'ELD2')
alpha_div_ELD2_ps

# only keeping inoculae
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, Actual_inoculum != 'no' | is.na(Actual_inoculum))
alpha_div_ELD2_ps <- subset_samples(alpha_div_ELD2_ps, SampleType != 'cecal')
alpha_div_ELD2_ps

results_ELD2 = estimate_richness(alpha_div_ELD2_ps, measures = "Shannon")
results_ELD2_df <- as.data.frame(results_ELD2)
results_ELD2_df$SampleID <- rownames(results_ELD2_df)
results_ELD2_df <- results_ELD2_df[c(2,1)]
View(results_ELD2_df)

ELD2_alpha_div_df <- merge(results_ELD2_df, alpha_mapping_file, by='SampleID')
View(ELD2_alpha_div_df)

p1 <- ggplot(ELD2_alpha_div_df, aes(x=Alpha_Desc, y=Shannon, fill=Species)) + geom_boxplot()
p2 <- p1 + scale_x_discrete(limits=c("donor_inocula", "48hrs_post_inoc", "1week_post_inoc", "2weeks_post_inoc","3weeks_post_inoc", "4weeks_post_inoc", "5weeks_post_inoc", "Final_collection")) + theme(axis.text.x=element_text(angle=45,hjust=1))
p2
```


##**Hierarchical clustering of samples**## :

```{r, echo=TRUE}

## INF2 Donor

# do transformation of data (count table) using DESeq
library('DESeq2'); packageVersion('DESeq2')
pig_mouse_comp_INF2_comparison_ps
deseq2_obj_INF2 <- phyloseq_to_deseq2(pig_mouse_comp_INF2_comparison_ps, ~ Species)
class(deseq2_obj_INF2)
trans_count_tab_INF2 <- assay(deseq2_obj_INF2)
library("vegan")
euc_dist_INF2 <- vegdist(t(trans_count_tab_INF2), method = "bray" ) # if wanting bray-curtis
#euc_dist_INF2 <- dist(t(trans_count_tab_INF2))

#pig_mouse_comp_INF2_comparison_ps.rarefied <- rarefy_even_depth(pig_mouse_comp_INF2_comparison_ps, rngseed = 1, sample.size = min(sample_sums(pig_mouse_comp_INF2_comparison_ps)), replace = F)  

uwnifrac_dist_INF2 <- phyloseq::distance(pig_mouse_comp_INF2_comparison_ps.rarefied, method = "unifrac", weighted = F)

# performing hierarchical clustering and plotting
euc_clust_INF2 <- hclust(euc_dist_INF2, method = "ward.D2")
plot(euc_clust_INF2)
unifr_clust_INF2 <- hclust(uwnifrac_dist_INF2, method = "ward.D2")

# need to color plot by 'Species' to see clustering patterns 
# install.packages("dendextend"); packageVersion("dendextend")
  library(dendextend)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_INF2 <- read.table("mapping_files_for_dendrograms/INF2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
View(mapping_file_dendro_INF2)
mapping_file_dendro_INF2$Color_Sp <- as.character(mapping_file_dendro_INF2$Color_Sp)

# making dendrogram
euc_dend_INF2 <- as.dendrogram(euc_clust_INF2, hang=0.1)
SampleIDs_dend_INF2 <- labels(euc_dend_INF2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_INF2

unifr_dend_INF2 <- as.dendrogram(unifr_clust_INF2, hang=0.1)
SampleIDs_dend_INF2 <- labels(unifr_dend_INF2)
SampleIDs_dend_INF2

sorted_mapping_INF2 <- mapping_file_dendro_INF2[match(SampleIDs_dend_INF2, mapping_file_dendro_INF2$SampleID),] # sorting mapping file by order of samples in dendrogram
View(sorted_mapping_INF2)

# color dendrogram labels 
labels_colors(euc_dend_INF2) <- sorted_mapping_INF2$Color_Sp
labels_colors(euc_dend_INF2) # check if sample IDs and colors correspond

labels_colors(unifr_dend_INF2) <- sorted_mapping_INF2$Color_Sp
labels_colors(unifr_dend_INF2)

# change labels to reflect time point, change label size and plot dendrogram
labels(euc_dend_INF2) <- sorted_mapping_INF2$Time_point
euc_dend_INF2 <- set(euc_dend_INF2, "labels_cex", 0.5)
plot(euc_dend_INF2, ylab="Euc. distances")

labels(unifr_dend_INF2) <- sorted_mapping_INF2$Time_point
unifr_dend_INF2 <- set(unifr_dend_INF2, "labels_cex", 0.5)
plot(unifr_dend_INF2, ylab="Unweighted Unifrac distances")

# adding distances to dendrogram
install.packages("pvclust"); packageVersion("pvclust")
plot(euc_dend_INF2, ylab="Euc. distances")
with(pvclust:::hc2axes(as.hclust(euc_dend_INF2)),text(x.axis, y.axis, round(y.axis, 2), adj = c(.5, 1), cex = 0.2))

  
## CHLD2 Donor

# do transformation of data (count table) using DESeq
deseq2_obj_CHLD2 <- phyloseq_to_deseq2(pig_mouse_comp_CHLD2_comparison_ps, ~ Species)
class(deseq2_obj_CHLD2)
trans_count_tab_CHLD2 <- assay(deseq2_obj_CHLD2)
euc_dist_CHLD2 <- dist(t(trans_count_tab_CHLD2))
jac_dist_CHLD2 <- vegdist(t(trans_count_tab_CHLD2), method = "jaccard", binary = TRUE)

uwnifrac_dist_CHLD2 <- phyloseq::distance(pig_mouse_comp_CHLD2_comparison_ps.rarefied, method = "unifrac", weighted = F)

# performing hierarchical clustering and plotting
jac_clust_CHLD2 <- hclust(jac_dist_CHLD2, method = "ward.D2")
plot(jac_clust_CHLD2)
unwunifrac_clust_CHLD2 <- hclust(uwnifrac_dist_CHLD2, method = "ward.D2")
plot(unwunifrac_clust_CHLD2)
# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
mapping_file_dendro_CHLD2 <- read.table("mapping_files_for_dendrograms/CHLD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
mapping_file_dendro_CHLD2$Color_sp <- as.character(mapping_file_dendro_CHLD2$Color_sp)

# making dendrogram
jac_dend_CHLD2 <- as.dendrogram(jac_clust_CHLD2, hang=0.1)
unwunifrac_dend_CHLD2 <- as.dendrogram(unwunifrac_clust_CHLD2, hang=0.1)

SampleIDs_dend_CHLD2 <- labels(unwunifrac_dend_CHLD2) # this shows the order in which the samples are arranged in the dendrogram

sorted_mapping_CHLD2 <- mapping_file_dendro_CHLD2[match(SampleIDs_dend_CHLD2, mapping_file_dendro_CHLD2$SampleID),] # sorting mapping file by order of samples in dendrogram

# color dendrogram labels 
labels_colors(jac_dend_CHLD2) <- sorted_mapping_CHLD2$Color_sp
labels_colors(jac_dend_CHLD2) # check if sample IDs and colors correspond
labels_colors(unwunifrac_dend_CHLD2) <- sorted_mapping_CHLD2$Color_sp
labels_colors(unwunifrac_dend_CHLD2)

# change label size and plot dendrogram
jac_dend_CHLD2 <- set(jac_dend_CHLD2, "labels_cex", 0.5)
labels(jac_dend_CHLD2) <- sorted_mapping_CHLD2$Time_point
plot(jac_dend_CHLD2, ylab="Binary Jaccard distances")

unwunifrac_dend_CHLD2 <- set(unwunifrac_dend_CHLD2, "labels_cex", 0.5)
labels(unwunifrac_dend_CHLD2) <- sorted_mapping_CHLD2$Time_point
plot(unwunifrac_dend_CHLD2, ylab="unweighted unifrac distances")


## ADLT2 Donor

pig_mouse_comp_ADLT2_comparison_ps
uwnifrac_dist_ADLT2 <- phyloseq::distance(pig_mouse_comp_ADLT2_comparison_ps.rarefied, method = "unifrac", weighted = F)

# do transformation of data (count table) using DESeq
deseq2_obj_ADLT2 <- phyloseq_to_deseq2(pig_mouse_comp_ADLT2_comparison_ps, ~ Species)
class(deseq2_obj_ADLT2)
trans_count_tab_ADLT2 <- assay(deseq2_obj_ADLT2)
euc_dist_ADLT2 <- dist(t(trans_count_tab_ADLT2))

# performing hierarchical clustering and plotting
euc_clust_ADLT2 <- hclust(euc_dist_ADLT2, method = "ward.D2")
plot(euc_clust_ADLT2)

unifr_clust_ADLT2 <- hclust(uwnifrac_dist_ADLT2, method = "ward.D2")
plot(unifr_clust_ADLT2)


# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
# write.table(sample_data(pig_mouse_comp_ADLT2_comparison_ps), "ADLT2_mapping_dendro.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE) # modify this file by adding a nw column 'Color'
mapping_file_dendro_ADLT2 <- read.table("mapping_files_for_dendrograms/ADLT2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
mapping_file_dendro_ADLT2$Color_sp <- as.character(mapping_file_dendro_ADLT2$Color_sp)

# making dendrogram
euc_dend_ADLT2 <- as.dendrogram(euc_clust_ADLT2, hang=0.1)
unifr_dend_ADLT2 <- as.dendrogram(unifr_clust_ADLT2, hang=0.1)

SampleIDs_dend_ADLT2 <- labels(euc_dend_ADLT2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ADLT2 <- labels(unifr_dend_ADLT2)

sorted_mapping_ADLT2 <- mapping_file_dendro_ADLT2[match(SampleIDs_dend_ADLT2, mapping_file_dendro_ADLT2$SampleID),] # sorting mapping file by order of samples in dendrogram

# color dendrogram labels 
labels_colors(euc_dend_ADLT2) <- sorted_mapping_ADLT2$Color_sp
labels_colors(euc_dend_ADLT2) # check if sample IDs and colors correspond

labels_colors(unifr_dend_ADLT2) <- sorted_mapping_ADLT2$Color_sp
labels_colors(unifr_dend_ADLT2)

# change label size and plot dendrogram
euc_dend_ADLT2 <- set(euc_dend_ADLT2, "labels_cex", 0.5)
labels(euc_dend_ADLT2) <- sorted_mapping_ADLT2$Time_point
plot(euc_dend_ADLT2, ylab="Euc. distances")

unifr_dend_ADLT2 <- set(unifr_dend_ADLT2, "labels_cex", 0.5)
labels(unifr_dend_ADLT2) <- sorted_mapping_ADLT2$Time_point
plot(unifr_dend_ADLT2, ylab="Unweighted Unifrac distances")


## ELDERLY donor

uwnifrac_dist_ELD2 <- phyloseq::distance(pig_mouse_comp_ELD2_comparison_ps.rarefied, method = "unifrac", weighted = F)

# performing hierarchical clustering and plotting
euc_clust_ELD2 <- hclust(euc_dist_ELD2, method = "ward.D2")
plot(euc_clust_ELD2)

unifr_clust_ELD2 <- hclust(uwnifrac_dist_ELD2, method = "ward.D2")
plot(unifr_clust_ELD2)

# reading in a new mapping file which has a column called 'Color' to facilitate coloring of dendrogram by 'Species'
# write.table(sample_data(pig_mouse_comp_ELD2_comparison_ps), "ELD2_mapping_dendro.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE) # modify this file by adding a nw column 'Color'
mapping_file_dendro_ELD2 <- read.table("mapping_files_for_dendrograms/ELD2_mapping_dendro.txt", sep = "\t", header = TRUE, comment.char = "")
mapping_file_dendro_ELD2$Color_sp <- as.character(mapping_file_dendro_ELD2$Color_sp)

# making dendrogram
euc_dend_ELD2 <- as.dendrogram(euc_clust_ELD2, hang=0.1)
unifr_dend_ELD2 <- as.dendrogram(unifr_clust_ELD2, hang=0.1)

SampleIDs_dend_ELD2 <- labels(euc_dend_ELD2) # this shows the order in which the samples are arranged in the dendrogram
SampleIDs_dend_ELD2 <- labels(unifr_dend_ELD2)

sorted_mapping_ELD2 <- mapping_file_dendro_ELD2[match(SampleIDs_dend_ELD2, mapping_file_dendro_ELD2$SampleID),] # sorting mapping file by order of samples in dendrogram

# color dendrogram labels 
labels_colors(euc_dend_ELD2) <- sorted_mapping_ELD2$Color_sp
labels_colors(euc_dend_ELD2) # check if sample IDs and colors correspond

labels_colors(unifr_dend_ELD2) <- sorted_mapping_ELD2$Color_sp
labels_colors(unifr_dend_ELD2)

# change label size and plot dendrogram
euc_dend_ELD2 <- set(euc_dend_ELD2, "labels_cex", 0.5)
labels(euc_dend_ELD2) <- sorted_mapping_ELD2$Time_point
plot(euc_dend_ELD2, ylab="Euc. distances")

unifr_dend_ELD2 <- set(unifr_dend_ELD2, "labels_cex", 0.5)
labels(unifr_dend_ELD2) <- sorted_mapping_ELD2$Time_point
plot(unifr_dend_ELD2, ylab="Unweighted Unifrac distances")

plot(unifr_dend_INF2, ylab="Unweighted Unifrac distances")
plot(unwunifrac_dend_CHLD2, ylab="unweighted unifrac distances")
plot(unifr_dend_ADLT2, ylab="Unweighted Unifrac distances")
plot(unifr_dend_ELD2, ylab="Unweighted Unifrac distances")
```


##**Variation of abundance with time point relative to donor**##

We also thought that it would be useful to compare the relative abundances of core donor ASVs to the HMA animal models at each time point

**INF2 donor**

```{r, echo=TRUE}

library('DESeq2'); packageVersion('DESeq2')

pig_mouse_comp_INF2_comparison_ps # phyloseq object with late inoc piglets, donors, and mice

### for mice
pig_mouse_comp_INF2_donor_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species != 'piglet')
pig_mouse_comp_INF2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_INF2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_INF2_mice_48hrs_ps # phyloseq object with the eight 48 hrs tp samples

pig_mouse_comp_INF2_donors <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_INF2_donors

pig_mouse_comp_INF2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_48hrs_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice_core <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
View(sigtab_subset_48hrs_mice_core)
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_INF2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_mice_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_mice_df)[colnames(sigtab_0.05_48hrs_core_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_mice_df)

# Performing mean,SD calculations for relative abundances (as described in https://github.com/joey711/phyloseq/issues/1085)
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_INF2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% dplyr::ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_INF2_taxa,]
View(ps_mean_stats_core_48hrs)
ps_mean_stats_core_48hrs_rounded <- ps_mean_stats_core_48hrs %>% dplyr::mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_48hrs_rounded)
ps_mean_stats_core_48hrs_rounded_donors <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_core_48hrs_rounded_donors, "DONORS_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_core_48hrs_rounded_mice <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_core_48hrs_rounded_mice, "MICE_INF2_core_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_INF2_mice_1week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_INF2_mice_1week_ps # phyloseq object with the eight 1week tp samples

pig_mouse_comp_INF2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_1week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_1week_core_mice_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_mice_df)[colnames(sigtab_0.05_1week_core_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_1week_rounded_mice <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_mice, "MICE_INF2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_INF2_mice_2week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_INF2_mice_2week_ps # phyloseq object with the eight 2weeks tp samples

pig_mouse_comp_INF2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_2week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_2weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_core <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_2weeks_core_mice_df <- subset(sigtab_0.05_2weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_core_mice_df)[colnames(sigtab_0.05_2weeks_core_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_core_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_2weeks_rounded_mice <- ps_mean_stats_core_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_2weeks_rounded_mice)
write.table(ps_mean_stats_core_2weeks_rounded_mice, "MICE_INF2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_INF2_mice_3week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_INF2_mice_3week_ps # phyloseq object with the seven 3weeks tp samples

pig_mouse_comp_INF2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_3week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_3weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_core <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_3weeks_core_mice_df <- subset(sigtab_0.05_3weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_core_mice_df)[colnames(sigtab_0.05_3weeks_core_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_core_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_3weeks_rounded_mice <- ps_mean_stats_core_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_3weeks_rounded_mice)
write.table(ps_mean_stats_core_3weeks_rounded_mice, "MICE_INF2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_INF2_mice_4week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_INF2_mice_4week_ps # phyloseq object with the seven 4weeks tp samples

pig_mouse_comp_INF2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_4week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_4weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_core <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_4weeks_core_mice_df <- subset(sigtab_0.05_4weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_core_mice_df)[colnames(sigtab_0.05_4weeks_core_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_core_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_4weeks_rounded_mice <- ps_mean_stats_core_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_4weeks_rounded_mice)
write.table(ps_mean_stats_core_4weeks_rounded_mice, "MICE_INF2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_INF2_mice_5week_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_INF2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_INF2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_5week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_5weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_core <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_5weeks_core_mice_df <- subset(sigtab_0.05_5weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_core_mice_df)[colnames(sigtab_0.05_5weeks_core_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_core_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_INF2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_5weeks_rounded_mice <- ps_mean_stats_core_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_5weeks_rounded_mice)
write.table(ps_mean_stats_core_5weeks_rounded_mice, "MICE_INF2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_INF2_mice_40_days_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_INF2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_INF2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_INF2_mice_40_days_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_mice_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_40days_core_mice_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_mice_df)[colnames(sigtab_0.05_40days_core_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_INF2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_INF2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_INF2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_40days_rounded_mice <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_mice)
write.table(ps_mean_stats_core_40days_rounded_mice, "MICE_INF2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of INF2 donor
merged_sigtabs_0.05_core_mice_INF2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_mice_df,sigtab_0.05_1week_core_mice_df,sigtab_0.05_2weeks_core_mice_df,sigtab_0.05_3weeks_core_mice_df,sigtab_0.05_4weeks_core_mice_df,sigtab_0.05_5weeks_core_mice_df,sigtab_0.05_40days_core_mice_df))
View(merged_sigtabs_0.05_core_mice_INF2_df)

# Separating out only those core ASVs which were able to colonize the mice ('core_INF2_taxa_colonizing_mice')
core_INF2_taxa_colonizing_mice
merged_sigtabs_0.05_core_mice_INF2_df <- merged_sigtabs_0.05_core_mice_INF2_df[merged_sigtabs_0.05_core_mice_INF2_df$ASV_ID %in% core_INF2_taxa_colonizing_mice,]


# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_mice_INF2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_mice_INF2_df),1,sum)
View(merged_sigtabs_0.05_core_mice_INF2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
INF2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_core_mice_INF2_df, non_sig_time_points >= '4') 
View(INF2_perst_rel_adund_mice_df)
perst_rel_abund_core_INF2_taxa_mice <- INF2_perst_rel_adund_mice_df[,1]
perst_rel_abund_core_INF2_taxa_mice

# ASVs 113, 134, and 96 didn't have a single time point where they were significantly different from each other
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_113","ASV_134","ASV_96")
perst_rel_abund_core_INF2_taxa_mice <- c(perst_rel_abund_core_INF2_taxa_mice,not_sig_diff_ASVs)
str(perst_rel_abund_core_INF2_taxa_mice) # 9 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_mice_INF2_df, "overall_sigdiff_ASVs_core_mice_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_INF2_donor_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species != 'mouse')
pig_mouse_comp_INF2_donor_piglets_ps # has donor samples and piglet samples

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_INF2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_INF2_piglets_48hrs_ps # phyloseq object with the four 48 hrs tp samples

pig_mouse_comp_INF2_donors <- subset_samples(pig_mouse_comp_INF2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_INF2_donors

pig_mouse_comp_INF2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_48hrs_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_INF2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_piglets_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_piglets_df)[colnames(sigtab_0.05_48hrs_core_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_piglets_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_48hrs_rounded_piglets <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_48hrs_rounded_piglets)
write.table(ps_mean_stats_core_48hrs_rounded_piglets, "PIGLETS_INF2_core_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_INF2_piglets_1week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_INF2_piglets_1week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_INF2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_1week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_1week_core_piglets_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_piglets_df)[colnames(sigtab_0.05_1week_core_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_1week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_INF2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_1week_rounded_piglets <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_piglets, "PIGLETS_INF2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_INF2_piglets_2week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_INF2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_INF2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_2week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_2week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_core <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_2week_core_piglets_df <- subset(sigtab_0.05_2week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_core_piglets_df)[colnames(sigtab_0.05_2week_core_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_core_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_2week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% core_INF2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_2week_rounded_piglets <- ps_mean_stats_core_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_2week_rounded_piglets, "PIGLETS_INF2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_INF2_piglets_3week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_INF2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_INF2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_3week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_3week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_core <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_3week_core_piglets_df <- subset(sigtab_0.05_3week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_core_piglets_df)[colnames(sigtab_0.05_3week_core_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_core_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_3week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% core_INF2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_3week_rounded_piglets <- ps_mean_stats_core_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_3week_rounded_piglets, "PIGLETS_INF2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_INF2_piglets_4week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_INF2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_INF2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_4week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_4week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_core <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_4week_core_piglets_df <- subset(sigtab_0.05_4week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_core_piglets_df)[colnames(sigtab_0.05_4week_core_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_core_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_4week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% core_INF2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_4week_rounded_piglets <- ps_mean_stats_core_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_4week_rounded_piglets, "PIGLETS_INF2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_INF2_piglets_5week_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_INF2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_INF2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_5week_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_5week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_core <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_5week_core_piglets_df <- subset(sigtab_0.05_5week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_core_piglets_df)[colnames(sigtab_0.05_5week_core_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_core_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_5week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps
pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% core_INF2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_5week_rounded_piglets <- ps_mean_stats_core_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_5week_rounded_piglets, "PIGLETS_INF2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_INF2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_INF2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_INF2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_INF2_piglets_40_days_ps,pig_mouse_comp_INF2_donors)
pig_mouse_comp_INF2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_INF2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_INF2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_INF2_donors_vs_piglets_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_INF2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_INF2_taxa,]
sigtab_0.05_40days_core_piglets_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_piglets_df)[colnames(sigtab_0.05_40days_core_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_piglets_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_INF2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_INF2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_INF2_taxa,]
ps_mean_stats_core_40days_rounded_piglets <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_piglets)
write.table(ps_mean_stats_core_40days_rounded_piglets, "PIGLETS_INF2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for piglets of INF2 donor
merged_sigtabs_0.05_core_piglets_INF2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_piglets_df,sigtab_0.05_1week_core_piglets_df,sigtab_0.05_2week_core_piglets_df,sigtab_0.05_3week_core_piglets_df,sigtab_0.05_4week_core_piglets_df,sigtab_0.05_5week_core_piglets_df,sigtab_0.05_40days_core_piglets_df))
View(merged_sigtabs_0.05_core_piglets_INF2_df)

# Separating out only those core ASVs which were able to colonize the piglets ('core_INF2_taxa_colonizing_piglets')
merged_sigtabs_0.05_core_piglets_INF2_df <- merged_sigtabs_0.05_core_piglets_INF2_df[merged_sigtabs_0.05_core_piglets_INF2_df$ASV_ID %in% core_INF2_taxa_colonizing_piglets,]

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_piglets_INF2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_piglets_INF2_df),1,sum)
View(merged_sigtabs_0.05_core_piglets_INF2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
INF2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_core_piglets_INF2_df, non_sig_time_points >= '4') 
View(INF2_perst_rel_adund_piglets_df)
perst_rel_abund_core_INF2_taxa_piglets <- INF2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_core_INF2_taxa_piglets

# ASVs 67, 71, 86, and 5 didn't have a single time point where they were significantly different from each other
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_5", "ASV_67", "ASV_71", "ASV_86")
perst_rel_abund_core_INF2_taxa_piglets <- c(perst_rel_abund_core_INF2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_core_INF2_taxa_piglets) 
perst_rel_abund_core_INF2_taxa_piglets # 10 ASVs with 'persistent' relative abundance 

write.table(merged_sigtabs_0.05_core_piglets_INF2_df, "overall_sigdiff_ASVs_core_piglets_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_core_perst_ASVs_INF2 <- Reduce(intersect, list(perst_rel_abund_core_INF2_taxa_mice, perst_rel_abund_core_INF2_taxa_piglets))
common_core_perst_ASVs_INF2 # only ASV 94
```


**CHLD2 donor**

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_comparison_ps

### for mice
pig_mouse_comp_CHLD2_donor_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species != 'piglet')
pig_mouse_comp_CHLD2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_CHLD2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_CHLD2_mice_48hrs_ps # phyloseq object with the eight 48 hrs tp samples

pig_mouse_comp_CHLD2_donors <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_CHLD2_donors

pig_mouse_comp_CHLD2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_48hrs_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_CHLD2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_mice_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_mice_df)[colnames(sigtab_0.05_48hrs_core_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_mice_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_CHLD2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_48hrs_rounded <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_core_48hrs_rounded_donors <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_core_48hrs_rounded_donors, "DONORS_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_core_48hrs_rounded_mice <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == '48_hrs')
View(ps_mean_stats_core_48hrs_rounded_mice)
write.table(ps_mean_stats_core_48hrs_rounded_mice, "MICE_CHLD2_core_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_CHLD2_mice_1week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_CHLD2_mice_1week_ps # phyloseq object with the eight 1week tp samples

pig_mouse_comp_CHLD2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_1week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_1week_core_mice_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_mice_df)[colnames(sigtab_0.05_1week_core_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_1week_rounded_mice <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_mice, "MICE_CHLD2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_CHLD2_mice_2week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_CHLD2_mice_2week_ps # phyloseq object with the eight 2weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_2week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_2weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_core <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_2weeks_core_mice_df <- subset(sigtab_0.05_2weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_core_mice_df)[colnames(sigtab_0.05_2weeks_core_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_core_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_2weeks_rounded_mice <- ps_mean_stats_core_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_2weeks_rounded_mice)
write.table(ps_mean_stats_core_2weeks_rounded_mice, "MICE_CHLD2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_CHLD2_mice_3week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_CHLD2_mice_3week_ps # phyloseq object with the seven 3weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_3week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_3weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_core <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_3weeks_core_mice_df <- subset(sigtab_0.05_3weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_core_mice_df)[colnames(sigtab_0.05_3weeks_core_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_core_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_3weeks_rounded_mice <- ps_mean_stats_core_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_3weeks_rounded_mice)
write.table(ps_mean_stats_core_3weeks_rounded_mice, "MICE_CHLD2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_CHLD2_mice_4week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_CHLD2_mice_4week_ps # phyloseq object with the seven 4weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_4week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_4weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_core <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_4weeks_core_mice_df <- subset(sigtab_0.05_4weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_core_mice_df)[colnames(sigtab_0.05_4weeks_core_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_core_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_4weeks_rounded_mice <- ps_mean_stats_core_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_4weeks_rounded_mice)
write.table(ps_mean_stats_core_4weeks_rounded_mice, "MICE_CHLD2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_CHLD2_mice_5week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_CHLD2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_CHLD2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_5week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_5weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_core <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_5weeks_core_mice_df <- subset(sigtab_0.05_5weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_core_mice_df)[colnames(sigtab_0.05_5weeks_core_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_core_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_5weeks_rounded_mice <- ps_mean_stats_core_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_5weeks_rounded_mice)
write.table(ps_mean_stats_core_5weeks_rounded_mice, "MICE_CHLD2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_CHLD2_mice_40_days_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_CHLD2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_CHLD2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_mice_40_days_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_mice_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_40days_core_mice_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_mice_df)[colnames(sigtab_0.05_40days_core_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_40days_rounded_mice <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_mice)
write.table(ps_mean_stats_core_40days_rounded_mice, "MICE_CHLD2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of CHLD2 donor
merged_sigtabs_0.05_core_mice_CHLD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_mice_df,sigtab_0.05_1week_core_mice_df,sigtab_0.05_2weeks_core_mice_df,sigtab_0.05_3weeks_core_mice_df,sigtab_0.05_4weeks_core_mice_df,sigtab_0.05_5weeks_core_mice_df,sigtab_0.05_40days_core_mice_df))
View(merged_sigtabs_0.05_core_mice_CHLD2_df)

# Separating out only those core ASVs which were able to colonize the mice ('core_CHLD2_taxa_colonizing_mice')
merged_sigtabs_0.05_core_mice_CHLD2_df <- merged_sigtabs_0.05_core_mice_CHLD2_df[merged_sigtabs_0.05_core_mice_CHLD2_df$ASV_ID %in% core_CHLD2_taxa_colonizing_mice,]
View(merged_sigtabs_0.05_core_mice_CHLD2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_mice_CHLD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_mice_CHLD2_df),1,sum)
View(merged_sigtabs_0.05_core_mice_CHLD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
CHLD2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_core_mice_CHLD2_df, non_sig_time_points >= '4') 
View(CHLD2_perst_rel_adund_mice_df)
perst_rel_abund_core_CHLD2_taxa_mice <- CHLD2_perst_rel_adund_mice_df[,1]
perst_rel_abund_core_CHLD2_taxa_mice

# ASVs 176, and 147 were not siginificantly different in abundance from the donor in any of the time points
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_176","ASV_147")
perst_rel_abund_core_CHLD2_taxa_mice <- c(perst_rel_abund_core_CHLD2_taxa_mice,not_sig_diff_ASVs)
str(perst_rel_abund_core_CHLD2_taxa_mice) # 7 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_mice_CHLD2_df, "overall_sigdiff_ASVs_core_mice_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_CHLD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species != 'mouse')
pig_mouse_comp_CHLD2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_CHLD2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_CHLD2_piglets_48hrs_ps # phyloseq object with the three 48 hrs tp samples

pig_mouse_comp_CHLD2_donors <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_CHLD2_donors

pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_48hrs_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_CHLD2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_piglets_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_piglets_df)[colnames(sigtab_0.05_48hrs_core_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_piglets_df)

# Performing mean,SD calculations for relative abundances
library(dplyr); packageVersion("dplyr")
pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_48hrs_rounded_piglets <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_48hrs_rounded_piglets)
write.table(ps_mean_stats_core_48hrs_rounded_piglets, "PIGLETS_CHLD2_core_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_CHLD2_piglets_1week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_CHLD2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_CHLD2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_1week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_1week_core_piglets_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_piglets_df)[colnames(sigtab_0.05_1week_core_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_1week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_CHLD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_1week_rounded_piglets <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_piglets, "PIGLETS_CHLD2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_CHLD2_piglets_2week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_CHLD2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_CHLD2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_2week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESeq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_2week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_core <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_2week_core_piglets_df <- subset(sigtab_0.05_2week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_core_piglets_df)[colnames(sigtab_0.05_2week_core_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_core_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_2week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
View(ps_mean_stats_2week)
ps_mean_stats_core_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% core_CHLD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_2week_rounded_piglets <- ps_mean_stats_core_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_2week_rounded_piglets)
write.table(ps_mean_stats_core_2week_rounded_piglets, "PIGLETS_CHLD2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_CHLD2_piglets_3week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_CHLD2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_CHLD2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_3week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_3week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_core <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_3week_core_piglets_df <- subset(sigtab_0.05_3week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_core_piglets_df)[colnames(sigtab_0.05_3week_core_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_core_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_3week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% core_CHLD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_3week_rounded_piglets <- ps_mean_stats_core_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_3week_rounded_piglets, "PIGLETS_CHLD2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_CHLD2_piglets_4week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_CHLD2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_CHLD2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_4week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_4week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_core <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_4week_core_piglets_df <- subset(sigtab_0.05_4week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_core_piglets_df)[colnames(sigtab_0.05_4week_core_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_core_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_4week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% core_CHLD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_4week_rounded_piglets <- ps_mean_stats_core_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_4week_rounded_piglets, "PIGLETS_CHLD2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_CHLD2_piglets_5week_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_CHLD2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_CHLD2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_5week_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_5week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_core <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_5week_core_piglets_df <- subset(sigtab_0.05_5week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_core_piglets_df)[colnames(sigtab_0.05_5week_core_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_core_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_5week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps
pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% core_CHLD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_5week_rounded_piglets <- ps_mean_stats_core_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_5week_rounded_piglets, "PIGLETS_CHLD2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_CHLD2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_CHLD2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_CHLD2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_CHLD2_piglets_40_days_ps,pig_mouse_comp_CHLD2_donors)
pig_mouse_comp_CHLD2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_CHLD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_CHLD2_donors_vs_piglets_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_CHLD2_taxa,]
sigtab_0.05_40days_core_piglets_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_piglets_df)[colnames(sigtab_0.05_40days_core_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_piglets_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_CHLD2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% dplyr::group_by(OTU, Time_point) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_core_40days_rounded_piglets <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_piglets)
write.table(ps_mean_stats_core_40days_rounded_piglets, "PIGLETS_CHLD2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


# Merging core significance tables (sigtab_0.05) for piglets of CHLD2 donor
merged_sigtabs_0.05_core_piglets_CHLD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_piglets_df,sigtab_0.05_1week_core_piglets_df,sigtab_0.05_2week_core_piglets_df,sigtab_0.05_3week_core_piglets_df,sigtab_0.05_4week_core_piglets_df,sigtab_0.05_5week_core_piglets_df,sigtab_0.05_40days_core_piglets_df))
View(merged_sigtabs_0.05_core_piglets_CHLD2_df)

# Separating out only those core ASVs which were able to colonize the piglets ('core_CHLD2_taxa_colonizing_piglets')
core_CHLD2_taxa_colonizing_piglets
merged_sigtabs_0.05_core_piglets_CHLD2_df <- merged_sigtabs_0.05_core_piglets_CHLD2_df[merged_sigtabs_0.05_core_piglets_CHLD2_df$ASV_ID %in% core_CHLD2_taxa_colonizing_piglets,]
View(merged_sigtabs_0.05_core_piglets_CHLD2_df)


# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_piglets_CHLD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_piglets_CHLD2_df),1,sum)
View(merged_sigtabs_0.05_core_piglets_CHLD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
CHLD2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_core_piglets_CHLD2_df, non_sig_time_points >= '4') 
View(CHLD2_perst_rel_adund_piglets_df)
perst_rel_abund_core_CHLD2_taxa_piglets <- CHLD2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_core_CHLD2_taxa_piglets

# ASVs 158 & 310 were not siginificantly different in abundance from the donor in any of the time points
# Let's add this ASV to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <-c( "ASV_158", "ASV_310")
perst_rel_abund_core_CHLD2_taxa_piglets <- c(perst_rel_abund_core_CHLD2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_core_CHLD2_taxa_piglets) # 15 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_piglets_CHLD2_df, "overall_sigdiff_ASVs_core_piglets_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_core_perst_ASVs_CHLD2 <- Reduce(intersect, list(perst_rel_abund_core_CHLD2_taxa_mice, perst_rel_abund_core_CHLD2_taxa_piglets))
common_core_perst_ASVs_CHLD2 # 2 ASVs in both (ASV_107 and ASV_55)
```


**ADLT2 donor**

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_comparison_ps

### for mice
pig_mouse_comp_ADLT2_donor_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species != 'piglet')
pig_mouse_comp_ADLT2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ADLT2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_ADLT2_mice_48hrs_ps # phyloseq object with the eight 48 hrs tp samples

pig_mouse_comp_ADLT2_donors <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ADLT2_donors

pig_mouse_comp_ADLT2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_48hrs_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_ADLT2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_mice_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_mice_df)[colnames(sigtab_0.05_48hrs_core_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_mice_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ADLT2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_48hrs_rounded <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_core_48hrs_rounded_donors <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_core_48hrs_rounded_donors, "DONORS_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_core_48hrs_rounded_mice <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_core_48hrs_rounded_mice, "MICE_ADLT2_core_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ADLT2_mice_1week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_ADLT2_mice_1week_ps # phyloseq object with the eight 1week tp samples

pig_mouse_comp_ADLT2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_1week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_1week_core_mice_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_mice_df)[colnames(sigtab_0.05_1week_core_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_1week_rounded_mice <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_mice, "MICE_ADLT2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_ADLT2_mice_2week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_ADLT2_mice_2week_ps # phyloseq object with the eight 2weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_2week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_2weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_core <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_2weeks_core_mice_df <- subset(sigtab_0.05_2weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_core_mice_df)[colnames(sigtab_0.05_2weeks_core_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_core_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_2weeks_rounded_mice <- ps_mean_stats_core_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_2weeks_rounded_mice)
write.table(ps_mean_stats_core_2weeks_rounded_mice, "MICE_ADLT2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_ADLT2_mice_3week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_ADLT2_mice_3week_ps # phyloseq object with the seven 3weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_3week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_3weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_core <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_3weeks_core_mice_df <- subset(sigtab_0.05_3weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_core_mice_df)[colnames(sigtab_0.05_3weeks_core_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_core_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_3weeks_rounded_mice <- ps_mean_stats_core_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_3weeks_rounded_mice)
write.table(ps_mean_stats_core_3weeks_rounded_mice, "MICE_ADLT2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_ADLT2_mice_4week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_ADLT2_mice_4week_ps # phyloseq object with the seven 4weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_4week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_4weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_core <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_4weeks_core_mice_df <- subset(sigtab_0.05_4weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_core_mice_df)[colnames(sigtab_0.05_4weeks_core_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_core_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_4weeks_rounded_mice <- ps_mean_stats_core_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_4weeks_rounded_mice)
write.table(ps_mean_stats_core_4weeks_rounded_mice, "MICE_ADLT2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_ADLT2_mice_5week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_ADLT2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_ADLT2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_5week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_5weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_core <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_5weeks_core_mice_df <- subset(sigtab_0.05_5weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_core_mice_df)[colnames(sigtab_0.05_5weeks_core_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_core_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_5weeks_rounded_mice <- ps_mean_stats_core_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_5weeks_rounded_mice)
write.table(ps_mean_stats_core_5weeks_rounded_mice, "MICE_ADLT2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ADLT2_mice_40_days_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_ADLT2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_ADLT2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_mice_40_days_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_mice_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_40days_core_mice_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_mice_df)[colnames(sigtab_0.05_40days_core_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_40days_rounded_mice <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_mice)
write.table(ps_mean_stats_core_40days_rounded_mice, "MICE_ADLT2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of ADLT2 donor
merged_sigtabs_0.05_core_mice_ADLT2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_mice_df,sigtab_0.05_1week_core_mice_df,sigtab_0.05_2weeks_core_mice_df,sigtab_0.05_3weeks_core_mice_df,sigtab_0.05_4weeks_core_mice_df,sigtab_0.05_5weeks_core_mice_df,sigtab_0.05_40days_core_mice_df))
View(merged_sigtabs_0.05_core_mice_ADLT2_df)

# Separating out only those core ASVs which were able to colonize the mice ('core_ADLT2_taxa_colonizing_mice')
merged_sigtabs_0.05_core_mice_ADLT2_df <- merged_sigtabs_0.05_core_mice_ADLT2_df[merged_sigtabs_0.05_core_mice_ADLT2_df$ASV_ID %in% core_ADLT2_taxa_colonizing_mice,]
View(merged_sigtabs_0.05_core_mice_ADLT2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_mice_ADLT2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_mice_ADLT2_df),1,sum)
View(merged_sigtabs_0.05_core_mice_ADLT2_df)

# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ADLT2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_core_mice_ADLT2_df, non_sig_time_points >= '4') 
View(ADLT2_perst_rel_adund_mice_df)
perst_rel_abund_core_ADLT2_taxa_mice <- ADLT2_perst_rel_adund_mice_df[,1]
perst_rel_abund_core_ADLT2_taxa_mice

# ASV 300 was not significantly different from the donor in abundances in any of the time points
# Let's add this ASV to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_300")
perst_rel_abund_core_ADLT2_taxa_mice <- c(perst_rel_abund_core_ADLT2_taxa_mice,not_sig_diff_ASVs)
str(perst_rel_abund_core_ADLT2_taxa_mice) # 13 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_mice_ADLT2_df, "overall_sigdiff_ASVs_core_mice_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_ADLT2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species != 'mouse')
pig_mouse_comp_ADLT2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ADLT2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_ADLT2_piglets_48hrs_ps # phyloseq object with the two 48 hrs tp samples (the 3rd sample had been removed as an outlier)

pig_mouse_comp_ADLT2_donors <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ADLT2_donors

pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_48hrs_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_ADLT2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_piglets_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_piglets_df)[colnames(sigtab_0.05_48hrs_core_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_piglets_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_48hrs_rounded_piglets <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_48hrs_rounded_piglets)
write.table(ps_mean_stats_core_48hrs_rounded_piglets, "PIGLETS_ADLT2_core_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ADLT2_piglets_1week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_ADLT2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_ADLT2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_1week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_1week_core_piglets_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_piglets_df)[colnames(sigtab_0.05_1week_core_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_1week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_ADLT2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_1week_rounded_piglets <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_piglets, "PIGLETS_ADLT2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_ADLT2_piglets_2week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_ADLT2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_ADLT2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_2week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_2week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_core <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_2week_core_piglets_df <- subset(sigtab_0.05_2week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_core_piglets_df)[colnames(sigtab_0.05_2week_core_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_core_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_2week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% core_ADLT2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_2week_rounded_piglets <- ps_mean_stats_core_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_2week_rounded_piglets, "PIGLETS_ADLT2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_ADLT2_piglets_3week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_ADLT2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_ADLT2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_3week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_3week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_core <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_3week_core_piglets_df <- subset(sigtab_0.05_3week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_core_piglets_df)[colnames(sigtab_0.05_3week_core_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_core_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_3week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% core_ADLT2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_3week_rounded_piglets <- ps_mean_stats_core_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_3week_rounded_piglets, "PIGLETS_ADLT2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_ADLT2_piglets_4week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_ADLT2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_ADLT2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_4week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_4week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_core <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_4week_core_piglets_df <- subset(sigtab_0.05_4week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_core_piglets_df)[colnames(sigtab_0.05_4week_core_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_core_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_4week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% core_ADLT2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_4week_rounded_piglets <- ps_mean_stats_core_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_4week_rounded_piglets, "PIGLETS_ADLT2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_ADLT2_piglets_5week_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_ADLT2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_ADLT2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_5week_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_5week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_core <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_5week_core_piglets_df <- subset(sigtab_0.05_5week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_core_piglets_df)[colnames(sigtab_0.05_5week_core_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_core_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps
pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% core_ADLT2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_5week_rounded_piglets <- ps_mean_stats_core_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_5week_rounded_piglets, "PIGLETS_ADLT2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ADLT2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_ADLT2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_ADLT2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_ADLT2_piglets_40_days_ps,pig_mouse_comp_ADLT2_donors)
pig_mouse_comp_ADLT2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ADLT2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ADLT2_donors_vs_piglets_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_ADLT2_taxa,]
sigtab_0.05_40days_core_piglets_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_piglets_df)[colnames(sigtab_0.05_40days_core_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_piglets_df)

# calculating mean and SD for 40days for piglet relative abundances
pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ADLT2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_core_40days_rounded_piglets <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_piglets)
write.table(ps_mean_stats_core_40days_rounded_piglets, "PIGLETS_ADLT2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for piglets of ADLT2 donor
merged_sigtabs_0.05_core_piglets_ADLT2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_piglets_df,sigtab_0.05_1week_core_piglets_df,sigtab_0.05_2week_core_piglets_df,sigtab_0.05_3week_core_piglets_df,sigtab_0.05_4week_core_piglets_df,sigtab_0.05_5week_core_piglets_df,sigtab_0.05_40days_core_piglets_df))
View(merged_sigtabs_0.05_core_piglets_ADLT2_df)

# Separating out only those core ASVs which were able to colonize the piglets ('core_ADLT2_taxa_colonizing_piglets')
core_ADLT2_taxa_colonizing_piglets
merged_sigtabs_0.05_core_piglets_ADLT2_df <- merged_sigtabs_0.05_core_piglets_ADLT2_df[merged_sigtabs_0.05_core_piglets_ADLT2_df$ASV_ID %in% core_ADLT2_taxa_colonizing_piglets,]
View(merged_sigtabs_0.05_core_piglets_ADLT2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_piglets_ADLT2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_piglets_ADLT2_df),1,sum)
View(merged_sigtabs_0.05_core_piglets_ADLT2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ADLT2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_core_piglets_ADLT2_df, non_sig_time_points >= '4') 
View(ADLT2_perst_rel_adund_piglets_df)
perst_rel_abund_core_ADLT2_taxa_piglets <- ADLT2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_core_ADLT2_taxa_piglets

# ASVs 147 & 419 were not significantly diffrent in abundance from the donor in any of the time points
# Let's add these ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <-c( "ASV_147", "ASV_419")
perst_rel_abund_core_ADLT2_taxa_piglets <- c(perst_rel_abund_core_ADLT2_taxa_piglets,not_sig_diff_ASVs)
str(perst_rel_abund_core_ADLT2_taxa_piglets) # 45 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_piglets_ADLT2_df, "overall_sigdiff_ASVs_core_piglets_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_core_perst_ASVs_ADLT2 <- Reduce(intersect, list(perst_rel_abund_core_ADLT2_taxa_mice, perst_rel_abund_core_ADLT2_taxa_piglets))
common_core_perst_ASVs_ADLT2 # 3 ASVs common (ASV_187, ASV_254, ASV_57)
```


**ELD2 donor**

```{r, echo=TRUE}
pig_mouse_comp_ELD2_comparison_ps

### for mice
pig_mouse_comp_ELD2_donor_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species != 'piglet')
pig_mouse_comp_ELD2_donor_mice_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ELD2_mice_48hrs_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '48_hrs')
pig_mouse_comp_ELD2_mice_48hrs_ps # phyloseq object with the eight 48 hrs tp samples

pig_mouse_comp_ELD2_donors <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ELD2_donors

pig_mouse_comp_ELD2_donor_mice_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_48hrs_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_ELD2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_mice_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_mice_df)[colnames(sigtab_0.05_48hrs_core_mice_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_mice_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ELD2_donor_mice_48hrs_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_48hrs_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_48hrs_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_48hrs_rounded <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
ps_mean_stats_core_48hrs_rounded_donors <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == 'donor')
write.table(ps_mean_stats_core_48hrs_rounded_donors, "DONORS_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
ps_mean_stats_core_48hrs_rounded_mice <- subset(ps_mean_stats_core_48hrs_rounded, Time_point == '48_hrs')
write.table(ps_mean_stats_core_48hrs_rounded_mice, "MICE_ELD2_core_abundances_48_hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ELD2_mice_1week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '1week')
pig_mouse_comp_ELD2_mice_1week_ps # phyloseq object with the nine 1week tp samples

pig_mouse_comp_ELD2_donor_mice_1week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_1week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_1week_core_mice_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_mice_df)[colnames(sigtab_0.05_1week_core_mice_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_mice_df)

# calculating mean and SD for week1 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_1week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_1week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_1week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_1week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_1week_mice_ps.prop) %>% as_tibble
View(ps_mean_stats)
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_1week_rounded_mice <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_mice, "MICE_ELD2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2weeks time point samples
pig_mouse_comp_ELD2_mice_2week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '2weeks')
pig_mouse_comp_ELD2_mice_2week_ps # phyloseq object with the eight 2weeks tp samples

pig_mouse_comp_ELD2_donor_mice_2week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_2week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_2weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2weeks$ASV_ID <- rownames(sigtab_0.05_2weeks)
sigtab_0.05_2weeks <- sigtab_0.05_2weeks[, c(14,1:13)]
sigtab_0.05_2weeks_core <- sigtab_0.05_2weeks[sigtab_0.05_2weeks$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_2weeks_core_mice_df <- subset(sigtab_0.05_2weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2weeks_core_mice_df)[colnames(sigtab_0.05_2weeks_core_mice_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2weeks_core_mice_df)

# calculating mean and SD for week2 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_2week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_2week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_2week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_2week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_2week_mice_ps.prop) %>% as_tibble
ps_mean_stats_2weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2weeks <- ps_mean_stats_2weeks[ps_mean_stats_2weeks$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_2weeks_rounded_mice <- ps_mean_stats_core_2weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_2weeks_rounded_mice)
write.table(ps_mean_stats_core_2weeks_rounded_mice, "MICE_ELD2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3weeks time point samples
pig_mouse_comp_ELD2_mice_3week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '3weeks')
pig_mouse_comp_ELD2_mice_3week_ps # phyloseq object with the seven 3weeks tp samples

pig_mouse_comp_ELD2_donor_mice_3week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_3week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_3weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3weeks$ASV_ID <- rownames(sigtab_0.05_3weeks)
sigtab_0.05_3weeks <- sigtab_0.05_3weeks[, c(14,1:13)]
sigtab_0.05_3weeks_core <- sigtab_0.05_3weeks[sigtab_0.05_3weeks$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_3weeks_core_mice_df <- subset(sigtab_0.05_3weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3weeks_core_mice_df)[colnames(sigtab_0.05_3weeks_core_mice_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3weeks_core_mice_df)

# calculating mean and SD for week3 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_3week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_3week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_3week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_3week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_3week_mice_ps.prop) %>% as_tibble
ps_mean_stats_3weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3weeks <- ps_mean_stats_3weeks[ps_mean_stats_3weeks$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_3weeks_rounded_mice <- ps_mean_stats_core_3weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_3weeks_rounded_mice)
write.table(ps_mean_stats_core_3weeks_rounded_mice, "MICE_ELD2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4weeks time point samples
pig_mouse_comp_ELD2_mice_4week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '4weeks')
pig_mouse_comp_ELD2_mice_4week_ps # phyloseq object with the ten 4weeks tp samples

pig_mouse_comp_ELD2_donor_mice_4week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_4week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_4weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4weeks$ASV_ID <- rownames(sigtab_0.05_4weeks)
sigtab_0.05_4weeks <- sigtab_0.05_4weeks[, c(14,1:13)]
sigtab_0.05_4weeks_core <- sigtab_0.05_4weeks[sigtab_0.05_4weeks$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_4weeks_core_mice_df <- subset(sigtab_0.05_4weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4weeks_core_mice_df)[colnames(sigtab_0.05_4weeks_core_mice_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4weeks_core_mice_df)

# calculating mean and SD for week4 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_4week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_4week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_4week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_4week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_4week_mice_ps.prop) %>% as_tibble
ps_mean_stats_4weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4weeks <- ps_mean_stats_4weeks[ps_mean_stats_4weeks$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_4weeks_rounded_mice <- ps_mean_stats_core_4weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_4weeks_rounded_mice)
write.table(ps_mean_stats_core_4weeks_rounded_mice, "MICE_ELD2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5weeks time point samples
pig_mouse_comp_ELD2_mice_5week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '5weeks')
pig_mouse_comp_ELD2_mice_5week_ps # phyloseq object with the seven 5weeks tp samples

pig_mouse_comp_ELD2_donor_mice_5week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_5week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_5weeks_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5weeks_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5weeks <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5weeks$ASV_ID <- rownames(sigtab_0.05_5weeks)
sigtab_0.05_5weeks <- sigtab_0.05_5weeks[, c(14,1:13)]
sigtab_0.05_5weeks_core <- sigtab_0.05_5weeks[sigtab_0.05_5weeks$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_5weeks_core_mice_df <- subset(sigtab_0.05_5weeks_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5weeks_core_mice_df)[colnames(sigtab_0.05_5weeks_core_mice_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5weeks_core_mice_df)

# calculating mean and SD for week5 mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_5week_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_5week_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_5week_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_5week_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_5week_mice_ps.prop) %>% as_tibble
ps_mean_stats_5weeks <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5weeks <- ps_mean_stats_5weeks[ps_mean_stats_5weeks$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_5weeks_rounded_mice <- ps_mean_stats_core_5weeks %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_5weeks_rounded_mice)
write.table(ps_mean_stats_core_5weeks_rounded_mice, "MICE_ELD2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ELD2_mice_40_days_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_ps, Time_point == '40_days')
pig_mouse_comp_ELD2_mice_40_days_ps # phyloseq object with the seven 40 days tp samples

pig_mouse_comp_ELD2_donor_mice_40_days_ps <- merge_phyloseq(pig_mouse_comp_ELD2_mice_40_days_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_mice_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for mice

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_mice_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_mice_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_mice <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_mice_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_40days_core_mice_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_mice_df)[colnames(sigtab_0.05_40days_core_mice_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_mice_df)

# calculating mean and SD for 40days mouse relative abundances
pig_mouse_comp_ELD2_donor_mice_40days_mice_ps <- subset_samples(pig_mouse_comp_ELD2_donor_mice_40_days_ps, Species == "mouse") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_mice_40days_mice_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_mice_40days_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_mice_40days_mice_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_40days_rounded_mice <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_mice)
write.table(ps_mean_stats_core_40days_rounded_mice, "MICE_ELD2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

# Merging core significance tables (sigtab_0.05) for mice of ELD2 donor
merged_sigtabs_0.05_core_mice_ELD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_mice_df,sigtab_0.05_1week_core_mice_df,sigtab_0.05_2weeks_core_mice_df,sigtab_0.05_3weeks_core_mice_df,sigtab_0.05_4weeks_core_mice_df,sigtab_0.05_5weeks_core_mice_df,sigtab_0.05_40days_core_mice_df))
View(merged_sigtabs_0.05_core_mice_ELD2_df)

# Separating out only those core ASVs which were able to colonize the mice ('core_ELD2_taxa_colonizing_mice')
merged_sigtabs_0.05_core_mice_ELD2_df <- merged_sigtabs_0.05_core_mice_ELD2_df[merged_sigtabs_0.05_core_mice_ELD2_df$ASV_ID %in% core_ELD2_taxa_colonizing_mice,]
View(merged_sigtabs_0.05_core_mice_ELD2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_mice_ELD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_mice_ELD2_df),1,sum)
View(merged_sigtabs_0.05_core_mice_ELD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ELD2_perst_rel_adund_mice_df <- subset(merged_sigtabs_0.05_core_mice_ELD2_df, non_sig_time_points >= '4') 
View(ELD2_perst_rel_adund_mice_df)
perst_rel_abund_core_ELD2_taxa_mice <- ELD2_perst_rel_adund_mice_df[,1]
perst_rel_abund_core_ELD2_taxa_mice

# ASV 87 was not significantly different in abundance from the donor in any of the time points
# Let's add this ASVs to a vector and merge with the 'persistent' relative abundance ASVs vector made above

not_sig_diff_ASVs <- c("ASV_87")
perst_rel_abund_core_ELD2_taxa_mice <- c(perst_rel_abund_core_ELD2_taxa_mice,not_sig_diff_ASVs)
str(perst_rel_abund_core_ELD2_taxa_mice) # 17 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_mice_ELD2_df, "overall_sigdiff_ASVs_core_mice_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### for piglets

pig_mouse_comp_ELD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species != 'mouse')
pig_mouse_comp_ELD2_donor_piglets_ps

## make phyloseq object with donor samples and 48 hrs time point samples
pig_mouse_comp_ELD2_piglets_48hrs_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '48_hrs')
pig_mouse_comp_ELD2_piglets_48hrs_ps # phyloseq object with the three 48 hrs tp samples 

pig_mouse_comp_ELD2_donors <- subset_samples(pig_mouse_comp_ELD2_comparison_ps, Species == 'human') # human donor phyloseq object
pig_mouse_comp_ELD2_donors

pig_mouse_comp_ELD2_donor_piglets_48hrs_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_48hrs_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_48hrs_ps # merge the above phyloseq objects to get 48hrs phyloseq object with donor and 48hrs tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_48hrs_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_48hrs_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_48hrs <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_48hrs$ASV_ID <- rownames(sigtab_0.05_48hrs)
sigtab_0.05_48hrs <- sigtab_0.05_48hrs[, c(14,1:13)]
sigtab_0.05_48hrs_core <- sigtab_0.05_48hrs[sigtab_0.05_48hrs$ASV_ID %in% core_ELD2_taxa,]
View(sigtab_0.05_48hrs_core)
sigtab_0.05_48hrs_core_piglets_df <- subset(sigtab_0.05_48hrs_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_48hrs_core_piglets_df)[colnames(sigtab_0.05_48hrs_core_piglets_df) == "padj"] <- "Padj-48hrs"
View(sigtab_0.05_48hrs_core_piglets_df)

# Performing mean,SD calculations for relative abundances
pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_48hrs_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_48hrs_piglets_ps.prop) %>% as_tibble
ps_mean_stats_48hrs <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_48hrs <- ps_mean_stats_48hrs[ps_mean_stats_48hrs$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_48hrs_rounded_piglets <- ps_mean_stats_core_48hrs %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_48hrs_rounded_piglets)
write.table(ps_mean_stats_core_48hrs_rounded_piglets, "PIGLETS_ELD2_core_abundances_48hrs.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 1week time point samples
pig_mouse_comp_ELD2_piglets_1week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '1week')
pig_mouse_comp_ELD2_piglets_1week_ps # phyloseq object with the three 1week tp samples

pig_mouse_comp_ELD2_donor_piglets_1week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_1week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_1week_ps # merge the above phyloseq objects to get 1week phyloseq object with donor and 1week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_1week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_1week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_1week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_1week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_1week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_1week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_1week$ASV_ID <- rownames(sigtab_0.05_1week)
sigtab_0.05_1week <- sigtab_0.05_1week[, c(14,1:13)]
sigtab_0.05_1week_core <- sigtab_0.05_1week[sigtab_0.05_1week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_1week_core_piglets_df <- subset(sigtab_0.05_1week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_1week_core_piglets_df)[colnames(sigtab_0.05_1week_core_piglets_df) == "padj"] <- "Padj-1week"
View(sigtab_0.05_1week_core_piglets_df)

# calculating mean and SD for week1 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_1week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_1week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_1week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_1week <- ps_mean_stats_1week[ps_mean_stats_1week$OTU %in% core_ELD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_1week_rounded_piglets <- ps_mean_stats_core_1week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_1week_rounded_piglets, "PIGLETS_ELD2_core_abundances_1week.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 2week time point samples
pig_mouse_comp_ELD2_piglets_2week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '2weeks')
pig_mouse_comp_ELD2_piglets_2week_ps # phyloseq object with the three 2week tp samples

pig_mouse_comp_ELD2_donor_piglets_2week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_2week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_2week_ps # merge the above phyloseq objects to get 2week phyloseq object with donor and 2week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_2week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_2week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_2week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_2week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_2week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_2week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_2week$ASV_ID <- rownames(sigtab_0.05_2week)
sigtab_0.05_2week <- sigtab_0.05_2week[, c(14,1:13)]
sigtab_0.05_2week_core <- sigtab_0.05_2week[sigtab_0.05_2week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_2week_core_piglets_df <- subset(sigtab_0.05_2week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_2week_core_piglets_df)[colnames(sigtab_0.05_2week_core_piglets_df) == "padj"] <- "Padj-2weeks"
View(sigtab_0.05_2week_core_piglets_df)

# calculating mean and SD for week2 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_2week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_2week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_2week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_2week <- ps_mean_stats_2week[ps_mean_stats_2week$OTU %in% core_ELD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_2week_rounded_piglets <- ps_mean_stats_core_2week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_2week_rounded_piglets, "PIGLETS_ELD2_core_abundances_2weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 3week time point samples
pig_mouse_comp_ELD2_piglets_3week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '3weeks')
pig_mouse_comp_ELD2_piglets_3week_ps # phyloseq object with the three 3week tp samples

pig_mouse_comp_ELD2_donor_piglets_3week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_3week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_3week_ps # merge the above phyloseq objects to get 3week phyloseq object with donor and 3week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_3week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_3week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_3week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_3week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_3week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_3week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_3week$ASV_ID <- rownames(sigtab_0.05_3week)
sigtab_0.05_3week <- sigtab_0.05_3week[, c(14,1:13)]
sigtab_0.05_3week_core <- sigtab_0.05_3week[sigtab_0.05_3week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_3week_core_piglets_df <- subset(sigtab_0.05_3week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_3week_core_piglets_df)[colnames(sigtab_0.05_3week_core_piglets_df) == "padj"] <- "Padj-3weeks"
View(sigtab_0.05_3week_core_piglets_df)

# calculating mean and SD for week3 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_3week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_3week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_3week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_3week <- ps_mean_stats_3week[ps_mean_stats_3week$OTU %in% core_ELD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_3week_rounded_piglets <- ps_mean_stats_core_3week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_3week_rounded_piglets, "PIGLETS_ELD2_core_abundances_3weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 4week time point samples
pig_mouse_comp_ELD2_piglets_4week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '4weeks')
pig_mouse_comp_ELD2_piglets_4week_ps # phyloseq object with the three 4week tp samples

pig_mouse_comp_ELD2_donor_piglets_4week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_4week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_4week_ps # merge the above phyloseq objects to get 4week phyloseq object with donor and 4week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_4week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_4week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_4week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_4week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_4week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_4week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_4week$ASV_ID <- rownames(sigtab_0.05_4week)
sigtab_0.05_4week <- sigtab_0.05_4week[, c(14,1:13)]
sigtab_0.05_4week_core <- sigtab_0.05_4week[sigtab_0.05_4week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_4week_core_piglets_df <- subset(sigtab_0.05_4week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_4week_core_piglets_df)[colnames(sigtab_0.05_4week_core_piglets_df) == "padj"] <- "Padj-4weeks"
View(sigtab_0.05_4week_core_piglets_df)

# calculating mean and SD for week4 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_4week_ps, Species == "piglet") # we need only the mouse samples
pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_4week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_4week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_4week <- ps_mean_stats_4week[ps_mean_stats_4week$OTU %in% core_ELD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_4week_rounded_piglets <- ps_mean_stats_core_4week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_4week_rounded_piglets, "PIGLETS_ELD2_core_abundances_4weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 5week time point samples
pig_mouse_comp_ELD2_piglets_5week_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '5weeks')
pig_mouse_comp_ELD2_piglets_5week_ps # phyloseq object with the three 5week tp samples

pig_mouse_comp_ELD2_donor_piglets_5week_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_5week_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_5week_ps # merge the above phyloseq objects to get 5week phyloseq object with donor and 5week tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_5week_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_5week_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_5week_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_5week_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_5week <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_5week_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_5week$ASV_ID <- rownames(sigtab_0.05_5week)
sigtab_0.05_5week <- sigtab_0.05_5week[, c(14,1:13)]
sigtab_0.05_5week_core <- sigtab_0.05_5week[sigtab_0.05_5week$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_5week_core_piglets_df <- subset(sigtab_0.05_5week_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_5week_core_piglets_df)[colnames(sigtab_0.05_5week_core_piglets_df) == "padj"] <- "Padj-5weeks"
View(sigtab_0.05_5week_core_piglets_df)

# calculating mean and SD for week5 piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_5week_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps
pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_5week_piglets_ps.prop) %>% as_tibble
ps_mean_stats_5week <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_5week <- ps_mean_stats_5week[ps_mean_stats_5week$OTU %in% core_ELD2_taxa,]
View(ps_mean_stats)
ps_mean_stats_core_5week_rounded_piglets <- ps_mean_stats_core_5week %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
write.table(ps_mean_stats_core_5week_rounded_piglets, "PIGLETS_ELD2_core_abundances_5weeks.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


## make phyloseq object with donor samples and 40 days time point samples
pig_mouse_comp_ELD2_piglets_40_days_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_ps, Time_point == '40_days')
pig_mouse_comp_ELD2_piglets_40_days_ps # phyloseq object with the three 40 days tp samples

pig_mouse_comp_ELD2_donor_piglets_40_days_ps <- merge_phyloseq(pig_mouse_comp_ELD2_piglets_40_days_ps,pig_mouse_comp_ELD2_donors)
pig_mouse_comp_ELD2_donor_piglets_40_days_ps # merge the above phyloseq objects to get 40 days phyloseq object with donor and 40 days tp samples for piglets

# performing DESeq comparisons
diagadds = phyloseq_to_deseq2(pig_mouse_comp_ELD2_donor_piglets_40_days_ps, ~Species)
gm_mean = function(x, na.rm=TRUE) {
  exp(sum(log(x[x > 0]), na.rm = na.rm) / length(x))
}
geoMeans = apply(counts(diagadds), 1, gm_mean)
diagadds = estimateSizeFactors(diagadds, geoMeans = geoMeans)
diagadds = DESeq(diagadds, fitType = "local") # carries out Benjamini-Hochberg multiple corrections within the DESEq function
res_DESeq2 <- results(diagadds, cooksCutoff = FALSE)
sigtab <- res_DESeq2
sigtab <- cbind(as(sigtab, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_40_days_ps)[rownames(sigtab), ], "matrix"))
sigtab$ASV_ID <- rownames(sigtab) # adding ASV_ID as a column header
sigtab <- sigtab[, c(14,1:13)] # setting dataframe columns in the desired order
sigtab_core <- sigtab[sigtab$ASV_ID %in% core_ELD2_taxa,] # Now let's separate out the core ASVs 
View(sigtab_core)
write.table(sigtab_core, 'DESeq2_results_ELD2_donors_vs_piglets_40_days_core_taxa.txt', sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE) 
sigtab_subset_40_days_piglets <- subset(sigtab_core, select=c("ASV_ID", "log2FoldChange", "padj"))
alpha = 0.05
sigtab_0.05 <- res_DESeq2[which(res_DESeq2$padj < alpha), ] # only has the significantly differentially abundant ASVs
sigtab_0.05_40days <- cbind(as(sigtab_0.05, "data.frame"), as(tax_table(pig_mouse_comp_ELD2_donor_piglets_40_days_ps)[rownames(sigtab_0.05), ], "matrix"))
sigtab_0.05_40days$ASV_ID <- rownames(sigtab_0.05_40days)
sigtab_0.05_40days <- sigtab_0.05_40days[, c(14,1:13)]
sigtab_0.05_40days_core <- sigtab_0.05_40days[sigtab_0.05_40days$ASV_ID %in% core_ELD2_taxa,]
sigtab_0.05_40days_core_piglets_df <- subset(sigtab_0.05_40days_core, select=c("ASV_ID","padj"))
colnames(sigtab_0.05_40days_core_piglets_df)[colnames(sigtab_0.05_40days_core_piglets_df) == "padj"] <- "Padj-40days"
View(sigtab_0.05_40days_core_piglets_df)

# calculating mean and SD for 40days for piglet relative abundances
pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_donor_piglets_40_days_ps, Species == "piglet") # we need only the piglet samples
pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(pig_mouse_comp_ELD2_donor_piglets_40days_piglets_ps.prop) %>% as_tibble
ps_mean_stats_40days <- ps_mean_stats %>% group_by(OTU, Time_point) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_core_40days <- ps_mean_stats_40days[ps_mean_stats_40days$OTU %in% core_ELD2_taxa,]
ps_mean_stats_core_40days_rounded_piglets <- ps_mean_stats_core_40days %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_core_40days_rounded_piglets)
write.table(ps_mean_stats_core_40days_rounded_piglets, "PIGLETS_ELD2_core_abundances_40days.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)


# Merging core significance tables (sigtab_0.05) for piglets of ELD2 donor
merged_sigtabs_0.05_core_piglets_ELD2_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(sigtab_0.05_48hrs_core_piglets_df,sigtab_0.05_1week_core_piglets_df,sigtab_0.05_2week_core_piglets_df,sigtab_0.05_3week_core_piglets_df,sigtab_0.05_4week_core_piglets_df,sigtab_0.05_5week_core_piglets_df,sigtab_0.05_40days_core_piglets_df))
View(merged_sigtabs_0.05_core_piglets_ELD2_df)

# Separating out only those core ASVs which were able to colonize the mice ('core_ELD2_taxa_colonizing_piglets')
merged_sigtabs_0.05_core_piglets_ELD2_df <- merged_sigtabs_0.05_core_piglets_ELD2_df[merged_sigtabs_0.05_core_piglets_ELD2_df$ASV_ID %in% core_ELD2_taxa_colonizing_piglets,]
View(merged_sigtabs_0.05_core_piglets_ELD2_df)

# calculating in how many time points each ASV has a relative abundance which is not sig. different from donor :
merged_sigtabs_0.05_core_piglets_ELD2_df$non_sig_time_points <- apply(is.na(merged_sigtabs_0.05_core_piglets_ELD2_df),1,sum)
View(merged_sigtabs_0.05_core_piglets_ELD2_df)
 
# Let's see which ASVs have maintained a donor-like relative abundance in at least 4 of the 7 sampling time points
ELD2_perst_rel_adund_piglets_df <- subset(merged_sigtabs_0.05_core_piglets_ELD2_df, non_sig_time_points >= '4') 
View(ELD2_perst_rel_adund_piglets_df)
perst_rel_abund_core_ELD2_taxa_piglets <- ELD2_perst_rel_adund_piglets_df[,1]
perst_rel_abund_core_ELD2_taxa_piglets

# All 92 ELD core ASVs which colonized the HMA piglets were significantly different from the donor in relative abundance in at least 1 of the 7 sampling time points

str(perst_rel_abund_core_ELD2_taxa_piglets) # 19 ASVs with 'persistent' relative abundance

write.table(merged_sigtabs_0.05_core_piglets_ELD2_df, "overall_sigdiff_ASVs_core_piglets_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

common_core_perst_ASVs_ELD2 <- Reduce(intersect, list(perst_rel_abund_core_ELD2_taxa_mice, perst_rel_abund_core_ELD2_taxa_piglets))
common_core_perst_ASVs_ELD2 # 4 ASVs common (ASV_101, ASV_265, ASV_86, ASV_92)
```


##**Taxonomic breakdown of ASVs that had persistent 'donor-like' relative abundances**##

```{r, echo=TRUE}

### INF2 donor

## for mice
perst_rel_abund_core_INF2_taxa_mice
INF2_HMA_mice_ps
INF2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_INF2_taxa_mice,INF2_HMA_mice_ps)
INF2_perst_abund_ps

# at PHYLUM level
INF2_perst_abund_core_taxa_PHYLUM <- table(tax_table(INF2_perst_abund_ps)[, "Phylum"], exclude = NULL)
INF2_perst_abund_core_taxa_PHYLUM_table_mice <- as.data.frame(INF2_perst_abund_core_taxa_PHYLUM)
colnames(INF2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(INF2_perst_abund_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(INF2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(INF2_perst_abund_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_PHYLUM_table_mice)
write.table(INF2_perst_abund_core_taxa_PHYLUM_table_mice, "INF2_perst_abund_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
INF2_perst_abund_core_taxa_FAMILY <- table(tax_table(INF2_perst_abund_ps)[, "Family"], exclude = NULL)
INF2_perst_abund_core_taxa_FAMILY_table_mice <- as.data.frame(INF2_perst_abund_core_taxa_FAMILY)
colnames(INF2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(INF2_perst_abund_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(INF2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(INF2_perst_abund_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_FAMILY_table_mice)
write.table(INF2_perst_abund_core_taxa_FAMILY_table_mice, "INF2_perst_abund_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
INF2_perst_abund_core_taxa_GENUS <- table(tax_table(INF2_perst_abund_ps)[, "Genus"], exclude = NULL)
INF2_perst_abund_core_taxa_GENUS_table_mice <- as.data.frame(INF2_perst_abund_core_taxa_GENUS)
colnames(INF2_perst_abund_core_taxa_GENUS_table_mice)[colnames(INF2_perst_abund_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(INF2_perst_abund_core_taxa_GENUS_table_mice)[colnames(INF2_perst_abund_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_GENUS_table_mice)
write.table(INF2_perst_abund_core_taxa_GENUS_table_mice, "INF2_perst_abund_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_rel_abund_core_INF2_taxa_piglets
INF2_HMA_piglets_ps
INF2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_INF2_taxa_piglets,INF2_HMA_piglets_ps)
INF2_perst_abund_ps

# at PHYLUM level
INF2_perst_abund_core_taxa_PHYLUM <- table(tax_table(INF2_perst_abund_ps)[, "Phylum"], exclude = NULL)
INF2_perst_abund_core_taxa_PHYLUM_table_piglets <- as.data.frame(INF2_perst_abund_core_taxa_PHYLUM)
colnames(INF2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(INF2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(INF2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(INF2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_PHYLUM_table_piglets)
write.table(INF2_perst_abund_core_taxa_PHYLUM_table_piglets, "INF2_perst_abund_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
INF2_perst_abund_core_taxa_FAMILY <- table(tax_table(INF2_perst_abund_ps)[, "Family"], exclude = NULL)
INF2_perst_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(INF2_perst_abund_core_taxa_FAMILY)
colnames(INF2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(INF2_perst_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(INF2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(INF2_perst_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_FAMILY_table_piglets)
write.table(INF2_perst_abund_core_taxa_FAMILY_table_piglets, "INF2_perst_abund_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
INF2_perst_abund_core_taxa_GENUS <- table(tax_table(INF2_perst_abund_ps)[, "Genus"], exclude = NULL)
INF2_perst_abund_core_taxa_GENUS_table_piglets <- as.data.frame(INF2_perst_abund_core_taxa_GENUS)
colnames(INF2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(INF2_perst_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(INF2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(INF2_perst_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(INF2_perst_abund_core_taxa_GENUS_table_piglets)
write.table(INF2_perst_abund_core_taxa_GENUS_table_piglets, "INF2_perst_abund_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### CHLD2 donor

## for mice
perst_rel_abund_core_CHLD2_taxa_mice
CHLD2_HMA_mice_ps
CHLD2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_CHLD2_taxa_mice,CHLD2_HMA_mice_ps)
CHLD2_perst_abund_ps

# at PHYLUM level
CHLD2_perst_abund_core_taxa_PHYLUM <- table(tax_table(CHLD2_perst_abund_ps)[, "Phylum"], exclude = NULL)
CHLD2_perst_abund_core_taxa_PHYLUM_table_mice <- as.data.frame(CHLD2_perst_abund_core_taxa_PHYLUM)
colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice)
write.table(CHLD2_perst_abund_core_taxa_PHYLUM_table_mice, "CHLD2_perst_abund_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
CHLD2_perst_abund_core_taxa_FAMILY <- table(tax_table(CHLD2_perst_abund_ps)[, "Family"], exclude = NULL)
CHLD2_perst_abund_core_taxa_FAMILY_table_mice <- as.data.frame(CHLD2_perst_abund_core_taxa_FAMILY)
colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_FAMILY_table_mice)
write.table(CHLD2_perst_abund_core_taxa_FAMILY_table_mice, "CHLD2_perst_abund_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
CHLD2_perst_abund_core_taxa_GENUS <- table(tax_table(CHLD2_perst_abund_ps)[, "Genus"], exclude = NULL)
CHLD2_perst_abund_core_taxa_GENUS_table_mice <- as.data.frame(CHLD2_perst_abund_core_taxa_GENUS)
colnames(CHLD2_perst_abund_core_taxa_GENUS_table_mice)[colnames(CHLD2_perst_abund_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(CHLD2_perst_abund_core_taxa_GENUS_table_mice)[colnames(CHLD2_perst_abund_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_GENUS_table_mice)
write.table(CHLD2_perst_abund_core_taxa_GENUS_table_mice, "CHLD2_perst_abund_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_rel_abund_core_CHLD2_taxa_piglets
CHLD2_HMA_piglets_ps
CHLD2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_CHLD2_taxa_piglets,CHLD2_HMA_piglets_ps)
CHLD2_perst_abund_ps

# at PHYLUM level
CHLD2_perst_abund_core_taxa_PHYLUM <- table(tax_table(CHLD2_perst_abund_ps)[, "Phylum"], exclude = NULL)
CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets <- as.data.frame(CHLD2_perst_abund_core_taxa_PHYLUM)
colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets)
write.table(CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets, "CHLD2_perst_abund_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
CHLD2_perst_abund_core_taxa_FAMILY <- table(tax_table(CHLD2_perst_abund_ps)[, "Family"], exclude = NULL)
CHLD2_perst_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(CHLD2_perst_abund_core_taxa_FAMILY)
colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets)
write.table(CHLD2_perst_abund_core_taxa_FAMILY_table_piglets, "CHLD2_perst_abund_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
CHLD2_perst_abund_core_taxa_GENUS <- table(tax_table(CHLD2_perst_abund_ps)[, "Genus"], exclude = NULL)
CHLD2_perst_abund_core_taxa_GENUS_table_piglets <- as.data.frame(CHLD2_perst_abund_core_taxa_GENUS)
colnames(CHLD2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(CHLD2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_abund_core_taxa_GENUS_table_piglets)
write.table(CHLD2_perst_abund_core_taxa_GENUS_table_piglets, "CHLD2_perst_abund_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### ADLT2 donor

## for mice
perst_rel_abund_core_ADLT2_taxa_mice
ADLT2_HMA_mice_ps
ADLT2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_ADLT2_taxa_mice,ADLT2_HMA_mice_ps)
ADLT2_perst_abund_ps

# at PHYLUM level
ADLT2_perst_abund_core_taxa_PHYLUM <- table(tax_table(ADLT2_perst_abund_ps)[, "Phylum"], exclude = NULL)
ADLT2_perst_abund_core_taxa_PHYLUM_table_mice <- as.data.frame(ADLT2_perst_abund_core_taxa_PHYLUM)
colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice)
write.table(ADLT2_perst_abund_core_taxa_PHYLUM_table_mice, "ADLT2_perst_abund_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ADLT2_perst_abund_core_taxa_FAMILY <- table(tax_table(ADLT2_perst_abund_ps)[, "Family"], exclude = NULL)
ADLT2_perst_abund_core_taxa_FAMILY_table_mice <- as.data.frame(ADLT2_perst_abund_core_taxa_FAMILY)
colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_FAMILY_table_mice)
write.table(ADLT2_perst_abund_core_taxa_FAMILY_table_mice, "ADLT2_perst_abund_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ADLT2_perst_abund_core_taxa_GENUS <- table(tax_table(ADLT2_perst_abund_ps)[, "Genus"], exclude = NULL)
ADLT2_perst_abund_core_taxa_GENUS_table_mice <- as.data.frame(ADLT2_perst_abund_core_taxa_GENUS)
colnames(ADLT2_perst_abund_core_taxa_GENUS_table_mice)[colnames(ADLT2_perst_abund_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(ADLT2_perst_abund_core_taxa_GENUS_table_mice)[colnames(ADLT2_perst_abund_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_GENUS_table_mice)
write.table(ADLT2_perst_abund_core_taxa_GENUS_table_mice, "ADLT2_perst_abund_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_rel_abund_core_ADLT2_taxa_piglets
ADLT2_HMA_piglets_ps
ADLT2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_ADLT2_taxa_piglets,ADLT2_HMA_piglets_ps)
ADLT2_perst_abund_ps

# at PHYLUM level
ADLT2_perst_abund_core_taxa_PHYLUM <- table(tax_table(ADLT2_perst_abund_ps)[, "Phylum"], exclude = NULL)
ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets <- as.data.frame(ADLT2_perst_abund_core_taxa_PHYLUM)
colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets)
write.table(ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets, "ADLT2_perst_abund_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ADLT2_perst_abund_core_taxa_FAMILY <- table(tax_table(ADLT2_perst_abund_ps)[, "Family"], exclude = NULL)
ADLT2_perst_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(ADLT2_perst_abund_core_taxa_FAMILY)
colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets)
write.table(ADLT2_perst_abund_core_taxa_FAMILY_table_piglets, "ADLT2_perst_abund_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ADLT2_perst_abund_core_taxa_GENUS <- table(tax_table(ADLT2_perst_abund_ps)[, "Genus"], exclude = NULL)
ADLT2_perst_abund_core_taxa_GENUS_table_piglets <- as.data.frame(ADLT2_perst_abund_core_taxa_GENUS)
colnames(ADLT2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ADLT2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_abund_core_taxa_GENUS_table_piglets)
write.table(ADLT2_perst_abund_core_taxa_GENUS_table_piglets, "ADLT2_perst_abund_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### ELD2 donor

## for mice
perst_rel_abund_core_ELD2_taxa_mice
ELD2_HMA_mice_ps
ELD2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_ELD2_taxa_mice,ELD2_HMA_mice_ps)
ELD2_perst_abund_ps

# at PHYLUM level
ELD2_perst_abund_core_taxa_PHYLUM <- table(tax_table(ELD2_perst_abund_ps)[, "Phylum"], exclude = NULL)
ELD2_perst_abund_core_taxa_PHYLUM_table_mice <- as.data.frame(ELD2_perst_abund_core_taxa_PHYLUM)
colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_mice) == "Var1"] <- "Phylum"
colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_mice)[colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_PHYLUM_table_mice)
write.table(ELD2_perst_abund_core_taxa_PHYLUM_table_mice, "ELD2_perst_abund_core_taxa_PHYLUM_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ELD2_perst_abund_core_taxa_FAMILY <- table(tax_table(ELD2_perst_abund_ps)[, "Family"], exclude = NULL)
ELD2_perst_abund_core_taxa_FAMILY_table_mice <- as.data.frame(ELD2_perst_abund_core_taxa_FAMILY)
colnames(ELD2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(ELD2_perst_abund_core_taxa_FAMILY_table_mice) == "Var1"] <- "Family"
colnames(ELD2_perst_abund_core_taxa_FAMILY_table_mice)[colnames(ELD2_perst_abund_core_taxa_FAMILY_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_FAMILY_table_mice)
write.table(ELD2_perst_abund_core_taxa_FAMILY_table_mice, "ELD2_perst_abund_core_taxa_FAMILY_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ELD2_perst_abund_core_taxa_GENUS <- table(tax_table(ELD2_perst_abund_ps)[, "Genus"], exclude = NULL)
ELD2_perst_abund_core_taxa_GENUS_table_mice <- as.data.frame(ELD2_perst_abund_core_taxa_GENUS)
colnames(ELD2_perst_abund_core_taxa_GENUS_table_mice)[colnames(ELD2_perst_abund_core_taxa_GENUS_table_mice) == "Var1"] <- "Genus"
colnames(ELD2_perst_abund_core_taxa_GENUS_table_mice)[colnames(ELD2_perst_abund_core_taxa_GENUS_table_mice) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_GENUS_table_mice)
write.table(ELD2_perst_abund_core_taxa_GENUS_table_mice, "ELD2_perst_abund_core_taxa_GENUS_table_mice.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## for piglets
perst_rel_abund_core_ELD2_taxa_piglets
ELD2_HMA_piglets_ps
ELD2_perst_abund_ps <- prune_taxa(perst_rel_abund_core_ELD2_taxa_piglets,ELD2_HMA_piglets_ps)
ELD2_perst_abund_ps

# at PHYLUM level
ELD2_perst_abund_core_taxa_PHYLUM <- table(tax_table(ELD2_perst_abund_ps)[, "Phylum"], exclude = NULL)
ELD2_perst_abund_core_taxa_PHYLUM_table_piglets <- as.data.frame(ELD2_perst_abund_core_taxa_PHYLUM)
colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Var1"] <- "Phylum"
colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets)[colnames(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets)
write.table(ELD2_perst_abund_core_taxa_PHYLUM_table_piglets, "ELD2_perst_abund_core_taxa_PHYLUM_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ELD2_perst_abund_core_taxa_FAMILY <- table(tax_table(ELD2_perst_abund_ps)[, "Family"], exclude = NULL)
ELD2_perst_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(ELD2_perst_abund_core_taxa_FAMILY)
colnames(ELD2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ELD2_perst_abund_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_FAMILY_table_piglets)
write.table(ELD2_perst_abund_core_taxa_FAMILY_table_piglets, "ELD2_perst_abund_core_taxa_FAMILY_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ELD2_perst_abund_core_taxa_GENUS <- table(tax_table(ELD2_perst_abund_ps)[, "Genus"], exclude = NULL)
ELD2_perst_abund_core_taxa_GENUS_table_piglets <- as.data.frame(ELD2_perst_abund_core_taxa_GENUS)
colnames(ELD2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ELD2_perst_abund_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_abund_core_taxa_GENUS_table_piglets)
write.table(ELD2_perst_abund_core_taxa_GENUS_table_piglets, "ELD2_perst_abund_core_taxa_GENUS_table_piglets.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### Checking if 'persistent colonizers' are represented within the ASVs which maintain donor-like relative abundances

## INF2 Donor

# for mice
perst_colonizing_core_taxa_INF2_mice
perst_rel_abund_core_INF2_taxa_mice
perst_coloniz_and_abund_INF2_taxa_mice <- Reduce(intersect, list(perst_colonizing_core_taxa_INF2_mice, perst_rel_abund_core_INF2_taxa_mice))
perst_coloniz_and_abund_INF2_taxa_mice # 7 ASVs 


# for piglets
perst_colonizing_core_taxa_INF2_piglets
perst_rel_abund_core_INF2_taxa_piglets
perst_coloniz_and_abund_INF2_taxa_piglets <- Reduce(intersect, list(perst_colonizing_core_taxa_INF2_piglets, perst_rel_abund_core_INF2_taxa_piglets))
perst_coloniz_and_abund_INF2_taxa_piglets # 8 ASVs 

# commonly persistent and donor-like abundant in both animal models
common_perst_coloniz_and_abund_INF2_taxa <- Reduce(intersect,list(perst_coloniz_and_abund_INF2_taxa_mice, perst_coloniz_and_abund_INF2_taxa_piglets))
common_perst_coloniz_and_abund_INF2_taxa # 0 taxa

 
## CHLD2 Donor

# for mice
perst_colonizing_core_taxa_CHLD2_mice
perst_rel_abund_core_CHLD2_taxa_mice
perst_coloniz_and_abund_CHLD2_taxa_mice <- Reduce(intersect, list(perst_colonizing_core_taxa_CHLD2_mice, perst_rel_abund_core_CHLD2_taxa_mice))
perst_coloniz_and_abund_CHLD2_taxa_mice # 1 ASV 

# for piglets
perst_colonizing_core_taxa_CHLD2_piglets
perst_rel_abund_core_CHLD2_taxa_piglets
perst_coloniz_and_abund_CHLD2_taxa_piglets <- Reduce(intersect, list(perst_colonizing_core_taxa_CHLD2_piglets, perst_rel_abund_core_CHLD2_taxa_piglets))
perst_coloniz_and_abund_CHLD2_taxa_piglets
 # 15 ASVs 

# Let's look at the classification of these 15 ASVs
CHLD2_perst_col_and_abund_ps <- prune_taxa(perst_coloniz_and_abund_CHLD2_taxa_piglets,CHLD2_HMA_piglets_ps)
CHLD2_perst_col_and_abund_ps

# at FAMILY level
CHLD2_perst_and_abund_core_taxa_FAMILY <- table(tax_table(CHLD2_perst_col_and_abund_ps)[, "Family"], exclude = NULL)
CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(CHLD2_perst_and_abund_core_taxa_FAMILY)
colnames(CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_and_abund_core_taxa_FAMILY_table_piglets)

# at GENUS level
CHLD2_perst_and_abund_core_taxa_GENUS <- table(tax_table(CHLD2_perst_col_and_abund_ps)[, "Genus"], exclude = NULL)
CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets <- as.data.frame(CHLD2_perst_and_abund_core_taxa_GENUS)
colnames(CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(CHLD2_perst_and_abund_core_taxa_GENUS_table_piglets)

# commonly persistent and donor-like abundant in both animal models
common_perst_coloniz_and_abund_CHLD2_taxa <- Reduce(intersect,list(perst_coloniz_and_abund_CHLD2_taxa_mice, perst_coloniz_and_abund_CHLD2_taxa_piglets))
common_perst_coloniz_and_abund_CHLD2_taxa # A single ASV (ASV_107)


## ADLT2 Donor

# for mice
perst_colonizing_core_taxa_ADLT2_mice
perst_rel_abund_core_ADLT2_taxa_mice
perst_coloniz_and_abund_ADLT2_taxa_mice <- Reduce(intersect, list(perst_colonizing_core_taxa_ADLT2_mice, perst_rel_abund_core_ADLT2_taxa_mice))
perst_coloniz_and_abund_ADLT2_taxa_mice # 2 ASVs (ASV_187 and ASV_285)

# for piglets
perst_colonizing_core_taxa_ADLT2_piglets
perst_rel_abund_core_ADLT2_taxa_piglets
perst_coloniz_and_abund_ADLT2_taxa_piglets <- Reduce(intersect, list(perst_colonizing_core_taxa_ADLT2_piglets, perst_rel_abund_core_ADLT2_taxa_piglets))
perst_coloniz_and_abund_ADLT2_taxa_piglets # 40 ASVs

# Let's look at the classification of these 40 ASVs
ADLT2_perst_col_and_abund_ps <- prune_taxa(perst_coloniz_and_abund_ADLT2_taxa_piglets,ADLT2_HMA_piglets_ps)
ADLT2_perst_col_and_abund_ps

# at FAMILY level
ADLT2_perst_and_abund_core_taxa_FAMILY <- table(tax_table(ADLT2_perst_col_and_abund_ps)[, "Family"], exclude = NULL)
ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(ADLT2_perst_and_abund_core_taxa_FAMILY)
colnames(ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_and_abund_core_taxa_FAMILY_table_piglets)

# at GENUS level
ADLT2_perst_and_abund_core_taxa_GENUS <- table(tax_table(ADLT2_perst_col_and_abund_ps)[, "Genus"], exclude = NULL)
ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets <- as.data.frame(ADLT2_perst_and_abund_core_taxa_GENUS)
colnames(ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ADLT2_perst_and_abund_core_taxa_GENUS_table_piglets)

# commonly persistent and donor-like abundant in both animal models
common_perst_coloniz_and_abund_ADLT2_taxa <- Reduce(intersect,list(perst_coloniz_and_abund_ADLT2_taxa_mice, perst_coloniz_and_abund_ADLT2_taxa_piglets))
common_perst_coloniz_and_abund_ADLT2_taxa # A single ASV (ASV_87)


## ELD2 Donor

# for mice
perst_colonizing_core_taxa_ELD2_mice
perst_rel_abund_core_ELD2_taxa_mice
perst_coloniz_and_abund_ELD2_taxa_mice <- Reduce(intersect, list(perst_colonizing_core_taxa_ELD2_mice, perst_rel_abund_core_ELD2_taxa_mice))
perst_coloniz_and_abund_ELD2_taxa_mice # 1 ASV (ASV_69)

# for piglets
perst_colonizing_core_taxa_ELD2_piglets
perst_rel_abund_core_ELD2_taxa_piglets
perst_coloniz_and_abund_ELD2_taxa_piglets <- Reduce(intersect, list(perst_colonizing_core_taxa_ELD2_piglets, perst_rel_abund_core_ELD2_taxa_piglets))
perst_coloniz_and_abund_ELD2_taxa_piglets # 16 ASVs

# Let's look at the classification of these 16 ASVs
ELD2_perst_col_and_abund_ps <- prune_taxa(perst_coloniz_and_abund_ELD2_taxa_piglets,ELD2_HMA_piglets_ps)
ELD2_perst_col_and_abund_ps

# at FAMILY level
ELD2_perst_and_abund_core_taxa_FAMILY <- table(tax_table(ELD2_perst_col_and_abund_ps)[, "Family"], exclude = NULL)
ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets <- as.data.frame(ELD2_perst_and_abund_core_taxa_FAMILY)
colnames(ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Var1"] <- "Family"
colnames(ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets)[colnames(ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_and_abund_core_taxa_FAMILY_table_piglets)

# at GENUS level
ELD2_perst_and_abund_core_taxa_GENUS <- table(tax_table(ELD2_perst_col_and_abund_ps)[, "Genus"], exclude = NULL)
ELD2_perst_and_abund_core_taxa_GENUS_table_piglets <- as.data.frame(ELD2_perst_and_abund_core_taxa_GENUS)
colnames(ELD2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Var1"] <- "Genus"
colnames(ELD2_perst_and_abund_core_taxa_GENUS_table_piglets)[colnames(ELD2_perst_and_abund_core_taxa_GENUS_table_piglets) == "Freq"] <- "number of ASVs"
View(ELD2_perst_and_abund_core_taxa_GENUS_table_piglets)

# commonly persistent and donor-like abundant in both animal models
common_perst_coloniz_and_abund_ELD2_taxa <- Reduce(intersect,list(perst_coloniz_and_abund_ELD2_taxa_mice, perst_coloniz_and_abund_ELD2_taxa_piglets))
common_perst_coloniz_and_abund_ELD2_taxa # 0 common ASVs
```


##**Common core ASVs across donors**##

We want to look at which ASVs are core ASVs across the four different donors :

```{r, echo=TRUE}
str(core_INF2_taxa)
str(core_CHLD2_taxa)
str(core_ADLT2_taxa)
str(core_ELD2_taxa)
common_core_ASVs <- Reduce(intersect, list(core_INF2_taxa,core_CHLD2_taxa,core_ADLT2_taxa,core_ELD2_taxa))
common_core_ASVs # only ASV 115 is commonly found in all the cores!

# Let's try looking at a common core omitting the INF2 donor core
common_core_ASVs_no_INF2 <- Reduce(intersect, list(core_CHLD2_taxa,core_ADLT2_taxa,core_ELD2_taxa))
common_core_ASVs_no_INF2 # there are 27 shared core ASVs among CHLD2, ADLT2, and ELD2 cores
```


##**Core ASVs not colonizing animal models**##

```{r, echo=TRUE}

## INF2 donor
core_INF2_taxa
core_INF2_taxa_colonizing_mice
core_INF2_taxa_colonizing_piglets

# core INF2 ASVs not colonizing mice (using 'setdiff' function)
core_INF2_taxa_not_colonizing_mice <- setdiff(core_INF2_taxa, core_INF2_taxa_colonizing_mice)
core_INF2_taxa_not_colonizing_mice # 5 ASVs 

# core INF2 ASVs not colonizing piglets
core_INF2_taxa_not_colonizing_piglets <- setdiff(core_INF2_taxa, core_INF2_taxa_colonizing_piglets)
core_INF2_taxa_not_colonizing_piglets # 6 ASVs

# core INF2 ASVs not colonizing either animal model
core_INF2_taxa_not_colonizing <- Reduce(intersect, list(core_INF2_taxa_not_colonizing_mice,core_INF2_taxa_not_colonizing_piglets))
core_INF2_taxa_not_colonizing # 4 ASVs

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_INF2_comparison_core_ps
INF2_not_colonizing_ps <- prune_taxa(core_INF2_taxa_not_colonizing,pig_mouse_comp_INF2_comparison_core_ps)
INF2_not_colonizing_ps

tax<-as(tax_table(INF2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax.txt", quote=FALSE, col.names=FALSE, sep="\t")

## CHLD2 donor
core_CHLD2_taxa
core_CHLD2_taxa_colonizing_mice
core_CHLD2_taxa_colonizing_piglets

# core CHLD2 ASVs not colonizing mice (using 'setdiff' function)
core_CHLD2_taxa_not_colonizing_mice <- setdiff(core_CHLD2_taxa, core_CHLD2_taxa_colonizing_mice)
core_CHLD2_taxa_not_colonizing_mice # 39 ASVs 

# core CHLD2 ASVs not colonizing piglets
core_CHLD2_taxa_not_colonizing_piglets <- setdiff(core_CHLD2_taxa, core_CHLD2_taxa_colonizing_piglets)
core_CHLD2_taxa_not_colonizing_piglets # 11 ASVs

# core CHLD2 ASVs not colonizing mice but colonizing pigs
core_CHLD2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_CHLD2_taxa_colonizing_piglets,core_CHLD2_taxa_not_colonizing_mice))
core_CHLD2_taxa_no_mice_yes_pigs # 28 ASVs

# core CHLD2 ASVs not colonizing piglets but colonizing mice
core_CHLD2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_CHLD2_taxa_colonizing_mice,core_CHLD2_taxa_not_colonizing_piglets))
core_CHLD2_taxa_no_pigs_yes_mice # 0 ASVs

# core CHLD2 ASVs not colonizing either animal model
core_CHLD2_taxa_not_colonizing <- Reduce(intersect, list(core_CHLD2_taxa_not_colonizing_mice,core_CHLD2_taxa_not_colonizing_piglets))  
core_CHLD2_taxa_not_colonizing # 11 ASVs

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_CHLD2_comparison_core_ps
CHLD2_not_colonizing_ps <- prune_taxa(core_CHLD2_taxa_not_colonizing,pig_mouse_comp_CHLD2_comparison_core_ps)
CHLD2_not_colonizing_ps

tax<-as(tax_table(CHLD2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax.txt", quote=FALSE, col.names=FALSE, sep="\t")


## ADLT2 donor
core_ADLT2_taxa
core_ADLT2_taxa_colonizing_mice
core_ADLT2_taxa_colonizing_piglets

# core ADLT2 ASVs not colonizing mice (using 'setdiff' function)
core_ADLT2_taxa_not_colonizing_mice <- setdiff(core_ADLT2_taxa, core_ADLT2_taxa_colonizing_mice)
core_ADLT2_taxa_not_colonizing_mice # 74 ASVs 

# core ADLT2 ASVs not colonizing piglets
core_ADLT2_taxa_not_colonizing_piglets <- setdiff(core_ADLT2_taxa, core_ADLT2_taxa_colonizing_piglets)
core_ADLT2_taxa_not_colonizing_piglets # 32 ASVs

# core ADLT2 ASVs not colonizing mice but colonizing pigs
core_ADLT2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_ADLT2_taxa_colonizing_piglets,core_ADLT2_taxa_not_colonizing_mice))
core_ADLT2_taxa_no_mice_yes_pigs # 44 ASVs

# core CHLD2 ASVs not colonizing piglets but colonizing mice
core_ADLT2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_ADLT2_taxa_colonizing_mice,core_ADLT2_taxa_not_colonizing_piglets))
core_ADLT2_taxa_no_pigs_yes_mice
                   
# core ADLT2 ASVs not colonizing either animal model
core_ADLT2_taxa_not_colonizing <- Reduce(intersect, list(core_ADLT2_taxa_not_colonizing_mice,core_ADLT2_taxa_not_colonizing_piglets))
core_ADLT2_taxa_not_colonizing # 31 core ASVs

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_ADLT2_comparison_core_ps
ADLT2_not_colonizing_ps <- prune_taxa(core_ADLT2_taxa_not_colonizing,pig_mouse_comp_ADLT2_comparison_core_ps)
ADLT2_not_colonizing_ps

tax<-as(tax_table(ADLT2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax.txt", quote=FALSE, col.names=FALSE, sep="\t")


## ELD2 donor
core_ELD2_taxa
core_ELD2_taxa_colonizing_mice
core_ELD2_taxa_colonizing_piglets

# core ELD2 ASVs not colonizing mice (using 'setdiff' function)
core_ELD2_taxa_not_colonizing_mice <- setdiff(core_ELD2_taxa, core_ELD2_taxa_colonizing_mice)
core_ELD2_taxa_not_colonizing_mice # 79 ASVs 

# core ELD2 ASVs not colonizing piglets
core_ELD2_taxa_not_colonizing_piglets <- setdiff(core_ELD2_taxa, core_ELD2_taxa_colonizing_piglets)
core_ELD2_taxa_not_colonizing_piglets # 42 ASVs

# core ELD2 ASVs not colonizing mice but colonizing pigs
core_ELD2_taxa_no_mice_yes_pigs <- Reduce(intersect, list(core_ELD2_taxa_colonizing_piglets,core_ELD2_taxa_not_colonizing_mice))
core_ELD2_taxa_no_mice_yes_pigs # 41 ASVs

# core ELD2 ASVs not colonizing piglets but colonizing mice
core_ELD2_taxa_no_pigs_yes_mice <- Reduce(intersect, list(core_ELD2_taxa_colonizing_mice,core_ELD2_taxa_not_colonizing_piglets))
core_ELD2_taxa_no_pigs_yes_mice # 4 ASVs

# core ELD2 ASVs not colonizing either animal model
core_ELD2_taxa_not_colonizing <- Reduce(intersect, list(core_ELD2_taxa_not_colonizing_mice,core_ELD2_taxa_not_colonizing_piglets))
core_ELD2_taxa_not_colonizing # 38 ASVs not colonizing either animal model

# Let's get the taxonomic classifications for these ASVs
pig_mouse_comp_ELD2_comparison_core_ps
ELD2_not_colonizing_ps <- prune_taxa(core_ELD2_taxa_not_colonizing,pig_mouse_comp_ELD2_comparison_core_ps)
ELD2_not_colonizing_ps

tax<-as(tax_table(ELD2_not_colonizing_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL
write.table(tax, "not_colonizing_tax.txt", quote=FALSE, col.names=FALSE, sep="\t")


### checking if there are common taxa (across donors) that fail to colonize a given animal model

# for HMA mice
common_taxa_not_colonizing_mice <- Reduce(intersect, list(core_INF2_taxa_not_colonizing_mice,core_CHLD2_taxa_not_colonizing_mice,core_ADLT2_taxa_not_colonizing_mice,core_ELD2_taxa_not_colonizing_mice))
common_taxa_not_colonizing_mice # 0 when INF2 is included

common_taxa_not_colonizing_mice_no_INF2 <- Reduce(intersect, list(core_CHLD2_taxa_not_colonizing_mice,core_ADLT2_taxa_not_colonizing_mice,core_ELD2_taxa_not_colonizing_mice))
common_taxa_not_colonizing_mice_no_INF2 # 10 ASVs (mostly Lachnospiraceae members)

# for HMA mice
common_taxa_not_colonizing_piglets <- Reduce(intersect, list(core_INF2_taxa_not_colonizing_piglets,core_CHLD2_taxa_not_colonizing_piglets,core_ADLT2_taxa_not_colonizing_piglets,core_ELD2_taxa_not_colonizing_piglets))
common_taxa_not_colonizing_piglets # 0 when INF2 is included

common_taxa_not_colonizing_piglets_no_INF2 <- Reduce(intersect, list(core_CHLD2_taxa_not_colonizing_piglets,core_ADLT2_taxa_not_colonizing_piglets,core_ELD2_taxa_not_colonizing_piglets))
common_taxa_not_colonizing_piglets_no_INF2 # 0 ASVs once again

### Looking at the taxonomy of core ASVs from each donor which have not colonized either of the two animal models

## INF2 donor
core_INF2_taxa_not_colonizing
pig_mouse_comp_INF2_comparison_ps
INF2_non_colonizing_ps <- prune_taxa(core_INF2_taxa_not_colonizing, pig_mouse_comp_INF2_comparison_ps)
INF2_non_colonizing_ps

# at PHYLUM level
INF2_non_coloniz_core_taxa_PHYLUM <- table(tax_table(INF2_non_colonizing_ps)[, "Phylum"], exclude = NULL)
INF2_non_coloniz_core_taxa_PHYLUM_table <- as.data.frame(INF2_non_coloniz_core_taxa_PHYLUM)
colnames(INF2_non_coloniz_core_taxa_PHYLUM_table)[colnames(INF2_non_coloniz_core_taxa_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(INF2_non_coloniz_core_taxa_PHYLUM_table)[colnames(INF2_non_coloniz_core_taxa_PHYLUM_table) == "Freq"] <- "number of ASVs"
View(INF2_non_coloniz_core_taxa_PHYLUM_table)
write.table(INF2_non_coloniz_core_taxa_PHYLUM_table, "INF2_non_coloniz_core_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
INF2_non_coloniz_core_taxa_FAMILY <- table(tax_table(INF2_non_colonizing_ps)[, "Family"], exclude = NULL)
INF2_non_coloniz_core_taxa_FAMILY_table <- as.data.frame(INF2_non_coloniz_core_taxa_FAMILY)
colnames(INF2_non_coloniz_core_taxa_FAMILY_table)[colnames(INF2_non_coloniz_core_taxa_FAMILY_table) == "Var1"] <- "Family"
colnames(INF2_non_coloniz_core_taxa_FAMILY_table)[colnames(INF2_non_coloniz_core_taxa_FAMILY_table) == "Freq"] <- "number of ASVs"
View(INF2_non_coloniz_core_taxa_FAMILY_table)
write.table(INF2_non_coloniz_core_taxa_FAMILY_table, "INF2_non_coloniz_core_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
INF2_non_coloniz_core_taxa_GENUS <- table(tax_table(INF2_non_colonizing_ps)[, "Genus"], exclude = NULL)
INF2_non_coloniz_core_taxa_GENUS_table <- as.data.frame(INF2_non_coloniz_core_taxa_GENUS)
colnames(INF2_non_coloniz_core_taxa_GENUS_table)[colnames(INF2_non_coloniz_core_taxa_GENUS_table) == "Var1"] <- "Genus"
colnames(INF2_non_coloniz_core_taxa_GENUS_table)[colnames(INF2_non_coloniz_core_taxa_GENUS_table) == "Freq"] <- "number of ASVs"
View(INF2_non_coloniz_core_taxa_GENUS_table)
write.table(INF2_non_coloniz_core_taxa_GENUS_table, "INF2_non_coloniz_core_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## CHLD2 donor
core_CHLD2_taxa_not_colonizing
pig_mouse_comp_CHLD2_comparison_ps
CHLD2_non_colonizing_ps <- prune_taxa(core_CHLD2_taxa_not_colonizing, pig_mouse_comp_CHLD2_comparison_ps)
CHLD2_non_colonizing_ps

# at PHYLUM level
CHLD2_non_coloniz_core_taxa_PHYLUM <- table(tax_table(CHLD2_non_colonizing_ps)[, "Phylum"], exclude = NULL)
CHLD2_non_coloniz_core_taxa_PHYLUM_table <- as.data.frame(CHLD2_non_coloniz_core_taxa_PHYLUM)
colnames(CHLD2_non_coloniz_core_taxa_PHYLUM_table)[colnames(CHLD2_non_coloniz_core_taxa_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(CHLD2_non_coloniz_core_taxa_PHYLUM_table)[colnames(CHLD2_non_coloniz_core_taxa_PHYLUM_table) == "Freq"] <- "number of ASVs"
View(CHLD2_non_coloniz_core_taxa_PHYLUM_table)
write.table(CHLD2_non_coloniz_core_taxa_PHYLUM_table, "CHLD2_non_coloniz_core_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
CHLD2_non_coloniz_core_taxa_FAMILY <- table(tax_table(CHLD2_non_colonizing_ps)[, "Family"], exclude = NULL)
CHLD2_non_coloniz_core_taxa_FAMILY_table <- as.data.frame(CHLD2_non_coloniz_core_taxa_FAMILY)
colnames(CHLD2_non_coloniz_core_taxa_FAMILY_table)[colnames(CHLD2_non_coloniz_core_taxa_FAMILY_table) == "Var1"] <- "Family"
colnames(CHLD2_non_coloniz_core_taxa_FAMILY_table)[colnames(CHLD2_non_coloniz_core_taxa_FAMILY_table) == "Freq"] <- "number of ASVs"
View(CHLD2_non_coloniz_core_taxa_FAMILY_table)
write.table(CHLD2_non_coloniz_core_taxa_FAMILY_table, "CHLD2_non_coloniz_core_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
CHLD2_non_coloniz_core_taxa_GENUS <- table(tax_table(CHLD2_non_colonizing_ps)[, "Genus"], exclude = NULL)
CHLD2_non_coloniz_core_taxa_GENUS_table <- as.data.frame(CHLD2_non_coloniz_core_taxa_GENUS)
colnames(CHLD2_non_coloniz_core_taxa_GENUS_table)[colnames(CHLD2_non_coloniz_core_taxa_GENUS_table) == "Var1"] <- "Genus"
colnames(CHLD2_non_coloniz_core_taxa_GENUS_table)[colnames(CHLD2_non_coloniz_core_taxa_GENUS_table) == "Freq"] <- "number of ASVs"
View(CHLD2_non_coloniz_core_taxa_GENUS_table)
write.table(CHLD2_non_coloniz_core_taxa_GENUS_table, "CHLD2_non_coloniz_core_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## ADLT2 donor
core_ADLT2_taxa_not_colonizing
pig_mouse_comp_ADLT2_comparison_ps
ADLT2_non_colonizing_ps <- prune_taxa(core_ADLT2_taxa_not_colonizing, pig_mouse_comp_ADLT2_comparison_ps)
ADLT2_non_colonizing_ps

# at PHYLUM level
ADLT2_non_coloniz_core_taxa_PHYLUM <- table(tax_table(ADLT2_non_colonizing_ps)[, "Phylum"], exclude = NULL)
ADLT2_non_coloniz_core_taxa_PHYLUM_table <- as.data.frame(ADLT2_non_coloniz_core_taxa_PHYLUM)
colnames(ADLT2_non_coloniz_core_taxa_PHYLUM_table)[colnames(ADLT2_non_coloniz_core_taxa_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(ADLT2_non_coloniz_core_taxa_PHYLUM_table)[colnames(ADLT2_non_coloniz_core_taxa_PHYLUM_table) == "Freq"] <- "number of ASVs"
View(ADLT2_non_coloniz_core_taxa_PHYLUM_table)
write.table(ADLT2_non_coloniz_core_taxa_PHYLUM_table, "ADLT2_non_coloniz_core_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ADLT2_non_coloniz_core_taxa_FAMILY <- table(tax_table(ADLT2_non_colonizing_ps)[, "Family"], exclude = NULL)
ADLT2_non_coloniz_core_taxa_FAMILY_table <- as.data.frame(ADLT2_non_coloniz_core_taxa_FAMILY)
colnames(ADLT2_non_coloniz_core_taxa_FAMILY_table)[colnames(ADLT2_non_coloniz_core_taxa_FAMILY_table) == "Var1"] <- "Family"
colnames(ADLT2_non_coloniz_core_taxa_FAMILY_table)[colnames(ADLT2_non_coloniz_core_taxa_FAMILY_table) == "Freq"] <- "number of ASVs"
View(ADLT2_non_coloniz_core_taxa_FAMILY_table)
write.table(ADLT2_non_coloniz_core_taxa_FAMILY_table, "ADLT2_non_coloniz_core_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ADLT2_non_coloniz_core_taxa_GENUS <- table(tax_table(ADLT2_non_colonizing_ps)[, "Genus"], exclude = NULL)
ADLT2_non_coloniz_core_taxa_GENUS_table <- as.data.frame(ADLT2_non_coloniz_core_taxa_GENUS)
colnames(ADLT2_non_coloniz_core_taxa_GENUS_table)[colnames(ADLT2_non_coloniz_core_taxa_GENUS_table) == "Var1"] <- "Genus"
colnames(ADLT2_non_coloniz_core_taxa_GENUS_table)[colnames(ADLT2_non_coloniz_core_taxa_GENUS_table) == "Freq"] <- "number of ASVs"
View(ADLT2_non_coloniz_core_taxa_GENUS_table)
write.table(ADLT2_non_coloniz_core_taxa_GENUS_table, "ADLT2_non_coloniz_core_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


## ELD2 donor
core_ELD2_taxa_not_colonizing
pig_mouse_comp_ELD2_comparison_ps
ELD2_non_colonizing_ps <- prune_taxa(core_ELD2_taxa_not_colonizing, pig_mouse_comp_ELD2_comparison_ps)
ELD2_non_colonizing_ps

# at PHYLUM level
ELD2_non_coloniz_core_taxa_PHYLUM <- table(tax_table(ELD2_non_colonizing_ps)[, "Phylum"], exclude = NULL)
ELD2_non_coloniz_core_taxa_PHYLUM_table <- as.data.frame(ELD2_non_coloniz_core_taxa_PHYLUM)
colnames(ELD2_non_coloniz_core_taxa_PHYLUM_table)[colnames(ELD2_non_coloniz_core_taxa_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(ELD2_non_coloniz_core_taxa_PHYLUM_table)[colnames(ELD2_non_coloniz_core_taxa_PHYLUM_table) == "Freq"] <- "number of ASVs"
View(ELD2_non_coloniz_core_taxa_PHYLUM_table)
write.table(ELD2_non_coloniz_core_taxa_PHYLUM_table, "ELD2_non_coloniz_core_PHYLUM.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at FAMILY level
ELD2_non_coloniz_core_taxa_FAMILY <- table(tax_table(ELD2_non_colonizing_ps)[, "Family"], exclude = NULL)
ELD2_non_coloniz_core_taxa_FAMILY_table <- as.data.frame(ELD2_non_coloniz_core_taxa_FAMILY)
colnames(ELD2_non_coloniz_core_taxa_FAMILY_table)[colnames(ELD2_non_coloniz_core_taxa_FAMILY_table) == "Var1"] <- "Family"
colnames(ELD2_non_coloniz_core_taxa_FAMILY_table)[colnames(ELD2_non_coloniz_core_taxa_FAMILY_table) == "Freq"] <- "number of ASVs"
View(ELD2_non_coloniz_core_taxa_FAMILY_table)
write.table(ELD2_non_coloniz_core_taxa_FAMILY_table, "ELD2_non_coloniz_core_FAMILY.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

# at GENUS level
ELD2_non_coloniz_core_taxa_GENUS <- table(tax_table(ELD2_non_colonizing_ps)[, "Genus"], exclude = NULL)
ELD2_non_coloniz_core_taxa_GENUS_table <- as.data.frame(ELD2_non_coloniz_core_taxa_GENUS)
colnames(ELD2_non_coloniz_core_taxa_GENUS_table)[colnames(ELD2_non_coloniz_core_taxa_GENUS_table) == "Var1"] <- "Genus"
colnames(ELD2_non_coloniz_core_taxa_GENUS_table)[colnames(ELD2_non_coloniz_core_taxa_GENUS_table) == "Freq"] <- "number of ASVs"
View(ELD2_non_coloniz_core_taxa_GENUS_table)
write.table(ELD2_non_coloniz_core_taxa_GENUS_table, "ELD2_non_coloniz_core_GENUS.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


##**Calculating mean and standard deviation for core ASVs in each donor among the three different species and assigning taxonomy**## 

**INF2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_INF2_comparison_ps # cecal samples and 'early' inoculation piglet samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_INF2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_INF2_comparison_ps_core.prop <- prune_taxa(core_INF2_taxa, pig_mouse_comp_INF2_comparison_ps.prop)
pig_mouse_comp_INF2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_INF2_core_ASVs <- psmelt(pig_mouse_comp_INF2_comparison_ps_core.prop) %>% as_tibble
tb_INF2_core_ASVs_0 <- tb_INF2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_0_rounded <- tb_INF2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_0_rounded)

df_donor_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_INF2 <- subset(tb_INF2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_INF2)[colnames(df_mouse_INF2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_INF2)[colnames(df_piglet_INF2) == "OTU"] <- "ASV_ID"
View(df_donor_INF2)
View(df_mouse_INF2)
View(df_piglet_INF2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_mouse_INF2,df_piglet_INF2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = TRUE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_INF2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Phylum")

tb_INF2_core_ASVs_PHYLUM_df <- psmelt(tb_INF2_core_ASVs_PHYLUM) %>% as_tibble
tb_INF2_core_ASVs_PHYLUM_df_0 <- tb_INF2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_PHYLUM_df_0_rounded <- tb_INF2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_PHYLUM <- subset(tb_INF2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_PHYLUM)[colnames(df_mouse_INF2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_PHYLUM)[colnames(df_mouse_INF2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_PHYLUM)[colnames(df_piglet_INF2_PHYLUM) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_INF2_PHYLUM)[colnames(df_piglet_INF2_PHYLUM) == "SD"] <- "SD_piglet"

View(df_donor_INF2_PHYLUM)
df_donor_INF2_PHYLUM$sample_Species <- NULL

INF2_donor_PHYLUM_df <- merge(human_donors_INF2_PHYLUM_table,df_donor_INF2_PHYLUM, by="Phylum")
View(INF2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_INF2_mice_ps <- prune_taxa(core_INF2_taxa_colonizing_mice, INF2_HMA_mice_ps)
colonizing_core_taxa_INF2_mice_ps

# at PHYLUM level
INF2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_INF2_PHYLUM_table <- as.data.frame(INF2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_INF2_PHYLUM_table)[colnames(core_taxa_mice_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_PHYLUM_table)

df_mouse_INF2_PHYLUM$sample_Species <- NULL
INF2_mouse_PHYLUM_df <- merge(core_taxa_mice_INF2_PHYLUM_table,df_mouse_INF2_PHYLUM, by="Phylum")
View(INF2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_INF2_piglets_ps <- prune_taxa(core_INF2_taxa_colonizing_piglets, INF2_HMA_piglets_ps)
colonizing_core_taxa_INF2_piglets_ps

# at PHYLUM level
INF2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_INF2_PHYLUM_table <- as.data.frame(INF2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_INF2_PHYLUM_table)[colnames(core_taxa_piglets_INF2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_PHYLUM_table)

df_piglet_INF2_PHYLUM$sample_Species <- NULL
INF2_piglet_PHYLUM_df <- merge(core_taxa_piglets_INF2_PHYLUM_table,df_piglet_INF2_PHYLUM, by="Phylum")
View(INF2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_INF2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_PHYLUM_df,INF2_mouse_PHYLUM_df,INF2_piglet_PHYLUM_df))
View(merged_INF2_PHYLUM)

write.table(merged_INF2_PHYLUM, "merged_core_PHYLUM_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_INF2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Family")

tb_INF2_core_ASVs_FAMILY_df <- psmelt(tb_INF2_core_ASVs_FAMILY) %>% as_tibble
tb_INF2_core_ASVs_FAMILY_df_0 <- tb_INF2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_FAMILY_df_0_rounded <- tb_INF2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_FAMILY_df_0_rounded)

df_donor_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_FAMILY <- subset(tb_INF2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_FAMILY)[colnames(df_mouse_INF2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_FAMILY)[colnames(df_mouse_INF2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_FAMILY)[colnames(df_piglet_INF2_FAMILY) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_INF2_FAMILY)[colnames(df_piglet_INF2_FAMILY) == "SD"] <- "SD_piglet"


View(df_donor_INF2_FAMILY)
df_donor_INF2_FAMILY$sample_Species <- NULL

INF2_donor_FAMILY_df <- merge(human_donors_INF2_FAMILY_table,df_donor_INF2_FAMILY, by="Family")
View(INF2_donor_FAMILY_df)


## for mice

# at FAMILY level
INF2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_INF2_FAMILY_table <- as.data.frame(INF2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_INF2_FAMILY_table)[colnames(core_taxa_mice_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_FAMILY_table)

df_mouse_INF2_FAMILY$sample_Species <- NULL
INF2_mouse_FAMILY_df <- merge(core_taxa_mice_INF2_FAMILY_table,df_mouse_INF2_FAMILY, by="Family")
View(INF2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
INF2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_INF2_FAMILY_table <- as.data.frame(INF2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_INF2_FAMILY_table)[colnames(core_taxa_piglets_INF2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_FAMILY_table)

df_piglet_INF2_FAMILY$sample_Species <- NULL
INF2_piglet_FAMILY_df <- merge(core_taxa_piglets_INF2_FAMILY_table,df_piglet_INF2_FAMILY, by="Family")
View(INF2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_INF2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_FAMILY_df,INF2_mouse_FAMILY_df,INF2_piglet_FAMILY_df))
View(merged_INF2_FAMILY)

write.table(merged_INF2_FAMILY, "merged_core_FAMILY_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_INF2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_INF2_comparison_ps_core.prop, "Genus", NArm = FALSE) # NArm = FALSE so that unclassified genus level ASVs are not removed
tb_INF2_core_ASVs_GENUS_df <- psmelt(tb_INF2_core_ASVs_GENUS) %>% as_tibble
View(tb_INF2_core_ASVs_GENUS_df)
tb_INF2_core_ASVs_GENUS_df_0 <- tb_INF2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_core_ASVs_GENUS_df_0_rounded <- tb_INF2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_core_ASVs_GENUS_df_0_rounded)

df_donor_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_INF2_GENUS <- subset(tb_INF2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_INF2_GENUS)[colnames(df_mouse_INF2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_INF2_GENUS)[colnames(df_mouse_INF2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_INF2_GENUS)[colnames(df_piglet_INF2_GENUS) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_INF2_GENUS)[colnames(df_piglet_INF2_GENUS) == "SD"] <- "SD_piglet"

View(df_donor_INF2_GENUS)
df_donor_INF2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
INF2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_INF2_GENUS_table,df_donor_INF2_GENUS))
View(INF2_donor_GENUS_df)


## for mice

# at GENUS level
INF2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_INF2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_INF2_GENUS_table <- as.data.frame(INF2_core_taxa_mouse_table_GENUS)
View(INF2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_INF2_GENUS_table)[colnames(core_taxa_mice_INF2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_INF2_GENUS_table)

df_mouse_INF2_GENUS$sample_Species <- NULL
View(df_mouse_INF2_GENUS)
INF2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_INF2_GENUS_table,df_mouse_INF2_GENUS))
View(INF2_mouse_GENUS_df)

## for piglets

# at GENUS level
INF2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_INF2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_INF2_GENUS_table <- as.data.frame(INF2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_INF2_GENUS_table)[colnames(core_taxa_piglets_INF2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_INF2_GENUS_table)

df_piglet_INF2_GENUS$sample_Species <- NULL
INF2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_INF2_GENUS_table,df_piglet_INF2_GENUS))
View(INF2_piglet_GENUS_df)


## merged the data frames from all three species
merged_INF2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(INF2_donor_GENUS_df,INF2_mouse_GENUS_df,INF2_piglet_GENUS_df))
View(merged_INF2_GENUS)

write.table(merged_INF2_GENUS, "merged_core_GENUS_INF2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```

**CHLD2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_CHLD2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_CHLD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_CHLD2_comparison_ps_core.prop <- prune_taxa(core_CHLD2_taxa, pig_mouse_comp_CHLD2_comparison_ps.prop)
pig_mouse_comp_CHLD2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_CHLD2_core_ASVs <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_core.prop) %>% as_tibble
tb_CHLD2_core_ASVs_0 <- tb_CHLD2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_0_rounded <- tb_CHLD2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2 <- subset(tb_CHLD2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_CHLD2)[colnames(df_mouse_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_CHLD2)[colnames(df_piglet_CHLD2) == "OTU"] <- "ASV_ID"
View(df_donor_CHLD2)
View(df_mouse_CHLD2)
View(df_piglet_CHLD2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_mouse_CHLD2,df_piglet_CHLD2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_CHLD2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Phylum")

tb_CHLD2_core_ASVs_PHYLUM_df <- psmelt(tb_CHLD2_core_ASVs_PHYLUM) %>% as_tibble
tb_CHLD2_core_ASVs_PHYLUM_df_0 <- tb_CHLD2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded <- tb_CHLD2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_PHYLUM <- subset(tb_CHLD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_PHYLUM)[colnames(df_mouse_CHLD2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_PHYLUM)[colnames(df_mouse_CHLD2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_PHYLUM)[colnames(df_piglet_CHLD2_PHYLUM) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_CHLD2_PHYLUM)[colnames(df_piglet_CHLD2_PHYLUM) == "SD"] <- "SD_piglet"

df_donor_CHLD2_PHYLUM$sample_Species <- NULL

CHLD2_donor_PHYLUM_df <- merge(human_donors_CHLD2_PHYLUM_table,df_donor_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_CHLD2_mice_ps <- prune_taxa(core_CHLD2_taxa_colonizing_mice, CHLD2_HMA_mice_ps)
colonizing_core_taxa_CHLD2_mice_ps # 37 ASVs

# at PHYLUM level
CHLD2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_CHLD2_PHYLUM_table)[colnames(core_taxa_mice_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_PHYLUM_table)

df_mouse_CHLD2_PHYLUM$sample_Species <- NULL
CHLD2_mouse_PHYLUM_df <- merge(core_taxa_mice_CHLD2_PHYLUM_table,df_mouse_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_CHLD2_piglets_ps <- prune_taxa(core_CHLD2_taxa_colonizing_piglets, CHLD2_HMA_piglets_ps)
colonizing_core_taxa_CHLD2_piglets_ps # 65 ASVs

# at PHYLUM level
CHLD2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_CHLD2_PHYLUM_table <- as.data.frame(CHLD2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_CHLD2_PHYLUM_table)[colnames(core_taxa_piglets_CHLD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_PHYLUM_table)

df_piglet_CHLD2_PHYLUM$sample_Species <- NULL
CHLD2_piglet_PHYLUM_df <- merge(core_taxa_piglets_CHLD2_PHYLUM_table,df_piglet_CHLD2_PHYLUM, by="Phylum")
View(CHLD2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_CHLD2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_PHYLUM_df,CHLD2_mouse_PHYLUM_df,CHLD2_piglet_PHYLUM_df))
View(merged_CHLD2_PHYLUM)

write.table(merged_CHLD2_PHYLUM, "merged_core_PHYLUM_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_CHLD2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_CHLD2_core_ASVs_FAMILY_df <- psmelt(tb_CHLD2_core_ASVs_FAMILY) %>% as_tibble
tb_CHLD2_core_ASVs_FAMILY_df_0 <- tb_CHLD2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_FAMILY_df_0_rounded <- tb_CHLD2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded)

df_donor_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_FAMILY <- subset(tb_CHLD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_FAMILY)[colnames(df_mouse_CHLD2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_FAMILY)[colnames(df_mouse_CHLD2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_FAMILY)[colnames(df_piglet_CHLD2_FAMILY) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_CHLD2_FAMILY)[colnames(df_piglet_CHLD2_FAMILY) == "SD"] <- "SD_piglet"


View(df_donor_CHLD2_FAMILY)
df_donor_CHLD2_FAMILY$sample_Species <- NULL

CHLD2_donor_FAMILY_df <- merge(human_donors_CHLD2_FAMILY_table,df_donor_CHLD2_FAMILY, by="Family")
View(CHLD2_donor_FAMILY_df)


## for mice

# at FAMILY level
CHLD2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_CHLD2_FAMILY_table <- as.data.frame(CHLD2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_CHLD2_FAMILY_table)[colnames(core_taxa_mice_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_FAMILY_table)

df_mouse_CHLD2_FAMILY$sample_Species <- NULL
CHLD2_mouse_FAMILY_df <- merge(core_taxa_mice_CHLD2_FAMILY_table,df_mouse_CHLD2_FAMILY, by="Family")
View(CHLD2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
CHLD2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_CHLD2_FAMILY_table <- as.data.frame(CHLD2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_CHLD2_FAMILY_table)[colnames(core_taxa_piglets_CHLD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_FAMILY_table)

df_piglet_CHLD2_FAMILY$sample_Species <- NULL
CHLD2_piglet_FAMILY_df <- merge(core_taxa_piglets_CHLD2_FAMILY_table,df_piglet_CHLD2_FAMILY, by="Family")
View(CHLD2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_CHLD2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_FAMILY_df,CHLD2_mouse_FAMILY_df,CHLD2_piglet_FAMILY_df))
View(merged_CHLD2_FAMILY)

write.table(merged_CHLD2_FAMILY, "merged_core_FAMILY_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_CHLD2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_CHLD2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_CHLD2_core_ASVs_GENUS_df <- psmelt(tb_CHLD2_core_ASVs_GENUS) %>% as_tibble
tb_CHLD2_core_ASVs_GENUS_df_0 <- tb_CHLD2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_core_ASVs_GENUS_df_0_rounded <- tb_CHLD2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_core_ASVs_GENUS_df_0_rounded)

df_donor_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_CHLD2_GENUS <- subset(tb_CHLD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_CHLD2_GENUS)[colnames(df_mouse_CHLD2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_CHLD2_GENUS)[colnames(df_mouse_CHLD2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_CHLD2_GENUS)[colnames(df_piglet_CHLD2_GENUS) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_CHLD2_GENUS)[colnames(df_piglet_CHLD2_GENUS) == "SD"] <- "SD_piglet"

View(df_donor_CHLD2_GENUS)
df_donor_CHLD2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
CHLD2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_CHLD2_GENUS_table,df_donor_CHLD2_GENUS))
View(CHLD2_donor_GENUS_df)


## for mice

# at GENUS level
CHLD2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_CHLD2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_CHLD2_GENUS_table <- as.data.frame(CHLD2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_CHLD2_GENUS_table)[colnames(core_taxa_mice_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_CHLD2_GENUS_table)

df_mouse_CHLD2_GENUS$sample_Species <- NULL
View(df_mouse_CHLD2_GENUS)
CHLD2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_CHLD2_GENUS_table,df_mouse_CHLD2_GENUS))
View(CHLD2_mouse_GENUS_df)

## for piglets

# at GENUS level
CHLD2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_CHLD2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_CHLD2_GENUS_table <- as.data.frame(CHLD2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_CHLD2_GENUS_table)[colnames(core_taxa_piglets_CHLD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_CHLD2_GENUS_table)

df_piglet_CHLD2_GENUS$sample_Species <- NULL
CHLD2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_CHLD2_GENUS_table,df_piglet_CHLD2_GENUS))
View(CHLD2_piglet_GENUS_df)


## merged the data frames from all three species
merged_CHLD2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(CHLD2_donor_GENUS_df,CHLD2_mouse_GENUS_df,CHLD2_piglet_GENUS_df))
View(merged_CHLD2_GENUS)

write.table(merged_CHLD2_GENUS, "merged_core_GENUS_CHLD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**ADLT2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_ADLT2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_ADLT2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_ADLT2_comparison_ps_core.prop <- prune_taxa(core_ADLT2_taxa, pig_mouse_comp_ADLT2_comparison_ps.prop)
pig_mouse_comp_ADLT2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_ADLT2_core_ASVs <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_core.prop) %>% as_tibble
tb_ADLT2_core_ASVs_0 <- tb_ADLT2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_0_rounded <- tb_ADLT2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2 <- subset(tb_ADLT2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ADLT2)[colnames(df_mouse_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ADLT2)[colnames(df_piglet_ADLT2) == "OTU"] <- "ASV_ID"
View(df_donor_ADLT2)
View(df_mouse_ADLT2)
View(df_piglet_ADLT2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_mouse_ADLT2,df_piglet_ADLT2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_ADLT2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Phylum")

tb_ADLT2_core_ASVs_PHYLUM_df <- psmelt(tb_ADLT2_core_ASVs_PHYLUM) %>% as_tibble
tb_ADLT2_core_ASVs_PHYLUM_df_0 <- tb_ADLT2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded <- tb_ADLT2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_PHYLUM <- subset(tb_ADLT2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_PHYLUM)[colnames(df_mouse_ADLT2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_PHYLUM)[colnames(df_mouse_ADLT2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_PHYLUM)[colnames(df_piglet_ADLT2_PHYLUM) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ADLT2_PHYLUM)[colnames(df_piglet_ADLT2_PHYLUM) == "SD"] <- "SD_piglet"

df_donor_ADLT2_PHYLUM$sample_Species <- NULL

ADLT2_donor_PHYLUM_df <- merge(human_donors_ADLT2_PHYLUM_table,df_donor_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_ADLT2_mice_ps <- prune_taxa(core_ADLT2_taxa_colonizing_mice, ADLT2_HMA_mice_ps)
colonizing_core_taxa_ADLT2_mice_ps # 66 ASVs

# at PHYLUM level
ADLT2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ADLT2_PHYLUM_table)[colnames(core_taxa_mice_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_PHYLUM_table)

df_mouse_ADLT2_PHYLUM$sample_Species <- NULL
ADLT2_mouse_PHYLUM_df <- merge(core_taxa_mice_ADLT2_PHYLUM_table,df_mouse_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_ADLT2_piglets_ps <- prune_taxa(core_ADLT2_taxa_colonizing_piglets, ADLT2_HMA_piglets_ps)
colonizing_core_taxa_ADLT2_piglets_ps # 106 ASVs

# at PHYLUM level
ADLT2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_ADLT2_PHYLUM_table <- as.data.frame(ADLT2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ADLT2_PHYLUM_table)[colnames(core_taxa_piglets_ADLT2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_PHYLUM_table)

df_piglet_ADLT2_PHYLUM$sample_Species <- NULL
ADLT2_piglet_PHYLUM_df <- merge(core_taxa_piglets_ADLT2_PHYLUM_table,df_piglet_ADLT2_PHYLUM, by="Phylum")
View(ADLT2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_ADLT2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_PHYLUM_df,ADLT2_mouse_PHYLUM_df,ADLT2_piglet_PHYLUM_df))
View(merged_ADLT2_PHYLUM)

write.table(merged_ADLT2_PHYLUM, "merged_core_PHYLUM_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_ADLT2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_ADLT2_core_ASVs_FAMILY_df <- psmelt(tb_ADLT2_core_ASVs_FAMILY) %>% as_tibble
tb_ADLT2_core_ASVs_FAMILY_df_0 <- tb_ADLT2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_FAMILY_df_0_rounded <- tb_ADLT2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded)

df_donor_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_FAMILY <- subset(tb_ADLT2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_FAMILY)[colnames(df_mouse_ADLT2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_FAMILY)[colnames(df_mouse_ADLT2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_FAMILY)[colnames(df_piglet_ADLT2_FAMILY) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ADLT2_FAMILY)[colnames(df_piglet_ADLT2_FAMILY) == "SD"] <- "SD_piglet"

View(df_donor_ADLT2_FAMILY)
df_donor_ADLT2_FAMILY$sample_Species <- NULL

ADLT2_donor_FAMILY_df <- merge(human_donors_ADLT2_FAMILY_table,df_donor_ADLT2_FAMILY, by="Family")
View(ADLT2_donor_FAMILY_df)


## for mice

# at FAMILY level
ADLT2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_ADLT2_FAMILY_table <- as.data.frame(ADLT2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ADLT2_FAMILY_table)[colnames(core_taxa_mice_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_FAMILY_table)

df_mouse_ADLT2_FAMILY$sample_Species <- NULL
ADLT2_mouse_FAMILY_df <- merge(core_taxa_mice_ADLT2_FAMILY_table,df_mouse_ADLT2_FAMILY, by="Family")
View(ADLT2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
ADLT2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_ADLT2_FAMILY_table <- as.data.frame(ADLT2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ADLT2_FAMILY_table)[colnames(core_taxa_piglets_ADLT2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_FAMILY_table)

df_piglet_ADLT2_FAMILY$sample_Species <- NULL
ADLT2_piglet_FAMILY_df <- merge(core_taxa_piglets_ADLT2_FAMILY_table,df_piglet_ADLT2_FAMILY, by="Family")
View(ADLT2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_ADLT2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_FAMILY_df,ADLT2_mouse_FAMILY_df,ADLT2_piglet_FAMILY_df))
View(merged_ADLT2_FAMILY)

write.table(merged_ADLT2_FAMILY, "merged_core_FAMILY_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_ADLT2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_ADLT2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_ADLT2_core_ASVs_GENUS_df <- psmelt(tb_ADLT2_core_ASVs_GENUS) %>% as_tibble
tb_ADLT2_core_ASVs_GENUS_df_0 <- tb_ADLT2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_core_ASVs_GENUS_df_0_rounded <- tb_ADLT2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_core_ASVs_GENUS_df_0_rounded)

df_donor_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_ADLT2_GENUS <- subset(tb_ADLT2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ADLT2_GENUS)[colnames(df_mouse_ADLT2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ADLT2_GENUS)[colnames(df_mouse_ADLT2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_ADLT2_GENUS)[colnames(df_piglet_ADLT2_GENUS) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ADLT2_GENUS)[colnames(df_piglet_ADLT2_GENUS) == "SD"] <- "SD_piglet"

View(df_donor_ADLT2_GENUS)
df_donor_ADLT2_GENUS$sample_Species <- NULL

# merged_prev_df_ADLT2_core_mice[is.na(merged_prev_df_ADLT2_core_mice)] <- 0
ADLT2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ADLT2_GENUS_table,df_donor_ADLT2_GENUS))
View(ADLT2_donor_GENUS_df)


## for mice

# at GENUS level
ADLT2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_ADLT2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_ADLT2_GENUS_table <- as.data.frame(ADLT2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ADLT2_GENUS_table)[colnames(core_taxa_mice_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ADLT2_GENUS_table)

df_mouse_ADLT2_GENUS$sample_Species <- NULL
View(df_mouse_ADLT2_GENUS)
ADLT2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ADLT2_GENUS_table,df_mouse_ADLT2_GENUS))
View(ADLT2_mouse_GENUS_df)

## for piglets

# at GENUS level
ADLT2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_ADLT2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_ADLT2_GENUS_table <- as.data.frame(ADLT2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ADLT2_GENUS_table)[colnames(core_taxa_piglets_ADLT2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ADLT2_GENUS_table)

df_piglet_ADLT2_GENUS$sample_Species <- NULL
ADLT2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ADLT2_GENUS_table,df_piglet_ADLT2_GENUS))
View(ADLT2_piglet_GENUS_df)


## merged the data frames from all three species
merged_ADLT2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ADLT2_donor_GENUS_df,ADLT2_mouse_GENUS_df,ADLT2_piglet_GENUS_df))
View(merged_ADLT2_GENUS)

write.table(merged_ADLT2_GENUS, "merged_core_GENUS_ADLT2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**ELD2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_ELD2_comparison_ps 

# Convert raw read counts to relative abundances
pig_mouse_comp_ELD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x) x/sum(x))

# retain just the core INF2 ASVs
pig_mouse_comp_ELD2_comparison_ps_core.prop <- prune_taxa(core_ELD2_taxa, pig_mouse_comp_ELD2_comparison_ps.prop)
pig_mouse_comp_ELD2_comparison_ps_core.prop

# calculating mean and standard deviation for each core phylum across the 3 different species
tb_ELD2_core_ASVs <- psmelt(pig_mouse_comp_ELD2_comparison_ps_core.prop) %>% as_tibble
tb_ELD2_core_ASVs_0 <- tb_ELD2_core_ASVs %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_0_rounded <- tb_ELD2_core_ASVs_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "human")
df_mouse_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "mouse")
df_piglet_ELD2 <- subset(tb_ELD2_core_ASVs_0_rounded, sample_Species == "piglet")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ELD2)[colnames(df_mouse_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ELD2)[colnames(df_piglet_ELD2) == "OTU"] <- "ASV_ID"
View(df_donor_ELD2)
View(df_mouse_ELD2)
View(df_piglet_ELD2)

merged <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_mouse_ELD2,df_piglet_ELD2, by="ASV_ID"))
View(merged)
merged[[5]] <- NULL
merged[[9]] <- NULL
View(merged)
write.table(merged, "merged_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_core.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_core_tax.txt", quote = FALSE, col.names = FALSE, sep = "\t")


### at PHYLUM level
tb_ELD2_core_ASVs_PHYLUM <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Phylum")

tb_ELD2_core_ASVs_PHYLUM_df <- psmelt(tb_ELD2_core_ASVs_PHYLUM) %>% as_tibble
tb_ELD2_core_ASVs_PHYLUM_df_0 <- tb_ELD2_core_ASVs_PHYLUM_df %>% dplyr::group_by(Phylum, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_PHYLUM_df_0_rounded <- tb_ELD2_core_ASVs_PHYLUM_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded)

df_donor_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_PHYLUM <- subset(tb_ELD2_core_ASVs_PHYLUM_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_PHYLUM)[colnames(df_mouse_ELD2_PHYLUM) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_PHYLUM)[colnames(df_mouse_ELD2_PHYLUM) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_PHYLUM)[colnames(df_piglet_ELD2_PHYLUM) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ELD2_PHYLUM)[colnames(df_piglet_ELD2_PHYLUM) == "SD"] <- "SD_piglet"

df_donor_ELD2_PHYLUM$sample_Species <- NULL

ELD2_donor_PHYLUM_df <- merge(human_donors_ELD2_PHYLUM_table,df_donor_ELD2_PHYLUM, by="Phylum")
View(ELD2_donor_PHYLUM_df)

## for mice
# core_taxa colonizing mice (found in at least a single fecal sample)
colonizing_core_taxa_ELD2_mice_ps <- prune_taxa(core_ELD2_taxa_colonizing_mice, ELD2_HMA_mice_ps)
colonizing_core_taxa_ELD2_mice_ps # 55 ASVs

# at PHYLUM level
ELD2_core_taxa_mouse_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Phylum"], exclude = NULL)
core_taxa_mice_ELD2_PHYLUM_table <- as.data.frame(ELD2_core_taxa_mouse_table_PHYLUM)
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_mice_ELD2_PHYLUM_table)[colnames(core_taxa_mice_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_PHYLUM_table)

df_mouse_ELD2_PHYLUM$sample_Species <- NULL
ELD2_mouse_PHYLUM_df <- merge(core_taxa_mice_ELD2_PHYLUM_table,df_mouse_ELD2_PHYLUM, by="Phylum")
View(ELD2_mouse_PHYLUM_df)

## for piglets
# core_taxa colonizing piglets (found in at least a single fecal sample)
colonizing_core_taxa_ELD2_piglets_ps <- prune_taxa(core_ELD2_taxa_colonizing_piglets, ELD2_HMA_piglets_ps)
colonizing_core_taxa_ELD2_piglets_ps # 92 ASVs

# at PHYLUM level
ELD2_core_taxa_piglet_table_PHYLUM <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Phylum"], exclude = NULL)
core_taxa_piglets_ELD2_PHYLUM_table <- as.data.frame(ELD2_core_taxa_piglet_table_PHYLUM)
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Var1"] <- "Phylum"
colnames(core_taxa_piglets_ELD2_PHYLUM_table)[colnames(core_taxa_piglets_ELD2_PHYLUM_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_PHYLUM_table)

df_piglet_ELD2_PHYLUM$sample_Species <- NULL
ELD2_piglet_PHYLUM_df <- merge(core_taxa_piglets_ELD2_PHYLUM_table,df_piglet_ELD2_PHYLUM, by="Phylum")
View(ELD2_piglet_PHYLUM_df)

# merged the data frames from all three species
merged_ELD2_PHYLUM <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_PHYLUM_df,ELD2_mouse_PHYLUM_df,ELD2_piglet_PHYLUM_df))
View(merged_ELD2_PHYLUM)

write.table(merged_ELD2_PHYLUM, "merged_core_PHYLUM_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at FAMILY level
tb_ELD2_core_ASVs_FAMILY <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Family", NArm = FALSE)

tb_ELD2_core_ASVs_FAMILY_df <- psmelt(tb_ELD2_core_ASVs_FAMILY) %>% as_tibble
tb_ELD2_core_ASVs_FAMILY_df_0 <- tb_ELD2_core_ASVs_FAMILY_df %>% dplyr::group_by(Family, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_FAMILY_df_0_rounded <- tb_ELD2_core_ASVs_FAMILY_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_FAMILY_df_0_rounded)

df_donor_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_FAMILY <- subset(tb_ELD2_core_ASVs_FAMILY_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_FAMILY)[colnames(df_mouse_ELD2_FAMILY) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_FAMILY)[colnames(df_mouse_ELD2_FAMILY) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_FAMILY)[colnames(df_piglet_ELD2_FAMILY) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ELD2_FAMILY)[colnames(df_piglet_ELD2_FAMILY) == "SD"] <- "SD_piglet"

View(df_donor_ELD2_FAMILY)
df_donor_ELD2_FAMILY$sample_Species <- NULL

ELD2_donor_FAMILY_df <- merge(human_donors_ELD2_FAMILY_table,df_donor_ELD2_FAMILY, by="Family")
View(ELD2_donor_FAMILY_df)


## for mice

# at FAMILY level
ELD2_core_taxa_mouse_table_FAMILY <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Family"], exclude = NULL)
core_taxa_mice_ELD2_FAMILY_table <- as.data.frame(ELD2_core_taxa_mouse_table_FAMILY)
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_mice_ELD2_FAMILY_table)[colnames(core_taxa_mice_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_FAMILY_table)

df_mouse_ELD2_FAMILY$sample_Species <- NULL
ELD2_mouse_FAMILY_df <- merge(core_taxa_mice_ELD2_FAMILY_table,df_mouse_ELD2_FAMILY, by="Family")
View(ELD2_mouse_FAMILY_df)

## for piglets

# at FAMILY level
ELD2_core_taxa_piglet_table_FAMILY <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Family"], exclude = NULL)
core_taxa_piglets_ELD2_FAMILY_table <- as.data.frame(ELD2_core_taxa_piglet_table_FAMILY)
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Var1"] <- "Family"
colnames(core_taxa_piglets_ELD2_FAMILY_table)[colnames(core_taxa_piglets_ELD2_FAMILY_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_FAMILY_table)

df_piglet_ELD2_FAMILY$sample_Species <- NULL
ELD2_piglet_FAMILY_df <- merge(core_taxa_piglets_ELD2_FAMILY_table,df_piglet_ELD2_FAMILY, by="Family")
View(ELD2_piglet_FAMILY_df)

# merged the data frames from all three species
merged_ELD2_FAMILY <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_FAMILY_df,ELD2_mouse_FAMILY_df,ELD2_piglet_FAMILY_df))
View(merged_ELD2_FAMILY)

write.table(merged_ELD2_FAMILY, "merged_core_FAMILY_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)


### at GENUS level
tb_ELD2_core_ASVs_GENUS <- tax_glom(pig_mouse_comp_ELD2_comparison_ps_core.prop, "Genus", NArm = FALSE)

tb_ELD2_core_ASVs_GENUS_df <- psmelt(tb_ELD2_core_ASVs_GENUS) %>% as_tibble
tb_ELD2_core_ASVs_GENUS_df_0 <- tb_ELD2_core_ASVs_GENUS_df %>% dplyr::group_by(Genus, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_core_ASVs_GENUS_df_0_rounded <- tb_ELD2_core_ASVs_GENUS_df_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_core_ASVs_GENUS_df_0_rounded)

df_donor_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "human")
df_mouse_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "mouse")
df_piglet_ELD2_GENUS <- subset(tb_ELD2_core_ASVs_GENUS_df_0_rounded, sample_Species == "piglet")

colnames(df_mouse_ELD2_GENUS)[colnames(df_mouse_ELD2_GENUS) == "Mean"] <- "Mean_mice" 
colnames(df_mouse_ELD2_GENUS)[colnames(df_mouse_ELD2_GENUS) == "SD"] <- "SD_mice"
colnames(df_piglet_ELD2_GENUS)[colnames(df_piglet_ELD2_GENUS) == "Mean"] <- "Mean_piglet" 
colnames(df_piglet_ELD2_GENUS)[colnames(df_piglet_ELD2_GENUS) == "SD"] <- "SD_piglet"

View(df_donor_ELD2_GENUS)
df_donor_ELD2_GENUS$sample_Species <- NULL

# merged_prev_df_CHLD2_core_mice[is.na(merged_prev_df_CHLD2_core_mice)] <- 0
ELD2_donor_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(human_donors_ELD2_GENUS_table,df_donor_ELD2_GENUS))
View(ELD2_donor_GENUS_df)


## for mice

# at GENUS level
ELD2_core_taxa_mouse_table_GENUS <- table(tax_table(colonizing_core_taxa_ELD2_mice_ps)[, "Genus"], exclude = NULL)
core_taxa_mice_ELD2_GENUS_table <- as.data.frame(ELD2_core_taxa_mouse_table_GENUS)
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_mice_ELD2_GENUS_table)[colnames(core_taxa_mice_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in mice"
View(core_taxa_mice_ELD2_GENUS_table)

df_mouse_ELD2_GENUS$sample_Species <- NULL
View(df_mouse_ELD2_GENUS)
ELD2_mouse_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_mice_ELD2_GENUS_table,df_mouse_ELD2_GENUS))
View(ELD2_mouse_GENUS_df)

## for piglets

# at GENUS level
ELD2_core_taxa_piglet_table_GENUS <- table(tax_table(colonizing_core_taxa_ELD2_piglets_ps)[, "Genus"], exclude = NULL)
core_taxa_piglets_ELD2_GENUS_table <- as.data.frame(ELD2_core_taxa_piglet_table_GENUS)
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Var1"] <- "Genus"
colnames(core_taxa_piglets_ELD2_GENUS_table)[colnames(core_taxa_piglets_ELD2_GENUS_table) == "Freq"] <- "number of ASVs in piglets"
View(core_taxa_piglets_ELD2_GENUS_table)

df_piglet_ELD2_GENUS$sample_Species <- NULL
ELD2_piglet_GENUS_df <- Reduce(function(x, y) merge(x, y, all=TRUE), list(core_taxa_piglets_ELD2_GENUS_table,df_piglet_ELD2_GENUS))
View(ELD2_piglet_GENUS_df)


## merged the data frames from all three species
merged_ELD2_GENUS <- Reduce(function(x, y) merge(x, y, all=TRUE), list(ELD2_donor_GENUS_df,ELD2_mouse_GENUS_df,ELD2_piglet_GENUS_df))
View(merged_ELD2_GENUS)

write.table(merged_ELD2_GENUS, "merged_core_GENUS_ELD2.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)
perst_colonizing_core_taxa_CHLD2_mice
```


##**Comparing relative abundances at the Phylum, Family, and Genus levels (Not sure of the best way to do this)


**INF2 donor**

```{r, echo=TRUE}
pig_mouse_comp_INF2_comparison_ps # cecal samples and 'early' inoculation piglet samples taken out

## PHYLUM level

INF2_phylum_ps <- tax_glom(pig_mouse_comp_INF2_comparison_ps, "Phylum", NArm = FALSE)

# Donor vs HMA mice

persistently_colonizing_taxa_INF2_mice
```


##**Calculating mean and standard deviation for persistently colonizing core ASVs in each donor among the three different species and assigning taxonomy**## 

**INF2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_INF2_comparison_ps # cecal samples and 'early' inoculation piglet samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_INF2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_INF2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_INF2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_INF2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_INF2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_taxa_INF2_mice
pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_INF2_mice, pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop)

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_INF2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_INF2_perst_core_ASVs_mice_0 <- tb_INF2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_perst_core_ASVs_mice_0_rounded <- tb_INF2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_perst_core_ASVs_mice_0_rounded)

df_donor_INF2 <- subset(tb_INF2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_INF2 <- subset(tb_INF2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_INF2)[colnames(df_mouse_INF2) == "OTU"] <- "ASV_ID"

View(df_donor_INF2)
View(df_mouse_INF2)

merged_INF2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_mouse_INF2, by="ASV_ID"))
View(merged_INF2_perst_mice)
merged_INF2_perst_mice[[5]] <- NULL
merged_INF2_perst_mice[[9]] <- NULL
merged_INF2_perst_mice[[10]] <- NULL
merged_INF2_perst_mice[[11]] <- NULL
View(merged_INF2_perst_mice)
write.table(merged_INF2_perst_mice, "merged_INF2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglet

# retain just the mouse and donor samples
pig_mouse_comp_INF2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_INF2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_INF2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_taxa_INF2_piglets
pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_INF2_piglets, pig_mouse_comp_INF2_comparison_ps_piglets.prop)

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_INF2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_INF2_perst_core_ASVs_piglets_0 <- tb_INF2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_INF2_perst_core_ASVs_piglets_0_rounded <- tb_INF2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_INF2_perst_core_ASVs_piglets_0_rounded)

df_donor_INF2 <- subset(tb_INF2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_INF2 <- subset(tb_INF2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_INF2)[colnames(df_donor_INF2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_INF2)[colnames(df_piglet_INF2) == "OTU"] <- "ASV_ID"

View(df_donor_INF2)
View(df_piglet_INF2)

merged_INF2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_INF2,df_piglet_INF2, by="ASV_ID"))
View(merged_INF2_perst_piglets)
merged_INF2_perst_piglets[[5]] <- NULL
merged_INF2_perst_piglets[[9]] <- NULL
merged_INF2_perst_piglets[[10]] <- NULL
merged_INF2_perst_piglets[[11]] <- NULL
View(merged_INF2_perst_piglets)
write.table(merged_INF2_perst_piglets, "merged_INF2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_INF2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "INF2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**CHLD2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_CHLD2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_CHLD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_CHLD2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_CHLD2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_CHLD2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_taxa_CHLD2_mice
pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_CHLD2_mice,pig_mouse_comp_CHLD2_comparison_ps_mice.prop)
pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_CHLD2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_CHLD2_perst_core_ASVs_mice_0 <- tb_CHLD2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_perst_core_ASVs_mice_0_rounded <- tb_CHLD2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_perst_core_ASVs_mice_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_CHLD2)[colnames(df_mouse_CHLD2) == "OTU"] <- "ASV_ID"

View(df_donor_CHLD2)
View(df_mouse_CHLD2)

merged_CHLD2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_mouse_CHLD2, by="ASV_ID"))
View(merged_CHLD2_perst_mice)
merged_CHLD2_perst_mice[[5]] <- NULL
merged_CHLD2_perst_mice[[9]] <- NULL
merged_CHLD2_perst_mice[[10]] <- NULL
merged_CHLD2_perst_mice[[11]] <- NULL
View(merged_CHLD2_perst_mice)
write.table(merged_CHLD2_perst_mice, "merged_CHLD2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglet

# retain just the mouse and donor samples
pig_mouse_comp_CHLD2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_CHLD2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_taxa_CHLD2_piglets
pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_CHLD2_piglets, pig_mouse_comp_CHLD2_comparison_ps_piglets.prop)
pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_CHLD2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_CHLD2_perst_core_ASVs_piglets_0 <- tb_CHLD2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_CHLD2_perst_core_ASVs_piglets_0_rounded <- tb_CHLD2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_CHLD2_perst_core_ASVs_piglets_0_rounded)

df_donor_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_CHLD2 <- subset(tb_CHLD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_CHLD2)[colnames(df_donor_CHLD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_CHLD2)[colnames(df_piglet_CHLD2) == "OTU"] <- "ASV_ID"

View(df_donor_CHLD2)
View(df_piglet_CHLD2)

merged_CHLD2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_CHLD2,df_piglet_CHLD2, by="ASV_ID"))
View(merged_CHLD2_perst_piglets)
merged_CHLD2_perst_piglets[[5]] <- NULL
merged_CHLD2_perst_piglets[[9]] <- NULL
merged_CHLD2_perst_piglets[[10]] <- NULL
merged_CHLD2_perst_piglets[[11]] <- NULL
View(merged_CHLD2_perst_piglets)
write.table(merged_CHLD2_perst_piglets, "merged_CHLD2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_CHLD2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "CHLD2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**ADLT2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_ADLT2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_ADLT2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ADLT2_comparison_ps, function(x) x/sum(x))

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_ADLT2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_ADLT2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_taxa_ADLT2_mice
pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_ADLT2_mice,pig_mouse_comp_ADLT2_comparison_ps_mice.prop)
pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ADLT2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_ADLT2_perst_core_ASVs_mice_0 <- tb_ADLT2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_perst_core_ASVs_mice_0_rounded <- tb_ADLT2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_perst_core_ASVs_mice_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ADLT2)[colnames(df_mouse_ADLT2) == "OTU"] <- "ASV_ID"

View(df_donor_ADLT2)
View(df_mouse_ADLT2)

merged_ADLT2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_mouse_ADLT2, by="ASV_ID"))
View(merged_ADLT2_perst_mice)
merged_ADLT2_perst_mice[[5]] <- NULL
merged_ADLT2_perst_mice[[9]] <- NULL
merged_ADLT2_perst_mice[[10]] <- NULL
merged_ADLT2_perst_mice[[11]] <- NULL
View(merged_ADLT2_perst_mice)
write.table(merged_ADLT2_perst_mice, "merged_ADLT2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglet

# retain just the mouse and donor samples
pig_mouse_comp_ADLT2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_ADLT2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_taxa_ADLT2_piglets
pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_ADLT2_piglets, pig_mouse_comp_ADLT2_comparison_ps_piglets.prop)
pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ADLT2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_ADLT2_perst_core_ASVs_piglets_0 <- tb_ADLT2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ADLT2_perst_core_ASVs_piglets_0_rounded <- tb_ADLT2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ADLT2_perst_core_ASVs_piglets_0_rounded)

df_donor_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_ADLT2 <- subset(tb_ADLT2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_ADLT2)[colnames(df_donor_ADLT2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ADLT2)[colnames(df_piglet_ADLT2) == "OTU"] <- "ASV_ID"

View(df_donor_ADLT2)
View(df_piglet_ADLT2)

merged_ADLT2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ADLT2,df_piglet_ADLT2, by="ASV_ID"))
View(merged_ADLT2_perst_piglets)
merged_ADLT2_perst_piglets[[5]] <- NULL
merged_ADLT2_perst_piglets[[9]] <- NULL
merged_ADLT2_perst_piglets[[10]] <- NULL
merged_ADLT2_perst_piglets[[11]] <- NULL
View(merged_ADLT2_perst_piglets)
write.table(merged_ADLT2_perst_piglets, "merged_ADLT2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ADLT2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ADLT2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")
```

**ELD2 Donor**

```{r, echo=TRUE}

pig_mouse_comp_ELD2_comparison_ps # cecal samples taken out

# Convert raw read counts to relative abundances
pig_mouse_comp_ELD2_comparison_ps.prop <- transform_sample_counts(pig_mouse_comp_ELD2_comparison_ps, function(x) x/sum(x))
pig_mouse_comp_ELD2_comparison_ps.prop

## for HMA mice

# retain just the mouse and donor samples
pig_mouse_comp_ELD2_comparison_ps_mice.prop <- subset_samples(pig_mouse_comp_ELD2_comparison_ps.prop, Species != "piglet")
pig_mouse_comp_ELD2_comparison_ps_mice.prop

# retain only the core persistent colonizers of mice
perst_colonizing_core_taxa_ELD2_mice
pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_ELD2_mice,pig_mouse_comp_ELD2_comparison_ps_mice.prop)
pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ELD2_perst_core_ASVs_mice <- psmelt(pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop) %>% as_tibble
tb_ELD2_perst_core_ASVs_mice_0 <- tb_ELD2_perst_core_ASVs_mice %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_perst_core_ASVs_mice_0_rounded <- tb_ELD2_perst_core_ASVs_mice_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_perst_core_ASVs_mice_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_perst_core_ASVs_mice_0_rounded, sample_Species == "human")
df_mouse_ELD2 <- subset(tb_ELD2_perst_core_ASVs_mice_0_rounded, sample_Species == "mouse")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_mouse_ELD2)[colnames(df_mouse_ELD2) == "OTU"] <- "ASV_ID"

View(df_donor_ELD2)
View(df_mouse_ELD2)

merged_ELD2_perst_mice <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_mouse_ELD2, by="ASV_ID"))
View(merged_ELD2_perst_mice)
merged_ELD2_perst_mice[[5]] <- NULL
merged_ELD2_perst_mice[[9]] <- NULL
merged_ELD2_perst_mice[[10]] <- NULL
merged_ELD2_perst_mice[[11]] <- NULL
View(merged_ELD2_perst_mice)
write.table(merged_ELD2_perst_mice, "merged_ELD2_perst_mice_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_mice_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_perst_core_tax_mice.txt", quote = FALSE, col.names = FALSE, sep = "\t")


## for HMA piglet

# retain just the mouse and donor samples
pig_mouse_comp_ELD2_comparison_ps_piglets.prop <- subset_samples(pig_mouse_comp_ELD2_comparison_ps.prop, Species != "mouse")
pig_mouse_comp_ELD2_comparison_ps_piglets.prop

# retain only the core persistent colonizers of piglets
perst_colonizing_core_taxa_ELD2_piglets
pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop <- prune_taxa(perst_colonizing_core_taxa_ELD2_piglets, pig_mouse_comp_ELD2_comparison_ps_piglets.prop)
pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop

# calculating mean and standard deviation for each perst. core ASV across the 2 different species
tb_ELD2_perst_core_ASVs_piglets <- psmelt(pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop) %>% as_tibble
tb_ELD2_perst_core_ASVs_piglets_0 <- tb_ELD2_perst_core_ASVs_piglets %>% dplyr::group_by(OTU, sample_Species) %>% dplyr::summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup
tb_ELD2_perst_core_ASVs_piglets_0_rounded <- tb_ELD2_perst_core_ASVs_piglets_0 %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(tb_ELD2_perst_core_ASVs_piglets_0_rounded)

df_donor_ELD2 <- subset(tb_ELD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "human")
df_piglet_ELD2 <- subset(tb_ELD2_perst_core_ASVs_piglets_0_rounded, sample_Species == "piglet")

colnames(df_donor_ELD2)[colnames(df_donor_ELD2) == "OTU"] <- "ASV_ID"
colnames(df_piglet_ELD2)[colnames(df_piglet_ELD2) == "OTU"] <- "ASV_ID"

View(df_donor_ELD2)
View(df_piglet_ELD2)

merged_ELD2_perst_piglets <- Reduce(function(x,y) cbind(x,y, all=TRUE), list(df_donor_ELD2,df_piglet_ELD2, by="ASV_ID"))
View(merged_ELD2_perst_piglets)
merged_ELD2_perst_piglets[[5]] <- NULL
merged_ELD2_perst_piglets[[9]] <- NULL
merged_ELD2_perst_piglets[[10]] <- NULL
merged_ELD2_perst_piglets[[11]] <- NULL
View(merged_ELD2_perst_piglets)
write.table(merged_ELD2_perst_piglets, "merged_ELD2_perst_piglets_table.txt", sep = "\t", col.names = TRUE, row.names = FALSE, quote = FALSE)

tax <- as(tax_table(pig_mouse_comp_ELD2_comparison_ps_piglets_core_perst.prop), "matrix")
tax_cols <- colnames(tax)
tax <- as.data.frame(tax)
tax$taxonomy <- do.call(paste, c(tax[tax_cols], sep = ";"))
for (co in tax_cols) tax[co] <- NULL
write.table(tax, "ELD2_perst_core_tax_piglets.txt", quote = FALSE, col.names = FALSE, sep = "\t")

perst_rel_abund_core_INF2_taxa_mice
perst_rel_abund_core_INF2_taxa_piglets

perst_rel_abund_core_CHLD2_taxa_mice
perst_rel_abund_core_CHLD2_taxa_piglets

# INF
length(perst_rel_abund_core_INF2_taxa_piglets)/length(core_INF2_taxa_colonizing_piglets) * 100

# CHLD
length(perst_rel_abund_core_CHLD2_taxa_piglets)/length(core_CHLD2_taxa_colonizing_piglets) * 100

# ADLT
length(perst_rel_abund_core_ADLT2_taxa_piglets)/length(core_ADLT2_taxa_colonizing_piglets) * 100

# ELD
length(perst_rel_abund_core_ELD2_taxa_piglets)/length(core_ELD2_taxa_colonizing_piglets) * 100

```

## Generating data for Circos plots ##


Getting relative abundance values for making Circos plots :


**INF2 Donor**

```{r, echo=TRUE}
pig_mouse_comp_INF2_comparison_ps

## Calculating abundance values for donor inocula
INF2_donor_inocula_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "human")
INF2_donor_inocula_ps

# Calculate relative abundances
INF2_donor_inocula_ps.prop <- transform_sample_counts(INF2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_INF2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
INF2_donor_mice_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "mouse")
INF2_donor_mice_ps

# Calculate relative abundances
INF2_donor_mice_ps.prop <- transform_sample_counts(INF2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_INF2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
INF2_donor_piglets_ps <- subset_samples(pig_mouse_comp_INF2_comparison_ps,Species == "piglet")
INF2_donor_piglets_ps

# Calculate relative abundances
INF2_donor_piglets_ps.prop <- transform_sample_counts(INF2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(INF2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_INF2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_INF2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**CHLD2 Donor** 

```{r, echo=TRUE}
pig_mouse_comp_CHLD2_comparison_ps

## Calculating abundance values for donor inocula
CHLD2_donor_inocula_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "human")
CHLD2_donor_inocula_ps

# Calculate relative abundances
CHLD2_donor_inocula_ps.prop <- transform_sample_counts(CHLD2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
CHLD2_donor_mice_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "mouse")
CHLD2_donor_mice_ps

# Calculate relative abundances
CHLD2_donor_mice_ps.prop <- transform_sample_counts(CHLD2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
CHLD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_CHLD2_comparison_ps,Species == "piglet")
CHLD2_donor_piglets_ps

# Calculate relative abundances
CHLD2_donor_piglets_ps.prop <- transform_sample_counts(CHLD2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(CHLD2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_CHLD2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_CHLD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**ADLT2 Donor**

```{r, echo=TRUE}
pig_mouse_comp_ADLT2_comparison_ps

## Calculating abundance values for donor inocula
ADLT2_donor_inocula_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "human")
ADLT2_donor_inocula_ps

# Calculate relative abundances
ADLT2_donor_inocula_ps.prop <- transform_sample_counts(ADLT2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
ADLT2_donor_mice_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "mouse")
ADLT2_donor_mice_ps

# Calculate relative abundances
ADLT2_donor_mice_ps.prop <- transform_sample_counts(ADLT2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
ADLT2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ADLT2_comparison_ps,Species == "piglet")
ADLT2_donor_piglets_ps

# Calculate relative abundances
ADLT2_donor_piglets_ps.prop <- transform_sample_counts(ADLT2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ADLT2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_ADLT2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core, "Piglets_ADLT2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


**ELD2 Donor**

```{r, echo=TRUE}
pig_mouse_comp_ELD2_comparison_ps

## Calculating abundance values for donor inocula
ELD2_donor_inocula_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "human")
ELD2_donor_inocula_ps

# Calculate relative abundances
ELD2_donor_inocula_ps.prop <- transform_sample_counts(ELD2_donor_inocula_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_inocula_ps.prop) %>% as_tibble
ps_mean_stats_inocula <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_inocula_core <- ps_mean_stats_inocula[ps_mean_stats_inocula$OTU %in% core_ELD2_taxa,]
ps_mean_stats_inocula_core_rounded <- ps_mean_stats_inocula_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_inocula_core_rounded)
write.table(ps_mean_stats_inocula_core_rounded, "Inocula_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the mouse fecal samples
ELD2_donor_mice_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "mouse")
ELD2_donor_mice_ps

# Calculate relative abundances
ELD2_donor_mice_ps.prop <- transform_sample_counts(ELD2_donor_mice_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_mice_ps.prop) %>% as_tibble
ps_mean_stats_mice <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_mice_core <- ps_mean_stats_mice[ps_mean_stats_mice$OTU %in% core_ELD2_taxa,]
ps_mean_stats_mice_core_rounded <- ps_mean_stats_mice_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_mice_core_rounded)
write.table(ps_mean_stats_mice_core_rounded, "Mice_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)

## Calculating abundance values for the piglet fecal samples
ELD2_donor_piglets_ps <- subset_samples(pig_mouse_comp_ELD2_comparison_ps,Species == "piglet")
ELD2_donor_piglets_ps

# Calculate relative abundances
ELD2_donor_piglets_ps.prop <- transform_sample_counts(ELD2_donor_piglets_ps, function(otu) otu/sum(otu))
ps_mean_stats <- psmelt(ELD2_donor_piglets_ps.prop) %>% as_tibble
ps_mean_stats_piglets <- ps_mean_stats %>% group_by(OTU, Donor) %>% summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup()
ps_mean_stats_piglets_core <- ps_mean_stats_piglets[ps_mean_stats_piglets$OTU %in% core_ELD2_taxa,]
ps_mean_stats_piglets_core_rounded <- ps_mean_stats_piglets_core %>% mutate_at(vars("Mean", "SD"), list(~ round(., 4)))
View(ps_mean_stats_piglets_core_rounded)
write.table(ps_mean_stats_piglets_core_rounded, "Piglets_ELD2_core_abundances.txt", sep = '\t', col.names = TRUE, row.names = FALSE, quote = FALSE)
```


Drawing circos plots with the relative abundance data generated above :

**Infant donor**

```{r, echo=TRUE}

## Based on information in https://jokergoo.github.io/circlize_book/book/

# Install 'circlize' package
install.packages("circlize"); packageVersion('circlize')

# Load circlize package and initiate
library(circlize)
circos.par("track.height" = 0.1)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/INF2_core/INF2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)
View(input_df)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/INF2_core/INF2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame
View(cols_df)

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/INF2_core/INF2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)
View(grid_cols_df)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/INF2_core/INF2_df1.txt", sep = "\t", header = T)
INF2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/INF2_core/INF2_df2.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
INF2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/INF2_core/INF2_df3.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
INF2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(INF2_df1$Samp)
M_ASVs <- as.vector(INF2_df2$Samp)
P_ASVs <- as.vector(INF2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.3)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "core ASVs of INF Donor", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "core INF ASVs colonizing  HMA mice ", cex = 0.55, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "core INF ASVs colonizing HMA piglets", cex = 0.35, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**CHLD donor**

```{r, echo=TRUE}
# Load circlize package and initiate
library(circlize)
circos.par("track.height" = 0.1)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/CHLD2_core/CHLD2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/CHLD2_core/CHLD2_df1.txt", sep = "\t", header = T)
CHLD2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/CHLD2_core/CHLD2_df2.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
CHLD2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/CHLD2_core/CHLD2_df3.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
CHLD2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(CHLD2_df1$Samp)
M_ASVs <- as.vector(CHLD2_df2$Samp)
P_ASVs <- as.vector(CHLD2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.3)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "core ASVs of CHLD Donor", cex = 0.7, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "core CHLD ASVs colonizing  HMA mice ", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "core CHLD ASVs colonizing HMA piglets", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**ADLT donor**

```{r, echo=TRUE}
# Load circlize package and initiate
library(circlize)
circos.par("track.height" = 0.1, gap.degree = 0.8)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/ADLT2_core/ADLT2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)
View(grid_cols)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]
View(link_cols)

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/ADLT2_core/ADLT2_df1.txt", sep = "\t", header = T)
ADLT2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/ADLT2_core/ADLT2_df2.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
ADLT2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/ADLT2_core/ADLT2_df3.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
ADLT2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(ADLT2_df1$Samp)
M_ASVs <- as.vector(ADLT2_df2$Samp)
P_ASVs <- as.vector(ADLT2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.3)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "core ASVs of ADLT Donor", cex = 0.7, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "core ADLT ASVs colonizing  HMA mice ", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "core ADLT ASVs colonizing HMA piglets", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


**ELD donor**

```{r, echo=TRUE}
# Load circlize package and initiate
library(circlize)
circos.par("track.height" = 0.1, gap.degree = 0.8)

# Read in table with relative abundances for donor and HMA animals
input <- read.table("Circos_plots/ELD2_core/ELD2_circos_df.txt", sep = "\t", header = T)
input_df <- as.data.frame(input) # convert table to a data frame (ASVs not colonizing either animal model have been removed from this table)

# Read in table designating colors for the chord diagram links
colors <- read.table("Circos_plots/ELD2_core/ELD2_circos_link_colors.txt", sep = "\t", header = T)
cols_df <- as.data.frame(colors) # convert table to a data frame

# Read in tables specifying colors for grids (sectors)
grid_cols <- read.table("Circos_plots/ELD2_core/ELD2_circos_grid_colors.txt", sep = "\t", header = T)
grid_cols_df <- as.data.frame(grid_cols)

# Convert grid colors to a vector
grid_cols <- as.vector(grid_cols_df$Col)
View(grid_cols)

# Convert link colors to a vector
link_cols <- as.vector(cols_df$Col) # another syntax: cols <- cols_df[['Col']]
View(link_cols)

# Need to import following 3 tables for specifying samples for outer sector

# Donor samples
df1 <- read.table("Circos_plots/ELD2_core/ELD2_df1.txt", sep = "\t", header = T)
ELD2_df1 <- as.data.frame(df1) 

# Mouse samples
df2 <- read.table("Circos_plots/ELD2_core/ELD2_df2.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
ELD2_df2 <- as.data.frame(df2)

# Piglet samples
df3 <- read.table("Circos_plots/ELD2_core/ELD2_df3.txt", sep = "\t", header = T) # any ASV which gas a zero relative abundance needs to be removed
ELD2_df3 <- as.data.frame(df3)

# Specify sample names as vectors 
H_ASVs <- as.vector(ELD2_df1$Samp)
M_ASVs <- as.vector(ELD2_df2$Samp)
P_ASVs <- as.vector(ELD2_df3$Samp)

chordDiagram(input_df, grid.col = grid_cols, col = link_cols, annotationTrack = c("grid"), annotationTrackHeight = uh(6, "mm"), transparency = 0.3, big.gap = 5, preAllocateTracks = list(track.height = uh(4, "mm"),track.margin = c(uh(1, "mm"), 0)))
for(si in get.all.sector.index()) {
  xlim = get.cell.meta.data("xlim", sector.index = si, track.index = 2)
  ylim = get.cell.meta.data("ylim", sector.index = si, track.index = 2)
  circos.text(mean(xlim), mean(ylim), si, sector.index = si, track.index = 2, facing = "clockwise", niceFacing = TRUE, col = "white", cex = 0.3)
}

# Setting the outer sectors 
highlight.sector(H_ASVs, track.index = 1, col = "red", text = "core ASVs of SNR Donor", cex = 0.7, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(M_ASVs, track.index = 1, col = "dark green", text = "core SNR ASVs colonizing  HMA mice ", cex = 0.55, text.col = "white", niceFacing = TRUE, facing = "bending.inside")
highlight.sector(P_ASVs, track.index = 1, col = "dark blue", text = "core SNR ASVs colonizing HMA piglets", cex = 0.6, text.col = "white", niceFacing = TRUE, facing = "bending.inside")

circos.clear() # Need to clear before starting to make a new chord diagram using different sector options
```


